{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "270caf2b",
   "metadata": {},
   "source": [
    "# Import"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "ca18b51d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import json\n",
    "import sys\n",
    "\n",
    "sys.path.append(\"../../\")\n",
    "from curation_tools.curation_tools import CuratedDataset\n",
    "from curation_tools.perturbseq_anndata_schema import ObsSchema, VarSchema\n",
    "\n",
    "from curation_tools.unified_metadata_schema.unified_metadata_schema import Experiment\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d0e5973",
   "metadata": {},
   "source": [
    "# Initialise the dataset object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "659fd697",
   "metadata": {},
   "outputs": [],
   "source": [
    "cur_data = CuratedDataset(\n",
    "    obs_schema=ObsSchema,\n",
    "    var_schema=VarSchema,\n",
    "    exp_metadata_schema=Experiment,\n",
    "    data_source_link = \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406677_10X005.h5ad\",\n",
    "    noncurated_path = \"../non_curated/h5ad/adamson_2016_upr_epistasis.h5ad\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "848c76f2",
   "metadata": {},
   "source": [
    "# Download the dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c7a28f0c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File ../non_curated/h5ad/adamson_2016_upr_epistasis.h5ad already exists. Skipping download.\n",
      "Loading data from ../non_curated/h5ad/adamson_2016_upr_epistasis.h5ad\n",
      "Observation data:\n",
      "DataFrame shape: (15006, 15)\n",
      "--------------------------------------------------\n",
      "                         perturbation  read count  UMI count tissue_type  \\\n",
      "cell_barcode                                                               \n",
      "AAACATACACTCAG   3x_neg_ctrl_pMJ144-1       261.0       59.0   cell_line   \n",
      "AAACATACTCCTAT   3x_neg_ctrl_pMJ144-2       132.0       37.0   cell_line   \n",
      "AAACATTGCAGAGG   3x_neg_ctrl_pMJ144-2       560.0      117.0   cell_line   \n",
      "AAACATTGGCGAAG  ATF6_PERK_IRE1_pMJ158       215.0       49.0   cell_line   \n",
      "AAACCGTGATACCG       ATF6_PERK_pMJ150       567.0      124.0   cell_line   \n",
      "...                               ...         ...        ...         ...   \n",
      "TTTGACTGGCTTAG       PERK_only_pMJ146       215.0       60.0   cell_line   \n",
      "TTTGACTGGGGATG       PERK_IRE1_pMJ154        64.0       16.0   cell_line   \n",
      "TTTGACTGTGGTCA   3x_neg_ctrl_pMJ144-1       218.0       43.0   cell_line   \n",
      "TTTGCATGCGGAGA       PERK_IRE1_pMJ154       284.0       45.0   cell_line   \n",
      "TTTGCATGTATCTC   3x_neg_ctrl_pMJ144-1       126.0       27.0   cell_line   \n",
      "\n",
      "               cell_line  cancer                       disease  \\\n",
      "cell_barcode                                                     \n",
      "AAACATACACTCAG      K562    True  chronic myelogenous leukemia   \n",
      "AAACATACTCCTAT      K562    True  chronic myelogenous leukemia   \n",
      "AAACATTGCAGAGG      K562    True  chronic myelogenous leukemia   \n",
      "AAACATTGGCGAAG      K562    True  chronic myelogenous leukemia   \n",
      "AAACCGTGATACCG      K562    True  chronic myelogenous leukemia   \n",
      "...                  ...     ...                           ...   \n",
      "TTTGACTGGCTTAG      K562    True  chronic myelogenous leukemia   \n",
      "TTTGACTGGGGATG      K562    True  chronic myelogenous leukemia   \n",
      "TTTGACTGTGGTCA      K562    True  chronic myelogenous leukemia   \n",
      "TTTGCATGCGGAGA      K562    True  chronic myelogenous leukemia   \n",
      "TTTGCATGTATCTC      K562    True  chronic myelogenous leukemia   \n",
      "\n",
      "               perturbation_type      celltype organism  ncounts  ngenes  \\\n",
      "cell_barcode                                                               \n",
      "AAACATACACTCAG            CRISPR  lymphoblasts    human  24343.0    4164   \n",
      "AAACATACTCCTAT            CRISPR  lymphoblasts    human  27678.0    4428   \n",
      "AAACATTGCAGAGG            CRISPR  lymphoblasts    human  24745.0    4371   \n",
      "AAACATTGGCGAAG            CRISPR  lymphoblasts    human  30836.0    4551   \n",
      "AAACCGTGATACCG            CRISPR  lymphoblasts    human  33314.0    4539   \n",
      "...                          ...           ...      ...      ...     ...   \n",
      "TTTGACTGGCTTAG            CRISPR  lymphoblasts    human  34472.0    4645   \n",
      "TTTGACTGGGGATG            CRISPR  lymphoblasts    human  27671.0    4655   \n",
      "TTTGACTGTGGTCA            CRISPR  lymphoblasts    human  26795.0    4238   \n",
      "TTTGCATGCGGAGA            CRISPR  lymphoblasts    human  28974.0    4382   \n",
      "TTTGCATGTATCTC            CRISPR  lymphoblasts    human  30050.0    4489   \n",
      "\n",
      "                percent_mito  percent_ribo  nperts  \n",
      "cell_barcode                                        \n",
      "AAACATACACTCAG      4.563941     32.629505       4  \n",
      "AAACATACTCCTAT      4.508996     28.658140       4  \n",
      "AAACATTGCAGAGG      3.200647     31.117397       4  \n",
      "AAACATTGGCGAAG      3.132702     29.890388       4  \n",
      "AAACCGTGATACCG      6.588821     35.984871       3  \n",
      "...                      ...           ...     ...  \n",
      "TTTGACTGGCTTAG      3.437572     31.933163       3  \n",
      "TTTGACTGGGGATG      4.033103     27.219833       3  \n",
      "TTTGACTGTGGTCA      4.545624     31.991043       4  \n",
      "TTTGCATGCGGAGA      2.626493     30.271969       3  \n",
      "TTTGCATGTATCTC      3.221298     32.013309       4  \n",
      "\n",
      "[15006 rows x 15 columns]\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.download_data()\n",
    "cur_data.load_data(path=\"../non_curated/h5ad/adamson_2016_upr_epistasis.h5ad\")\n",
    "# show the data\n",
    "cur_data.show_obs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fe64c691",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variable data:\n",
      "DataFrame shape: (32738, 3)\n",
      "--------------------------------------------------\n",
      "                   ensembl_id  ncounts  ncells\n",
      "gene_symbol                                   \n",
      "MIR1302-10    ENSG00000243485      2.0       2\n",
      "FAM138A       ENSG00000237613      0.0       0\n",
      "OR4F5         ENSG00000186092      0.0       0\n",
      "RP11-34P13.7  ENSG00000238009      1.0       1\n",
      "RP11-34P13.8  ENSG00000239945      2.0       2\n",
      "...                       ...      ...     ...\n",
      "AC145205.1    ENSG00000215635      0.0       0\n",
      "BAGE5         ENSG00000268590      0.0       0\n",
      "CU459201.1    ENSG00000251180      0.0       0\n",
      "AC002321.2    ENSG00000215616      0.0       0\n",
      "AC002321.1    ENSG00000215611      0.0       0\n",
      "\n",
      "[32738 rows x 3 columns]\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.show_var()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "383e2f64",
   "metadata": {},
   "source": [
    "# OBS slot curation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c08e4be",
   "metadata": {},
   "source": [
    "### Show unique perturbations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8c35039f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique values in adata.obs.perturbation: 21\n",
      "--------------------------------------------------\n",
      "{nan,\n",
      " '*',\n",
      " '3x_neg_ctrl_pMJ144-1',\n",
      " '3x_neg_ctrl_pMJ144-2',\n",
      " 'ATF4_pBA576',\n",
      " 'ATF6_IRE1_pMJ152',\n",
      " 'ATF6_PERK_IRE1_pMJ158',\n",
      " 'ATF6_PERK_pMJ150',\n",
      " 'ATF6_only_pMJ145',\n",
      " 'C7orf26_pDS004',\n",
      " 'Gal4-4(mod)_pBA582',\n",
      " 'IER3IP1_pDS003',\n",
      " 'IRE1_only_pMJ148',\n",
      " 'PERK_IRE1_pMJ154',\n",
      " 'PERK_only_pMJ146',\n",
      " 'PSMA1_pDS007',\n",
      " 'PSMD12_pDS009',\n",
      " 'SNAI1_pDS266',\n",
      " 'XBP1_pBA578',\n",
      " 'XBP1_pBA579',\n",
      " 'YIPF5_pDS001'}\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.show_unique(slot = 'obs', column = 'perturbation')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa8a54c1",
   "metadata": {},
   "source": [
    "### Drop NAs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3e2eda1b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removed 296 NA entries from column perturbation of adata.obs\n"
     ]
    }
   ],
   "source": [
    "cur_data.remove_na(slot = 'obs', column = 'perturbation')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c6a9f654",
   "metadata": {},
   "source": [
    "### Drop \"*\" entries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "65ce4911",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removed 13 entries \\* from column perturbation of adata.obs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<>:1: SyntaxWarning: invalid escape sequence '\\*'\n",
      "<>:1: SyntaxWarning: invalid escape sequence '\\*'\n",
      "/var/folders/f_/7vds5wkn7vn5h991mk4_ybt80000gp/T/ipykernel_15284/814522886.py:1: SyntaxWarning: invalid escape sequence '\\*'\n",
      "  cur_data.remove_entries(slot = 'obs', column = 'perturbation', to_remove = '\\*')\n"
     ]
    }
   ],
   "source": [
    "cur_data.remove_entries(slot = 'obs', column = 'perturbation', to_remove = '\\*')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "dbb9c89d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique values in adata.obs.perturbation: 19\n",
      "--------------------------------------------------\n",
      "{'3x_neg_ctrl_pMJ144-1',\n",
      " '3x_neg_ctrl_pMJ144-2',\n",
      " 'ATF4_pBA576',\n",
      " 'ATF6_IRE1_pMJ152',\n",
      " 'ATF6_PERK_IRE1_pMJ158',\n",
      " 'ATF6_PERK_pMJ150',\n",
      " 'ATF6_only_pMJ145',\n",
      " 'C7orf26_pDS004',\n",
      " 'Gal4-4(mod)_pBA582',\n",
      " 'IER3IP1_pDS003',\n",
      " 'IRE1_only_pMJ148',\n",
      " 'PERK_IRE1_pMJ154',\n",
      " 'PERK_only_pMJ146',\n",
      " 'PSMA1_pDS007',\n",
      " 'PSMD12_pDS009',\n",
      " 'SNAI1_pDS266',\n",
      " 'XBP1_pBA578',\n",
      " 'XBP1_pBA579',\n",
      " 'YIPF5_pDS001'}\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.show_unique(slot = 'obs', column = 'perturbation')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d2b5bd29",
   "metadata": {},
   "source": [
    "### Rename `perturbation` to `perturbation_name`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b57adb98",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Renamed columns in adata.obs: {'perturbation': 'perturbation_name'}\n"
     ]
    }
   ],
   "source": [
    "cur_data.rename_columns(slot = 'obs', name_dict = {'perturbation': 'perturbation_name'})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9903d8b",
   "metadata": {},
   "source": [
    "### Add guide RNA information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ec94c92f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Column guide_sequence added to adata.obs\n"
     ]
    }
   ],
   "source": [
    "# Not reported by study authors for this dataset\n",
    "cur_data.create_columns(\n",
    "    slot='obs',\n",
    "    col_dict={\n",
    "        'guide_sequence': None\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "89d4b801",
   "metadata": {},
   "source": [
    "### Extract perturbation symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "830d42d2",
   "metadata": {},
   "source": [
    "#### Add `perturbation_target_symbol` column based on the `perturbation_name`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b4b6462f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Column perturbed_target_symbol added to adata.obs\n",
      "Unique values in adata.obs.perturbed_target_symbol: 19\n",
      "--------------------------------------------------\n",
      "{'3x_neg_ctrl_pMJ144-1',\n",
      " '3x_neg_ctrl_pMJ144-2',\n",
      " 'ATF4_pBA576',\n",
      " 'ATF6_IRE1_pMJ152',\n",
      " 'ATF6_PERK_IRE1_pMJ158',\n",
      " 'ATF6_PERK_pMJ150',\n",
      " 'ATF6_only_pMJ145',\n",
      " 'C7orf26_pDS004',\n",
      " 'Gal4-4(mod)_pBA582',\n",
      " 'IER3IP1_pDS003',\n",
      " 'IRE1_only_pMJ148',\n",
      " 'PERK_IRE1_pMJ154',\n",
      " 'PERK_only_pMJ146',\n",
      " 'PSMA1_pDS007',\n",
      " 'PSMD12_pDS009',\n",
      " 'SNAI1_pDS266',\n",
      " 'XBP1_pBA578',\n",
      " 'XBP1_pBA579',\n",
      " 'YIPF5_pDS001'}\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.create_columns(\n",
    "    slot=\"obs\", col_dict={\"perturbed_target_symbol\": cur_data.adata.obs[\"perturbation_name\"]},\n",
    "    overwrite=True\n",
    ")\n",
    "cur_data.show_unique(slot = 'obs', column = 'perturbed_target_symbol')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8c490dea",
   "metadata": {},
   "source": [
    "#### Clean up `perturbation_target_symbol` column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4fcbf8f2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Replaced '3x_neg_ctrl.*' with 'control_nontargeting' in column perturbed_target_symbol of adata.obs\n",
      "Replaced 'Gal4-4.*' with 'control_nontargeting' in column perturbed_target_symbol of adata.obs\n",
      "Replaced '_(pM|pD|pB|only).*' with '' in column perturbed_target_symbol of adata.obs\n",
      "Unique values in adata.obs.perturbed_target_symbol: 16\n",
      "--------------------------------------------------\n",
      "{'ATF4',\n",
      " 'ATF6',\n",
      " 'ATF6_IRE1',\n",
      " 'ATF6_PERK',\n",
      " 'ATF6_PERK_IRE1',\n",
      " 'C7ORF26',\n",
      " 'CONTROL_NONTARGETING',\n",
      " 'IER3IP1',\n",
      " 'IRE1',\n",
      " 'PERK',\n",
      " 'PERK_IRE1',\n",
      " 'PSMA1',\n",
      " 'PSMD12',\n",
      " 'SNAI1',\n",
      " 'XBP1',\n",
      " 'YIPF5'}\n",
      "--------------------------------------------------\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/zakirov/Documents/GitHub/PerturbationCatalogue/data_exploration/Perturbseq/curation_notebooks/../../curation_tools/curation_tools.py:569: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  if df[column].str.upper().str.contains(old_val.upper()).any():\n"
     ]
    }
   ],
   "source": [
    "cur_data.replace_entries(\n",
    "    slot=\"obs\",\n",
    "    column=\"perturbed_target_symbol\",\n",
    "    map_dict={\n",
    "        r\"3x_neg_ctrl.*\": \"control_nontargeting\",\n",
    "        r\"Gal4-4.*\": \"control_nontargeting\",\n",
    "        r\"_(pM|pD|pB|only).*\": \"\"\n",
    "    }\n",
    ")\n",
    "\n",
    "cur_data.show_unique(slot = 'obs', column = 'perturbed_target_symbol')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02b02e25",
   "metadata": {},
   "source": [
    "### Standardise perturbation targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9e0c0790",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/zakirov/Documents/GitHub/PerturbationCatalogue/data_exploration/Perturbseq/curation_notebooks/../../curation_tools/curation_tools.py:1533: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  map_df[\"synonyms\"] = map_df[\"synonyms\"].str.split(\"|\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mapped potential synonyms in perturbed_target_symbol of the provided dataframe to gene symbols\n",
      "Converted 11/13 gene symbols/ENSG IDs to standardized gene symbols/ENSG IDs\n",
      "--------------------------------------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collapsed column index using separator |\n"
     ]
    }
   ],
   "source": [
    "cur_data.standardize_genes(\n",
    "    slot='obs',\n",
    "    input_column='perturbed_target_symbol',\n",
    "    input_column_type='gene_symbol',\n",
    "    multiple_entries=True,\n",
    "    multiple_entries_sep='_'\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "98719978",
   "metadata": {},
   "source": [
    "### Add `perturbed_target_number` column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "168c267a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Counted entries in column perturbed_target_symbol of adata.obs and stored in perturbed_target_number\n"
     ]
    }
   ],
   "source": [
    "cur_data.count_entries(\n",
    "    slot='obs',\n",
    "    input_column='perturbed_target_symbol',\n",
    "    count_column_name='perturbed_target_number',\n",
    "    sep='|'\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a74b8907",
   "metadata": {},
   "source": [
    "### Encode chromosomes as integers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "184dc961",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chromosome encoding applied to perturbed_target_chromosome in adata.obs and stored as 'perturbed_target_chromosome_encoding'.\n"
     ]
    }
   ],
   "source": [
    "cur_data.chromosome_encoding()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c345b3a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Observation data:\n",
      "DataFrame shape: (14697, 2)\n",
      "--------------------------------------------------\n",
      "                    perturbation_name  perturbed_target_chromosome_encoding\n",
      "index                                                                      \n",
      "AAACATACACTCAG   3x_neg_ctrl_pMJ144-1                                     0\n",
      "AAACATACTCCTAT   3x_neg_ctrl_pMJ144-2                                     0\n",
      "AAACATTGCAGAGG   3x_neg_ctrl_pMJ144-2                                     0\n",
      "AAACATTGGCGAAG  ATF6_PERK_IRE1_pMJ158                                     0\n",
      "AAACCGTGATACCG       ATF6_PERK_pMJ150                                     0\n",
      "...                               ...                                   ...\n",
      "TTTGACTGGCTTAG       PERK_only_pMJ146                                     2\n",
      "TTTGACTGGGGATG       PERK_IRE1_pMJ154                                     0\n",
      "TTTGACTGTGGTCA   3x_neg_ctrl_pMJ144-1                                     0\n",
      "TTTGCATGCGGAGA       PERK_IRE1_pMJ154                                     0\n",
      "TTTGCATGTATCTC   3x_neg_ctrl_pMJ144-1                                     0\n",
      "\n",
      "[14697 rows x 2 columns]\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.show_obs(['perturbation_name', 'perturbed_target_chromosome_encoding'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa009f43",
   "metadata": {},
   "source": [
    "### Add metadata"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e85e0827",
   "metadata": {},
   "source": [
    "### Add treatment information"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1af1a26",
   "metadata": {},
   "source": [
    "Data is stored in a separate file, as part of the `cell BC` column's suffix\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "907589fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "cell BC",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "guide identity",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "c51988b2-1ed2-4b24-9a02-20b4855961e8",
       "rows": [
        [
         "0",
         "ACGGTATGCTTAGG-3",
         "PERK_IRE1_pMJ154"
        ],
        [
         "1",
         "ACAATCCTACCCTC-1",
         "PERK_IRE1_pMJ154"
        ],
        [
         "2",
         "ACGAACACGTGCTA-3",
         "ATF6_PERK_IRE1_pMJ158"
        ],
        [
         "3",
         "CTGTGAGATTGGTG-1",
         "ATF6_PERK_IRE1_pMJ158"
        ],
        [
         "4",
         "ATGTTGCTAATCGC-2",
         "3x_neg_ctrl_pMJ144-2"
        ]
       ],
       "shape": {
        "columns": 2,
        "rows": 5
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cell BC</th>\n",
       "      <th>guide identity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACGGTATGCTTAGG-3</td>\n",
       "      <td>PERK_IRE1_pMJ154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACAATCCTACCCTC-1</td>\n",
       "      <td>PERK_IRE1_pMJ154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACGAACACGTGCTA-3</td>\n",
       "      <td>ATF6_PERK_IRE1_pMJ158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CTGTGAGATTGGTG-1</td>\n",
       "      <td>ATF6_PERK_IRE1_pMJ158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATGTTGCTAATCGC-2</td>\n",
       "      <td>3x_neg_ctrl_pMJ144-2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            cell BC         guide identity\n",
       "0  ACGGTATGCTTAGG-3       PERK_IRE1_pMJ154\n",
       "1  ACAATCCTACCCTC-1       PERK_IRE1_pMJ154\n",
       "2  ACGAACACGTGCTA-3  ATF6_PERK_IRE1_pMJ158\n",
       "3  CTGTGAGATTGGTG-1  ATF6_PERK_IRE1_pMJ158\n",
       "4  ATGTTGCTAATCGC-2   3x_neg_ctrl_pMJ144-2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "cell_barcode",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "treatment_temp",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "a075fbe3-e333-4831-b2a3-843b8e4b0ad0",
       "rows": [
        [
         "ACGGTATGCTTAGG",
         "DMSO"
        ],
        [
         "ACAATCCTACCCTC",
         "tunicamycin"
        ],
        [
         "ACGAACACGTGCTA",
         "DMSO"
        ],
        [
         "CTGTGAGATTGGTG",
         "tunicamycin"
        ],
        [
         "ATGTTGCTAATCGC",
         "thapsigargin"
        ],
        [
         "GTCAACGATCGCTC",
         "thapsigargin"
        ],
        [
         "AAGCGACTTACGCA",
         "thapsigargin"
        ],
        [
         "AGCCTCTGGGAACG",
         "tunicamycin"
        ],
        [
         "GACGTATGACACTG",
         "tunicamycin"
        ],
        [
         "GGGCCAACGTTACG",
         "DMSO"
        ],
        [
         "CTCGAGCTATGCCA",
         "DMSO"
        ],
        [
         "TACCATTGCGCCTT",
         "thapsigargin"
        ],
        [
         "GGAACTACGTACCA",
         "DMSO"
        ],
        [
         "GACAGTACTTCTAC",
         "DMSO"
        ],
        [
         "CAATATGACTACTT",
         "thapsigargin"
        ],
        [
         "AAGTCCGATAAGGA",
         "tunicamycin"
        ],
        [
         "AACCGATGTCCTAT",
         "DMSO"
        ],
        [
         "AGATCGTGGTACGT",
         "thapsigargin"
        ],
        [
         "TTGTACACAGGCGA",
         "tunicamycin"
        ],
        [
         "CATCGCTGAAGCAA",
         "thapsigargin"
        ],
        [
         "GCCCAACTTGTGGT",
         "thapsigargin"
        ],
        [
         "TGCCGACTCCGTTC",
         "thapsigargin"
        ],
        [
         "GTAAGCTGCGACAT",
         "DMSO"
        ],
        [
         "TCCTACCTTGAGAA",
         "DMSO"
        ],
        [
         "CTGTAACTAAGGGC",
         "tunicamycin"
        ],
        [
         "TTTATCCTCCATGA",
         "tunicamycin"
        ],
        [
         "CTTTACGAGGGTGA",
         "tunicamycin"
        ],
        [
         "GTAGCCCTTTCCGC",
         "thapsigargin"
        ],
        [
         "CACTGCACCGGTAT",
         "DMSO"
        ],
        [
         "ACTATCACACACAC",
         "tunicamycin"
        ],
        [
         "AGTATAACCTGTAG",
         "DMSO"
        ],
        [
         "CGTACAGAGTACCA",
         "thapsigargin"
        ],
        [
         "AACTTGCTGGAGGT",
         "thapsigargin"
        ],
        [
         "TAGTAAACCCAATG",
         "DMSO"
        ],
        [
         "TCCATCCTAAGAAC",
         "thapsigargin"
        ],
        [
         "ACCATTACAAAGCA",
         "thapsigargin"
        ],
        [
         "GGGACCACCGCATA",
         "thapsigargin"
        ],
        [
         "AGTTCTACGACTAC",
         "thapsigargin"
        ],
        [
         "TTAGGTCTGGTCTA",
         "tunicamycin"
        ],
        [
         "CACGACCTAATCGC",
         "DMSO"
        ],
        [
         "TGGATCGAATGCCA",
         "thapsigargin"
        ],
        [
         "AAAGACGATCTTCA",
         "DMSO"
        ],
        [
         "TGCATGGAAACCGT",
         "thapsigargin"
        ],
        [
         "CTTTAGACTCGACA",
         "thapsigargin"
        ],
        [
         "AACACTCTAGCATC",
         "DMSO"
        ],
        [
         "ACAGTGTGTATGGC",
         "thapsigargin"
        ],
        [
         "AGCATGACGTCACA",
         "thapsigargin"
        ],
        [
         "ACTTGTACGGTCAT",
         "DMSO"
        ],
        [
         "CAGTTGGAGTCGTA",
         "thapsigargin"
        ],
        [
         "ATGTTAGATACAGC",
         "DMSO"
        ]
       ],
       "shape": {
        "columns": 1,
        "rows": 14754
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>treatment_temp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_barcode</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACGGTATGCTTAGG</th>\n",
       "      <td>DMSO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAATCCTACCCTC</th>\n",
       "      <td>tunicamycin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACGAACACGTGCTA</th>\n",
       "      <td>DMSO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTGTGAGATTGGTG</th>\n",
       "      <td>tunicamycin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATGTTGCTAATCGC</th>\n",
       "      <td>thapsigargin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGTAACGAGTTGCA</th>\n",
       "      <td>DMSO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCATGCTGGCTTCC</th>\n",
       "      <td>DMSO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCCTCAGAAAAGTG</th>\n",
       "      <td>DMSO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCAAGTCTAGGTCT</th>\n",
       "      <td>thapsigargin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TATAGCCTCGGGAA</th>\n",
       "      <td>DMSO</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14754 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               treatment_temp\n",
       "cell_barcode                 \n",
       "ACGGTATGCTTAGG           DMSO\n",
       "ACAATCCTACCCTC    tunicamycin\n",
       "ACGAACACGTGCTA           DMSO\n",
       "CTGTGAGATTGGTG    tunicamycin\n",
       "ATGTTGCTAATCGC   thapsigargin\n",
       "...                       ...\n",
       "CGTAACGAGTTGCA           DMSO\n",
       "CCATGCTGGCTTCC           DMSO\n",
       "CCCTCAGAAAAGTG           DMSO\n",
       "TCAAGTCTAGGTCT   thapsigargin\n",
       "TATAGCCTCGGGAA           DMSO\n",
       "\n",
       "[14754 rows x 1 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_cell_ident_link = r\"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2406nnn/GSM2406677/suppl/GSM2406677%5F10X005%5Fcell%5Fidentities.csv.gz\"\n",
    "orig_cell_ident = pd.read_csv(orig_cell_ident_link)\n",
    "\n",
    "# the number after the dash is the cell barcode indicates the treatment\n",
    "display(orig_cell_ident[[\"cell BC\", \"guide identity\"]].head())\n",
    "\n",
    "treatment_label_map = {\"1\": \"tunicamycin\", \"2\": \"thapsigargin\", \"3\": \"DMSO\"}\n",
    "\n",
    "treatment_df = (\n",
    "    orig_cell_ident[\"cell BC\"]\n",
    "    .str.split(\"-\", expand=True)\n",
    "    .rename(columns={0: \"cell_barcode\", 1: \"treatment_number\"})\n",
    ")\n",
    "\n",
    "treatment_df[\"treatment_temp\"] = treatment_df[\"treatment_number\"].map(\n",
    "    treatment_label_map\n",
    ")\n",
    "\n",
    "treatment_df = (\n",
    "    treatment_df.drop_duplicates(subset=[\"cell_barcode\"])\n",
    "    .drop(columns=[\"treatment_number\"])\n",
    "    .set_index(\"cell_barcode\")\n",
    ")\n",
    "\n",
    "treatment_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7067e6b1",
   "metadata": {},
   "source": [
    "Add treatment information to the dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f0d26809",
   "metadata": {},
   "outputs": [],
   "source": [
    "cur_data.adata.obs = cur_data.adata.obs.merge(\n",
    "    treatment_df,\n",
    "    left_index=True,\n",
    "    right_index=True,\n",
    "    how=\"left\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "07e0a9c9",
   "metadata": {},
   "source": [
    "Map treatment compounds to CHEBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "617db07b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found standardized name for compound 'tunicamycin': tunicamycin (ChEBI ID: CHEBI:29699)\n",
      "Found standardized name for compound 'thapsigargin': thapsigargin (ChEBI ID: CHEBI:9516)\n",
      "Found standardized name for compound 'DMSO': dimethyl sulfoxide (ChEBI ID: CHEBI:28262)\n",
      "Standardized compound names in column 'treatment_temp' and added 'treatment_label' and 'treatment_id' columns.\n"
     ]
    }
   ],
   "source": [
    "cur_data.standardize_compounds(column='treatment_temp')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "09b1485f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Column dataset_id added to adata.obs\n",
      "Column sample_id added to adata.obs\n",
      "Column perturbation_type_label added to adata.obs\n",
      "Column perturbation_type_id added to adata.obs\n",
      "Column model_system_label added to adata.obs\n",
      "Column model_system_id added to adata.obs\n",
      "Column tissue added to adata.obs\n",
      "Column timepoint added to adata.obs\n",
      "Column species added to adata.obs\n",
      "Column sex_label added to adata.obs\n",
      "Column sex_id added to adata.obs\n",
      "Column developmental_stage_label added to adata.obs\n",
      "Column developmental_stage_id added to adata.obs\n",
      "Column study_title added to adata.obs\n",
      "Column study_uri added to adata.obs\n",
      "Column study_year added to adata.obs\n",
      "Column first_author added to adata.obs\n",
      "Column last_author added to adata.obs\n",
      "Column experiment_title added to adata.obs\n",
      "Column experiment_summary added to adata.obs\n",
      "Column number_of_perturbed_targets added to adata.obs\n",
      "Column number_of_perturbed_samples added to adata.obs\n",
      "Column library_generation_type_id added to adata.obs\n",
      "Column library_generation_type_label added to adata.obs\n",
      "Column library_generation_method_id added to adata.obs\n",
      "Column library_generation_method_label added to adata.obs\n",
      "Column enzyme_delivery_method_id added to adata.obs\n",
      "Column enzyme_delivery_method_label added to adata.obs\n",
      "Column library_delivery_method_id added to adata.obs\n",
      "Column library_delivery_method_label added to adata.obs\n",
      "Column enzyme_integration_state_id added to adata.obs\n",
      "Column enzyme_integration_state_label added to adata.obs\n",
      "Column library_integration_state_id added to adata.obs\n",
      "Column library_integration_state_label added to adata.obs\n",
      "Column enzyme_expression_control_id added to adata.obs\n",
      "Column enzyme_expression_control_label added to adata.obs\n",
      "Column library_expression_control_id added to adata.obs\n",
      "Column library_expression_control_label added to adata.obs\n",
      "Column library_name added to adata.obs\n",
      "Column library_uri added to adata.obs\n",
      "Column library_format_id added to adata.obs\n",
      "Column library_format_label added to adata.obs\n",
      "Column library_scope_id added to adata.obs\n",
      "Column library_scope_label added to adata.obs\n",
      "Column library_perturbation_type_id added to adata.obs\n",
      "Column library_perturbation_type_label added to adata.obs\n",
      "Column library_manufacturer added to adata.obs\n",
      "Column library_lentiviral_generation added to adata.obs\n",
      "Column library_grnas_per_target added to adata.obs\n",
      "Column library_total_grnas added to adata.obs\n",
      "Column library_total_variants added to adata.obs\n",
      "Column readout_dimensionality_id added to adata.obs\n",
      "Column readout_dimensionality_label added to adata.obs\n",
      "Column readout_type_id added to adata.obs\n",
      "Column readout_type_label added to adata.obs\n",
      "Column readout_technology_id added to adata.obs\n",
      "Column readout_technology_label added to adata.obs\n",
      "Column method_name_id added to adata.obs\n",
      "Column method_name_label added to adata.obs\n",
      "Column method_uri added to adata.obs\n",
      "Column sequencing_library_kit_id added to adata.obs\n",
      "Column sequencing_library_kit_label added to adata.obs\n",
      "Column sequencing_platform_id added to adata.obs\n",
      "Column sequencing_platform_label added to adata.obs\n",
      "Column sequencing_strategy_id added to adata.obs\n",
      "Column sequencing_strategy_label added to adata.obs\n",
      "Column software_counts_id added to adata.obs\n",
      "Column software_counts_label added to adata.obs\n",
      "Column software_analysis_id added to adata.obs\n",
      "Column software_analysis_label added to adata.obs\n",
      "Column reference_genome_id added to adata.obs\n",
      "Column reference_genome_label added to adata.obs\n",
      "Column associated_datasets added to adata.obs\n"
     ]
    }
   ],
   "source": [
    "cur_data.create_columns(\n",
    "    slot=\"obs\",\n",
    "    col_dict={\n",
    "        \"dataset_id\": cur_data.dataset_id,\n",
    "        \"sample_id\": range(1, cur_data.adata.obs.shape[0] + 1),\n",
    "        # treatment\n",
    "        # \"treatment_label\": None, \n",
    "        # \"treatment_id\": None,\n",
    "        # perturbation type\n",
    "        \"perturbation_type_label\": \"CRISPRi\", \n",
    "        \"perturbation_type_id\": None,\n",
    "        # model system\n",
    "        \"model_system_label\": \"cell line\", \n",
    "        \"model_system_id\": None,\n",
    "        'tissue': 'blood',\n",
    "        \"timepoint\": \"P0DT0H0M0S\",\n",
    "        \n",
    "        \"species\": \"Homo sapiens\",\n",
    "        \"sex_label\": \"female\", \n",
    "        \"sex_id\": None,\n",
    "        \"developmental_stage_label\": \"adult\", \n",
    "        \"developmental_stage_id\": None,\n",
    "\n",
    "        \"study_title\": \"A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response\",\n",
    "        \"study_uri\": \"https://doi.org/10.1016/j.cell.2016.11.048\",\n",
    "        \"study_year\": 2016,\n",
    "        \"first_author\": \"Britt Adamson\",\n",
    "        \"last_author\": \"Jonathan Weissman\",\n",
    "        \n",
    "        \"experiment_title\": \"14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).\",\n",
    "        \"experiment_summary\": \"Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).\",\n",
    "        \"number_of_perturbed_targets\": len(set(cur_data.adata.obs['perturbed_target_coord'])),\n",
    "        \"number_of_perturbed_samples\": cur_data.adata.obs.shape[0],\n",
    "        \n",
    "        \"library_generation_type_id\": \"EFO:0022868\",\n",
    "        \"library_generation_type_label\": \"endogenous\",\n",
    "\n",
    "        \"library_generation_method_id\": \"EFO:0022895\",\n",
    "        \"library_generation_method_label\": \"dCas9-KRAB\",\n",
    "\n",
    "        \"enzyme_delivery_method_id\": None,\n",
    "        \"enzyme_delivery_method_label\": \"retroviral transduction\",\n",
    "\n",
    "        \"library_delivery_method_id\": None,\n",
    "        \"library_delivery_method_label\": \"lentiviral transduction\",\n",
    "\n",
    "        \"enzyme_integration_state_id\": None,\n",
    "        \"enzyme_integration_state_label\": \"random locus integration\",\n",
    "\n",
    "        \"library_integration_state_id\": None,\n",
    "        \"library_integration_state_label\": \"random locus integration\",\n",
    "\n",
    "        \"enzyme_expression_control_id\": None,\n",
    "        \"enzyme_expression_control_label\": \"constitutive expression\",\n",
    "\n",
    "        \"library_expression_control_id\": None,\n",
    "        \"library_expression_control_label\": \"constitutive expression\",\n",
    "\n",
    "        \"library_name\": \"custom\",\n",
    "        \"library_uri\": None,\n",
    "\n",
    "        \"library_format_id\": None,\n",
    "        \"library_format_label\": \"pooled\",\n",
    "\n",
    "        \"library_scope_id\": None,\n",
    "        \"library_scope_label\": \"focused\",\n",
    "\n",
    "        \"library_perturbation_type_id\": None,\n",
    "        \"library_perturbation_type_label\": \"inhibition\",\n",
    "        \n",
    "        \"library_manufacturer\": \"Weissman\",\n",
    "        \"library_lentiviral_generation\": \"3\",\n",
    "        \"library_grnas_per_target\": \"1\",\n",
    "        \"library_total_grnas\": \"16\",\n",
    "        \"library_total_variants\": None,\n",
    "        \n",
    "        \"readout_dimensionality_id\": None,\n",
    "        \"readout_dimensionality_label\": \"high-dimensional assay\",\n",
    "\n",
    "        \"readout_type_id\": None,\n",
    "        \"readout_type_label\": \"transcriptomic\",\n",
    "\n",
    "        \"readout_technology_id\": None,\n",
    "        \"readout_technology_label\": \"single-cell rna-seq\",\n",
    "\n",
    "        \"method_name_id\": None,\n",
    "        \"method_name_label\": \"Perturb-seq\",\n",
    "\n",
    "        \"method_uri\": None,\n",
    "\n",
    "        \"sequencing_library_kit_id\": None,\n",
    "        \"sequencing_library_kit_label\": \"10x Genomics Single Cell 3-prime\",\n",
    "\n",
    "        \"sequencing_platform_id\": None,\n",
    "        \"sequencing_platform_label\": \"Illumina HiSeq 4000\",\n",
    "\n",
    "        \"sequencing_strategy_id\": None,\n",
    "        \"sequencing_strategy_label\": \"barcode sequencing\",\n",
    "\n",
    "        \"software_counts_id\": None,\n",
    "        \"software_counts_label\": \"CellRanger\",\n",
    "\n",
    "        \"software_analysis_id\": None,\n",
    "        \"software_analysis_label\": \"MAGeCK\",\n",
    "\n",
    "        \"reference_genome_id\": None,\n",
    "        \"reference_genome_label\": \"GRCh37\",\n",
    "\n",
    "        \"associated_datasets\": json.dumps([\n",
    "            {\n",
    "                \"dataset_accession\": \"GSM2406675\",\n",
    "                \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\",\n",
    "                \"dataset_description\": \"Raw counts\",\n",
    "                \"dataset_file_name\": \"GSE90546_RAW.tar\",\n",
    "            },\n",
    "            {\n",
    "                \"dataset_accession\": \"GSM2406675_10X001\",\n",
    "                \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\",\n",
    "                \"dataset_description\": \"Processed .h5ad file\",\n",
    "                \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\",\n",
    "            }\n",
    "        ])\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5fd97b2e",
   "metadata": {},
   "source": [
    "### Curate tissue information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "f713f6f9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mapped 1 tissue ontology terms from `tissue` column to ontology terms\n",
      "DataFrame shape: (1, 4)\n",
      "--------------------------------------------------\n",
      "  input_column input_column_lower name_lower     ontology_id\n",
      "0        blood              blood      blood  UBERON:0000178\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "\n",
    "cur_data.standardize_ontology(\n",
    "    input_column='tissue',\n",
    "    column_type='term_name',\n",
    "    ontology_type='tissue',\n",
    "    overwrite=True\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "10c287aa",
   "metadata": {},
   "source": [
    "### Curate cell type information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ceb2464d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mapped 1 cell_type ontology terms from `celltype` column to ontology terms\n",
      "DataFrame shape: (1, 4)\n",
      "--------------------------------------------------\n",
      "   input_column input_column_lower    name_lower ontology_id\n",
      "0  lymphoblasts       lymphoblasts  lymphoblasts  CL:0017005\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.standardize_ontology(\n",
    "    input_column='celltype',\n",
    "    column_type='term_name',\n",
    "    ontology_type='cell_type'\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65633fff",
   "metadata": {},
   "source": [
    "### Curate cell line information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "a6227529",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mapped 1 cell_line ontology terms from `cell_line` column to ontology terms\n",
      "DataFrame shape: (1, 4)\n",
      "--------------------------------------------------\n",
      "  input_column input_column_lower name_lower  ontology_id\n",
      "0         K562               k562       k562  CLO:0007050\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.standardize_ontology(\n",
    "    input_column='cell_line',\n",
    "    column_type='term_name',\n",
    "    ontology_type='cell_line'\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "87f279a1",
   "metadata": {},
   "source": [
    "### Curate disease information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "1f7fd146",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mapped 1 disease ontology terms from `disease` column to ontology terms\n",
      "DataFrame shape: (1, 4)\n",
      "--------------------------------------------------\n",
      "                   input_column            input_column_lower  \\\n",
      "0  chronic myelogenous leukemia  chronic myelogenous leukemia   \n",
      "\n",
      "                     name_lower    ontology_id  \n",
      "0  chronic myelogenous leukemia  MONDO:0011996  \n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.standardize_ontology(\n",
    "    input_column='disease',\n",
    "    column_type='term_name',\n",
    "    ontology_type='disease'\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "daa472e9",
   "metadata": {},
   "source": [
    "### Match schema column order"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "96599719",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Matched columns of adata.obs to the obs_schema.\n"
     ]
    }
   ],
   "source": [
    "cur_data.match_schema_columns(slot='obs')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ba1623c",
   "metadata": {},
   "source": [
    "### Validate obs metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "952a97cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "adata.obs is valid according to the obs_schema.\n",
      "Validated data:\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "dataset_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "sample_id",
         "rawType": "Int64",
         "type": "integer"
        },
        {
         "name": "perturbation_name",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "perturbed_target_coord",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "perturbed_target_chromosome",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "perturbed_target_chromosome_encoding",
         "rawType": "Int64",
         "type": "integer"
        },
        {
         "name": "perturbed_target_number",
         "rawType": "Int64",
         "type": "integer"
        },
        {
         "name": "perturbed_target_ensg",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "perturbed_target_symbol",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "perturbed_target_biotype",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "guide_sequence",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "perturbation_type_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "perturbation_type_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "timepoint",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "treatment_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "treatment_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "model_system_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "model_system_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "species",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "tissue_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "tissue_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "cell_type_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "cell_type_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "cell_line_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "cell_line_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "sex_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "sex_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "developmental_stage_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "developmental_stage_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "disease_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "disease_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "study_title",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "study_uri",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "study_year",
         "rawType": "Int64",
         "type": "integer"
        },
        {
         "name": "first_author",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "last_author",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "experiment_title",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "experiment_summary",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "number_of_perturbed_targets",
         "rawType": "Int64",
         "type": "integer"
        },
        {
         "name": "number_of_perturbed_samples",
         "rawType": "Int64",
         "type": "integer"
        },
        {
         "name": "library_generation_type_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_generation_type_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_generation_method_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_generation_method_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "enzyme_delivery_method_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "enzyme_delivery_method_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_delivery_method_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_delivery_method_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "enzyme_integration_state_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "enzyme_integration_state_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_integration_state_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_integration_state_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "enzyme_expression_control_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "enzyme_expression_control_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_expression_control_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_expression_control_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_name",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_uri",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_format_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_format_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_scope_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_scope_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_perturbation_type_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_perturbation_type_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_manufacturer",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_lentiviral_generation",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_grnas_per_target",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "library_total_grnas",
         "rawType": "Int64",
         "type": "integer"
        },
        {
         "name": "library_total_variants",
         "rawType": "Int64",
         "type": "integer"
        },
        {
         "name": "readout_dimensionality_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "readout_dimensionality_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "readout_type_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "readout_type_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "readout_technology_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "readout_technology_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "method_name_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "method_name_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "method_uri",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "sequencing_library_kit_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "sequencing_library_kit_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "sequencing_platform_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "sequencing_platform_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "sequencing_strategy_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "sequencing_strategy_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "software_counts_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "software_counts_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "software_analysis_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "software_analysis_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "reference_genome_id",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "reference_genome_label",
         "rawType": "string",
         "type": "string"
        },
        {
         "name": "associated_datasets",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "f706be5c-1e70-4c45-8b1a-12266f562c10",
       "rows": [
        [
         "0",
         "adamson_2016_upr_epistasis",
         "1",
         "3x_neg_ctrl_pMJ144-1",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "1",
         "adamson_2016_upr_epistasis",
         "2",
         "3x_neg_ctrl_pMJ144-2",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "2",
         "adamson_2016_upr_epistasis",
         "3",
         "3x_neg_ctrl_pMJ144-2",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "3",
         "adamson_2016_upr_epistasis",
         "4",
         "ATF6_PERK_IRE1_pMJ158",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "1|2|17",
         "0",
         "3",
         "ENSG00000118217|ENSG00000172071|ENSG00000178607",
         "ATF6|EIF2AK3|ERN1",
         "protein_coding|protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "4",
         "adamson_2016_upr_epistasis",
         "5",
         "ATF6_PERK_pMJ150",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1",
         "1|2",
         "0",
         "2",
         "ENSG00000118217|ENSG00000172071",
         "ATF6|EIF2AK3",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "5",
         "adamson_2016_upr_epistasis",
         "6",
         "ATF6_PERK_pMJ150",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1",
         "1|2",
         "0",
         "2",
         "ENSG00000118217|ENSG00000172071",
         "ATF6|EIF2AK3",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "6",
         "adamson_2016_upr_epistasis",
         "7",
         "ATF6_only_pMJ145",
         "chr1:161766298-161977574;1",
         "1",
         "1",
         "1",
         "ENSG00000118217",
         "ATF6",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "7",
         "adamson_2016_upr_epistasis",
         "8",
         "3x_neg_ctrl_pMJ144-1",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "8",
         "adamson_2016_upr_epistasis",
         "9",
         "PERK_IRE1_pMJ154",
         "chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "2|17",
         "0",
         "2",
         "ENSG00000172071|ENSG00000178607",
         "EIF2AK3|ERN1",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "9",
         "adamson_2016_upr_epistasis",
         "10",
         "ATF6_IRE1_pMJ152",
         "chr1:161766298-161977574;1|chr17:64039080-64130819;-1",
         "1|17",
         "0",
         "2",
         "ENSG00000118217|ENSG00000178607",
         "ATF6|ERN1",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "10",
         "adamson_2016_upr_epistasis",
         "11",
         "3x_neg_ctrl_pMJ144-1",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "11",
         "adamson_2016_upr_epistasis",
         "12",
         "3x_neg_ctrl_pMJ144-1",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "12",
         "adamson_2016_upr_epistasis",
         "13",
         "ATF6_IRE1_pMJ152",
         "chr1:161766298-161977574;1|chr17:64039080-64130819;-1",
         "1|17",
         "0",
         "2",
         "ENSG00000118217|ENSG00000178607",
         "ATF6|ERN1",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "13",
         "adamson_2016_upr_epistasis",
         "14",
         "3x_neg_ctrl_pMJ144-1",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "14",
         "adamson_2016_upr_epistasis",
         "15",
         "3x_neg_ctrl_pMJ144-2",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "15",
         "adamson_2016_upr_epistasis",
         "16",
         "IRE1_only_pMJ148",
         "chr17:64039080-64130819;-1",
         "17",
         "17",
         "1",
         "ENSG00000178607",
         "ERN1",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "16",
         "adamson_2016_upr_epistasis",
         "17",
         "ATF6_PERK_IRE1_pMJ158",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "1|2|17",
         "0",
         "3",
         "ENSG00000118217|ENSG00000172071|ENSG00000178607",
         "ATF6|EIF2AK3|ERN1",
         "protein_coding|protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "17",
         "adamson_2016_upr_epistasis",
         "18",
         "ATF6_PERK_pMJ150",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1",
         "1|2",
         "0",
         "2",
         "ENSG00000118217|ENSG00000172071",
         "ATF6|EIF2AK3",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "18",
         "adamson_2016_upr_epistasis",
         "19",
         "ATF6_IRE1_pMJ152",
         "chr1:161766298-161977574;1|chr17:64039080-64130819;-1",
         "1|17",
         "0",
         "2",
         "ENSG00000118217|ENSG00000178607",
         "ATF6|ERN1",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "19",
         "adamson_2016_upr_epistasis",
         "20",
         "3x_neg_ctrl_pMJ144-1",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "20",
         "adamson_2016_upr_epistasis",
         "21",
         "PERK_only_pMJ146",
         "chr2:88556741-88691518;-1",
         "2",
         "2",
         "1",
         "ENSG00000172071",
         "EIF2AK3",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "21",
         "adamson_2016_upr_epistasis",
         "22",
         "ATF6_PERK_pMJ150",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1",
         "1|2",
         "0",
         "2",
         "ENSG00000118217|ENSG00000172071",
         "ATF6|EIF2AK3",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "22",
         "adamson_2016_upr_epistasis",
         "23",
         "ATF6_PERK_pMJ150",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1",
         "1|2",
         "0",
         "2",
         "ENSG00000118217|ENSG00000172071",
         "ATF6|EIF2AK3",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "23",
         "adamson_2016_upr_epistasis",
         "24",
         "ATF6_only_pMJ145",
         "chr1:161766298-161977574;1",
         "1",
         "1",
         "1",
         "ENSG00000118217",
         "ATF6",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "24",
         "adamson_2016_upr_epistasis",
         "25",
         "ATF6_PERK_pMJ150",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1",
         "1|2",
         "0",
         "2",
         "ENSG00000118217|ENSG00000172071",
         "ATF6|EIF2AK3",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "25",
         "adamson_2016_upr_epistasis",
         "26",
         "PERK_IRE1_pMJ154",
         "chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "2|17",
         "0",
         "2",
         "ENSG00000172071|ENSG00000178607",
         "EIF2AK3|ERN1",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "26",
         "adamson_2016_upr_epistasis",
         "27",
         "ATF6_only_pMJ145",
         "chr1:161766298-161977574;1",
         "1",
         "1",
         "1",
         "ENSG00000118217",
         "ATF6",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "27",
         "adamson_2016_upr_epistasis",
         "28",
         "ATF6_PERK_IRE1_pMJ158",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "1|2|17",
         "0",
         "3",
         "ENSG00000118217|ENSG00000172071|ENSG00000178607",
         "ATF6|EIF2AK3|ERN1",
         "protein_coding|protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "28",
         "adamson_2016_upr_epistasis",
         "29",
         "ATF6_PERK_IRE1_pMJ158",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "1|2|17",
         "0",
         "3",
         "ENSG00000118217|ENSG00000172071|ENSG00000178607",
         "ATF6|EIF2AK3|ERN1",
         "protein_coding|protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "29",
         "adamson_2016_upr_epistasis",
         "30",
         "3x_neg_ctrl_pMJ144-1",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "30",
         "adamson_2016_upr_epistasis",
         "31",
         "3x_neg_ctrl_pMJ144-2",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "31",
         "adamson_2016_upr_epistasis",
         "32",
         "ATF6_PERK_pMJ150",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1",
         "1|2",
         "0",
         "2",
         "ENSG00000118217|ENSG00000172071",
         "ATF6|EIF2AK3",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "32",
         "adamson_2016_upr_epistasis",
         "33",
         "IRE1_only_pMJ148",
         "chr17:64039080-64130819;-1",
         "17",
         "17",
         "1",
         "ENSG00000178607",
         "ERN1",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "33",
         "adamson_2016_upr_epistasis",
         "34",
         "3x_neg_ctrl_pMJ144-2",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "34",
         "adamson_2016_upr_epistasis",
         "35",
         "PERK_IRE1_pMJ154",
         "chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "2|17",
         "0",
         "2",
         "ENSG00000172071|ENSG00000178607",
         "EIF2AK3|ERN1",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "35",
         "adamson_2016_upr_epistasis",
         "36",
         "PERK_only_pMJ146",
         "chr2:88556741-88691518;-1",
         "2",
         "2",
         "1",
         "ENSG00000172071",
         "EIF2AK3",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "36",
         "adamson_2016_upr_epistasis",
         "37",
         "3x_neg_ctrl_pMJ144-1",
         null,
         null,
         "0",
         "1",
         "None|None",
         "None|None",
         null,
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "37",
         "adamson_2016_upr_epistasis",
         "38",
         "ATF6_IRE1_pMJ152",
         "chr1:161766298-161977574;1|chr17:64039080-64130819;-1",
         "1|17",
         "0",
         "2",
         "ENSG00000118217|ENSG00000178607",
         "ATF6|ERN1",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "38",
         "adamson_2016_upr_epistasis",
         "39",
         "IRE1_only_pMJ148",
         "chr17:64039080-64130819;-1",
         "17",
         "17",
         "1",
         "ENSG00000178607",
         "ERN1",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "39",
         "adamson_2016_upr_epistasis",
         "40",
         "PERK_only_pMJ146",
         "chr2:88556741-88691518;-1",
         "2",
         "2",
         "1",
         "ENSG00000172071",
         "EIF2AK3",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "40",
         "adamson_2016_upr_epistasis",
         "41",
         "ATF6_PERK_pMJ150",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1",
         "1|2",
         "0",
         "2",
         "ENSG00000118217|ENSG00000172071",
         "ATF6|EIF2AK3",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "41",
         "adamson_2016_upr_epistasis",
         "42",
         "ATF6_PERK_IRE1_pMJ158",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "1|2|17",
         "0",
         "3",
         "ENSG00000118217|ENSG00000172071|ENSG00000178607",
         "ATF6|EIF2AK3|ERN1",
         "protein_coding|protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "42",
         "adamson_2016_upr_epistasis",
         "43",
         "IRE1_only_pMJ148",
         "chr17:64039080-64130819;-1",
         "17",
         "17",
         "1",
         "ENSG00000178607",
         "ERN1",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "43",
         "adamson_2016_upr_epistasis",
         "44",
         "ATF6_only_pMJ145",
         "chr1:161766298-161977574;1",
         "1",
         "1",
         "1",
         "ENSG00000118217",
         "ATF6",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "44",
         "adamson_2016_upr_epistasis",
         "45",
         "PERK_only_pMJ146",
         "chr2:88556741-88691518;-1",
         "2",
         "2",
         "1",
         "ENSG00000172071",
         "EIF2AK3",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "45",
         "adamson_2016_upr_epistasis",
         "46",
         "ATF6_PERK_pMJ150",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1",
         "1|2",
         "0",
         "2",
         "ENSG00000118217|ENSG00000172071",
         "ATF6|EIF2AK3",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "46",
         "adamson_2016_upr_epistasis",
         "47",
         "ATF6_only_pMJ145",
         "chr1:161766298-161977574;1",
         "1",
         "1",
         "1",
         "ENSG00000118217",
         "ATF6",
         "protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "47",
         "adamson_2016_upr_epistasis",
         "48",
         "ATF6_PERK_IRE1_pMJ158",
         "chr1:161766298-161977574;1|chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "1|2|17",
         "0",
         "3",
         "ENSG00000118217|ENSG00000172071|ENSG00000178607",
         "ATF6|EIF2AK3|ERN1",
         "protein_coding|protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "48",
         "adamson_2016_upr_epistasis",
         "49",
         "ATF6_IRE1_pMJ152",
         "chr1:161766298-161977574;1|chr17:64039080-64130819;-1",
         "1|17",
         "0",
         "2",
         "ENSG00000118217|ENSG00000178607",
         "ATF6|ERN1",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ],
        [
         "49",
         "adamson_2016_upr_epistasis",
         "50",
         "PERK_IRE1_pMJ154",
         "chr2:88556741-88691518;-1|chr17:64039080-64130819;-1",
         "2|17",
         "0",
         "2",
         "ENSG00000172071|ENSG00000178607",
         "EIF2AK3|ERN1",
         "protein_coding|protein_coding",
         null,
         "CRISPRi",
         null,
         "P0DT0H0M0S",
         "tunicamycin",
         "CHEBI:29699",
         "cell line",
         null,
         "Homo sapiens",
         "blood",
         "UBERON:0000178",
         "lymphoblast",
         "CL:0017005",
         "K 562 cell",
         "CLO:0007050",
         "female",
         null,
         "adult",
         null,
         "chronic myelogenous leukemia, BCR-ABL1 positive",
         "MONDO:0011996",
         "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
         "https://doi.org/10.1016/j.cell.2016.11.048",
         "2016",
         "Britt Adamson",
         "Jonathan Weissman",
         "14595 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs in every combination (singly with controls, doubly with a control, or triply).",
         "Using our final three-guide Perturb-seq vector to simultaneously deliver 3 sgRNAs, we individually transduced K562 cells expressing dCas9-KRAB (cBA010) with constructs that targeted all three UPR sensor genes in every combination (singly with controls, doubly with a control, or triply). Transduced cells were then pooled and selected. After 2 days of combined growth, the cells were treated with DMSO for 6 hr, 4 μg/mL tunicamycin (Tm) for 6 hr, or 100 nM thapsigargin (Tg) for 4 hr and were profiled by Perturb-seq (24 conditions in total).",
         "16",
         "14697",
         "EFO:0022868",
         "endogenous",
         "EFO:0022895",
         "dCas9-KRAB",
         null,
         "retroviral transduction",
         null,
         "lentiviral transduction",
         null,
         "random locus integration",
         null,
         "random locus integration",
         null,
         "constitutive expression",
         null,
         "constitutive expression",
         "custom",
         null,
         null,
         "pooled",
         null,
         "focused",
         null,
         "inhibition",
         "Weissman",
         "3",
         "1",
         "16",
         null,
         null,
         "high-dimensional assay",
         null,
         "transcriptomic",
         null,
         "single-cell rna-seq",
         null,
         "Perturb-seq",
         null,
         null,
         "10x Genomics Single Cell 3-prime",
         null,
         "Illumina HiSeq 4000",
         null,
         "barcode sequencing",
         null,
         "CellRanger",
         null,
         "MAGeCK",
         null,
         "GRCh37",
         "[{\"dataset_accession\": \"GSM2406675\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406675\", \"dataset_description\": \"Raw counts\", \"dataset_file_name\": \"GSE90546_RAW.tar\"}, {\"dataset_accession\": \"GSM2406675_10X001\", \"dataset_uri\": \"https://zenodo.org/record/7041849/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406675_10X001.h5ad\"}]"
        ]
       ],
       "shape": {
        "columns": 91,
        "rows": 14697
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>perturbation_name</th>\n",
       "      <th>perturbed_target_coord</th>\n",
       "      <th>perturbed_target_chromosome</th>\n",
       "      <th>perturbed_target_chromosome_encoding</th>\n",
       "      <th>perturbed_target_number</th>\n",
       "      <th>perturbed_target_ensg</th>\n",
       "      <th>perturbed_target_symbol</th>\n",
       "      <th>perturbed_target_biotype</th>\n",
       "      <th>...</th>\n",
       "      <th>sequencing_platform_label</th>\n",
       "      <th>sequencing_strategy_id</th>\n",
       "      <th>sequencing_strategy_label</th>\n",
       "      <th>software_counts_id</th>\n",
       "      <th>software_counts_label</th>\n",
       "      <th>software_analysis_id</th>\n",
       "      <th>software_analysis_label</th>\n",
       "      <th>reference_genome_id</th>\n",
       "      <th>reference_genome_label</th>\n",
       "      <th>associated_datasets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>1</td>\n",
       "      <td>3x_neg_ctrl_pMJ144-1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>None|None</td>\n",
       "      <td>None|None</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>2</td>\n",
       "      <td>3x_neg_ctrl_pMJ144-2</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>None|None</td>\n",
       "      <td>None|None</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>3</td>\n",
       "      <td>3x_neg_ctrl_pMJ144-2</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>None|None</td>\n",
       "      <td>None|None</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>4</td>\n",
       "      <td>ATF6_PERK_IRE1_pMJ158</td>\n",
       "      <td>chr1:161766298-161977574;1|chr2:88556741-88691...</td>\n",
       "      <td>1|2|17</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000118217|ENSG00000172071|ENSG00000178607</td>\n",
       "      <td>ATF6|EIF2AK3|ERN1</td>\n",
       "      <td>protein_coding|protein_coding|protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>5</td>\n",
       "      <td>ATF6_PERK_pMJ150</td>\n",
       "      <td>chr1:161766298-161977574;1|chr2:88556741-88691...</td>\n",
       "      <td>1|2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSG00000118217|ENSG00000172071</td>\n",
       "      <td>ATF6|EIF2AK3</td>\n",
       "      <td>protein_coding|protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14692</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>14693</td>\n",
       "      <td>PERK_only_pMJ146</td>\n",
       "      <td>chr2:88556741-88691518;-1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000172071</td>\n",
       "      <td>EIF2AK3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14693</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>14694</td>\n",
       "      <td>PERK_IRE1_pMJ154</td>\n",
       "      <td>chr2:88556741-88691518;-1|chr17:64039080-64130...</td>\n",
       "      <td>2|17</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSG00000172071|ENSG00000178607</td>\n",
       "      <td>EIF2AK3|ERN1</td>\n",
       "      <td>protein_coding|protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14694</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>14695</td>\n",
       "      <td>3x_neg_ctrl_pMJ144-1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>None|None</td>\n",
       "      <td>None|None</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14695</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>14696</td>\n",
       "      <td>PERK_IRE1_pMJ154</td>\n",
       "      <td>chr2:88556741-88691518;-1|chr17:64039080-64130...</td>\n",
       "      <td>2|17</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSG00000172071|ENSG00000178607</td>\n",
       "      <td>EIF2AK3|ERN1</td>\n",
       "      <td>protein_coding|protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14696</th>\n",
       "      <td>adamson_2016_upr_epistasis</td>\n",
       "      <td>14697</td>\n",
       "      <td>3x_neg_ctrl_pMJ144-1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>None|None</td>\n",
       "      <td>None|None</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>barcode sequencing</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>CellRanger</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>MAGeCK</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>[{\"dataset_accession\": \"GSM2406675\", \"dataset_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14697 rows × 91 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       dataset_id  sample_id      perturbation_name  \\\n",
       "0      adamson_2016_upr_epistasis          1   3x_neg_ctrl_pMJ144-1   \n",
       "1      adamson_2016_upr_epistasis          2   3x_neg_ctrl_pMJ144-2   \n",
       "2      adamson_2016_upr_epistasis          3   3x_neg_ctrl_pMJ144-2   \n",
       "3      adamson_2016_upr_epistasis          4  ATF6_PERK_IRE1_pMJ158   \n",
       "4      adamson_2016_upr_epistasis          5       ATF6_PERK_pMJ150   \n",
       "...                           ...        ...                    ...   \n",
       "14692  adamson_2016_upr_epistasis      14693       PERK_only_pMJ146   \n",
       "14693  adamson_2016_upr_epistasis      14694       PERK_IRE1_pMJ154   \n",
       "14694  adamson_2016_upr_epistasis      14695   3x_neg_ctrl_pMJ144-1   \n",
       "14695  adamson_2016_upr_epistasis      14696       PERK_IRE1_pMJ154   \n",
       "14696  adamson_2016_upr_epistasis      14697   3x_neg_ctrl_pMJ144-1   \n",
       "\n",
       "                                  perturbed_target_coord  \\\n",
       "0                                                   <NA>   \n",
       "1                                                   <NA>   \n",
       "2                                                   <NA>   \n",
       "3      chr1:161766298-161977574;1|chr2:88556741-88691...   \n",
       "4      chr1:161766298-161977574;1|chr2:88556741-88691...   \n",
       "...                                                  ...   \n",
       "14692                          chr2:88556741-88691518;-1   \n",
       "14693  chr2:88556741-88691518;-1|chr17:64039080-64130...   \n",
       "14694                                               <NA>   \n",
       "14695  chr2:88556741-88691518;-1|chr17:64039080-64130...   \n",
       "14696                                               <NA>   \n",
       "\n",
       "      perturbed_target_chromosome  perturbed_target_chromosome_encoding  \\\n",
       "0                            <NA>                                     0   \n",
       "1                            <NA>                                     0   \n",
       "2                            <NA>                                     0   \n",
       "3                          1|2|17                                     0   \n",
       "4                             1|2                                     0   \n",
       "...                           ...                                   ...   \n",
       "14692                           2                                     2   \n",
       "14693                        2|17                                     0   \n",
       "14694                        <NA>                                     0   \n",
       "14695                        2|17                                     0   \n",
       "14696                        <NA>                                     0   \n",
       "\n",
       "       perturbed_target_number  \\\n",
       "0                            1   \n",
       "1                            1   \n",
       "2                            1   \n",
       "3                            3   \n",
       "4                            2   \n",
       "...                        ...   \n",
       "14692                        1   \n",
       "14693                        2   \n",
       "14694                        1   \n",
       "14695                        2   \n",
       "14696                        1   \n",
       "\n",
       "                                 perturbed_target_ensg  \\\n",
       "0                                            None|None   \n",
       "1                                            None|None   \n",
       "2                                            None|None   \n",
       "3      ENSG00000118217|ENSG00000172071|ENSG00000178607   \n",
       "4                      ENSG00000118217|ENSG00000172071   \n",
       "...                                                ...   \n",
       "14692                                  ENSG00000172071   \n",
       "14693                  ENSG00000172071|ENSG00000178607   \n",
       "14694                                        None|None   \n",
       "14695                  ENSG00000172071|ENSG00000178607   \n",
       "14696                                        None|None   \n",
       "\n",
       "      perturbed_target_symbol                      perturbed_target_biotype  \\\n",
       "0                   None|None                                          <NA>   \n",
       "1                   None|None                                          <NA>   \n",
       "2                   None|None                                          <NA>   \n",
       "3           ATF6|EIF2AK3|ERN1  protein_coding|protein_coding|protein_coding   \n",
       "4                ATF6|EIF2AK3                 protein_coding|protein_coding   \n",
       "...                       ...                                           ...   \n",
       "14692                 EIF2AK3                                protein_coding   \n",
       "14693            EIF2AK3|ERN1                 protein_coding|protein_coding   \n",
       "14694               None|None                                          <NA>   \n",
       "14695            EIF2AK3|ERN1                 protein_coding|protein_coding   \n",
       "14696               None|None                                          <NA>   \n",
       "\n",
       "       ... sequencing_platform_label sequencing_strategy_id  \\\n",
       "0      ...       Illumina HiSeq 4000                   <NA>   \n",
       "1      ...       Illumina HiSeq 4000                   <NA>   \n",
       "2      ...       Illumina HiSeq 4000                   <NA>   \n",
       "3      ...       Illumina HiSeq 4000                   <NA>   \n",
       "4      ...       Illumina HiSeq 4000                   <NA>   \n",
       "...    ...                       ...                    ...   \n",
       "14692  ...       Illumina HiSeq 4000                   <NA>   \n",
       "14693  ...       Illumina HiSeq 4000                   <NA>   \n",
       "14694  ...       Illumina HiSeq 4000                   <NA>   \n",
       "14695  ...       Illumina HiSeq 4000                   <NA>   \n",
       "14696  ...       Illumina HiSeq 4000                   <NA>   \n",
       "\n",
       "      sequencing_strategy_label software_counts_id software_counts_label  \\\n",
       "0            barcode sequencing               <NA>            CellRanger   \n",
       "1            barcode sequencing               <NA>            CellRanger   \n",
       "2            barcode sequencing               <NA>            CellRanger   \n",
       "3            barcode sequencing               <NA>            CellRanger   \n",
       "4            barcode sequencing               <NA>            CellRanger   \n",
       "...                         ...                ...                   ...   \n",
       "14692        barcode sequencing               <NA>            CellRanger   \n",
       "14693        barcode sequencing               <NA>            CellRanger   \n",
       "14694        barcode sequencing               <NA>            CellRanger   \n",
       "14695        barcode sequencing               <NA>            CellRanger   \n",
       "14696        barcode sequencing               <NA>            CellRanger   \n",
       "\n",
       "      software_analysis_id software_analysis_label reference_genome_id  \\\n",
       "0                     <NA>                  MAGeCK                <NA>   \n",
       "1                     <NA>                  MAGeCK                <NA>   \n",
       "2                     <NA>                  MAGeCK                <NA>   \n",
       "3                     <NA>                  MAGeCK                <NA>   \n",
       "4                     <NA>                  MAGeCK                <NA>   \n",
       "...                    ...                     ...                 ...   \n",
       "14692                 <NA>                  MAGeCK                <NA>   \n",
       "14693                 <NA>                  MAGeCK                <NA>   \n",
       "14694                 <NA>                  MAGeCK                <NA>   \n",
       "14695                 <NA>                  MAGeCK                <NA>   \n",
       "14696                 <NA>                  MAGeCK                <NA>   \n",
       "\n",
       "      reference_genome_label  \\\n",
       "0                     GRCh37   \n",
       "1                     GRCh37   \n",
       "2                     GRCh37   \n",
       "3                     GRCh37   \n",
       "4                     GRCh37   \n",
       "...                      ...   \n",
       "14692                 GRCh37   \n",
       "14693                 GRCh37   \n",
       "14694                 GRCh37   \n",
       "14695                 GRCh37   \n",
       "14696                 GRCh37   \n",
       "\n",
       "                                     associated_datasets  \n",
       "0      [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "1      [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "2      [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "3      [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "4      [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "...                                                  ...  \n",
       "14692  [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "14693  [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "14694  [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "14695  [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "14696  [{\"dataset_accession\": \"GSM2406675\", \"dataset_...  \n",
       "\n",
       "[14697 rows x 91 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cur_data.validate_data(slot='obs')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c86800d",
   "metadata": {},
   "source": [
    "# VAR slot curation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e8d9842",
   "metadata": {},
   "source": [
    "### Standardise genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "d61f4089",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variable data:\n",
      "DataFrame shape: (32738, 3)\n",
      "--------------------------------------------------\n",
      "                   ensembl_id  ncounts  ncells\n",
      "gene_symbol                                   \n",
      "MIR1302-10    ENSG00000243485      2.0       2\n",
      "FAM138A       ENSG00000237613      0.0       0\n",
      "OR4F5         ENSG00000186092      0.0       0\n",
      "RP11-34P13.7  ENSG00000238009      1.0       1\n",
      "RP11-34P13.8  ENSG00000239945      2.0       2\n",
      "...                       ...      ...     ...\n",
      "AC145205.1    ENSG00000215635      0.0       0\n",
      "BAGE5         ENSG00000268590      0.0       0\n",
      "CU459201.1    ENSG00000251180      0.0       0\n",
      "AC002321.2    ENSG00000215616      0.0       0\n",
      "AC002321.1    ENSG00000215611      0.0       0\n",
      "\n",
      "[32738 rows x 3 columns]\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.show_var()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "8d29f7b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converted 30168/32738 gene symbols/ENSG IDs to standardized gene symbols/ENSG IDs\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "cur_data.standardize_genes(\n",
    "    slot=\"var\", input_column=\"ensembl_id\", input_column_type=\"ensembl_gene_id\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "451e5138",
   "metadata": {},
   "source": [
    "### Validate var metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "6bba9183",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "adata.var is valid according to the var_schema.\n",
      "Validated data:\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "ensembl_gene_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "gene_symbol",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "ref": "4a81da27-e8df-4f0e-9a78-a809334fed2a",
       "rows": [
        [
         "MIR1302-10",
         "ENSG00000243485",
         "MIR1302-2HG"
        ],
        [
         "FAM138A",
         "ENSG00000237613",
         "FAM138A"
        ],
        [
         "OR4F5",
         "ENSG00000186092",
         "OR4F5"
        ],
        [
         "RP11-34P13.7",
         "ENSG00000238009",
         null
        ],
        [
         "RP11-34P13.8",
         "ENSG00000239945",
         null
        ],
        [
         "AL627309.1",
         "ENSG00000237683",
         null
        ],
        [
         "RP11-34P13.14",
         "ENSG00000239906",
         null
        ],
        [
         "RP11-34P13.9",
         "ENSG00000241599",
         null
        ],
        [
         "AP006222.2",
         "ENSG00000228463",
         null
        ],
        [
         "RP4-669L17.10",
         "ENSG00000237094",
         null
        ],
        [
         "OR4F29",
         "ENSG00000235249",
         null
        ],
        [
         "RP4-669L17.2",
         "ENSG00000236601",
         null
        ],
        [
         "RP5-857K21.15",
         "ENSG00000236743",
         null
        ],
        [
         "RP5-857K21.1",
         "ENSG00000231709",
         null
        ],
        [
         "RP5-857K21.2",
         "ENSG00000235146",
         null
        ],
        [
         "RP5-857K21.3",
         "ENSG00000239664",
         null
        ],
        [
         "RP5-857K21.4",
         "ENSG00000230021",
         null
        ],
        [
         "RP5-857K21.5",
         "ENSG00000223659",
         null
        ],
        [
         "OR4F16",
         "ENSG00000185097",
         null
        ],
        [
         "RP11-206L10.3",
         "ENSG00000235373",
         null
        ],
        [
         "RP11-206L10.5",
         "ENSG00000240618",
         null
        ],
        [
         "RP11-206L10.4",
         "ENSG00000229905",
         null
        ],
        [
         "RP11-206L10.2",
         "ENSG00000228327",
         null
        ],
        [
         "RP11-206L10.9",
         "ENSG00000237491",
         "LINC01409"
        ],
        [
         "AL669831.1",
         "ENSG00000269831",
         null
        ],
        [
         "FAM87B",
         "ENSG00000177757",
         "FAM87B"
        ],
        [
         "LINC00115",
         "ENSG00000225880",
         "LINC00115"
        ],
        [
         "FAM41C",
         "ENSG00000230368",
         "FAM41C"
        ],
        [
         "AL645608.2",
         "ENSG00000269308",
         null
        ],
        [
         "RP11-54O7.16",
         "ENSG00000272438",
         null
        ],
        [
         "RP11-54O7.1",
         "ENSG00000230699",
         null
        ],
        [
         "RP11-54O7.2",
         "ENSG00000241180",
         null
        ],
        [
         "RP11-54O7.3",
         "ENSG00000223764",
         "LINC02593"
        ],
        [
         "SAMD11",
         "ENSG00000187634",
         "SAMD11"
        ],
        [
         "AL645608.1",
         "ENSG00000268179",
         null
        ],
        [
         "NOC2L",
         "ENSG00000188976",
         "NOC2L"
        ],
        [
         "KLHL17",
         "ENSG00000187961",
         "KLHL17"
        ],
        [
         "PLEKHN1",
         "ENSG00000187583",
         "PLEKHN1"
        ],
        [
         "C1orf170",
         "ENSG00000187642",
         "PERM1"
        ],
        [
         "RP11-54O7.17",
         "ENSG00000272512",
         null
        ],
        [
         "HES4",
         "ENSG00000188290",
         "HES4"
        ],
        [
         "RP11-54O7.11",
         "ENSG00000224969",
         null
        ],
        [
         "ISG15",
         "ENSG00000187608",
         "ISG15"
        ],
        [
         "AGRN",
         "ENSG00000188157",
         "AGRN"
        ],
        [
         "RP11-54O7.18",
         "ENSG00000273443",
         null
        ],
        [
         "RNF223",
         "ENSG00000237330",
         "RNF223"
        ],
        [
         "C1orf159",
         "ENSG00000131591",
         "C1orf159"
        ],
        [
         "RP11-465B22.5",
         "ENSG00000223823",
         "LINC01342"
        ],
        [
         "RP11-465B22.8",
         "ENSG00000272141",
         null
        ],
        [
         "TTLL10-AS1",
         "ENSG00000205231",
         "TTLL10-AS1"
        ]
       ],
       "shape": {
        "columns": 2,
        "rows": 32738
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MIR1302-10</th>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>MIR1302-2HG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAM138A</th>\n",
       "      <td>ENSG00000237613</td>\n",
       "      <td>FAM138A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OR4F5</th>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RP11-34P13.7</th>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RP11-34P13.8</th>\n",
       "      <td>ENSG00000239945</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC145205.1</th>\n",
       "      <td>ENSG00000215635</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BAGE5</th>\n",
       "      <td>ENSG00000268590</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CU459201.1</th>\n",
       "      <td>ENSG00000251180</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC002321.2</th>\n",
       "      <td>ENSG00000215616</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC002321.1</th>\n",
       "      <td>ENSG00000215611</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32738 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              ensembl_gene_id  gene_symbol\n",
       "index                                     \n",
       "MIR1302-10    ENSG00000243485  MIR1302-2HG\n",
       "FAM138A       ENSG00000237613      FAM138A\n",
       "OR4F5         ENSG00000186092        OR4F5\n",
       "RP11-34P13.7  ENSG00000238009         None\n",
       "RP11-34P13.8  ENSG00000239945         None\n",
       "...                       ...          ...\n",
       "AC145205.1    ENSG00000215635         None\n",
       "BAGE5         ENSG00000268590         None\n",
       "CU459201.1    ENSG00000251180         None\n",
       "AC002321.2    ENSG00000215616         None\n",
       "AC002321.1    ENSG00000215611         None\n",
       "\n",
       "[32738 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cur_data.validate_data(slot='var')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e96453ed",
   "metadata": {},
   "source": [
    "# Save the dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "6bd01cb7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/zakirov/Documents/GitHub/PerturbationCatalogue/data_exploration/Perturbseq/curation_notebooks/../../curation_tools/curation_tools.py:281: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
      "  adata.obs = adata.obs.fillna(value=np.nan)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Curated data saved to ../curated/h5ad/adamson_2016_upr_epistasis_curated.h5ad\n"
     ]
    }
   ],
   "source": [
    "cur_data.save_curated_data_h5ad()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1efb3070",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting the conversion of adata to a long format DataFrame...\n",
      "Starting the conversion to long format...\n",
      "Processing 32738 genes in 164 chunks of size 200...\n",
      "Created ParquetWriter and wrote chunk 1/164\n",
      "Appended chunk 2/164 to parquet file\n",
      "Appended chunk 3/164 to parquet file\n",
      "Appended chunk 4/164 to parquet file\n",
      "Appended chunk 5/164 to parquet file\n",
      "Appended chunk 6/164 to parquet file\n",
      "Appended chunk 7/164 to parquet file\n",
      "Appended chunk 8/164 to parquet file\n",
      "Appended chunk 9/164 to parquet file\n",
      "Appended chunk 10/164 to parquet file\n",
      "Appended chunk 11/164 to parquet file\n",
      "Appended chunk 12/164 to parquet file\n",
      "Appended chunk 13/164 to parquet file\n",
      "Appended chunk 14/164 to parquet file\n",
      "Appended chunk 15/164 to parquet file\n",
      "Appended chunk 16/164 to parquet file\n",
      "Appended chunk 17/164 to parquet file\n",
      "Appended chunk 18/164 to parquet file\n",
      "Appended chunk 19/164 to parquet file\n",
      "Appended chunk 20/164 to parquet file\n",
      "Appended chunk 21/164 to parquet file\n",
      "Appended chunk 22/164 to parquet file\n",
      "Appended chunk 23/164 to parquet file\n",
      "Appended chunk 24/164 to parquet file\n",
      "Appended chunk 25/164 to parquet file\n",
      "Appended chunk 26/164 to parquet file\n",
      "Appended chunk 27/164 to parquet file\n",
      "Appended chunk 28/164 to parquet file\n",
      "Appended chunk 29/164 to parquet file\n",
      "Appended chunk 30/164 to parquet file\n",
      "Appended chunk 31/164 to parquet file\n",
      "Appended chunk 32/164 to parquet file\n",
      "Appended chunk 33/164 to parquet file\n",
      "Appended chunk 34/164 to parquet file\n",
      "Appended chunk 35/164 to parquet file\n",
      "Appended chunk 36/164 to parquet file\n",
      "Appended chunk 37/164 to parquet file\n",
      "Appended chunk 38/164 to parquet file\n",
      "Appended chunk 39/164 to parquet file\n",
      "Appended chunk 40/164 to parquet file\n",
      "Appended chunk 41/164 to parquet file\n",
      "Appended chunk 42/164 to parquet file\n",
      "Appended chunk 43/164 to parquet file\n",
      "Appended chunk 44/164 to parquet file\n",
      "Appended chunk 45/164 to parquet file\n",
      "Appended chunk 46/164 to parquet file\n",
      "Appended chunk 47/164 to parquet file\n",
      "Appended chunk 48/164 to parquet file\n",
      "Appended chunk 49/164 to parquet file\n",
      "Appended chunk 50/164 to parquet file\n",
      "Appended chunk 51/164 to parquet file\n",
      "Appended chunk 52/164 to parquet file\n",
      "Appended chunk 53/164 to parquet file\n",
      "Appended chunk 54/164 to parquet file\n",
      "Appended chunk 55/164 to parquet file\n",
      "Appended chunk 56/164 to parquet file\n",
      "Appended chunk 57/164 to parquet file\n",
      "Appended chunk 58/164 to parquet file\n",
      "Appended chunk 59/164 to parquet file\n",
      "Appended chunk 60/164 to parquet file\n",
      "Appended chunk 61/164 to parquet file\n",
      "Appended chunk 62/164 to parquet file\n",
      "Appended chunk 63/164 to parquet file\n",
      "Appended chunk 64/164 to parquet file\n",
      "Appended chunk 65/164 to parquet file\n",
      "Appended chunk 66/164 to parquet file\n",
      "Appended chunk 67/164 to parquet file\n",
      "Appended chunk 68/164 to parquet file\n",
      "Appended chunk 69/164 to parquet file\n",
      "Appended chunk 70/164 to parquet file\n",
      "Appended chunk 71/164 to parquet file\n",
      "Appended chunk 72/164 to parquet file\n",
      "Appended chunk 73/164 to parquet file\n",
      "Appended chunk 74/164 to parquet file\n",
      "Appended chunk 75/164 to parquet file\n",
      "Appended chunk 76/164 to parquet file\n",
      "Appended chunk 77/164 to parquet file\n",
      "Appended chunk 78/164 to parquet file\n",
      "Appended chunk 79/164 to parquet file\n",
      "Appended chunk 80/164 to parquet file\n",
      "Appended chunk 81/164 to parquet file\n",
      "Appended chunk 82/164 to parquet file\n",
      "Appended chunk 83/164 to parquet file\n",
      "Appended chunk 84/164 to parquet file\n",
      "Appended chunk 85/164 to parquet file\n",
      "Appended chunk 86/164 to parquet file\n",
      "Appended chunk 87/164 to parquet file\n",
      "Appended chunk 88/164 to parquet file\n",
      "Appended chunk 89/164 to parquet file\n",
      "Appended chunk 90/164 to parquet file\n",
      "Appended chunk 91/164 to parquet file\n",
      "Appended chunk 92/164 to parquet file\n",
      "Appended chunk 93/164 to parquet file\n",
      "Appended chunk 94/164 to parquet file\n",
      "Appended chunk 95/164 to parquet file\n",
      "Appended chunk 96/164 to parquet file\n",
      "Appended chunk 97/164 to parquet file\n",
      "Appended chunk 98/164 to parquet file\n",
      "Appended chunk 99/164 to parquet file\n",
      "Appended chunk 100/164 to parquet file\n",
      "Appended chunk 101/164 to parquet file\n",
      "Appended chunk 102/164 to parquet file\n",
      "Appended chunk 103/164 to parquet file\n",
      "Appended chunk 104/164 to parquet file\n",
      "Appended chunk 105/164 to parquet file\n",
      "Appended chunk 106/164 to parquet file\n",
      "Appended chunk 107/164 to parquet file\n",
      "Appended chunk 108/164 to parquet file\n",
      "Appended chunk 109/164 to parquet file\n",
      "Appended chunk 110/164 to parquet file\n",
      "Appended chunk 111/164 to parquet file\n",
      "Appended chunk 112/164 to parquet file\n",
      "Appended chunk 113/164 to parquet file\n",
      "Appended chunk 114/164 to parquet file\n",
      "Appended chunk 115/164 to parquet file\n",
      "Appended chunk 116/164 to parquet file\n",
      "Appended chunk 117/164 to parquet file\n",
      "Appended chunk 118/164 to parquet file\n",
      "Appended chunk 119/164 to parquet file\n",
      "Appended chunk 120/164 to parquet file\n",
      "Appended chunk 121/164 to parquet file\n",
      "Appended chunk 122/164 to parquet file\n",
      "Appended chunk 123/164 to parquet file\n",
      "Appended chunk 124/164 to parquet file\n",
      "Appended chunk 125/164 to parquet file\n",
      "Appended chunk 126/164 to parquet file\n",
      "Appended chunk 127/164 to parquet file\n",
      "Appended chunk 128/164 to parquet file\n",
      "Appended chunk 129/164 to parquet file\n",
      "Appended chunk 130/164 to parquet file\n",
      "Appended chunk 131/164 to parquet file\n",
      "Appended chunk 132/164 to parquet file\n",
      "Appended chunk 133/164 to parquet file\n",
      "Appended chunk 134/164 to parquet file\n",
      "Appended chunk 135/164 to parquet file\n",
      "Appended chunk 136/164 to parquet file\n",
      "Appended chunk 137/164 to parquet file\n",
      "Appended chunk 138/164 to parquet file\n",
      "Appended chunk 139/164 to parquet file\n",
      "Appended chunk 140/164 to parquet file\n",
      "Appended chunk 141/164 to parquet file\n",
      "Appended chunk 142/164 to parquet file\n",
      "Appended chunk 143/164 to parquet file\n",
      "Appended chunk 144/164 to parquet file\n",
      "Appended chunk 145/164 to parquet file\n",
      "Appended chunk 146/164 to parquet file\n",
      "Appended chunk 147/164 to parquet file\n",
      "Appended chunk 148/164 to parquet file\n",
      "Appended chunk 149/164 to parquet file\n",
      "Appended chunk 150/164 to parquet file\n",
      "Appended chunk 151/164 to parquet file\n",
      "Appended chunk 152/164 to parquet file\n",
      "Appended chunk 153/164 to parquet file\n",
      "Appended chunk 154/164 to parquet file\n",
      "Appended chunk 155/164 to parquet file\n",
      "Appended chunk 156/164 to parquet file\n",
      "Appended chunk 157/164 to parquet file\n",
      "Appended chunk 158/164 to parquet file\n",
      "Appended chunk 159/164 to parquet file\n",
      "Appended chunk 160/164 to parquet file\n",
      "Appended chunk 161/164 to parquet file\n",
      "Appended chunk 162/164 to parquet file\n",
      "Appended chunk 163/164 to parquet file\n",
      "Appended chunk 164/164 to parquet file\n",
      "All chunks written and ParquetWriter closed.\n",
      "Metadata written to ../curated/parquet/adamson_2016_upr_epistasis_curated_long_metadata.parquet\n"
     ]
    }
   ],
   "source": [
    "cur_data.save_curated_data_parquet(split_metadata=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "4c82b13f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting the conversion of adata to a long format DataFrame...\n",
      "Starting the conversion to long format...\n",
      "Processing 32738 genes in 164 chunks of size 200...\n",
      "Created ParquetWriter and wrote chunk 1/164\n",
      "Appended chunk 2/164 to parquet file\n",
      "Appended chunk 3/164 to parquet file\n",
      "Appended chunk 4/164 to parquet file\n",
      "Appended chunk 5/164 to parquet file\n",
      "Appended chunk 6/164 to parquet file\n",
      "Appended chunk 7/164 to parquet file\n",
      "Appended chunk 8/164 to parquet file\n",
      "Appended chunk 9/164 to parquet file\n",
      "Appended chunk 10/164 to parquet file\n",
      "Appended chunk 11/164 to parquet file\n",
      "Appended chunk 12/164 to parquet file\n",
      "Appended chunk 13/164 to parquet file\n",
      "Appended chunk 14/164 to parquet file\n",
      "Appended chunk 15/164 to parquet file\n",
      "Appended chunk 16/164 to parquet file\n",
      "Appended chunk 17/164 to parquet file\n",
      "Appended chunk 18/164 to parquet file\n",
      "Appended chunk 19/164 to parquet file\n",
      "Appended chunk 20/164 to parquet file\n",
      "Appended chunk 21/164 to parquet file\n",
      "Appended chunk 22/164 to parquet file\n",
      "Appended chunk 23/164 to parquet file\n",
      "Appended chunk 24/164 to parquet file\n",
      "Appended chunk 25/164 to parquet file\n",
      "Appended chunk 26/164 to parquet file\n",
      "Appended chunk 27/164 to parquet file\n",
      "Appended chunk 28/164 to parquet file\n",
      "Appended chunk 29/164 to parquet file\n",
      "Appended chunk 30/164 to parquet file\n",
      "Appended chunk 31/164 to parquet file\n",
      "Appended chunk 32/164 to parquet file\n",
      "Appended chunk 33/164 to parquet file\n",
      "Appended chunk 34/164 to parquet file\n",
      "Appended chunk 35/164 to parquet file\n",
      "Appended chunk 36/164 to parquet file\n",
      "Appended chunk 37/164 to parquet file\n",
      "Appended chunk 38/164 to parquet file\n",
      "Appended chunk 39/164 to parquet file\n",
      "Appended chunk 40/164 to parquet file\n",
      "Appended chunk 41/164 to parquet file\n",
      "Appended chunk 42/164 to parquet file\n",
      "Appended chunk 43/164 to parquet file\n",
      "Appended chunk 44/164 to parquet file\n",
      "Appended chunk 45/164 to parquet file\n",
      "Appended chunk 46/164 to parquet file\n",
      "Appended chunk 47/164 to parquet file\n",
      "Appended chunk 48/164 to parquet file\n",
      "Appended chunk 49/164 to parquet file\n",
      "Appended chunk 50/164 to parquet file\n",
      "Appended chunk 51/164 to parquet file\n",
      "Appended chunk 52/164 to parquet file\n",
      "Appended chunk 53/164 to parquet file\n",
      "Appended chunk 54/164 to parquet file\n",
      "Appended chunk 55/164 to parquet file\n",
      "Appended chunk 56/164 to parquet file\n",
      "Appended chunk 57/164 to parquet file\n",
      "Appended chunk 58/164 to parquet file\n",
      "Appended chunk 59/164 to parquet file\n",
      "Appended chunk 60/164 to parquet file\n",
      "Appended chunk 61/164 to parquet file\n",
      "Appended chunk 62/164 to parquet file\n",
      "Appended chunk 63/164 to parquet file\n",
      "Appended chunk 64/164 to parquet file\n",
      "Appended chunk 65/164 to parquet file\n",
      "Appended chunk 66/164 to parquet file\n",
      "Appended chunk 67/164 to parquet file\n",
      "Appended chunk 68/164 to parquet file\n",
      "Appended chunk 69/164 to parquet file\n",
      "Appended chunk 70/164 to parquet file\n",
      "Appended chunk 71/164 to parquet file\n",
      "Appended chunk 72/164 to parquet file\n",
      "Appended chunk 73/164 to parquet file\n",
      "Appended chunk 74/164 to parquet file\n",
      "Appended chunk 75/164 to parquet file\n",
      "Appended chunk 76/164 to parquet file\n",
      "Appended chunk 77/164 to parquet file\n",
      "Appended chunk 78/164 to parquet file\n",
      "Appended chunk 79/164 to parquet file\n",
      "Appended chunk 80/164 to parquet file\n",
      "Appended chunk 81/164 to parquet file\n",
      "Appended chunk 82/164 to parquet file\n",
      "Appended chunk 83/164 to parquet file\n",
      "Appended chunk 84/164 to parquet file\n",
      "Appended chunk 85/164 to parquet file\n",
      "Appended chunk 86/164 to parquet file\n",
      "Appended chunk 87/164 to parquet file\n",
      "Appended chunk 88/164 to parquet file\n",
      "Appended chunk 89/164 to parquet file\n",
      "Appended chunk 90/164 to parquet file\n",
      "Appended chunk 91/164 to parquet file\n",
      "Appended chunk 92/164 to parquet file\n",
      "Appended chunk 93/164 to parquet file\n",
      "Appended chunk 94/164 to parquet file\n",
      "Appended chunk 95/164 to parquet file\n",
      "Appended chunk 96/164 to parquet file\n",
      "Appended chunk 97/164 to parquet file\n",
      "Appended chunk 98/164 to parquet file\n",
      "Appended chunk 99/164 to parquet file\n",
      "Appended chunk 100/164 to parquet file\n",
      "Appended chunk 101/164 to parquet file\n",
      "Appended chunk 102/164 to parquet file\n",
      "Appended chunk 103/164 to parquet file\n",
      "Appended chunk 104/164 to parquet file\n",
      "Appended chunk 105/164 to parquet file\n",
      "Appended chunk 106/164 to parquet file\n",
      "Appended chunk 107/164 to parquet file\n",
      "Appended chunk 108/164 to parquet file\n",
      "Appended chunk 109/164 to parquet file\n",
      "Appended chunk 110/164 to parquet file\n",
      "Appended chunk 111/164 to parquet file\n",
      "Appended chunk 112/164 to parquet file\n",
      "Appended chunk 113/164 to parquet file\n",
      "Appended chunk 114/164 to parquet file\n",
      "Appended chunk 115/164 to parquet file\n",
      "Appended chunk 116/164 to parquet file\n",
      "Appended chunk 117/164 to parquet file\n",
      "Appended chunk 118/164 to parquet file\n",
      "Appended chunk 119/164 to parquet file\n",
      "Appended chunk 120/164 to parquet file\n",
      "Appended chunk 121/164 to parquet file\n",
      "Appended chunk 122/164 to parquet file\n",
      "Appended chunk 123/164 to parquet file\n",
      "Appended chunk 124/164 to parquet file\n",
      "Appended chunk 125/164 to parquet file\n",
      "Appended chunk 126/164 to parquet file\n",
      "Appended chunk 127/164 to parquet file\n",
      "Appended chunk 128/164 to parquet file\n",
      "Appended chunk 129/164 to parquet file\n",
      "Appended chunk 130/164 to parquet file\n",
      "Appended chunk 131/164 to parquet file\n",
      "Appended chunk 132/164 to parquet file\n",
      "Appended chunk 133/164 to parquet file\n",
      "Appended chunk 134/164 to parquet file\n",
      "Appended chunk 135/164 to parquet file\n",
      "Appended chunk 136/164 to parquet file\n",
      "Appended chunk 137/164 to parquet file\n",
      "Appended chunk 138/164 to parquet file\n",
      "Appended chunk 139/164 to parquet file\n",
      "Appended chunk 140/164 to parquet file\n",
      "Appended chunk 141/164 to parquet file\n",
      "Appended chunk 142/164 to parquet file\n",
      "Appended chunk 143/164 to parquet file\n",
      "Appended chunk 144/164 to parquet file\n",
      "Appended chunk 145/164 to parquet file\n",
      "Appended chunk 146/164 to parquet file\n",
      "Appended chunk 147/164 to parquet file\n",
      "Appended chunk 148/164 to parquet file\n",
      "Appended chunk 149/164 to parquet file\n",
      "Appended chunk 150/164 to parquet file\n",
      "Appended chunk 151/164 to parquet file\n",
      "Appended chunk 152/164 to parquet file\n",
      "Appended chunk 153/164 to parquet file\n",
      "Appended chunk 154/164 to parquet file\n",
      "Appended chunk 155/164 to parquet file\n",
      "Appended chunk 156/164 to parquet file\n",
      "Appended chunk 157/164 to parquet file\n",
      "Appended chunk 158/164 to parquet file\n",
      "Appended chunk 159/164 to parquet file\n",
      "Appended chunk 160/164 to parquet file\n",
      "Appended chunk 161/164 to parquet file\n",
      "Appended chunk 162/164 to parquet file\n",
      "Appended chunk 163/164 to parquet file\n",
      "Appended chunk 164/164 to parquet file\n",
      "All chunks written and ParquetWriter closed.\n"
     ]
    }
   ],
   "source": [
    "cur_data.save_curated_data_parquet(split_metadata=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "PerturbationCatalogue",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
