{
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "## Note that this notebook should be run in GCP Colab environment\n",
        "\n",
        "Runtime requirements:\n",
        "- min. 750 GB disk space\n",
        "- min. 500 GB RAM"
      ],
      "metadata": {
        "id": "xnKMhbDWt6Wh"
      },
      "id": "xnKMhbDWt6Wh"
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "IfVEJsY2-cD0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IfVEJsY2-cD0",
        "outputId": "05c2a8ef-fce9-4f46-a1e6-d5e517afbde1"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content\n"
          ]
        }
      ],
      "source": [
        "!pwd"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "1wjh-CBq-fLv",
      "metadata": {
        "id": "1wjh-CBq-fLv",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "2a60d11f-2a2c-4ffc-9836-316c1f848987"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'PerturbationCatalogue'...\n",
            "remote: Enumerating objects: 4128, done.\u001b[K\n",
            "remote: Counting objects: 100% (1326/1326), done.\u001b[K\n",
            "remote: Compressing objects: 100% (602/602), done.\u001b[K\n",
            "remote: Total 4128 (delta 822), reused 798 (delta 724), pack-reused 2802 (from 2)\u001b[K\n",
            "Receiving objects: 100% (4128/4128), 58.79 MiB | 46.74 MiB/s, done.\n",
            "Resolving deltas: 100% (2728/2728), done.\n"
          ]
        }
      ],
      "source": [
        "!git clone https://github.com/EMBL-EBI-ABC/PerturbationCatalogue.git"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "gjFGZPGu-hlO",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gjFGZPGu-hlO",
        "outputId": "bbbc9ecd-28c3-40e6-8714-44053ae49b84"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/PerturbationCatalogue/data_exploration\n"
          ]
        }
      ],
      "source": [
        "%cd PerturbationCatalogue/data_exploration/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "Hvdndxyp-jio",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Hvdndxyp-jio",
        "outputId": "35bf9b90-eac2-42ab-c7fd-2affc73b0353"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "remote: Enumerating objects: 27, done.\u001b[K\n",
            "remote: Counting objects:   4% (1/25)\u001b[K\rremote: Counting objects:   8% (2/25)\u001b[K\rremote: Counting objects:  12% (3/25)\u001b[K\rremote: Counting objects:  16% (4/25)\u001b[K\rremote: Counting objects:  20% (5/25)\u001b[K\rremote: Counting objects:  24% (6/25)\u001b[K\rremote: Counting objects:  28% (7/25)\u001b[K\rremote: Counting objects:  32% (8/25)\u001b[K\rremote: Counting objects:  36% (9/25)\u001b[K\rremote: Counting objects:  40% (10/25)\u001b[K\rremote: Counting objects:  44% (11/25)\u001b[K\rremote: Counting objects:  48% (12/25)\u001b[K\rremote: Counting objects:  52% (13/25)\u001b[K\rremote: Counting objects:  56% (14/25)\u001b[K\rremote: Counting objects:  60% (15/25)\u001b[K\rremote: Counting objects:  64% (16/25)\u001b[K\rremote: Counting objects:  68% (17/25)\u001b[K\rremote: Counting objects:  72% (18/25)\u001b[K\rremote: Counting objects:  76% (19/25)\u001b[K\rremote: Counting objects:  80% (20/25)\u001b[K\rremote: Counting objects:  84% (21/25)\u001b[K\rremote: Counting objects:  88% (22/25)\u001b[K\rremote: Counting objects:  92% (23/25)\u001b[K\rremote: Counting objects:  96% (24/25)\u001b[K\rremote: Counting objects: 100% (25/25)\u001b[K\rremote: Counting objects: 100% (25/25), done.\u001b[K\n",
            "remote: Compressing objects:  33% (1/3)\u001b[K\rremote: Compressing objects:  66% (2/3)\u001b[K\rremote: Compressing objects: 100% (3/3)\u001b[K\rremote: Compressing objects: 100% (3/3), done.\u001b[K\n",
            "remote: Total 11 (delta 8), reused 11 (delta 8), pack-reused 0 (from 0)\u001b[K\n",
            "Unpacking objects: 100% (11/11), 4.34 KiB | 2.00 KiB/s, done.\n",
            "From https://github.com/EMBL-EBI-ABC/PerturbationCatalogue\n",
            "   769b40f..827153c  aleks/curate_nadig_2025 -> origin/aleks/curate_nadig_2025\n",
            "   62127cc..4019a01  dev                     -> origin/dev\n",
            "Updating 769b40f..827153c\n",
            "error: Your local changes to the following files would be overwritten by merge:\n",
            "\tdata_exploration/curation_tools/ontologies/gene_ont.parquet\n",
            "Please commit your changes or stash them before you merge.\n",
            "Aborting\n",
            "error: Your local changes to the following files would be overwritten by checkout:\n",
            "\tdata_exploration/curation_tools/ontologies/gene_ont.parquet\n",
            "\tdata_exploration/local_curation_tools.egg-info/SOURCES.txt\n",
            "Please commit your changes or stash them before you switch branches.\n",
            "Aborting\n"
          ]
        }
      ],
      "source": [
        "!git pull\n",
        "!git switch aleks/bq_table"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "pwzakG-S-rkM",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pwzakG-S-rkM",
        "outputId": "0f19e6a3-72ca-4e2a-cf63-264019fc36ed",
        "collapsed": true
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: ipykernel in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 2)) (6.17.1)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 5)) (3.10.0)\n",
            "Collecting scipy<1.15.0 (from -r requirements.txt (line 6))\n",
            "  Downloading scipy-1.14.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m60.8/60.8 kB\u001b[0m \u001b[31m13.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 7)) (2.2.2)\n",
            "Collecting decoupler==2.0.1 (from -r requirements.txt (line 10))\n",
            "  Downloading decoupler-2.0.1-py3-none-any.whl.metadata (7.5 kB)\n",
            "Collecting pertpy==0.11.4 (from -r requirements.txt (line 11))\n",
            "  Downloading pertpy-0.11.4-py3-none-any.whl.metadata (8.7 kB)\n",
            "Collecting scanpy==1.11.1 (from -r requirements.txt (line 12))\n",
            "  Downloading scanpy-1.11.1-py3-none-any.whl.metadata (9.9 kB)\n",
            "Collecting anndata==0.12.2 (from -r requirements.txt (line 13))\n",
            "  Downloading anndata-0.12.2-py3-none-any.whl.metadata (9.6 kB)\n",
            "Collecting pandera (from -r requirements.txt (line 16))\n",
            "  Downloading pandera-0.26.1-py3-none-any.whl.metadata (10 kB)\n",
            "Collecting libchebipy (from -r requirements.txt (line 17))\n",
            "  Downloading libChEBIpy-1.0.10-py3-none-any.whl.metadata (677 bytes)\n",
            "Collecting gprofiler-official (from -r requirements.txt (line 18))\n",
            "  Downloading gprofiler_official-1.0.0-py3-none-any.whl.metadata (11 kB)\n",
            "Collecting adjusttext (from decoupler==2.0.1->-r requirements.txt (line 10))\n",
            "  Downloading adjustText-1.3.0-py3-none-any.whl.metadata (3.1 kB)\n",
            "Collecting docrep (from decoupler==2.0.1->-r requirements.txt (line 10))\n",
            "  Downloading docrep-0.3.2.tar.gz (33 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting igraph (from decoupler==2.0.1->-r requirements.txt (line 10))\n",
            "  Downloading igraph-0.11.9-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.4 kB)\n",
            "Collecting marsilea (from decoupler==2.0.1->-r requirements.txt (line 10))\n",
            "  Downloading marsilea-0.5.5-py3-none-any.whl.metadata (7.3 kB)\n",
            "Requirement already satisfied: numba in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (0.60.0)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (2.32.3)\n",
            "Requirement already satisfied: seaborn in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (0.13.2)\n",
            "Collecting session-info2 (from decoupler==2.0.1->-r requirements.txt (line 10))\n",
            "  Downloading session_info2-0.2.1-py3-none-any.whl.metadata (3.4 kB)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (4.67.1)\n",
            "Requirement already satisfied: xgboost in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (3.0.4)\n",
            "Requirement already satisfied: arviz in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (0.22.0)\n",
            "Collecting blitzgsea (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading blitzgsea-1.3.54.tar.gz (626 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m627.0/627.0 kB\u001b[0m \u001b[31m89.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting fast-array-utils (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading fast_array_utils-1.2.3-py3-none-any.whl.metadata (3.7 kB)\n",
            "Collecting lamin-utils (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading lamin_utils-0.15.0-py2.py3-none-any.whl.metadata (961 bytes)\n",
            "Collecting mudata (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading mudata-0.3.2-py3-none-any.whl.metadata (8.4 kB)\n",
            "Requirement already satisfied: openpyxl in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (3.1.5)\n",
            "Collecting ott-jax (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading ott_jax-0.5.1-py3-none-any.whl.metadata (20 kB)\n",
            "Collecting pubchempy (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading PubChemPy-1.0.4.tar.gz (29 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: pyarrow in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (18.1.0)\n",
            "Requirement already satisfied: rich in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (13.9.4)\n",
            "Requirement already satisfied: scikit-learn>=1.4 in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (1.6.1)\n",
            "Collecting scikit-misc (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading scikit_misc-0.5.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.0 kB)\n",
            "Collecting scvi-tools (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading scvi_tools-1.3.3-py3-none-any.whl.metadata (20 kB)\n",
            "Collecting sparsecca (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading sparsecca-0.3.1-py3-none-any.whl.metadata (1.9 kB)\n",
            "Requirement already satisfied: h5py>=3.7 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (3.14.0)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (1.5.1)\n",
            "Collecting legacy-api-wrap>=1.4 (from scanpy==1.11.1->-r requirements.txt (line 12))\n",
            "  Downloading legacy_api_wrap-1.4.1-py3-none-any.whl.metadata (2.1 kB)\n",
            "Requirement already satisfied: natsort in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (8.4.0)\n",
            "Requirement already satisfied: networkx>=2.7 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (3.5)\n",
            "Requirement already satisfied: numpy>=1.24 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (2.0.2)\n",
            "Requirement already satisfied: packaging>=21.3 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (25.0)\n",
            "Requirement already satisfied: patsy!=1.0.0 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (1.0.1)\n",
            "Requirement already satisfied: pynndescent>=0.5 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (0.5.13)\n",
            "Collecting scikit-learn>=1.4 (from pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading scikit_learn-1.5.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)\n",
            "Requirement already satisfied: statsmodels>=0.13 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (0.14.5)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (4.14.1)\n",
            "Requirement already satisfied: umap-learn!=0.5.0,>=0.5 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (0.5.9.post2)\n",
            "Collecting array-api-compat>=1.7.1 (from anndata==0.12.2->-r requirements.txt (line 13))\n",
            "  Downloading array_api_compat-1.12.0-py3-none-any.whl.metadata (2.5 kB)\n",
            "Collecting zarr!=3.0.*,>=2.18.7 (from anndata==0.12.2->-r requirements.txt (line 13))\n",
            "  Downloading zarr-3.1.2-py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: debugpy>=1.0 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (1.8.15)\n",
            "Requirement already satisfied: ipython>=7.23.1 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (7.34.0)\n",
            "Requirement already satisfied: jupyter-client>=6.1.12 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (6.1.12)\n",
            "Requirement already satisfied: matplotlib-inline>=0.1 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (0.1.7)\n",
            "Requirement already satisfied: nest-asyncio in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (1.6.0)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (5.9.5)\n",
            "Requirement already satisfied: pyzmq>=17 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (26.2.1)\n",
            "Requirement already satisfied: tornado>=6.1 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (6.4.2)\n",
            "Requirement already satisfied: traitlets>=5.1.0 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (5.7.1)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (1.3.3)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (0.12.1)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (4.59.0)\n",
            "Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (1.4.9)\n",
            "Requirement already satisfied: pillow>=8 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (11.3.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (3.2.3)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->-r requirements.txt (line 7)) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas->-r requirements.txt (line 7)) (2025.2)\n",
            "Requirement already satisfied: pydantic in /usr/local/lib/python3.11/dist-packages (from pandera->-r requirements.txt (line 16)) (2.11.7)\n",
            "Requirement already satisfied: typeguard in /usr/local/lib/python3.11/dist-packages (from pandera->-r requirements.txt (line 16)) (4.4.4)\n",
            "Collecting typing_inspect>=0.6.0 (from pandera->-r requirements.txt (line 16))\n",
            "  Downloading typing_inspect-0.9.0-py3-none-any.whl.metadata (1.5 kB)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.11/dist-packages (from libchebipy->-r requirements.txt (line 17)) (1.17.0)\n",
            "Requirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (75.2.0)\n",
            "Requirement already satisfied: jedi>=0.16 in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.19.2)\n",
            "Requirement already satisfied: decorator in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (4.4.2)\n",
            "Requirement already satisfied: pickleshare in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.7.5)\n",
            "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (3.0.51)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (2.19.2)\n",
            "Requirement already satisfied: backcall in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.2.0)\n",
            "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (4.9.0)\n",
            "Requirement already satisfied: jupyter-core>=4.6.0 in /usr/local/lib/python3.11/dist-packages (from jupyter-client>=6.1.12->ipykernel->-r requirements.txt (line 2)) (5.8.1)\n",
            "Requirement already satisfied: llvmlite<0.44,>=0.43.0dev0 in /usr/local/lib/python3.11/dist-packages (from numba->decoupler==2.0.1->-r requirements.txt (line 10)) (0.43.0)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.11/dist-packages (from scikit-learn>=1.4->pertpy==0.11.4->-r requirements.txt (line 11)) (3.6.0)\n",
            "Collecting mypy-extensions>=0.3.0 (from typing_inspect>=0.6.0->pandera->-r requirements.txt (line 16))\n",
            "  Downloading mypy_extensions-1.1.0-py3-none-any.whl.metadata (1.1 kB)\n",
            "INFO: pip is looking at multiple versions of umap-learn to determine which version is compatible with other requirements. This could take a while.\n",
            "Collecting umap-learn!=0.5.0,>=0.5 (from scanpy==1.11.1->-r requirements.txt (line 12))\n",
            "  Downloading umap_learn-0.5.8-py3-none-any.whl.metadata (23 kB)\n",
            "  Downloading umap_learn-0.5.7-py3-none-any.whl.metadata (21 kB)\n",
            "Collecting donfig>=0.8 (from zarr!=3.0.*,>=2.18.7->anndata==0.12.2->-r requirements.txt (line 13))\n",
            "  Downloading donfig-0.8.1.post1-py3-none-any.whl.metadata (5.0 kB)\n",
            "Collecting numcodecs>=0.14 (from numcodecs[crc32c]>=0.14->zarr!=3.0.*,>=2.18.7->anndata==0.12.2->-r requirements.txt (line 13))\n",
            "  Downloading numcodecs-0.16.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.3 kB)\n",
            "Requirement already satisfied: xarray>=2023.7.0 in /usr/local/lib/python3.11/dist-packages (from arviz->pertpy==0.11.4->-r requirements.txt (line 11)) (2025.7.1)\n",
            "Requirement already satisfied: h5netcdf>=1.0.2 in /usr/local/lib/python3.11/dist-packages (from arviz->pertpy==0.11.4->-r requirements.txt (line 11)) (1.6.4)\n",
            "Requirement already satisfied: xarray-einstats>=0.3 in /usr/local/lib/python3.11/dist-packages (from arviz->pertpy==0.11.4->-r requirements.txt (line 11)) (0.9.1)\n",
            "Requirement already satisfied: mpmath in /usr/local/lib/python3.11/dist-packages (from blitzgsea->pertpy==0.11.4->-r requirements.txt (line 11)) (1.3.0)\n",
            "Collecting texttable>=1.6.2 (from igraph->decoupler==2.0.1->-r requirements.txt (line 10))\n",
            "  Downloading texttable-1.7.0-py2.py3-none-any.whl.metadata (9.8 kB)\n",
            "Collecting legendkit (from marsilea->decoupler==2.0.1->-r requirements.txt (line 10))\n",
            "  Downloading legendkit-0.3.6-py3-none-any.whl.metadata (2.6 kB)\n",
            "Requirement already satisfied: platformdirs in /usr/local/lib/python3.11/dist-packages (from marsilea->decoupler==2.0.1->-r requirements.txt (line 10)) (4.3.8)\n",
            "Requirement already satisfied: et-xmlfile in /usr/local/lib/python3.11/dist-packages (from openpyxl->pertpy==0.11.4->-r requirements.txt (line 11)) (2.0.0)\n",
            "Requirement already satisfied: jax>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.5.3)\n",
            "Collecting jaxopt>=0.8 (from ott-jax->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading jaxopt-0.8.5-py3-none-any.whl.metadata (3.3 kB)\n",
            "Collecting lineax>=0.0.7 (from ott-jax->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading lineax-0.0.8-py3-none-any.whl.metadata (18 kB)\n",
            "Requirement already satisfied: optax>=0.2.4 in /usr/local/lib/python3.11/dist-packages (from ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.2.5)\n",
            "Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.11/dist-packages (from pydantic->pandera->-r requirements.txt (line 16)) (0.7.0)\n",
            "Requirement already satisfied: pydantic-core==2.33.2 in /usr/local/lib/python3.11/dist-packages (from pydantic->pandera->-r requirements.txt (line 16)) (2.33.2)\n",
            "Requirement already satisfied: typing-inspection>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from pydantic->pandera->-r requirements.txt (line 16)) (0.4.1)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->decoupler==2.0.1->-r requirements.txt (line 10)) (3.4.3)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->decoupler==2.0.1->-r requirements.txt (line 10)) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->decoupler==2.0.1->-r requirements.txt (line 10)) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->decoupler==2.0.1->-r requirements.txt (line 10)) (2025.8.3)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.11/dist-packages (from rich->pertpy==0.11.4->-r requirements.txt (line 11)) (4.0.0)\n",
            "Requirement already satisfied: flax in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.10.6)\n",
            "Requirement already satisfied: jaxlib<0.7.0 in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.5.3)\n",
            "Collecting lightning>=2.0 (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading lightning-2.5.4-py3-none-any.whl.metadata (39 kB)\n",
            "Collecting ml-collections>=0.1.1 (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading ml_collections-1.1.0-py3-none-any.whl.metadata (22 kB)\n",
            "Collecting numpyro>=0.12.1 (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading numpyro-0.19.0-py3-none-any.whl.metadata (37 kB)\n",
            "Collecting pyro-ppl>=1.6.0 (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading pyro_ppl-1.9.1-py3-none-any.whl.metadata (7.8 kB)\n",
            "Collecting sparse>=0.14.0 (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading sparse-0.17.0-py2.py3-none-any.whl.metadata (5.3 kB)\n",
            "Requirement already satisfied: tensorboard>=2.0 in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2.19.0)\n",
            "Requirement already satisfied: torch in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2.6.0+cu124)\n",
            "Collecting torchmetrics>=0.11.0 (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading torchmetrics-1.8.1-py3-none-any.whl.metadata (22 kB)\n",
            "Requirement already satisfied: pyomo in /usr/local/lib/python3.11/dist-packages (from sparsecca->pertpy==0.11.4->-r requirements.txt (line 11)) (6.9.3)\n",
            "Requirement already satisfied: nvidia-nccl-cu12 in /usr/local/lib/python3.11/dist-packages (from xgboost->decoupler==2.0.1->-r requirements.txt (line 10)) (2.23.4)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.11/dist-packages (from donfig>=0.8->zarr!=3.0.*,>=2.18.7->anndata==0.12.2->-r requirements.txt (line 13)) (6.0.2)\n",
            "Requirement already satisfied: ml_dtypes>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from jax>=0.4.0->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.5.3)\n",
            "Requirement already satisfied: opt_einsum in /usr/local/lib/python3.11/dist-packages (from jax>=0.4.0->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (3.4.0)\n",
            "Requirement already satisfied: parso<0.9.0,>=0.8.4 in /usr/local/lib/python3.11/dist-packages (from jedi>=0.16->ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.8.4)\n",
            "Requirement already satisfied: fsspec<2027.0,>=2022.5.0 in /usr/local/lib/python3.11/dist-packages (from fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2025.3.0)\n",
            "Collecting lightning-utilities<2.0,>=0.10.0 (from lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading lightning_utilities-0.15.2-py3-none-any.whl.metadata (5.7 kB)\n",
            "Collecting pytorch-lightning (from lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading pytorch_lightning-2.5.4-py3-none-any.whl.metadata (20 kB)\n",
            "Collecting equinox>=0.11.10 (from lineax>=0.0.7->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading equinox-0.13.0-py3-none-any.whl.metadata (18 kB)\n",
            "Collecting jaxtyping>=0.2.24 (from lineax>=0.0.7->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading jaxtyping-0.3.2-py3-none-any.whl.metadata (7.0 kB)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.11/dist-packages (from markdown-it-py>=2.2.0->rich->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.2)\n",
            "Requirement already satisfied: absl-py in /usr/local/lib/python3.11/dist-packages (from ml-collections>=0.1.1->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.4.0)\n",
            "Collecting crc32c>=2.7 (from numcodecs[crc32c]>=0.14->zarr!=3.0.*,>=2.18.7->anndata==0.12.2->-r requirements.txt (line 13))\n",
            "  Downloading crc32c-2.7.1-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.3 kB)\n",
            "Requirement already satisfied: multipledispatch in /usr/local/lib/python3.11/dist-packages (from numpyro>=0.12.1->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.0.0)\n",
            "Requirement already satisfied: chex>=0.1.87 in /usr/local/lib/python3.11/dist-packages (from optax>=0.2.4->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.90)\n",
            "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.11/dist-packages (from pexpect>4.3->ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.7.0)\n",
            "Requirement already satisfied: wcwidth in /usr/local/lib/python3.11/dist-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.2.13)\n",
            "Collecting pyro-api>=0.1.1 (from pyro-ppl>=1.6.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading pyro_api-0.1.2-py3-none-any.whl.metadata (2.5 kB)\n",
            "Requirement already satisfied: grpcio>=1.48.2 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.74.0)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.8.2)\n",
            "Requirement already satisfied: protobuf!=4.24.0,>=3.19.6 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (6.32.0)\n",
            "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.7.2)\n",
            "Requirement already satisfied: werkzeug>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.1.3)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.18.0)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.1.6)\n",
            "Collecting nvidia-cuda-nvrtc-cu12==12.4.127 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_cuda_nvrtc_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-cuda-runtime-cu12==12.4.127 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_cuda_runtime_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-cuda-cupti-cu12==12.4.127 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_cuda_cupti_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl.metadata (1.6 kB)\n",
            "Collecting nvidia-cudnn-cu12==9.1.0.70 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_cudnn_cu12-9.1.0.70-py3-none-manylinux2014_x86_64.whl.metadata (1.6 kB)\n",
            "Collecting nvidia-cublas-cu12==12.4.5.8 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_cublas_cu12-12.4.5.8-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-cufft-cu12==11.2.1.3 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_cufft_cu12-11.2.1.3-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-curand-cu12==10.3.5.147 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_curand_cu12-10.3.5.147-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-cusolver-cu12==11.6.1.9 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_cusolver_cu12-11.6.1.9-py3-none-manylinux2014_x86_64.whl.metadata (1.6 kB)\n",
            "Collecting nvidia-cusparse-cu12==12.3.1.170 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_cusparse_cu12-12.3.1.170-py3-none-manylinux2014_x86_64.whl.metadata (1.6 kB)\n",
            "Requirement already satisfied: nvidia-cusparselt-cu12==0.6.2 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.6.2)\n",
            "Collecting nvidia-nccl-cu12 (from xgboost->decoupler==2.0.1->-r requirements.txt (line 10))\n",
            "  Downloading nvidia_nccl_cu12-2.21.5-py3-none-manylinux2014_x86_64.whl.metadata (1.8 kB)\n",
            "Requirement already satisfied: nvidia-nvtx-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (12.4.127)\n",
            "Collecting nvidia-nvjitlink-cu12==12.4.127 (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading nvidia_nvjitlink_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Requirement already satisfied: triton==3.2.0 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.2.0)\n",
            "Requirement already satisfied: sympy==1.13.1 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.13.1)\n",
            "Requirement already satisfied: msgpack in /usr/local/lib/python3.11/dist-packages (from flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.1.1)\n",
            "Requirement already satisfied: orbax-checkpoint in /usr/local/lib/python3.11/dist-packages (from flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.11.21)\n",
            "Requirement already satisfied: tensorstore in /usr/local/lib/python3.11/dist-packages (from flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.76)\n",
            "Requirement already satisfied: treescope>=0.1.7 in /usr/local/lib/python3.11/dist-packages (from flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.10)\n",
            "Requirement already satisfied: ply in /usr/local/lib/python3.11/dist-packages (from pyomo->sparsecca->pertpy==0.11.4->-r requirements.txt (line 11)) (3.11)\n",
            "Requirement already satisfied: toolz>=0.9.0 in /usr/local/lib/python3.11/dist-packages (from chex>=0.1.87->optax>=0.2.4->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.12.1)\n",
            "Collecting wadler-lindig>=0.1.0 (from equinox>=0.11.10->lineax>=0.0.7->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11))\n",
            "  Downloading wadler_lindig-0.1.7-py3-none-any.whl.metadata (17 kB)\n",
            "Requirement already satisfied: aiohttp!=4.0.0a0,!=4.0.0a1 in /usr/local/lib/python3.11/dist-packages (from fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.12.15)\n",
            "Requirement already satisfied: MarkupSafe>=2.1.1 in /usr/local/lib/python3.11/dist-packages (from werkzeug>=1.0.1->tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.0.2)\n",
            "Requirement already satisfied: etils[epath,epy] in /usr/local/lib/python3.11/dist-packages (from orbax-checkpoint->flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.13.0)\n",
            "Requirement already satisfied: aiofiles in /usr/local/lib/python3.11/dist-packages (from orbax-checkpoint->flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (24.1.0)\n",
            "Requirement already satisfied: humanize in /usr/local/lib/python3.11/dist-packages (from orbax-checkpoint->flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (4.12.3)\n",
            "Requirement already satisfied: simplejson>=3.16.0 in /usr/local/lib/python3.11/dist-packages (from orbax-checkpoint->flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.20.1)\n",
            "Requirement already satisfied: aiohappyeyeballs>=2.5.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2.6.1)\n",
            "Requirement already satisfied: aiosignal>=1.4.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.4.0)\n",
            "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (25.3.0)\n",
            "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.7.0)\n",
            "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (6.6.4)\n",
            "Requirement already satisfied: propcache>=0.2.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.3.2)\n",
            "Requirement already satisfied: yarl<2.0,>=1.17.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.20.1)\n",
            "Requirement already satisfied: importlib_resources in /usr/local/lib/python3.11/dist-packages (from etils[epath,epy]->orbax-checkpoint->flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (6.5.2)\n",
            "Requirement already satisfied: zipp in /usr/local/lib/python3.11/dist-packages (from etils[epath,epy]->orbax-checkpoint->flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.23.0)\n",
            "Downloading decoupler-2.0.1-py3-none-any.whl (110 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m110.7/110.7 kB\u001b[0m \u001b[31m24.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pertpy-0.11.4-py3-none-any.whl (215 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m215.0/215.0 kB\u001b[0m \u001b[31m45.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading scanpy-1.11.1-py3-none-any.whl (2.1 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.1/2.1 MB\u001b[0m \u001b[31m106.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading anndata-0.12.2-py3-none-any.whl (169 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m169.9/169.9 kB\u001b[0m \u001b[31m43.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading scipy-1.14.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (41.2 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m41.2/41.2 MB\u001b[0m \u001b[31m76.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pandera-0.26.1-py3-none-any.whl (292 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m292.9/292.9 kB\u001b[0m \u001b[31m59.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading libChEBIpy-1.0.10-py3-none-any.whl (18 kB)\n",
            "Downloading gprofiler_official-1.0.0-py3-none-any.whl (9.3 kB)\n",
            "Downloading array_api_compat-1.12.0-py3-none-any.whl (58 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m58.2/58.2 kB\u001b[0m \u001b[31m16.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading legacy_api_wrap-1.4.1-py3-none-any.whl (10.0 kB)\n",
            "Downloading scikit_learn-1.5.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.3/13.3 MB\u001b[0m \u001b[31m176.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading typing_inspect-0.9.0-py3-none-any.whl (8.8 kB)\n",
            "Downloading umap_learn-0.5.7-py3-none-any.whl (88 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m88.8/88.8 kB\u001b[0m \u001b[31m24.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading zarr-3.1.2-py3-none-any.whl (261 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m261.0/261.0 kB\u001b[0m \u001b[31m59.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading adjustText-1.3.0-py3-none-any.whl (13 kB)\n",
            "Downloading fast_array_utils-1.2.3-py3-none-any.whl (34 kB)\n",
            "Downloading igraph-0.11.9-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.4 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.4/4.4 MB\u001b[0m \u001b[31m153.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading lamin_utils-0.15.0-py2.py3-none-any.whl (25 kB)\n",
            "Downloading marsilea-0.5.5-py3-none-any.whl (84 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m84.8/84.8 kB\u001b[0m \u001b[31m22.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading mudata-0.3.2-py3-none-any.whl (41 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m41.9/41.9 kB\u001b[0m \u001b[31m10.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading ott_jax-0.5.1-py3-none-any.whl (291 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m291.9/291.9 kB\u001b[0m \u001b[31m64.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading scikit_misc-0.5.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (188 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m189.0/189.0 kB\u001b[0m \u001b[31m47.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading scvi_tools-1.3.3-py3-none-any.whl (560 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m560.6/560.6 kB\u001b[0m \u001b[31m95.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading session_info2-0.2.1-py3-none-any.whl (16 kB)\n",
            "Downloading sparsecca-0.3.1-py3-none-any.whl (12 kB)\n",
            "Downloading donfig-0.8.1.post1-py3-none-any.whl (21 kB)\n",
            "Downloading jaxopt-0.8.5-py3-none-any.whl (172 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m172.4/172.4 kB\u001b[0m \u001b[31m27.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading lightning-2.5.4-py3-none-any.whl (825 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m825.2/825.2 kB\u001b[0m \u001b[31m112.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading lineax-0.0.8-py3-none-any.whl (68 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m68.0/68.0 kB\u001b[0m \u001b[31m18.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading ml_collections-1.1.0-py3-none-any.whl (76 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m76.7/76.7 kB\u001b[0m \u001b[31m20.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading mypy_extensions-1.1.0-py3-none-any.whl (5.0 kB)\n",
            "Downloading numcodecs-0.16.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (8.8 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m8.8/8.8 MB\u001b[0m \u001b[31m164.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading numpyro-0.19.0-py3-none-any.whl (370 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m370.9/370.9 kB\u001b[0m \u001b[31m74.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyro_ppl-1.9.1-py3-none-any.whl (755 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m756.0/756.0 kB\u001b[0m \u001b[31m89.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading sparse-0.17.0-py2.py3-none-any.whl (259 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m259.4/259.4 kB\u001b[0m \u001b[31m45.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading texttable-1.7.0-py2.py3-none-any.whl (10 kB)\n",
            "Downloading nvidia_nccl_cu12-2.21.5-py3-none-manylinux2014_x86_64.whl (188.7 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m188.7/188.7 MB\u001b[0m \u001b[31m14.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cublas_cu12-12.4.5.8-py3-none-manylinux2014_x86_64.whl (363.4 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m363.4/363.4 MB\u001b[0m \u001b[31m7.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cuda_cupti_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl (13.8 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.8/13.8 MB\u001b[0m \u001b[31m204.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cuda_nvrtc_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl (24.6 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m24.6/24.6 MB\u001b[0m \u001b[31m165.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cuda_runtime_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl (883 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m883.7/883.7 kB\u001b[0m \u001b[31m121.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cudnn_cu12-9.1.0.70-py3-none-manylinux2014_x86_64.whl (664.8 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m664.8/664.8 MB\u001b[0m \u001b[31m3.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cufft_cu12-11.2.1.3-py3-none-manylinux2014_x86_64.whl (211.5 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m211.5/211.5 MB\u001b[0m \u001b[31m12.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_curand_cu12-10.3.5.147-py3-none-manylinux2014_x86_64.whl (56.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m56.3/56.3 MB\u001b[0m \u001b[31m52.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cusolver_cu12-11.6.1.9-py3-none-manylinux2014_x86_64.whl (127.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m127.9/127.9 MB\u001b[0m \u001b[31m20.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cusparse_cu12-12.3.1.170-py3-none-manylinux2014_x86_64.whl (207.5 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m207.5/207.5 MB\u001b[0m \u001b[31m12.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_nvjitlink_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl (21.1 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m21.1/21.1 MB\u001b[0m \u001b[31m177.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading torchmetrics-1.8.1-py3-none-any.whl (982 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m983.0/983.0 kB\u001b[0m \u001b[31m111.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading legendkit-0.3.6-py3-none-any.whl (24 kB)\n",
            "Downloading crc32c-2.7.1-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (53 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m53.7/53.7 kB\u001b[0m \u001b[31m12.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading equinox-0.13.0-py3-none-any.whl (177 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m177.7/177.7 kB\u001b[0m \u001b[31m43.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading jaxtyping-0.3.2-py3-none-any.whl (55 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m55.4/55.4 kB\u001b[0m \u001b[31m13.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading lightning_utilities-0.15.2-py3-none-any.whl (29 kB)\n",
            "Downloading pyro_api-0.1.2-py3-none-any.whl (11 kB)\n",
            "Downloading pytorch_lightning-2.5.4-py3-none-any.whl (829 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m829.2/829.2 kB\u001b[0m \u001b[31m119.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading wadler_lindig-0.1.7-py3-none-any.whl (20 kB)\n",
            "Building wheels for collected packages: blitzgsea, docrep, pubchempy\n",
            "  Building wheel for blitzgsea (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for blitzgsea: filename=blitzgsea-1.3.54-py3-none-any.whl size=625671 sha256=3129bedbf1cf50fbd92fd19a77a4d59ac927bfce45d45eacdaad2305ea522ddf\n",
            "  Stored in directory: /root/.cache/pip/wheels/3d/df/0c/a1a09a7eabc171ece48873363bc79f78141d008b3429098e5f\n",
            "  Building wheel for docrep (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for docrep: filename=docrep-0.3.2-py3-none-any.whl size=19876 sha256=f1a5ebc5e1263d5d8dd0b0aae3d205ff3b4f02d7767962ea27c50a85a0b14583\n",
            "  Stored in directory: /root/.cache/pip/wheels/06/76/8f/0ecb7d357c0bff71a2bd1940671be2d07a200752da9189bb55\n",
            "  Building wheel for pubchempy (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pubchempy: filename=PubChemPy-1.0.4-py3-none-any.whl size=13818 sha256=80b86460cba8e3f768d789a277e64019da49b61989ed7722e6e88e075773d905\n",
            "  Stored in directory: /root/.cache/pip/wheels/8b/e3/6c/3385b2db08b0985a87f5b117f98d0cb61a3ae3ca3bcbbd8307\n",
            "Successfully built blitzgsea docrep pubchempy\n",
            "Installing collected packages: texttable, pyro-api, pubchempy, lamin-utils, wadler-lindig, session-info2, scipy, scikit-misc, nvidia-nvjitlink-cu12, nvidia-nccl-cu12, nvidia-curand-cu12, nvidia-cufft-cu12, nvidia-cuda-runtime-cu12, nvidia-cuda-nvrtc-cu12, nvidia-cuda-cupti-cu12, nvidia-cublas-cu12, numcodecs, mypy-extensions, ml-collections, lightning-utilities, legacy-api-wrap, igraph, fast-array-utils, donfig, docrep, crc32c, array-api-compat, typing_inspect, sparse, scikit-learn, nvidia-cusparse-cu12, nvidia-cudnn-cu12, libchebipy, jaxtyping, gprofiler-official, zarr, pandera, nvidia-cusolver-cu12, legendkit, adjusttext, umap-learn, sparsecca, numpyro, marsilea, jaxopt, equinox, blitzgsea, anndata, torchmetrics, scanpy, pyro-ppl, mudata, lineax, decoupler, pytorch-lightning, ott-jax, lightning, scvi-tools, pertpy\n",
            "  Attempting uninstall: scipy\n",
            "    Found existing installation: scipy 1.16.1\n",
            "    Uninstalling scipy-1.16.1:\n",
            "      Successfully uninstalled scipy-1.16.1\n",
            "  Attempting uninstall: nvidia-nvjitlink-cu12\n",
            "    Found existing installation: nvidia-nvjitlink-cu12 12.5.82\n",
            "    Uninstalling nvidia-nvjitlink-cu12-12.5.82:\n",
            "      Successfully uninstalled nvidia-nvjitlink-cu12-12.5.82\n",
            "  Attempting uninstall: nvidia-nccl-cu12\n",
            "    Found existing installation: nvidia-nccl-cu12 2.23.4\n",
            "    Uninstalling nvidia-nccl-cu12-2.23.4:\n",
            "      Successfully uninstalled nvidia-nccl-cu12-2.23.4\n",
            "  Attempting uninstall: nvidia-curand-cu12\n",
            "    Found existing installation: nvidia-curand-cu12 10.3.6.82\n",
            "    Uninstalling nvidia-curand-cu12-10.3.6.82:\n",
            "      Successfully uninstalled nvidia-curand-cu12-10.3.6.82\n",
            "  Attempting uninstall: nvidia-cufft-cu12\n",
            "    Found existing installation: nvidia-cufft-cu12 11.2.3.61\n",
            "    Uninstalling nvidia-cufft-cu12-11.2.3.61:\n",
            "      Successfully uninstalled nvidia-cufft-cu12-11.2.3.61\n",
            "  Attempting uninstall: nvidia-cuda-runtime-cu12\n",
            "    Found existing installation: nvidia-cuda-runtime-cu12 12.5.82\n",
            "    Uninstalling nvidia-cuda-runtime-cu12-12.5.82:\n",
            "      Successfully uninstalled nvidia-cuda-runtime-cu12-12.5.82\n",
            "  Attempting uninstall: nvidia-cuda-nvrtc-cu12\n",
            "    Found existing installation: nvidia-cuda-nvrtc-cu12 12.5.82\n",
            "    Uninstalling nvidia-cuda-nvrtc-cu12-12.5.82:\n",
            "      Successfully uninstalled nvidia-cuda-nvrtc-cu12-12.5.82\n",
            "  Attempting uninstall: nvidia-cuda-cupti-cu12\n",
            "    Found existing installation: nvidia-cuda-cupti-cu12 12.5.82\n",
            "    Uninstalling nvidia-cuda-cupti-cu12-12.5.82:\n",
            "      Successfully uninstalled nvidia-cuda-cupti-cu12-12.5.82\n",
            "  Attempting uninstall: nvidia-cublas-cu12\n",
            "    Found existing installation: nvidia-cublas-cu12 12.5.3.2\n",
            "    Uninstalling nvidia-cublas-cu12-12.5.3.2:\n",
            "      Successfully uninstalled nvidia-cublas-cu12-12.5.3.2\n",
            "  Attempting uninstall: scikit-learn\n",
            "    Found existing installation: scikit-learn 1.6.1\n",
            "    Uninstalling scikit-learn-1.6.1:\n",
            "      Successfully uninstalled scikit-learn-1.6.1\n",
            "  Attempting uninstall: nvidia-cusparse-cu12\n",
            "    Found existing installation: nvidia-cusparse-cu12 12.5.1.3\n",
            "    Uninstalling nvidia-cusparse-cu12-12.5.1.3:\n",
            "      Successfully uninstalled nvidia-cusparse-cu12-12.5.1.3\n",
            "  Attempting uninstall: nvidia-cudnn-cu12\n",
            "    Found existing installation: nvidia-cudnn-cu12 9.3.0.75\n",
            "    Uninstalling nvidia-cudnn-cu12-9.3.0.75:\n",
            "      Successfully uninstalled nvidia-cudnn-cu12-9.3.0.75\n",
            "  Attempting uninstall: nvidia-cusolver-cu12\n",
            "    Found existing installation: nvidia-cusolver-cu12 11.6.3.83\n",
            "    Uninstalling nvidia-cusolver-cu12-11.6.3.83:\n",
            "      Successfully uninstalled nvidia-cusolver-cu12-11.6.3.83\n",
            "  Attempting uninstall: umap-learn\n",
            "    Found existing installation: umap-learn 0.5.9.post2\n",
            "    Uninstalling umap-learn-0.5.9.post2:\n",
            "      Successfully uninstalled umap-learn-0.5.9.post2\n",
            "Successfully installed adjusttext-1.3.0 anndata-0.12.2 array-api-compat-1.12.0 blitzgsea-1.3.54 crc32c-2.7.1 decoupler-2.0.1 docrep-0.3.2 donfig-0.8.1.post1 equinox-0.13.0 fast-array-utils-1.2.3 gprofiler-official-1.0.0 igraph-0.11.9 jaxopt-0.8.5 jaxtyping-0.3.2 lamin-utils-0.15.0 legacy-api-wrap-1.4.1 legendkit-0.3.6 libchebipy-1.0.10 lightning-2.5.4 lightning-utilities-0.15.2 lineax-0.0.8 marsilea-0.5.5 ml-collections-1.1.0 mudata-0.3.2 mypy-extensions-1.1.0 numcodecs-0.16.2 numpyro-0.19.0 nvidia-cublas-cu12-12.4.5.8 nvidia-cuda-cupti-cu12-12.4.127 nvidia-cuda-nvrtc-cu12-12.4.127 nvidia-cuda-runtime-cu12-12.4.127 nvidia-cudnn-cu12-9.1.0.70 nvidia-cufft-cu12-11.2.1.3 nvidia-curand-cu12-10.3.5.147 nvidia-cusolver-cu12-11.6.1.9 nvidia-cusparse-cu12-12.3.1.170 nvidia-nccl-cu12-2.21.5 nvidia-nvjitlink-cu12-12.4.127 ott-jax-0.5.1 pandera-0.26.1 pertpy-0.11.4 pubchempy-1.0.4 pyro-api-0.1.2 pyro-ppl-1.9.1 pytorch-lightning-2.5.4 scanpy-1.11.1 scikit-learn-1.5.2 scikit-misc-0.5.1 scipy-1.14.1 scvi-tools-1.3.3 session-info2-0.2.1 sparse-0.17.0 sparsecca-0.3.1 texttable-1.7.0 torchmetrics-1.8.1 typing_inspect-0.9.0 umap-learn-0.5.7 wadler-lindig-0.1.7 zarr-3.1.2\n"
          ]
        }
      ],
      "source": [
        "%pip install -r requirements.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "OeDtRFPz-xvt",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OeDtRFPz-xvt",
        "outputId": "f2a3adf2-7d28-4304-ab11-7c15148d30a9",
        "collapsed": true
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Welcome! This command will take you through the configuration of gcloud.\n",
            "\n",
            "Your current configuration has been set to: [default]\n",
            "\n",
            "You can skip diagnostics next time by using the following flag:\n",
            "  gcloud init --skip-diagnostics\n",
            "\n",
            "Network diagnostic detects and fixes local network connection issues.\n",
            "Reachability Check passed.\n",
            "Network diagnostic passed (1/1 checks passed).\n",
            "\n",
            "You are signed in as: [zakirov@ebi.ac.uk].\n",
            "\n",
            "Your current project has been set to: [prj-ext-dev-pertcat-437314].\n",
            "\n",
            "Do you want to configure a default Compute Region and Zone? (Y/n)?  n\n",
            "\n",
            "The Google Cloud CLI is configured and ready to use!\n",
            "\n",
            "* Commands that require authentication will use zakirov@ebi.ac.uk by default\n",
            "* Commands will reference project `prj-ext-dev-pertcat-437314` by default\n",
            "Run `gcloud help config` to learn how to change individual settings\n",
            "\n",
            "This gcloud configuration is called [default]. You can create additional configurations if you work with multiple accounts and/or projects.\n",
            "Run `gcloud topic configurations` to learn more.\n",
            "\n",
            "Some things to try next:\n",
            "\n",
            "* Run `gcloud --help` to see the Cloud Platform services you can interact with. And run `gcloud help COMMAND` to get help on any gcloud command.\n",
            "* Run `gcloud topic --help` to learn about advanced features of the CLI like arg files and output formatting\n",
            "* Run `gcloud cheat-sheet` to see a roster of go-to `gcloud` commands.\n"
          ]
        }
      ],
      "source": [
        "!gcloud init --project \"prj-ext-dev-pertcat-437314\" --account \"zakirov@ebi.ac.uk\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "aAczvpxw-3sR",
      "metadata": {
        "id": "aAczvpxw-3sR"
      },
      "outputs": [],
      "source": [
        "!mkdir -p Perturbseq/non_curated/h5ad/"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gcloud storage cp gs://perturbation-catalogue-lake/perturbseq/non_curated/orion_2025_hct116.h5ad Perturbseq/non_curated/h5ad/"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "G124pxpA3VSB",
        "outputId": "851877d1-60d0-4cf8-d856-53291b1a5c33"
      },
      "id": "G124pxpA3VSB",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Copying gs://perturbation-catalogue-lake/perturbseq/non_curated/orion_2025_hct116.h5ad to file://Perturbseq/non_curated/h5ad/orion_2025_hct116.h5ad\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\u001b[1;33mWARNING:\u001b[0m Component check failed. Could not verify SDK install path.\n",
            "\n",
            "\n",
            "Command killed by keyboard interrupt\n",
            "\n",
            "\n",
            "\n",
            "Command killed by keyboard interrupt\n",
            "\n",
            "\n",
            "\n",
            "Command killed by keyboard interrupt\n",
            "\n",
            "\n",
            "\n",
            "Command killed by keyboard interrupt\n",
            "\n",
            "\n",
            "\n",
            "Command killed by keyboard interrupt\n",
            "\n",
            "\n",
            "\n",
            "Command killed by keyboard interrupt\n",
            "\n",
            "\n",
            "\n",
            "Command killed by keyboard interrupt\n",
            "\n",
            "\n",
            "\n",
            "Command killed by keyboard interrupt\n",
            "\n",
            "\n",
            "\n",
            "Command killed by keyboard interrupt\n",
            "\n",
            "\n",
            "Average throughput: 882.8MiB/s\n",
            "\u001b[1;31mERROR:\u001b[0m gcloud crashed (RuntimeError): release unlocked lock\n",
            "\n",
            "If you would like to report this issue, please run the following command:\n",
            "  gcloud feedback\n",
            "\n",
            "To check gcloud for common problems, please run the following command:\n",
            "  gcloud info --run-diagnostics\n",
            "^C\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "id": "8c9fe243",
      "metadata": {
        "id": "8c9fe243"
      },
      "source": [
        "# Import"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "fd9fdd06",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fd9fdd06",
        "outputId": "8548ad2c-6e13-48d3-c04f-63625cd8d34e"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.12/dist-packages/pandera/_pandas_deprecated.py:157: FutureWarning: Importing pandas-specific classes and functions from the\n",
            "top-level pandera module will be **removed in a future version of pandera**.\n",
            "If you're using pandera to validate pandas objects, we highly recommend updating\n",
            "your import:\n",
            "\n",
            "```\n",
            "# old import\n",
            "import pandera as pa\n",
            "\n",
            "# new import\n",
            "import pandera.pandas as pa\n",
            "```\n",
            "\n",
            "If you're using pandera to validate objects from other compatible libraries\n",
            "like pyspark or polars, see the supported libraries section of the documentation\n",
            "for more information on how to import pandera:\n",
            "\n",
            "https://pandera.readthedocs.io/en/stable/supported_libraries.html\n",
            "\n",
            "To disable this warning, set the environment variable:\n",
            "\n",
            "```\n",
            "export DISABLE_PANDERA_IMPORT_WARNING=True\n",
            "```\n",
            "\n",
            "  warnings.warn(_future_warning, FutureWarning)\n"
          ]
        }
      ],
      "source": [
        "import pandas as pd\n",
        "import json\n",
        "\n",
        "from curation_tools.curation_tools import (\n",
        "    CuratedDataset,\n",
        "    ObsSchema,\n",
        "    VarSchema,\n",
        "    Experiment,\n",
        "    download_file,\n",
        "    upload_parquet_to_bq\n",
        ")\n",
        "\n",
        "# import logging\n",
        "# logging.basicConfig(\n",
        "#     level=logging.DEBUG,\n",
        "#     format=\"%(asctime)s %(levelname)s %(name)s: %(message)s\",\n",
        "#     handlers=[\n",
        "#         logging.FileHandler(\"curation.log\"),\n",
        "#         logging.StreamHandler(),  # keep console output too\n",
        "#     ],\n",
        "#     force=True,\n",
        "# )"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "e1f448e7",
      "metadata": {
        "id": "e1f448e7"
      },
      "source": [
        "# Pre-process the data"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "3392c826",
      "metadata": {
        "id": "3392c826"
      },
      "source": [
        "### Load the data and construct the dummy anndata object to explore the dataset offline"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "e948b59b",
      "metadata": {
        "id": "e948b59b"
      },
      "outputs": [],
      "source": [
        "# from scipy.sparse import csr_matrix\n",
        "# import numpy as np\n",
        "# import scanpy as sc\n",
        "\n",
        "# N1 = obs.shape[0]  # Number of objects (e.g., cells)\n",
        "# N2 = var.shape[0]  # Number of categories (e.g., genes)\n",
        "# rows = np.arange(N1)\n",
        "# cols = (np.floor(np.random.permutation(N1)/float(N1)*N2)).astype(int) # Randomly pick N1 objects and assign to N2 categories in almost equal proportion\n",
        "# w = np.ones(N1)\n",
        "# conn_matrix = csr_matrix((w, (rows, cols)), shape=(N1, N2), dtype=int)\n",
        "\n",
        "# adata = sc.AnnData(X=conn_matrix, obs=obs, var=var)\n",
        "\n",
        "# # save the non-curated AnnData object\n",
        "# adata.write_h5ad(\"Perturbseq/non_curated/h5ad/orion_2025_dummy.h5ad\")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "c0b2a8d6",
      "metadata": {
        "id": "c0b2a8d6"
      },
      "source": [
        "# Initialise the dataset object"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "3782a1d4",
      "metadata": {
        "id": "3782a1d4",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ae36dd17-405d-419d-860f-f4c16c3cdfad"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Loading data from Perturbseq/non_curated/h5ad/orion_2025_hct116.h5ad\n"
          ]
        }
      ],
      "source": [
        "noncurated_path = \"Perturbseq/non_curated/h5ad/orion_2025_hct116.h5ad\"\n",
        "cur_data = CuratedDataset(\n",
        "    obs_schema=ObsSchema,\n",
        "    var_schema=VarSchema,\n",
        "    exp_metadata_schema=Experiment,\n",
        "    noncurated_path=noncurated_path\n",
        ")\n",
        "\n",
        "cur_data.load_data()"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "cur_data.adata.obs"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 669
        },
        "id": "lwSOqo_73hlk",
        "outputId": "a9cd08b5-b561-4af2-f236-41a07fd99642"
      },
      "id": "lwSOqo_73hlk",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                           sample  num_features  \\\n",
              "AAACCAAAGACATGTT-HCT116_Batch1      HCT116_Batch1             2   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1      HCT116_Batch1             2   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1      HCT116_Batch1             2   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1      HCT116_Batch1             2   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1      HCT116_Batch1             2   \n",
              "...                                           ...           ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109  HCT116_Batch109             2   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109  HCT116_Batch109             2   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109  HCT116_Batch109             2   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109  HCT116_Batch109             2   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  HCT116_Batch109             2   \n",
              "\n",
              "                                                   guide_target gene_target  \\\n",
              "AAACCAAAGACATGTT-HCT116_Batch1          ST14_P1P2-1|ST14_P1P2-2        ST14   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1    SIGLEC5_P1P2-1|SIGLEC5_P1P2-2     SIGLEC5   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1        VSNL1_P1P2-1|VSNL1_P1P2-2       VSNL1   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1        KCNK7_P1P2-1|KCNK7_P1P2-2       KCNK7   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1        APOA4_P1P2-1|APOA4_P1P2-2       APOA4   \n",
              "...                                                         ...         ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109        ACP6_P1P2-1|ACP6_P1P2-2        ACP6   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109      RNF39_P1P2-1|RNF39_P1P2-2       RNF39   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109              AFP_P2-1|AFP_P2-2         AFP   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109    METTL9_P1P2-1|METTL9_P1P2-2      METTL9   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  SLC5A10_P1P2-1|SLC5A10_P1P2-2     SLC5A10   \n",
              "\n",
              "                                  n_genes_by_counts  total_counts  \\\n",
              "AAACCAAAGACATGTT-HCT116_Batch1                 4883       19136.0   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                 8130       47916.0   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                 6531       28435.0   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                 5931       26080.0   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                 7157       38366.0   \n",
              "...                                             ...           ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109               5107       17882.0   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109               3643        8155.0   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109               2831        6334.0   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109               3868        8766.0   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109               4095        9686.0   \n",
              "\n",
              "                                  total_counts_mt  pct_counts_mt  \\\n",
              "AAACCAAAGACATGTT-HCT116_Batch1             1179.0       6.161162   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1             1562.0       3.259871   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1             1042.0       3.664498   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1             1087.0       4.167945   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1              955.0       2.489183   \n",
              "...                                           ...            ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109            767.0       4.289229   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109            664.0       8.142244   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109           1553.0      24.518473   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109            868.0       9.901894   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109            738.0       7.619245   \n",
              "\n",
              "                                  pass_guide_filter  \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                 True  \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                 True  \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                 True  \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                 True  \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                 True  \n",
              "...                                             ...  \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109               True  \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109               True  \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109               True  \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109               True  \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109               True  \n",
              "\n",
              "[3409169 rows x 9 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-aa39e725-d7ca-40cc-9c8d-72d20ba48674\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>sample</th>\n",
              "      <th>num_features</th>\n",
              "      <th>guide_target</th>\n",
              "      <th>gene_target</th>\n",
              "      <th>n_genes_by_counts</th>\n",
              "      <th>total_counts</th>\n",
              "      <th>total_counts_mt</th>\n",
              "      <th>pct_counts_mt</th>\n",
              "      <th>pass_guide_filter</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACATGTT-HCT116_Batch1</th>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>2</td>\n",
              "      <td>ST14_P1P2-1|ST14_P1P2-2</td>\n",
              "      <td>ST14</td>\n",
              "      <td>4883</td>\n",
              "      <td>19136.0</td>\n",
              "      <td>1179.0</td>\n",
              "      <td>6.161162</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACCCAAC-HCT116_Batch1</th>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>2</td>\n",
              "      <td>SIGLEC5_P1P2-1|SIGLEC5_P1P2-2</td>\n",
              "      <td>SIGLEC5</td>\n",
              "      <td>8130</td>\n",
              "      <td>47916.0</td>\n",
              "      <td>1562.0</td>\n",
              "      <td>3.259871</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGGTACG-HCT116_Batch1</th>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>2</td>\n",
              "      <td>VSNL1_P1P2-1|VSNL1_P1P2-2</td>\n",
              "      <td>VSNL1</td>\n",
              "      <td>6531</td>\n",
              "      <td>28435.0</td>\n",
              "      <td>1042.0</td>\n",
              "      <td>3.664498</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGCGATTAT-HCT116_Batch1</th>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>2</td>\n",
              "      <td>KCNK7_P1P2-1|KCNK7_P1P2-2</td>\n",
              "      <td>KCNK7</td>\n",
              "      <td>5931</td>\n",
              "      <td>26080.0</td>\n",
              "      <td>1087.0</td>\n",
              "      <td>4.167945</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGGCTTAAT-HCT116_Batch1</th>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>2</td>\n",
              "      <td>APOA4_P1P2-1|APOA4_P1P2-2</td>\n",
              "      <td>APOA4</td>\n",
              "      <td>7157</td>\n",
              "      <td>38366.0</td>\n",
              "      <td>955.0</td>\n",
              "      <td>2.489183</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGCAGGGAATCG-HCT116_Batch109</th>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>2</td>\n",
              "      <td>ACP6_P1P2-1|ACP6_P1P2-2</td>\n",
              "      <td>ACP6</td>\n",
              "      <td>5107</td>\n",
              "      <td>17882.0</td>\n",
              "      <td>767.0</td>\n",
              "      <td>4.289229</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCACGCGG-HCT116_Batch109</th>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>2</td>\n",
              "      <td>RNF39_P1P2-1|RNF39_P1P2-2</td>\n",
              "      <td>RNF39</td>\n",
              "      <td>3643</td>\n",
              "      <td>8155.0</td>\n",
              "      <td>664.0</td>\n",
              "      <td>8.142244</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCTGACTC-HCT116_Batch109</th>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>2</td>\n",
              "      <td>AFP_P2-1|AFP_P2-2</td>\n",
              "      <td>AFP</td>\n",
              "      <td>2831</td>\n",
              "      <td>6334.0</td>\n",
              "      <td>1553.0</td>\n",
              "      <td>24.518473</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCTTGATT-HCT116_Batch109</th>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>2</td>\n",
              "      <td>METTL9_P1P2-1|METTL9_P1P2-2</td>\n",
              "      <td>METTL9</td>\n",
              "      <td>3868</td>\n",
              "      <td>8766.0</td>\n",
              "      <td>868.0</td>\n",
              "      <td>9.901894</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTGTAGTTA-HCT116_Batch109</th>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>2</td>\n",
              "      <td>SLC5A10_P1P2-1|SLC5A10_P1P2-2</td>\n",
              "      <td>SLC5A10</td>\n",
              "      <td>4095</td>\n",
              "      <td>9686.0</td>\n",
              "      <td>738.0</td>\n",
              "      <td>7.619245</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3409169 rows × 9 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-aa39e725-d7ca-40cc-9c8d-72d20ba48674')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-aa39e725-d7ca-40cc-9c8d-72d20ba48674 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-aa39e725-d7ca-40cc-9c8d-72d20ba48674');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-8cbb205b-5d1f-4f08-820b-fa63b33e3c45\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-8cbb205b-5d1f-4f08-820b-fa63b33e3c45')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-8cbb205b-5d1f-4f08-820b-fa63b33e3c45 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            }
          },
          "metadata": {},
          "execution_count": 25
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "aafa2dc5",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aafa2dc5",
        "outputId": "2f6afa3d-3f32-4f3c-b64e-1e72d6c1cb06"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Variable data:\n",
            "DataFrame shape: (38606, 6)\n",
            "--------------------------------------------------\n",
            "                    mt  n_cells_by_counts   mean_counts  \\\n",
            "DDX11L2          False               2719  2.777434e-04   \n",
            "MIR1302-2HG      False               2620  2.673255e-04   \n",
            "FAM138A          False                  3  3.034341e-07   \n",
            "ENSG00000290826  False                764  8.698444e-05   \n",
            "OR4F5            False                263  2.700564e-05   \n",
            "...                ...                ...           ...   \n",
            "ENSG00000277836  False                  0  0.000000e+00   \n",
            "ENSG00000278633  False                106  1.092363e-05   \n",
            "ENSG00000276017  False                663  6.756467e-05   \n",
            "ENSG00000278817  False              35369  3.616227e-03   \n",
            "ENSG00000277196  False              48372  5.025274e-03   \n",
            "\n",
            "                 pct_dropout_by_counts  total_counts      gene_symbol  \n",
            "DDX11L2                      99.972499        2746.0          DDX11L2  \n",
            "MIR1302-2HG                  99.973500        2643.0      MIR1302-2HG  \n",
            "FAM138A                      99.999970           3.0          FAM138A  \n",
            "ENSG00000290826              99.992273         860.0  ENSG00000290826  \n",
            "OR4F5                        99.997340         267.0            OR4F5  \n",
            "...                                ...           ...              ...  \n",
            "ENSG00000277836             100.000000           0.0  ENSG00000277836  \n",
            "ENSG00000278633              99.998928         108.0  ENSG00000278633  \n",
            "ENSG00000276017              99.993294         668.0  ENSG00000276017  \n",
            "ENSG00000278817              99.642261       35753.0  ENSG00000278817  \n",
            "ENSG00000277196              99.510743       49684.0  ENSG00000277196  \n",
            "\n",
            "[38606 rows x 6 columns]\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.adata.var['gene_symbol'] = cur_data.adata.var.index\n",
        "cur_data.show_var()"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "44a7835d",
      "metadata": {
        "id": "44a7835d"
      },
      "source": [
        "# OBS slot curation"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "7c2b269f",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 366
        },
        "id": "7c2b269f",
        "outputId": "1346954b-53c6-4c50-b433-01e8c57454cb"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "sample               0\n",
              "num_features         0\n",
              "guide_target         0\n",
              "gene_target          0\n",
              "n_genes_by_counts    0\n",
              "total_counts         0\n",
              "total_counts_mt      0\n",
              "pct_counts_mt        0\n",
              "pass_guide_filter    0\n",
              "dtype: int64"
            ],
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>sample</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>num_features</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>guide_target</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>gene_target</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>n_genes_by_counts</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>total_counts</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>total_counts_mt</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>pct_counts_mt</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>pass_guide_filter</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div><br><label><b>dtype:</b> int64</label>"
            ]
          },
          "metadata": {},
          "execution_count": 27
        }
      ],
      "source": [
        "cur_data.adata.obs.isna().sum()"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "57acacbc",
      "metadata": {
        "id": "57acacbc"
      },
      "source": [
        "### Extract perturbation symbols"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "a26c14a6",
      "metadata": {
        "id": "a26c14a6"
      },
      "source": [
        "### Rename `guide_target` to `perturbation_name`"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "0566e196",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0566e196",
        "outputId": "301aeb0a-5215-4103-aa09-222714b29a5b"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Renamed columns in adata.obs: {'guide_target': 'perturbation_name'}\n"
          ]
        }
      ],
      "source": [
        "cur_data.rename_columns(slot = 'obs', name_dict = {'guide_target': 'perturbation_name'})"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "b3deae0f",
      "metadata": {
        "id": "b3deae0f"
      },
      "source": [
        "### Rename `gene_target` to `perturbed_target_symbol`"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "10a4deb2",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "10a4deb2",
        "outputId": "b2a0ae8c-a848-4792-b3f9-ad45861ba68a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Renamed columns in adata.obs: {'gene_target': 'perturbed_target_symbol'}\n"
          ]
        }
      ],
      "source": [
        "cur_data.rename_columns(slot = 'obs', name_dict = {'gene_target': 'perturbed_target_symbol'})"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "b9b45777",
      "metadata": {
        "id": "b9b45777"
      },
      "source": [
        "#### Clean up `perturbed_target_symbol` column"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "c5d8cd5e",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c5d8cd5e",
        "outputId": "d2112f72-dfad-4935-9d70-cecb1f93b34a",
        "collapsed": true
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[1;30;43mStreaming output truncated to the last 5000 lines.\u001b[0m\n",
            " 'RP1L1',\n",
            " 'RP2',\n",
            " 'RP9',\n",
            " 'RPA1',\n",
            " 'RPA2',\n",
            " 'RPA3',\n",
            " 'RPA4',\n",
            " 'RPAIN',\n",
            " 'RPAP1',\n",
            " 'RPAP2',\n",
            " 'RPAP3',\n",
            " 'RPE',\n",
            " 'RPE65',\n",
            " 'RPF1',\n",
            " 'RPF2',\n",
            " 'RPGR',\n",
            " 'RPGRIP1',\n",
            " 'RPGRIP1L',\n",
            " 'RPH3A',\n",
            " 'RPH3AL',\n",
            " 'RPIA',\n",
            " 'RPL10',\n",
            " 'RPL10A',\n",
            " 'RPL10L',\n",
            " 'RPL11',\n",
            " 'RPL12',\n",
            " 'RPL13',\n",
            " 'RPL13A',\n",
            " 'RPL14',\n",
            " 'RPL15',\n",
            " 'RPL17',\n",
            " 'RPL18',\n",
            " 'RPL18A',\n",
            " 'RPL19',\n",
            " 'RPL21',\n",
            " 'RPL22',\n",
            " 'RPL22L1',\n",
            " 'RPL23',\n",
            " 'RPL23A',\n",
            " 'RPL24',\n",
            " 'RPL26',\n",
            " 'RPL26L1',\n",
            " 'RPL27',\n",
            " 'RPL27A',\n",
            " 'RPL28',\n",
            " 'RPL29',\n",
            " 'RPL3',\n",
            " 'RPL30',\n",
            " 'RPL31',\n",
            " 'RPL32',\n",
            " 'RPL34',\n",
            " 'RPL35',\n",
            " 'RPL35A',\n",
            " 'RPL36',\n",
            " 'RPL36A',\n",
            " 'RPL36AL',\n",
            " 'RPL37',\n",
            " 'RPL37A',\n",
            " 'RPL38',\n",
            " 'RPL39',\n",
            " 'RPL39L',\n",
            " 'RPL3L',\n",
            " 'RPL4',\n",
            " 'RPL41',\n",
            " 'RPL5',\n",
            " 'RPL6',\n",
            " 'RPL7',\n",
            " 'RPL7A',\n",
            " 'RPL7L1',\n",
            " 'RPL8',\n",
            " 'RPL9',\n",
            " 'RPLP0',\n",
            " 'RPLP1',\n",
            " 'RPLP2',\n",
            " 'RPN1',\n",
            " 'RPN2',\n",
            " 'RPP14',\n",
            " 'RPP21',\n",
            " 'RPP25',\n",
            " 'RPP25L',\n",
            " 'RPP30',\n",
            " 'RPP40',\n",
            " 'RPRD1A',\n",
            " 'RPRD1B',\n",
            " 'RPRD2',\n",
            " 'RPRM',\n",
            " 'RPRML',\n",
            " 'RPS10',\n",
            " 'RPS11',\n",
            " 'RPS12',\n",
            " 'RPS13',\n",
            " 'RPS14',\n",
            " 'RPS15',\n",
            " 'RPS15A',\n",
            " 'RPS16',\n",
            " 'RPS17',\n",
            " 'RPS18',\n",
            " 'RPS19',\n",
            " 'RPS19BP1',\n",
            " 'RPS2',\n",
            " 'RPS20',\n",
            " 'RPS21',\n",
            " 'RPS23',\n",
            " 'RPS24',\n",
            " 'RPS25',\n",
            " 'RPS26',\n",
            " 'RPS27',\n",
            " 'RPS27A',\n",
            " 'RPS27L',\n",
            " 'RPS28',\n",
            " 'RPS29',\n",
            " 'RPS3',\n",
            " 'RPS3A',\n",
            " 'RPS4X',\n",
            " 'RPS4Y1',\n",
            " 'RPS4Y2',\n",
            " 'RPS5',\n",
            " 'RPS6',\n",
            " 'RPS6KA1',\n",
            " 'RPS6KA2',\n",
            " 'RPS6KA3',\n",
            " 'RPS6KA4',\n",
            " 'RPS6KA5',\n",
            " 'RPS6KA6',\n",
            " 'RPS6KB1',\n",
            " 'RPS6KB2',\n",
            " 'RPS6KC1',\n",
            " 'RPS6KL1',\n",
            " 'RPS7',\n",
            " 'RPS8',\n",
            " 'RPS9',\n",
            " 'RPSA',\n",
            " 'RPTN',\n",
            " 'RPTOR',\n",
            " 'RPUSD1',\n",
            " 'RPUSD2',\n",
            " 'RPUSD3',\n",
            " 'RRAD',\n",
            " 'RRAGA',\n",
            " 'RRAGB',\n",
            " 'RRAGC',\n",
            " 'RRAGD',\n",
            " 'RRAS',\n",
            " 'RRAS2',\n",
            " 'RRBP1',\n",
            " 'RREB1',\n",
            " 'RRH',\n",
            " 'RRM1',\n",
            " 'RRM2',\n",
            " 'RRM2B',\n",
            " 'RRN3',\n",
            " 'RRP1',\n",
            " 'RRP12',\n",
            " 'RRP15',\n",
            " 'RRP1B',\n",
            " 'RRP36',\n",
            " 'RRP7A',\n",
            " 'RRP8',\n",
            " 'RRP9',\n",
            " 'RRS1',\n",
            " 'RS1',\n",
            " 'RSAD1',\n",
            " 'RSAD2',\n",
            " 'RSBN1',\n",
            " 'RSBN1L',\n",
            " 'RSC1A1',\n",
            " 'RSF1',\n",
            " 'RSL1D1',\n",
            " 'RSL24D1',\n",
            " 'RSPH1',\n",
            " 'RSPH10B',\n",
            " 'RSPH10B2',\n",
            " 'RSPH14',\n",
            " 'RSPH3',\n",
            " 'RSPH4A',\n",
            " 'RSPH6A',\n",
            " 'RSPH9',\n",
            " 'RSPO1',\n",
            " 'RSPO2',\n",
            " 'RSPO3',\n",
            " 'RSPO4',\n",
            " 'RSPRY1',\n",
            " 'RSRC1',\n",
            " 'RSRC2',\n",
            " 'RSRP1',\n",
            " 'RSU1',\n",
            " 'RTBDN',\n",
            " 'RTCA',\n",
            " 'RTCB',\n",
            " 'RTEL1',\n",
            " 'RTF1',\n",
            " 'RTF2',\n",
            " 'RTKN',\n",
            " 'RTKN2',\n",
            " 'RTL1',\n",
            " 'RTL10',\n",
            " 'RTL3',\n",
            " 'RTL4',\n",
            " 'RTL5',\n",
            " 'RTL6',\n",
            " 'RTL8A',\n",
            " 'RTL8B',\n",
            " 'RTL8C',\n",
            " 'RTL9',\n",
            " 'RTN1',\n",
            " 'RTN2',\n",
            " 'RTN3',\n",
            " 'RTN4',\n",
            " 'RTN4IP1',\n",
            " 'RTN4R',\n",
            " 'RTN4RL1',\n",
            " 'RTN4RL2',\n",
            " 'RTP1',\n",
            " 'RTP2',\n",
            " 'RTP3',\n",
            " 'RTP4',\n",
            " 'RTP5',\n",
            " 'RTRAF',\n",
            " 'RTTN',\n",
            " 'RUBCN',\n",
            " 'RUBCNL',\n",
            " 'RUFY1',\n",
            " 'RUFY2',\n",
            " 'RUFY3',\n",
            " 'RUFY4',\n",
            " 'RUNDC1',\n",
            " 'RUNDC3A',\n",
            " 'RUNDC3B',\n",
            " 'RUNX1',\n",
            " 'RUNX1T1',\n",
            " 'RUNX2',\n",
            " 'RUNX3',\n",
            " 'RUSC1',\n",
            " 'RUSC1-AS1',\n",
            " 'RUSC2',\n",
            " 'RUSF1',\n",
            " 'RUVBL1',\n",
            " 'RUVBL2',\n",
            " 'RWDD1',\n",
            " 'RWDD2A',\n",
            " 'RWDD2B',\n",
            " 'RWDD3',\n",
            " 'RWDD4',\n",
            " 'RXFP1',\n",
            " 'RXFP2',\n",
            " 'RXFP3',\n",
            " 'RXFP4',\n",
            " 'RXRA',\n",
            " 'RXRB',\n",
            " 'RXRG',\n",
            " 'RXYLT1',\n",
            " 'RYBP',\n",
            " 'RYK',\n",
            " 'RYR1',\n",
            " 'RYR2',\n",
            " 'RYR3',\n",
            " 'S100A1',\n",
            " 'S100A10',\n",
            " 'S100A11',\n",
            " 'S100A12',\n",
            " 'S100A13',\n",
            " 'S100A14',\n",
            " 'S100A16',\n",
            " 'S100A2',\n",
            " 'S100A3',\n",
            " 'S100A4',\n",
            " 'S100A5',\n",
            " 'S100A6',\n",
            " 'S100A7',\n",
            " 'S100A7A',\n",
            " 'S100A8',\n",
            " 'S100A9',\n",
            " 'S100B',\n",
            " 'S100G',\n",
            " 'S100P',\n",
            " 'S100PBP',\n",
            " 'S100Z',\n",
            " 'S1PR1',\n",
            " 'S1PR2',\n",
            " 'S1PR3',\n",
            " 'S1PR4',\n",
            " 'S1PR5',\n",
            " 'SAA1',\n",
            " 'SAA2',\n",
            " 'SAA4',\n",
            " 'SAAL1',\n",
            " 'SAC3D1',\n",
            " 'SACM1L',\n",
            " 'SACS',\n",
            " 'SAE1',\n",
            " 'SAFB',\n",
            " 'SAFB2',\n",
            " 'SAG',\n",
            " 'SAGE1',\n",
            " 'SALL1',\n",
            " 'SALL2',\n",
            " 'SALL3',\n",
            " 'SALL4',\n",
            " 'SAMD1',\n",
            " 'SAMD10',\n",
            " 'SAMD11',\n",
            " 'SAMD12',\n",
            " 'SAMD13',\n",
            " 'SAMD14',\n",
            " 'SAMD15',\n",
            " 'SAMD3',\n",
            " 'SAMD4A',\n",
            " 'SAMD4B',\n",
            " 'SAMD5',\n",
            " 'SAMD7',\n",
            " 'SAMD8',\n",
            " 'SAMD9',\n",
            " 'SAMD9L',\n",
            " 'SAMHD1',\n",
            " 'SAMM50',\n",
            " 'SAMSN1',\n",
            " 'SANBR',\n",
            " 'SAP130',\n",
            " 'SAP18',\n",
            " 'SAP25',\n",
            " 'SAP30',\n",
            " 'SAP30BP',\n",
            " 'SAP30L',\n",
            " 'SAPCD1',\n",
            " 'SAPCD2',\n",
            " 'SAR1A',\n",
            " 'SAR1B',\n",
            " 'SARAF',\n",
            " 'SARDH',\n",
            " 'SARM1',\n",
            " 'SARNP',\n",
            " 'SARS1',\n",
            " 'SARS2',\n",
            " 'SART1',\n",
            " 'SART3',\n",
            " 'SASH1',\n",
            " 'SASH3',\n",
            " 'SASS6',\n",
            " 'SAT1',\n",
            " 'SAT2',\n",
            " 'SATB1',\n",
            " 'SATB2',\n",
            " 'SATL1',\n",
            " 'SAV1',\n",
            " 'SAXO1',\n",
            " 'SAXO2',\n",
            " 'SAYSD1',\n",
            " 'SBDS',\n",
            " 'SBF1',\n",
            " 'SBF2',\n",
            " 'SBK1',\n",
            " 'SBK2',\n",
            " 'SBNO1',\n",
            " 'SBNO2',\n",
            " 'SBSN',\n",
            " 'SBSPON',\n",
            " 'SC5D',\n",
            " 'SCAF1',\n",
            " 'SCAF11',\n",
            " 'SCAF4',\n",
            " 'SCAF8',\n",
            " 'SCAI',\n",
            " 'SCAMP1',\n",
            " 'SCAMP2',\n",
            " 'SCAMP3',\n",
            " 'SCAMP4',\n",
            " 'SCAMP5',\n",
            " 'SCAND1',\n",
            " 'SCAND3',\n",
            " 'SCAP',\n",
            " 'SCAPER',\n",
            " 'SCARA3',\n",
            " 'SCARA5',\n",
            " 'SCARB1',\n",
            " 'SCARB2',\n",
            " 'SCARF1',\n",
            " 'SCARF2',\n",
            " 'SCCPDH',\n",
            " 'SCD',\n",
            " 'SCD5',\n",
            " 'SCEL',\n",
            " 'SCFD1',\n",
            " 'SCFD2',\n",
            " 'SCG2',\n",
            " 'SCG3',\n",
            " 'SCG5',\n",
            " 'SCGB1A1',\n",
            " 'SCGB1C1',\n",
            " 'SCGB1D1',\n",
            " 'SCGB1D2',\n",
            " 'SCGB1D4',\n",
            " 'SCGB2A1',\n",
            " 'SCGB2A2',\n",
            " 'SCGB2B2',\n",
            " 'SCGB3A1',\n",
            " 'SCGB3A2',\n",
            " 'SCGN',\n",
            " 'SCHIP1',\n",
            " 'SCIMP',\n",
            " 'SCIN',\n",
            " 'SCLT1',\n",
            " 'SCLY',\n",
            " 'SCMH1',\n",
            " 'SCML1',\n",
            " 'SCML2',\n",
            " 'SCML4',\n",
            " 'SCN10A',\n",
            " 'SCN11A',\n",
            " 'SCN1A',\n",
            " 'SCN1B',\n",
            " 'SCN2A',\n",
            " 'SCN2B',\n",
            " 'SCN3A',\n",
            " 'SCN3B',\n",
            " 'SCN4A',\n",
            " 'SCN4B',\n",
            " 'SCN5A',\n",
            " 'SCN7A',\n",
            " 'SCN8A',\n",
            " 'SCN9A',\n",
            " 'SCNM1',\n",
            " 'SCNN1A',\n",
            " 'SCNN1B',\n",
            " 'SCNN1D',\n",
            " 'SCNN1G',\n",
            " 'SCO1',\n",
            " 'SCO2',\n",
            " 'SCOC',\n",
            " 'SCP2',\n",
            " 'SCP2D1',\n",
            " 'SCP2D1-AS1',\n",
            " 'SCPEP1',\n",
            " 'SCRG1',\n",
            " 'SCRIB',\n",
            " 'SCRN1',\n",
            " 'SCRN2',\n",
            " 'SCRN3',\n",
            " 'SCRT1',\n",
            " 'SCRT2',\n",
            " 'SCT',\n",
            " 'SCTR',\n",
            " 'SCUBE1',\n",
            " 'SCUBE2',\n",
            " 'SCUBE3',\n",
            " 'SCX',\n",
            " 'SCYL1',\n",
            " 'SCYL2',\n",
            " 'SCYL3',\n",
            " 'SDAD1',\n",
            " 'SDC1',\n",
            " 'SDC2',\n",
            " 'SDC3',\n",
            " 'SDC4',\n",
            " 'SDCBP',\n",
            " 'SDCBP2',\n",
            " 'SDCCAG8',\n",
            " 'SDE2',\n",
            " 'SDF2',\n",
            " 'SDF2L1',\n",
            " 'SDF4',\n",
            " 'SDHA',\n",
            " 'SDHAF1',\n",
            " 'SDHAF2',\n",
            " 'SDHAF3',\n",
            " 'SDHAF4',\n",
            " 'SDHB',\n",
            " 'SDHC',\n",
            " 'SDHD',\n",
            " 'SDK1',\n",
            " 'SDK2',\n",
            " 'SDR16C5',\n",
            " 'SDR39U1',\n",
            " 'SDR42E1',\n",
            " 'SDR42E2',\n",
            " 'SDR9C7',\n",
            " 'SDS',\n",
            " 'SDSL',\n",
            " 'SEC11A',\n",
            " 'SEC11C',\n",
            " 'SEC13',\n",
            " 'SEC14L1',\n",
            " 'SEC14L2',\n",
            " 'SEC14L3',\n",
            " 'SEC14L4',\n",
            " 'SEC14L5',\n",
            " 'SEC14L6',\n",
            " 'SEC16A',\n",
            " 'SEC16B',\n",
            " 'SEC22A',\n",
            " 'SEC22B',\n",
            " 'SEC22C',\n",
            " 'SEC23A',\n",
            " 'SEC23B',\n",
            " 'SEC23IP',\n",
            " 'SEC24A',\n",
            " 'SEC24B',\n",
            " 'SEC24C',\n",
            " 'SEC24D',\n",
            " 'SEC31A',\n",
            " 'SEC31B',\n",
            " 'SEC61A1',\n",
            " 'SEC61A2',\n",
            " 'SEC61B',\n",
            " 'SEC61G',\n",
            " 'SEC62',\n",
            " 'SEC63',\n",
            " 'SECISBP2',\n",
            " 'SECISBP2L',\n",
            " 'SECTM1',\n",
            " 'SEH1L',\n",
            " 'SEL1L',\n",
            " 'SEL1L2',\n",
            " 'SEL1L3',\n",
            " 'SELE',\n",
            " 'SELENBP1',\n",
            " 'SELENOF',\n",
            " 'SELENOH',\n",
            " 'SELENOI',\n",
            " 'SELENOK',\n",
            " 'SELENOM',\n",
            " 'SELENON',\n",
            " 'SELENOO',\n",
            " 'SELENOP',\n",
            " 'SELENOS',\n",
            " 'SELENOT',\n",
            " 'SELENOV',\n",
            " 'SELENOW',\n",
            " 'SELL',\n",
            " 'SELP',\n",
            " 'SELPLG',\n",
            " 'SEM1',\n",
            " 'SEMA3A',\n",
            " 'SEMA3B',\n",
            " 'SEMA3C',\n",
            " 'SEMA3D',\n",
            " 'SEMA3E',\n",
            " 'SEMA3F',\n",
            " 'SEMA3G',\n",
            " 'SEMA4A',\n",
            " 'SEMA4B',\n",
            " 'SEMA4C',\n",
            " 'SEMA4D',\n",
            " 'SEMA4F',\n",
            " 'SEMA4G',\n",
            " 'SEMA5A',\n",
            " 'SEMA5B',\n",
            " 'SEMA6A',\n",
            " 'SEMA6B',\n",
            " 'SEMA6C',\n",
            " 'SEMA6D',\n",
            " 'SEMA7A',\n",
            " 'SENP1',\n",
            " 'SENP2',\n",
            " 'SENP3',\n",
            " 'SENP5',\n",
            " 'SENP6',\n",
            " 'SENP7',\n",
            " 'SENP8',\n",
            " 'SEPHS1',\n",
            " 'SEPHS2',\n",
            " 'SEPSECS',\n",
            " 'SEPTIN1',\n",
            " 'SEPTIN10',\n",
            " 'SEPTIN11',\n",
            " 'SEPTIN12',\n",
            " 'SEPTIN14',\n",
            " 'SEPTIN2',\n",
            " 'SEPTIN3',\n",
            " 'SEPTIN4',\n",
            " 'SEPTIN5',\n",
            " 'SEPTIN6',\n",
            " 'SEPTIN7',\n",
            " 'SEPTIN8',\n",
            " 'SEPTIN9',\n",
            " 'SERAC1',\n",
            " 'SERBP1',\n",
            " 'SERF2',\n",
            " 'SERGEF',\n",
            " 'SERHL2',\n",
            " 'SERINC1',\n",
            " 'SERINC2',\n",
            " 'SERINC3',\n",
            " 'SERINC4',\n",
            " 'SERINC5',\n",
            " 'SERP1',\n",
            " 'SERP2',\n",
            " 'SERPINA1',\n",
            " 'SERPINA10',\n",
            " 'SERPINA11',\n",
            " 'SERPINA12',\n",
            " 'SERPINA3',\n",
            " 'SERPINA4',\n",
            " 'SERPINA5',\n",
            " 'SERPINA6',\n",
            " 'SERPINA7',\n",
            " 'SERPINA9',\n",
            " 'SERPINB1',\n",
            " 'SERPINB10',\n",
            " 'SERPINB11',\n",
            " 'SERPINB12',\n",
            " 'SERPINB13',\n",
            " 'SERPINB2',\n",
            " 'SERPINB3',\n",
            " 'SERPINB4',\n",
            " 'SERPINB5',\n",
            " 'SERPINB6',\n",
            " 'SERPINB7',\n",
            " 'SERPINB8',\n",
            " 'SERPINB9',\n",
            " 'SERPINC1',\n",
            " 'SERPIND1',\n",
            " 'SERPINE1',\n",
            " 'SERPINE2',\n",
            " 'SERPINE3',\n",
            " 'SERPINF1',\n",
            " 'SERPINF2',\n",
            " 'SERPING1',\n",
            " 'SERPINH1',\n",
            " 'SERPINI1',\n",
            " 'SERPINI2',\n",
            " 'SERTAD1',\n",
            " 'SERTAD2',\n",
            " 'SERTAD3',\n",
            " 'SERTAD4',\n",
            " 'SERTM1',\n",
            " 'SESN1',\n",
            " 'SESN2',\n",
            " 'SESN3',\n",
            " 'SESTD1',\n",
            " 'SET',\n",
            " 'SETBP1',\n",
            " 'SETD1A',\n",
            " 'SETD1B',\n",
            " 'SETD2',\n",
            " 'SETD3',\n",
            " 'SETD4',\n",
            " 'SETD5',\n",
            " 'SETD6',\n",
            " 'SETD7',\n",
            " 'SETD9',\n",
            " 'SETDB1',\n",
            " 'SETDB2',\n",
            " 'SETMAR',\n",
            " 'SETX',\n",
            " 'SEZ6',\n",
            " 'SEZ6L',\n",
            " 'SEZ6L2',\n",
            " 'SF1',\n",
            " 'SF3A1',\n",
            " 'SF3A2',\n",
            " 'SF3A3',\n",
            " 'SF3B1',\n",
            " 'SF3B2',\n",
            " 'SF3B3',\n",
            " 'SF3B4',\n",
            " 'SF3B5',\n",
            " 'SFI1',\n",
            " 'SFMBT1',\n",
            " 'SFMBT2',\n",
            " 'SFN',\n",
            " 'SFPQ',\n",
            " 'SFR1',\n",
            " 'SFRP1',\n",
            " 'SFRP2',\n",
            " 'SFRP4',\n",
            " 'SFRP5',\n",
            " 'SFSWAP',\n",
            " 'SFT2D1',\n",
            " 'SFT2D2',\n",
            " 'SFT2D3',\n",
            " 'SFTA2',\n",
            " 'SFTA3',\n",
            " 'SFTPA1',\n",
            " 'SFTPA2',\n",
            " 'SFTPB',\n",
            " 'SFTPC',\n",
            " 'SFTPD',\n",
            " 'SFXN1',\n",
            " 'SFXN2',\n",
            " 'SFXN3',\n",
            " 'SFXN4',\n",
            " 'SFXN5',\n",
            " 'SGCA',\n",
            " 'SGCB',\n",
            " 'SGCE',\n",
            " 'SGCG',\n",
            " 'SGCZ',\n",
            " 'SGF29',\n",
            " 'SGIP1',\n",
            " 'SGK1',\n",
            " 'SGK2',\n",
            " 'SGK3',\n",
            " 'SGMS1',\n",
            " 'SGMS2',\n",
            " 'SGO1',\n",
            " 'SGO2',\n",
            " 'SGPL1',\n",
            " 'SGPP1',\n",
            " 'SGPP2',\n",
            " 'SGSH',\n",
            " 'SGSM1',\n",
            " 'SGSM2',\n",
            " 'SGSM3',\n",
            " 'SGTA',\n",
            " 'SGTB',\n",
            " 'SH2B1',\n",
            " 'SH2B2',\n",
            " 'SH2B3',\n",
            " 'SH2D1A',\n",
            " 'SH2D1B',\n",
            " 'SH2D2A',\n",
            " 'SH2D3A',\n",
            " 'SH2D3C',\n",
            " 'SH2D4A',\n",
            " 'SH2D4B',\n",
            " 'SH2D5',\n",
            " 'SH2D6',\n",
            " 'SH2D7',\n",
            " 'SH3BGR',\n",
            " 'SH3BGRL',\n",
            " 'SH3BGRL2',\n",
            " 'SH3BGRL3',\n",
            " 'SH3BP1',\n",
            " 'SH3BP2',\n",
            " 'SH3BP4',\n",
            " 'SH3BP5',\n",
            " 'SH3BP5L',\n",
            " 'SH3D19',\n",
            " 'SH3GL1',\n",
            " 'SH3GL2',\n",
            " 'SH3GL3',\n",
            " 'SH3GLB1',\n",
            " 'SH3GLB2',\n",
            " 'SH3KBP1',\n",
            " 'SH3PXD2A',\n",
            " 'SH3PXD2B',\n",
            " 'SH3RF1',\n",
            " 'SH3RF2',\n",
            " 'SH3RF3',\n",
            " 'SH3TC1',\n",
            " 'SH3TC2',\n",
            " 'SH3YL1',\n",
            " 'SHANK1',\n",
            " 'SHANK2',\n",
            " 'SHANK3',\n",
            " 'SHARPIN',\n",
            " 'SHB',\n",
            " 'SHBG',\n",
            " 'SHC1',\n",
            " 'SHC2',\n",
            " 'SHC3',\n",
            " 'SHC4',\n",
            " 'SHCBP1',\n",
            " 'SHCBP1L',\n",
            " 'SHD',\n",
            " 'SHF',\n",
            " 'SHFL',\n",
            " 'SHH',\n",
            " 'SHISA2',\n",
            " 'SHISA3',\n",
            " 'SHISA4',\n",
            " 'SHISA5',\n",
            " 'SHISA6',\n",
            " 'SHISA7',\n",
            " 'SHISA8',\n",
            " 'SHISA9',\n",
            " 'SHISAL1',\n",
            " 'SHISAL2A',\n",
            " 'SHISAL2B',\n",
            " 'SHKBP1',\n",
            " 'SHLD1',\n",
            " 'SHLD2',\n",
            " 'SHMT1',\n",
            " 'SHMT2',\n",
            " 'SHOC1',\n",
            " 'SHOC2',\n",
            " 'SHOX',\n",
            " 'SHOX2',\n",
            " 'SHPK',\n",
            " 'SHPRH',\n",
            " 'SHQ1',\n",
            " 'SHROOM1',\n",
            " 'SHROOM2',\n",
            " 'SHROOM3',\n",
            " 'SHROOM4',\n",
            " 'SHTN1',\n",
            " 'SI',\n",
            " 'SIAE',\n",
            " 'SIAH1',\n",
            " 'SIAH3',\n",
            " 'SIDT1',\n",
            " 'SIDT2',\n",
            " 'SIGIRR',\n",
            " 'SIGLEC1',\n",
            " 'SIGLEC10',\n",
            " 'SIGLEC11',\n",
            " 'SIGLEC12',\n",
            " 'SIGLEC14',\n",
            " 'SIGLEC15',\n",
            " 'SIGLEC5',\n",
            " 'SIGLEC6',\n",
            " 'SIGLEC7',\n",
            " 'SIGLEC8',\n",
            " 'SIGLEC9',\n",
            " 'SIGLECL1',\n",
            " 'SIGMAR1',\n",
            " 'SIK1',\n",
            " 'SIK2',\n",
            " 'SIK3',\n",
            " 'SIKE1',\n",
            " 'SIL1',\n",
            " 'SIM1',\n",
            " 'SIM2',\n",
            " 'SIMC1',\n",
            " 'SIN3A',\n",
            " 'SIN3B',\n",
            " 'SINHCAF',\n",
            " 'SIPA1',\n",
            " 'SIPA1L1',\n",
            " 'SIPA1L2',\n",
            " 'SIPA1L3',\n",
            " 'SIRPA',\n",
            " 'SIRPB1',\n",
            " 'SIRPB2',\n",
            " 'SIRPD',\n",
            " 'SIRPG',\n",
            " 'SIRT1',\n",
            " 'SIRT2',\n",
            " 'SIRT3',\n",
            " 'SIRT4',\n",
            " 'SIRT5',\n",
            " 'SIRT6',\n",
            " 'SIRT7',\n",
            " 'SIT1',\n",
            " 'SIVA1',\n",
            " 'SIX1',\n",
            " 'SIX2',\n",
            " 'SIX3',\n",
            " 'SIX4',\n",
            " 'SIX5',\n",
            " 'SIX6',\n",
            " 'SKA1',\n",
            " 'SKA2',\n",
            " 'SKA3',\n",
            " 'SKAP1',\n",
            " 'SKAP2',\n",
            " 'SKI',\n",
            " 'SKIC2',\n",
            " 'SKIC3',\n",
            " 'SKIC8',\n",
            " 'SKIDA1',\n",
            " 'SKIL',\n",
            " 'SKOR1',\n",
            " 'SKOR2',\n",
            " 'SKP1',\n",
            " 'SKP2',\n",
            " 'SLA',\n",
            " 'SLA2',\n",
            " 'SLAIN1',\n",
            " 'SLAIN2',\n",
            " 'SLAMF1',\n",
            " 'SLAMF6',\n",
            " 'SLAMF7',\n",
            " 'SLAMF8',\n",
            " 'SLAMF9',\n",
            " 'SLBP',\n",
            " 'SLC10A1',\n",
            " 'SLC10A2',\n",
            " 'SLC10A3',\n",
            " 'SLC10A4',\n",
            " 'SLC10A5',\n",
            " 'SLC10A6',\n",
            " 'SLC10A7',\n",
            " 'SLC11A1',\n",
            " 'SLC11A2',\n",
            " 'SLC12A1',\n",
            " 'SLC12A2',\n",
            " 'SLC12A3',\n",
            " 'SLC12A4',\n",
            " 'SLC12A5',\n",
            " 'SLC12A6',\n",
            " 'SLC12A7',\n",
            " 'SLC12A8',\n",
            " 'SLC12A9',\n",
            " 'SLC13A1',\n",
            " 'SLC13A2',\n",
            " 'SLC13A3',\n",
            " 'SLC13A4',\n",
            " 'SLC13A5',\n",
            " 'SLC14A1',\n",
            " 'SLC14A2',\n",
            " 'SLC15A1',\n",
            " 'SLC15A2',\n",
            " 'SLC15A3',\n",
            " 'SLC15A4',\n",
            " 'SLC15A5',\n",
            " 'SLC16A1',\n",
            " 'SLC16A10',\n",
            " 'SLC16A11',\n",
            " 'SLC16A12',\n",
            " 'SLC16A13',\n",
            " 'SLC16A14',\n",
            " 'SLC16A2',\n",
            " 'SLC16A3',\n",
            " 'SLC16A4',\n",
            " 'SLC16A5',\n",
            " 'SLC16A6',\n",
            " 'SLC16A7',\n",
            " 'SLC16A8',\n",
            " 'SLC16A9',\n",
            " 'SLC17A1',\n",
            " 'SLC17A2',\n",
            " 'SLC17A3',\n",
            " 'SLC17A4',\n",
            " 'SLC17A5',\n",
            " 'SLC17A6',\n",
            " 'SLC17A7',\n",
            " 'SLC17A8',\n",
            " 'SLC18A1',\n",
            " 'SLC18A2',\n",
            " 'SLC18A3',\n",
            " 'SLC18B1',\n",
            " 'SLC19A1',\n",
            " 'SLC19A2',\n",
            " 'SLC19A3',\n",
            " 'SLC1A1',\n",
            " 'SLC1A2',\n",
            " 'SLC1A3',\n",
            " 'SLC1A4',\n",
            " 'SLC1A5',\n",
            " 'SLC1A6',\n",
            " 'SLC1A7',\n",
            " 'SLC20A1',\n",
            " 'SLC22A1',\n",
            " 'SLC22A10',\n",
            " 'SLC22A11',\n",
            " 'SLC22A12',\n",
            " 'SLC22A13',\n",
            " 'SLC22A14',\n",
            " 'SLC22A15',\n",
            " 'SLC22A16',\n",
            " 'SLC22A17',\n",
            " 'SLC22A18',\n",
            " 'SLC22A18AS',\n",
            " 'SLC22A2',\n",
            " 'SLC22A23',\n",
            " 'SLC22A24',\n",
            " 'SLC22A25',\n",
            " 'SLC22A3',\n",
            " 'SLC22A31',\n",
            " 'SLC22A4',\n",
            " 'SLC22A5',\n",
            " 'SLC22A6',\n",
            " 'SLC22A7',\n",
            " 'SLC22A8',\n",
            " 'SLC22A9',\n",
            " 'SLC23A1',\n",
            " 'SLC23A2',\n",
            " 'SLC23A3',\n",
            " 'SLC24A1',\n",
            " 'SLC24A2',\n",
            " 'SLC24A3',\n",
            " 'SLC24A4',\n",
            " 'SLC24A5',\n",
            " 'SLC25A1',\n",
            " 'SLC25A11',\n",
            " 'SLC25A12',\n",
            " 'SLC25A13',\n",
            " 'SLC25A14',\n",
            " 'SLC25A15',\n",
            " 'SLC25A16',\n",
            " 'SLC25A17',\n",
            " 'SLC25A18',\n",
            " 'SLC25A19',\n",
            " 'SLC25A2',\n",
            " 'SLC25A20',\n",
            " 'SLC25A21',\n",
            " 'SLC25A22',\n",
            " 'SLC25A23',\n",
            " 'SLC25A24',\n",
            " 'SLC25A25',\n",
            " 'SLC25A26',\n",
            " 'SLC25A27',\n",
            " 'SLC25A28',\n",
            " 'SLC25A29',\n",
            " 'SLC25A3',\n",
            " 'SLC25A30',\n",
            " 'SLC25A31',\n",
            " 'SLC25A32',\n",
            " 'SLC25A33',\n",
            " 'SLC25A34',\n",
            " 'SLC25A35',\n",
            " 'SLC25A36',\n",
            " 'SLC25A37',\n",
            " 'SLC25A38',\n",
            " 'SLC25A39',\n",
            " 'SLC25A4',\n",
            " 'SLC25A40',\n",
            " 'SLC25A41',\n",
            " 'SLC25A42',\n",
            " 'SLC25A43',\n",
            " 'SLC25A44',\n",
            " 'SLC25A45',\n",
            " 'SLC25A46',\n",
            " 'SLC25A47',\n",
            " 'SLC25A48',\n",
            " 'SLC25A5',\n",
            " 'SLC25A51',\n",
            " 'SLC25A52',\n",
            " 'SLC25A53',\n",
            " 'SLC25A6',\n",
            " 'SLC26A1',\n",
            " 'SLC26A11',\n",
            " 'SLC26A2',\n",
            " 'SLC26A3',\n",
            " 'SLC26A4',\n",
            " 'SLC26A5',\n",
            " 'SLC26A6',\n",
            " 'SLC26A7',\n",
            " 'SLC26A8',\n",
            " 'SLC26A9',\n",
            " 'SLC27A1',\n",
            " 'SLC27A2',\n",
            " 'SLC27A3',\n",
            " 'SLC27A4',\n",
            " 'SLC27A5',\n",
            " 'SLC27A6',\n",
            " 'SLC28A1',\n",
            " 'SLC28A2',\n",
            " 'SLC28A3',\n",
            " 'SLC29A1',\n",
            " 'SLC29A2',\n",
            " 'SLC29A3',\n",
            " 'SLC29A4',\n",
            " 'SLC2A1',\n",
            " 'SLC2A10',\n",
            " 'SLC2A11',\n",
            " 'SLC2A12',\n",
            " 'SLC2A13',\n",
            " 'SLC2A14',\n",
            " 'SLC2A2',\n",
            " 'SLC2A3',\n",
            " 'SLC2A4',\n",
            " 'SLC2A4RG',\n",
            " 'SLC2A5',\n",
            " 'SLC2A6',\n",
            " 'SLC2A7',\n",
            " 'SLC2A8',\n",
            " 'SLC2A9',\n",
            " 'SLC30A1',\n",
            " 'SLC30A10',\n",
            " 'SLC30A2',\n",
            " 'SLC30A3',\n",
            " 'SLC30A4',\n",
            " 'SLC30A5',\n",
            " 'SLC30A6',\n",
            " 'SLC30A7',\n",
            " 'SLC30A8',\n",
            " 'SLC30A9',\n",
            " 'SLC31A1',\n",
            " 'SLC31A2',\n",
            " 'SLC32A1',\n",
            " 'SLC33A1',\n",
            " 'SLC34A1',\n",
            " 'SLC34A2',\n",
            " 'SLC34A3',\n",
            " 'SLC35A1',\n",
            " 'SLC35A2',\n",
            " 'SLC35A3',\n",
            " 'SLC35A4',\n",
            " 'SLC35A5',\n",
            " 'SLC35B1',\n",
            " 'SLC35B2',\n",
            " 'SLC35B3',\n",
            " 'SLC35B4',\n",
            " 'SLC35C1',\n",
            " 'SLC35C2',\n",
            " 'SLC35D1',\n",
            " 'SLC35D2',\n",
            " 'SLC35D3',\n",
            " 'SLC35E1',\n",
            " 'SLC35E2B',\n",
            " 'SLC35E3',\n",
            " 'SLC35E4',\n",
            " 'SLC35F1',\n",
            " 'SLC35F2',\n",
            " 'SLC35F3',\n",
            " 'SLC35F4',\n",
            " 'SLC35F5',\n",
            " 'SLC35G1',\n",
            " 'SLC35G2',\n",
            " 'SLC35G3',\n",
            " 'SLC35G4',\n",
            " 'SLC35G5',\n",
            " 'SLC35G6',\n",
            " 'SLC36A1',\n",
            " 'SLC36A2',\n",
            " 'SLC36A3',\n",
            " 'SLC36A4',\n",
            " 'SLC37A1',\n",
            " 'SLC37A2',\n",
            " 'SLC37A3',\n",
            " 'SLC37A4',\n",
            " 'SLC38A1',\n",
            " 'SLC38A10',\n",
            " 'SLC38A11',\n",
            " 'SLC38A2',\n",
            " 'SLC38A3',\n",
            " 'SLC38A4',\n",
            " 'SLC38A5',\n",
            " 'SLC38A6',\n",
            " 'SLC38A7',\n",
            " 'SLC38A8',\n",
            " 'SLC38A9',\n",
            " 'SLC39A1',\n",
            " 'SLC39A10',\n",
            " 'SLC39A11',\n",
            " 'SLC39A12',\n",
            " 'SLC39A13',\n",
            " 'SLC39A14',\n",
            " 'SLC39A2',\n",
            " 'SLC39A3',\n",
            " 'SLC39A4',\n",
            " 'SLC39A5',\n",
            " 'SLC39A6',\n",
            " 'SLC39A7',\n",
            " 'SLC39A8',\n",
            " 'SLC39A9',\n",
            " 'SLC3A1',\n",
            " 'SLC3A2',\n",
            " 'SLC40A1',\n",
            " 'SLC41A1',\n",
            " 'SLC41A2',\n",
            " 'SLC41A3',\n",
            " 'SLC43A1',\n",
            " 'SLC43A2',\n",
            " 'SLC43A3',\n",
            " 'SLC44A1',\n",
            " 'SLC44A2',\n",
            " 'SLC44A3',\n",
            " 'SLC44A4',\n",
            " 'SLC44A5',\n",
            " 'SLC45A1',\n",
            " 'SLC45A2',\n",
            " 'SLC45A3',\n",
            " 'SLC45A4',\n",
            " 'SLC46A1',\n",
            " 'SLC46A2',\n",
            " 'SLC46A3',\n",
            " 'SLC47A1',\n",
            " 'SLC47A2',\n",
            " 'SLC48A1',\n",
            " 'SLC49A3',\n",
            " 'SLC49A4',\n",
            " 'SLC4A1',\n",
            " 'SLC4A10',\n",
            " 'SLC4A11',\n",
            " 'SLC4A1AP',\n",
            " 'SLC4A2',\n",
            " 'SLC4A3',\n",
            " 'SLC4A4',\n",
            " 'SLC4A5',\n",
            " 'SLC4A7',\n",
            " 'SLC4A8',\n",
            " 'SLC4A9',\n",
            " 'SLC50A1',\n",
            " 'SLC51A',\n",
            " 'SLC51B',\n",
            " 'SLC52A1',\n",
            " 'SLC52A2',\n",
            " 'SLC52A3',\n",
            " 'SLC5A1',\n",
            " 'SLC5A10',\n",
            " 'SLC5A11',\n",
            " 'SLC5A12',\n",
            " 'SLC5A2',\n",
            " 'SLC5A4',\n",
            " 'SLC5A5',\n",
            " 'SLC5A6',\n",
            " 'SLC5A7',\n",
            " 'SLC5A8',\n",
            " 'SLC5A9',\n",
            " 'SLC66A1',\n",
            " 'SLC66A2',\n",
            " 'SLC66A3',\n",
            " 'SLC6A1',\n",
            " 'SLC6A11',\n",
            " 'SLC6A12',\n",
            " 'SLC6A13',\n",
            " 'SLC6A14',\n",
            " 'SLC6A15',\n",
            " 'SLC6A16',\n",
            " 'SLC6A17',\n",
            " 'SLC6A18',\n",
            " 'SLC6A19',\n",
            " 'SLC6A2',\n",
            " 'SLC6A20',\n",
            " 'SLC6A3',\n",
            " 'SLC6A4',\n",
            " 'SLC6A5',\n",
            " 'SLC6A6',\n",
            " 'SLC6A7',\n",
            " 'SLC6A8',\n",
            " 'SLC6A9',\n",
            " 'SLC7A1',\n",
            " 'SLC7A10',\n",
            " 'SLC7A11',\n",
            " 'SLC7A13',\n",
            " 'SLC7A14',\n",
            " 'SLC7A2',\n",
            " 'SLC7A3',\n",
            " 'SLC7A4',\n",
            " 'SLC7A5',\n",
            " 'SLC7A6',\n",
            " 'SLC7A6OS',\n",
            " 'SLC7A7',\n",
            " 'SLC7A8',\n",
            " 'SLC7A9',\n",
            " 'SLC8A1',\n",
            " 'SLC8A2',\n",
            " 'SLC8A3',\n",
            " 'SLC8B1',\n",
            " 'SLC9A1',\n",
            " 'SLC9A2',\n",
            " 'SLC9A3',\n",
            " 'SLC9A4',\n",
            " 'SLC9A5',\n",
            " 'SLC9A6',\n",
            " 'SLC9A7',\n",
            " 'SLC9A8',\n",
            " 'SLC9A9',\n",
            " 'SLC9B1',\n",
            " 'SLC9B2',\n",
            " 'SLC9C1',\n",
            " 'SLC9C2',\n",
            " 'SLCO1A2',\n",
            " 'SLCO1B1',\n",
            " 'SLCO1B3',\n",
            " 'SLCO1C1',\n",
            " 'SLCO2A1',\n",
            " 'SLCO2B1',\n",
            " 'SLCO3A1',\n",
            " 'SLCO4A1',\n",
            " 'SLCO4C1',\n",
            " 'SLCO5A1',\n",
            " 'SLCO6A1',\n",
            " 'SLF1',\n",
            " 'SLF2',\n",
            " 'SLFN11',\n",
            " 'SLFN12',\n",
            " 'SLFN12L',\n",
            " 'SLFN13',\n",
            " 'SLFN14',\n",
            " 'SLFN5',\n",
            " 'SLFNL1',\n",
            " 'SLIRP',\n",
            " 'SLIT1',\n",
            " 'SLIT2',\n",
            " 'SLIT3',\n",
            " 'SLITRK1',\n",
            " 'SLITRK2',\n",
            " 'SLITRK3',\n",
            " 'SLITRK4',\n",
            " 'SLITRK5',\n",
            " 'SLITRK6',\n",
            " 'SLK',\n",
            " 'SLMAP',\n",
            " 'SLN',\n",
            " 'SLPI',\n",
            " 'SLTM',\n",
            " 'SLU7',\n",
            " 'SLURP1',\n",
            " 'SLX4',\n",
            " 'SLX4IP',\n",
            " 'SLX9',\n",
            " 'SMAD1',\n",
            " 'SMAD2',\n",
            " 'SMAD3',\n",
            " 'SMAD4',\n",
            " 'SMAD5',\n",
            " 'SMAD6',\n",
            " 'SMAD7',\n",
            " 'SMAD9',\n",
            " 'SMAGP',\n",
            " 'SMAP1',\n",
            " 'SMAP2',\n",
            " 'SMARCA1',\n",
            " 'SMARCA2',\n",
            " 'SMARCA4',\n",
            " 'SMARCA5',\n",
            " 'SMARCAD1',\n",
            " 'SMARCAL1',\n",
            " 'SMARCB1',\n",
            " 'SMARCC1',\n",
            " 'SMARCC2',\n",
            " 'SMARCD1',\n",
            " 'SMARCD2',\n",
            " 'SMARCD3',\n",
            " 'SMARCE1',\n",
            " 'SMC1B',\n",
            " 'SMC2',\n",
            " 'SMC3',\n",
            " 'SMC4',\n",
            " 'SMC5',\n",
            " 'SMC6',\n",
            " 'SMCHD1',\n",
            " 'SMCO1',\n",
            " 'SMCO2',\n",
            " 'SMCO3',\n",
            " 'SMCO4',\n",
            " 'SMCP',\n",
            " 'SMCR8',\n",
            " 'SMDT1',\n",
            " 'SMG1',\n",
            " 'SMG5',\n",
            " 'SMG6',\n",
            " 'SMG7',\n",
            " 'SMG8',\n",
            " 'SMG9',\n",
            " 'SMIM1',\n",
            " 'SMIM10',\n",
            " 'SMIM11',\n",
            " 'SMIM12',\n",
            " 'SMIM13',\n",
            " 'SMIM14',\n",
            " 'SMIM15',\n",
            " 'SMIM17',\n",
            " 'SMIM18',\n",
            " 'SMIM2',\n",
            " 'SMIM20',\n",
            " 'SMIM21',\n",
            " 'SMIM22',\n",
            " 'SMIM23',\n",
            " 'SMIM24',\n",
            " 'SMIM29',\n",
            " 'SMIM3',\n",
            " 'SMIM5',\n",
            " 'SMIM6',\n",
            " 'SMIM7',\n",
            " 'SMIM8',\n",
            " 'SMIM9',\n",
            " 'SMKR1',\n",
            " 'SMLR1',\n",
            " 'SMNDC1',\n",
            " 'SMO',\n",
            " 'SMOC1',\n",
            " 'SMOC2',\n",
            " 'SMOX',\n",
            " 'SMPD1',\n",
            " 'SMPD2',\n",
            " 'SMPD3',\n",
            " 'SMPD4',\n",
            " 'SMPDL3A',\n",
            " 'SMPDL3B',\n",
            " 'SMPX',\n",
            " 'SMR3A',\n",
            " 'SMR3B',\n",
            " 'SMS',\n",
            " 'SMTN',\n",
            " 'SMTNL1',\n",
            " 'SMTNL2',\n",
            " 'SMU1',\n",
            " 'SMUG1',\n",
            " 'SMURF1',\n",
            " 'SMURF2',\n",
            " 'SMYD1',\n",
            " 'SMYD2',\n",
            " 'SMYD3',\n",
            " 'SMYD4',\n",
            " 'SMYD5',\n",
            " 'SNAI1',\n",
            " 'SNAI3',\n",
            " 'SNAP23',\n",
            " 'SNAP25',\n",
            " 'SNAP29',\n",
            " 'SNAP47',\n",
            " 'SNAP91',\n",
            " 'SNAPC1',\n",
            " 'SNAPC2',\n",
            " 'SNAPC3',\n",
            " 'SNAPC4',\n",
            " 'SNAPC5',\n",
            " 'SNAPIN',\n",
            " 'SNCA',\n",
            " 'SNCAIP',\n",
            " 'SNCB',\n",
            " 'SNCG',\n",
            " 'SND1',\n",
            " 'SNED1',\n",
            " 'SNF8',\n",
            " 'SNHG28',\n",
            " 'SNHG32',\n",
            " 'SNIP1',\n",
            " 'SNN',\n",
            " 'SNORC',\n",
            " 'SNPH',\n",
            " 'SNRK',\n",
            " 'SNRNP200',\n",
            " 'SNRNP25',\n",
            " 'SNRNP27',\n",
            " 'SNRNP35',\n",
            " 'SNRNP40',\n",
            " 'SNRNP48',\n",
            " 'SNRNP70',\n",
            " 'SNRPA',\n",
            " 'SNRPA1',\n",
            " 'SNRPB',\n",
            " 'SNRPB2',\n",
            " 'SNRPC',\n",
            " 'SNRPD1',\n",
            " 'SNRPD2',\n",
            " 'SNRPD3',\n",
            " 'SNRPE',\n",
            " 'SNRPF',\n",
            " 'SNRPG',\n",
            " 'SNTA1',\n",
            " 'SNTB1',\n",
            " 'SNTB2',\n",
            " 'SNTG1',\n",
            " 'SNTG2',\n",
            " 'SNTN',\n",
            " 'SNU13',\n",
            " 'SNUPN',\n",
            " 'SNURF',\n",
            " 'SNW1',\n",
            " 'SNX1',\n",
            " 'SNX10',\n",
            " 'SNX11',\n",
            " 'SNX12',\n",
            " 'SNX13',\n",
            " 'SNX14',\n",
            " 'SNX15',\n",
            " 'SNX16',\n",
            " 'SNX17',\n",
            " 'SNX18',\n",
            " 'SNX19',\n",
            " 'SNX2',\n",
            " 'SNX20',\n",
            " 'SNX21',\n",
            " 'SNX22',\n",
            " 'SNX24',\n",
            " 'SNX25',\n",
            " 'SNX27',\n",
            " 'SNX29',\n",
            " 'SNX3',\n",
            " 'SNX30',\n",
            " 'SNX31',\n",
            " 'SNX32',\n",
            " 'SNX33',\n",
            " 'SNX4',\n",
            " 'SNX5',\n",
            " 'SNX6',\n",
            " 'SNX7',\n",
            " 'SNX8',\n",
            " 'SNX9',\n",
            " 'SOAT1',\n",
            " 'SOAT2',\n",
            " 'SOBP',\n",
            " 'SOCS1',\n",
            " 'SOCS2',\n",
            " 'SOCS3',\n",
            " 'SOCS4',\n",
            " 'SOCS5',\n",
            " 'SOCS6',\n",
            " 'SOCS7',\n",
            " 'SOD1',\n",
            " 'SOD2',\n",
            " 'SOD3',\n",
            " 'SOGA1',\n",
            " 'SOGA3',\n",
            " 'SOHLH1',\n",
            " 'SOHLH2',\n",
            " 'SON',\n",
            " 'SORBS1',\n",
            " 'SORBS2',\n",
            " 'SORBS3',\n",
            " 'SORCS1',\n",
            " 'SORCS2',\n",
            " 'SORCS3',\n",
            " 'SORD',\n",
            " 'SORL1',\n",
            " 'SORT1',\n",
            " 'SOS1',\n",
            " 'SOS2',\n",
            " 'SOST',\n",
            " 'SOSTDC1',\n",
            " 'SOWAHA',\n",
            " 'SOWAHB',\n",
            " 'SOWAHC',\n",
            " 'SOWAHD',\n",
            " 'SOX1',\n",
            " 'SOX10',\n",
            " 'SOX11',\n",
            " 'SOX12',\n",
            " 'SOX13',\n",
            " 'SOX14',\n",
            " 'SOX15',\n",
            " 'SOX17',\n",
            " 'SOX18',\n",
            " 'SOX2',\n",
            " 'SOX21',\n",
            " 'SOX3',\n",
            " 'SOX30',\n",
            " 'SOX4',\n",
            " 'SOX5',\n",
            " 'SOX6',\n",
            " 'SOX8',\n",
            " 'SOX9',\n",
            " 'SP1',\n",
            " 'SP100',\n",
            " 'SP110',\n",
            " 'SP140',\n",
            " 'SP140L',\n",
            " 'SP2',\n",
            " 'SP3',\n",
            " 'SP4',\n",
            " 'SP5',\n",
            " 'SP6',\n",
            " 'SP7',\n",
            " 'SP8',\n",
            " 'SP9',\n",
            " 'SPA17',\n",
            " 'SPACA1',\n",
            " 'SPACA3',\n",
            " 'SPACA4',\n",
            " 'SPACA5',\n",
            " 'SPACA5B',\n",
            " 'SPACA7',\n",
            " 'SPACA9',\n",
            " 'SPACDR',\n",
            " 'SPAG1',\n",
            " 'SPAG16',\n",
            " 'SPAG17',\n",
            " 'SPAG4',\n",
            " 'SPAG5',\n",
            " 'SPAG6',\n",
            " 'SPAG7',\n",
            " 'SPAG8',\n",
            " 'SPAG9',\n",
            " 'SPAM1',\n",
            " 'SPANXC',\n",
            " 'SPANXN1',\n",
            " 'SPANXN2',\n",
            " 'SPANXN3',\n",
            " 'SPANXN4',\n",
            " 'SPANXN5',\n",
            " 'SPARC',\n",
            " 'SPARCL1',\n",
            " 'SPART',\n",
            " 'SPAST',\n",
            " 'SPATA12',\n",
            " 'SPATA13',\n",
            " 'SPATA16',\n",
            " 'SPATA17',\n",
            " 'SPATA18',\n",
            " 'SPATA19',\n",
            " 'SPATA2',\n",
            " 'SPATA20',\n",
            " 'SPATA21',\n",
            " 'SPATA22',\n",
            " 'SPATA24',\n",
            " 'SPATA25',\n",
            " 'SPATA2L',\n",
            " 'SPATA3',\n",
            " 'SPATA31D1',\n",
            " 'SPATA31E1',\n",
            " 'SPATA32',\n",
            " 'SPATA33',\n",
            " 'SPATA4',\n",
            " 'SPATA45',\n",
            " 'SPATA46',\n",
            " 'SPATA6',\n",
            " 'SPATA6L',\n",
            " 'SPATA7',\n",
            " 'SPATA8',\n",
            " 'SPATA9',\n",
            " 'SPATC1',\n",
            " 'SPATC1L',\n",
            " 'SPATS1',\n",
            " 'SPATS2',\n",
            " 'SPATS2L',\n",
            " 'SPC24',\n",
            " 'SPC25',\n",
            " 'SPCS1',\n",
            " 'SPCS2',\n",
            " 'SPCS3',\n",
            " 'SPDEF',\n",
            " 'SPDL1',\n",
            " 'SPDYA',\n",
            " 'SPDYC',\n",
            " 'SPDYE2',\n",
            " 'SPDYE2B',\n",
            " 'SPDYE4',\n",
            " 'SPECC1',\n",
            " 'SPECC1L',\n",
            " 'SPEF1',\n",
            " 'SPEF2',\n",
            " 'SPEG',\n",
            " 'SPEM1',\n",
            " 'SPEM2',\n",
            " 'SPEN',\n",
            " 'SPG11',\n",
            " 'SPG21',\n",
            " 'SPG7',\n",
            " 'SPHK1',\n",
            " 'SPHK2',\n",
            " 'SPHKAP',\n",
            " 'SPI1',\n",
            " 'SPIB',\n",
            " 'SPIC',\n",
            " 'SPICE1',\n",
            " 'SPIDR',\n",
            " 'SPIN1',\n",
            " 'SPIN2A',\n",
            " 'SPIN2B',\n",
            " 'SPIN3',\n",
            " 'SPIN4',\n",
            " 'SPINDOC',\n",
            " 'SPINK1',\n",
            " 'SPINK13',\n",
            " 'SPINK14',\n",
            " 'SPINK2',\n",
            " 'SPINK4',\n",
            " 'SPINK5',\n",
            " 'SPINK6',\n",
            " 'SPINK7',\n",
            " 'SPINK8',\n",
            " 'SPINT1',\n",
            " 'SPINT2',\n",
            " 'SPINT3',\n",
            " 'SPINT4',\n",
            " 'SPIRE1',\n",
            " 'SPIRE2',\n",
            " 'SPN',\n",
            " 'SPNS1',\n",
            " 'SPNS2',\n",
            " 'SPNS3',\n",
            " 'SPO11',\n",
            " 'SPOCD1',\n",
            " 'SPOCK1',\n",
            " 'SPOCK2',\n",
            " 'SPOCK3',\n",
            " 'SPON1',\n",
            " 'SPON2',\n",
            " 'SPOP',\n",
            " 'SPOPL',\n",
            " 'SPOUT1',\n",
            " 'SPP1',\n",
            " 'SPP2',\n",
            " 'SPPL2A',\n",
            " 'SPPL2B',\n",
            " 'SPPL2C',\n",
            " 'SPPL3',\n",
            " 'SPR',\n",
            " 'SPRED1',\n",
            " 'SPRED2',\n",
            " 'SPRED3',\n",
            " 'SPRING1',\n",
            " 'SPRN',\n",
            " 'SPRR1A',\n",
            " 'SPRR1B',\n",
            " 'SPRR2A',\n",
            " 'SPRR2B',\n",
            " 'SPRR2D',\n",
            " 'SPRR2E',\n",
            " 'SPRR2F',\n",
            " 'SPRR2G',\n",
            " 'SPRR3',\n",
            " 'SPRR4',\n",
            " 'SPRTN',\n",
            " 'SPRY1',\n",
            " 'SPRY2',\n",
            " 'SPRY3',\n",
            " 'SPRY4',\n",
            " 'SPRYD3',\n",
            " 'SPRYD4',\n",
            " 'SPRYD7',\n",
            " 'SPSB1',\n",
            " 'SPSB2',\n",
            " 'SPSB3',\n",
            " 'SPSB4',\n",
            " 'SPTA1',\n",
            " 'SPTAN1',\n",
            " 'SPTB',\n",
            " 'SPTBN1',\n",
            " 'SPTBN2',\n",
            " 'SPTBN4',\n",
            " 'SPTBN5',\n",
            " 'SPTLC1',\n",
            " 'SPTLC2',\n",
            " 'SPTLC3',\n",
            " 'SPTSSA',\n",
            " 'SPTY2D1',\n",
            " 'SPX',\n",
            " 'SPZ1',\n",
            " 'SQLE',\n",
            " 'SQOR',\n",
            " 'SQSTM1',\n",
            " 'SRA1',\n",
            " 'SRARP',\n",
            " 'SRBD1',\n",
            " 'SRC',\n",
            " 'SRCAP',\n",
            " 'SRCIN1',\n",
            " 'SRD5A1',\n",
            " 'SRD5A2',\n",
            " 'SRD5A3',\n",
            " 'SREBF1',\n",
            " 'SREBF2',\n",
            " 'SREK1',\n",
            " 'SREK1IP1',\n",
            " 'SRF',\n",
            " 'SRFBP1',\n",
            " 'SRGAP1',\n",
            " 'SRGAP2',\n",
            " 'SRGAP3',\n",
            " 'SRGN',\n",
            " 'SRI',\n",
            " 'SRL',\n",
            " 'SRM',\n",
            " 'SRMS',\n",
            " 'SRP14',\n",
            " 'SRP19',\n",
            " 'SRP54',\n",
            " 'SRP68',\n",
            " 'SRP72',\n",
            " 'SRP9',\n",
            " 'SRPK1',\n",
            " 'SRPK2',\n",
            " 'SRPK3',\n",
            " 'SRPRA',\n",
            " 'SRPRB',\n",
            " 'SRPX',\n",
            " 'SRPX2',\n",
            " 'SRR',\n",
            " 'SRRD',\n",
            " 'SRRM1',\n",
            " 'SRRM2',\n",
            " 'SRRM3',\n",
            " 'SRRM4',\n",
            " 'SRRM5',\n",
            " 'SRRT',\n",
            " 'SRSF1',\n",
            " 'SRSF10',\n",
            " 'SRSF11',\n",
            " 'SRSF12',\n",
            " 'SRSF2',\n",
            " 'SRSF3',\n",
            " 'SRSF4',\n",
            " 'SRSF5',\n",
            " 'SRSF6',\n",
            " 'SRSF7',\n",
            " 'SRSF8',\n",
            " 'SRSF9',\n",
            " 'SRXN1',\n",
            " 'SRY',\n",
            " 'SS18',\n",
            " 'SS18L1',\n",
            " 'SS18L2',\n",
            " 'SSB',\n",
            " 'SSBP1',\n",
            " 'SSBP2',\n",
            " 'SSBP3',\n",
            " 'SSBP4',\n",
            " 'SSC4D',\n",
            " 'SSC5D',\n",
            " 'SSH1',\n",
            " 'SSH2',\n",
            " 'SSH3',\n",
            " 'SSMEM1',\n",
            " 'SSNA1',\n",
            " 'SSPN',\n",
            " 'SSR1',\n",
            " 'SSR2',\n",
            " 'SSR3',\n",
            " 'SSR4',\n",
            " 'SSRP1',\n",
            " 'SST',\n",
            " 'SSTR1',\n",
            " 'SSTR2',\n",
            " 'SSTR3',\n",
            " 'SSTR4',\n",
            " 'SSTR5',\n",
            " 'SSU72',\n",
            " 'SSUH2',\n",
            " 'SSX1',\n",
            " 'SSX2IP',\n",
            " 'SSX3',\n",
            " 'SSX4',\n",
            " 'SSX4B',\n",
            " 'SSX7',\n",
            " 'ST13',\n",
            " 'ST14',\n",
            " 'ST18',\n",
            " 'ST20',\n",
            " 'ST3GAL1',\n",
            " 'ST3GAL2',\n",
            " 'ST3GAL3',\n",
            " 'ST3GAL4',\n",
            " 'ST3GAL5',\n",
            " 'ST3GAL6',\n",
            " 'ST6GAL1',\n",
            " 'ST6GAL2',\n",
            " 'ST6GALNAC1',\n",
            " 'ST6GALNAC2',\n",
            " 'ST6GALNAC3',\n",
            " 'ST6GALNAC4',\n",
            " 'ST6GALNAC5',\n",
            " 'ST6GALNAC6',\n",
            " 'ST7',\n",
            " 'ST7L',\n",
            " 'ST8SIA1',\n",
            " 'ST8SIA2',\n",
            " 'ST8SIA3',\n",
            " 'ST8SIA4',\n",
            " 'ST8SIA5',\n",
            " 'ST8SIA6',\n",
            " 'STAB1',\n",
            " 'STAB2',\n",
            " 'STAC',\n",
            " 'STAC2',\n",
            " 'STAC3',\n",
            " 'STAG1',\n",
            " 'STAG2',\n",
            " 'STAG3',\n",
            " 'STAM',\n",
            " 'STAM2',\n",
            " 'STAMBP',\n",
            " 'STAMBPL1',\n",
            " 'STAP1',\n",
            " 'STAP2',\n",
            " 'STAR',\n",
            " 'STARD10',\n",
            " 'STARD13',\n",
            " 'STARD3',\n",
            " 'STARD3NL',\n",
            " 'STARD4',\n",
            " 'STARD5',\n",
            " 'STARD6',\n",
            " 'STARD7',\n",
            " 'STARD8',\n",
            " 'STARD9',\n",
            " 'STAT1',\n",
            " 'STAT2',\n",
            " 'STAT3',\n",
            " 'STAT4',\n",
            " 'STAT5A',\n",
            " 'STAT5B',\n",
            " 'STAT6',\n",
            " 'STATH',\n",
            " 'STAU1',\n",
            " 'STAU2',\n",
            " 'STBD1',\n",
            " 'STC1',\n",
            " 'STC2',\n",
            " 'STEAP1',\n",
            " 'STEAP1B',\n",
            " 'STEAP2',\n",
            " 'STEAP3',\n",
            " 'STEAP4',\n",
            " 'STEEP1',\n",
            " 'STH',\n",
            " 'STIL',\n",
            " 'STIM1',\n",
            " 'STIM2',\n",
            " 'STIMATE',\n",
            " 'STING1',\n",
            " 'STIP1',\n",
            " 'STK10',\n",
            " 'STK11',\n",
            " 'STK11IP',\n",
            " 'STK16',\n",
            " 'STK17A',\n",
            " 'STK17B',\n",
            " 'STK19',\n",
            " 'STK24',\n",
            " 'STK25',\n",
            " 'STK26',\n",
            " 'STK3',\n",
            " 'STK31',\n",
            " 'STK32A',\n",
            " 'STK32B',\n",
            " 'STK32C',\n",
            " 'STK33',\n",
            " 'STK35',\n",
            " 'STK36',\n",
            " 'STK38',\n",
            " 'STK38L',\n",
            " 'STK39',\n",
            " 'STK4',\n",
            " 'STK40',\n",
            " 'STKLD1',\n",
            " 'STMN1',\n",
            " 'STMN2',\n",
            " 'STMN3',\n",
            " 'STMN4',\n",
            " 'STMND1',\n",
            " 'STMP1',\n",
            " 'STN1',\n",
            " 'STOM',\n",
            " 'STOML1',\n",
            " 'STOML2',\n",
            " 'STOML3',\n",
            " 'STON1',\n",
            " 'STON2',\n",
            " 'STOX1',\n",
            " 'STOX2',\n",
            " 'STPG1',\n",
            " 'STPG2',\n",
            " 'STPG3',\n",
            " 'STPG4',\n",
            " 'STRA6',\n",
            " 'STRA8',\n",
            " 'STRADA',\n",
            " 'STRADB',\n",
            " 'STRAP',\n",
            " 'STRBP',\n",
            " 'STRC',\n",
            " 'STRIP1',\n",
            " 'STRIP2',\n",
            " 'STRN',\n",
            " 'STRN3',\n",
            " 'STRN4',\n",
            " 'STS',\n",
            " 'STT3A',\n",
            " 'STT3B',\n",
            " 'STUB1',\n",
            " 'STUM',\n",
            " 'STX10',\n",
            " 'STX11',\n",
            " 'STX12',\n",
            " 'STX16',\n",
            " 'STX17',\n",
            " 'STX18',\n",
            " 'STX19',\n",
            " 'STX1A',\n",
            " 'STX1B',\n",
            " 'STX2',\n",
            " 'STX3',\n",
            " 'STX4',\n",
            " 'STX5',\n",
            " 'STX6',\n",
            " 'STX7',\n",
            " 'STX8',\n",
            " 'STXBP1',\n",
            " 'STXBP2',\n",
            " 'STXBP3',\n",
            " 'STXBP4',\n",
            " 'STXBP5',\n",
            " 'STXBP5L',\n",
            " 'STXBP6',\n",
            " 'STYK1',\n",
            " 'STYX',\n",
            " 'STYXL1',\n",
            " 'STYXL2',\n",
            " 'SUB1',\n",
            " 'SUCLA2',\n",
            " 'SUCLG1',\n",
            " 'SUCLG2',\n",
            " 'SUCNR1',\n",
            " 'SUCO',\n",
            " 'SUDS3',\n",
            " 'SUFU',\n",
            " 'SUGCT',\n",
            " 'SUGP1',\n",
            " 'SUGP2',\n",
            " 'SUGT1',\n",
            " 'SULF1',\n",
            " 'SULF2',\n",
            " 'SULT1A1',\n",
            " 'SULT1A2',\n",
            " 'SULT1A3',\n",
            " 'SULT1A4',\n",
            " 'SULT1B1',\n",
            " 'SULT1C2',\n",
            " 'SULT1C3',\n",
            " 'SULT1C4',\n",
            " 'SULT1E1',\n",
            " 'SULT2A1',\n",
            " 'SULT2B1',\n",
            " 'SULT4A1',\n",
            " 'SULT6B1',\n",
            " 'SUMF1',\n",
            " 'SUMF2',\n",
            " 'SUMO1',\n",
            " 'SUMO2',\n",
            " 'SUMO3',\n",
            " 'SUMO4',\n",
            " 'SUN1',\n",
            " 'SUN2',\n",
            " 'SUN3',\n",
            " 'SUN5',\n",
            " 'SUOX',\n",
            " 'SUPT16H',\n",
            " 'SUPT20H',\n",
            " 'SUPT3H',\n",
            " 'SUPT4H1',\n",
            " 'SUPT5H',\n",
            " 'SUPT6H',\n",
            " 'SUPT7L',\n",
            " 'SUPV3L1',\n",
            " 'SURF1',\n",
            " 'SURF2',\n",
            " 'SURF4',\n",
            " 'SURF6',\n",
            " 'SUSD1',\n",
            " 'SUSD2',\n",
            " 'SUSD3',\n",
            " 'SUSD4',\n",
            " 'SUSD5',\n",
            " 'SUSD6',\n",
            " 'SUV39H1',\n",
            " 'SUV39H2',\n",
            " 'SUZ12',\n",
            " 'SV2A',\n",
            " 'SV2B',\n",
            " 'SV2C',\n",
            " 'SVBP',\n",
            " 'SVEP1',\n",
            " 'SVIL',\n",
            " 'SVIP',\n",
            " 'SVOP',\n",
            " 'SVOPL',\n",
            " 'SWAP70',\n",
            " 'SWI5',\n",
            " 'SWSAP1',\n",
            " 'SWT1',\n",
            " 'SYAP1',\n",
            " 'SYBU',\n",
            " 'SYCE1',\n",
            " 'SYCE1L',\n",
            " 'SYCE2',\n",
            " 'SYCE3',\n",
            " 'SYCN',\n",
            " 'SYCP1',\n",
            " 'SYCP2',\n",
            " 'SYCP2L',\n",
            " 'SYCP3',\n",
            " 'SYDE1',\n",
            " 'SYDE2',\n",
            " 'SYF2',\n",
            " 'SYK',\n",
            " 'SYMPK',\n",
            " 'SYN1',\n",
            " 'SYN2',\n",
            " 'SYN3',\n",
            " 'SYNC',\n",
            " 'SYNCRIP',\n",
            " 'SYNDIG1',\n",
            " 'SYNDIG1L',\n",
            " 'SYNE1',\n",
            " 'SYNE2',\n",
            " 'SYNE3',\n",
            " 'SYNE4',\n",
            " 'SYNGAP1',\n",
            " 'SYNGR1',\n",
            " 'SYNGR2',\n",
            " 'SYNGR3',\n",
            " 'SYNGR4',\n",
            " 'SYNJ1',\n",
            " 'SYNJ2',\n",
            " 'SYNJ2BP',\n",
            " 'SYNM',\n",
            " 'SYNPO',\n",
            " 'SYNPO2',\n",
            " 'SYNPO2L',\n",
            " 'SYNPR',\n",
            " 'SYNRG',\n",
            " 'SYP',\n",
            " 'SYPL1',\n",
            " 'SYPL2',\n",
            " 'SYS1',\n",
            " 'SYT1',\n",
            " 'SYT10',\n",
            " 'SYT11',\n",
            " 'SYT12',\n",
            " 'SYT13',\n",
            " 'SYT14',\n",
            " 'SYT15',\n",
            " 'SYT16',\n",
            " 'SYT17',\n",
            " 'SYT2',\n",
            " 'SYT3',\n",
            " 'SYT4',\n",
            " 'SYT5',\n",
            " 'SYT6',\n",
            " 'SYT7',\n",
            " 'SYT8',\n",
            " 'SYT9',\n",
            " 'SYTL1',\n",
            " 'SYTL2',\n",
            " 'SYTL3',\n",
            " 'SYTL4',\n",
            " 'SYTL5',\n",
            " 'SYVN1',\n",
            " 'SZRD1',\n",
            " 'SZT2',\n",
            " 'TAAR1',\n",
            " 'TAAR2',\n",
            " 'TAAR5',\n",
            " 'TAAR6',\n",
            " 'TAAR8',\n",
            " 'TAAR9',\n",
            " 'TAB1',\n",
            " 'TAB2',\n",
            " 'TAB3',\n",
            " 'TAC1',\n",
            " 'TAC3',\n",
            " 'TAC4',\n",
            " 'TACC1',\n",
            " 'TACC2',\n",
            " 'TACC3',\n",
            " 'TACO1',\n",
            " 'TACR1',\n",
            " 'TACR2',\n",
            " 'TACR3',\n",
            " 'TACSTD2',\n",
            " 'TADA1',\n",
            " 'TADA2A',\n",
            " 'TADA2B',\n",
            " 'TADA3',\n",
            " 'TAF1',\n",
            " 'TAF10',\n",
            " 'TAF11',\n",
            " 'TAF12',\n",
            " 'TAF13',\n",
            " 'TAF15',\n",
            " 'TAF1A',\n",
            " 'TAF1B',\n",
            " 'TAF1C',\n",
            " 'TAF1D',\n",
            " 'TAF1L',\n",
            " 'TAF2',\n",
            " 'TAF3',\n",
            " 'TAF4',\n",
            " 'TAF4B',\n",
            " 'TAF5',\n",
            " 'TAF5L',\n",
            " 'TAF6',\n",
            " 'TAF6L',\n",
            " 'TAF7',\n",
            " 'TAF7L',\n",
            " 'TAF8',\n",
            " 'TAF9',\n",
            " 'TAF9B',\n",
            " 'TAFA1',\n",
            " 'TAFA2',\n",
            " 'TAFA3',\n",
            " 'TAFA4',\n",
            " 'TAFA5',\n",
            " 'TAFAZZIN',\n",
            " 'TAGAP',\n",
            " 'TAGLN',\n",
            " 'TAGLN3',\n",
            " 'TAL1',\n",
            " 'TAL2',\n",
            " 'TALDO1',\n",
            " 'TAMALIN',\n",
            " 'TAMM41',\n",
            " 'TANC1',\n",
            " 'TANC2',\n",
            " 'TANGO2',\n",
            " 'TANGO6',\n",
            " 'TANK',\n",
            " 'TAOK1',\n",
            " 'TAOK2',\n",
            " 'TAOK3',\n",
            " 'TAP1',\n",
            " 'TAP2',\n",
            " 'TAPBP',\n",
            " 'TAPBPL',\n",
            " 'TAPT1',\n",
            " 'TARBP1',\n",
            " 'TARBP2',\n",
            " 'TARDBP',\n",
            " 'TARM1',\n",
            " 'TARP',\n",
            " 'TARS1',\n",
            " 'TARS2',\n",
            " 'TARS3',\n",
            " 'TAS1R1',\n",
            " 'TAS1R2',\n",
            " 'TAS1R3',\n",
            " 'TAS2R1',\n",
            " 'TAS2R10',\n",
            " 'TAS2R13',\n",
            " 'TAS2R14',\n",
            " 'TAS2R16',\n",
            " 'TAS2R19',\n",
            " 'TAS2R20',\n",
            " 'TAS2R3',\n",
            " 'TAS2R30',\n",
            " 'TAS2R31',\n",
            " 'TAS2R38',\n",
            " 'TAS2R39',\n",
            " 'TAS2R4',\n",
            " 'TAS2R40',\n",
            " 'TAS2R41',\n",
            " 'TAS2R42',\n",
            " 'TAS2R43',\n",
            " 'TAS2R46',\n",
            " 'TAS2R5',\n",
            " 'TAS2R50',\n",
            " 'TAS2R60',\n",
            " 'TAS2R7',\n",
            " 'TAS2R8',\n",
            " 'TAS2R9',\n",
            " 'TASL',\n",
            " 'TASOR',\n",
            " 'TASOR2',\n",
            " 'TASP1',\n",
            " 'TAT',\n",
            " 'TATDN1',\n",
            " 'TATDN2',\n",
            " 'TATDN3',\n",
            " 'TAX1BP1',\n",
            " 'TAX1BP3',\n",
            " 'TBATA',\n",
            " 'TBC1D1',\n",
            " 'TBC1D10A',\n",
            " 'TBC1D10B',\n",
            " 'TBC1D10C',\n",
            " 'TBC1D12',\n",
            " 'TBC1D13',\n",
            " 'TBC1D14',\n",
            " 'TBC1D15',\n",
            " 'TBC1D16',\n",
            " 'TBC1D17',\n",
            " 'TBC1D19',\n",
            " 'TBC1D2',\n",
            " 'TBC1D20',\n",
            " 'TBC1D21',\n",
            " 'TBC1D22A',\n",
            " 'TBC1D22B',\n",
            " 'TBC1D23',\n",
            " 'TBC1D24',\n",
            " 'TBC1D25',\n",
            " 'TBC1D28',\n",
            " 'TBC1D2B',\n",
            " 'TBC1D3',\n",
            " 'TBC1D30',\n",
            " 'TBC1D31',\n",
            " 'TBC1D32',\n",
            " 'TBC1D4',\n",
            " 'TBC1D5',\n",
            " 'TBC1D7',\n",
            " 'TBC1D8',\n",
            " 'TBC1D8B',\n",
            " 'TBC1D9',\n",
            " 'TBC1D9B',\n",
            " 'TBCA',\n",
            " 'TBCB',\n",
            " 'TBCC',\n",
            " 'TBCCD1',\n",
            " 'TBCD',\n",
            " 'TBCE',\n",
            " 'TBCEL',\n",
            " 'TBCK',\n",
            " 'TBK1',\n",
            " 'TBKBP1',\n",
            " 'TBL1X',\n",
            " 'TBL1XR1',\n",
            " 'TBL1Y',\n",
            " 'TBL2',\n",
            " 'TBL3',\n",
            " 'TBP',\n",
            " 'TBPL1',\n",
            " 'TBPL2',\n",
            " 'TBR1',\n",
            " 'TBRG1',\n",
            " 'TBRG4',\n",
            " 'TBX1',\n",
            " 'TBX10',\n",
            " 'TBX15',\n",
            " 'TBX18',\n",
            " 'TBX19',\n",
            " 'TBX2',\n",
            " 'TBX20',\n",
            " 'TBX21',\n",
            " 'TBX22',\n",
            " 'TBX3',\n",
            " 'TBX4',\n",
            " 'TBX5',\n",
            " 'TBX6',\n",
            " 'TBXA2R',\n",
            " 'TBXAS1',\n",
            " 'TBXT',\n",
            " 'TC2N',\n",
            " 'TCAF1',\n",
            " 'TCAF2',\n",
            " 'TCAIM',\n",
            " 'TCAP',\n",
            " 'TCEA1',\n",
            " 'TCEA2',\n",
            " 'TCEA3',\n",
            " 'TCEAL1',\n",
            " 'TCEAL2',\n",
            " 'TCEAL3',\n",
            " 'TCEAL4',\n",
            " 'TCEAL5',\n",
            " 'TCEAL6',\n",
            " 'TCEAL7',\n",
            " 'TCEAL8',\n",
            " 'TCEAL9',\n",
            " 'TCEANC',\n",
            " 'TCEANC2',\n",
            " 'TCERG1',\n",
            " 'TCERG1L',\n",
            " 'TCF12',\n",
            " 'TCF15',\n",
            " 'TCF19',\n",
            " 'TCF20',\n",
            " 'TCF21',\n",
            " 'TCF23',\n",
            " 'TCF24',\n",
            " 'TCF25',\n",
            " 'TCF3',\n",
            " 'TCF4',\n",
            " 'TCF7',\n",
            " 'TCF7L1',\n",
            " 'TCF7L2',\n",
            " 'TCFL5',\n",
            " 'TCHH',\n",
            " 'TCHHL1',\n",
            " 'TCHP',\n",
            " 'TCIM',\n",
            " 'TCIRG1',\n",
            " 'TCL1A',\n",
            " 'TCL1B',\n",
            " 'TCN1',\n",
            " 'TCN2',\n",
            " 'TCOF1',\n",
            " 'TCP1',\n",
            " 'TCP10L',\n",
            " 'TCP10L3',\n",
            " 'TCP11',\n",
            " 'TCP11L1',\n",
            " 'TCP11L2',\n",
            " 'TCTA',\n",
            " 'TCTE1',\n",
            " 'TCTN1',\n",
            " 'TCTN2',\n",
            " 'TCTN3',\n",
            " 'TDG',\n",
            " 'TDO2',\n",
            " 'TDP1',\n",
            " 'TDP2',\n",
            " 'TDRD1',\n",
            " 'TDRD12',\n",
            " 'TDRD15',\n",
            " 'TDRD3',\n",
            " 'TDRD5',\n",
            " 'TDRD6',\n",
            " 'TDRD7',\n",
            " 'TDRD9',\n",
            " 'TDRKH',\n",
            " 'TDRP',\n",
            " 'TEAD1',\n",
            " 'TEAD2',\n",
            " 'TEAD3',\n",
            " 'TEAD4',\n",
            " 'TEC',\n",
            " 'TECPR1',\n",
            " 'TECPR2',\n",
            " 'TECR',\n",
            " 'TECRL',\n",
            " 'TECTA',\n",
            " 'TECTB',\n",
            " 'TEDC1',\n",
            " 'TEDC2',\n",
            " 'TEDDM1',\n",
            " 'TEF',\n",
            " 'TEFM',\n",
            " 'TEK',\n",
            " 'TEKT1',\n",
            " 'TEKT2',\n",
            " 'TEKT3',\n",
            " 'TEKT4',\n",
            " 'TEKT5',\n",
            " 'TEKTIP1',\n",
            " 'TELO2',\n",
            " 'TENM1',\n",
            " 'TENM2',\n",
            " 'TENM3',\n",
            " 'TENM4',\n",
            " 'TENT2',\n",
            " 'TENT4A',\n",
            " 'TENT4B',\n",
            " 'TENT5A',\n",
            " 'TENT5B',\n",
            " 'TENT5C',\n",
            " 'TENT5D',\n",
            " 'TEP1',\n",
            " 'TEPSIN',\n",
            " 'TERB1',\n",
            " 'TERB2',\n",
            " 'TERF1',\n",
            " 'TERF2',\n",
            " 'TERF2IP',\n",
            " 'TERT',\n",
            " 'TES',\n",
            " 'TESC',\n",
            " 'TESK1',\n",
            " 'TESK2',\n",
            " 'TESMIN',\n",
            " 'TESPA1',\n",
            " 'TET1',\n",
            " 'TET2',\n",
            " 'TET3',\n",
            " 'TEX10',\n",
            " 'TEX101',\n",
            " 'TEX11',\n",
            " 'TEX12',\n",
            " 'TEX13A',\n",
            " 'TEX13B',\n",
            " 'TEX14',\n",
            " 'TEX15',\n",
            " 'TEX19',\n",
            " 'TEX2',\n",
            " 'TEX22',\n",
            " 'TEX26',\n",
            " 'TEX261',\n",
            " 'TEX264',\n",
            " 'TEX28',\n",
            " 'TEX29',\n",
            " 'TEX30',\n",
            " 'TEX35',\n",
            " 'TEX36',\n",
            " 'TEX38',\n",
            " 'TEX44',\n",
            " 'TEX46',\n",
            " 'TEX47',\n",
            " 'TEX55',\n",
            " 'TEX9',\n",
            " 'TF',\n",
            " 'TFAM',\n",
            " 'TFAP2A',\n",
            " 'TFAP2B',\n",
            " 'TFAP2C',\n",
            " 'TFAP2D',\n",
            " 'TFAP2E',\n",
            " 'TFAP4',\n",
            " 'TFB1M',\n",
            " 'TFB2M',\n",
            " 'TFCP2',\n",
            " 'TFCP2L1',\n",
            " 'TFDP1',\n",
            " 'TFDP2',\n",
            " 'TFDP3',\n",
            " 'TFE3',\n",
            " 'TFEB',\n",
            " 'TFEC',\n",
            " 'TFF1',\n",
            " 'TFF2',\n",
            " 'TFF3',\n",
            " 'TFG',\n",
            " 'TFIP11',\n",
            " 'TFPI',\n",
            " 'TFPI2',\n",
            " 'TFPT',\n",
            " 'TFR2',\n",
            " 'TFRC',\n",
            " 'TG',\n",
            " 'TGDS',\n",
            " 'TGFA',\n",
            " 'TGFB1',\n",
            " 'TGFB1I1',\n",
            " 'TGFB2',\n",
            " 'TGFB3',\n",
            " 'TGFBI',\n",
            " 'TGFBR1',\n",
            " 'TGFBR2',\n",
            " 'TGFBR3',\n",
            " 'TGFBR3L',\n",
            " 'TGFBRAP1',\n",
            " 'TGIF1',\n",
            " 'TGIF2',\n",
            " 'TGIF2LX',\n",
            " 'TGIF2LY',\n",
            " 'TGM1',\n",
            " 'TGM2',\n",
            " 'TGM3',\n",
            " 'TGM4',\n",
            " 'TGM5',\n",
            " 'TGM6',\n",
            " 'TGM7',\n",
            " 'TGOLN2',\n",
            " 'TGS1',\n",
            " 'TH',\n",
            " 'THADA',\n",
            " 'THAP1',\n",
            " 'THAP10',\n",
            " 'THAP11',\n",
            " 'THAP12',\n",
            " 'THAP2',\n",
            " 'THAP3',\n",
            " 'THAP4',\n",
            " 'THAP5',\n",
            " 'THAP6',\n",
            " 'THAP7',\n",
            " 'THAP8',\n",
            " 'THAP9',\n",
            " 'THBD',\n",
            " 'THBS1',\n",
            " 'THBS2',\n",
            " 'THBS3',\n",
            " 'THBS4',\n",
            " 'THEM4',\n",
            " 'THEM5',\n",
            " 'THEM6',\n",
            " 'THEMIS',\n",
            " 'THEMIS2',\n",
            " 'THG1L',\n",
            " 'THNSL1',\n",
            " 'THNSL2',\n",
            " 'THOC1',\n",
            " 'THOC2',\n",
            " 'THOC3',\n",
            " 'THOC5',\n",
            " 'THOC6',\n",
            " 'THOC7',\n",
            " 'THOP1',\n",
            " 'THPO',\n",
            " 'THRA',\n",
            " 'THRAP3',\n",
            " 'THRB',\n",
            " 'THRSP',\n",
            " 'THSD1',\n",
            " 'THSD4',\n",
            " 'THSD7A',\n",
            " 'THSD7B',\n",
            " 'THTPA',\n",
            " 'THUMPD1',\n",
            " 'THUMPD2',\n",
            " 'THUMPD3',\n",
            " 'THY1',\n",
            " 'THYN1',\n",
            " 'TIA1',\n",
            " 'TIAL1',\n",
            " 'TIAM1',\n",
            " 'TIAM2',\n",
            " 'TICAM1',\n",
            " 'TICAM2',\n",
            " 'TICRR',\n",
            " 'TIE1',\n",
            " 'TIFA',\n",
            " 'TIFAB',\n",
            " 'TIGAR',\n",
            " 'TIGD1',\n",
            " 'TIGD2',\n",
            " 'TIGD3',\n",
            " 'TIGD4',\n",
            " 'TIGD5',\n",
            " 'TIGD6',\n",
            " 'TIGD7',\n",
            " 'TIGIT',\n",
            " 'TIMD4',\n",
            " 'TIMELESS',\n",
            " 'TIMM10',\n",
            " 'TIMM10B',\n",
            " 'TIMM13',\n",
            " 'TIMM17A',\n",
            " 'TIMM17B',\n",
            " 'TIMM21',\n",
            " 'TIMM22',\n",
            " 'TIMM29',\n",
            " 'TIMM44',\n",
            " 'TIMM50',\n",
            " 'TIMM8A',\n",
            " 'TIMM8B',\n",
            " 'TIMM9',\n",
            " 'TIMMDC1',\n",
            " 'TIMP1',\n",
            " 'TIMP2',\n",
            " 'TIMP3',\n",
            " 'TIMP4',\n",
            " 'TINAG',\n",
            " 'TINAGL1',\n",
            " 'TINF2',\n",
            " 'TIPARP',\n",
            " 'TIPIN',\n",
            " 'TIPRL',\n",
            " 'TIRAP',\n",
            " 'TJAP1',\n",
            " 'TJP1',\n",
            " 'TJP2',\n",
            " 'TJP3',\n",
            " 'TK1',\n",
            " 'TK2',\n",
            " 'TKFC',\n",
            " 'TKT',\n",
            " 'TKTL1',\n",
            " 'TKTL2',\n",
            " 'TLCD1',\n",
            " 'TLCD2',\n",
            " 'TLCD3A',\n",
            " 'TLCD3B',\n",
            " 'TLCD4',\n",
            " 'TLCD5',\n",
            " 'TLDC2',\n",
            " 'TLE1',\n",
            " 'TLE2',\n",
            " 'TLE3',\n",
            " 'TLE4',\n",
            " 'TLE5',\n",
            " 'TLE6',\n",
            " 'TLK1',\n",
            " 'TLK2',\n",
            " 'TLL1',\n",
            " 'TLL2',\n",
            " 'TLN1',\n",
            " 'TLN2',\n",
            " 'TLNRD1',\n",
            " 'TLR1',\n",
            " 'TLR10',\n",
            " 'TLR2',\n",
            " 'TLR3',\n",
            " 'TLR4',\n",
            " 'TLR5',\n",
            " 'TLR6',\n",
            " 'TLR7',\n",
            " 'TLR8',\n",
            " 'TLR9',\n",
            " 'TLX2',\n",
            " 'TLX3',\n",
            " 'TM2D1',\n",
            " 'TM2D2',\n",
            " 'TM2D3',\n",
            " 'TM4SF1',\n",
            " 'TM4SF18',\n",
            " 'TM4SF19',\n",
            " 'TM4SF20',\n",
            " 'TM4SF4',\n",
            " 'TM4SF5',\n",
            " 'TM6SF1',\n",
            " 'TM7SF2',\n",
            " 'TM7SF3',\n",
            " 'TM9SF1',\n",
            " 'TM9SF2',\n",
            " 'TM9SF3',\n",
            " 'TM9SF4',\n",
            " 'TMA16',\n",
            " 'TMA7',\n",
            " 'TMBIM1',\n",
            " 'TMBIM4',\n",
            " 'TMBIM6',\n",
            " 'TMC1',\n",
            " 'TMC2',\n",
            " 'TMC3',\n",
            " 'TMC4',\n",
            " 'TMC5',\n",
            " 'TMC6',\n",
            " 'TMC7',\n",
            " 'TMC8',\n",
            " 'TMCC1',\n",
            " 'TMCC2',\n",
            " 'TMCC3',\n",
            " 'TMCO1',\n",
            " 'TMCO2',\n",
            " 'TMCO3',\n",
            " 'TMCO4',\n",
            " 'TMCO5A',\n",
            " 'TMCO6',\n",
            " 'TMED1',\n",
            " 'TMED10',\n",
            " 'TMED2',\n",
            " 'TMED3',\n",
            " 'TMED4',\n",
            " 'TMED5',\n",
            " 'TMED6',\n",
            " 'TMED7',\n",
            " 'TMED8',\n",
            " 'TMED9',\n",
            " 'TMEFF1',\n",
            " 'TMEFF2',\n",
            " 'TMEM100',\n",
            " 'TMEM101',\n",
            " 'TMEM102',\n",
            " 'TMEM104',\n",
            " 'TMEM105',\n",
            " 'TMEM106A',\n",
            " 'TMEM106B',\n",
            " 'TMEM106C',\n",
            " 'TMEM107',\n",
            " 'TMEM108',\n",
            " 'TMEM109',\n",
            " 'TMEM11',\n",
            " 'TMEM114',\n",
            " 'TMEM115',\n",
            " 'TMEM116',\n",
            " 'TMEM117',\n",
            " 'TMEM119',\n",
            " 'TMEM120A',\n",
            " 'TMEM120B',\n",
            " 'TMEM121',\n",
            " 'TMEM121B',\n",
            " 'TMEM123',\n",
            " 'TMEM125',\n",
            " 'TMEM126A',\n",
            " 'TMEM126B',\n",
            " 'TMEM127',\n",
            " 'TMEM128',\n",
            " 'TMEM129',\n",
            " 'TMEM130',\n",
            " 'TMEM131',\n",
            " 'TMEM131L',\n",
            " 'TMEM132A',\n",
            " 'TMEM132B',\n",
            " 'TMEM132C',\n",
            " 'TMEM132D',\n",
            " 'TMEM132E',\n",
            " 'TMEM132E-DT',\n",
            " 'TMEM134',\n",
            " 'TMEM135',\n",
            " 'TMEM138',\n",
            " 'TMEM139',\n",
            " 'TMEM140',\n",
            " 'TMEM141',\n",
            " 'TMEM143',\n",
            " 'TMEM144',\n",
            " 'TMEM145',\n",
            " 'TMEM147',\n",
            " 'TMEM14A',\n",
            " 'TMEM14B',\n",
            " 'TMEM14C',\n",
            " 'TMEM150A',\n",
            " 'TMEM150B',\n",
            " 'TMEM150C',\n",
            " 'TMEM151A',\n",
            " 'TMEM151B',\n",
            " 'TMEM154',\n",
            " 'TMEM156',\n",
            " 'TMEM158',\n",
            " 'TMEM160',\n",
            " 'TMEM161A',\n",
            " 'TMEM161B',\n",
            " 'TMEM163',\n",
            " 'TMEM164',\n",
            " 'TMEM165',\n",
            " 'TMEM167A',\n",
            " 'TMEM167B',\n",
            " 'TMEM168',\n",
            " 'TMEM169',\n",
            " 'TMEM17',\n",
            " 'TMEM170A',\n",
            " 'TMEM170B',\n",
            " 'TMEM171',\n",
            " 'TMEM174',\n",
            " 'TMEM175',\n",
            " 'TMEM176A',\n",
            " 'TMEM176B',\n",
            " 'TMEM177',\n",
            " 'TMEM178A',\n",
            " 'TMEM178B',\n",
            " 'TMEM179',\n",
            " 'TMEM179B',\n",
            " 'TMEM18',\n",
            " 'TMEM181',\n",
            " 'TMEM182',\n",
            " 'TMEM183A',\n",
            " 'TMEM184A',\n",
            " 'TMEM184B',\n",
            " 'TMEM184C',\n",
            " 'TMEM185A',\n",
            " 'TMEM185B',\n",
            " 'TMEM186',\n",
            " 'TMEM187',\n",
            " 'TMEM19',\n",
            " 'TMEM190',\n",
            " 'TMEM191B',\n",
            " 'TMEM191C',\n",
            " 'TMEM192',\n",
            " 'TMEM196',\n",
            " 'TMEM198',\n",
            " 'TMEM199',\n",
            " 'TMEM200A',\n",
            " 'TMEM200B',\n",
            " 'TMEM200C',\n",
            " 'TMEM201',\n",
            " 'TMEM202',\n",
            " 'TMEM203',\n",
            " 'TMEM204',\n",
            " 'TMEM205',\n",
            " 'TMEM207',\n",
            " 'TMEM208',\n",
            " 'TMEM209',\n",
            " 'TMEM210',\n",
            " 'TMEM212',\n",
            " 'TMEM213',\n",
            " 'TMEM214',\n",
            " 'TMEM215',\n",
            " 'TMEM216',\n",
            " 'TMEM217',\n",
            " 'TMEM218',\n",
            " 'TMEM219',\n",
            " 'TMEM220',\n",
            " 'TMEM221',\n",
            " 'TMEM222',\n",
            " 'TMEM223',\n",
            " 'TMEM225',\n",
            " 'TMEM229A',\n",
            " 'TMEM229B',\n",
            " 'TMEM230',\n",
            " 'TMEM231',\n",
            " 'TMEM232',\n",
            " 'TMEM233',\n",
            " 'TMEM234',\n",
            " 'TMEM235',\n",
            " 'TMEM236',\n",
            " 'TMEM237',\n",
            " 'TMEM238',\n",
            " 'TMEM239',\n",
            " 'TMEM240',\n",
            " 'TMEM241',\n",
            " 'TMEM242',\n",
            " 'TMEM243',\n",
            " 'TMEM244',\n",
            " 'TMEM245',\n",
            " 'TMEM247',\n",
            " 'TMEM248',\n",
            " 'TMEM249',\n",
            " 'TMEM25',\n",
            " 'TMEM250',\n",
            " 'TMEM252',\n",
            " 'TMEM253',\n",
            " 'TMEM254',\n",
            " 'TMEM255A',\n",
            " 'TMEM255B',\n",
            " 'TMEM256',\n",
            " 'TMEM258',\n",
            " 'TMEM259',\n",
            " 'TMEM26',\n",
            " 'TMEM260',\n",
            " 'TMEM263',\n",
            " 'TMEM266',\n",
            " 'TMEM267',\n",
            " 'TMEM268',\n",
            " 'TMEM270',\n",
            " 'TMEM273',\n",
            " 'TMEM30A',\n",
            " 'TMEM30B',\n",
            " 'TMEM31',\n",
            " 'TMEM33',\n",
            " 'TMEM35A',\n",
            " 'TMEM35B',\n",
            " 'TMEM37',\n",
            " 'TMEM38A',\n",
            " 'TMEM38B',\n",
            " 'TMEM39A',\n",
            " 'TMEM39B',\n",
            " 'TMEM40',\n",
            " 'TMEM41A',\n",
            " 'TMEM41B',\n",
            " 'TMEM42',\n",
            " 'TMEM43',\n",
            " 'TMEM44',\n",
            " 'TMEM45A',\n",
            " 'TMEM45B',\n",
            " 'TMEM47',\n",
            " 'TMEM50A',\n",
            " 'TMEM50B',\n",
            " 'TMEM51',\n",
            " 'TMEM51-AS2',\n",
            " 'TMEM52',\n",
            " 'TMEM52B',\n",
            " 'TMEM53',\n",
            " 'TMEM54',\n",
            " 'TMEM59',\n",
            " 'TMEM59L',\n",
            " 'TMEM60',\n",
            " 'TMEM61',\n",
            " 'TMEM62',\n",
            " 'TMEM63A',\n",
            " 'TMEM63B',\n",
            " 'TMEM63C',\n",
            " 'TMEM64',\n",
            " 'TMEM65',\n",
            " 'TMEM67',\n",
            " 'TMEM68',\n",
            " 'TMEM69',\n",
            " 'TMEM70',\n",
            " 'TMEM71',\n",
            " 'TMEM72',\n",
            " 'TMEM74',\n",
            " 'TMEM74B',\n",
            " 'TMEM79',\n",
            " 'TMEM80',\n",
            " 'TMEM81',\n",
            " 'TMEM82',\n",
            " 'TMEM86A',\n",
            " 'TMEM86B',\n",
            " 'TMEM87A',\n",
            " 'TMEM87B',\n",
            " 'TMEM88',\n",
            " 'TMEM88B',\n",
            " 'TMEM89',\n",
            " 'TMEM8B',\n",
            " 'TMEM9',\n",
            " 'TMEM91',\n",
            " 'TMEM92',\n",
            " 'TMEM94',\n",
            " 'TMEM95',\n",
            " 'TMEM97',\n",
            " 'TMEM98',\n",
            " 'TMEM9B',\n",
            " 'TMF1',\n",
            " 'TMIE',\n",
            " 'TMIGD1',\n",
            " 'TMIGD2',\n",
            " 'TMLHE',\n",
            " 'TMOD1',\n",
            " 'TMOD2',\n",
            " 'TMOD3',\n",
            " 'TMOD4',\n",
            " 'TMPO',\n",
            " 'TMPPE',\n",
            " 'TMPRSS11A',\n",
            " 'TMPRSS11B',\n",
            " 'TMPRSS11D',\n",
            " 'TMPRSS11E',\n",
            " 'TMPRSS11F',\n",
            " 'TMPRSS12',\n",
            " 'TMPRSS13',\n",
            " 'TMPRSS15',\n",
            " 'TMPRSS2',\n",
            " 'TMPRSS3',\n",
            " 'TMPRSS4',\n",
            " 'TMPRSS5',\n",
            " 'TMPRSS6',\n",
            " 'TMPRSS7',\n",
            " 'TMPRSS9',\n",
            " 'TMSB10',\n",
            " 'TMSB15A',\n",
            " 'TMSB15B',\n",
            " 'TMSB4X',\n",
            " 'TMSB4Y',\n",
            " 'TMTC1',\n",
            " 'TMTC2',\n",
            " 'TMTC3',\n",
            " 'TMTC4',\n",
            " 'TMUB1',\n",
            " 'TMUB2',\n",
            " 'TMX1',\n",
            " 'TMX2',\n",
            " 'TMX3',\n",
            " 'TMX4',\n",
            " 'TNC',\n",
            " 'TNF',\n",
            " 'TNFAIP1',\n",
            " 'TNFAIP2',\n",
            " 'TNFAIP3',\n",
            " 'TNFAIP6',\n",
            " 'TNFAIP8',\n",
            " 'TNFAIP8L1',\n",
            " 'TNFAIP8L2',\n",
            " 'TNFAIP8L3',\n",
            " 'TNFRSF10A',\n",
            " 'TNFRSF10B',\n",
            " 'TNFRSF10C',\n",
            " 'TNFRSF10D',\n",
            " 'TNFRSF11A',\n",
            " 'TNFRSF11B',\n",
            " 'TNFRSF12A',\n",
            " 'TNFRSF13B',\n",
            " 'TNFRSF13C',\n",
            " 'TNFRSF14',\n",
            " 'TNFRSF17',\n",
            " 'TNFRSF18',\n",
            " 'TNFRSF19',\n",
            " 'TNFRSF1A',\n",
            " 'TNFRSF1B',\n",
            " 'TNFRSF21',\n",
            " 'TNFRSF25',\n",
            " 'TNFRSF4',\n",
            " 'TNFRSF6B',\n",
            " 'TNFRSF8',\n",
            " 'TNFRSF9',\n",
            " 'TNFSF10',\n",
            " 'TNFSF11',\n",
            " 'TNFSF12',\n",
            " 'TNFSF13',\n",
            " 'TNFSF13B',\n",
            " 'TNFSF14',\n",
            " 'TNFSF15',\n",
            " 'TNFSF18',\n",
            " 'TNFSF4',\n",
            " 'TNFSF8',\n",
            " 'TNFSF9',\n",
            " 'TNIK',\n",
            " 'TNIP1',\n",
            " 'TNIP2',\n",
            " 'TNIP3',\n",
            " 'TNK1',\n",
            " 'TNK2',\n",
            " 'TNKS',\n",
            " 'TNKS1BP1',\n",
            " 'TNKS2',\n",
            " 'TNMD',\n",
            " 'TNN',\n",
            " 'TNNC1',\n",
            " 'TNNC2',\n",
            " 'TNNI1',\n",
            " 'TNNI2',\n",
            " 'TNNI3',\n",
            " 'TNNI3K',\n",
            " 'TNNT1',\n",
            " 'TNNT2',\n",
            " 'TNNT3',\n",
            " 'TNP1',\n",
            " 'TNP2',\n",
            " 'TNPO1',\n",
            " 'TNPO2',\n",
            " 'TNPO3',\n",
            " 'TNR',\n",
            " 'TNRC18',\n",
            " 'TNRC6A',\n",
            " 'TNRC6B',\n",
            " 'TNRC6C',\n",
            " 'TNS1',\n",
            " 'TNS2',\n",
            " 'TNS3',\n",
            " 'TNS4',\n",
            " 'TNXB',\n",
            " 'TOB1',\n",
            " 'TOB2',\n",
            " 'TOE1',\n",
            " 'TOGARAM1',\n",
            " 'TOGARAM2',\n",
            " 'TOLLIP',\n",
            " 'TOM1',\n",
            " 'TOM1L1',\n",
            " 'TOM1L2',\n",
            " 'TOMM20',\n",
            " 'TOMM20L',\n",
            " 'TOMM22',\n",
            " 'TOMM34',\n",
            " 'TOMM40',\n",
            " 'TOMM40L',\n",
            " 'TOMM5',\n",
            " 'TOMM6',\n",
            " 'TOMM7',\n",
            " 'TOMM70',\n",
            " 'TONSL',\n",
            " 'TOP1',\n",
            " 'TOP1MT',\n",
            " 'TOP2A',\n",
            " 'TOP2B',\n",
            " 'TOP3A',\n",
            " 'TOP3B',\n",
            " 'TOPAZ1',\n",
            " 'TOPBP1',\n",
            " 'TOPORS',\n",
            " 'TOR1A',\n",
            " 'TOR1AIP1',\n",
            " 'TOR1AIP2',\n",
            " 'TOR1B',\n",
            " 'TOR2A',\n",
            " 'TOR3A',\n",
            " 'TOR4A',\n",
            " 'TOX',\n",
            " 'TOX2',\n",
            " 'TOX3',\n",
            " 'TOX4',\n",
            " 'TP53',\n",
            " 'TP53AIP1',\n",
            " 'TP53BP1',\n",
            " 'TP53BP2',\n",
            " 'TP53I11',\n",
            " 'TP53I13',\n",
            " 'TP53I3',\n",
            " 'TP53INP1',\n",
            " 'TP53INP2',\n",
            " 'TP53RK',\n",
            " 'TP53TG3C',\n",
            " 'TP53TG5',\n",
            " 'TP63',\n",
            " 'TP73',\n",
            " 'TPBG',\n",
            " 'TPBGL',\n",
            " 'TPCN1',\n",
            " 'TPCN2',\n",
            " 'TPD52',\n",
            " 'TPD52L1',\n",
            " 'TPD52L2',\n",
            " 'TPD52L3',\n",
            " 'TPGS1',\n",
            " 'TPGS2',\n",
            " 'TPH1',\n",
            " 'TPH2',\n",
            " 'TPI1',\n",
            " 'TPK1',\n",
            " 'TPM1',\n",
            " 'TPM3',\n",
            " 'TPM4',\n",
            " 'TPMT',\n",
            " 'TPO',\n",
            " 'TPP1',\n",
            " 'TPP2',\n",
            " 'TPPP',\n",
            " 'TPPP2',\n",
            " 'TPPP3',\n",
            " 'TPR',\n",
            " 'TPRA1',\n",
            " 'TPRG1',\n",
            " 'TPRG1L',\n",
            " 'TPRKB',\n",
            " 'TPRN',\n",
            " 'TPRX1',\n",
            " 'TPSAB1',\n",
            " 'TPSB2',\n",
            " 'TPSD1',\n",
            " 'TPSG1',\n",
            " 'TPST1',\n",
            " 'TPST2',\n",
            " 'TPT1',\n",
            " 'TPTE',\n",
            " 'TPTE2',\n",
            " 'TPX2',\n",
            " 'TRA2A',\n",
            " 'TRA2B',\n",
            " 'TRABD',\n",
            " 'TRABD2A',\n",
            " 'TRABD2B',\n",
            " 'TRADD',\n",
            " 'TRAF1',\n",
            " 'TRAF2',\n",
            " 'TRAF3',\n",
            " 'TRAF3IP1',\n",
            " 'TRAF3IP2',\n",
            " 'TRAF3IP3',\n",
            " 'TRAF4',\n",
            " 'TRAF5',\n",
            " 'TRAF6',\n",
            " 'TRAF7',\n",
            " 'TRAFD1',\n",
            " 'TRAIP',\n",
            " 'TRAK1',\n",
            " 'TRAK2',\n",
            " 'TRAM1',\n",
            " 'TRAM1L1',\n",
            " 'TRAM2',\n",
            " 'TRANK1',\n",
            " 'TRAP1',\n",
            " 'TRAPPC1',\n",
            " 'TRAPPC10',\n",
            " 'TRAPPC11',\n",
            " 'TRAPPC12',\n",
            " 'TRAPPC13',\n",
            " 'TRAPPC14',\n",
            " 'TRAPPC2',\n",
            " 'TRAPPC2L',\n",
            " 'TRAPPC3',\n",
            " 'TRAPPC3L',\n",
            " 'TRAPPC4',\n",
            " 'TRAPPC5',\n",
            " 'TRAPPC6A',\n",
            " 'TRAPPC6B',\n",
            " 'TRAPPC8',\n",
            " 'TRAPPC9',\n",
            " 'TRARG1',\n",
            " 'TRAT1',\n",
            " 'TRDMT1',\n",
            " 'TRDN',\n",
            " 'TREH',\n",
            " 'TREM1',\n",
            " 'TREM2',\n",
            " 'TREML1',\n",
            " 'TREML2',\n",
            " 'TREML4',\n",
            " 'TRERF1',\n",
            " 'TREX1',\n",
            " 'TRH',\n",
            " 'TRHDE',\n",
            " 'TRHR',\n",
            " 'TRIAP1',\n",
            " 'TRIB1',\n",
            " 'TRIB2',\n",
            " 'TRIB3',\n",
            " 'TRIL',\n",
            " 'TRIM10',\n",
            " 'TRIM11',\n",
            " 'TRIM13',\n",
            " 'TRIM14',\n",
            " 'TRIM15',\n",
            " 'TRIM16',\n",
            " 'TRIM16L',\n",
            " 'TRIM17',\n",
            " 'TRIM2',\n",
            " 'TRIM21',\n",
            " 'TRIM22',\n",
            " 'TRIM23',\n",
            " 'TRIM24',\n",
            " 'TRIM25',\n",
            " 'TRIM26',\n",
            " 'TRIM27',\n",
            " 'TRIM28',\n",
            " 'TRIM29',\n",
            " 'TRIM3',\n",
            " 'TRIM31',\n",
            " 'TRIM32',\n",
            " 'TRIM33',\n",
            " 'TRIM34',\n",
            " 'TRIM35',\n",
            " 'TRIM36',\n",
            " 'TRIM37',\n",
            " 'TRIM38',\n",
            " 'TRIM39',\n",
            " 'TRIM4',\n",
            " 'TRIM40',\n",
            " 'TRIM41',\n",
            " 'TRIM42',\n",
            " 'TRIM43',\n",
            " 'TRIM44',\n",
            " 'TRIM45',\n",
            " 'TRIM46',\n",
            " 'TRIM47',\n",
            " 'TRIM49',\n",
            " 'TRIM49C',\n",
            " 'TRIM49D1',\n",
            " 'TRIM5',\n",
            " 'TRIM50',\n",
            " 'TRIM51',\n",
            " 'TRIM52',\n",
            " 'TRIM54',\n",
            " 'TRIM55',\n",
            " 'TRIM56',\n",
            " 'TRIM58',\n",
            " 'TRIM59',\n",
            " 'TRIM6',\n",
            " 'TRIM60',\n",
            " 'TRIM61',\n",
            " 'TRIM62',\n",
            " 'TRIM63',\n",
            " 'TRIM65',\n",
            " 'TRIM66',\n",
            " 'TRIM67',\n",
            " 'TRIM68',\n",
            " 'TRIM69',\n",
            " 'TRIM7',\n",
            " 'TRIM71',\n",
            " 'TRIM72',\n",
            " 'TRIM73',\n",
            " 'TRIM74',\n",
            " 'TRIM77',\n",
            " 'TRIM8',\n",
            " 'TRIM9',\n",
            " 'TRIML1',\n",
            " 'TRIML2',\n",
            " 'TRIO',\n",
            " 'TRIOBP',\n",
            " 'TRIP10',\n",
            " 'TRIP11',\n",
            " 'TRIP12',\n",
            " 'TRIP13',\n",
            " 'TRIP4',\n",
            " 'TRIP6',\n",
            " 'TRIQK',\n",
            " 'TRIR',\n",
            " 'TRIT1',\n",
            " 'TRMO',\n",
            " 'TRMT1',\n",
            " 'TRMT10A',\n",
            " 'TRMT10B',\n",
            " 'TRMT10C',\n",
            " 'TRMT11',\n",
            " 'TRMT112',\n",
            " 'TRMT12',\n",
            " 'TRMT13',\n",
            " 'TRMT1L',\n",
            " 'TRMT2A',\n",
            " 'TRMT2B',\n",
            " 'TRMT44',\n",
            " 'TRMT5',\n",
            " 'TRMT6',\n",
            " 'TRMT61A',\n",
            " 'TRMT61B',\n",
            " 'TRMT9B',\n",
            " 'TRMU',\n",
            " 'TRNAU1AP',\n",
            " 'TRNP1',\n",
            " 'TRNT1',\n",
            " 'TRO',\n",
            " 'TROAP',\n",
            " 'TRPA1',\n",
            " 'TRPC1',\n",
            " 'TRPC3',\n",
            " 'TRPC4',\n",
            " 'TRPC4AP',\n",
            " 'TRPC5',\n",
            " 'TRPC5OS',\n",
            " 'TRPC6',\n",
            " 'TRPC7',\n",
            " 'TRPM1',\n",
            " 'TRPM2',\n",
            " 'TRPM3',\n",
            " 'TRPM4',\n",
            " 'TRPM5',\n",
            " 'TRPM6',\n",
            " 'TRPM7',\n",
            " 'TRPM8',\n",
            " 'TRPS1',\n",
            " 'TRPT1',\n",
            " 'TRPV1',\n",
            " 'TRPV2',\n",
            " 'TRPV3',\n",
            " 'TRPV4',\n",
            " 'TRPV5',\n",
            " 'TRPV6',\n",
            " 'TRRAP',\n",
            " 'TRUB1',\n",
            " 'TRUB2',\n",
            " 'TSACC',\n",
            " 'TSBP1',\n",
            " 'TSC1',\n",
            " 'TSC2',\n",
            " 'TSC22D1',\n",
            " 'TSC22D2',\n",
            " 'TSC22D3',\n",
            " 'TSC22D4',\n",
            " 'TSEN15',\n",
            " 'TSEN2',\n",
            " 'TSEN34',\n",
            " 'TSEN54',\n",
            " 'TSFM',\n",
            " 'TSG101',\n",
            " 'TSGA10',\n",
            " 'TSGA10IP',\n",
            " 'TSGA13',\n",
            " 'TSHB',\n",
            " 'TSHR',\n",
            " 'TSHZ1',\n",
            " 'TSHZ2',\n",
            " 'TSHZ3',\n",
            " 'TSKS',\n",
            " 'TSKU',\n",
            " 'TSLP',\n",
            " 'TSN',\n",
            " 'TSNARE1',\n",
            " 'TSNAX',\n",
            " 'TSNAXIP1',\n",
            " 'TSPAN1',\n",
            " 'TSPAN10',\n",
            " 'TSPAN11',\n",
            " 'TSPAN12',\n",
            " 'TSPAN13',\n",
            " 'TSPAN14',\n",
            " 'TSPAN15',\n",
            " 'TSPAN16',\n",
            " 'TSPAN17',\n",
            " 'TSPAN18',\n",
            " 'TSPAN19',\n",
            " 'TSPAN2',\n",
            " 'TSPAN3',\n",
            " 'TSPAN31',\n",
            " 'TSPAN32',\n",
            " 'TSPAN33',\n",
            " 'TSPAN4',\n",
            " 'TSPAN5',\n",
            " 'TSPAN6',\n",
            " 'TSPAN7',\n",
            " 'TSPAN8',\n",
            " 'TSPAN9',\n",
            " 'TSPEAR',\n",
            " 'TSPO',\n",
            " 'TSPO2',\n",
            " 'TSPOAP1',\n",
            " 'TSPY1',\n",
            " 'TSPYL1',\n",
            " 'TSPYL2',\n",
            " 'TSPYL4',\n",
            " 'TSPYL5',\n",
            " 'TSPYL6',\n",
            " 'TSR1',\n",
            " 'TSR2',\n",
            " 'TSR3',\n",
            " 'TSSC4',\n",
            " 'TSSK1B',\n",
            " 'TSSK2',\n",
            " 'TSSK3',\n",
            " 'TSSK4',\n",
            " 'TSSK6',\n",
            " 'TST',\n",
            " 'TSTD1',\n",
            " 'TSTD2',\n",
            " 'TSTD3',\n",
            " 'TTBK1',\n",
            " 'TTBK2',\n",
            " 'TTC1',\n",
            " 'TTC12',\n",
            " 'TTC13',\n",
            " 'TTC14',\n",
            " 'TTC16',\n",
            " 'TTC17',\n",
            " 'TTC19',\n",
            " 'TTC21A',\n",
            " 'TTC21B',\n",
            " 'TTC22',\n",
            " 'TTC23',\n",
            " 'TTC23L',\n",
            " 'TTC24',\n",
            " 'TTC27',\n",
            " 'TTC28',\n",
            " 'TTC29',\n",
            " 'TTC3',\n",
            " 'TTC31',\n",
            " 'TTC32',\n",
            " 'TTC33',\n",
            " 'TTC34',\n",
            " 'TTC36',\n",
            " 'TTC38',\n",
            " 'TTC39A',\n",
            " 'TTC39B',\n",
            " 'TTC39C',\n",
            " 'TTC4',\n",
            " 'TTC5',\n",
            " 'TTC6',\n",
            " 'TTC7B',\n",
            " 'TTC8',\n",
            " 'TTC9',\n",
            " 'TTC9B',\n",
            " 'TTC9C',\n",
            " 'TTF1',\n",
            " 'TTF2',\n",
            " 'TTI1',\n",
            " 'TTI2',\n",
            " 'TTK',\n",
            " 'TTL',\n",
            " 'TTLL1',\n",
            " 'TTLL10',\n",
            " 'TTLL11',\n",
            " 'TTLL12',\n",
            " 'TTLL13',\n",
            " 'TTLL2',\n",
            " 'TTLL3',\n",
            " 'TTLL4',\n",
            " 'TTLL5',\n",
            " 'TTLL6',\n",
            " 'TTLL7',\n",
            " 'TTLL8',\n",
            " 'TTLL9',\n",
            " 'TTN',\n",
            " 'TTPA',\n",
            " 'TTPAL',\n",
            " 'TTR',\n",
            " 'TTYH1',\n",
            " 'TTYH2',\n",
            " 'TTYH3',\n",
            " 'TUB',\n",
            " 'TUBA1A',\n",
            " 'TUBA1B',\n",
            " 'TUBA1C',\n",
            " 'TUBA3C',\n",
            " 'TUBA3D',\n",
            " 'TUBA3E',\n",
            " 'TUBA4A',\n",
            " 'TUBA8',\n",
            " 'TUBAL3',\n",
            " 'TUBB',\n",
            " 'TUBB1',\n",
            " 'TUBB2A',\n",
            " 'TUBB2B',\n",
            " 'TUBB3',\n",
            " 'TUBB4A',\n",
            " 'TUBB4B',\n",
            " 'TUBB6',\n",
            " 'TUBB8',\n",
            " 'TUBD1',\n",
            " 'TUBE1',\n",
            " 'TUBG1',\n",
            " 'TUBG2',\n",
            " 'TUBGCP2',\n",
            " 'TUBGCP3',\n",
            " 'TUBGCP4',\n",
            " 'TUBGCP5',\n",
            " 'TUBGCP6',\n",
            " 'TUFM',\n",
            " 'TUFT1',\n",
            " 'TULP1',\n",
            " 'TULP2',\n",
            " 'TULP3',\n",
            " 'TULP4',\n",
            " 'TUSC1',\n",
            " 'TUSC2',\n",
            " 'TUSC3',\n",
            " 'TUT1',\n",
            " 'TUT4',\n",
            " 'TUT7',\n",
            " 'TVP23A',\n",
            " 'TVP23B',\n",
            " 'TVP23C',\n",
            " 'TWF1',\n",
            " 'TWF2',\n",
            " 'TWIST1',\n",
            " 'TWIST2',\n",
            " 'TWNK',\n",
            " 'TWSG1',\n",
            " 'TXK',\n",
            " 'TXLNA',\n",
            " 'TXLNB',\n",
            " 'TXLNG',\n",
            " 'TXN',\n",
            " 'TXN2',\n",
            " 'TXNDC11',\n",
            " 'TXNDC12',\n",
            " 'TXNDC15',\n",
            " 'TXNDC16',\n",
            " 'TXNDC17',\n",
            " 'TXNDC2',\n",
            " 'TXNDC5',\n",
            " 'TXNDC8',\n",
            " 'TXNDC9',\n",
            " 'TXNIP',\n",
            " 'TXNL1',\n",
            " 'TXNL4A',\n",
            " 'TXNL4B',\n",
            " 'TXNRD1',\n",
            " 'TXNRD2',\n",
            " 'TXNRD3',\n",
            " 'TYK2',\n",
            " 'TYMP',\n",
            " 'TYR',\n",
            " 'TYRO3',\n",
            " 'TYROBP',\n",
            " 'TYRP1',\n",
            " 'TYSND1',\n",
            " 'TYW1',\n",
            " 'TYW1B',\n",
            " 'TYW3',\n",
            " 'TYW5',\n",
            " 'U2AF1',\n",
            " 'U2AF1L4',\n",
            " 'U2AF2',\n",
            " 'U2SURP',\n",
            " 'UACA',\n",
            " 'UAP1',\n",
            " 'UAP1L1',\n",
            " 'UBA1',\n",
            " 'UBA2',\n",
            " 'UBA3',\n",
            " 'UBA5',\n",
            " 'UBA52',\n",
            " 'UBA6',\n",
            " 'UBA7',\n",
            " 'UBAC1',\n",
            " 'UBAC2',\n",
            " 'UBALD1',\n",
            " 'UBALD2',\n",
            " 'UBAP1',\n",
            " 'UBAP1L',\n",
            " 'UBAP2',\n",
            " 'UBAP2L',\n",
            " 'UBASH3A',\n",
            " 'UBASH3B',\n",
            " 'UBB',\n",
            " 'UBC',\n",
            " 'UBD',\n",
            " 'UBE2A',\n",
            " 'UBE2B',\n",
            " 'UBE2C',\n",
            " 'UBE2D1',\n",
            " 'UBE2D2',\n",
            " 'UBE2D3',\n",
            " 'UBE2D4',\n",
            " 'UBE2E1',\n",
            " 'UBE2E2',\n",
            " 'UBE2E3',\n",
            " 'UBE2F',\n",
            " 'UBE2G1',\n",
            " 'UBE2G2',\n",
            " 'UBE2H',\n",
            " 'UBE2I',\n",
            " 'UBE2J1',\n",
            " 'UBE2J2',\n",
            " 'UBE2K',\n",
            " 'UBE2L3',\n",
            " 'UBE2L6',\n",
            " 'UBE2M',\n",
            " 'UBE2N',\n",
            " 'UBE2NL',\n",
            " 'UBE2O',\n",
            " 'UBE2Q1',\n",
            " 'UBE2Q2',\n",
            " 'UBE2QL1',\n",
            " 'UBE2R2',\n",
            " 'UBE2S',\n",
            " 'UBE2T',\n",
            " 'UBE2U',\n",
            " 'UBE2V1',\n",
            " 'UBE2V2',\n",
            " 'UBE2W',\n",
            " 'UBE2Z',\n",
            " 'UBE3A',\n",
            " 'UBE3B',\n",
            " 'UBE3C',\n",
            " 'UBE3D',\n",
            " 'UBE4A',\n",
            " 'UBE4B',\n",
            " 'UBFD1',\n",
            " 'UBIAD1',\n",
            " 'UBL3',\n",
            " 'UBL4A',\n",
            " 'UBL4B',\n",
            " 'UBL5',\n",
            " 'UBL7',\n",
            " 'UBLCP1',\n",
            " 'UBN1',\n",
            " 'UBN2',\n",
            " 'UBP1',\n",
            " 'UBQLN1',\n",
            " 'UBQLN2',\n",
            " 'UBQLN3',\n",
            " 'UBQLN4',\n",
            " 'UBQLNL',\n",
            " 'UBR1',\n",
            " 'UBR2',\n",
            " 'UBR3',\n",
            " 'UBR4',\n",
            " 'UBR5',\n",
            " 'UBR7',\n",
            " 'UBTD1',\n",
            " 'UBTD2',\n",
            " 'UBTF',\n",
            " 'UBXN1',\n",
            " 'UBXN10',\n",
            " 'UBXN11',\n",
            " 'UBXN2A',\n",
            " 'UBXN2B',\n",
            " 'UBXN4',\n",
            " 'UBXN6',\n",
            " 'UBXN7',\n",
            " 'UBXN8',\n",
            " 'UCHL1',\n",
            " 'UCHL3',\n",
            " 'UCHL5',\n",
            " 'UCK1',\n",
            " 'UCK2',\n",
            " 'UCKL1',\n",
            " 'UCMA',\n",
            " 'UCN',\n",
            " 'UCN2',\n",
            " 'UCN3',\n",
            " 'UCP1',\n",
            " 'UCP3',\n",
            " 'UEVLD',\n",
            " 'UFC1',\n",
            " 'UFD1',\n",
            " 'UFL1',\n",
            " 'UFM1',\n",
            " 'UFSP1',\n",
            " 'UFSP2',\n",
            " 'UGCG',\n",
            " 'UGDH',\n",
            " 'UGGT1',\n",
            " 'UGGT2',\n",
            " 'UGP2',\n",
            " 'UGT1A1',\n",
            " 'UGT1A10',\n",
            " 'UGT1A3',\n",
            " 'UGT1A4',\n",
            " 'UGT1A5',\n",
            " 'UGT1A6',\n",
            " 'UGT1A7',\n",
            " 'UGT1A8',\n",
            " 'UGT2A3',\n",
            " 'UGT2B15',\n",
            " 'UGT2B17',\n",
            " 'UGT2B4',\n",
            " 'UGT2B7',\n",
            " 'UGT3A1',\n",
            " 'UGT8',\n",
            " 'UHMK1',\n",
            " 'UHRF1',\n",
            " 'UHRF2',\n",
            " 'UIMC1',\n",
            " 'ULBP1',\n",
            " 'ULBP2',\n",
            " 'ULBP3',\n",
            " 'ULK1',\n",
            " 'ULK2',\n",
            " 'ULK3',\n",
            " 'ULK4',\n",
            " 'UMOD',\n",
            " 'UMODL1',\n",
            " 'UMPS',\n",
            " 'UNC119',\n",
            " 'UNC119B',\n",
            " 'UNC13A',\n",
            " 'UNC13B',\n",
            " 'UNC13C',\n",
            " 'UNC13D',\n",
            " 'UNC45A',\n",
            " 'UNC45B',\n",
            " 'UNC50',\n",
            " 'UNC5A',\n",
            " 'UNC5B',\n",
            " 'UNC5C',\n",
            " 'UNC5CL',\n",
            " 'UNC5D',\n",
            " 'UNC79',\n",
            " 'UNC80',\n",
            " 'UNC93A',\n",
            " 'UNC93B1',\n",
            " 'UNCX',\n",
            " 'UNG',\n",
            " 'UNK',\n",
            " 'UNKL',\n",
            " 'UPB1',\n",
            " 'UPF1',\n",
            " 'UPF2',\n",
            " 'UPF3A',\n",
            " 'UPF3B',\n",
            " 'UPK1A',\n",
            " 'UPK1B',\n",
            " 'UPK2',\n",
            " 'UPK3A',\n",
            " 'UPK3B',\n",
            " 'UPK3BL1',\n",
            " 'UPP1',\n",
            " 'UPP2',\n",
            " 'UPRT',\n",
            " 'UQCC1',\n",
            " 'UQCC2',\n",
            " 'UQCC3',\n",
            " 'UQCC4',\n",
            " 'UQCC5',\n",
            " 'UQCC6',\n",
            " 'UQCR10',\n",
            " 'UQCR11',\n",
            " 'UQCRB',\n",
            " 'UQCRC1',\n",
            " 'UQCRC2',\n",
            " 'UQCRFS1',\n",
            " 'UQCRH',\n",
            " 'UQCRHL',\n",
            " 'UQCRQ',\n",
            " 'URAD',\n",
            " 'URB1',\n",
            " 'URB2',\n",
            " 'URGCP',\n",
            " 'URI1',\n",
            " 'URM1',\n",
            " 'UROC1',\n",
            " 'UROD',\n",
            " 'UROS',\n",
            " 'USB1',\n",
            " 'USE1',\n",
            " 'USF1',\n",
            " 'USF2',\n",
            " 'USF3',\n",
            " 'USH1C',\n",
            " 'USH1G',\n",
            " 'USH2A',\n",
            " 'USHBP1',\n",
            " 'USO1',\n",
            " 'USP1',\n",
            " 'USP10',\n",
            " 'USP11',\n",
            " 'USP12',\n",
            " 'USP13',\n",
            " 'USP14',\n",
            " 'USP15',\n",
            " 'USP16',\n",
            " 'USP18',\n",
            " 'USP19',\n",
            " 'USP2',\n",
            " 'USP20',\n",
            " 'USP21',\n",
            " 'USP22',\n",
            " 'USP24',\n",
            " 'USP25',\n",
            " 'USP26',\n",
            " 'USP27X',\n",
            " 'USP28',\n",
            " 'USP29',\n",
            " 'USP3',\n",
            " 'USP30',\n",
            " 'USP31',\n",
            " 'USP32',\n",
            " 'USP33',\n",
            " 'USP34',\n",
            " 'USP35',\n",
            " 'USP36',\n",
            " 'USP37',\n",
            " 'USP38',\n",
            " 'USP39',\n",
            " 'USP4',\n",
            " 'USP40',\n",
            " 'USP42',\n",
            " 'USP43',\n",
            " 'USP44',\n",
            " 'USP45',\n",
            " 'USP46',\n",
            " 'USP47',\n",
            " 'USP48',\n",
            " 'USP49',\n",
            " 'USP5',\n",
            " 'USP50',\n",
            " 'USP51',\n",
            " 'USP53',\n",
            " 'USP54',\n",
            " 'USP6',\n",
            " 'USP6NL',\n",
            " 'USP7',\n",
            " 'USP8',\n",
            " 'USP9X',\n",
            " 'USP9Y',\n",
            " 'USPL1',\n",
            " 'UST',\n",
            " 'UTF1',\n",
            " 'UTP11',\n",
            " 'UTP14A',\n",
            " 'UTP15',\n",
            " 'UTP18',\n",
            " 'UTP20',\n",
            " 'UTP23',\n",
            " 'UTP25',\n",
            " 'UTP3',\n",
            " 'UTP4',\n",
            " 'UTP6',\n",
            " 'UTRN',\n",
            " 'UTS2',\n",
            " 'UTS2B',\n",
            " 'UTS2R',\n",
            " 'UTY',\n",
            " 'UVRAG',\n",
            " 'UVSSA',\n",
            " 'UXS1',\n",
            " 'UXT',\n",
            " 'VAC14',\n",
            " 'VAMP1',\n",
            " 'VAMP2',\n",
            " 'VAMP3',\n",
            " 'VAMP4',\n",
            " 'VAMP5',\n",
            " 'VAMP7',\n",
            " 'VAMP8',\n",
            " 'VANGL1',\n",
            " 'VANGL2',\n",
            " 'VAPA',\n",
            " 'VAPB',\n",
            " 'VARS1',\n",
            " 'VARS2',\n",
            " 'VASH1',\n",
            " 'VASH2',\n",
            " 'VASN',\n",
            " 'VASP',\n",
            " 'VAT1',\n",
            " 'VAT1L',\n",
            " 'VAV1',\n",
            " 'VAV2',\n",
            " 'VAV3',\n",
            " 'VAX1',\n",
            " 'VAX2',\n",
            " 'VBP1',\n",
            " 'VCAM1',\n",
            " 'VCAN',\n",
            " 'VCL',\n",
            " 'VCP',\n",
            " 'VCPIP1',\n",
            " 'VCPKMT',\n",
            " 'VCX',\n",
            " 'VDAC1',\n",
            " 'VDAC2',\n",
            " 'VDAC3',\n",
            " 'VDR',\n",
            " 'VEGFA',\n",
            " 'VEGFB',\n",
            " 'VEGFC',\n",
            " 'VEGFD',\n",
            " 'VENTX',\n",
            " 'VEPH1',\n",
            " 'VEZF1',\n",
            " 'VEZT',\n",
            " 'VGF',\n",
            " 'VGLL1',\n",
            " 'VGLL2',\n",
            " 'VGLL3',\n",
            " 'VGLL4',\n",
            " 'VHL',\n",
            " 'VHLL',\n",
            " 'VIL1',\n",
            " 'VILL',\n",
            " 'VIM',\n",
            " 'VIP',\n",
            " 'VIPAS39',\n",
            " 'VIPR1',\n",
            " 'VIPR2',\n",
            " 'VIRMA',\n",
            " 'VIT',\n",
            " 'VKORC1',\n",
            " 'VKORC1L1',\n",
            " 'VLDLR',\n",
            " 'VMA21',\n",
            " 'VMAC',\n",
            " 'VMO1',\n",
            " 'VMP1',\n",
            " 'VN1R1',\n",
            " 'VN1R2',\n",
            " 'VN1R4',\n",
            " 'VNN1',\n",
            " 'VNN2',\n",
            " 'VOPP1',\n",
            " 'VPREB1',\n",
            " 'VPREB3',\n",
            " 'VPS11',\n",
            " 'VPS13A',\n",
            " 'VPS13B',\n",
            " 'VPS13C',\n",
            " 'VPS13D',\n",
            " 'VPS16',\n",
            " 'VPS18',\n",
            " 'VPS25',\n",
            " 'VPS26A',\n",
            " 'VPS26B',\n",
            " 'VPS26C',\n",
            " 'VPS28',\n",
            " 'VPS33A',\n",
            " 'VPS33B',\n",
            " 'VPS35',\n",
            " 'VPS35L',\n",
            " 'VPS36',\n",
            " 'VPS37A',\n",
            " 'VPS37B',\n",
            " 'VPS37C',\n",
            " 'VPS37D',\n",
            " 'VPS39',\n",
            " 'VPS41',\n",
            " 'VPS45',\n",
            " 'VPS4A',\n",
            " 'VPS4B',\n",
            " 'VPS50',\n",
            " 'VPS51',\n",
            " 'VPS52',\n",
            " 'VPS53',\n",
            " 'VPS54',\n",
            " 'VPS72',\n",
            " 'VPS8',\n",
            " 'VPS9D1',\n",
            " 'VRK1',\n",
            " 'VRK2',\n",
            " 'VRK3',\n",
            " 'VRTN',\n",
            " 'VSIG1',\n",
            " 'VSIG10',\n",
            " 'VSIG10L',\n",
            " 'VSIG2',\n",
            " 'VSIG4',\n",
            " 'VSIG8',\n",
            " 'VSIR',\n",
            " 'VSNL1',\n",
            " 'VSTM1',\n",
            " 'VSTM2A',\n",
            " 'VSTM2B',\n",
            " 'VSTM2L',\n",
            " 'VSTM4',\n",
            " 'VSTM5',\n",
            " 'VSX1',\n",
            " 'VSX2',\n",
            " 'VTA1',\n",
            " 'VTCN1',\n",
            " 'VTI1A',\n",
            " 'VTI1B',\n",
            " 'VWA1',\n",
            " 'VWA2',\n",
            " 'VWA3A',\n",
            " 'VWA3B',\n",
            " 'VWA5A',\n",
            " 'VWA5B1',\n",
            " 'VWA5B2',\n",
            " 'VWA7',\n",
            " 'VWA8',\n",
            " 'VWC2',\n",
            " 'VWC2L',\n",
            " 'VWCE',\n",
            " 'VWDE',\n",
            " 'VWF',\n",
            " 'VXN',\n",
            " 'WAC',\n",
            " 'WAPL',\n",
            " 'WARS1',\n",
            " 'WARS2',\n",
            " 'WAS',\n",
            " 'WASF1',\n",
            " 'WASF2',\n",
            " 'WASF3',\n",
            " 'WASHC2A',\n",
            " 'WASHC2C',\n",
            " 'WASHC3',\n",
            " 'WASHC4',\n",
            " 'WASHC5',\n",
            " 'WASL',\n",
            " 'WBP1',\n",
            " 'WBP11',\n",
            " 'WBP1L',\n",
            " 'WBP2',\n",
            " 'WBP2NL',\n",
            " 'WBP4',\n",
            " 'WDCP',\n",
            " 'WDFY1',\n",
            " 'WDFY2',\n",
            " 'WDFY3',\n",
            " 'WDFY4',\n",
            " 'WDHD1',\n",
            " 'WDPCP',\n",
            " 'WDR1',\n",
            " 'WDR11',\n",
            " 'WDR12',\n",
            " 'WDR13',\n",
            " 'WDR17',\n",
            " 'WDR18',\n",
            " 'WDR19',\n",
            " 'WDR20',\n",
            " 'WDR24',\n",
            " 'WDR25',\n",
            " 'WDR26',\n",
            " 'WDR27',\n",
            " 'WDR3',\n",
            " 'WDR31',\n",
            " 'WDR33',\n",
            " 'WDR35',\n",
            " 'WDR36',\n",
            " 'WDR37',\n",
            " 'WDR38',\n",
            " 'WDR4',\n",
            " 'WDR41',\n",
            " 'WDR43',\n",
            " 'WDR44',\n",
            " 'WDR45B',\n",
            " 'WDR46',\n",
            " 'WDR47',\n",
            " 'WDR48',\n",
            " 'WDR49',\n",
            " 'WDR5',\n",
            " 'WDR53',\n",
            " 'WDR54',\n",
            " 'WDR55',\n",
            " 'WDR59',\n",
            " 'WDR5B',\n",
            " 'WDR6',\n",
            " 'WDR62',\n",
            " 'WDR64',\n",
            " 'WDR7',\n",
            " 'WDR70',\n",
            " 'WDR72',\n",
            " 'WDR73',\n",
            " 'WDR74',\n",
            " 'WDR75',\n",
            " 'WDR76',\n",
            " 'WDR77',\n",
            " 'WDR81',\n",
            " 'WDR82',\n",
            " 'WDR83',\n",
            " 'WDR83OS',\n",
            " 'WDR86',\n",
            " 'WDR87',\n",
            " 'WDR88',\n",
            " 'WDR89',\n",
            " 'WDR90',\n",
            " 'WDR91',\n",
            " 'WDR93',\n",
            " 'WDSUB1',\n",
            " 'WDTC1',\n",
            " 'WEE1',\n",
            " 'WEE2',\n",
            " 'WFDC1',\n",
            " 'WFDC10A',\n",
            " 'WFDC10B',\n",
            " 'WFDC11',\n",
            " 'WFDC12',\n",
            " 'WFDC13',\n",
            " 'WFDC2',\n",
            " 'WFDC3',\n",
            " 'WFDC5',\n",
            " 'WFDC6',\n",
            " 'WFDC8',\n",
            " 'WFDC9',\n",
            " 'WFIKKN1',\n",
            " 'WFIKKN2',\n",
            " 'WFS1',\n",
            " 'WHAMM',\n",
            " 'WHRN',\n",
            " 'WIF1',\n",
            " 'WIPF1',\n",
            " 'WIPF2',\n",
            " 'WIPF3',\n",
            " 'WIPI1',\n",
            " 'WIPI2',\n",
            " 'WIZ',\n",
            " 'WLS',\n",
            " 'WNK1',\n",
            " 'WNK2',\n",
            " 'WNK3',\n",
            " 'WNK4',\n",
            " 'WNT1',\n",
            " 'WNT10A',\n",
            " 'WNT10B',\n",
            " 'WNT11',\n",
            " 'WNT16',\n",
            " 'WNT2',\n",
            " 'WNT2B',\n",
            " 'WNT3',\n",
            " 'WNT3A',\n",
            " 'WNT4',\n",
            " 'WNT5A',\n",
            " 'WNT5B',\n",
            " 'WNT6',\n",
            " 'WNT7A',\n",
            " 'WNT7B',\n",
            " 'WNT8A',\n",
            " 'WNT8B',\n",
            " 'WNT9A',\n",
            " 'WNT9B',\n",
            " 'WRAP53',\n",
            " 'WRAP73',\n",
            " 'WRN',\n",
            " 'WRNIP1',\n",
            " 'WSB1',\n",
            " 'WSB2',\n",
            " 'WSCD1',\n",
            " 'WSCD2',\n",
            " 'WT1',\n",
            " 'WTAP',\n",
            " 'WTIP',\n",
            " 'WWC1',\n",
            " 'WWC2',\n",
            " 'WWC3',\n",
            " 'WWOX',\n",
            " 'WWP1',\n",
            " 'WWP2',\n",
            " 'WWTR1',\n",
            " 'XAB2',\n",
            " 'XAF1',\n",
            " 'XAGE1B',\n",
            " 'XAGE2',\n",
            " 'XAGE3',\n",
            " 'XAGE5',\n",
            " 'XBP1',\n",
            " 'XCL1',\n",
            " 'XCL2',\n",
            " 'XCR1',\n",
            " 'XDH',\n",
            " 'XG',\n",
            " 'XIAP',\n",
            " 'XIRP1',\n",
            " 'XIRP2',\n",
            " 'XK',\n",
            " 'XKR3',\n",
            " 'XKR4',\n",
            " 'XKR5',\n",
            " 'XKR6',\n",
            " 'XKR7',\n",
            " 'XKR8',\n",
            " 'XKR9',\n",
            " 'XKRX',\n",
            " 'XPA',\n",
            " 'XPC',\n",
            " 'XPNPEP1',\n",
            " 'XPNPEP2',\n",
            " 'XPNPEP3',\n",
            " 'XPO1',\n",
            " 'XPO4',\n",
            " 'XPO5',\n",
            " 'XPO6',\n",
            " 'XPO7',\n",
            " 'XPOT',\n",
            " 'XPR1',\n",
            " 'XRCC1',\n",
            " 'XRCC2',\n",
            " 'XRCC3',\n",
            " 'XRCC4',\n",
            " 'XRCC5',\n",
            " 'XRCC6',\n",
            " 'XRN1',\n",
            " 'XRN2',\n",
            " 'XRRA1',\n",
            " 'XXYLT1',\n",
            " 'XYLB',\n",
            " 'XYLT1',\n",
            " 'XYLT2',\n",
            " 'YAE1',\n",
            " 'YAF2',\n",
            " 'YAP1',\n",
            " 'YARS1',\n",
            " 'YARS2',\n",
            " 'YBEY',\n",
            " 'YBX1',\n",
            " 'YBX2',\n",
            " 'YBX3',\n",
            " 'YDJC',\n",
            " 'YEATS2',\n",
            " 'YEATS4',\n",
            " 'YES1',\n",
            " 'YIF1A',\n",
            " 'YIF1B',\n",
            " 'YIPF1',\n",
            " 'YIPF2',\n",
            " 'YIPF3',\n",
            " 'YIPF4',\n",
            " 'YIPF5',\n",
            " 'YIPF6',\n",
            " 'YIPF7',\n",
            " 'YJEFN3',\n",
            " 'YJU2',\n",
            " 'YJU2B',\n",
            " 'YKT6',\n",
            " 'YLPM1',\n",
            " 'YME1L1',\n",
            " 'YOD1',\n",
            " 'YPEL1',\n",
            " 'YPEL2',\n",
            " 'YPEL3',\n",
            " 'YPEL4',\n",
            " 'YPEL5',\n",
            " 'YRDC',\n",
            " 'YTHDC1',\n",
            " 'YTHDC2',\n",
            " 'YTHDF1',\n",
            " 'YTHDF2',\n",
            " 'YTHDF3',\n",
            " 'YWHAB',\n",
            " 'YWHAE',\n",
            " 'YWHAG',\n",
            " 'YWHAH',\n",
            " 'YWHAH-AS1',\n",
            " 'YWHAQ',\n",
            " 'YWHAZ',\n",
            " 'YY1',\n",
            " 'YY1AP1',\n",
            " 'YY2',\n",
            " 'ZACN',\n",
            " 'ZAN',\n",
            " 'ZAP70',\n",
            " 'ZAR1',\n",
            " 'ZAR1L',\n",
            " 'ZBBX',\n",
            " 'ZBED2',\n",
            " 'ZBED3',\n",
            " 'ZBED4',\n",
            " 'ZBED6',\n",
            " 'ZBP1',\n",
            " 'ZBTB1',\n",
            " 'ZBTB10',\n",
            " 'ZBTB11',\n",
            " 'ZBTB12',\n",
            " 'ZBTB14',\n",
            " 'ZBTB16',\n",
            " 'ZBTB17',\n",
            " 'ZBTB18',\n",
            " 'ZBTB2',\n",
            " 'ZBTB20',\n",
            " 'ZBTB21',\n",
            " 'ZBTB22',\n",
            " 'ZBTB24',\n",
            " 'ZBTB25',\n",
            " 'ZBTB26',\n",
            " 'ZBTB3',\n",
            " 'ZBTB32',\n",
            " 'ZBTB33',\n",
            " 'ZBTB34',\n",
            " 'ZBTB37',\n",
            " 'ZBTB38',\n",
            " 'ZBTB39',\n",
            " 'ZBTB4',\n",
            " 'ZBTB40',\n",
            " 'ZBTB41',\n",
            " 'ZBTB42',\n",
            " 'ZBTB43',\n",
            " 'ZBTB44',\n",
            " 'ZBTB45',\n",
            " 'ZBTB46',\n",
            " 'ZBTB47',\n",
            " 'ZBTB48',\n",
            " 'ZBTB49',\n",
            " 'ZBTB5',\n",
            " 'ZBTB6',\n",
            " 'ZBTB7A',\n",
            " 'ZBTB7B',\n",
            " 'ZBTB7C',\n",
            " 'ZBTB8A',\n",
            " 'ZBTB8B',\n",
            " 'ZBTB8OS',\n",
            " 'ZBTB9',\n",
            " 'ZC2HC1A',\n",
            " 'ZC2HC1B',\n",
            " 'ZC2HC1C',\n",
            " 'ZC3H12A',\n",
            " 'ZC3H12B',\n",
            " 'ZC3H12C',\n",
            " 'ZC3H12D',\n",
            " 'ZC3H13',\n",
            " 'ZC3H14',\n",
            " 'ZC3H15',\n",
            " 'ZC3H18',\n",
            " 'ZC3H3',\n",
            " 'ZC3H4',\n",
            " 'ZC3H6',\n",
            " 'ZC3H7A',\n",
            " 'ZC3H7B',\n",
            " 'ZC3H8',\n",
            " 'ZC3HAV1',\n",
            " 'ZC3HAV1L',\n",
            " 'ZC3HC1',\n",
            " 'ZC4H2',\n",
            " 'ZCCHC10',\n",
            " 'ZCCHC12',\n",
            " 'ZCCHC13',\n",
            " 'ZCCHC14',\n",
            " 'ZCCHC17',\n",
            " 'ZCCHC18',\n",
            " 'ZCCHC2',\n",
            " 'ZCCHC24',\n",
            " 'ZCCHC3',\n",
            " 'ZCCHC4',\n",
            " 'ZCCHC7',\n",
            " 'ZCCHC8',\n",
            " 'ZCCHC9',\n",
            " 'ZCRB1',\n",
            " 'ZCWPW1',\n",
            " 'ZCWPW2',\n",
            " 'ZDBF2',\n",
            " 'ZDHHC1',\n",
            " 'ZDHHC11',\n",
            " 'ZDHHC11B',\n",
            " 'ZDHHC12',\n",
            " 'ZDHHC13',\n",
            " 'ZDHHC14',\n",
            " 'ZDHHC15',\n",
            " 'ZDHHC16',\n",
            " 'ZDHHC17',\n",
            " 'ZDHHC18',\n",
            " 'ZDHHC19',\n",
            " 'ZDHHC2',\n",
            " 'ZDHHC20',\n",
            " 'ZDHHC21',\n",
            " 'ZDHHC22',\n",
            " 'ZDHHC23',\n",
            " 'ZDHHC24',\n",
            " 'ZDHHC3',\n",
            " 'ZDHHC4',\n",
            " 'ZDHHC5',\n",
            " 'ZDHHC6',\n",
            " 'ZDHHC7',\n",
            " 'ZDHHC8',\n",
            " 'ZDHHC9',\n",
            " 'ZEB1',\n",
            " 'ZEB2',\n",
            " 'ZER1',\n",
            " 'ZFAND1',\n",
            " 'ZFAND2A',\n",
            " 'ZFAND2B',\n",
            " 'ZFAND3',\n",
            " 'ZFAND4',\n",
            " 'ZFAND5',\n",
            " 'ZFAND6',\n",
            " 'ZFAT',\n",
            " 'ZFC3H1',\n",
            " 'ZFHX2',\n",
            " 'ZFHX3',\n",
            " 'ZFHX4',\n",
            " 'ZFP1',\n",
            " 'ZFP14',\n",
            " 'ZFP2',\n",
            " 'ZFP28',\n",
            " 'ZFP3',\n",
            " 'ZFP30',\n",
            " 'ZFP36',\n",
            " 'ZFP36L1',\n",
            " 'ZFP36L2',\n",
            " 'ZFP37',\n",
            " 'ZFP41',\n",
            " 'ZFP42',\n",
            " 'ZFP57',\n",
            " 'ZFP62',\n",
            " 'ZFP64',\n",
            " 'ZFP69',\n",
            " 'ZFP69B',\n",
            " 'ZFP82',\n",
            " 'ZFP90',\n",
            " 'ZFP91',\n",
            " 'ZFP92',\n",
            " 'ZFPL1',\n",
            " 'ZFPM1',\n",
            " 'ZFPM2',\n",
            " 'ZFR',\n",
            " 'ZFR2',\n",
            " 'ZFTA',\n",
            " 'ZFTRAF1',\n",
            " 'ZFX',\n",
            " 'ZFY',\n",
            " 'ZFYVE1',\n",
            " 'ZFYVE16',\n",
            " 'ZFYVE19',\n",
            " 'ZFYVE21',\n",
            " 'ZFYVE26',\n",
            " 'ZFYVE27',\n",
            " 'ZFYVE28',\n",
            " 'ZFYVE9',\n",
            " 'ZG16',\n",
            " 'ZG16B',\n",
            " 'ZGLP1',\n",
            " 'ZGPAT',\n",
            " 'ZGRF1',\n",
            " 'ZHX1',\n",
            " 'ZHX2',\n",
            " 'ZHX3',\n",
            " 'ZIC1',\n",
            " 'ZIC2',\n",
            " 'ZIC3',\n",
            " 'ZIC4',\n",
            " 'ZIC5',\n",
            " 'ZIK1',\n",
            " 'ZIM2',\n",
            " 'ZIM3',\n",
            " 'ZKSCAN1',\n",
            " 'ZKSCAN2',\n",
            " 'ZKSCAN3',\n",
            " 'ZKSCAN4',\n",
            " 'ZKSCAN5',\n",
            " 'ZKSCAN7',\n",
            " 'ZKSCAN8',\n",
            " 'ZMAT1',\n",
            " 'ZMAT2',\n",
            " 'ZMAT3',\n",
            " 'ZMAT4',\n",
            " 'ZMAT5',\n",
            " 'ZMIZ1',\n",
            " 'ZMIZ2',\n",
            " 'ZMPSTE24',\n",
            " 'ZMYM1',\n",
            " 'ZMYM2',\n",
            " 'ZMYM3',\n",
            " 'ZMYM4',\n",
            " 'ZMYM5',\n",
            " 'ZMYM6',\n",
            " 'ZMYND10',\n",
            " 'ZMYND11',\n",
            " 'ZMYND12',\n",
            " 'ZMYND19',\n",
            " 'ZMYND8',\n",
            " 'ZNF10',\n",
            " 'ZNF100',\n",
            " 'ZNF101',\n",
            " 'ZNF106',\n",
            " 'ZNF107',\n",
            " 'ZNF112',\n",
            " 'ZNF114',\n",
            " 'ZNF117',\n",
            " 'ZNF12',\n",
            " 'ZNF121',\n",
            " 'ZNF124',\n",
            " 'ZNF131',\n",
            " 'ZNF132',\n",
            " 'ZNF133',\n",
            " 'ZNF134',\n",
            " 'ZNF135',\n",
            " 'ZNF136',\n",
            " 'ZNF138',\n",
            " 'ZNF14',\n",
            " 'ZNF140',\n",
            " 'ZNF141',\n",
            " 'ZNF142',\n",
            " 'ZNF143',\n",
            " 'ZNF146',\n",
            " 'ZNF148',\n",
            " 'ZNF154',\n",
            " 'ZNF155',\n",
            " 'ZNF157',\n",
            " 'ZNF16',\n",
            " 'ZNF160',\n",
            " 'ZNF165',\n",
            " 'ZNF169',\n",
            " 'ZNF17',\n",
            " 'ZNF174',\n",
            " 'ZNF175',\n",
            " 'ZNF177',\n",
            " 'ZNF18',\n",
            " 'ZNF180',\n",
            " 'ZNF181',\n",
            " 'ZNF182',\n",
            " 'ZNF184',\n",
            " 'ZNF185',\n",
            " 'ZNF189',\n",
            " 'ZNF19',\n",
            " 'ZNF195',\n",
            " 'ZNF197',\n",
            " 'ZNF2',\n",
            " 'ZNF20',\n",
            " 'ZNF200',\n",
            " 'ZNF202',\n",
            " 'ZNF205',\n",
            " 'ZNF207',\n",
            " 'ZNF208',\n",
            " 'ZNF211',\n",
            " 'ZNF212',\n",
            " 'ZNF213',\n",
            " 'ZNF214',\n",
            " 'ZNF215',\n",
            " 'ZNF217',\n",
            " 'ZNF219',\n",
            " 'ZNF22',\n",
            " 'ZNF22-AS1',\n",
            " 'ZNF221',\n",
            " 'ZNF222',\n",
            " 'ZNF223',\n",
            " 'ZNF224',\n",
            " 'ZNF225',\n",
            " 'ZNF227',\n",
            " 'ZNF229',\n",
            " 'ZNF23',\n",
            " 'ZNF230',\n",
            " 'ZNF232',\n",
            " 'ZNF233',\n",
            " 'ZNF235',\n",
            " 'ZNF236',\n",
            " 'ZNF239',\n",
            " 'ZNF24',\n",
            " 'ZNF248',\n",
            " 'ZNF25',\n",
            " 'ZNF250',\n",
            " 'ZNF251',\n",
            " 'ZNF253',\n",
            " 'ZNF254',\n",
            " 'ZNF256',\n",
            " 'ZNF257',\n",
            " 'ZNF26',\n",
            " 'ZNF260',\n",
            " 'ZNF263',\n",
            " 'ZNF264',\n",
            " 'ZNF266',\n",
            " 'ZNF267',\n",
            " 'ZNF268',\n",
            " 'ZNF273',\n",
            " 'ZNF274',\n",
            " 'ZNF275',\n",
            " 'ZNF276',\n",
            " 'ZNF277',\n",
            " 'ZNF28',\n",
            " 'ZNF280A',\n",
            " 'ZNF280B',\n",
            " 'ZNF280C',\n",
            " 'ZNF280D',\n",
            " 'ZNF281',\n",
            " 'ZNF282',\n",
            " 'ZNF283',\n",
            " 'ZNF284',\n",
            " 'ZNF285',\n",
            " 'ZNF286A',\n",
            " 'ZNF286B',\n",
            " 'ZNF287',\n",
            " 'ZNF292',\n",
            " 'ZNF296',\n",
            " 'ZNF3',\n",
            " 'ZNF30',\n",
            " 'ZNF300',\n",
            " 'ZNF302',\n",
            " 'ZNF304',\n",
            " 'ZNF311',\n",
            " 'ZNF317',\n",
            " 'ZNF318',\n",
            " 'ZNF319',\n",
            " 'ZNF32',\n",
            " 'ZNF320',\n",
            " 'ZNF322',\n",
            " 'ZNF324',\n",
            " 'ZNF324B',\n",
            " 'ZNF326',\n",
            " 'ZNF329',\n",
            " 'ZNF330',\n",
            " 'ZNF331',\n",
            " 'ZNF333',\n",
            " 'ZNF334',\n",
            " 'ZNF335',\n",
            " 'ZNF337',\n",
            " 'ZNF33A',\n",
            " 'ZNF33B',\n",
            " 'ZNF34',\n",
            " 'ZNF341',\n",
            " 'ZNF343',\n",
            " 'ZNF345',\n",
            " 'ZNF346',\n",
            " 'ZNF347',\n",
            " 'ZNF35',\n",
            " 'ZNF350',\n",
            " 'ZNF354A',\n",
            " 'ZNF354B',\n",
            " 'ZNF354C',\n",
            " 'ZNF358',\n",
            " 'ZNF362',\n",
            " 'ZNF365',\n",
            " 'ZNF366',\n",
            " 'ZNF367',\n",
            " 'ZNF37A',\n",
            " 'ZNF382',\n",
            " 'ZNF383',\n",
            " 'ZNF384',\n",
            " 'ZNF385A',\n",
            " 'ZNF385B',\n",
            " 'ZNF385C',\n",
            " 'ZNF385D',\n",
            " 'ZNF391',\n",
            " 'ZNF394',\n",
            " 'ZNF395',\n",
            " 'ZNF396',\n",
            " 'ZNF397',\n",
            " 'ZNF398',\n",
            " 'ZNF404',\n",
            " 'ZNF407',\n",
            " 'ZNF408',\n",
            " 'ZNF41',\n",
            " 'ZNF410',\n",
            " 'ZNF414',\n",
            " 'ZNF415',\n",
            " 'ZNF416',\n",
            " 'ZNF417',\n",
            " 'ZNF418',\n",
            " 'ZNF419',\n",
            " 'ZNF420',\n",
            " 'ZNF423',\n",
            " 'ZNF425',\n",
            " 'ZNF426',\n",
            " 'ZNF428',\n",
            " 'ZNF429',\n",
            " 'ZNF43',\n",
            " 'ZNF430',\n",
            " 'ZNF431',\n",
            " 'ZNF432',\n",
            " 'ZNF433',\n",
            " 'ZNF436',\n",
            " 'ZNF438',\n",
            " 'ZNF439',\n",
            " 'ZNF44',\n",
            " 'ZNF440',\n",
            " 'ZNF441',\n",
            " 'ZNF442',\n",
            " 'ZNF443',\n",
            " 'ZNF444',\n",
            " 'ZNF445',\n",
            " 'ZNF446',\n",
            " 'ZNF449',\n",
            " 'ZNF45',\n",
            " 'ZNF451',\n",
            " 'ZNF454',\n",
            " 'ZNF460',\n",
            " 'ZNF461',\n",
            " 'ZNF462',\n",
            " 'ZNF467',\n",
            " 'ZNF468',\n",
            " 'ZNF469',\n",
            " 'ZNF470',\n",
            " 'ZNF471',\n",
            " 'ZNF473',\n",
            " 'ZNF474',\n",
            " 'ZNF479',\n",
            " 'ZNF48',\n",
            " 'ZNF480',\n",
            " 'ZNF483',\n",
            " 'ZNF484',\n",
            " 'ZNF485',\n",
            " 'ZNF486',\n",
            " 'ZNF488',\n",
            " 'ZNF490',\n",
            " 'ZNF491',\n",
            " 'ZNF492',\n",
            " 'ZNF493',\n",
            " 'ZNF496',\n",
            " 'ZNF497',\n",
            " 'ZNF500',\n",
            " 'ZNF501',\n",
            " 'ZNF502',\n",
            " 'ZNF503',\n",
            " 'ZNF506',\n",
            " 'ZNF507',\n",
            " 'ZNF510',\n",
            " 'ZNF511',\n",
            " 'ZNF512',\n",
            " 'ZNF512B',\n",
            " 'ZNF513',\n",
            " 'ZNF514',\n",
            " 'ZNF516',\n",
            " 'ZNF517',\n",
            " 'ZNF518A',\n",
            " 'ZNF518B',\n",
            " 'ZNF519',\n",
            " 'ZNF521',\n",
            " 'ZNF524',\n",
            " 'ZNF526',\n",
            " 'ZNF527',\n",
            " 'ZNF528',\n",
            " 'ZNF529',\n",
            " 'ZNF530',\n",
            " 'ZNF532',\n",
            " 'ZNF534',\n",
            " 'ZNF536',\n",
            " 'ZNF540',\n",
            " 'ZNF541',\n",
            " 'ZNF543',\n",
            " 'ZNF544',\n",
            " 'ZNF546',\n",
            " 'ZNF547',\n",
            " 'ZNF548',\n",
            " 'ZNF549',\n",
            " 'ZNF550',\n",
            " 'ZNF551',\n",
            " 'ZNF552',\n",
            " 'ZNF554',\n",
            " 'ZNF555',\n",
            " 'ZNF556',\n",
            " 'ZNF557',\n",
            " 'ZNF558',\n",
            " 'ZNF559',\n",
            " 'ZNF560',\n",
            " 'ZNF561',\n",
            " 'ZNF561-AS1',\n",
            " 'ZNF562',\n",
            " 'ZNF563',\n",
            " 'ZNF565',\n",
            " 'ZNF566',\n",
            " 'ZNF567',\n",
            " 'ZNF568',\n",
            " 'ZNF569',\n",
            " 'ZNF57',\n",
            " 'ZNF570',\n",
            " 'ZNF571',\n",
            " 'ZNF572',\n",
            " 'ZNF573',\n",
            " 'ZNF574',\n",
            " 'ZNF575',\n",
            " 'ZNF577',\n",
            " 'ZNF578',\n",
            " 'ZNF579',\n",
            " 'ZNF580',\n",
            " 'ZNF581',\n",
            " 'ZNF582',\n",
            " 'ZNF583',\n",
            " 'ZNF584',\n",
            " 'ZNF585B',\n",
            " 'ZNF586',\n",
            " 'ZNF587',\n",
            " 'ZNF587B',\n",
            " 'ZNF589',\n",
            " 'ZNF592',\n",
            " 'ZNF593',\n",
            " 'ZNF594',\n",
            " 'ZNF595',\n",
            " 'ZNF596',\n",
            " 'ZNF597',\n",
            " 'ZNF598',\n",
            " 'ZNF599',\n",
            " 'ZNF600',\n",
            " 'ZNF605',\n",
            " 'ZNF606',\n",
            " 'ZNF607',\n",
            " 'ZNF608',\n",
            " 'ZNF609',\n",
            " 'ZNF610',\n",
            " 'ZNF611',\n",
            " 'ZNF613',\n",
            " 'ZNF614',\n",
            " 'ZNF615',\n",
            " 'ZNF616',\n",
            " 'ZNF618',\n",
            " 'ZNF619',\n",
            " 'ZNF620',\n",
            " 'ZNF621',\n",
            " 'ZNF622',\n",
            " 'ZNF623',\n",
            " 'ZNF624',\n",
            " 'ZNF626',\n",
            " 'ZNF627',\n",
            " 'ZNF628',\n",
            " 'ZNF629',\n",
            " 'ZNF630',\n",
            " 'ZNF638',\n",
            " 'ZNF639',\n",
            " 'ZNF641',\n",
            " 'ZNF644',\n",
            " 'ZNF646',\n",
            " 'ZNF648',\n",
            " 'ZNF649',\n",
            " 'ZNF652',\n",
            " 'ZNF653',\n",
            " 'ZNF654',\n",
            " 'ZNF655',\n",
            " 'ZNF658',\n",
            " 'ZNF660',\n",
            " 'ZNF662',\n",
            " 'ZNF664',\n",
            " 'ZNF665',\n",
            " 'ZNF667',\n",
            " 'ZNF668',\n",
            " 'ZNF669',\n",
            " 'ZNF670',\n",
            " 'ZNF671',\n",
            " 'ZNF672',\n",
            " 'ZNF674',\n",
            " 'ZNF675',\n",
            " 'ZNF676',\n",
            " 'ZNF677',\n",
            " 'ZNF678',\n",
            " 'ZNF679',\n",
            " 'ZNF680',\n",
            " 'ZNF681',\n",
            " 'ZNF682',\n",
            " 'ZNF683',\n",
            " 'ZNF684',\n",
            " 'ZNF687',\n",
            " 'ZNF688',\n",
            " 'ZNF689',\n",
            " 'ZNF69',\n",
            " 'ZNF691',\n",
            " 'ZNF692',\n",
            " 'ZNF695',\n",
            " 'ZNF696',\n",
            " 'ZNF697',\n",
            " 'ZNF699',\n",
            " 'ZNF7',\n",
            " 'ZNF70',\n",
            " 'ZNF700',\n",
            " 'ZNF701',\n",
            " 'ZNF703',\n",
            " 'ZNF704',\n",
            " 'ZNF705A',\n",
            " 'ZNF705B',\n",
            " 'ZNF705D',\n",
            " 'ZNF705G',\n",
            " 'ZNF706',\n",
            " 'ZNF707',\n",
            " 'ZNF708',\n",
            " 'ZNF71',\n",
            " 'ZNF710',\n",
            " 'ZNF711',\n",
            " 'ZNF713',\n",
            " 'ZNF714',\n",
            " 'ZNF716',\n",
            " 'ZNF717',\n",
            " 'ZNF718',\n",
            " 'ZNF721',\n",
            " 'ZNF726',\n",
            " 'ZNF727',\n",
            " 'ZNF728',\n",
            " 'ZNF729',\n",
            " 'ZNF730',\n",
            " 'ZNF732',\n",
            " 'ZNF735',\n",
            " 'ZNF736',\n",
            " 'ZNF737',\n",
            " 'ZNF74',\n",
            " 'ZNF740',\n",
            " 'ZNF746',\n",
            " 'ZNF747',\n",
            " 'ZNF749',\n",
            " 'ZNF750',\n",
            " 'ZNF75A',\n",
            " 'ZNF75D',\n",
            " 'ZNF76',\n",
            " 'ZNF761',\n",
            " 'ZNF763',\n",
            " 'ZNF764',\n",
            " 'ZNF765',\n",
            " 'ZNF766',\n",
            " 'ZNF768',\n",
            " 'ZNF77',\n",
            " 'ZNF770',\n",
            " 'ZNF771',\n",
            " 'ZNF772',\n",
            " 'ZNF773',\n",
            " 'ZNF774',\n",
            " 'ZNF775',\n",
            " 'ZNF776',\n",
            " 'ZNF777',\n",
            " 'ZNF778',\n",
            " 'ZNF780A',\n",
            " 'ZNF780B',\n",
            " 'ZNF781',\n",
            " 'ZNF782',\n",
            " 'ZNF783',\n",
            " 'ZNF784',\n",
            " 'ZNF785',\n",
            " 'ZNF786',\n",
            " 'ZNF787',\n",
            " 'ZNF789',\n",
            " 'ZNF79',\n",
            " 'ZNF790',\n",
            " 'ZNF791',\n",
            " 'ZNF792',\n",
            " 'ZNF793',\n",
            " 'ZNF799',\n",
            " 'ZNF8',\n",
            " 'ZNF80',\n",
            " 'ZNF800',\n",
            " 'ZNF804A',\n",
            " 'ZNF804B',\n",
            " 'ZNF805',\n",
            " 'ZNF808',\n",
            " 'ZNF81',\n",
            " 'ZNF813',\n",
            " 'ZNF814',\n",
            " 'ZNF816',\n",
            " 'ZNF821',\n",
            " 'ZNF823',\n",
            " 'ZNF827',\n",
            " 'ZNF829',\n",
            " 'ZNF83',\n",
            " 'ZNF830',\n",
            " 'ZNF831',\n",
            " 'ZNF835',\n",
            " 'ZNF836',\n",
            " 'ZNF837',\n",
            " 'ZNF839',\n",
            " 'ZNF84',\n",
            " 'ZNF841',\n",
            " 'ZNF843',\n",
            " 'ZNF844',\n",
            " 'ZNF845',\n",
            " 'ZNF846',\n",
            " 'ZNF85',\n",
            " 'ZNF850',\n",
            " 'ZNF852',\n",
            " 'ZNF853',\n",
            " 'ZNF860',\n",
            " 'ZNF862',\n",
            " 'ZNF865',\n",
            " 'ZNF875',\n",
            " 'ZNF878',\n",
            " 'ZNF879',\n",
            " 'ZNF880',\n",
            " 'ZNF883',\n",
            " 'ZNF888',\n",
            " 'ZNF891',\n",
            " 'ZNF90',\n",
            " 'ZNF91',\n",
            " 'ZNF92',\n",
            " 'ZNF93',\n",
            " 'ZNF98',\n",
            " 'ZNF99',\n",
            " 'ZNFX1',\n",
            " 'ZNG1A',\n",
            " 'ZNHIT1',\n",
            " 'ZNHIT2',\n",
            " 'ZNHIT3',\n",
            " 'ZNHIT6',\n",
            " 'ZNRD2',\n",
            " 'ZNRF1',\n",
            " 'ZNRF2',\n",
            " 'ZNRF3',\n",
            " 'ZNRF4',\n",
            " 'ZP1',\n",
            " 'ZP2',\n",
            " 'ZP3',\n",
            " 'ZP4',\n",
            " 'ZPBP',\n",
            " 'ZPBP2',\n",
            " 'ZPLD1',\n",
            " 'ZPR1',\n",
            " 'ZRANB1',\n",
            " 'ZRANB2',\n",
            " 'ZRANB3',\n",
            " 'ZRSR2',\n",
            " 'ZSCAN1',\n",
            " 'ZSCAN10',\n",
            " 'ZSCAN12',\n",
            " 'ZSCAN16',\n",
            " 'ZSCAN18',\n",
            " 'ZSCAN2',\n",
            " 'ZSCAN20',\n",
            " 'ZSCAN21',\n",
            " 'ZSCAN22',\n",
            " 'ZSCAN23',\n",
            " 'ZSCAN25',\n",
            " 'ZSCAN26',\n",
            " 'ZSCAN29',\n",
            " 'ZSCAN30',\n",
            " 'ZSCAN31',\n",
            " 'ZSCAN32',\n",
            " 'ZSCAN5A',\n",
            " 'ZSCAN5B',\n",
            " 'ZSCAN9',\n",
            " 'ZSWIM1',\n",
            " 'ZSWIM2',\n",
            " 'ZSWIM3',\n",
            " 'ZSWIM4',\n",
            " 'ZSWIM5',\n",
            " 'ZSWIM6',\n",
            " 'ZSWIM7',\n",
            " 'ZSWIM8',\n",
            " 'ZSWIM9',\n",
            " 'ZUP1',\n",
            " 'ZW10',\n",
            " 'ZWILCH',\n",
            " 'ZWINT',\n",
            " 'ZXDA',\n",
            " 'ZXDB',\n",
            " 'ZXDC',\n",
            " 'ZYG11A',\n",
            " 'ZYG11B',\n",
            " 'ZYX',\n",
            " 'ZZEF1',\n",
            " 'ZZZ3',\n",
            " 'control_nontargeting'}\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.replace_entries(\n",
        "    slot=\"obs\",\n",
        "    column=\"perturbed_target_symbol\",\n",
        "    map_dict={\n",
        "        \"Non-Targeting\": \"control_nontargeting\"\n",
        "    }\n",
        ")\n",
        "\n",
        "cur_data.show_unique(slot = 'obs', column = 'perturbed_target_symbol')"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "f1a48435",
      "metadata": {
        "id": "f1a48435"
      },
      "source": [
        "### Standardise perturbation targets"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "438c975e",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "438c975e",
        "outputId": "3e32d701-af37-4f1a-aca0-d04affa8029c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Mapping gene symbols: 100%|████████████████████████████████| 18294/18294 [00:00<00:00, 25285.63it/s]\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--------------------------------------------------\n",
            "Successfully mapped 18288 out of 18294 gene symbols.\n",
            "--------------------------------------------------\n",
            "Couldn't map gene symbols: ['control_nontargeting', 'PRY2', 'PRY', 'LINC02911', 'SMIM2', 'LINC02913']\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.standardize_genes(\n",
        "    slot='obs',\n",
        "    input_column='perturbed_target_symbol',\n",
        "    input_column_type='gene_symbol',\n",
        "    multiple_entries=False\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "962e76b5",
      "metadata": {
        "id": "962e76b5"
      },
      "source": [
        "### Add `perturbed_target_number` column"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "cur_data.adata.obs = cur_data.adata.obs.iloc[:,[0,1,2,3,4,5,6,7,8,9,11,12,13]]"
      ],
      "metadata": {
        "id": "oOmaEX_332ro"
      },
      "id": "oOmaEX_332ro",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "a8faf759",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "a8faf759",
        "outputId": "7cd5e921-18d1-44c9-de3b-dd302955110b"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Counted entries in column perturbed_target_symbol of adata.obs and stored in perturbed_target_number\n"
          ]
        }
      ],
      "source": [
        "cur_data.count_entries(\n",
        "    slot='obs',\n",
        "    input_column='perturbed_target_symbol',\n",
        "    count_column_name='perturbed_target_number',\n",
        "    sep='|'\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "cd2d53ef",
      "metadata": {
        "id": "cd2d53ef"
      },
      "source": [
        "### Encode chromosomes as integers"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "e3b73c56",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "e3b73c56",
        "outputId": "d395229a-7d76-4b29-d4c2-152b8d233f0d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Chromosome encoding applied to perturbed_target_chromosome in adata.obs and stored as 'perturbed_target_chromosome_encoding'.\n"
          ]
        }
      ],
      "source": [
        "cur_data.chromosome_encoding()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "02a7c724",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "02a7c724",
        "outputId": "0d92fa99-3f97-4534-956a-9a04725c96e5"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Observation data:\n",
            "DataFrame shape: (3409169, 2)\n",
            "--------------------------------------------------\n",
            "                                              perturbation_name  \\\n",
            "index                                                             \n",
            "AAACCAAAGACATGTT-HCT116_Batch1          ST14_P1P2-1|ST14_P1P2-2   \n",
            "AAACCAAAGACCCAAC-HCT116_Batch1    SIGLEC5_P1P2-1|SIGLEC5_P1P2-2   \n",
            "AAACCAAAGAGGTACG-HCT116_Batch1        VSNL1_P1P2-1|VSNL1_P1P2-2   \n",
            "AAACCAAAGCGATTAT-HCT116_Batch1        KCNK7_P1P2-1|KCNK7_P1P2-2   \n",
            "AAACCAAAGGCTTAAT-HCT116_Batch1        APOA4_P1P2-1|APOA4_P1P2-2   \n",
            "...                                                         ...   \n",
            "GTTGTGCAGGGAATCG-HCT116_Batch109        ACP6_P1P2-1|ACP6_P1P2-2   \n",
            "GTTGTGGGTCACGCGG-HCT116_Batch109      RNF39_P1P2-1|RNF39_P1P2-2   \n",
            "GTTGTGGGTCTGACTC-HCT116_Batch109              AFP_P2-1|AFP_P2-2   \n",
            "GTTGTGGGTCTTGATT-HCT116_Batch109    METTL9_P1P2-1|METTL9_P1P2-2   \n",
            "GTTGTGGGTGTAGTTA-HCT116_Batch109  SLC5A10_P1P2-1|SLC5A10_P1P2-2   \n",
            "\n",
            "                                  perturbed_target_chromosome_encoding  \n",
            "index                                                                   \n",
            "AAACCAAAGACATGTT-HCT116_Batch1                                      11  \n",
            "AAACCAAAGACCCAAC-HCT116_Batch1                                      19  \n",
            "AAACCAAAGAGGTACG-HCT116_Batch1                                       2  \n",
            "AAACCAAAGCGATTAT-HCT116_Batch1                                      11  \n",
            "AAACCAAAGGCTTAAT-HCT116_Batch1                                      11  \n",
            "...                                                                ...  \n",
            "GTTGTGCAGGGAATCG-HCT116_Batch109                                     1  \n",
            "GTTGTGGGTCACGCGG-HCT116_Batch109                                     6  \n",
            "GTTGTGGGTCTGACTC-HCT116_Batch109                                     4  \n",
            "GTTGTGGGTCTTGATT-HCT116_Batch109                                    16  \n",
            "GTTGTGGGTGTAGTTA-HCT116_Batch109                                    17  \n",
            "\n",
            "[3409169 rows x 2 columns]\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.show_obs(['perturbation_name', 'perturbed_target_chromosome_encoding'])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "tbHMJF1UDeex",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 344
        },
        "id": "tbHMJF1UDeex",
        "outputId": "54bccf52-e4a5-4df7-93b8-7683cb877697"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                pct_counts_mt  pass_guide_filter  \\\n",
              "index                                                              \n",
              "AAACCAAAGACATGTT-HCT116_Batch1       6.161162               True   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1       3.259871               True   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1       3.664498               True   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1       4.167945               True   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1       2.489183               True   \n",
              "\n",
              "                                num_features  total_counts_mt  total_counts  \\\n",
              "index                                                                         \n",
              "AAACCAAAGACATGTT-HCT116_Batch1             2           1179.0       19136.0   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1             2           1562.0       47916.0   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1             2           1042.0       28435.0   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1             2           1087.0       26080.0   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1             2            955.0       38366.0   \n",
              "\n",
              "                                       sample  n_genes_by_counts  \\\n",
              "index                                                              \n",
              "AAACCAAAGACATGTT-HCT116_Batch1  HCT116_Batch1               4883   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1  HCT116_Batch1               8130   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1  HCT116_Batch1               6531   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1  HCT116_Batch1               5931   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1  HCT116_Batch1               7157   \n",
              "\n",
              "                                            perturbation_name  \\\n",
              "index                                                           \n",
              "AAACCAAAGACATGTT-HCT116_Batch1        ST14_P1P2-1|ST14_P1P2-2   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1  SIGLEC5_P1P2-1|SIGLEC5_P1P2-2   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1      VSNL1_P1P2-1|VSNL1_P1P2-2   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1      KCNK7_P1P2-1|KCNK7_P1P2-2   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1      APOA4_P1P2-1|APOA4_P1P2-2   \n",
              "\n",
              "                               perturbed_target_ensg perturbed_target_symbol  \\\n",
              "index                                                                          \n",
              "AAACCAAAGACATGTT-HCT116_Batch1       ENSG00000149418                    ST14   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1       ENSG00000105501                 SIGLEC5   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1       ENSG00000163032                   VSNL1   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1       ENSG00000173338                   KCNK7   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1       ENSG00000110244                   APOA4   \n",
              "\n",
              "                               perturbed_target_biotype  \\\n",
              "index                                                     \n",
              "AAACCAAAGACATGTT-HCT116_Batch1           protein_coding   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                   lncRNA   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1           protein_coding   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1           protein_coding   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1           protein_coding   \n",
              "\n",
              "                                      perturbed_target_coord  \\\n",
              "index                                                          \n",
              "AAACCAAAGACATGTT-HCT116_Batch1   chr11:130159746-130210367;1   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1    chr19:51630101-51645545;-1   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1      chr2:17539020-17657018;1   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1    chr11:65592836-65595996;-1   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1  chr11:116820700-116823304;-1   \n",
              "\n",
              "                               perturbed_target_chromosome  \\\n",
              "index                                                        \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                          11   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                          19   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                           2   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                          11   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                          11   \n",
              "\n",
              "                                perturbed_target_number  \\\n",
              "index                                                     \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                        1   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                        1   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                        1   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                        1   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                        1   \n",
              "\n",
              "                                perturbed_target_chromosome_encoding  \n",
              "index                                                                 \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                                    11  \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                                    19  \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                                     2  \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                                    11  \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                                    11  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-5440254d-6592-42f0-9deb-c64902f2bf17\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pct_counts_mt</th>\n",
              "      <th>pass_guide_filter</th>\n",
              "      <th>num_features</th>\n",
              "      <th>total_counts_mt</th>\n",
              "      <th>total_counts</th>\n",
              "      <th>sample</th>\n",
              "      <th>n_genes_by_counts</th>\n",
              "      <th>perturbation_name</th>\n",
              "      <th>perturbed_target_ensg</th>\n",
              "      <th>perturbed_target_symbol</th>\n",
              "      <th>perturbed_target_biotype</th>\n",
              "      <th>perturbed_target_coord</th>\n",
              "      <th>perturbed_target_chromosome</th>\n",
              "      <th>perturbed_target_number</th>\n",
              "      <th>perturbed_target_chromosome_encoding</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACATGTT-HCT116_Batch1</th>\n",
              "      <td>6.161162</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1179.0</td>\n",
              "      <td>19136.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>4883</td>\n",
              "      <td>ST14_P1P2-1|ST14_P1P2-2</td>\n",
              "      <td>ENSG00000149418</td>\n",
              "      <td>ST14</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:130159746-130210367;1</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>11</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACCCAAC-HCT116_Batch1</th>\n",
              "      <td>3.259871</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1562.0</td>\n",
              "      <td>47916.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>8130</td>\n",
              "      <td>SIGLEC5_P1P2-1|SIGLEC5_P1P2-2</td>\n",
              "      <td>ENSG00000105501</td>\n",
              "      <td>SIGLEC5</td>\n",
              "      <td>lncRNA</td>\n",
              "      <td>chr19:51630101-51645545;-1</td>\n",
              "      <td>19</td>\n",
              "      <td>1</td>\n",
              "      <td>19</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGGTACG-HCT116_Batch1</th>\n",
              "      <td>3.664498</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1042.0</td>\n",
              "      <td>28435.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>6531</td>\n",
              "      <td>VSNL1_P1P2-1|VSNL1_P1P2-2</td>\n",
              "      <td>ENSG00000163032</td>\n",
              "      <td>VSNL1</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr2:17539020-17657018;1</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGCGATTAT-HCT116_Batch1</th>\n",
              "      <td>4.167945</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1087.0</td>\n",
              "      <td>26080.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>5931</td>\n",
              "      <td>KCNK7_P1P2-1|KCNK7_P1P2-2</td>\n",
              "      <td>ENSG00000173338</td>\n",
              "      <td>KCNK7</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:65592836-65595996;-1</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>11</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGGCTTAAT-HCT116_Batch1</th>\n",
              "      <td>2.489183</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>955.0</td>\n",
              "      <td>38366.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>7157</td>\n",
              "      <td>APOA4_P1P2-1|APOA4_P1P2-2</td>\n",
              "      <td>ENSG00000110244</td>\n",
              "      <td>APOA4</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:116820700-116823304;-1</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>11</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-5440254d-6592-42f0-9deb-c64902f2bf17')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-5440254d-6592-42f0-9deb-c64902f2bf17 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-5440254d-6592-42f0-9deb-c64902f2bf17');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-7d0a2ab0-25bf-41a7-8cb1-9f9963dd6b3f\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-7d0a2ab0-25bf-41a7-8cb1-9f9963dd6b3f')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-7d0a2ab0-25bf-41a7-8cb1-9f9963dd6b3f button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"cur_data\",\n  \"rows\": 5,\n  \"fields\": [\n    {\n      \"column\": \"index\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"AAACCAAAGACCCAAC-HCT116_Batch1\",\n          \"AAACCAAAGGCTTAAT-HCT116_Batch1\",\n          \"AAACCAAAGAGGTACG-HCT116_Batch1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pct_counts_mt\",\n      \"properties\": {\n        \"dtype\": \"float32\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          3.259871482849121,\n          2.489183187484741,\n          3.6644980907440186\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pass_guide_filter\",\n      \"properties\": {\n        \"dtype\": \"boolean\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          true\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"num_features\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 2,\n        \"max\": 2,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"total_counts_mt\",\n      \"properties\": {\n        \"dtype\": \"float32\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          1562.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"total_counts\",\n      \"properties\": {\n        \"dtype\": \"float32\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          47916.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"sample\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"HCT116_Batch1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"n_genes_by_counts\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1227,\n        \"min\": 4883,\n        \"max\": 8130,\n        \"num_unique_values\": 5,\n        \"samples\": [\n          8130\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbation_name\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"SIGLEC5_P1P2-1|SIGLEC5_P1P2-2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_ensg\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"ENSG00000105501\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_symbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"SIGLEC5\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_biotype\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"lncRNA\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_coord\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"chr19:51630101-51645545;-1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_chromosome\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 3,\n        \"samples\": [\n          \"11\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_number\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 1,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_chromosome_encoding\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 6,\n        \"min\": 2,\n        \"max\": 19,\n        \"num_unique_values\": 3,\n        \"samples\": [\n          11\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 36
        }
      ],
      "source": [
        "cur_data.adata.obs.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "1dc8df38",
      "metadata": {
        "id": "1dc8df38"
      },
      "source": [
        "### Add guide RNA information"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "d0d8699f",
      "metadata": {
        "id": "d0d8699f"
      },
      "source": [
        "The original pre-print does not contain information about guide RNAs in the `obs` slot.\n",
        "\n",
        "However, the pre-print references the source of the library - [Maximizing CRISPRi efficacy and accessibility with dual-sgRNA libraries and optimal effectors. Replogle et al 2022](https://doi.org/10.7554/eLife.81856).\n",
        "\n",
        "Supplementary table s4 contains the information about guide RNAs used in the experiment:\n",
        "\n",
        "Supplementary table S4: https://elifesciences.org/download/aHR0cHM6Ly9jZG4uZWxpZmVzY2llbmNlcy5vcmcvYXJ0aWNsZXMvODE4NTYvZWxpZmUtODE4NTYtc3VwcDQtdjIueGxzeA--/elife-81856-supp4-v2.xlsx?_hash=AmEXp%2BOl1z1Y4JCBAaVhST5ImQmKAGdfEjGhuJbSJ9Q%3D"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "41bf47fe",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 649
        },
        "id": "41bf47fe",
        "outputId": "6cc041a8-cb0f-4907-d52c-e827b6228237"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                             guide_sequence\n",
              "index                                                                      \n",
              "AAACCAAAGACATGTT-HCT116_Batch1    GGCCCAGCGCGCCCGACCGG|GAGAGAGCGCGCCAGGGCGA\n",
              "AAACCAAAGACCCAAC-HCT116_Batch1    GCCCGAGGCCCTGACTCCCA|GTATGGGACCTGGTGACCTC\n",
              "AAACCAAAGAGGTACG-HCT116_Batch1    GTGCAAGGCGCGATCCAAGA|GGATTTACCCGAGCGTGTTG\n",
              "AAACCAAAGCGATTAT-HCT116_Batch1    GAAGGCCCTGGTCCCGATAC|GAGTTCCCACAGCACCCCAG\n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1    GGCGCAGGTGAGCTCTCCTG|GTTGTAGGGAGGATCCAGTG\n",
              "...                                                                     ...\n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109  GAGACCCCGAGTGGGAACGG|GCCTGAGGGAGACCCCGAGT\n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109  GTGAAGAGAGAGGCAATCCC|GGGCAATCCCAGGAGAGGGG\n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109  GTCCGACGAGGTCAGCTGGG|GATGGCCATCAGCTCCGACG\n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109  GGCGATGGCTCGAGCTCGGG|GGCCCATCCAGCCCCCTTCA\n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  GATGAGCCCGAGGGACTGAG|GTGGAGGTCGCTGGTGGAGT\n",
              "\n",
              "[3409169 rows x 1 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-bd452635-c863-4151-9478-d50927d1af41\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>guide_sequence</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACATGTT-HCT116_Batch1</th>\n",
              "      <td>GGCCCAGCGCGCCCGACCGG|GAGAGAGCGCGCCAGGGCGA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACCCAAC-HCT116_Batch1</th>\n",
              "      <td>GCCCGAGGCCCTGACTCCCA|GTATGGGACCTGGTGACCTC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGGTACG-HCT116_Batch1</th>\n",
              "      <td>GTGCAAGGCGCGATCCAAGA|GGATTTACCCGAGCGTGTTG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGCGATTAT-HCT116_Batch1</th>\n",
              "      <td>GAAGGCCCTGGTCCCGATAC|GAGTTCCCACAGCACCCCAG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGGCTTAAT-HCT116_Batch1</th>\n",
              "      <td>GGCGCAGGTGAGCTCTCCTG|GTTGTAGGGAGGATCCAGTG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGCAGGGAATCG-HCT116_Batch109</th>\n",
              "      <td>GAGACCCCGAGTGGGAACGG|GCCTGAGGGAGACCCCGAGT</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCACGCGG-HCT116_Batch109</th>\n",
              "      <td>GTGAAGAGAGAGGCAATCCC|GGGCAATCCCAGGAGAGGGG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCTGACTC-HCT116_Batch109</th>\n",
              "      <td>GTCCGACGAGGTCAGCTGGG|GATGGCCATCAGCTCCGACG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCTTGATT-HCT116_Batch109</th>\n",
              "      <td>GGCGATGGCTCGAGCTCGGG|GGCCCATCCAGCCCCCTTCA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTGTAGTTA-HCT116_Batch109</th>\n",
              "      <td>GATGAGCCCGAGGGACTGAG|GTGGAGGTCGCTGGTGGAGT</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3409169 rows × 1 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-bd452635-c863-4151-9478-d50927d1af41')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-bd452635-c863-4151-9478-d50927d1af41 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-bd452635-c863-4151-9478-d50927d1af41');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-125388ed-4de5-47a7-96ae-a1c5ece59b12\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-125388ed-4de5-47a7-96ae-a1c5ece59b12')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-125388ed-4de5-47a7-96ae-a1c5ece59b12 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "  <div id=\"id_417ce68c-a0f0-416d-a646-065f283fbc08\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('obj_df')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_417ce68c-a0f0-416d-a646-065f283fbc08 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('obj_df');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "obj_df"
            }
          },
          "metadata": {},
          "execution_count": 37
        }
      ],
      "source": [
        "# get synonyms from the gene ontology\n",
        "syn_df = cur_data.gene_ont[[\"synonym\", \"gene_symbol\"]].drop_duplicates()\n",
        "syn_df[\"gene_symbol\"] = syn_df[\"gene_symbol\"].fillna(\"\")\n",
        "syn_df[\"synonym\"] = syn_df[\"synonym\"].fillna(\"\")\n",
        "syn_df[\"all_genes\"] = (\n",
        "    (syn_df[\"gene_symbol\"] + \"|\" + syn_df[\"synonym\"]).str.lstrip(\"|\").str.rstrip(\"|\")\n",
        ")\n",
        "syn_df[\"all_genes_match\"] = syn_df[\"all_genes\"].str.split(\"|\")\n",
        "syn_df = syn_df.explode(\"all_genes_match\")\n",
        "syn_df[\"all_genes_match\"] = syn_df[\"all_genes_match\"].str.upper()\n",
        "\n",
        "# read data from supplementary files\n",
        "guide_ids_df = pd.read_csv(\n",
        "    \"Perturbseq/supplementary/orion_2025_weissman_dual_crispri_sgrnas.csv\"\n",
        ")\n",
        "guide_ids_df[\"gene\"] = guide_ids_df[\"gene\"].str.upper()\n",
        "\n",
        "guide_ids_df = guide_ids_df.merge(\n",
        "    syn_df, left_on=\"gene\", right_on=\"all_genes_match\", how=\"left\"\n",
        ")\n",
        "\n",
        "guide_ids_df[\"all_genes\"] = guide_ids_df[\"all_genes\"].str.split(\"|\")\n",
        "guide_ids_df = guide_ids_df.explode(\"all_genes\").drop_duplicates()\n",
        "\n",
        "# create a perturbation name column\n",
        "guide_ids_df[\"perturbation_name\"] = (\n",
        "    guide_ids_df[\"all_genes\"]\n",
        "    + \"_\"\n",
        "    + guide_ids_df[\"transcript\"]\n",
        "    + \"|\"\n",
        "    + guide_ids_df[\"all_genes\"]\n",
        "    + \"_\"\n",
        "    + guide_ids_df[\"transcript\"]\n",
        ")\n",
        "\n",
        "guide_ids_df[\"perturbation_name\"] = guide_ids_df[\"perturbation_name\"].str.upper()\n",
        "\n",
        "# replace commas with underscores in the perturbation name\n",
        "guide_ids_df[\"perturbation_name\"] = guide_ids_df[\"perturbation_name\"].str.replace(\n",
        "    \",\", \"_\"\n",
        ")\n",
        "\n",
        "# create a dual guide RNA sequence column\n",
        "guide_ids_df[\"dual_guide_sequence\"] = (\n",
        "    guide_ids_df[\"protospacer_A\"] + \"|\" + guide_ids_df[\"protospacer_B\"]\n",
        ").str.upper()\n",
        "\n",
        "guide_ids_df.loc[\n",
        "    guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"perturbation_name\"\n",
        "] = (\n",
        "    guide_ids_df.loc[\n",
        "        guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"gene\"\n",
        "    ]\n",
        "    + \"_\"\n",
        "    + guide_ids_df.loc[\n",
        "        guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"transcript\"\n",
        "    ]\n",
        "    + \"|\"\n",
        "    + guide_ids_df.loc[\n",
        "        guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"gene\"\n",
        "    ]\n",
        "    + \"_\"\n",
        "    + guide_ids_df.loc[\n",
        "        guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"transcript\"\n",
        "    ]\n",
        ")\n",
        "\n",
        "# replace NEGATIVE_CONTROL with actual non targeting guide names\n",
        "guide_ids_df.loc[guide_ids_df[\"gene\"] == \"NEGATIVE_CONTROL\", \"perturbation_name\"] = (\n",
        "    guide_ids_df.loc[guide_ids_df[\"gene\"] == \"NEGATIVE_CONTROL\", \"sgID_AB\"].str.upper()\n",
        ")\n",
        "\n",
        "# create dictionaries for guide sequences mapping\n",
        "guide_ids_dict = (\n",
        "    guide_ids_df[[\"perturbation_name\", \"dual_guide_sequence\"]]\n",
        "    .dropna()\n",
        "    .drop_duplicates()\n",
        "    .set_index(\"perturbation_name\")[\"dual_guide_sequence\"]\n",
        "    .to_dict()\n",
        ")\n",
        "\n",
        "# add guide sequences to the obs dataframe\n",
        "obj_df = cur_data.adata.obs.copy()\n",
        "\n",
        "# remove the numeric suffices from the perturbation names\n",
        "obj_df[\"perturbation_name_matching\"] = obj_df[\"perturbation_name\"].str.replace(\n",
        "    r\"-\\d(?=\\||$)\", \"\", regex=True\n",
        ")\n",
        "obj_df[\"perturbation_name_matching\"] = obj_df[\"perturbation_name_matching\"].str.upper()\n",
        "\n",
        "# map the guide sequences to the perturbation names\n",
        "obj_df[\"guide_sequence\"] = obj_df[\"perturbation_name_matching\"].map(guide_ids_dict)\n",
        "\n",
        "obj_df = obj_df[[\"guide_sequence\"]]\n",
        "\n",
        "obj_df"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "51ed3fec",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 649
        },
        "id": "51ed3fec",
        "outputId": "af42c954-3be5-4d03-f882-ab51fde1a069"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                  pct_counts_mt  pass_guide_filter  \\\n",
              "index                                                                \n",
              "AAACCAAAGACATGTT-HCT116_Batch1         6.161162               True   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1         3.259871               True   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1         3.664498               True   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1         4.167945               True   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1         2.489183               True   \n",
              "...                                         ...                ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109       4.289229               True   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109       8.142244               True   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109      24.518473               True   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109       9.901894               True   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109       7.619245               True   \n",
              "\n",
              "                                  num_features  total_counts_mt  total_counts  \\\n",
              "index                                                                           \n",
              "AAACCAAAGACATGTT-HCT116_Batch1               2           1179.0       19136.0   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1               2           1562.0       47916.0   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1               2           1042.0       28435.0   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1               2           1087.0       26080.0   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1               2            955.0       38366.0   \n",
              "...                                        ...              ...           ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109             2            767.0       17882.0   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109             2            664.0        8155.0   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109             2           1553.0        6334.0   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109             2            868.0        8766.0   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109             2            738.0        9686.0   \n",
              "\n",
              "                                           sample  n_genes_by_counts  \\\n",
              "index                                                                  \n",
              "AAACCAAAGACATGTT-HCT116_Batch1      HCT116_Batch1               4883   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1      HCT116_Batch1               8130   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1      HCT116_Batch1               6531   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1      HCT116_Batch1               5931   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1      HCT116_Batch1               7157   \n",
              "...                                           ...                ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109  HCT116_Batch109               5107   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109  HCT116_Batch109               3643   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109  HCT116_Batch109               2831   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109  HCT116_Batch109               3868   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  HCT116_Batch109               4095   \n",
              "\n",
              "                                              perturbation_name  \\\n",
              "index                                                             \n",
              "AAACCAAAGACATGTT-HCT116_Batch1          ST14_P1P2-1|ST14_P1P2-2   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1    SIGLEC5_P1P2-1|SIGLEC5_P1P2-2   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1        VSNL1_P1P2-1|VSNL1_P1P2-2   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1        KCNK7_P1P2-1|KCNK7_P1P2-2   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1        APOA4_P1P2-1|APOA4_P1P2-2   \n",
              "...                                                         ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109        ACP6_P1P2-1|ACP6_P1P2-2   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109      RNF39_P1P2-1|RNF39_P1P2-2   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109              AFP_P2-1|AFP_P2-2   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109    METTL9_P1P2-1|METTL9_P1P2-2   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  SLC5A10_P1P2-1|SLC5A10_P1P2-2   \n",
              "\n",
              "                                 perturbed_target_ensg  \\\n",
              "index                                                    \n",
              "AAACCAAAGACATGTT-HCT116_Batch1         ENSG00000149418   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1         ENSG00000105501   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1         ENSG00000163032   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1         ENSG00000173338   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1         ENSG00000110244   \n",
              "...                                                ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109       ENSG00000162836   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109       ENSG00000204618   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109       ENSG00000081051   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109       ENSG00000197006   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109       ENSG00000154025   \n",
              "\n",
              "                                 perturbed_target_symbol  \\\n",
              "index                                                      \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                      ST14   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                   SIGLEC5   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                     VSNL1   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                     KCNK7   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                     APOA4   \n",
              "...                                                  ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109                    ACP6   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109                   RNF39   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109                     AFP   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109                  METTL9   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109                 SLC5A10   \n",
              "\n",
              "                                 perturbed_target_biotype  \\\n",
              "index                                                       \n",
              "AAACCAAAGACATGTT-HCT116_Batch1             protein_coding   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                     lncRNA   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1             protein_coding   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1             protein_coding   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1             protein_coding   \n",
              "...                                                   ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109           protein_coding   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109           protein_coding   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109           protein_coding   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109           protein_coding   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109           protein_coding   \n",
              "\n",
              "                                        perturbed_target_coord  \\\n",
              "index                                                            \n",
              "AAACCAAAGACATGTT-HCT116_Batch1     chr11:130159746-130210367;1   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1      chr19:51630101-51645545;-1   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1        chr2:17539020-17657018;1   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1      chr11:65592836-65595996;-1   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1    chr11:116820700-116823304;-1   \n",
              "...                                                        ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109   chr1:147629652-147670531;-1   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109     chr6:30070266-30075849;-1   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109      chr4:73431138-73456174;1   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109     chr16:21597218-21657471;1   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109     chr17:18950345-19022595;1   \n",
              "\n",
              "                                 perturbed_target_chromosome  \\\n",
              "index                                                          \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                            11   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                            19   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                             2   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                            11   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                            11   \n",
              "...                                                      ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109                           1   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109                           6   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109                           4   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109                          16   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109                          17   \n",
              "\n",
              "                                  perturbed_target_number  \\\n",
              "index                                                       \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                          1   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                          1   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                          1   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                          1   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                          1   \n",
              "...                                                   ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109                        1   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109                        1   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109                        1   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109                        1   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109                        1   \n",
              "\n",
              "                                  perturbed_target_chromosome_encoding  \\\n",
              "index                                                                    \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                                      11   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                                      19   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                                       2   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                                      11   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                                      11   \n",
              "...                                                                ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109                                     1   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109                                     6   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109                                     4   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109                                    16   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109                                    17   \n",
              "\n",
              "                                                             guide_sequence  \n",
              "index                                                                        \n",
              "AAACCAAAGACATGTT-HCT116_Batch1    GGCCCAGCGCGCCCGACCGG|GAGAGAGCGCGCCAGGGCGA  \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1    GCCCGAGGCCCTGACTCCCA|GTATGGGACCTGGTGACCTC  \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1    GTGCAAGGCGCGATCCAAGA|GGATTTACCCGAGCGTGTTG  \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1    GAAGGCCCTGGTCCCGATAC|GAGTTCCCACAGCACCCCAG  \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1    GGCGCAGGTGAGCTCTCCTG|GTTGTAGGGAGGATCCAGTG  \n",
              "...                                                                     ...  \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109  GAGACCCCGAGTGGGAACGG|GCCTGAGGGAGACCCCGAGT  \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109  GTGAAGAGAGAGGCAATCCC|GGGCAATCCCAGGAGAGGGG  \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109  GTCCGACGAGGTCAGCTGGG|GATGGCCATCAGCTCCGACG  \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109  GGCGATGGCTCGAGCTCGGG|GGCCCATCCAGCCCCCTTCA  \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  GATGAGCCCGAGGGACTGAG|GTGGAGGTCGCTGGTGGAGT  \n",
              "\n",
              "[3409169 rows x 16 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-9d3d6204-e601-43d7-aa07-79d09b33b453\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pct_counts_mt</th>\n",
              "      <th>pass_guide_filter</th>\n",
              "      <th>num_features</th>\n",
              "      <th>total_counts_mt</th>\n",
              "      <th>total_counts</th>\n",
              "      <th>sample</th>\n",
              "      <th>n_genes_by_counts</th>\n",
              "      <th>perturbation_name</th>\n",
              "      <th>perturbed_target_ensg</th>\n",
              "      <th>perturbed_target_symbol</th>\n",
              "      <th>perturbed_target_biotype</th>\n",
              "      <th>perturbed_target_coord</th>\n",
              "      <th>perturbed_target_chromosome</th>\n",
              "      <th>perturbed_target_number</th>\n",
              "      <th>perturbed_target_chromosome_encoding</th>\n",
              "      <th>guide_sequence</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACATGTT-HCT116_Batch1</th>\n",
              "      <td>6.161162</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1179.0</td>\n",
              "      <td>19136.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>4883</td>\n",
              "      <td>ST14_P1P2-1|ST14_P1P2-2</td>\n",
              "      <td>ENSG00000149418</td>\n",
              "      <td>ST14</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:130159746-130210367;1</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>11</td>\n",
              "      <td>GGCCCAGCGCGCCCGACCGG|GAGAGAGCGCGCCAGGGCGA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACCCAAC-HCT116_Batch1</th>\n",
              "      <td>3.259871</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1562.0</td>\n",
              "      <td>47916.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>8130</td>\n",
              "      <td>SIGLEC5_P1P2-1|SIGLEC5_P1P2-2</td>\n",
              "      <td>ENSG00000105501</td>\n",
              "      <td>SIGLEC5</td>\n",
              "      <td>lncRNA</td>\n",
              "      <td>chr19:51630101-51645545;-1</td>\n",
              "      <td>19</td>\n",
              "      <td>1</td>\n",
              "      <td>19</td>\n",
              "      <td>GCCCGAGGCCCTGACTCCCA|GTATGGGACCTGGTGACCTC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGGTACG-HCT116_Batch1</th>\n",
              "      <td>3.664498</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1042.0</td>\n",
              "      <td>28435.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>6531</td>\n",
              "      <td>VSNL1_P1P2-1|VSNL1_P1P2-2</td>\n",
              "      <td>ENSG00000163032</td>\n",
              "      <td>VSNL1</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr2:17539020-17657018;1</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>GTGCAAGGCGCGATCCAAGA|GGATTTACCCGAGCGTGTTG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGCGATTAT-HCT116_Batch1</th>\n",
              "      <td>4.167945</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1087.0</td>\n",
              "      <td>26080.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>5931</td>\n",
              "      <td>KCNK7_P1P2-1|KCNK7_P1P2-2</td>\n",
              "      <td>ENSG00000173338</td>\n",
              "      <td>KCNK7</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:65592836-65595996;-1</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>11</td>\n",
              "      <td>GAAGGCCCTGGTCCCGATAC|GAGTTCCCACAGCACCCCAG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGGCTTAAT-HCT116_Batch1</th>\n",
              "      <td>2.489183</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>955.0</td>\n",
              "      <td>38366.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>7157</td>\n",
              "      <td>APOA4_P1P2-1|APOA4_P1P2-2</td>\n",
              "      <td>ENSG00000110244</td>\n",
              "      <td>APOA4</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:116820700-116823304;-1</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>11</td>\n",
              "      <td>GGCGCAGGTGAGCTCTCCTG|GTTGTAGGGAGGATCCAGTG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGCAGGGAATCG-HCT116_Batch109</th>\n",
              "      <td>4.289229</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>767.0</td>\n",
              "      <td>17882.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>5107</td>\n",
              "      <td>ACP6_P1P2-1|ACP6_P1P2-2</td>\n",
              "      <td>ENSG00000162836</td>\n",
              "      <td>ACP6</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr1:147629652-147670531;-1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>GAGACCCCGAGTGGGAACGG|GCCTGAGGGAGACCCCGAGT</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCACGCGG-HCT116_Batch109</th>\n",
              "      <td>8.142244</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>664.0</td>\n",
              "      <td>8155.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>3643</td>\n",
              "      <td>RNF39_P1P2-1|RNF39_P1P2-2</td>\n",
              "      <td>ENSG00000204618</td>\n",
              "      <td>RNF39</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr6:30070266-30075849;-1</td>\n",
              "      <td>6</td>\n",
              "      <td>1</td>\n",
              "      <td>6</td>\n",
              "      <td>GTGAAGAGAGAGGCAATCCC|GGGCAATCCCAGGAGAGGGG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCTGACTC-HCT116_Batch109</th>\n",
              "      <td>24.518473</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1553.0</td>\n",
              "      <td>6334.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>2831</td>\n",
              "      <td>AFP_P2-1|AFP_P2-2</td>\n",
              "      <td>ENSG00000081051</td>\n",
              "      <td>AFP</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr4:73431138-73456174;1</td>\n",
              "      <td>4</td>\n",
              "      <td>1</td>\n",
              "      <td>4</td>\n",
              "      <td>GTCCGACGAGGTCAGCTGGG|GATGGCCATCAGCTCCGACG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCTTGATT-HCT116_Batch109</th>\n",
              "      <td>9.901894</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>868.0</td>\n",
              "      <td>8766.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>3868</td>\n",
              "      <td>METTL9_P1P2-1|METTL9_P1P2-2</td>\n",
              "      <td>ENSG00000197006</td>\n",
              "      <td>METTL9</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr16:21597218-21657471;1</td>\n",
              "      <td>16</td>\n",
              "      <td>1</td>\n",
              "      <td>16</td>\n",
              "      <td>GGCGATGGCTCGAGCTCGGG|GGCCCATCCAGCCCCCTTCA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTGTAGTTA-HCT116_Batch109</th>\n",
              "      <td>7.619245</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>738.0</td>\n",
              "      <td>9686.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>4095</td>\n",
              "      <td>SLC5A10_P1P2-1|SLC5A10_P1P2-2</td>\n",
              "      <td>ENSG00000154025</td>\n",
              "      <td>SLC5A10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr17:18950345-19022595;1</td>\n",
              "      <td>17</td>\n",
              "      <td>1</td>\n",
              "      <td>17</td>\n",
              "      <td>GATGAGCCCGAGGGACTGAG|GTGGAGGTCGCTGGTGGAGT</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3409169 rows × 16 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-9d3d6204-e601-43d7-aa07-79d09b33b453')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-9d3d6204-e601-43d7-aa07-79d09b33b453 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-9d3d6204-e601-43d7-aa07-79d09b33b453');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-261e6026-4ba8-4066-8e12-31f904fb23ef\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-261e6026-4ba8-4066-8e12-31f904fb23ef')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-261e6026-4ba8-4066-8e12-31f904fb23ef button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            }
          },
          "metadata": {},
          "execution_count": 38
        }
      ],
      "source": [
        "# merge guide sequences with main\n",
        "cur_data.adata.obs = cur_data.adata.obs.merge(\n",
        "    obj_df,\n",
        "    left_index=True,\n",
        "    right_index=True,\n",
        "    how=\"left\",\n",
        ")\n",
        "\n",
        "cur_data.adata.obs"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "8297b633",
      "metadata": {
        "id": "8297b633"
      },
      "source": [
        "### Add metadata"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "f7c92f51",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f7c92f51",
        "outputId": "50426875-f230-41fb-d365-6e1ce40fda61"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Column dataset_id added to adata.obs\n",
            "Column sample_id added to adata.obs\n",
            "Column treatment_label added to adata.obs\n",
            "Column treatment_id added to adata.obs\n",
            "Column perturbation_type_label added to adata.obs\n",
            "Column perturbation_type_id added to adata.obs\n",
            "Column model_system_label added to adata.obs\n",
            "Column model_system_id added to adata.obs\n",
            "Column data_modality added to adata.obs\n",
            "Column significant added to adata.obs\n",
            "Column significance_criteria added to adata.obs\n",
            "Column score_interpretation added to adata.obs\n",
            "Column technical_replicate added to adata.obs\n",
            "Column biological_replicate added to adata.obs\n",
            "Column tissue added to adata.obs\n",
            "Column cell_type added to adata.obs\n",
            "Column cell_line added to adata.obs\n",
            "Column timepoint added to adata.obs\n",
            "Column disease added to adata.obs\n",
            "Column species added to adata.obs\n",
            "Column sex_label added to adata.obs\n",
            "Column sex_id added to adata.obs\n",
            "Column developmental_stage_label added to adata.obs\n",
            "Column developmental_stage_id added to adata.obs\n",
            "Column study_title added to adata.obs\n",
            "Column study_uri added to adata.obs\n",
            "Column study_year added to adata.obs\n",
            "Column first_author added to adata.obs\n",
            "Column last_author added to adata.obs\n",
            "Column experiment_title added to adata.obs\n",
            "Column experiment_summary added to adata.obs\n",
            "Column number_of_perturbed_targets added to adata.obs\n",
            "Column number_of_perturbed_samples added to adata.obs\n",
            "Column library_generation_type_id added to adata.obs\n",
            "Column library_generation_type_label added to adata.obs\n",
            "Column library_generation_method_id added to adata.obs\n",
            "Column library_generation_method_label added to adata.obs\n",
            "Column enzyme_delivery_method_id added to adata.obs\n",
            "Column enzyme_delivery_method_label added to adata.obs\n",
            "Column library_delivery_method_id added to adata.obs\n",
            "Column library_delivery_method_label added to adata.obs\n",
            "Column enzyme_integration_state_id added to adata.obs\n",
            "Column enzyme_integration_state_label added to adata.obs\n",
            "Column library_integration_state_id added to adata.obs\n",
            "Column library_integration_state_label added to adata.obs\n",
            "Column enzyme_expression_control_id added to adata.obs\n",
            "Column enzyme_expression_control_label added to adata.obs\n",
            "Column library_expression_control_id added to adata.obs\n",
            "Column library_expression_control_label added to adata.obs\n",
            "Column library_name added to adata.obs\n",
            "Column library_uri added to adata.obs\n",
            "Column library_format_id added to adata.obs\n",
            "Column library_format_label added to adata.obs\n",
            "Column library_scope_id added to adata.obs\n",
            "Column library_scope_label added to adata.obs\n",
            "Column library_perturbation_type_id added to adata.obs\n",
            "Column library_perturbation_type_label added to adata.obs\n",
            "Column library_manufacturer added to adata.obs\n",
            "Column library_lentiviral_generation added to adata.obs\n",
            "Column library_grnas_per_target added to adata.obs\n",
            "Column library_total_grnas added to adata.obs\n",
            "Column library_total_variants added to adata.obs\n",
            "Column readout_dimensionality_id added to adata.obs\n",
            "Column readout_dimensionality_label added to adata.obs\n",
            "Column readout_type_id added to adata.obs\n",
            "Column readout_type_label added to adata.obs\n",
            "Column readout_technology_id added to adata.obs\n",
            "Column readout_technology_label added to adata.obs\n",
            "Column method_name_id added to adata.obs\n",
            "Column method_name_label added to adata.obs\n",
            "Column method_uri added to adata.obs\n",
            "Column sequencing_library_kit_id added to adata.obs\n",
            "Column sequencing_library_kit_label added to adata.obs\n",
            "Column sequencing_platform_id added to adata.obs\n",
            "Column sequencing_platform_label added to adata.obs\n",
            "Column sequencing_strategy_id added to adata.obs\n",
            "Column sequencing_strategy_label added to adata.obs\n",
            "Column software_counts_id added to adata.obs\n",
            "Column software_counts_label added to adata.obs\n",
            "Column software_analysis_id added to adata.obs\n",
            "Column software_analysis_label added to adata.obs\n",
            "Column reference_genome_id added to adata.obs\n",
            "Column reference_genome_label added to adata.obs\n",
            "Column license_label added to adata.obs\n",
            "Column license_id added to adata.obs\n",
            "Column associated_datasets added to adata.obs\n"
          ]
        }
      ],
      "source": [
        "cur_data.create_columns(\n",
        "    slot=\"obs\",\n",
        "    col_dict={\n",
        "        \"dataset_id\": cur_data.dataset_id,\n",
        "        \"sample_id\": range(1, cur_data.adata.obs.shape[0] + 1),\n",
        "        # treatment\n",
        "        \"treatment_label\": None,\n",
        "        \"treatment_id\": None,\n",
        "        # perturbation type\n",
        "        \"perturbation_type_label\": \"CRISPRi\",\n",
        "        \"perturbation_type_id\": None,\n",
        "        # model system\n",
        "        \"model_system_label\": \"cell_line\",\n",
        "        \"model_system_id\": None,\n",
        "\n",
        "        \"data_modality\": \"Perturb-seq\",\n",
        "        \"significant\": None,\n",
        "        \"significance_criteria\": None,\n",
        "        \"score_interpretation\": None,\n",
        "\n",
        "        # replicates\n",
        "        \"technical_replicate\": None,\n",
        "        \"biological_replicate\": None,\n",
        "\n",
        "        'tissue': 'colon',\n",
        "        'cell_type': 'colon epithelial cell',\n",
        "        'cell_line': 'HCT 116 cell',\n",
        "        \"timepoint\": \"P0DT0H0M0S\",\n",
        "        'disease': 'colon carcinoma',\n",
        "\n",
        "        \"species\": \"Homo sapiens\",\n",
        "        \"sex_label\": \"male\",\n",
        "        \"sex_id\": None,\n",
        "        \"developmental_stage_label\": \"adult\",\n",
        "        \"developmental_stage_id\": None,\n",
        "\n",
        "        \"study_title\": \"X-Atlas/Orion: Genome-wide Perturb-seq Datasets via a Scalable Fix-Cryopreserve Platform for Training Dose-Dependent Biological Foundation Models\",\n",
        "        \"study_uri\": \"https://doi.org/10.1101/2025.06.11.659105\",\n",
        "        \"study_year\": 2025,\n",
        "        \"first_author\": \"Ann Huang\",\n",
        "        \"last_author\": \"Ci Chu\",\n",
        "\n",
        "        \"experiment_title\": \"Over 3.4 million HCT116 cells were transfected with dual gRNAs targeting the same gene, and whole genome CRISPRi Perturb-seq was performed.\",\n",
        "        \"experiment_summary\": \"The researchers developed the Fix-Cryopreserve-ScRNAseq (FiCS) Perturb-seq platform to enable scalable and reproducible genome-wide perturbation screening in human cell lines. Using this platform, they generated a genome-wide dual-guide CRISPRi Perturb-seq dataset in HCT116 colon carcinoma cells. The dataset includes over 3.4 million deeply sequenced cells, with each cell yielding over 16,000 unique molecular identifiers (UMIs).\",\n",
        "        \"number_of_perturbed_targets\": len(set(cur_data.adata.obs['perturbed_target_coord'])),\n",
        "        \"number_of_perturbed_samples\": cur_data.adata.obs.shape[0],\n",
        "\n",
        "        \"library_generation_type_id\": \"EFO:0022868\",\n",
        "        \"library_generation_type_label\": \"endogenous\",\n",
        "\n",
        "        \"library_generation_method_id\": \"EFO:0022895\",\n",
        "        \"library_generation_method_label\": \"dCas9-KRAB\",\n",
        "\n",
        "        \"enzyme_delivery_method_id\": None,\n",
        "        \"enzyme_delivery_method_label\": \"nanoparticle-mediated transfection\",\n",
        "\n",
        "        \"library_delivery_method_id\": None,\n",
        "        \"library_delivery_method_label\": \"lentivirus transduction\",\n",
        "\n",
        "        \"enzyme_integration_state_id\": None,\n",
        "        \"enzyme_integration_state_label\": \"random locus integration\",\n",
        "\n",
        "        \"library_integration_state_id\": None,\n",
        "        \"library_integration_state_label\": \"random locus integration\",\n",
        "\n",
        "        \"enzyme_expression_control_id\": None,\n",
        "        \"enzyme_expression_control_label\": \"constitutive transgene expression\",\n",
        "\n",
        "        \"library_expression_control_id\": None,\n",
        "        \"library_expression_control_label\": \"constitutive transgene expression\",\n",
        "\n",
        "        \"library_name\": \"custom\",\n",
        "        \"library_uri\": None,\n",
        "\n",
        "        \"library_format_id\": None,\n",
        "        \"library_format_label\": \"pooled\",\n",
        "\n",
        "        \"library_scope_id\": None,\n",
        "        \"library_scope_label\": \"genome-wide\",\n",
        "\n",
        "        \"library_perturbation_type_id\": None,\n",
        "        \"library_perturbation_type_label\": \"inhibition\",\n",
        "\n",
        "        \"library_manufacturer\": \"Weissman\",\n",
        "        \"library_lentiviral_generation\": \"3\",\n",
        "        \"library_grnas_per_target\": \"2\",\n",
        "        \"library_total_grnas\": \"20845\",\n",
        "        \"library_total_variants\": None,\n",
        "\n",
        "        \"readout_dimensionality_id\": None,\n",
        "        \"readout_dimensionality_label\": \"high-dimensional assay\",\n",
        "\n",
        "        \"readout_type_id\": None,\n",
        "        \"readout_type_label\": \"transcriptomic\",\n",
        "\n",
        "        \"readout_technology_id\": None,\n",
        "        \"readout_technology_label\": \"single-cell rna-seq\",\n",
        "\n",
        "        \"method_name_id\": None,\n",
        "        \"method_name_label\": \"Perturb-seq\",\n",
        "\n",
        "        \"method_uri\": None,\n",
        "\n",
        "        \"sequencing_library_kit_id\": None,\n",
        "        \"sequencing_library_kit_label\": \"10x Genomics Chromium GEM-X Single Cell 5-prime kit v3\",\n",
        "\n",
        "        \"sequencing_platform_id\": None,\n",
        "        \"sequencing_platform_label\": \"Illumina NovaSeq X Plus\",\n",
        "\n",
        "        \"sequencing_strategy_id\": None,\n",
        "        \"sequencing_strategy_label\": \"barcode sequencing\",\n",
        "\n",
        "        \"software_counts_id\": None,\n",
        "        \"software_counts_label\": \"CellRanger\",\n",
        "\n",
        "        \"software_analysis_id\": None,\n",
        "        \"software_analysis_label\": \"custom\",\n",
        "\n",
        "        \"reference_genome_id\": None,\n",
        "        \"reference_genome_label\": \"GRCh38\",\n",
        "\n",
        "        \"license_label\": \"CC BY-NC-SA 4.0\",\n",
        "        \"license_id\": \"SWO:1000090\",\n",
        "\n",
        "        \"associated_datasets\": json.dumps([\n",
        "            {\n",
        "                \"dataset_accession\": \"HCT116_filtered_dual_guide_cells\",\n",
        "                \"dataset_uri\": \"https://plus.figshare.com/ndownloader/files/55021257\",\n",
        "                \"dataset_description\": \"Filtered raw counts\",\n",
        "                \"dataset_file_name\": \"HCT116_filtered_dual_guide_cells.h5ad\",\n",
        "            }\n",
        "        ])\n",
        "    }\n",
        ")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "302952a7",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 649
        },
        "id": "302952a7",
        "outputId": "fa555384-5f26-4013-d65c-c6e56437bd48"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                  pct_counts_mt  pass_guide_filter  \\\n",
              "index                                                                \n",
              "AAACCAAAGACATGTT-HCT116_Batch1         6.161162               True   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1         3.259871               True   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1         3.664498               True   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1         4.167945               True   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1         2.489183               True   \n",
              "...                                         ...                ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109       4.289229               True   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109       8.142244               True   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109      24.518473               True   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109       9.901894               True   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109       7.619245               True   \n",
              "\n",
              "                                  num_features  total_counts_mt  total_counts  \\\n",
              "index                                                                           \n",
              "AAACCAAAGACATGTT-HCT116_Batch1               2           1179.0       19136.0   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1               2           1562.0       47916.0   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1               2           1042.0       28435.0   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1               2           1087.0       26080.0   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1               2            955.0       38366.0   \n",
              "...                                        ...              ...           ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109             2            767.0       17882.0   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109             2            664.0        8155.0   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109             2           1553.0        6334.0   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109             2            868.0        8766.0   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109             2            738.0        9686.0   \n",
              "\n",
              "                                           sample  n_genes_by_counts  \\\n",
              "index                                                                  \n",
              "AAACCAAAGACATGTT-HCT116_Batch1      HCT116_Batch1               4883   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1      HCT116_Batch1               8130   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1      HCT116_Batch1               6531   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1      HCT116_Batch1               5931   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1      HCT116_Batch1               7157   \n",
              "...                                           ...                ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109  HCT116_Batch109               5107   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109  HCT116_Batch109               3643   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109  HCT116_Batch109               2831   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109  HCT116_Batch109               3868   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  HCT116_Batch109               4095   \n",
              "\n",
              "                                              perturbation_name  \\\n",
              "index                                                             \n",
              "AAACCAAAGACATGTT-HCT116_Batch1          ST14_P1P2-1|ST14_P1P2-2   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1    SIGLEC5_P1P2-1|SIGLEC5_P1P2-2   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1        VSNL1_P1P2-1|VSNL1_P1P2-2   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1        KCNK7_P1P2-1|KCNK7_P1P2-2   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1        APOA4_P1P2-1|APOA4_P1P2-2   \n",
              "...                                                         ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109        ACP6_P1P2-1|ACP6_P1P2-2   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109      RNF39_P1P2-1|RNF39_P1P2-2   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109              AFP_P2-1|AFP_P2-2   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109    METTL9_P1P2-1|METTL9_P1P2-2   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  SLC5A10_P1P2-1|SLC5A10_P1P2-2   \n",
              "\n",
              "                                 perturbed_target_ensg  \\\n",
              "index                                                    \n",
              "AAACCAAAGACATGTT-HCT116_Batch1         ENSG00000149418   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1         ENSG00000105501   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1         ENSG00000163032   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1         ENSG00000173338   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1         ENSG00000110244   \n",
              "...                                                ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109       ENSG00000162836   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109       ENSG00000204618   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109       ENSG00000081051   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109       ENSG00000197006   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109       ENSG00000154025   \n",
              "\n",
              "                                 perturbed_target_symbol  ...  \\\n",
              "index                                                     ...   \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                      ST14  ...   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                   SIGLEC5  ...   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                     VSNL1  ...   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                     KCNK7  ...   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                     APOA4  ...   \n",
              "...                                                  ...  ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109                    ACP6  ...   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109                   RNF39  ...   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109                     AFP  ...   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109                  METTL9  ...   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109                 SLC5A10  ...   \n",
              "\n",
              "                                 sequencing_strategy_label software_counts_id  \\\n",
              "index                                                                           \n",
              "AAACCAAAGACATGTT-HCT116_Batch1          barcode sequencing               None   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1          barcode sequencing               None   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1          barcode sequencing               None   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1          barcode sequencing               None   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1          barcode sequencing               None   \n",
              "...                                                    ...                ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109        barcode sequencing               None   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109        barcode sequencing               None   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109        barcode sequencing               None   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109        barcode sequencing               None   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109        barcode sequencing               None   \n",
              "\n",
              "                                 software_counts_label  software_analysis_id  \\\n",
              "index                                                                          \n",
              "AAACCAAAGACATGTT-HCT116_Batch1              CellRanger                  None   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1              CellRanger                  None   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1              CellRanger                  None   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1              CellRanger                  None   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1              CellRanger                  None   \n",
              "...                                                ...                   ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109            CellRanger                  None   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109            CellRanger                  None   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109            CellRanger                  None   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109            CellRanger                  None   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109            CellRanger                  None   \n",
              "\n",
              "                                  software_analysis_label reference_genome_id  \\\n",
              "index                                                                           \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                     custom                None   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                     custom                None   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                     custom                None   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                     custom                None   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                     custom                None   \n",
              "...                                                   ...                 ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109                   custom                None   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109                   custom                None   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109                   custom                None   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109                   custom                None   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109                   custom                None   \n",
              "\n",
              "                                 reference_genome_label    license_label  \\\n",
              "index                                                                      \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                   GRCh38  CC BY-NC-SA 4.0   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                   GRCh38  CC BY-NC-SA 4.0   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                   GRCh38  CC BY-NC-SA 4.0   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                   GRCh38  CC BY-NC-SA 4.0   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                   GRCh38  CC BY-NC-SA 4.0   \n",
              "...                                                 ...              ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109                 GRCh38  CC BY-NC-SA 4.0   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109                 GRCh38  CC BY-NC-SA 4.0   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109                 GRCh38  CC BY-NC-SA 4.0   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109                 GRCh38  CC BY-NC-SA 4.0   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109                 GRCh38  CC BY-NC-SA 4.0   \n",
              "\n",
              "                                   license_id  \\\n",
              "index                                           \n",
              "AAACCAAAGACATGTT-HCT116_Batch1    SWO:1000090   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1    SWO:1000090   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1    SWO:1000090   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1    SWO:1000090   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1    SWO:1000090   \n",
              "...                                       ...   \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109  SWO:1000090   \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109  SWO:1000090   \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109  SWO:1000090   \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109  SWO:1000090   \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  SWO:1000090   \n",
              "\n",
              "                                                                associated_datasets  \n",
              "index                                                                                \n",
              "AAACCAAAGACATGTT-HCT116_Batch1    [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1    [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1    [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1    [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1    [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "...                                                                             ...  \n",
              "GTTGTGCAGGGAATCG-HCT116_Batch109  [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "GTTGTGGGTCACGCGG-HCT116_Batch109  [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "GTTGTGGGTCTGACTC-HCT116_Batch109  [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "GTTGTGGGTCTTGATT-HCT116_Batch109  [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "GTTGTGGGTGTAGTTA-HCT116_Batch109  [{\"dataset_accession\": \"HCT116_filtered_dual_g...  \n",
              "\n",
              "[3409169 rows x 102 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-d8a7c25a-617a-4ab7-a548-e95d87864763\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pct_counts_mt</th>\n",
              "      <th>pass_guide_filter</th>\n",
              "      <th>num_features</th>\n",
              "      <th>total_counts_mt</th>\n",
              "      <th>total_counts</th>\n",
              "      <th>sample</th>\n",
              "      <th>n_genes_by_counts</th>\n",
              "      <th>perturbation_name</th>\n",
              "      <th>perturbed_target_ensg</th>\n",
              "      <th>perturbed_target_symbol</th>\n",
              "      <th>...</th>\n",
              "      <th>sequencing_strategy_label</th>\n",
              "      <th>software_counts_id</th>\n",
              "      <th>software_counts_label</th>\n",
              "      <th>software_analysis_id</th>\n",
              "      <th>software_analysis_label</th>\n",
              "      <th>reference_genome_id</th>\n",
              "      <th>reference_genome_label</th>\n",
              "      <th>license_label</th>\n",
              "      <th>license_id</th>\n",
              "      <th>associated_datasets</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACATGTT-HCT116_Batch1</th>\n",
              "      <td>6.161162</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1179.0</td>\n",
              "      <td>19136.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>4883</td>\n",
              "      <td>ST14_P1P2-1|ST14_P1P2-2</td>\n",
              "      <td>ENSG00000149418</td>\n",
              "      <td>ST14</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACCCAAC-HCT116_Batch1</th>\n",
              "      <td>3.259871</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1562.0</td>\n",
              "      <td>47916.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>8130</td>\n",
              "      <td>SIGLEC5_P1P2-1|SIGLEC5_P1P2-2</td>\n",
              "      <td>ENSG00000105501</td>\n",
              "      <td>SIGLEC5</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGGTACG-HCT116_Batch1</th>\n",
              "      <td>3.664498</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1042.0</td>\n",
              "      <td>28435.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>6531</td>\n",
              "      <td>VSNL1_P1P2-1|VSNL1_P1P2-2</td>\n",
              "      <td>ENSG00000163032</td>\n",
              "      <td>VSNL1</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGCGATTAT-HCT116_Batch1</th>\n",
              "      <td>4.167945</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1087.0</td>\n",
              "      <td>26080.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>5931</td>\n",
              "      <td>KCNK7_P1P2-1|KCNK7_P1P2-2</td>\n",
              "      <td>ENSG00000173338</td>\n",
              "      <td>KCNK7</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGGCTTAAT-HCT116_Batch1</th>\n",
              "      <td>2.489183</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>955.0</td>\n",
              "      <td>38366.0</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>7157</td>\n",
              "      <td>APOA4_P1P2-1|APOA4_P1P2-2</td>\n",
              "      <td>ENSG00000110244</td>\n",
              "      <td>APOA4</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGCAGGGAATCG-HCT116_Batch109</th>\n",
              "      <td>4.289229</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>767.0</td>\n",
              "      <td>17882.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>5107</td>\n",
              "      <td>ACP6_P1P2-1|ACP6_P1P2-2</td>\n",
              "      <td>ENSG00000162836</td>\n",
              "      <td>ACP6</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCACGCGG-HCT116_Batch109</th>\n",
              "      <td>8.142244</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>664.0</td>\n",
              "      <td>8155.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>3643</td>\n",
              "      <td>RNF39_P1P2-1|RNF39_P1P2-2</td>\n",
              "      <td>ENSG00000204618</td>\n",
              "      <td>RNF39</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCTGACTC-HCT116_Batch109</th>\n",
              "      <td>24.518473</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>1553.0</td>\n",
              "      <td>6334.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>2831</td>\n",
              "      <td>AFP_P2-1|AFP_P2-2</td>\n",
              "      <td>ENSG00000081051</td>\n",
              "      <td>AFP</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTCTTGATT-HCT116_Batch109</th>\n",
              "      <td>9.901894</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>868.0</td>\n",
              "      <td>8766.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>3868</td>\n",
              "      <td>METTL9_P1P2-1|METTL9_P1P2-2</td>\n",
              "      <td>ENSG00000197006</td>\n",
              "      <td>METTL9</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGGGTGTAGTTA-HCT116_Batch109</th>\n",
              "      <td>7.619245</td>\n",
              "      <td>True</td>\n",
              "      <td>2</td>\n",
              "      <td>738.0</td>\n",
              "      <td>9686.0</td>\n",
              "      <td>HCT116_Batch109</td>\n",
              "      <td>4095</td>\n",
              "      <td>SLC5A10_P1P2-1|SLC5A10_P1P2-2</td>\n",
              "      <td>ENSG00000154025</td>\n",
              "      <td>SLC5A10</td>\n",
              "      <td>...</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>custom</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>CC BY-NC-SA 4.0</td>\n",
              "      <td>SWO:1000090</td>\n",
              "      <td>[{\"dataset_accession\": \"HCT116_filtered_dual_g...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3409169 rows × 102 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d8a7c25a-617a-4ab7-a548-e95d87864763')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-d8a7c25a-617a-4ab7-a548-e95d87864763 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-d8a7c25a-617a-4ab7-a548-e95d87864763');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-1d884116-8170-4a0b-a912-338945bcafda\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-1d884116-8170-4a0b-a912-338945bcafda')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-1d884116-8170-4a0b-a912-338945bcafda button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            }
          },
          "metadata": {},
          "execution_count": 40
        }
      ],
      "source": [
        "cur_data.adata.obs"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Curate replicate information"
      ],
      "metadata": {
        "id": "qKwT144Z576z"
      },
      "id": "qKwT144Z576z"
    },
    {
      "cell_type": "code",
      "source": [
        "cur_data.adata.obs['technical_replicate'] = cur_data.adata.obs['sample'].astype(str)"
      ],
      "metadata": {
        "id": "HXPvb212593M"
      },
      "id": "HXPvb212593M",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "id": "9bf05464",
      "metadata": {
        "id": "9bf05464"
      },
      "source": [
        "### Curate tissue information"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "b7ecab7b",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "b7ecab7b",
        "outputId": "ffd0f701-76c8-4f29-cb18-121ff9808081"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mapped 1 tissue ontology terms from `tissue` column to ontology terms\n",
            "DataFrame shape: (1, 4)\n",
            "--------------------------------------------------\n",
            "  input_column input_column_lower name_lower     ontology_id\n",
            "0        colon              colon      colon  UBERON:0001155\n",
            "--------------------------------------------------\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/lib/python3.12/functools.py:912: ImplicitModificationWarning: Transforming to str index.\n",
            "  return dispatch(args[0].__class__)(*args, **kw)\n"
          ]
        }
      ],
      "source": [
        "\n",
        "cur_data.standardize_ontology(\n",
        "    input_column='tissue',\n",
        "    column_type='term_name',\n",
        "    ontology_type='tissue',\n",
        "    overwrite=True\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "70436dda",
      "metadata": {
        "id": "70436dda"
      },
      "source": [
        "### Curate cell type information"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "859e388c",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "859e388c",
        "outputId": "bc56271f-a306-4ab6-8706-5c9511c262be"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mapped 1 cell_type ontology terms from `cell_type` column to ontology terms\n",
            "DataFrame shape: (1, 4)\n",
            "--------------------------------------------------\n",
            "            input_column     input_column_lower             name_lower  \\\n",
            "0  colon epithelial cell  colon epithelial cell  colon epithelial cell   \n",
            "\n",
            "  ontology_id  \n",
            "0  CL:0011108  \n",
            "--------------------------------------------------\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/lib/python3.12/functools.py:912: ImplicitModificationWarning: Transforming to str index.\n",
            "  return dispatch(args[0].__class__)(*args, **kw)\n"
          ]
        }
      ],
      "source": [
        "cur_data.standardize_ontology(\n",
        "    input_column='cell_type',\n",
        "    column_type='term_name',\n",
        "    ontology_type='cell_type'\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "fb440392",
      "metadata": {
        "id": "fb440392"
      },
      "source": [
        "### Curate cell line information"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "d63b5043",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "d63b5043",
        "outputId": "5ee06bbe-2324-4cae-a5cc-15f68edaa791"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mapped 1 cell_line ontology terms from `cell_line` column to ontology terms\n",
            "DataFrame shape: (1, 4)\n",
            "--------------------------------------------------\n",
            "   input_column input_column_lower    name_lower  ontology_id\n",
            "0  HCT 116 cell       hct 116 cell  hct 116 cell  CLO:0003665\n",
            "--------------------------------------------------\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/lib/python3.12/functools.py:912: ImplicitModificationWarning: Transforming to str index.\n",
            "  return dispatch(args[0].__class__)(*args, **kw)\n"
          ]
        }
      ],
      "source": [
        "cur_data.standardize_ontology(\n",
        "    input_column='cell_line',\n",
        "    column_type='term_name',\n",
        "    ontology_type='cell_line'\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "e09aecfa",
      "metadata": {
        "id": "e09aecfa"
      },
      "source": [
        "### Curate disease information"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "1ca9b7f8",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1ca9b7f8",
        "outputId": "8b452993-0875-4d62-94dd-c783c609500f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mapped 1 disease ontology terms from `disease` column to ontology terms\n",
            "DataFrame shape: (1, 4)\n",
            "--------------------------------------------------\n",
            "      input_column input_column_lower       name_lower    ontology_id\n",
            "0  colon carcinoma    colon carcinoma  colon carcinoma  MONDO:0002032\n",
            "--------------------------------------------------\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/lib/python3.12/functools.py:912: ImplicitModificationWarning: Transforming to str index.\n",
            "  return dispatch(args[0].__class__)(*args, **kw)\n"
          ]
        }
      ],
      "source": [
        "cur_data.standardize_ontology(\n",
        "    input_column='disease',\n",
        "    column_type='term_name',\n",
        "    ontology_type='disease'\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "32019fb8",
      "metadata": {
        "id": "32019fb8"
      },
      "source": [
        "### Match schema column order"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "a5d1e598",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "a5d1e598",
        "outputId": "fddaa235-9adf-45ed-bc94-fbc63957258d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Matched columns of adata.obs to the obs_schema.\n"
          ]
        }
      ],
      "source": [
        "cur_data.match_schema_columns(slot='obs')"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "e0048114",
      "metadata": {
        "id": "e0048114"
      },
      "source": [
        "### Validate obs metadata"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "sch = ObsSchema.to_schema()\n",
        "\n",
        "for col in list(sch.columns.keys()):\n",
        "    sch_subset = sch.select_columns([col])\n",
        "    try:\n",
        "        sch_subset.validate(cur_data.adata.obs[[col]], lazy=True)\n",
        "        print(f\"Validation successful for column {col}.\")\n",
        "    except Exception as e:\n",
        "        print(f\"Validation failed for column {col}: {e}\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "g9aaGF1l6ICi",
        "outputId": "ce421281-a575-4de4-9206-bc9c2e322a4e"
      },
      "id": "g9aaGF1l6ICi",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Validation successful for column dataset_id.\n",
            "Validation successful for column sample_id.\n",
            "Validation successful for column data_modality.\n",
            "Validation successful for column significant.\n",
            "Validation successful for column significance_criteria.\n",
            "Validation successful for column perturbation_name.\n",
            "Validation successful for column perturbed_target_coord.\n",
            "Validation successful for column perturbed_target_chromosome.\n",
            "Validation successful for column perturbed_target_chromosome_encoding.\n",
            "Validation successful for column perturbed_target_number.\n",
            "Validation successful for column perturbed_target_ensg.\n",
            "Validation successful for column perturbed_target_symbol.\n",
            "Validation successful for column perturbed_target_biotype.\n",
            "Validation successful for column guide_sequence.\n",
            "Validation successful for column perturbation_type_label.\n",
            "Validation successful for column perturbation_type_id.\n",
            "Validation successful for column timepoint.\n",
            "Validation successful for column treatment_label.\n",
            "Validation successful for column treatment_id.\n",
            "Validation successful for column technical_replicate.\n",
            "Validation successful for column biological_replicate.\n",
            "Validation successful for column model_system_label.\n",
            "Validation successful for column model_system_id.\n",
            "Validation successful for column species.\n",
            "Validation successful for column tissue_label.\n",
            "Validation successful for column tissue_id.\n",
            "Validation successful for column cell_type_label.\n",
            "Validation successful for column cell_type_id.\n",
            "Validation successful for column cell_line_label.\n",
            "Validation successful for column cell_line_id.\n",
            "Validation successful for column sex_label.\n",
            "Validation successful for column sex_id.\n",
            "Validation successful for column developmental_stage_label.\n",
            "Validation successful for column developmental_stage_id.\n",
            "Validation successful for column disease_label.\n",
            "Validation successful for column disease_id.\n",
            "Validation successful for column study_title.\n",
            "Validation successful for column study_uri.\n",
            "Validation successful for column study_year.\n",
            "Validation successful for column first_author.\n",
            "Validation successful for column last_author.\n",
            "Validation successful for column experiment_title.\n",
            "Validation successful for column experiment_summary.\n",
            "Validation successful for column number_of_perturbed_targets.\n",
            "Validation successful for column number_of_perturbed_samples.\n",
            "Validation successful for column library_generation_type_id.\n",
            "Validation successful for column library_generation_type_label.\n",
            "Validation successful for column library_generation_method_id.\n",
            "Validation successful for column library_generation_method_label.\n",
            "Validation successful for column enzyme_delivery_method_id.\n",
            "Validation successful for column enzyme_delivery_method_label.\n",
            "Validation successful for column library_delivery_method_id.\n",
            "Validation successful for column library_delivery_method_label.\n",
            "Validation successful for column enzyme_integration_state_id.\n",
            "Validation successful for column enzyme_integration_state_label.\n",
            "Validation successful for column library_integration_state_id.\n",
            "Validation successful for column library_integration_state_label.\n",
            "Validation successful for column enzyme_expression_control_id.\n",
            "Validation successful for column enzyme_expression_control_label.\n",
            "Validation successful for column library_expression_control_id.\n",
            "Validation successful for column library_expression_control_label.\n",
            "Validation successful for column library_name.\n",
            "Validation successful for column library_uri.\n",
            "Validation successful for column library_format_id.\n",
            "Validation successful for column library_format_label.\n",
            "Validation successful for column library_scope_id.\n",
            "Validation successful for column library_scope_label.\n",
            "Validation successful for column library_perturbation_type_id.\n",
            "Validation successful for column library_perturbation_type_label.\n",
            "Validation successful for column library_manufacturer.\n",
            "Validation successful for column library_lentiviral_generation.\n",
            "Validation successful for column library_grnas_per_target.\n",
            "Validation successful for column library_total_grnas.\n",
            "Validation failed for column library_total_variants: {\n",
            "    \"SCHEMA\": {\n",
            "        \"WRONG_DATATYPE\": [\n",
            "            {\n",
            "                \"schema\": \"ObsSchema\",\n",
            "                \"column\": \"library_total_variants\",\n",
            "                \"check\": \"dtype('int64')\",\n",
            "                \"error\": \"expected series 'library_total_variants' to have type int64, got object\"\n",
            "            }\n",
            "        ]\n",
            "    }\n",
            "}\n",
            "Validation successful for column readout_dimensionality_id.\n",
            "Validation successful for column readout_dimensionality_label.\n",
            "Validation successful for column readout_type_id.\n",
            "Validation successful for column readout_type_label.\n",
            "Validation successful for column readout_technology_id.\n",
            "Validation successful for column readout_technology_label.\n",
            "Validation successful for column method_name_id.\n",
            "Validation successful for column method_name_label.\n",
            "Validation successful for column method_uri.\n",
            "Validation successful for column sequencing_library_kit_id.\n",
            "Validation successful for column sequencing_library_kit_label.\n",
            "Validation successful for column sequencing_platform_id.\n",
            "Validation successful for column sequencing_platform_label.\n",
            "Validation successful for column sequencing_strategy_id.\n",
            "Validation successful for column sequencing_strategy_label.\n",
            "Validation successful for column software_counts_id.\n",
            "Validation successful for column software_counts_label.\n",
            "Validation successful for column software_analysis_id.\n",
            "Validation successful for column software_analysis_label.\n",
            "Validation successful for column score_interpretation.\n",
            "Validation successful for column reference_genome_id.\n",
            "Validation successful for column reference_genome_label.\n",
            "Validation successful for column associated_datasets.\n",
            "Validation successful for column license_label.\n",
            "Validation successful for column license_id.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "a0a39c5d",
      "metadata": {
        "id": "a0a39c5d"
      },
      "outputs": [],
      "source": [
        "# cur_data.validate_data(slot='obs')"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "723924a2",
      "metadata": {
        "id": "723924a2"
      },
      "source": [
        "# VAR slot curation"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "bff48ef8",
      "metadata": {
        "id": "bff48ef8"
      },
      "source": [
        "### Standardise genes"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "3a410193",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3a410193",
        "outputId": "ad85cf3e-600a-4654-c362-352301314624"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Variable data:\n",
            "DataFrame shape: (38606, 6)\n",
            "--------------------------------------------------\n",
            "                    mt  n_cells_by_counts   mean_counts  \\\n",
            "DDX11L2          False               2719  2.777434e-04   \n",
            "MIR1302-2HG      False               2620  2.673255e-04   \n",
            "FAM138A          False                  3  3.034341e-07   \n",
            "ENSG00000290826  False                764  8.698444e-05   \n",
            "OR4F5            False                263  2.700564e-05   \n",
            "...                ...                ...           ...   \n",
            "ENSG00000277836  False                  0  0.000000e+00   \n",
            "ENSG00000278633  False                106  1.092363e-05   \n",
            "ENSG00000276017  False                663  6.756467e-05   \n",
            "ENSG00000278817  False              35369  3.616227e-03   \n",
            "ENSG00000277196  False              48372  5.025274e-03   \n",
            "\n",
            "                 pct_dropout_by_counts  total_counts      gene_symbol  \n",
            "DDX11L2                      99.972499        2746.0          DDX11L2  \n",
            "MIR1302-2HG                  99.973500        2643.0      MIR1302-2HG  \n",
            "FAM138A                      99.999970           3.0          FAM138A  \n",
            "ENSG00000290826              99.992273         860.0  ENSG00000290826  \n",
            "OR4F5                        99.997340         267.0            OR4F5  \n",
            "...                                ...           ...              ...  \n",
            "ENSG00000277836             100.000000           0.0  ENSG00000277836  \n",
            "ENSG00000278633              99.998928         108.0  ENSG00000278633  \n",
            "ENSG00000276017              99.993294         668.0  ENSG00000276017  \n",
            "ENSG00000278817              99.642261       35753.0  ENSG00000278817  \n",
            "ENSG00000277196              99.510743       49684.0  ENSG00000277196  \n",
            "\n",
            "[38606 rows x 6 columns]\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.show_var()"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "cur_data.adata.var['ensembl_inputs'] = [e if e.startswith('ENSG') else None for e in cur_data.adata.var.index]\n",
        "cur_data.adata.var['gene_inputs'] = [e if not e.startswith('ENSG') else '' for e in cur_data.adata.var.index]\n",
        "\n",
        "var_df = cur_data.adata.var.copy()"
      ],
      "metadata": {
        "id": "rMnZ1cAP6pB9"
      },
      "id": "rMnZ1cAP6pB9",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "cur_data.create_columns(slot='var', col_dict={\n",
        "    'genes':cur_data.adata.var.index\n",
        "})"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "eLJlAP2w6thh",
        "outputId": "b83ed8c7-9d6d-4e5f-ffba-2c23f17fa8d8"
      },
      "id": "eLJlAP2w6thh",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Column genes added to adata.var\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "cur_data.standardize_genes(\n",
        "    slot=\"var\", input_column=\"gene_inputs\", input_column_type=\"gene_symbol\"\n",
        ")\n",
        "var_symb = cur_data.adata.var.copy()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ThSPcZ0E6v2F",
        "outputId": "6f3bc549-00f2-4178-fea3-d52a12f5a2aa"
      },
      "id": "ThSPcZ0E6v2F",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Mapping gene symbols: 100%|█████████████████████████████████| 25645/25645 [00:02<00:00, 9096.70it/s]\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--------------------------------------------------\n",
            "Successfully mapped 25586 out of 25645 gene symbols.\n",
            "--------------------------------------------------\n",
            "Couldn't map gene symbols: ['', 'ARHGEF19-AS1', 'LYST-AS1', 'LINC01115-1', 'LINC01119', 'LSP1P5-1', 'LINC01889', 'UPP2-IT1', 'LINC01238-1', 'ARF4-AS1', 'DENND6A-AS1', 'PRICKLE2-AS1-1', 'SLC41A3-AS1', 'IFT122P3', 'LINC00886', 'LINC02263', 'TEMN3-AS1', 'CTD-2350J17.1', 'LINC02119', 'LINC01033', 'CTC-338M12.4', 'MIR5689HG', 'RAET1E-AS1-1', 'LINC03021-1', 'LINC03023-1', 'TNFRSF10A-DT-1', 'LINC01605-1', 'NCRNA00250', 'PTGER4P2-CDK2AP2P2', 'LINC03025-1', 'LINC00484-1', 'LINC02913', 'DNAJC9-AS1-1', 'LINC02722', 'LINC02398', 'GPR84-AS1-1', 'LINC02399', 'SPATA13-1', 'SMIM2', 'SMIM2-IT1', 'PCDH9-AS4', 'HIF1A-AS1', 'HIF1A-AS3', 'VASH1-AS1', 'GOLGA8M-1', 'LINC02256-1', 'LINC01580', 'DDX11L16-1', 'NPIPA9-1', 'LINC02911', 'MKKS-1', 'MKKS-2', 'D21S2088E', 'LINC01422', 'ELFN2-1', 'ARMCX5-GPRASP2-1', 'PRY2', 'PRY', 'MAFIP']\n",
            "--------------------------------------------------\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "cur_data.standardize_genes(\n",
        "    slot=\"var\", input_column=\"ensembl_inputs\", input_column_type=\"ensembl_gene_id\"\n",
        ")\n",
        "var_ensg = cur_data.adata.var.copy()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fzYx3sON60G7",
        "outputId": "913665e0-694e-444e-dcff-502ef1d87b58"
      },
      "id": "fzYx3sON60G7",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Missing Ensembl IDs: ['ENSG00000272922', 'ENSG00000224481', 'ENSG00000259582', 'ENSG00000287139', 'ENSG00000251399', 'ENSG00000277559', 'ENSG00000276017', 'ENSG00000275636', 'ENSG00000286571', 'ENSG00000263331', 'ENSG00000290872', 'ENSG00000289531', 'ENSG00000203520', 'ENSG00000257962', 'ENSG00000241912', 'ENSG00000261386', 'ENSG00000287811', 'ENSG00000290709', 'ENSG00000258921', 'ENSG00000230649', 'ENSG00000286472', 'ENSG00000259274', 'ENSG00000225489', 'ENSG00000236412', 'ENSG00000259209', 'ENSG00000277630', 'ENSG00000286977', 'ENSG00000229454', 'ENSG00000290078', 'ENSG00000286633', 'ENSG00000264421', 'ENSG00000244260', 'ENSG00000249638', 'ENSG00000267105', 'ENSG00000255328', 'ENSG00000266588', 'ENSG00000286603', 'ENSG00000234405', 'ENSG00000258345', 'ENSG00000260163', 'ENSG00000250360', 'ENSG00000257454', 'ENSG00000232056', 'ENSG00000289448', 'ENSG00000267420', 'ENSG00000264790', 'ENSG00000267537', 'ENSG00000260511', 'ENSG00000287622', 'ENSG00000277423', 'ENSG00000288816', 'ENSG00000287614', 'ENSG00000235815', 'ENSG00000289176', 'ENSG00000230600', 'ENSG00000287400', 'ENSG00000254290', 'ENSG00000234223', 'ENSG00000291048', 'ENSG00000224819', 'ENSG00000258927', 'ENSG00000290100', 'ENSG00000251506', 'ENSG00000258918', 'ENSG00000286997', 'ENSG00000286087', 'ENSG00000248240', 'ENSG00000287411', 'ENSG00000287361', 'ENSG00000267058', 'ENSG00000287569', 'ENSG00000231212', 'ENSG00000232949', 'ENSG00000272948', 'ENSG00000224884', 'ENSG00000241772', 'ENSG00000287127', 'ENSG00000254364', 'ENSG00000277836', 'ENSG00000256321', 'ENSG00000288749', 'ENSG00000255160', 'ENSG00000232294', 'ENSG00000290183', 'ENSG00000285622', 'ENSG00000291258', 'ENSG00000254306', 'ENSG00000260759', 'ENSG00000261218', 'ENSG00000291260', 'ENSG00000268615', 'ENSG00000272953', 'ENSG00000249984', 'ENSG00000237899', 'ENSG00000272541', 'ENSG00000286911', 'ENSG00000273729', 'ENSG00000271792', 'ENSG00000265671', 'ENSG00000225916', 'ENSG00000248359', 'ENSG00000287941', 'ENSG00000291298', 'ENSG00000243176', 'ENSG00000290974', 'ENSG00000251678', 'ENSG00000287682', 'ENSG00000287954', 'ENSG00000257879', 'ENSG00000270157', 'ENSG00000261135', 'ENSG00000288950', 'ENSG00000283255', 'ENSG00000286906', 'ENSG00000289958', 'ENSG00000260983', 'ENSG00000261159', 'ENSG00000276269', 'ENSG00000231173', 'ENSG00000274312', 'ENSG00000289135', 'ENSG00000255279', 'ENSG00000253988', 'ENSG00000290101', 'ENSG00000291279', 'ENSG00000258642', 'ENSG00000226078', 'ENSG00000290884', 'ENSG00000131484', 'ENSG00000278084', 'ENSG00000233342', 'ENSG00000228561', 'ENSG00000285191', 'ENSG00000234296', 'ENSG00000188525', 'ENSG00000287176', 'ENSG00000289635', 'ENSG00000290659', 'ENSG00000271788', 'ENSG00000251532', 'ENSG00000286080', 'ENSG00000289084', 'ENSG00000283907', 'ENSG00000288850', 'ENSG00000259222', 'ENSG00000286164', 'ENSG00000270055', 'ENSG00000286293', 'ENSG00000289194', 'ENSG00000286889', 'ENSG00000250670', 'ENSG00000285782', 'ENSG00000286825', 'ENSG00000266490', 'ENSG00000286501', 'ENSG00000289919', 'ENSG00000290449', 'ENSG00000286755', 'ENSG00000229151', 'ENSG00000251454', 'ENSG00000272650', 'ENSG00000257183', 'ENSG00000257918', 'ENSG00000276760', 'ENSG00000226277', 'ENSG00000272979', 'ENSG00000257603', 'ENSG00000249122', 'ENSG00000248646', 'ENSG00000251416', 'ENSG00000276854', 'ENSG00000263588', 'ENSG00000250234', 'ENSG00000285108', 'ENSG00000254744', 'ENSG00000286420', 'ENSG00000271254', 'ENSG00000272137', 'ENSG00000283480', 'ENSG00000290843', 'ENSG00000229160', 'ENSG00000228008', 'ENSG00000230699', 'ENSG00000228434', 'ENSG00000231054', 'ENSG00000228648', 'ENSG00000263508', 'ENSG00000286941', 'ENSG00000226079', 'ENSG00000253197', 'ENSG00000248753', 'ENSG00000260581', 'ENSG00000288889', 'ENSG00000265174', 'ENSG00000225085', 'ENSG00000255817', 'ENSG00000288808', 'ENSG00000236856', 'ENSG00000272049', 'ENSG00000287824', 'ENSG00000233256', 'ENSG00000248346', 'ENSG00000276250', 'ENSG00000263316', 'ENSG00000253891', 'ENSG00000289082', 'ENSG00000288565', 'ENSG00000269888', 'ENSG00000238102', 'ENSG00000250957', 'ENSG00000267449', 'ENSG00000227683', 'ENSG00000273765', 'ENSG00000234540', 'ENSG00000256686', 'ENSG00000260517', 'ENSG00000236948', 'ENSG00000258302', 'ENSG00000273890', 'ENSG00000289036', 'ENSG00000267388', 'ENSG00000249125', 'ENSG00000229881', 'ENSG00000290580', 'ENSG00000286723', 'ENSG00000290038', 'ENSG00000250777', 'ENSG00000244063', 'ENSG00000232597', 'ENSG00000266002', 'ENSG00000254821', 'ENSG00000272341', 'ENSG00000248311', 'ENSG00000289988', 'ENSG00000237262', 'ENSG00000227917', 'ENSG00000239407', 'ENSG00000291342', 'ENSG00000240996', 'ENSG00000289255', 'ENSG00000290384', 'ENSG00000268333', 'ENSG00000287557', 'ENSG00000291272', 'ENSG00000290510', 'ENSG00000286494', 'ENSG00000267327', 'ENSG00000259760', 'ENSG00000224247', 'ENSG00000248103', 'ENSG00000290759', 'ENSG00000248533', 'ENSG00000255495', 'ENSG00000286511', 'ENSG00000271955', 'ENSG00000287884', 'ENSG00000240355', 'ENSG00000249008', 'ENSG00000288982', 'ENSG00000290459', 'ENSG00000253960', 'ENSG00000253235', 'ENSG00000286778', 'ENSG00000239674', 'ENSG00000289917', 'ENSG00000244545', 'ENSG00000287729', 'ENSG00000263574', 'ENSG00000260951', 'ENSG00000255325', 'ENSG00000282887', 'ENSG00000228421', 'ENSG00000289905', 'ENSG00000287649', 'ENSG00000255599', 'ENSG00000261222', 'ENSG00000263388', 'ENSG00000228156', 'ENSG00000286895', 'ENSG00000275249', 'ENSG00000259126', 'ENSG00000267131', 'ENSG00000269391', 'ENSG00000259995', 'ENSG00000283633', 'ENSG00000270154', 'ENSG00000255028', 'ENSG00000235994', 'ENSG00000273363', 'ENSG00000234902', 'ENSG00000290828', 'ENSG00000261797', 'ENSG00000250101', 'ENSG00000270016', 'ENSG00000289867', 'ENSG00000287263', 'ENSG00000269873', 'ENSG00000283504', 'ENSG00000249988', 'ENSG00000282440', 'ENSG00000271911', 'ENSG00000260100', 'ENSG00000290440', 'ENSG00000286330', 'ENSG00000236451', 'ENSG00000283517', 'ENSG00000228478', 'ENSG00000257277', 'ENSG00000225527', 'ENSG00000267279', 'ENSG00000287798', 'ENSG00000267406', 'ENSG00000289307', 'ENSG00000277901', 'ENSG00000229912', 'ENSG00000288553', 'ENSG00000232411', 'ENSG00000289142', 'ENSG00000260470', 'ENSG00000267243', 'ENSG00000285584', 'ENSG00000248991', 'ENSG00000262558', 'ENSG00000265091', 'ENSG00000253671', 'ENSG00000230090', 'ENSG00000236760', 'ENSG00000227192', 'ENSG00000233852', 'ENSG00000288806', 'ENSG00000251218', 'ENSG00000239216', 'ENSG00000251679', 'ENSG00000287440', 'ENSG00000287974', 'ENSG00000289315', 'ENSG00000288574', 'ENSG00000256694', 'ENSG00000273374', 'ENSG00000265943', 'ENSG00000287386', 'ENSG00000231128', 'ENSG00000263698', 'ENSG00000286542', 'ENSG00000285769', 'ENSG00000288963', 'ENSG00000240240', 'ENSG00000287446', 'ENSG00000258844', 'ENSG00000272369', 'ENSG00000277196', 'ENSG00000253369', 'ENSG00000288526', 'ENSG00000271973', 'ENSG00000223675', 'ENSG00000286915', 'ENSG00000287577', 'ENSG00000251314', 'ENSG00000232692', 'ENSG00000291287', 'ENSG00000233415', 'ENSG00000275232', 'ENSG00000290081', 'ENSG00000269986', 'ENSG00000236501', 'ENSG00000282021', 'ENSG00000249684', 'ENSG00000288789', 'ENSG00000236839', 'ENSG00000288957', 'ENSG00000258265', 'ENSG00000288657', 'ENSG00000287350', 'ENSG00000286817', 'ENSG00000235795', 'ENSG00000289160', 'ENSG00000225790', 'ENSG00000272872', 'ENSG00000261720', 'ENSG00000234936', 'ENSG00000288019', 'ENSG00000256739', 'ENSG00000285656', 'ENSG00000277999', 'ENSG00000258028', 'ENSG00000232034', 'ENSG00000290593', 'ENSG00000261462', 'ENSG00000228538', 'ENSG00000288772', 'ENSG00000258081', 'ENSG00000289563', 'ENSG00000289242', 'ENSG00000289994', 'ENSG00000274902', 'ENSG00000289411', 'ENSG00000291278', 'ENSG00000286622', 'ENSG00000241770', 'ENSG00000232636', 'ENSG00000260182', 'ENSG00000258386', 'ENSG00000267636', 'ENSG00000256732', 'ENSG00000237356', 'ENSG00000284700', 'ENSG00000264078', 'ENSG00000253805', 'ENSG00000270019', 'ENSG00000235578', 'ENSG00000290697', 'ENSG00000259359', 'ENSG00000225058', 'ENSG00000267115', 'ENSG00000258092', 'ENSG00000249166', 'ENSG00000288055', 'ENSG00000257322', 'ENSG00000278923', 'ENSG00000259616', 'ENSG00000255479', 'ENSG00000287248', 'ENSG00000242021', 'ENSG00000272465', 'ENSG00000286504', 'ENSG00000289486', 'ENSG00000287712', 'ENSG00000251144', 'ENSG00000272971', 'ENSG00000285766', 'ENSG00000286519', 'ENSG00000228353', 'ENSG00000288531', 'ENSG00000243179', 'ENSG00000254651', 'ENSG00000259998', 'ENSG00000231616', 'ENSG00000266957', 'ENSG00000291107', 'ENSG00000229191', 'ENSG00000259587', 'ENSG00000227029', 'ENSG00000257226', 'ENSG00000285535', 'ENSG00000235934', 'ENSG00000254836', 'ENSG00000261470', 'ENSG00000254180', 'ENSG00000258473', 'ENSG00000224621', 'ENSG00000224218', 'ENSG00000290032', 'ENSG00000272755', 'ENSG00000253872', 'ENSG00000275557', 'ENSG00000261273', 'ENSG00000258088', 'ENSG00000270147', 'ENSG00000233987', 'ENSG00000276278', 'ENSG00000275869', 'ENSG00000286931', 'ENSG00000260071', 'ENSG00000250046', 'ENSG00000264067', 'ENSG00000260183', 'ENSG00000286827', 'ENSG00000264080', 'ENSG00000257194', 'ENSG00000287788', 'ENSG00000290905', 'ENSG00000289920', 'ENSG00000262039', 'ENSG00000238158', 'ENSG00000228113', 'ENSG00000272218', 'ENSG00000289620', 'ENSG00000249159', 'ENSG00000277475', 'ENSG00000261775', 'ENSG00000263443', 'ENSG00000261757', 'ENSG00000250511', 'ENSG00000258631', 'ENSG00000272009', 'ENSG00000254757', 'ENSG00000290784', 'ENSG00000258342', 'ENSG00000231979', 'ENSG00000272468', 'ENSG00000278633', 'ENSG00000288101', 'ENSG00000250060', 'ENSG00000276075', 'ENSG00000288046', 'ENSG00000272564', 'ENSG00000253484', 'ENSG00000225393', 'ENSG00000286761', 'ENSG00000223808', 'ENSG00000289167', 'ENSG00000287906', 'ENSG00000290091', 'ENSG00000290242', 'ENSG00000204398', 'ENSG00000237422', 'ENSG00000264660', 'ENSG00000286564', 'ENSG00000271259', 'ENSG00000230205', 'ENSG00000289722', 'ENSG00000273119', 'ENSG00000265975', 'ENSG00000289321', 'ENSG00000253381', 'ENSG00000275063', 'ENSG00000289524', 'ENSG00000289467', 'ENSG00000285660', 'ENSG00000249335', 'ENSG00000288991', 'ENSG00000267521', 'ENSG00000288754', 'ENSG00000224525', 'ENSG00000244349', 'ENSG00000276631', 'ENSG00000228503', 'ENSG00000233906', 'ENSG00000224342', 'ENSG00000286133', 'ENSG00000289590', 'ENSG00000288954', 'ENSG00000286499', 'ENSG00000229628', 'ENSG00000231140', 'ENSG00000291246', 'ENSG00000260409', 'ENSG00000283886', 'ENSG00000276790', 'ENSG00000289975', 'ENSG00000277856', 'ENSG00000225643', 'ENSG00000286475', 'ENSG00000280366', 'ENSG00000290680', 'ENSG00000260780', 'ENSG00000268355', 'ENSG00000274652', 'ENSG00000266614', 'ENSG00000244268', 'ENSG00000255272', 'ENSG00000286397', 'ENSG00000235271', 'ENSG00000288972', 'ENSG00000234810', 'ENSG00000235337', 'ENSG00000223379', 'ENSG00000286919', 'ENSG00000290105', 'ENSG00000235215', 'ENSG00000259166', 'ENSG00000259306', 'ENSG00000236263', 'ENSG00000286863', 'ENSG00000251652', 'ENSG00000273748', 'ENSG00000249708', 'ENSG00000273771', 'ENSG00000229964', 'ENSG00000290856', 'ENSG00000253774', 'ENSG00000262227', 'ENSG00000286351', 'ENSG00000253444', 'ENSG00000233214', 'ENSG00000288002', 'ENSG00000238009', 'ENSG00000267399', 'ENSG00000289069', 'ENSG00000271751', 'ENSG00000243944', 'ENSG00000255945', 'ENSG00000278011', 'ENSG00000249837', 'ENSG00000287387', 'ENSG00000286462', 'ENSG00000268674', 'ENSG00000259407', 'ENSG00000271156', 'ENSG00000253762', 'ENSG00000287438', 'ENSG00000228216', 'ENSG00000273289', 'ENSG00000272954', 'ENSG00000274828', 'ENSG00000234134', 'ENSG00000273388', 'ENSG00000253139', 'ENSG00000223811', 'ENSG00000272583', 'ENSG00000248518', 'ENSG00000259594', 'ENSG00000250158', 'ENSG00000260060', 'ENSG00000273486', 'ENSG00000258772', 'ENSG00000285877', 'ENSG00000249604', 'ENSG00000259038', 'ENSG00000291288', 'ENSG00000290958', 'ENSG00000289455', 'ENSG00000260793', 'ENSG00000286292', 'ENSG00000287843', 'ENSG00000239462', 'ENSG00000285424', 'ENSG00000287101', 'ENSG00000260090', 'ENSG00000288819', 'ENSG00000262089', 'ENSG00000256001', 'ENSG00000235736', 'ENSG00000236129', 'ENSG00000241180', 'ENSG00000289413', 'ENSG00000284930', 'ENSG00000233005', 'ENSG00000257654', 'ENSG00000276649', 'ENSG00000278107', 'ENSG00000287944', 'ENSG00000288788', 'ENSG00000272382', 'ENSG00000240401', 'ENSG00000289541', 'ENSG00000228255', 'ENSG00000264072', 'ENSG00000290594', 'ENSG00000272592', 'ENSG00000256101', 'ENSG00000237461', 'ENSG00000259604', 'ENSG00000261445', 'ENSG00000291290', 'ENSG00000288896', 'ENSG00000260052', 'ENSG00000288111', 'ENSG00000290591', 'ENSG00000235209', 'ENSG00000233029', 'ENSG00000286329', 'ENSG00000258407', 'ENSG00000229458', 'ENSG00000287702', 'ENSG00000272515', 'ENSG00000287368', 'ENSG00000290816', 'ENSG00000254143', 'ENSG00000259730', 'ENSG00000276867', 'ENSG00000259376', 'ENSG00000260030', 'ENSG00000237359', 'ENSG00000288583', 'ENSG00000229403', 'ENSG00000250418', 'ENSG00000288554', 'ENSG00000232325', 'ENSG00000267643', 'ENSG00000289619', 'ENSG00000229646', 'ENSG00000290903', 'ENSG00000242396', 'ENSG00000227920', 'ENSG00000264212', 'ENSG00000274175', 'ENSG00000250078', 'ENSG00000258885', 'ENSG00000288986', 'ENSG00000286439', 'ENSG00000256542', 'ENSG00000285155', 'ENSG00000223504', 'ENSG00000289179', 'ENSG00000269072', 'ENSG00000233517', 'ENSG00000279159', 'ENSG00000286016', 'ENSG00000254695', 'ENSG00000239498', 'ENSG00000272482', 'ENSG00000280273', 'ENSG00000261350', 'ENSG00000236716', 'ENSG00000232680', 'ENSG00000260141', 'ENSG00000289122', 'ENSG00000291058', 'ENSG00000290010', 'ENSG00000254927', 'ENSG00000290407', 'ENSG00000254731', 'ENSG00000255644', 'ENSG00000283689', 'ENSG00000273733', 'ENSG00000279987', 'ENSG00000289005', 'ENSG00000232386', 'ENSG00000255015', 'ENSG00000224532', 'ENSG00000290616', 'ENSG00000285759', 'ENSG00000233408', 'ENSG00000291259', 'ENSG00000238358', 'ENSG00000253735', 'ENSG00000285933', 'ENSG00000249463', 'ENSG00000247810', 'ENSG00000286374', 'ENSG00000272519', 'ENSG00000255944', 'ENSG00000272444', 'ENSG00000253400', 'ENSG00000241280', 'ENSG00000287875', 'ENSG00000233970', 'ENSG00000290721', 'ENSG00000289950', 'ENSG00000263081', 'ENSG00000235111', 'ENSG00000258919', 'ENSG00000290463', 'ENSG00000283294', 'ENSG00000286213', 'ENSG00000268509', 'ENSG00000259972', 'ENSG00000289600', 'ENSG00000254619', 'ENSG00000273554', 'ENSG00000272070', 'ENSG00000232749', 'ENSG00000254138', 'ENSG00000291026', 'ENSG00000286262', 'ENSG00000258600', 'ENSG00000277501', 'ENSG00000291299', 'ENSG00000265964', 'ENSG00000253256', 'ENSG00000272316', 'ENSG00000290666', 'ENSG00000232754', 'ENSG00000289830', 'ENSG00000287523', 'ENSG00000291215', 'ENSG00000287858', 'ENSG00000286189', 'ENSG00000249731', 'ENSG00000251613', 'ENSG00000254587', 'ENSG00000259762', 'ENSG00000290818', 'ENSG00000258131', 'ENSG00000260095', 'ENSG00000291273', 'ENSG00000286103', 'ENSG00000266601', 'ENSG00000234698', 'ENSG00000286854', 'ENSG00000272940', 'ENSG00000289862', 'ENSG00000241475', 'ENSG00000253115', 'ENSG00000235047', 'ENSG00000267787', 'ENSG00000278878', 'ENSG00000289329', 'ENSG00000272736', 'ENSG00000283164', 'ENSG00000236938', 'ENSG00000226849', 'ENSG00000270265', 'ENSG00000278514', 'ENSG00000237954', 'ENSG00000287211', 'ENSG00000256923', 'ENSG00000242753', 'ENSG00000273181', 'ENSG00000273481', 'ENSG00000264334', 'ENSG00000254420', 'ENSG00000286824', 'ENSG00000242375', 'ENSG00000235852', 'ENSG00000259123', 'ENSG00000234141', 'ENSG00000265717', 'ENSG00000286792', 'ENSG00000268810', 'ENSG00000251379', 'ENSG00000289184', 'ENSG00000289614', 'ENSG00000234998', 'ENSG00000281920', 'ENSG00000290969', 'ENSG00000272732', 'ENSG00000225836', 'ENSG00000261120', 'ENSG00000253983', 'ENSG00000233799', 'ENSG00000285922', 'ENSG00000241792', 'ENSG00000214955', 'ENSG00000288810', 'ENSG00000261502', 'ENSG00000235450', 'ENSG00000286978', 'ENSG00000250567', 'ENSG00000250310', 'ENSG00000254115', 'ENSG00000248787', 'ENSG00000273211', 'ENSG00000288573', 'ENSG00000250415', 'ENSG00000266248', 'ENSG00000266893', 'ENSG00000278266', 'ENSG00000289048', 'ENSG00000274792', 'ENSG00000224228', 'ENSG00000227305', 'ENSG00000286362', 'ENSG00000254028', 'ENSG00000286687', 'ENSG00000267683', 'ENSG00000290396', 'ENSG00000254095', 'ENSG00000257715', 'ENSG00000253985', 'ENSG00000289636', 'ENSG00000272240', 'ENSG00000225655', 'ENSG00000263688', 'ENSG00000288904', 'ENSG00000291284', 'ENSG00000273138', 'ENSG00000234677', 'ENSG00000272030', 'ENSG00000272840', 'ENSG00000287837', 'ENSG00000286534', 'ENSG00000290412', 'ENSG00000274444', 'ENSG00000261650', 'ENSG00000290124', 'ENSG00000291025', 'ENSG00000291247', 'ENSG00000278817', 'ENSG00000289147', 'ENSG00000282863', 'ENSG00000290529', 'ENSG00000288580', 'ENSG00000249091', 'ENSG00000272085', 'ENSG00000259423', 'ENSG00000223795', 'ENSG00000250481', 'ENSG00000285719', 'ENSG00000291291', 'ENSG00000267316', 'ENSG00000253766', 'ENSG00000278384', 'ENSG00000289206', 'ENSG00000286466', 'ENSG00000249483', 'ENSG00000287492', 'ENSG00000289008', 'ENSG00000261305', 'ENSG00000249184', 'ENSG00000226376', 'ENSG00000231877', 'ENSG00000249269', 'ENSG00000289826', 'ENSG00000274367', 'ENSG00000234306', 'ENSG00000276256', 'ENSG00000286111', 'ENSG00000250431', 'ENSG00000290066', 'ENSG00000291011', 'ENSG00000258196', 'ENSG00000254039', 'ENSG00000289593', 'ENSG00000241520', 'ENSG00000278704', 'ENSG00000286572', 'ENSG00000284685', 'ENSG00000276345', 'ENSG00000254814', 'ENSG00000287346', 'ENSG00000276471', 'ENSG00000270114', 'ENSG00000284602', 'ENSG00000287316', 'ENSG00000290414', 'ENSG00000291055', 'ENSG00000266908', 'ENSG00000273123', 'ENSG00000231255', 'ENSG00000289209', 'ENSG00000263677', 'ENSG00000273203', 'ENSG00000288581', 'ENSG00000289512', 'ENSG00000253859', 'ENSG00000261044', 'ENSG00000235242', 'ENSG00000236453', 'ENSG00000289465', 'ENSG00000273319', 'ENSG00000270038', 'ENSG00000244620', 'ENSG00000244650', 'ENSG00000214970', 'ENSG00000230506', 'ENSG00000291188', 'ENSG00000250948', 'ENSG00000283003', 'ENSG00000287865', 'ENSG00000261193', 'ENSG00000290004', 'ENSG00000258831', 'ENSG00000234083', 'ENSG00000259255', 'ENSG00000246331', 'ENSG00000273403', 'ENSG00000269927', 'ENSG00000290514', 'ENSG00000258170', 'ENSG00000258216', 'ENSG00000290029', 'ENSG00000253842', 'ENSG00000287708', 'ENSG00000271904', 'ENSG00000280767', 'ENSG00000222022', 'ENSG00000258949', 'ENSG00000249492', 'ENSG00000250237', 'ENSG00000290443', 'ENSG00000278626', 'ENSG00000251654', 'ENSG00000286815', 'ENSG00000233193', 'ENSG00000277400', 'ENSG00000283648', 'ENSG00000287868', 'ENSG00000241317', 'ENSG00000276251', 'ENSG00000267070', 'ENSG00000267081', 'ENSG00000287168', 'ENSG00000276348', 'ENSG00000233536', 'ENSG00000286599', 'ENSG00000231883', 'ENSG00000287564', 'ENSG00000253737', 'ENSG00000233376', 'ENSG00000253844', 'ENSG00000253584', 'ENSG00000223469', 'ENSG00000260487', 'ENSG00000272304', 'ENSG00000230381', 'ENSG00000287313', 'ENSG00000266450', 'ENSG00000279949', 'ENSG00000257711', 'ENSG00000240440', 'ENSG00000289840', 'ENSG00000277010', 'ENSG00000250284', 'ENSG00000228334']; attempting to fetch latest IDs...\n",
            "Fetched latest Ensembl IDs: {'ENSG00000272922': nan, 'ENSG00000224481': 'ENSG00000238107', 'ENSG00000259582': nan, 'ENSG00000287139': 'ENSG00000300787', 'ENSG00000251399': 'ENSG00000250334', 'ENSG00000277559': 'ENSG00000287885', 'ENSG00000275636': 'ENSG00000248508', 'ENSG00000286571': 'ENSG00000287241', 'ENSG00000263331': 'ENSG00000289640', 'ENSG00000290872': 'ENSG00000290758', 'ENSG00000289531': 'ENSG00000214900', 'ENSG00000203520': 'ENSG00000256789', 'ENSG00000257962': 'ENSG00000258256', 'ENSG00000241912': 'ENSG00000241220', 'ENSG00000261386': 'ENSG00000259804', 'ENSG00000287811': 'ENSG00000198478', 'ENSG00000290709': 'ENSG00000196696', 'ENSG00000258921': nan, 'ENSG00000230649': 'ENSG00000232053', 'ENSG00000286472': 'ENSG00000230725', 'ENSG00000259274': 'ENSG00000245534', 'ENSG00000225489': 'ENSG00000236924', 'ENSG00000236412': 'ENSG00000224187', 'ENSG00000259209': 'ENSG00000259768', 'ENSG00000286977': 'ENSG00000226310', 'ENSG00000229454': 'ENSG00000230303', 'ENSG00000290078': 'ENSG00000254416', 'ENSG00000286633': 'ENSG00000288887', 'ENSG00000264421': 'ENSG00000266473', 'ENSG00000244260': 'ENSG00000240687', 'ENSG00000249638': nan, 'ENSG00000267105': 'ENSG00000266978', 'ENSG00000255328': 'ENSG00000251661', 'ENSG00000266588': 'ENSG00000266013', 'ENSG00000286603': 'ENSG00000287281', 'ENSG00000234405': 'ENSG00000281091', 'ENSG00000258345': nan, 'ENSG00000260163': 'ENSG00000286971', 'ENSG00000250360': 'ENSG00000251573', 'ENSG00000257454': 'ENSG00000258053', 'ENSG00000232056': 'ENSG00000271952', 'ENSG00000289448': 'ENSG00000223768', 'ENSG00000267420': 'ENSG00000261514', 'ENSG00000264790': 'ENSG00000265374', 'ENSG00000267537': 'ENSG00000290606', 'ENSG00000260511': 'ENSG00000260479', 'ENSG00000287622': 'ENSG00000286922', 'ENSG00000277423': nan, 'ENSG00000288816': 'ENSG00000227634', 'ENSG00000287614': 'ENSG00000227803', 'ENSG00000235815': 'ENSG00000289090', 'ENSG00000289176': 'ENSG00000238201', 'ENSG00000230600': 'ENSG00000232817', 'ENSG00000287400': 'ENSG00000235358', 'ENSG00000254290': 'ENSG00000253161', 'ENSG00000234223': 'ENSG00000232667', 'ENSG00000291048': 'ENSG00000253893', 'ENSG00000224819': 'ENSG00000288902', 'ENSG00000258927': 'ENSG00000187621', 'ENSG00000290100': 'ENSG00000236213', 'ENSG00000251506': 'ENSG00000251426', 'ENSG00000258918': 'ENSG00000259130', 'ENSG00000286997': 'ENSG00000267207', 'ENSG00000286087': 'ENSG00000234265', 'ENSG00000248240': 'ENSG00000248554', 'ENSG00000287411': 'ENSG00000230423', 'ENSG00000287361': 'ENSG00000232985', 'ENSG00000267058': 'ENSG00000176222', 'ENSG00000287569': 'ENSG00000125804', 'ENSG00000231212': 'ENSG00000204802', 'ENSG00000232949': 'ENSG00000232759', 'ENSG00000272948': 'ENSG00000242553', 'ENSG00000224884': 'ENSG00000227110', 'ENSG00000241772': 'ENSG00000144228', 'ENSG00000287127': 'ENSG00000246145', 'ENSG00000254364': 'ENSG00000253666', 'ENSG00000256321': 'ENSG00000256995', 'ENSG00000288749': 'ENSG00000232301', 'ENSG00000255160': 'ENSG00000255308', 'ENSG00000232294': 'ENSG00000234948', 'ENSG00000290183': 'ENSG00000006194', 'ENSG00000285622': nan, 'ENSG00000254306': 'ENSG00000253161', 'ENSG00000260759': 'ENSG00000291183', 'ENSG00000261218': 'ENSG00000197943', 'ENSG00000291260': 'ENSG00000291027', 'ENSG00000268615': 'ENSG00000233901', 'ENSG00000272953': 'ENSG00000273084', 'ENSG00000249984': 'ENSG00000249776', 'ENSG00000237899': 'ENSG00000272145', 'ENSG00000272541': nan, 'ENSG00000286911': 'ENSG00000260423', 'ENSG00000273729': 'ENSG00000246548', 'ENSG00000271792': 'ENSG00000249526', 'ENSG00000265671': 'ENSG00000265179', 'ENSG00000225916': 'ENSG00000229647', 'ENSG00000248359': 'ENSG00000248884', 'ENSG00000287941': 'ENSG00000241098', 'ENSG00000291298': nan, 'ENSG00000243176': 'ENSG00000241135', 'ENSG00000290974': nan, 'ENSG00000251678': 'ENSG00000223597', 'ENSG00000287682': 'ENSG00000214407', 'ENSG00000287954': 'ENSG00000230010', 'ENSG00000257879': 'ENSG00000257165', 'ENSG00000270157': 'ENSG00000244701', 'ENSG00000261135': 'ENSG00000224174', 'ENSG00000288950': 'ENSG00000286285', 'ENSG00000283255': 'ENSG00000228697', 'ENSG00000286906': 'ENSG00000226370', 'ENSG00000289958': 'ENSG00000234261', 'ENSG00000260983': 'ENSG00000286841', 'ENSG00000261159': nan, 'ENSG00000276269': nan, 'ENSG00000231173': 'ENSG00000223751', 'ENSG00000274312': 'ENSG00000277987', 'ENSG00000289135': 'ENSG00000273265', 'ENSG00000255279': 'ENSG00000255171', 'ENSG00000253988': 'ENSG00000254361', 'ENSG00000290101': 'ENSG00000233942', 'ENSG00000291279': 'ENSG00000284084', 'ENSG00000258642': 'ENSG00000259130', 'ENSG00000226078': 'ENSG00000233721', 'ENSG00000290884': 'ENSG00000289621', 'ENSG00000131484': 'ENSG00000267198', 'ENSG00000278084': 'ENSG00000256811', 'ENSG00000233342': 'ENSG00000231863', 'ENSG00000228561': 'ENSG00000228221', 'ENSG00000285191': 'ENSG00000289309', 'ENSG00000234296': 'ENSG00000290423', 'ENSG00000188525': 'ENSG00000289033', 'ENSG00000287176': 'ENSG00000287940', 'ENSG00000289635': 'ENSG00000255240', 'ENSG00000290659': 'ENSG00000273018', 'ENSG00000271788': 'ENSG00000177738', 'ENSG00000251532': 'ENSG00000291079', 'ENSG00000286080': 'ENSG00000266904', 'ENSG00000289084': 'ENSG00000233067', 'ENSG00000283907': 'ENSG00000224910', 'ENSG00000288850': 'ENSG00000260608', 'ENSG00000259222': 'ENSG00000212766', 'ENSG00000286164': 'ENSG00000248668', 'ENSG00000270055': nan, 'ENSG00000286293': 'ENSG00000293042', 'ENSG00000289194': nan, 'ENSG00000286889': 'ENSG00000287048', 'ENSG00000250670': 'ENSG00000245384', 'ENSG00000285782': 'ENSG00000233008', 'ENSG00000286825': 'ENSG00000288044', 'ENSG00000266490': 'ENSG00000290928', 'ENSG00000286501': 'ENSG00000243550', 'ENSG00000289919': nan, 'ENSG00000290449': 'ENSG00000290920', 'ENSG00000286755': 'ENSG00000232832', 'ENSG00000229151': 'ENSG00000230317', 'ENSG00000251454': 'ENSG00000251185', 'ENSG00000272650': nan, 'ENSG00000257183': nan, 'ENSG00000257918': 'ENSG00000257545', 'ENSG00000226277': nan, 'ENSG00000272979': 'ENSG00000290089', 'ENSG00000257603': 'ENSG00000257997', 'ENSG00000249122': 'ENSG00000250467', 'ENSG00000248646': 'ENSG00000251185', 'ENSG00000251416': 'ENSG00000251504', 'ENSG00000276854': 'ENSG00000227258', 'ENSG00000263588': 'ENSG00000265758', 'ENSG00000250234': 'ENSG00000271874', 'ENSG00000285108': 'ENSG00000229140', 'ENSG00000254744': 'ENSG00000259363', 'ENSG00000286420': 'ENSG00000250582', 'ENSG00000272137': 'ENSG00000198478', 'ENSG00000283480': 'ENSG00000280989', 'ENSG00000290843': 'ENSG00000262412', 'ENSG00000229160': 'ENSG00000227292', 'ENSG00000228008': 'ENSG00000230698', 'ENSG00000230699': 'ENSG00000272438', 'ENSG00000228434': 'ENSG00000290758', 'ENSG00000231054': nan, 'ENSG00000228648': 'ENSG00000280707', 'ENSG00000263508': 'ENSG00000263400', 'ENSG00000286941': 'ENSG00000290990', 'ENSG00000226079': 'ENSG00000232395', 'ENSG00000253197': 'ENSG00000253577', 'ENSG00000248753': 'ENSG00000224186', 'ENSG00000260581': 'ENSG00000288081', 'ENSG00000288889': 'ENSG00000306503', 'ENSG00000265174': 'ENSG00000265944', 'ENSG00000225085': 'ENSG00000290551', 'ENSG00000255817': 'ENSG00000287618', 'ENSG00000288808': 'ENSG00000259460', 'ENSG00000236856': 'ENSG00000233684', 'ENSG00000272049': 'ENSG00000245729', 'ENSG00000287824': 'ENSG00000231297', 'ENSG00000233256': 'ENSG00000286514', 'ENSG00000248346': 'ENSG00000248764', 'ENSG00000276250': 'ENSG00000290928', 'ENSG00000263316': 'ENSG00000267532', 'ENSG00000253891': 'ENSG00000253535', 'ENSG00000289082': 'ENSG00000288886', 'ENSG00000288565': 'ENSG00000233639', 'ENSG00000269888': nan, 'ENSG00000238102': nan, 'ENSG00000250957': 'ENSG00000248174', 'ENSG00000267449': nan, 'ENSG00000227683': 'ENSG00000165511', 'ENSG00000273765': nan, 'ENSG00000234540': 'ENSG00000286277', 'ENSG00000256686': 'ENSG00000256440', 'ENSG00000260517': 'ENSG00000103472', 'ENSG00000236948': 'ENSG00000289893', 'ENSG00000258302': 'ENSG00000270344', 'ENSG00000273890': 'ENSG00000257449', 'ENSG00000289036': 'ENSG00000288782', 'ENSG00000267388': 'ENSG00000267786', 'ENSG00000249125': 'ENSG00000251526', 'ENSG00000229881': nan, 'ENSG00000290580': 'ENSG00000249515', 'ENSG00000286723': 'ENSG00000290555', 'ENSG00000290038': nan, 'ENSG00000250777': 'ENSG00000250126', 'ENSG00000244063': 'ENSG00000228857', 'ENSG00000232597': 'ENSG00000236141', 'ENSG00000266002': 'ENSG00000265313', 'ENSG00000254821': 'ENSG00000234817', 'ENSG00000272341': 'ENSG00000229931', 'ENSG00000248311': 'ENSG00000248973', 'ENSG00000289988': 'ENSG00000296783', 'ENSG00000237262': 'ENSG00000231890', 'ENSG00000227917': 'ENSG00000231808', 'ENSG00000239407': 'ENSG00000223546', 'ENSG00000291342': 'ENSG00000290474', 'ENSG00000240996': 'ENSG00000235100', 'ENSG00000289255': 'ENSG00000267939', 'ENSG00000290384': 'ENSG00000282508', 'ENSG00000268333': 'ENSG00000235590', 'ENSG00000287557': 'ENSG00000286680', 'ENSG00000291272': 'ENSG00000290394', 'ENSG00000290510': 'ENSG00000223597', 'ENSG00000286494': 'ENSG00000248551', 'ENSG00000267327': 'ENSG00000267057', 'ENSG00000259760': 'ENSG00000259590', 'ENSG00000224247': nan, 'ENSG00000248103': 'ENSG00000248275', 'ENSG00000290759': 'ENSG00000283098', 'ENSG00000248533': 'ENSG00000248587', 'ENSG00000255495': nan, 'ENSG00000286511': 'ENSG00000203801', 'ENSG00000271955': 'ENSG00000233891', 'ENSG00000287884': 'ENSG00000205790', 'ENSG00000240355': 'ENSG00000230680', 'ENSG00000249008': 'ENSG00000287544', 'ENSG00000288982': nan, 'ENSG00000290459': 'ENSG00000196696', 'ENSG00000253960': 'ENSG00000246662', 'ENSG00000253235': 'ENSG00000250295', 'ENSG00000286778': 'ENSG00000226939', 'ENSG00000239674': 'ENSG00000290475', 'ENSG00000289917': 'ENSG00000259717', 'ENSG00000244545': nan, 'ENSG00000287729': 'ENSG00000238123', 'ENSG00000263574': 'ENSG00000234721', 'ENSG00000260951': 'ENSG00000233824', 'ENSG00000255325': 'ENSG00000253819', 'ENSG00000282887': nan, 'ENSG00000228421': 'ENSG00000272568', 'ENSG00000289905': nan, 'ENSG00000287649': 'ENSG00000243295', 'ENSG00000255599': 'ENSG00000255689', 'ENSG00000261222': 'ENSG00000182352', 'ENSG00000263388': 'ENSG00000272975', 'ENSG00000228156': 'ENSG00000204393', 'ENSG00000286895': 'ENSG00000215241', 'ENSG00000259126': 'ENSG00000258487', 'ENSG00000267131': 'ENSG00000267280', 'ENSG00000269391': 'ENSG00000233570', 'ENSG00000259995': 'ENSG00000250514', 'ENSG00000283633': 'ENSG00000290418', 'ENSG00000270154': 'ENSG00000227888', 'ENSG00000255028': 'ENSG00000254659', 'ENSG00000235994': 'ENSG00000229921', 'ENSG00000273363': 'ENSG00000224812', 'ENSG00000234902': 'ENSG00000223725', 'ENSG00000290828': 'ENSG00000290829', 'ENSG00000261797': 'ENSG00000261570', 'ENSG00000250101': 'ENSG00000291113', 'ENSG00000270016': nan, 'ENSG00000287263': 'ENSG00000177738', 'ENSG00000269873': nan, 'ENSG00000283504': 'ENSG00000287636', 'ENSG00000249988': 'ENSG00000250634', 'ENSG00000282440': 'ENSG00000226383', 'ENSG00000271911': 'ENSG00000271727', 'ENSG00000260100': 'ENSG00000232920', 'ENSG00000290440': 'ENSG00000228315', 'ENSG00000286330': 'ENSG00000235532', 'ENSG00000236451': 'ENSG00000288902', 'ENSG00000283517': 'ENSG00000228639', 'ENSG00000228478': 'ENSG00000183674', 'ENSG00000257277': 'ENSG00000231758', 'ENSG00000225527': 'ENSG00000240291', 'ENSG00000267279': 'ENSG00000267175', 'ENSG00000287798': 'ENSG00000290829', 'ENSG00000267406': 'ENSG00000267723', 'ENSG00000289307': 'ENSG00000245694', 'ENSG00000277901': 'ENSG00000234832', 'ENSG00000229912': 'ENSG00000235126', 'ENSG00000288553': 'ENSG00000233862', 'ENSG00000232411': 'ENSG00000229195', 'ENSG00000289142': 'ENSG00000257551', 'ENSG00000260470': 'ENSG00000250432', 'ENSG00000267243': 'ENSG00000290606', 'ENSG00000285584': 'ENSG00000259110', 'ENSG00000248991': 'ENSG00000251249', 'ENSG00000262558': 'ENSG00000262294', 'ENSG00000265091': 'ENSG00000264575', 'ENSG00000253671': 'ENSG00000253944', 'ENSG00000230090': 'ENSG00000224128', 'ENSG00000236760': 'ENSG00000234423', 'ENSG00000227192': 'ENSG00000229017', 'ENSG00000233852': 'ENSG00000236088', 'ENSG00000288806': 'ENSG00000239523', 'ENSG00000251218': 'ENSG00000253433', 'ENSG00000239216': 'ENSG00000226172', 'ENSG00000251679': 'ENSG00000250634', 'ENSG00000287440': 'ENSG00000250934', 'ENSG00000287974': 'ENSG00000236700', 'ENSG00000289315': 'ENSG00000239941', 'ENSG00000288574': 'ENSG00000287452', 'ENSG00000256694': 'ENSG00000256540', 'ENSG00000273374': 'ENSG00000296790', 'ENSG00000265943': 'ENSG00000266850', 'ENSG00000287386': 'ENSG00000257319', 'ENSG00000231128': 'ENSG00000226167', 'ENSG00000263698': nan, 'ENSG00000286542': 'ENSG00000245281', 'ENSG00000285769': 'ENSG00000254968', 'ENSG00000288963': 'ENSG00000232104', 'ENSG00000240240': 'ENSG00000290718', 'ENSG00000287446': 'ENSG00000290950', 'ENSG00000258844': 'ENSG00000257585', 'ENSG00000272369': 'ENSG00000257261', 'ENSG00000253369': 'ENSG00000237807', 'ENSG00000288526': 'ENSG00000234026', 'ENSG00000271973': 'ENSG00000228350', 'ENSG00000223675': 'ENSG00000293271', 'ENSG00000286915': 'ENSG00000261758', 'ENSG00000287577': 'ENSG00000260423', 'ENSG00000251314': 'ENSG00000153113', 'ENSG00000232692': 'ENSG00000197934', 'ENSG00000291287': 'ENSG00000290718', 'ENSG00000233415': 'ENSG00000237767', 'ENSG00000275232': 'ENSG00000256209', 'ENSG00000290081': 'ENSG00000225506', 'ENSG00000269986': 'ENSG00000260658', 'ENSG00000236501': 'ENSG00000229337', 'ENSG00000282021': 'ENSG00000253982', 'ENSG00000249684': 'ENSG00000291113', 'ENSG00000288789': 'ENSG00000255666', 'ENSG00000236839': 'ENSG00000225718', 'ENSG00000288957': 'ENSG00000293676', 'ENSG00000258265': 'ENSG00000257891', 'ENSG00000288657': 'ENSG00000226762', 'ENSG00000287350': 'ENSG00000258702', 'ENSG00000286817': 'ENSG00000293222', 'ENSG00000235795': nan, 'ENSG00000289160': 'ENSG00000231711', 'ENSG00000225790': 'ENSG00000231574', 'ENSG00000272872': 'ENSG00000206195', 'ENSG00000261720': 'ENSG00000292430', 'ENSG00000234936': 'ENSG00000279873', 'ENSG00000288019': 'ENSG00000287037', 'ENSG00000256739': 'ENSG00000255672', 'ENSG00000285656': 'ENSG00000255553', 'ENSG00000277999': 'ENSG00000103472', 'ENSG00000258028': 'ENSG00000258038', 'ENSG00000232034': 'ENSG00000289077', 'ENSG00000290593': 'ENSG00000267575', 'ENSG00000261462': 'ENSG00000307585', 'ENSG00000228538': 'ENSG00000229621', 'ENSG00000288772': 'ENSG00000229431', 'ENSG00000258081': nan, 'ENSG00000289563': 'ENSG00000260420', 'ENSG00000289242': 'ENSG00000253214', 'ENSG00000289994': 'ENSG00000286610', 'ENSG00000274902': 'ENSG00000257433', 'ENSG00000289411': 'ENSG00000277332', 'ENSG00000291278': 'ENSG00000286058', 'ENSG00000286622': 'ENSG00000260093', 'ENSG00000241770': 'ENSG00000241135', 'ENSG00000232636': 'ENSG00000228824', 'ENSG00000260182': nan, 'ENSG00000258386': 'ENSG00000258474', 'ENSG00000267636': 'ENSG00000267662', 'ENSG00000256732': 'ENSG00000256128', 'ENSG00000237356': 'ENSG00000258731', 'ENSG00000284700': 'ENSG00000225767', 'ENSG00000264078': 'ENSG00000235790', 'ENSG00000253805': 'ENSG00000289473', 'ENSG00000270019': 'ENSG00000228857', 'ENSG00000235578': 'ENSG00000290950', 'ENSG00000259359': 'ENSG00000275443', 'ENSG00000225058': 'ENSG00000224187', 'ENSG00000267115': 'ENSG00000266916', 'ENSG00000258092': 'ENSG00000258325', 'ENSG00000249166': nan, 'ENSG00000288055': 'ENSG00000253832', 'ENSG00000257322': 'ENSG00000257252', 'ENSG00000278923': 'ENSG00000293042', 'ENSG00000259616': 'ENSG00000259675', 'ENSG00000255479': 'ENSG00000255363', 'ENSG00000287248': 'ENSG00000267057', 'ENSG00000242021': 'ENSG00000228933', 'ENSG00000272465': 'ENSG00000250903', 'ENSG00000286504': 'ENSG00000226091', 'ENSG00000289486': 'ENSG00000288852', 'ENSG00000287712': 'ENSG00000236908', 'ENSG00000251144': 'ENSG00000251670', 'ENSG00000272971': 'ENSG00000229953', 'ENSG00000285766': 'ENSG00000227245', 'ENSG00000286519': 'ENSG00000286728', 'ENSG00000228353': 'ENSG00000229672', 'ENSG00000288531': 'ENSG00000228794', 'ENSG00000243179': 'ENSG00000234199', 'ENSG00000254651': 'ENSG00000255267', 'ENSG00000259998': 'ENSG00000258975', 'ENSG00000231616': 'ENSG00000226877', 'ENSG00000266957': 'ENSG00000167220', 'ENSG00000291107': 'ENSG00000284634', 'ENSG00000229191': 'ENSG00000234132', 'ENSG00000259587': nan, 'ENSG00000227029': 'ENSG00000233395', 'ENSG00000257226': 'ENSG00000231758', 'ENSG00000285535': 'ENSG00000267506', 'ENSG00000235934': 'ENSG00000234350', 'ENSG00000254836': 'ENSG00000254584', 'ENSG00000261470': 'ENSG00000261749', 'ENSG00000254180': 'ENSG00000253799', 'ENSG00000258473': 'ENSG00000258819', 'ENSG00000224621': 'ENSG00000227959', 'ENSG00000224218': 'ENSG00000233860', 'ENSG00000290032': nan, 'ENSG00000272755': 'ENSG00000274265', 'ENSG00000253872': 'ENSG00000253773', 'ENSG00000275557': 'ENSG00000291158', 'ENSG00000261273': 'ENSG00000260420', 'ENSG00000258088': 'ENSG00000258077', 'ENSG00000270147': 'ENSG00000269921', 'ENSG00000233987': 'ENSG00000237222', 'ENSG00000276278': nan, 'ENSG00000286931': 'ENSG00000291039', 'ENSG00000260071': 'ENSG00000261617', 'ENSG00000250046': 'ENSG00000249532', 'ENSG00000264067': 'ENSG00000272975', 'ENSG00000260183': 'ENSG00000259867', 'ENSG00000286827': 'ENSG00000243593', 'ENSG00000264080': 'ENSG00000266602', 'ENSG00000257194': nan, 'ENSG00000287788': 'ENSG00000288761', 'ENSG00000290905': 'ENSG00000284084', 'ENSG00000289920': 'ENSG00000225778', 'ENSG00000262039': 'ENSG00000248714', 'ENSG00000238158': 'ENSG00000233068', 'ENSG00000228113': 'ENSG00000254003', 'ENSG00000272218': 'ENSG00000250829', 'ENSG00000289620': 'ENSG00000223511', 'ENSG00000249159': 'ENSG00000245729', 'ENSG00000261775': 'ENSG00000247240', 'ENSG00000263443': 'ENSG00000253819', 'ENSG00000261757': nan, 'ENSG00000250511': 'ENSG00000250103', 'ENSG00000258631': 'ENSG00000257060', 'ENSG00000272009': 'ENSG00000269293', 'ENSG00000254757': 'ENSG00000287312', 'ENSG00000290784': 'ENSG00000225880', 'ENSG00000258342': 'ENSG00000257585', 'ENSG00000231979': 'ENSG00000234601', 'ENSG00000272468': nan, 'ENSG00000288101': 'ENSG00000220908', 'ENSG00000250060': 'ENSG00000250974', 'ENSG00000276075': 'ENSG00000259804', 'ENSG00000288046': 'ENSG00000287497', 'ENSG00000272564': 'ENSG00000232504', 'ENSG00000253484': 'ENSG00000289095', 'ENSG00000225393': 'ENSG00000224533', 'ENSG00000286761': 'ENSG00000249818', 'ENSG00000223808': 'ENSG00000234752', 'ENSG00000289167': 'ENSG00000281162', 'ENSG00000287906': 'ENSG00000255367', 'ENSG00000290091': 'ENSG00000273084', 'ENSG00000290242': nan, 'ENSG00000204398': 'ENSG00000215881', 'ENSG00000237422': 'ENSG00000230537', 'ENSG00000264660': 'ENSG00000233098', 'ENSG00000286564': 'ENSG00000285838', 'ENSG00000271259': nan, 'ENSG00000230205': 'ENSG00000287473', 'ENSG00000289722': 'ENSG00000162039', 'ENSG00000273119': 'ENSG00000267051', 'ENSG00000265975': 'ENSG00000266389', 'ENSG00000289321': 'ENSG00000291242', 'ENSG00000253381': 'ENSG00000253665', 'ENSG00000289524': 'ENSG00000243550', 'ENSG00000289467': nan, 'ENSG00000285660': 'ENSG00000241131', 'ENSG00000249335': 'ENSG00000249352', 'ENSG00000288991': nan, 'ENSG00000267521': 'ENSG00000267568', 'ENSG00000288754': 'ENSG00000234869', 'ENSG00000224525': 'ENSG00000272865', 'ENSG00000244349': nan, 'ENSG00000276631': nan, 'ENSG00000228503': 'ENSG00000234967', 'ENSG00000233906': 'ENSG00000283982', 'ENSG00000224342': 'ENSG00000229647', 'ENSG00000286133': 'ENSG00000287226', 'ENSG00000289590': 'ENSG00000258197', 'ENSG00000288954': 'ENSG00000226091', 'ENSG00000286499': 'ENSG00000287563', 'ENSG00000229628': 'ENSG00000237471', 'ENSG00000231140': 'ENSG00000224038', 'ENSG00000291246': 'ENSG00000290466', 'ENSG00000260409': 'ENSG00000310519', 'ENSG00000283886': nan, 'ENSG00000276790': 'ENSG00000263293', 'ENSG00000289975': 'ENSG00000272217', 'ENSG00000225643': 'ENSG00000157978', 'ENSG00000286475': 'ENSG00000290955', 'ENSG00000280366': 'ENSG00000230945', 'ENSG00000290680': 'ENSG00000103472', 'ENSG00000260780': 'ENSG00000259905', 'ENSG00000268355': 'ENSG00000270164', 'ENSG00000274652': nan, 'ENSG00000266614': 'ENSG00000266014', 'ENSG00000244268': 'ENSG00000288787', 'ENSG00000255272': 'ENSG00000254508', 'ENSG00000286397': 'ENSG00000228564', 'ENSG00000235271': 'ENSG00000244625', 'ENSG00000288972': 'ENSG00000235257', 'ENSG00000234810': 'ENSG00000233079', 'ENSG00000235337': 'ENSG00000288902', 'ENSG00000223379': 'ENSG00000231527', 'ENSG00000286919': 'ENSG00000289641', 'ENSG00000290105': 'ENSG00000272668', 'ENSG00000235215': 'ENSG00000241042', 'ENSG00000259166': 'ENSG00000259082', 'ENSG00000259306': 'ENSG00000259240', 'ENSG00000236263': 'ENSG00000291137', 'ENSG00000286863': 'ENSG00000227016', 'ENSG00000251652': 'ENSG00000227189', 'ENSG00000249708': nan, 'ENSG00000273771': 'ENSG00000247809', 'ENSG00000229964': 'ENSG00000276407', 'ENSG00000290856': 'ENSG00000273784', 'ENSG00000253774': 'ENSG00000253130', 'ENSG00000262227': 'ENSG00000262678', 'ENSG00000286351': nan, 'ENSG00000253444': 'ENSG00000282142', 'ENSG00000233214': 'ENSG00000288731', 'ENSG00000288002': 'ENSG00000245729', 'ENSG00000238009': 'ENSG00000241860', 'ENSG00000267399': 'ENSG00000267712', 'ENSG00000289069': 'ENSG00000242242', 'ENSG00000271751': 'ENSG00000272186', 'ENSG00000243944': 'ENSG00000244541', 'ENSG00000255945': 'ENSG00000286922', 'ENSG00000278011': nan, 'ENSG00000249837': 'ENSG00000286734', 'ENSG00000287387': 'ENSG00000279873', 'ENSG00000286462': 'ENSG00000248703', 'ENSG00000259407': 'ENSG00000259416', 'ENSG00000271156': 'ENSG00000286984', 'ENSG00000253762': 'ENSG00000287844', 'ENSG00000287438': 'ENSG00000226370', 'ENSG00000228216': 'ENSG00000230537', 'ENSG00000273289': 'ENSG00000197182', 'ENSG00000272954': 'ENSG00000207751', 'ENSG00000274828': 'ENSG00000178412', 'ENSG00000234134': 'ENSG00000224023', 'ENSG00000273388': 'ENSG00000272975', 'ENSG00000253139': 'ENSG00000246430', 'ENSG00000223811': 'ENSG00000289961', 'ENSG00000272583': 'ENSG00000228826', 'ENSG00000248518': 'ENSG00000248184', 'ENSG00000259594': 'ENSG00000259343', 'ENSG00000250158': 'ENSG00000153113', 'ENSG00000260060': nan, 'ENSG00000273486': nan, 'ENSG00000258772': 'ENSG00000259130', 'ENSG00000285877': 'ENSG00000266473', 'ENSG00000249604': 'ENSG00000245293', 'ENSG00000259038': 'ENSG00000258623', 'ENSG00000291288': 'ENSG00000223784', 'ENSG00000290958': 'ENSG00000228315', 'ENSG00000289455': 'ENSG00000225930', 'ENSG00000260793': 'ENSG00000267394', 'ENSG00000286292': 'ENSG00000271646', 'ENSG00000287843': 'ENSG00000231460', 'ENSG00000239462': 'ENSG00000240476', 'ENSG00000285424': 'ENSG00000284823', 'ENSG00000287101': 'ENSG00000248491', 'ENSG00000260090': 'ENSG00000248764', 'ENSG00000288819': nan, 'ENSG00000262089': 'ENSG00000267532', 'ENSG00000256001': 'ENSG00000249345', 'ENSG00000235736': 'ENSG00000293597', 'ENSG00000236129': 'ENSG00000237612', 'ENSG00000241180': 'ENSG00000272438', 'ENSG00000289413': 'ENSG00000288958', 'ENSG00000284930': 'ENSG00000133961', 'ENSG00000233005': 'ENSG00000231204', 'ENSG00000257654': 'ENSG00000257883', 'ENSG00000276649': 'ENSG00000286288', 'ENSG00000278107': 'ENSG00000264278', 'ENSG00000287944': 'ENSG00000267284', 'ENSG00000288788': 'ENSG00000255240', 'ENSG00000272382': nan, 'ENSG00000240401': 'ENSG00000293422', 'ENSG00000289541': 'ENSG00000256357', 'ENSG00000228255': 'ENSG00000225233', 'ENSG00000264072': 'ENSG00000266604', 'ENSG00000290594': 'ENSG00000287151', 'ENSG00000272592': 'ENSG00000233643', 'ENSG00000256101': 'ENSG00000215241', 'ENSG00000237461': 'ENSG00000271086', 'ENSG00000259604': 'ENSG00000272808', 'ENSG00000261445': 'ENSG00000290413', 'ENSG00000291290': 'ENSG00000281903', 'ENSG00000288896': 'ENSG00000242880', 'ENSG00000260052': nan, 'ENSG00000288111': 'ENSG00000249846', 'ENSG00000290591': 'ENSG00000286058', 'ENSG00000235209': 'ENSG00000288106', 'ENSG00000233029': 'ENSG00000291123', 'ENSG00000286329': 'ENSG00000242828', 'ENSG00000258407': 'ENSG00000258826', 'ENSG00000229458': 'ENSG00000225738', 'ENSG00000287702': 'ENSG00000287568', 'ENSG00000272515': nan, 'ENSG00000287368': 'ENSG00000230074', 'ENSG00000290816': 'ENSG00000244479', 'ENSG00000254143': 'ENSG00000253665', 'ENSG00000259730': 'ENSG00000259754', 'ENSG00000276867': nan, 'ENSG00000259376': 'ENSG00000259755', 'ENSG00000260030': nan, 'ENSG00000237359': 'ENSG00000232104', 'ENSG00000288583': 'ENSG00000230074', 'ENSG00000229403': 'ENSG00000285741', 'ENSG00000250418': nan, 'ENSG00000288554': 'ENSG00000261103', 'ENSG00000232325': 'ENSG00000287883', 'ENSG00000267643': 'ENSG00000293190', 'ENSG00000289619': nan, 'ENSG00000229646': 'ENSG00000234261', 'ENSG00000290903': 'ENSG00000226067', 'ENSG00000242396': 'ENSG00000233203', 'ENSG00000227920': 'ENSG00000204110', 'ENSG00000264212': 'ENSG00000266014', 'ENSG00000250078': 'ENSG00000250340', 'ENSG00000258885': nan, 'ENSG00000288986': 'ENSG00000234572', 'ENSG00000286439': 'ENSG00000270111', 'ENSG00000256542': 'ENSG00000256943', 'ENSG00000285155': 'ENSG00000115221', 'ENSG00000223504': 'ENSG00000285838', 'ENSG00000289179': 'ENSG00000248174', 'ENSG00000269072': 'ENSG00000268595', 'ENSG00000233517': 'ENSG00000272568', 'ENSG00000279159': nan, 'ENSG00000286016': 'ENSG00000286791', 'ENSG00000254695': 'ENSG00000254946', 'ENSG00000239498': 'ENSG00000288902', 'ENSG00000272482': 'ENSG00000162496', 'ENSG00000280273': 'ENSG00000236827', 'ENSG00000261350': nan, 'ENSG00000236716': 'ENSG00000232624', 'ENSG00000232680': 'ENSG00000288731', 'ENSG00000260141': 'ENSG00000291268', 'ENSG00000289122': 'ENSG00000223486', 'ENSG00000290010': 'ENSG00000294436', 'ENSG00000254927': 'ENSG00000254930', 'ENSG00000290407': 'ENSG00000291282', 'ENSG00000254731': nan, 'ENSG00000255644': nan, 'ENSG00000283689': 'ENSG00000287885', 'ENSG00000273733': nan, 'ENSG00000279987': 'ENSG00000276095', 'ENSG00000289005': 'ENSG00000244265', 'ENSG00000232386': 'ENSG00000272808', 'ENSG00000255015': 'ENSG00000255248', 'ENSG00000224532': 'ENSG00000226281', 'ENSG00000290616': 'ENSG00000287151', 'ENSG00000285759': 'ENSG00000229950', 'ENSG00000233408': 'ENSG00000230471', 'ENSG00000291259': 'ENSG00000291082', 'ENSG00000238358': nan, 'ENSG00000253735': 'ENSG00000254237', 'ENSG00000285933': 'ENSG00000290775', 'ENSG00000249463': 'ENSG00000251598', 'ENSG00000247810': 'ENSG00000261761', 'ENSG00000286374': 'ENSG00000255344', 'ENSG00000272519': nan, 'ENSG00000255944': 'ENSG00000256128', 'ENSG00000272444': nan, 'ENSG00000253400': 'ENSG00000254261', 'ENSG00000241280': 'ENSG00000214407', 'ENSG00000287875': 'ENSG00000293612', 'ENSG00000233970': 'ENSG00000227713', 'ENSG00000290721': 'ENSG00000167562', 'ENSG00000289950': 'ENSG00000268518', 'ENSG00000263081': 'ENSG00000277675', 'ENSG00000235111': 'ENSG00000188511', 'ENSG00000258919': 'ENSG00000258404', 'ENSG00000290463': 'ENSG00000197332', 'ENSG00000283294': 'ENSG00000290461', 'ENSG00000286213': 'ENSG00000228067', 'ENSG00000268509': 'ENSG00000233912', 'ENSG00000259972': 'ENSG00000290890', 'ENSG00000289600': 'ENSG00000232053', 'ENSG00000254619': 'ENSG00000254508', 'ENSG00000272070': 'ENSG00000288095', 'ENSG00000232749': 'ENSG00000290551', 'ENSG00000254138': 'ENSG00000289601', 'ENSG00000291026': 'ENSG00000291027', 'ENSG00000286262': 'ENSG00000223768', 'ENSG00000258600': 'ENSG00000258426', 'ENSG00000277501': 'ENSG00000277688', 'ENSG00000291299': 'ENSG00000228315', 'ENSG00000265964': 'ENSG00000265547', 'ENSG00000253256': 'ENSG00000253315', 'ENSG00000272316': 'ENSG00000290555', 'ENSG00000290666': 'ENSG00000290665', 'ENSG00000232754': 'ENSG00000231993', 'ENSG00000289830': 'ENSG00000287092', 'ENSG00000287523': 'ENSG00000228944', 'ENSG00000291215': 'ENSG00000225880', 'ENSG00000287858': 'ENSG00000222017', 'ENSG00000286189': 'ENSG00000180769', 'ENSG00000249731': 'ENSG00000248994', 'ENSG00000251613': 'ENSG00000249085', 'ENSG00000254587': 'ENSG00000256684', 'ENSG00000259762': 'ENSG00000259416', 'ENSG00000290818': 'ENSG00000244198', 'ENSG00000258131': nan, 'ENSG00000260095': 'ENSG00000260874', 'ENSG00000291273': nan, 'ENSG00000286103': 'ENSG00000293062', 'ENSG00000266601': 'ENSG00000234494', 'ENSG00000234698': nan, 'ENSG00000286854': 'ENSG00000291293', 'ENSG00000272940': 'ENSG00000205559', 'ENSG00000289862': 'ENSG00000259230', 'ENSG00000241475': 'ENSG00000228044', 'ENSG00000253115': 'ENSG00000254288', 'ENSG00000235047': 'ENSG00000289349', 'ENSG00000267787': 'ENSG00000267040', 'ENSG00000278878': 'ENSG00000280145', 'ENSG00000289329': nan, 'ENSG00000272736': 'ENSG00000272975', 'ENSG00000283164': 'ENSG00000230385', 'ENSG00000236938': 'ENSG00000285090', 'ENSG00000226849': 'ENSG00000230337', 'ENSG00000270265': 'ENSG00000251611', 'ENSG00000278514': 'ENSG00000214432', 'ENSG00000237954': 'ENSG00000230434', 'ENSG00000287211': 'ENSG00000238129', 'ENSG00000256923': nan, 'ENSG00000242753': 'ENSG00000233656', 'ENSG00000273181': 'ENSG00000272721', 'ENSG00000273481': 'ENSG00000237976', 'ENSG00000264334': 'ENSG00000265843', 'ENSG00000254420': 'ENSG00000288853', 'ENSG00000286824': nan, 'ENSG00000242375': 'ENSG00000203279', 'ENSG00000235852': 'ENSG00000228509', 'ENSG00000259123': 'ENSG00000258754', 'ENSG00000234141': 'ENSG00000233108', 'ENSG00000265717': 'ENSG00000266743', 'ENSG00000286792': 'ENSG00000265445', 'ENSG00000268810': 'ENSG00000269151', 'ENSG00000251379': 'ENSG00000249252', 'ENSG00000289184': 'ENSG00000287862', 'ENSG00000289614': 'ENSG00000291233', 'ENSG00000234998': 'ENSG00000291123', 'ENSG00000281920': 'ENSG00000204929', 'ENSG00000290969': 'ENSG00000290718', 'ENSG00000272732': nan, 'ENSG00000225836': nan, 'ENSG00000261120': 'ENSG00000258553', 'ENSG00000253983': 'ENSG00000254288', 'ENSG00000233799': 'ENSG00000249592', 'ENSG00000285922': 'ENSG00000225206', 'ENSG00000241792': 'ENSG00000241131', 'ENSG00000214955': 'ENSG00000227456', 'ENSG00000288810': 'ENSG00000225206', 'ENSG00000261502': 'ENSG00000260658', 'ENSG00000235450': 'ENSG00000223665', 'ENSG00000286978': 'ENSG00000287181', 'ENSG00000250567': 'ENSG00000293393', 'ENSG00000250310': 'ENSG00000249906', 'ENSG00000254115': 'ENSG00000253778', 'ENSG00000248787': 'ENSG00000241288', 'ENSG00000273211': 'ENSG00000223343', 'ENSG00000288573': 'ENSG00000237950', 'ENSG00000250415': 'ENSG00000246214', 'ENSG00000266248': 'ENSG00000264515', 'ENSG00000266893': 'ENSG00000290606', 'ENSG00000278266': 'ENSG00000286791', 'ENSG00000289048': 'ENSG00000254024', 'ENSG00000224228': 'ENSG00000224000', 'ENSG00000227305': 'ENSG00000234387', 'ENSG00000286362': nan, 'ENSG00000254028': 'ENSG00000253433', 'ENSG00000286687': 'ENSG00000287246', 'ENSG00000267683': 'ENSG00000267339', 'ENSG00000290396': 'ENSG00000290394', 'ENSG00000254095': 'ENSG00000253377', 'ENSG00000257715': 'ENSG00000257878', 'ENSG00000253985': 'ENSG00000250387', 'ENSG00000289636': 'ENSG00000305148', 'ENSG00000272240': 'ENSG00000288985', 'ENSG00000225655': 'ENSG00000291075', 'ENSG00000263688': 'ENSG00000285095', 'ENSG00000288904': nan, 'ENSG00000291284': 'ENSG00000290482', 'ENSG00000273138': 'ENSG00000233942', 'ENSG00000234677': 'ENSG00000287768', 'ENSG00000272030': 'ENSG00000189171', 'ENSG00000272840': 'ENSG00000284624', 'ENSG00000287837': 'ENSG00000290476', 'ENSG00000286534': 'ENSG00000267506', 'ENSG00000290412': 'ENSG00000254518', 'ENSG00000274444': 'ENSG00000259460', 'ENSG00000261650': 'ENSG00000214039', 'ENSG00000290124': 'ENSG00000290683', 'ENSG00000291025': 'ENSG00000291024', 'ENSG00000291247': 'ENSG00000290775', 'ENSG00000289147': 'ENSG00000264672', 'ENSG00000282863': 'ENSG00000230417', 'ENSG00000290529': 'ENSG00000165511', 'ENSG00000288580': 'ENSG00000253853', 'ENSG00000249091': 'ENSG00000248397', 'ENSG00000272085': 'ENSG00000254350', 'ENSG00000259423': 'ENSG00000175779', 'ENSG00000223795': 'ENSG00000228877', 'ENSG00000250481': 'ENSG00000251426', 'ENSG00000285719': 'ENSG00000284237', 'ENSG00000291291': nan, 'ENSG00000267316': 'ENSG00000267175', 'ENSG00000253766': 'ENSG00000291102', 'ENSG00000289206': 'ENSG00000177738', 'ENSG00000286466': 'ENSG00000225718', 'ENSG00000249483': 'ENSG00000287938', 'ENSG00000287492': 'ENSG00000289195', 'ENSG00000289008': 'ENSG00000245864', 'ENSG00000261305': 'ENSG00000301866', 'ENSG00000249184': 'ENSG00000251249', 'ENSG00000226376': 'ENSG00000231902', 'ENSG00000231877': 'ENSG00000234464', 'ENSG00000249269': 'ENSG00000251619', 'ENSG00000289826': 'ENSG00000293442', 'ENSG00000274367': 'ENSG00000262362', 'ENSG00000234306': 'ENSG00000270111', 'ENSG00000286111': nan, 'ENSG00000250431': 'ENSG00000248551', 'ENSG00000290066': 'ENSG00000285847', 'ENSG00000291011': 'ENSG00000232827', 'ENSG00000258196': 'ENSG00000257155', 'ENSG00000254039': 'ENSG00000248954', 'ENSG00000289593': 'ENSG00000303589', 'ENSG00000241520': 'ENSG00000232485', 'ENSG00000286572': 'ENSG00000176754', 'ENSG00000284685': 'ENSG00000103472', 'ENSG00000254814': 'ENSG00000293621', 'ENSG00000287346': 'ENSG00000258735', 'ENSG00000276471': 'ENSG00000241163', 'ENSG00000270114': 'ENSG00000203335', 'ENSG00000284602': 'ENSG00000284657', 'ENSG00000287316': 'ENSG00000236213', 'ENSG00000290414': nan, 'ENSG00000291055': 'ENSG00000291079', 'ENSG00000266908': 'ENSG00000230563', 'ENSG00000273123': nan, 'ENSG00000231255': 'ENSG00000233073', 'ENSG00000289209': nan, 'ENSG00000263677': 'ENSG00000260372', 'ENSG00000273203': 'ENSG00000290931', 'ENSG00000288581': 'ENSG00000286757', 'ENSG00000289512': 'ENSG00000225519', 'ENSG00000253859': 'ENSG00000253374', 'ENSG00000261044': 'ENSG00000277332', 'ENSG00000235242': 'ENSG00000286513', 'ENSG00000236453': 'ENSG00000236861', 'ENSG00000289465': 'ENSG00000254659', 'ENSG00000273319': 'ENSG00000233559', 'ENSG00000270038': nan, 'ENSG00000244620': 'ENSG00000226777', 'ENSG00000244650': 'ENSG00000241131', 'ENSG00000214970': 'ENSG00000272975', 'ENSG00000230506': 'ENSG00000232448', 'ENSG00000291188': 'ENSG00000103472', 'ENSG00000250948': 'ENSG00000248714', 'ENSG00000283003': 'ENSG00000261617', 'ENSG00000287865': 'ENSG00000228521', 'ENSG00000261193': 'ENSG00000205037', 'ENSG00000290004': 'ENSG00000288946', 'ENSG00000258831': 'ENSG00000258476', 'ENSG00000234083': 'ENSG00000178457', 'ENSG00000259255': 'ENSG00000287704', 'ENSG00000246331': 'ENSG00000285517', 'ENSG00000273403': nan, 'ENSG00000269927': 'ENSG00000286423', 'ENSG00000290514': 'ENSG00000310532', 'ENSG00000258170': 'ENSG00000257837', 'ENSG00000258216': 'ENSG00000271614', 'ENSG00000290029': 'ENSG00000236700', 'ENSG00000253842': 'ENSG00000253824', 'ENSG00000287708': 'ENSG00000225472', 'ENSG00000271904': 'ENSG00000245526', 'ENSG00000280767': 'ENSG00000281849', 'ENSG00000222022': 'ENSG00000222032', 'ENSG00000258949': 'ENSG00000258633', 'ENSG00000249492': 'ENSG00000248554', 'ENSG00000250237': 'ENSG00000248884', 'ENSG00000290443': 'ENSG00000243008', 'ENSG00000278626': nan, 'ENSG00000251654': 'ENSG00000248605', 'ENSG00000286815': 'ENSG00000293042', 'ENSG00000233193': 'ENSG00000233002', 'ENSG00000283648': 'ENSG00000287636', 'ENSG00000287868': 'ENSG00000286782', 'ENSG00000241317': 'ENSG00000293221', 'ENSG00000276251': 'ENSG00000267498', 'ENSG00000267070': 'ENSG00000285567', 'ENSG00000267081': 'ENSG00000261316', 'ENSG00000287168': 'ENSG00000288838', 'ENSG00000276348': 'ENSG00000277350', 'ENSG00000233536': 'ENSG00000255666', 'ENSG00000286599': 'ENSG00000287382', 'ENSG00000231883': 'ENSG00000223598', 'ENSG00000287564': 'ENSG00000272381', 'ENSG00000253737': 'ENSG00000253355', 'ENSG00000233376': 'ENSG00000229005', 'ENSG00000253844': 'ENSG00000228801', 'ENSG00000253584': 'ENSG00000251574', 'ENSG00000223469': 'ENSG00000283215', 'ENSG00000260487': 'ENSG00000261346', 'ENSG00000272304': 'ENSG00000249413', 'ENSG00000230381': 'ENSG00000233154', 'ENSG00000287313': 'ENSG00000235009', 'ENSG00000266450': 'ENSG00000266602', 'ENSG00000279949': nan, 'ENSG00000257711': nan, 'ENSG00000240440': 'ENSG00000223725', 'ENSG00000289840': nan, 'ENSG00000277010': 'ENSG00000196557', 'ENSG00000250284': 'ENSG00000249803', 'ENSG00000228334': 'ENSG00000217455'}\n",
            "--------------------------------------------------\n",
            "Successfully mapped 11976 out of 12044 Ensembl IDs.\n",
            "--------------------------------------------------\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Compile final gene symbols and ensg ids from the two mappings"
      ],
      "metadata": {
        "id": "Tx10pxWT67rr"
      },
      "id": "Tx10pxWT67rr"
    },
    {
      "cell_type": "code",
      "source": [
        "var_df_clean = var_symb.merge(var_ensg, left_index=True, right_index=True, suffixes=('_symb', '_ensg'))\n",
        "var_df_clean = var_df_clean.drop(columns=['gene_inputs_symb', 'gene_inputs_ensg', 'ensembl_inputs_symb', 'ensembl_inputs_ensg'])\n",
        "var_df_clean['clean_gene_symbol'] = None\n",
        "var_df_clean['clean_ensg_id'] = None\n",
        "\n",
        "ensg_starts_mask = var_df_clean.index.str.startswith('ENSG')\n",
        "\n",
        "# if original id starts with ENSG - assign ensg mapping\n",
        "var_df_clean.loc[ensg_starts_mask, 'clean_gene_symbol'] = var_df_clean.loc[ensg_starts_mask, 'gene_symbol_ensg']\n",
        "var_df_clean.loc[ensg_starts_mask, 'clean_ensg_id'] = var_df_clean.loc[ensg_starts_mask, 'ensembl_gene_id_ensg']\n",
        "\n",
        "# if not, assign symb mapping\n",
        "var_df_clean.loc[~ensg_starts_mask, 'clean_gene_symbol'] = var_df_clean.loc[~ensg_starts_mask, 'gene_symbol_symb']\n",
        "var_df_clean.loc[~ensg_starts_mask, 'clean_ensg_id'] = var_df_clean.loc[~ensg_starts_mask, 'ensembl_gene_id_symb']\n",
        "\n",
        "# replace missing ensg with\n",
        "var_df_clean.loc[var_df_clean['clean_ensg_id'].isna() & ensg_starts_mask, 'clean_ensg_id'] = var_df_clean.loc[var_df_clean['clean_ensg_id'].isna() & ensg_starts_mask].index\n",
        "\n",
        "var_df_clean = var_df_clean.rename(columns={'clean_gene_symbol':'gene_symbol', 'clean_ensg_id':'ensembl_gene_id'})[['ensembl_gene_id', 'gene_symbol']]\n",
        "\n",
        "var_df_clean\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 455
        },
        "id": "IZUfeXcW6-1h",
        "outputId": "19c54eec-a8b8-49fc-d461-c468b75c84ec"
      },
      "id": "IZUfeXcW6-1h",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                 ensembl_gene_id  gene_symbol\n",
              "index                                        \n",
              "DDX11L2          ENSG00000310539      DDX11L2\n",
              "MIR1302-2HG      ENSG00000243485  MIR1302-2HG\n",
              "FAM138A          ENSG00000237613      FAM138A\n",
              "ENSG00000290826  ENSG00000290826         None\n",
              "OR4F5            ENSG00000186092        OR4F5\n",
              "...                          ...          ...\n",
              "ENSG00000277836  ENSG00000277836          NaN\n",
              "ENSG00000278633  ENSG00000278633          NaN\n",
              "ENSG00000276017  ENSG00000276017          NaN\n",
              "ENSG00000278817  ENSG00000278817          NaN\n",
              "ENSG00000277196  ENSG00000277196          NaN\n",
              "\n",
              "[38606 rows x 2 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-9aa906f9-6af9-43b0-8936-c57040903d27\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>ensembl_gene_id</th>\n",
              "      <th>gene_symbol</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>DDX11L2</th>\n",
              "      <td>ENSG00000310539</td>\n",
              "      <td>DDX11L2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MIR1302-2HG</th>\n",
              "      <td>ENSG00000243485</td>\n",
              "      <td>MIR1302-2HG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>FAM138A</th>\n",
              "      <td>ENSG00000237613</td>\n",
              "      <td>FAM138A</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000290826</th>\n",
              "      <td>ENSG00000290826</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>OR4F5</th>\n",
              "      <td>ENSG00000186092</td>\n",
              "      <td>OR4F5</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000277836</th>\n",
              "      <td>ENSG00000277836</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000278633</th>\n",
              "      <td>ENSG00000278633</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000276017</th>\n",
              "      <td>ENSG00000276017</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000278817</th>\n",
              "      <td>ENSG00000278817</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000277196</th>\n",
              "      <td>ENSG00000277196</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>38606 rows × 2 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-9aa906f9-6af9-43b0-8936-c57040903d27')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-9aa906f9-6af9-43b0-8936-c57040903d27 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-9aa906f9-6af9-43b0-8936-c57040903d27');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-c6bc099a-fb9f-458b-a9a6-278679a6f149\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-c6bc099a-fb9f-458b-a9a6-278679a6f149')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-c6bc099a-fb9f-458b-a9a6-278679a6f149 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "  <div id=\"id_a535f02d-ad12-48e0-8e86-e8eebd568876\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('var_df_clean')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_a535f02d-ad12-48e0-8e86-e8eebd568876 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('var_df_clean');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "var_df_clean",
              "summary": "{\n  \"name\": \"var_df_clean\",\n  \"rows\": 38606,\n  \"fields\": [\n    {\n      \"column\": \"index\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 38606,\n        \"samples\": [\n          \"GARIN5A\",\n          \"ENSG00000286348\",\n          \"ENSG00000260986\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ensembl_gene_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 38438,\n        \"samples\": [\n          \"ENSG00000196565\",\n          \"ENSG00000289485\",\n          \"ENSG00000101558\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"gene_symbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 26004,\n        \"samples\": [\n          \"FARSB\",\n          \"WWC1\",\n          \"TBL3\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 54
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# assign fixed genes back to var\n",
        "cur_data.adata.var = var_df_clean"
      ],
      "metadata": {
        "id": "zUBIXz1S7E3w"
      },
      "id": "zUBIXz1S7E3w",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "var_df_clean.loc[var_df_clean['ensembl_gene_id'].notna() & var_df_clean['ensembl_gene_id'].str.startswith('LRG')]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 81
        },
        "id": "oLHcWgk67LbD",
        "outputId": "a5b19d6e-9aae-436d-f7e5-be6fd78c7d1e"
      },
      "id": "oLHcWgk67LbD",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "Empty DataFrame\n",
              "Columns: [ensembl_gene_id, gene_symbol]\n",
              "Index: []"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-171cdd69-8b89-4bc5-b315-fe539b3e1d75\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>ensembl_gene_id</th>\n",
              "      <th>gene_symbol</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-171cdd69-8b89-4bc5-b315-fe539b3e1d75')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-171cdd69-8b89-4bc5-b315-fe539b3e1d75 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-171cdd69-8b89-4bc5-b315-fe539b3e1d75');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "repr_error": "Out of range float values are not JSON compliant: nan"
            }
          },
          "metadata": {},
          "execution_count": 56
        }
      ]
    },
    {
      "cell_type": "markdown",
      "id": "38bca8db",
      "metadata": {
        "id": "38bca8db"
      },
      "source": [
        "### Validate var metadata"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "3fdf4305",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 455
        },
        "id": "3fdf4305",
        "outputId": "9fdeb065-0f81-4867-b53f-b39f8f5a3c88"
      },
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "                 ensembl_gene_id  gene_symbol\n",
              "index                                        \n",
              "DDX11L2          ENSG00000310539      DDX11L2\n",
              "MIR1302-2HG      ENSG00000243485  MIR1302-2HG\n",
              "FAM138A          ENSG00000237613      FAM138A\n",
              "ENSG00000290826  ENSG00000290826         None\n",
              "OR4F5            ENSG00000186092        OR4F5\n",
              "...                          ...          ...\n",
              "ENSG00000277836  ENSG00000277836          NaN\n",
              "ENSG00000278633  ENSG00000278633          NaN\n",
              "ENSG00000276017  ENSG00000276017          NaN\n",
              "ENSG00000278817  ENSG00000278817          NaN\n",
              "ENSG00000277196  ENSG00000277196          NaN\n",
              "\n",
              "[38606 rows x 2 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-6a2b3d4e-a35c-4cf8-95d4-b3c415ce3986\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>ensembl_gene_id</th>\n",
              "      <th>gene_symbol</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>DDX11L2</th>\n",
              "      <td>ENSG00000310539</td>\n",
              "      <td>DDX11L2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MIR1302-2HG</th>\n",
              "      <td>ENSG00000243485</td>\n",
              "      <td>MIR1302-2HG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>FAM138A</th>\n",
              "      <td>ENSG00000237613</td>\n",
              "      <td>FAM138A</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000290826</th>\n",
              "      <td>ENSG00000290826</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>OR4F5</th>\n",
              "      <td>ENSG00000186092</td>\n",
              "      <td>OR4F5</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000277836</th>\n",
              "      <td>ENSG00000277836</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000278633</th>\n",
              "      <td>ENSG00000278633</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000276017</th>\n",
              "      <td>ENSG00000276017</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000278817</th>\n",
              "      <td>ENSG00000278817</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000277196</th>\n",
              "      <td>ENSG00000277196</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>38606 rows × 2 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-6a2b3d4e-a35c-4cf8-95d4-b3c415ce3986')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-6a2b3d4e-a35c-4cf8-95d4-b3c415ce3986 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-6a2b3d4e-a35c-4cf8-95d4-b3c415ce3986');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-c45deb1a-d702-4d96-9198-892dcc4e430c\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-c45deb1a-d702-4d96-9198-892dcc4e430c')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-c45deb1a-d702-4d96-9198-892dcc4e430c button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"cur_data\",\n  \"rows\": 38606,\n  \"fields\": [\n    {\n      \"column\": \"index\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 38606,\n        \"samples\": [\n          \"GARIN5A\",\n          \"ENSG00000286348\",\n          \"ENSG00000260986\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ensembl_gene_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 38438,\n        \"samples\": [\n          \"ENSG00000196565\",\n          \"ENSG00000289485\",\n          \"ENSG00000101558\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"gene_symbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 26004,\n        \"samples\": [\n          \"FARSB\",\n          \"WWC1\",\n          \"TBL3\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        }
      ],
      "source": [
        "cur_data.validate_data(slot='var')"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "c72b1ccf",
      "metadata": {
        "id": "c72b1ccf"
      },
      "source": [
        "# Save the dataset"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "0f014cae",
      "metadata": {
        "id": "0f014cae",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0129ad30-2a7a-4b62-8c6a-d37baf43f570"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/content/PerturbationCatalogue/data_exploration/curation_tools/curation_tools.py:327: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
            "  adata.obs = adata.obs.fillna(value=np.nan)\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✅ Curated h5ad data saved to Perturbseq/curated/h5ad/orion_2025_hct116_curated.h5ad\n"
          ]
        }
      ],
      "source": [
        "cur_data.save_curated_data_h5ad()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "7e083fcb",
      "metadata": {
        "id": "7e083fcb",
        "outputId": "a7851f46-67ca-49fb-bd6a-64f055ca00b7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✅ Metadata saved to Perturbseq/curated/parquet/orion_2025_hct116_curated_metadata.parquet\n"
          ]
        }
      ],
      "source": [
        "cur_data.save_curated_data_parquet(split_metadata=True, save_metadata_only=True)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Upload to BigQuery"
      ],
      "metadata": {
        "id": "0QSGe4d37d6g"
      },
      "id": "0QSGe4d37d6g"
    },
    {
      "cell_type": "code",
      "source": [
        "upload_parquet_to_bq(\n",
        "    parquet_path='/content/PerturbationCatalogue/data_exploration/Perturbseq/curated/parquet/orion_2025_hct116_curated_metadata.parquet',\n",
        "    bq_dataset_id='prj-ext-dev-pertcat-437314.perturb_seq',\n",
        "    bq_table_name='metadata',\n",
        "    key_columns=['dataset_id', 'sample_id'],\n",
        "    verbose=True\n",
        ")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0Gt9w-al7h4j",
        "outputId": "1b088068-f4e8-4a52-b033-3e3854ec0174"
      },
      "id": "0Gt9w-al7h4j",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Staging table: loading `.parquet` file /content/PerturbationCatalogue/data_exploration/Perturbseq/curated/parquet/orion_2025_hct116_curated_metadata.parquet to prj-ext-dev-pertcat-437314.perturb_seq.metadata_staging...\n",
            "Staging table: loaded 3409169 rows to prj-ext-dev-pertcat-437314.perturb_seq.metadata_staging\n",
            "Staging table: added ingested_at timestamp column to prj-ext-dev-pertcat-437314.perturb_seq.metadata_staging\n",
            "Merge completed: staging → prj-ext-dev-pertcat-437314.perturb_seq.metadata with type-safe casting.\n",
            "Staging table: deleted prj-ext-dev-pertcat-437314.perturb_seq.metadata_staging\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Upload to GC Storage"
      ],
      "metadata": {
        "id": "6w22shDp7ldv"
      },
      "id": "6w22shDp7ldv"
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "H516751FbB1z",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "H516751FbB1z",
        "outputId": "020c785d-bfea-4b90-f481-43cae91fa4e5"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[1;33mWARNING:\u001b[0m Parallel composite upload was turned ON to get the best performance on\n",
            "uploading large objects. If you would like to opt-out and instead\n",
            "perform a normal upload, run:\n",
            "`gcloud config set storage/parallel_composite_upload_enabled False`\n",
            "If you would like to disable this warning, run:\n",
            "`gcloud config set storage/parallel_composite_upload_enabled True`\n",
            "Note that with parallel composite uploads, your object might be\n",
            "uploaded as a composite object\n",
            "(https://cloud.google.com/storage/docs/composite-objects), which means\n",
            "that any user who downloads your object will need to use crc32c\n",
            "checksums to verify data integrity. gcloud storage is capable of\n",
            "computing crc32c checksums, but this might pose a problem for other\n",
            "clients.\n",
            "\n",
            "Copying file://Perturbseq/curated/h5ad/orion_2025_hct116_curated.h5ad to gs://perturbation-catalogue-lake/perturbseq/curated/orion_2025_hct116_curated.h5ad\n"
          ]
        }
      ],
      "source": [
        "!gcloud storage cp Perturbseq/curated/h5ad/orion_2025_hct116_curated.h5ad gs://perturbation-catalogue-lake/perturbseq/curated/"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "name": "orion_2025_hct116.ipynb",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "PerturbationCatalogue",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.12.8"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}