{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 1,
      "id": "IfVEJsY2-cD0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IfVEJsY2-cD0",
        "outputId": "c81a5db5-0256-4cda-848a-0bc455ace237"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content\n"
          ]
        }
      ],
      "source": [
        "!pwd"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "id": "1wjh-CBq-fLv",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1wjh-CBq-fLv",
        "outputId": "b200e1d8-81de-42d8-f9b5-44cc57c26959"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "fatal: destination path 'PerturbationCatalogue' already exists and is not an empty directory.\n"
          ]
        }
      ],
      "source": [
        "!git clone https://github.com/EMBL-EBI-ABC/PerturbationCatalogue.git"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "id": "gjFGZPGu-hlO",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gjFGZPGu-hlO",
        "outputId": "2e60caa7-a232-46ad-cef6-a1b0ee251c46"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/PerturbationCatalogue/data_exploration\n"
          ]
        }
      ],
      "source": [
        "%cd PerturbationCatalogue/data_exploration/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "id": "Hvdndxyp-jio",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Hvdndxyp-jio",
        "outputId": "b60457e4-40a7-42f0-c3ea-36854706a362"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Already up to date.\n",
            "Already on 'aleks/curating_orion_2025'\n",
            "Your branch is up to date with 'origin/aleks/curating_orion_2025'.\n"
          ]
        }
      ],
      "source": [
        "!git pull\n",
        "!git switch aleks/curating_orion_2025"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "id": "pwzakG-S-rkM",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "pwzakG-S-rkM",
        "outputId": "60951be5-902a-42ba-f782-7390ffeef4eb"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: ipykernel in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 2)) (6.17.1)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 5)) (3.10.0)\n",
            "Requirement already satisfied: scipy<1.15.0 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 6)) (1.14.1)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 7)) (2.3.1)\n",
            "Requirement already satisfied: decoupler==2.0.1 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 10)) (2.0.1)\n",
            "Requirement already satisfied: pertpy==0.11.4 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 11)) (0.11.4)\n",
            "Requirement already satisfied: scanpy==1.11.1 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 12)) (1.11.1)\n",
            "Requirement already satisfied: snapatac2 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 13)) (2.8.0)\n",
            "Requirement already satisfied: pandera in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 16)) (0.25.0)\n",
            "Requirement already satisfied: libchebipy in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 17)) (1.0.10)\n",
            "Requirement already satisfied: gprofiler-official in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 18)) (1.0.0)\n",
            "Requirement already satisfied: adjusttext in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (1.3.0)\n",
            "Requirement already satisfied: anndata in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (0.10.9)\n",
            "Requirement already satisfied: docrep in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (0.3.2)\n",
            "Requirement already satisfied: igraph in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (0.11.9)\n",
            "Requirement already satisfied: marsilea in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (0.5.3)\n",
            "Requirement already satisfied: numba in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (0.60.0)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (2.32.3)\n",
            "Requirement already satisfied: seaborn in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (0.13.2)\n",
            "Requirement already satisfied: session-info2 in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (0.1.2)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (4.67.1)\n",
            "Requirement already satisfied: xgboost in /usr/local/lib/python3.11/dist-packages (from decoupler==2.0.1->-r requirements.txt (line 10)) (2.1.4)\n",
            "Requirement already satisfied: arviz in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (0.21.0)\n",
            "Requirement already satisfied: blitzgsea in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (1.3.54)\n",
            "Requirement already satisfied: fast-array-utils in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (1.2.1)\n",
            "Requirement already satisfied: lamin-utils in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (0.15.0)\n",
            "Requirement already satisfied: mudata in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (0.3.2)\n",
            "Requirement already satisfied: openpyxl in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (3.1.5)\n",
            "Requirement already satisfied: ott-jax in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (0.5.0)\n",
            "Requirement already satisfied: pubchempy in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (1.0.4)\n",
            "Requirement already satisfied: pyarrow in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (18.1.0)\n",
            "Requirement already satisfied: rich in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (13.9.4)\n",
            "Requirement already satisfied: scikit-learn>=1.4 in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (1.5.2)\n",
            "Requirement already satisfied: scikit-misc in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (0.5.1)\n",
            "Requirement already satisfied: scvi-tools in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (1.2.1)\n",
            "Requirement already satisfied: sparsecca in /usr/local/lib/python3.11/dist-packages (from pertpy==0.11.4->-r requirements.txt (line 11)) (0.3.1)\n",
            "Requirement already satisfied: h5py>=3.7 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (3.14.0)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (1.5.1)\n",
            "Requirement already satisfied: legacy-api-wrap>=1.4 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (1.4.1)\n",
            "Requirement already satisfied: natsort in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (8.4.0)\n",
            "Requirement already satisfied: networkx>=2.7 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (3.5)\n",
            "Requirement already satisfied: numpy>=1.24 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (2.3.1)\n",
            "Requirement already satisfied: packaging>=21.3 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (24.2)\n",
            "Requirement already satisfied: patsy!=1.0.0 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (1.0.1)\n",
            "Requirement already satisfied: pynndescent>=0.5 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (0.5.13)\n",
            "Requirement already satisfied: statsmodels>=0.13 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (0.14.4)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (4.14.0)\n",
            "Requirement already satisfied: umap-learn!=0.5.0,>=0.5 in /usr/local/lib/python3.11/dist-packages (from scanpy==1.11.1->-r requirements.txt (line 12)) (0.5.7)\n",
            "Requirement already satisfied: debugpy>=1.0 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (1.8.0)\n",
            "Requirement already satisfied: ipython>=7.23.1 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (7.34.0)\n",
            "Requirement already satisfied: jupyter-client>=6.1.12 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (6.1.12)\n",
            "Requirement already satisfied: matplotlib-inline>=0.1 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (0.1.7)\n",
            "Requirement already satisfied: nest-asyncio in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (1.6.0)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (5.9.5)\n",
            "Requirement already satisfied: pyzmq>=17 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (24.0.1)\n",
            "Requirement already satisfied: tornado>=6.1 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (6.4.2)\n",
            "Requirement already satisfied: traitlets>=5.1.0 in /usr/local/lib/python3.11/dist-packages (from ipykernel->-r requirements.txt (line 2)) (5.7.1)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (1.3.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (0.12.1)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (4.58.4)\n",
            "Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (1.4.8)\n",
            "Requirement already satisfied: pillow>=8 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (11.2.1)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (3.2.3)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.11/dist-packages (from matplotlib->-r requirements.txt (line 5)) (2.9.0.post0)\n",
            "Collecting numpy>=1.24 (from scanpy==1.11.1->-r requirements.txt (line 12))\n",
            "  Downloading numpy-2.2.6-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (62 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m62.0/62.0 kB\u001b[0m \u001b[31m11.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->-r requirements.txt (line 7)) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas->-r requirements.txt (line 7)) (2025.2)\n",
            "Requirement already satisfied: kaleido in /usr/local/lib/python3.11/dist-packages (from snapatac2->-r requirements.txt (line 13)) (1.0.0)\n",
            "Requirement already satisfied: multiprocess in /usr/local/lib/python3.11/dist-packages (from snapatac2->-r requirements.txt (line 13)) (0.70.15)\n",
            "Requirement already satisfied: macs3<3.1,>=3.0 in /usr/local/lib/python3.11/dist-packages (from snapatac2->-r requirements.txt (line 13)) (3.0.3)\n",
            "  Downloading numpy-1.26.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m61.0/61.0 kB\u001b[0m \u001b[31m12.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting pandas (from -r requirements.txt (line 7))\n",
            "  Downloading pandas-2.1.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (18 kB)\n",
            "Requirement already satisfied: plotly<6.0.0,>=5.19.0 in /usr/local/lib/python3.11/dist-packages (from snapatac2->-r requirements.txt (line 13)) (5.24.1)\n",
            "Requirement already satisfied: polars<2.0,>=1.0 in /usr/local/lib/python3.11/dist-packages (from snapatac2->-r requirements.txt (line 13)) (1.21.0)\n",
            "Requirement already satisfied: pooch<2.0.0,>=1.6.0 in /usr/local/lib/python3.11/dist-packages (from snapatac2->-r requirements.txt (line 13)) (1.8.2)\n",
            "Requirement already satisfied: pyfaidx<0.8.0,>=0.7.0 in /usr/local/lib/python3.11/dist-packages (from snapatac2->-r requirements.txt (line 13)) (0.7.2.2)\n",
            "Requirement already satisfied: rustworkx in /usr/local/lib/python3.11/dist-packages (from snapatac2->-r requirements.txt (line 13)) (0.16.0)\n",
            "Requirement already satisfied: typeguard>=4.0 in /usr/local/lib/python3.11/dist-packages (from snapatac2->-r requirements.txt (line 13)) (4.4.4)\n",
            "Requirement already satisfied: pydantic in /usr/local/lib/python3.11/dist-packages (from pandera->-r requirements.txt (line 16)) (2.11.7)\n",
            "Requirement already satisfied: typing_inspect>=0.6.0 in /usr/local/lib/python3.11/dist-packages (from pandera->-r requirements.txt (line 16)) (0.9.0)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.11/dist-packages (from libchebipy->-r requirements.txt (line 17)) (1.17.0)\n",
            "Requirement already satisfied: array-api-compat!=1.5,>1.4 in /usr/local/lib/python3.11/dist-packages (from anndata->decoupler==2.0.1->-r requirements.txt (line 10)) (1.12.0)\n",
            "Requirement already satisfied: texttable>=1.6.2 in /usr/local/lib/python3.11/dist-packages (from igraph->decoupler==2.0.1->-r requirements.txt (line 10)) (1.7.0)\n",
            "Requirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (69.5.1)\n",
            "Requirement already satisfied: jedi>=0.16 in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.19.2)\n",
            "Requirement already satisfied: decorator in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (4.4.2)\n",
            "Requirement already satisfied: pickleshare in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.7.5)\n",
            "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (3.0.51)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (2.19.2)\n",
            "Requirement already satisfied: backcall in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.2.0)\n",
            "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.11/dist-packages (from ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (4.9.0)\n",
            "Requirement already satisfied: jupyter-core>=4.6.0 in /usr/local/lib/python3.11/dist-packages (from jupyter-client>=6.1.12->ipykernel->-r requirements.txt (line 2)) (5.8.1)\n",
            "Requirement already satisfied: hmmlearn>=0.3.2 in /usr/local/lib/python3.11/dist-packages (from macs3<3.1,>=3.0->snapatac2->-r requirements.txt (line 13)) (0.3.3)\n",
            "Requirement already satisfied: cykhash>=2.0 in /usr/local/lib/python3.11/dist-packages (from macs3<3.1,>=3.0->snapatac2->-r requirements.txt (line 13)) (2.0.1)\n",
            "Requirement already satisfied: llvmlite<0.44,>=0.43.0dev0 in /usr/local/lib/python3.11/dist-packages (from numba->decoupler==2.0.1->-r requirements.txt (line 10)) (0.43.0)\n",
            "Requirement already satisfied: tenacity>=6.2.0 in /usr/local/lib/python3.11/dist-packages (from plotly<6.0.0,>=5.19.0->snapatac2->-r requirements.txt (line 13)) (8.5.0)\n",
            "Requirement already satisfied: platformdirs>=2.5.0 in /usr/local/lib/python3.11/dist-packages (from pooch<2.0.0,>=1.6.0->snapatac2->-r requirements.txt (line 13)) (4.3.8)\n",
            "Requirement already satisfied: importlib-metadata in /usr/local/lib/python3.11/dist-packages (from pyfaidx<0.8.0,>=0.7.0->snapatac2->-r requirements.txt (line 13)) (8.7.0)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->decoupler==2.0.1->-r requirements.txt (line 10)) (3.4.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->decoupler==2.0.1->-r requirements.txt (line 10)) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->decoupler==2.0.1->-r requirements.txt (line 10)) (2.4.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->decoupler==2.0.1->-r requirements.txt (line 10)) (2025.6.15)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.11/dist-packages (from scikit-learn>=1.4->pertpy==0.11.4->-r requirements.txt (line 11)) (3.6.0)\n",
            "Requirement already satisfied: mypy-extensions>=0.3.0 in /usr/local/lib/python3.11/dist-packages (from typing_inspect>=0.6.0->pandera->-r requirements.txt (line 16)) (1.1.0)\n",
            "Requirement already satisfied: xarray>=2022.6.0 in /usr/local/lib/python3.11/dist-packages (from arviz->pertpy==0.11.4->-r requirements.txt (line 11)) (2025.3.1)\n",
            "Requirement already satisfied: h5netcdf>=1.0.2 in /usr/local/lib/python3.11/dist-packages (from arviz->pertpy==0.11.4->-r requirements.txt (line 11)) (1.6.1)\n",
            "Requirement already satisfied: xarray-einstats>=0.3 in /usr/local/lib/python3.11/dist-packages (from arviz->pertpy==0.11.4->-r requirements.txt (line 11)) (0.9.1)\n",
            "Requirement already satisfied: mpmath in /usr/local/lib/python3.11/dist-packages (from blitzgsea->pertpy==0.11.4->-r requirements.txt (line 11)) (1.3.0)\n",
            "Requirement already satisfied: choreographer>=1.0.5 in /usr/local/lib/python3.11/dist-packages (from kaleido->snapatac2->-r requirements.txt (line 13)) (1.0.9)\n",
            "Requirement already satisfied: logistro>=1.0.8 in /usr/local/lib/python3.11/dist-packages (from kaleido->snapatac2->-r requirements.txt (line 13)) (1.1.0)\n",
            "Requirement already satisfied: orjson>=3.10.15 in /usr/local/lib/python3.11/dist-packages (from kaleido->snapatac2->-r requirements.txt (line 13)) (3.10.18)\n",
            "Requirement already satisfied: legendkit in /usr/local/lib/python3.11/dist-packages (from marsilea->decoupler==2.0.1->-r requirements.txt (line 10)) (0.3.6)\n",
            "Requirement already satisfied: dill>=0.3.7 in /usr/local/lib/python3.11/dist-packages (from multiprocess->snapatac2->-r requirements.txt (line 13)) (0.3.7)\n",
            "Requirement already satisfied: et-xmlfile in /usr/local/lib/python3.11/dist-packages (from openpyxl->pertpy==0.11.4->-r requirements.txt (line 11)) (2.0.0)\n",
            "Requirement already satisfied: jax>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.5.2)\n",
            "Requirement already satisfied: jaxopt>=0.8 in /usr/local/lib/python3.11/dist-packages (from ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.8.5)\n",
            "Requirement already satisfied: lineax>=0.0.7 in /usr/local/lib/python3.11/dist-packages (from ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.0.8)\n",
            "Requirement already satisfied: optax>=0.2.4 in /usr/local/lib/python3.11/dist-packages (from ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.2.5)\n",
            "Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.11/dist-packages (from pydantic->pandera->-r requirements.txt (line 16)) (0.7.0)\n",
            "Requirement already satisfied: pydantic-core==2.33.2 in /usr/local/lib/python3.11/dist-packages (from pydantic->pandera->-r requirements.txt (line 16)) (2.33.2)\n",
            "Requirement already satisfied: typing-inspection>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from pydantic->pandera->-r requirements.txt (line 16)) (0.4.1)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.11/dist-packages (from rich->pertpy==0.11.4->-r requirements.txt (line 11)) (3.0.0)\n",
            "Requirement already satisfied: flax in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.10.6)\n",
            "Requirement already satisfied: jaxlib in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.5.1)\n",
            "Requirement already satisfied: lightning>=2.0 in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2.5.2)\n",
            "Requirement already satisfied: ml-collections>=0.1.1 in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.1.0)\n",
            "Requirement already satisfied: numpyro>=0.12.1 in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.18.0)\n",
            "Requirement already satisfied: pyro-ppl>=1.6.0 in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.9.1)\n",
            "Requirement already satisfied: sparse>=0.14.0 in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.17.0)\n",
            "Requirement already satisfied: tensorboard>=2.0 in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2.18.0)\n",
            "Requirement already satisfied: torch in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2.6.0+cu124)\n",
            "Requirement already satisfied: torchmetrics>=0.11.0 in /usr/local/lib/python3.11/dist-packages (from scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.7.4)\n",
            "Requirement already satisfied: pyomo in /usr/local/lib/python3.11/dist-packages (from sparsecca->pertpy==0.11.4->-r requirements.txt (line 11)) (6.9.2)\n",
            "Requirement already satisfied: nvidia-nccl-cu12 in /usr/local/lib/python3.11/dist-packages (from xgboost->decoupler==2.0.1->-r requirements.txt (line 10)) (2.21.5)\n",
            "Requirement already satisfied: simplejson>=3.19.3 in /usr/local/lib/python3.11/dist-packages (from choreographer>=1.0.5->kaleido->snapatac2->-r requirements.txt (line 13)) (3.20.1)\n",
            "Requirement already satisfied: ml_dtypes>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from jax>=0.4.0->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.4.1)\n",
            "Requirement already satisfied: opt_einsum in /usr/local/lib/python3.11/dist-packages (from jax>=0.4.0->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (3.4.0)\n",
            "Requirement already satisfied: parso<0.9.0,>=0.8.4 in /usr/local/lib/python3.11/dist-packages (from jedi>=0.16->ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.8.4)\n",
            "Requirement already satisfied: PyYAML<8.0,>=5.4 in /usr/local/lib/python3.11/dist-packages (from lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (6.0.2)\n",
            "Requirement already satisfied: fsspec<2027.0,>=2022.5.0 in /usr/local/lib/python3.11/dist-packages (from fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2025.3.2)\n",
            "Requirement already satisfied: lightning-utilities<2.0,>=0.10.0 in /usr/local/lib/python3.11/dist-packages (from lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.14.3)\n",
            "Requirement already satisfied: pytorch-lightning in /usr/local/lib/python3.11/dist-packages (from lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2.5.2)\n",
            "Requirement already satisfied: equinox>=0.11.10 in /usr/local/lib/python3.11/dist-packages (from lineax>=0.0.7->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.13.0)\n",
            "Requirement already satisfied: jaxtyping>=0.2.24 in /usr/local/lib/python3.11/dist-packages (from lineax>=0.0.7->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.3.2)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.11/dist-packages (from markdown-it-py>=2.2.0->rich->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.2)\n",
            "Requirement already satisfied: absl-py in /usr/local/lib/python3.11/dist-packages (from ml-collections>=0.1.1->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.4.0)\n",
            "Requirement already satisfied: multipledispatch in /usr/local/lib/python3.11/dist-packages (from numpyro>=0.12.1->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.0.0)\n",
            "Requirement already satisfied: chex>=0.1.87 in /usr/local/lib/python3.11/dist-packages (from optax>=0.2.4->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.89)\n",
            "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.11/dist-packages (from pexpect>4.3->ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.7.0)\n",
            "Requirement already satisfied: wcwidth in /usr/local/lib/python3.11/dist-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=7.23.1->ipykernel->-r requirements.txt (line 2)) (0.2.13)\n",
            "Requirement already satisfied: pyro-api>=0.1.1 in /usr/local/lib/python3.11/dist-packages (from pyro-ppl>=1.6.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.2)\n",
            "Requirement already satisfied: grpcio>=1.48.2 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.73.0)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.8.2)\n",
            "Requirement already satisfied: protobuf!=4.24.0,>=3.19.6 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (5.29.5)\n",
            "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.7.2)\n",
            "Requirement already satisfied: werkzeug>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.1.3)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.18.0)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.1.6)\n",
            "Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (12.4.127)\n",
            "Requirement already satisfied: nvidia-cuda-runtime-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (12.4.127)\n",
            "Requirement already satisfied: nvidia-cuda-cupti-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (12.4.127)\n",
            "Requirement already satisfied: nvidia-cudnn-cu12==9.1.0.70 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (9.1.0.70)\n",
            "Requirement already satisfied: nvidia-cublas-cu12==12.4.5.8 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (12.4.5.8)\n",
            "Requirement already satisfied: nvidia-cufft-cu12==11.2.1.3 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (11.2.1.3)\n",
            "Requirement already satisfied: nvidia-curand-cu12==10.3.5.147 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (10.3.5.147)\n",
            "Requirement already satisfied: nvidia-cusolver-cu12==11.6.1.9 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (11.6.1.9)\n",
            "Requirement already satisfied: nvidia-cusparse-cu12==12.3.1.170 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (12.3.1.170)\n",
            "Requirement already satisfied: nvidia-cusparselt-cu12==0.6.2 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.6.2)\n",
            "Requirement already satisfied: nvidia-nvtx-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (12.4.127)\n",
            "Requirement already satisfied: nvidia-nvjitlink-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (12.4.127)\n",
            "Requirement already satisfied: triton==3.2.0 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.2.0)\n",
            "Requirement already satisfied: sympy==1.13.1 in /usr/local/lib/python3.11/dist-packages (from torch->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.13.1)\n",
            "Requirement already satisfied: msgpack in /usr/local/lib/python3.11/dist-packages (from flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.1.1)\n",
            "Requirement already satisfied: orbax-checkpoint in /usr/local/lib/python3.11/dist-packages (from flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.11.16)\n",
            "Requirement already satisfied: tensorstore in /usr/local/lib/python3.11/dist-packages (from flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.74)\n",
            "Requirement already satisfied: treescope>=0.1.7 in /usr/local/lib/python3.11/dist-packages (from flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.9)\n",
            "Requirement already satisfied: zipp>=3.20 in /usr/local/lib/python3.11/dist-packages (from importlib-metadata->pyfaidx<0.8.0,>=0.7.0->snapatac2->-r requirements.txt (line 13)) (3.23.0)\n",
            "Requirement already satisfied: ply in /usr/local/lib/python3.11/dist-packages (from pyomo->sparsecca->pertpy==0.11.4->-r requirements.txt (line 11)) (3.11)\n",
            "Requirement already satisfied: toolz>=0.9.0 in /usr/local/lib/python3.11/dist-packages (from chex>=0.1.87->optax>=0.2.4->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.12.1)\n",
            "Requirement already satisfied: wadler-lindig>=0.1.0 in /usr/local/lib/python3.11/dist-packages (from equinox>=0.11.10->lineax>=0.0.7->ott-jax->pertpy==0.11.4->-r requirements.txt (line 11)) (0.1.7)\n",
            "Requirement already satisfied: aiohttp!=4.0.0a0,!=4.0.0a1 in /usr/local/lib/python3.11/dist-packages (from fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.11.15)\n",
            "Requirement already satisfied: MarkupSafe>=2.1.1 in /usr/local/lib/python3.11/dist-packages (from werkzeug>=1.0.1->tensorboard>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (3.0.2)\n",
            "Requirement already satisfied: etils[epath,epy] in /usr/local/lib/python3.11/dist-packages (from orbax-checkpoint->flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.12.2)\n",
            "Requirement already satisfied: humanize in /usr/local/lib/python3.11/dist-packages (from orbax-checkpoint->flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (4.12.3)\n",
            "Requirement already satisfied: aiohappyeyeballs>=2.3.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (2.6.1)\n",
            "Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.3.2)\n",
            "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (25.3.0)\n",
            "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.7.0)\n",
            "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (6.4.4)\n",
            "Requirement already satisfied: propcache>=0.2.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (0.3.2)\n",
            "Requirement already satisfied: yarl<2.0,>=1.17.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2027.0,>=2022.5.0->lightning>=2.0->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (1.20.1)\n",
            "Requirement already satisfied: importlib_resources in /usr/local/lib/python3.11/dist-packages (from etils[epath,epy]->orbax-checkpoint->flax->scvi-tools->pertpy==0.11.4->-r requirements.txt (line 11)) (6.5.2)\n",
            "Downloading pandas-2.1.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.2 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m12.2/12.2 MB\u001b[0m \u001b[31m188.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading numpy-1.26.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m18.3/18.3 MB\u001b[0m \u001b[31m167.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: numpy, pandas\n",
            "  Attempting uninstall: numpy\n",
            "    Found existing installation: numpy 2.3.1\n",
            "    Uninstalling numpy-2.3.1:\n",
            "      Successfully uninstalled numpy-2.3.1\n",
            "  Attempting uninstall: pandas\n",
            "    Found existing installation: pandas 2.3.1\n",
            "    Uninstalling pandas-2.3.1:\n",
            "      Successfully uninstalled pandas-2.3.1\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "google-colab 1.0.0 requires pandas==2.2.2, but you have pandas 2.1.1 which is incompatible.\n",
            "mizani 0.13.5 requires pandas>=2.2.0, but you have pandas 2.1.1 which is incompatible.\n",
            "thinc 8.3.6 requires numpy<3.0.0,>=2.0.0, but you have numpy 1.26.4 which is incompatible.\n",
            "plotnine 0.14.5 requires pandas>=2.2.0, but you have pandas 2.1.1 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed numpy-1.26.4 pandas-2.1.1\n"
          ]
        },
        {
          "data": {
            "application/vnd.colab-display-data+json": {
              "id": "8543e1700dba4289b1c3707a442d01dd",
              "pip_warning": {
                "packages": [
                  "numpy",
                  "pandas"
                ]
              }
            }
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "!pip install -r requirements.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "id": "OeDtRFPz-xvt",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OeDtRFPz-xvt",
        "outputId": "095235ca-9ac7-4a6e-c056-2e0a685934c3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Welcome! This command will take you through the configuration of gcloud.\n",
            "\n",
            "Your current configuration has been set to: [default]\n",
            "\n",
            "You can skip diagnostics next time by using the following flag:\n",
            "  gcloud init --skip-diagnostics\n",
            "\n",
            "Network diagnostic detects and fixes local network connection issues.\n",
            "Reachability Check passed.\n",
            "Network diagnostic passed (1/1 checks passed).\n",
            "\n",
            "Choose the account you want to use for this configuration.\n",
            "To use a federated user account, exit this command and sign in to the gcloud CLI\n",
            " with your login configuration file, then run this command again.\n",
            "\n",
            "Select an account:\n",
            " [1] zakirov@ebi.ac.uk\n",
            " [2] Sign in with a new Google Account\n",
            " [3] Skip this step\n",
            "Please enter your numeric choice:  1\n",
            "\n",
            "You are signed in as: [zakirov@ebi.ac.uk].\n",
            "\n",
            "This account has a lot of projects! Listing them all can take a while.\n",
            " [1] Enter a project ID\n",
            " [2] Create a new project\n",
            " [3] List projects\n",
            "Please enter your numeric choice:  1\n",
            "\n",
            "Enter project ID you would like to use:  prj-ext-dev-pertcat-437314\n",
            "Your current project has been set to: [prj-ext-dev-pertcat-437314].\n",
            "\n",
            "Do you want to configure a default Compute Region and Zone? (Y/n)?  Y\n",
            "\n",
            "Which Google Compute Engine zone would you like to use as project default?\n",
            "If you do not specify a zone via a command line flag while working with Compute \n",
            "Engine resources, the default is assumed.\n",
            " [1] Do not set default zone\n",
            " [2] us-east1-b\n",
            " [3] us-east1-c\n",
            " [4] us-east1-d\n",
            " [5] us-east4-c\n",
            " [6] us-east4-b\n",
            " [7] us-east4-a\n",
            " [8] us-central1-c\n",
            " [9] us-central1-a\n",
            " [10] us-central1-f\n",
            " [11] us-central1-b\n",
            " [12] us-west1-b\n",
            " [13] us-west1-c\n",
            " [14] us-west1-a\n",
            " [15] europe-west4-a\n",
            " [16] europe-west4-b\n",
            " [17] europe-west4-c\n",
            " [18] europe-west1-b\n",
            " [19] europe-west1-d\n",
            " [20] europe-west1-c\n",
            " [21] europe-west3-c\n",
            " [22] europe-west3-a\n",
            " [23] europe-west3-b\n",
            " [24] europe-west2-c\n",
            " [25] europe-west2-b\n",
            " [26] europe-west2-a\n",
            " [27] asia-east1-b\n",
            " [28] asia-east1-a\n",
            " [29] asia-east1-c\n",
            " [30] asia-southeast1-b\n",
            " [31] asia-southeast1-a\n",
            " [32] asia-southeast1-c\n",
            " [33] asia-northeast1-b\n",
            " [34] asia-northeast1-c\n",
            " [35] asia-northeast1-a\n",
            " [36] asia-south1-c\n",
            " [37] asia-south1-b\n",
            " [38] asia-south1-a\n",
            " [39] australia-southeast1-b\n",
            " [40] australia-southeast1-c\n",
            " [41] australia-southeast1-a\n",
            " [42] southamerica-east1-b\n",
            " [43] southamerica-east1-c\n",
            " [44] southamerica-east1-a\n",
            " [45] africa-south1-a\n",
            " [46] africa-south1-b\n",
            " [47] africa-south1-c\n",
            " [48] asia-east2-a\n",
            " [49] asia-east2-b\n",
            " [50] asia-east2-c\n",
            "Did not print [78] options.\n",
            "Too many options [128]. Enter \"list\" at prompt to print choices fully.\n",
            "Please enter numeric choice or text value (must exactly match list item):  26\n",
            "\n",
            "Your project default Compute Engine zone has been set to [europe-west2-a].\n",
            "You can change it by running [gcloud config set compute/zone NAME].\n",
            "\n",
            "Your project default Compute Engine region has been set to [europe-west2].\n",
            "You can change it by running [gcloud config set compute/region NAME].\n",
            "\n",
            "The Google Cloud CLI is configured and ready to use!\n",
            "\n",
            "* Commands that require authentication will use zakirov@ebi.ac.uk by default\n",
            "* Commands will reference project `prj-ext-dev-pertcat-437314` by default\n",
            "* Compute Engine commands will use region `europe-west2` by default\n",
            "* Compute Engine commands will use zone `europe-west2-a` by default\n",
            "\n",
            "Run `gcloud help config` to learn how to change individual settings\n",
            "\n",
            "This gcloud configuration is called [default]. You can create additional configurations if you work with multiple accounts and/or projects.\n",
            "Run `gcloud topic configurations` to learn more.\n",
            "\n",
            "Some things to try next:\n",
            "\n",
            "* Run `gcloud --help` to see the Cloud Platform services you can interact with. And run `gcloud help COMMAND` to get help on any gcloud command.\n",
            "* Run `gcloud topic --help` to learn about advanced features of the CLI like arg files and output formatting\n",
            "* Run `gcloud cheat-sheet` to see a roster of go-to `gcloud` commands.\n"
          ]
        }
      ],
      "source": [
        "!gcloud init"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "id": "aAczvpxw-3sR",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aAczvpxw-3sR",
        "outputId": "e681749b-17de-409c-94e5-98b4a6d52c4e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Copying gs://perturbation-catalogue-lake/perturbseq/non_curated/orion_2025_hek293t.h5ad...\n",
            "/ [0/1 files][    0.0 B/326.1 GiB]   0% Done                                    \r==> NOTE: You are downloading one or more large file(s), which would\n",
            "run significantly faster if you enabled sliced object downloads. This\n",
            "feature is enabled by default but requires that compiled crcmod be\n",
            "installed (see \"gsutil help crcmod\").\n",
            "\n",
            "-\n",
            "Operation completed over 1 objects/326.1 GiB.                                    \n"
          ]
        }
      ],
      "source": [
        "!mkdir -p Perturbseq/non_curated/h5ad/\n",
        "!gsutil -m cp gs://perturbation-catalogue-lake/perturbseq/non_curated/orion_2025_hek293t.h5ad Perturbseq/non_curated/h5ad/"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "8c9fe243",
      "metadata": {
        "id": "8c9fe243"
      },
      "source": [
        "# Import"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "fd9fdd06",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fd9fdd06",
        "outputId": "9a8b48b6-08f0-4f37-ccf1-a128ce84b650"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/usr/local/lib/python3.11/dist-packages/pandera/_pandas_deprecated.py:157: FutureWarning: Importing pandas-specific classes and functions from the\n",
            "top-level pandera module will be **removed in a future version of pandera**.\n",
            "If you're using pandera to validate pandas objects, we highly recommend updating\n",
            "your import:\n",
            "\n",
            "```\n",
            "# old import\n",
            "import pandera as pa\n",
            "\n",
            "# new import\n",
            "import pandera.pandas as pa\n",
            "```\n",
            "\n",
            "If you're using pandera to validate objects from other compatible libraries\n",
            "like pyspark or polars, see the supported libraries section of the documentation\n",
            "for more information on how to import pandera:\n",
            "\n",
            "https://pandera.readthedocs.io/en/stable/supported_libraries.html\n",
            "\n",
            "To disable this warning, set the environment variable:\n",
            "\n",
            "```\n",
            "export DISABLE_PANDERA_IMPORT_WARNING=True\n",
            "```\n",
            "\n",
            "  warnings.warn(_future_warning, FutureWarning)\n"
          ]
        }
      ],
      "source": [
        "import pandas as pd\n",
        "import json\n",
        "import sys\n",
        "\n",
        "sys.path.append(\"../../\")\n",
        "\n",
        "from curation_tools.curation_tools import CuratedDataset\n",
        "from curation_tools.perturbseq_anndata_schema import ObsSchema, VarSchema\n",
        "\n",
        "from curation_tools.unified_metadata_schema.unified_metadata_schema import Experiment"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "e1f448e7",
      "metadata": {
        "id": "e1f448e7"
      },
      "source": [
        "# Pre-process the data"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "3392c826",
      "metadata": {
        "id": "3392c826"
      },
      "source": [
        "### Load the data and construct the dummy anndata object to explore the dataset offline"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "e948b59b",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 428
        },
        "id": "e948b59b",
        "outputId": "cbcd51b0-714d-42d8-ad79-bdb968650a5e"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/tmp/ipython-input-5-3476015825.py:8: DeprecationWarning: `Series._import_from_c` is deprecated. use _import_arrow_from_c; if you are using an extension, please compile it with latest 'pyo3-polars'\n",
            "  obs = backed_data.obs[:].to_pandas()\n"
          ]
        },
        {
          "ename": "ValueError",
          "evalue": "axis 1 index 0 exceeds matrix dimension 0",
          "output_type": "error",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
            "\u001b[0;32m/tmp/ipython-input-5-3476015825.py\u001b[0m in \u001b[0;36m<cell line: 0>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0mcols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfloor\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrandom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpermutation\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mN1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0mfloat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mN1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mN2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mint\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Randomly pick N1 objects and assign to N2 categories in almost equal proportion\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[0mw\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mones\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mN1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 16\u001b[0;31m \u001b[0mconn_matrix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcsr_matrix\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mw\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mrows\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcols\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mshape\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mN1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mN2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mint\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     17\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m \u001b[0madata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mAnnData\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mconn_matrix\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mobs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mobs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvar\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mvar\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/scipy/sparse/_compressed.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, arg1, shape, dtype, copy)\u001b[0m\n\u001b[1;32m     53\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg1\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m                     \u001b[0;31m# (data, ij) format\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 55\u001b[0;31m                     \u001b[0mcoo\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_coo_container\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mshape\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     56\u001b[0m                     \u001b[0marrays\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoo\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_coo_to_compressed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_swap\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindptr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindices\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_shape\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marrays\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/scipy/sparse/_coo.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, arg1, shape, dtype, copy)\u001b[0m\n\u001b[1;32m     97\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     98\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 99\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    100\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    101\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mproperty\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.11/dist-packages/scipy/sparse/_coo.py\u001b[0m in \u001b[0;36m_check\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    203\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0midx\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcoords\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    204\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0midx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmax\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 205\u001b[0;31m                     raise ValueError(f'axis {i} index {idx.max()} exceeds '\n\u001b[0m\u001b[1;32m    206\u001b[0m                                      f'matrix dimension {self.shape[i]}')\n\u001b[1;32m    207\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0midx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mValueError\u001b[0m: axis 1 index 0 exceeds matrix dimension 0"
          ]
        }
      ],
      "source": [
        "# from scipy.sparse import csr_matrix\n",
        "# import numpy as np\n",
        "# import scanpy as sc\n",
        "# import snapatac2 as snap\n",
        "\n",
        "# backed_data = snap.read('Perturbseq/non_curated/h5ad/orion_2025_hek293t.h5ad')\n",
        "\n",
        "# obs = backed_data.obs[:].to_pandas()\n",
        "# var = backed_data.var[:].to_pandas()\n",
        "\n",
        "# N1 = obs.shape[0]  # Number of objects (e.g., cells)\n",
        "# N2 = var.shape[0]  # Number of categories (e.g., genes)\n",
        "# rows = np.arange(N1)\n",
        "# cols = (np.floor(np.random.permutation(N1)/float(N1)*N2)).astype(int) # Randomly pick N1 objects and assign to N2 categories in almost equal proportion\n",
        "# w = np.ones(N1)\n",
        "# conn_matrix = csr_matrix((w, (rows, cols)), shape=(N1, N2), dtype=int)\n",
        "\n",
        "# adata = sc.AnnData(X=conn_matrix, obs=obs, var=var)\n",
        "\n",
        "# # save the non-curated AnnData object\n",
        "# adata.write_h5ad(\"Perturbseq/non_curated/h5ad/orion_2025_hek293t_dummy.h5ad\")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "c0b2a8d6",
      "metadata": {
        "id": "c0b2a8d6"
      },
      "source": [
        "# Initialise the dataset object"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "id": "3782a1d4",
      "metadata": {
        "id": "3782a1d4"
      },
      "outputs": [],
      "source": [
        "cur_data = CuratedDataset(\n",
        "    obs_schema=ObsSchema,\n",
        "    var_schema=VarSchema,\n",
        "    exp_metadata_schema=Experiment,\n",
        "    noncurated_path = \"Perturbseq/non_curated/h5ad/orion_2025_hek293t.h5ad\"\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "3d18e4db",
      "metadata": {
        "id": "3d18e4db"
      },
      "source": [
        "# Load the dataset"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "id": "13c5917d",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "13c5917d",
        "outputId": "e7d6536d-981c-4e49-d3a8-ff1fda7a4283"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Loading data from Perturbseq/non_curated/h5ad/orion_2025_hek293t.h5ad\n",
            "Observation data:\n",
            "DataFrame shape: (4534299, 9)\n",
            "--------------------------------------------------\n",
            "                                             sample  num_features  \\\n",
            "AAACCAAAGAAGTAGC-HEK293T_Batch1      HEK293T_Batch1             2   \n",
            "AAACCAAAGAGTACGG-HEK293T_Batch1      HEK293T_Batch1             2   \n",
            "AAACCAAAGAGTGGGC-HEK293T_Batch1      HEK293T_Batch1             2   \n",
            "AAACCAAAGAGTGTAA-HEK293T_Batch1      HEK293T_Batch1             2   \n",
            "AAACCAAAGATACGGG-HEK293T_Batch1      HEK293T_Batch1             2   \n",
            "...                                             ...           ...   \n",
            "GTTGTCCGTGGGATGG-HEK293T_Batch223  HEK293T_Batch223             2   \n",
            "GTTGTCTTCAACAGCC-HEK293T_Batch223  HEK293T_Batch223             2   \n",
            "GTTGTCTTCAGTACTG-HEK293T_Batch223  HEK293T_Batch223             2   \n",
            "GTTGTCTTCCGCCGTT-HEK293T_Batch223  HEK293T_Batch223             2   \n",
            "GTTGTGCAGGCTTGCG-HEK293T_Batch223  HEK293T_Batch223             2   \n",
            "\n",
            "                                                              guide_target  \\\n",
            "AAACCAAAGAAGTAGC-HEK293T_Batch1              FAM180A_P1P2-1|FAM180A_P1P2-2   \n",
            "AAACCAAAGAGTACGG-HEK293T_Batch1    non-targeting_03016|non-targeting_03214   \n",
            "AAACCAAAGAGTGGGC-HEK293T_Batch1              PCDHGC4_P1P2-1|PCDHGC4_P1P2-2   \n",
            "AAACCAAAGAGTGTAA-HEK293T_Batch1                  USP30_P1P2-1|USP30_P1P2-2   \n",
            "AAACCAAAGATACGGG-HEK293T_Batch1                  KLRD1_P1P2-1|KLRD1_P1P2-2   \n",
            "...                                                                    ...   \n",
            "GTTGTCCGTGGGATGG-HEK293T_Batch223  non-targeting_00183|non-targeting_01885   \n",
            "GTTGTCTTCAACAGCC-HEK293T_Batch223              AKAP13_P1P2-1|AKAP13_P1P2-2   \n",
            "GTTGTCTTCAGTACTG-HEK293T_Batch223                  TTF2_P1P2-1|TTF2_P1P2-2   \n",
            "GTTGTCTTCCGCCGTT-HEK293T_Batch223                  MMP7_P1P2-1|MMP7_P1P2-2   \n",
            "GTTGTGCAGGCTTGCG-HEK293T_Batch223                  AAK1_P1P2-1|AAK1_P1P2-2   \n",
            "\n",
            "                                     gene_target  n_genes_by_counts  \\\n",
            "AAACCAAAGAAGTAGC-HEK293T_Batch1          FAM180A               6856   \n",
            "AAACCAAAGAGTACGG-HEK293T_Batch1    Non-Targeting               6636   \n",
            "AAACCAAAGAGTGGGC-HEK293T_Batch1          PCDHGC4               5353   \n",
            "AAACCAAAGAGTGTAA-HEK293T_Batch1            USP30               6358   \n",
            "AAACCAAAGATACGGG-HEK293T_Batch1            KLRD1               5527   \n",
            "...                                          ...                ...   \n",
            "GTTGTCCGTGGGATGG-HEK293T_Batch223  Non-Targeting               8112   \n",
            "GTTGTCTTCAACAGCC-HEK293T_Batch223         AKAP13              10568   \n",
            "GTTGTCTTCAGTACTG-HEK293T_Batch223           TTF2              10735   \n",
            "GTTGTCTTCCGCCGTT-HEK293T_Batch223           MMP7               9248   \n",
            "GTTGTGCAGGCTTGCG-HEK293T_Batch223           AAK1               8682   \n",
            "\n",
            "                                   total_counts  total_counts_mt  \\\n",
            "AAACCAAAGAAGTAGC-HEK293T_Batch1         24243.0            647.0   \n",
            "AAACCAAAGAGTACGG-HEK293T_Batch1         20847.0            886.0   \n",
            "AAACCAAAGAGTGGGC-HEK293T_Batch1         14035.0            610.0   \n",
            "AAACCAAAGAGTGTAA-HEK293T_Batch1         20939.0            589.0   \n",
            "AAACCAAAGATACGGG-HEK293T_Batch1         14083.0            476.0   \n",
            "...                                         ...              ...   \n",
            "GTTGTCCGTGGGATGG-HEK293T_Batch223       29296.0            522.0   \n",
            "GTTGTCTTCAACAGCC-HEK293T_Batch223       88551.0           3035.0   \n",
            "GTTGTCTTCAGTACTG-HEK293T_Batch223       81286.0           1808.0   \n",
            "GTTGTCTTCCGCCGTT-HEK293T_Batch223       40733.0           1767.0   \n",
            "GTTGTGCAGGCTTGCG-HEK293T_Batch223       36045.0            945.0   \n",
            "\n",
            "                                   pct_counts_mt  pass_guide_filter  \n",
            "AAACCAAAGAAGTAGC-HEK293T_Batch1         2.668812               True  \n",
            "AAACCAAAGAGTACGG-HEK293T_Batch1         4.250012               True  \n",
            "AAACCAAAGAGTGGGC-HEK293T_Batch1         4.346277               True  \n",
            "AAACCAAAGAGTGTAA-HEK293T_Batch1         2.812933               True  \n",
            "AAACCAAAGATACGGG-HEK293T_Batch1         3.379961               True  \n",
            "...                                          ...                ...  \n",
            "GTTGTCCGTGGGATGG-HEK293T_Batch223       1.781813               True  \n",
            "GTTGTCTTCAACAGCC-HEK293T_Batch223       3.427403               True  \n",
            "GTTGTCTTCAGTACTG-HEK293T_Batch223       2.224245               True  \n",
            "GTTGTCTTCCGCCGTT-HEK293T_Batch223       4.338006               True  \n",
            "GTTGTGCAGGCTTGCG-HEK293T_Batch223       2.621723               True  \n",
            "\n",
            "[4534299 rows x 9 columns]\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "# cur_data.download_data()\n",
        "cur_data.load_data()\n",
        "# show the data\n",
        "cur_data.show_obs()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "id": "wrrqOGLNA3p0",
      "metadata": {
        "id": "wrrqOGLNA3p0"
      },
      "outputs": [],
      "source": [
        "cur_data.adata.var['gene_symbol'] = cur_data.adata.var.index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "id": "fRi2g-hdBM7A",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fRi2g-hdBM7A",
        "outputId": "932d70f1-0a74-49a3-e5f7-a2c8d2a5a322"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Variable data:\n",
            "DataFrame shape: (38606, 1)\n",
            "--------------------------------------------------\n",
            "                     gene_symbol\n",
            "DDX11L2                  DDX11L2\n",
            "MIR1302-2HG          MIR1302-2HG\n",
            "FAM138A                  FAM138A\n",
            "ENSG00000290826  ENSG00000290826\n",
            "OR4F5                      OR4F5\n",
            "...                          ...\n",
            "ENSG00000277836  ENSG00000277836\n",
            "ENSG00000278633  ENSG00000278633\n",
            "ENSG00000276017  ENSG00000276017\n",
            "ENSG00000278817  ENSG00000278817\n",
            "ENSG00000277196  ENSG00000277196\n",
            "\n",
            "[38606 rows x 1 columns]\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.show_var()"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "44a7835d",
      "metadata": {
        "id": "44a7835d"
      },
      "source": [
        "# OBS slot curation"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "id": "7c2b269f",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 366
        },
        "id": "7c2b269f",
        "outputId": "4a7817f8-b387-4174-8cda-d323f8274908"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>sample</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>num_features</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>guide_target</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>gene_target</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>n_genes_by_counts</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>total_counts</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>total_counts_mt</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>pct_counts_mt</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>pass_guide_filter</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div><br><label><b>dtype:</b> int64</label>"
            ],
            "text/plain": [
              "sample               0\n",
              "num_features         0\n",
              "guide_target         0\n",
              "gene_target          0\n",
              "n_genes_by_counts    0\n",
              "total_counts         0\n",
              "total_counts_mt      0\n",
              "pct_counts_mt        0\n",
              "pass_guide_filter    0\n",
              "dtype: int64"
            ]
          },
          "execution_count": 14,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "cur_data.adata.obs.isna().sum()"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "57acacbc",
      "metadata": {
        "id": "57acacbc"
      },
      "source": [
        "### Extract perturbation symbols"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "a26c14a6",
      "metadata": {
        "id": "a26c14a6"
      },
      "source": [
        "### Rename `guide_target` to `perturbation_name`"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "id": "0566e196",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0566e196",
        "outputId": "622e221f-c863-46b7-f95d-6a676f5aba4d"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Renamed columns in adata.obs: {'guide_target': 'perturbation_name'}\n"
          ]
        }
      ],
      "source": [
        "cur_data.rename_columns(slot = 'obs', name_dict = {'guide_target': 'perturbation_name'})"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "b3deae0f",
      "metadata": {
        "id": "b3deae0f"
      },
      "source": [
        "### Rename `gene_target` to `perturbed_target_symbol`"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 16,
      "id": "10a4deb2",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "10a4deb2",
        "outputId": "ea01fbf9-fc7b-4bb2-a322-1d9da5fc29c6"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Renamed columns in adata.obs: {'gene_target': 'perturbed_target_symbol'}\n"
          ]
        }
      ],
      "source": [
        "cur_data.rename_columns(slot = 'obs', name_dict = {'gene_target': 'perturbed_target_symbol'})"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "b9b45777",
      "metadata": {
        "id": "b9b45777"
      },
      "source": [
        "#### Clean up `perturbed_target_symbol` column"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 17,
      "id": "c5d8cd5e",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c5d8cd5e",
        "outputId": "be4e1805-b47e-4968-b7ba-3591a4c34f61"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;30;43mStreaming output truncated to the last 5000 lines.\u001b[0m\n",
            " 'RPA4',\n",
            " 'RPAIN',\n",
            " 'RPAP1',\n",
            " 'RPAP2',\n",
            " 'RPAP3',\n",
            " 'RPE',\n",
            " 'RPE65',\n",
            " 'RPF1',\n",
            " 'RPF2',\n",
            " 'RPGR',\n",
            " 'RPGRIP1',\n",
            " 'RPGRIP1L',\n",
            " 'RPH3A',\n",
            " 'RPH3AL',\n",
            " 'RPIA',\n",
            " 'RPL10',\n",
            " 'RPL10A',\n",
            " 'RPL10L',\n",
            " 'RPL11',\n",
            " 'RPL12',\n",
            " 'RPL13',\n",
            " 'RPL13A',\n",
            " 'RPL14',\n",
            " 'RPL15',\n",
            " 'RPL17',\n",
            " 'RPL18',\n",
            " 'RPL18A',\n",
            " 'RPL19',\n",
            " 'RPL21',\n",
            " 'RPL22',\n",
            " 'RPL22L1',\n",
            " 'RPL23',\n",
            " 'RPL23A',\n",
            " 'RPL24',\n",
            " 'RPL26',\n",
            " 'RPL26L1',\n",
            " 'RPL27',\n",
            " 'RPL27A',\n",
            " 'RPL28',\n",
            " 'RPL29',\n",
            " 'RPL3',\n",
            " 'RPL30',\n",
            " 'RPL31',\n",
            " 'RPL32',\n",
            " 'RPL34',\n",
            " 'RPL35',\n",
            " 'RPL35A',\n",
            " 'RPL36',\n",
            " 'RPL36A',\n",
            " 'RPL36AL',\n",
            " 'RPL37',\n",
            " 'RPL37A',\n",
            " 'RPL38',\n",
            " 'RPL39',\n",
            " 'RPL39L',\n",
            " 'RPL3L',\n",
            " 'RPL4',\n",
            " 'RPL41',\n",
            " 'RPL5',\n",
            " 'RPL6',\n",
            " 'RPL7',\n",
            " 'RPL7A',\n",
            " 'RPL7L1',\n",
            " 'RPL8',\n",
            " 'RPL9',\n",
            " 'RPLP0',\n",
            " 'RPLP1',\n",
            " 'RPLP2',\n",
            " 'RPN1',\n",
            " 'RPN2',\n",
            " 'RPP14',\n",
            " 'RPP21',\n",
            " 'RPP25',\n",
            " 'RPP25L',\n",
            " 'RPP30',\n",
            " 'RPP40',\n",
            " 'RPRD1A',\n",
            " 'RPRD1B',\n",
            " 'RPRD2',\n",
            " 'RPRM',\n",
            " 'RPRML',\n",
            " 'RPS10',\n",
            " 'RPS11',\n",
            " 'RPS12',\n",
            " 'RPS13',\n",
            " 'RPS14',\n",
            " 'RPS15',\n",
            " 'RPS15A',\n",
            " 'RPS16',\n",
            " 'RPS17',\n",
            " 'RPS18',\n",
            " 'RPS19',\n",
            " 'RPS19BP1',\n",
            " 'RPS2',\n",
            " 'RPS20',\n",
            " 'RPS21',\n",
            " 'RPS23',\n",
            " 'RPS24',\n",
            " 'RPS25',\n",
            " 'RPS26',\n",
            " 'RPS27',\n",
            " 'RPS27A',\n",
            " 'RPS27L',\n",
            " 'RPS28',\n",
            " 'RPS29',\n",
            " 'RPS3',\n",
            " 'RPS3A',\n",
            " 'RPS4X',\n",
            " 'RPS4Y1',\n",
            " 'RPS4Y2',\n",
            " 'RPS5',\n",
            " 'RPS6',\n",
            " 'RPS6KA1',\n",
            " 'RPS6KA2',\n",
            " 'RPS6KA3',\n",
            " 'RPS6KA4',\n",
            " 'RPS6KA5',\n",
            " 'RPS6KA6',\n",
            " 'RPS6KB1',\n",
            " 'RPS6KB2',\n",
            " 'RPS6KC1',\n",
            " 'RPS6KL1',\n",
            " 'RPS7',\n",
            " 'RPS8',\n",
            " 'RPS9',\n",
            " 'RPSA',\n",
            " 'RPTN',\n",
            " 'RPTOR',\n",
            " 'RPUSD1',\n",
            " 'RPUSD2',\n",
            " 'RPUSD3',\n",
            " 'RRAD',\n",
            " 'RRAGA',\n",
            " 'RRAGB',\n",
            " 'RRAGC',\n",
            " 'RRAGD',\n",
            " 'RRAS',\n",
            " 'RRAS2',\n",
            " 'RRBP1',\n",
            " 'RREB1',\n",
            " 'RRH',\n",
            " 'RRM1',\n",
            " 'RRM2',\n",
            " 'RRM2B',\n",
            " 'RRN3',\n",
            " 'RRP1',\n",
            " 'RRP12',\n",
            " 'RRP15',\n",
            " 'RRP1B',\n",
            " 'RRP36',\n",
            " 'RRP7A',\n",
            " 'RRP8',\n",
            " 'RRP9',\n",
            " 'RRS1',\n",
            " 'RS1',\n",
            " 'RSAD1',\n",
            " 'RSAD2',\n",
            " 'RSBN1',\n",
            " 'RSBN1L',\n",
            " 'RSC1A1',\n",
            " 'RSF1',\n",
            " 'RSL1D1',\n",
            " 'RSL24D1',\n",
            " 'RSPH1',\n",
            " 'RSPH10B',\n",
            " 'RSPH10B2',\n",
            " 'RSPH14',\n",
            " 'RSPH3',\n",
            " 'RSPH4A',\n",
            " 'RSPH6A',\n",
            " 'RSPH9',\n",
            " 'RSPO1',\n",
            " 'RSPO2',\n",
            " 'RSPO3',\n",
            " 'RSPO4',\n",
            " 'RSPRY1',\n",
            " 'RSRC1',\n",
            " 'RSRC2',\n",
            " 'RSRP1',\n",
            " 'RSU1',\n",
            " 'RTBDN',\n",
            " 'RTCA',\n",
            " 'RTCB',\n",
            " 'RTEL1',\n",
            " 'RTF1',\n",
            " 'RTF2',\n",
            " 'RTKN',\n",
            " 'RTKN2',\n",
            " 'RTL1',\n",
            " 'RTL10',\n",
            " 'RTL3',\n",
            " 'RTL4',\n",
            " 'RTL5',\n",
            " 'RTL6',\n",
            " 'RTL8A',\n",
            " 'RTL8B',\n",
            " 'RTL8C',\n",
            " 'RTL9',\n",
            " 'RTN1',\n",
            " 'RTN2',\n",
            " 'RTN3',\n",
            " 'RTN4',\n",
            " 'RTN4IP1',\n",
            " 'RTN4R',\n",
            " 'RTN4RL1',\n",
            " 'RTN4RL2',\n",
            " 'RTP1',\n",
            " 'RTP2',\n",
            " 'RTP3',\n",
            " 'RTP4',\n",
            " 'RTP5',\n",
            " 'RTRAF',\n",
            " 'RTTN',\n",
            " 'RUBCN',\n",
            " 'RUBCNL',\n",
            " 'RUFY1',\n",
            " 'RUFY2',\n",
            " 'RUFY3',\n",
            " 'RUFY4',\n",
            " 'RUNDC1',\n",
            " 'RUNDC3A',\n",
            " 'RUNDC3B',\n",
            " 'RUNX1',\n",
            " 'RUNX1T1',\n",
            " 'RUNX2',\n",
            " 'RUNX3',\n",
            " 'RUSC1',\n",
            " 'RUSC1-AS1',\n",
            " 'RUSC2',\n",
            " 'RUSF1',\n",
            " 'RUVBL1',\n",
            " 'RUVBL2',\n",
            " 'RWDD1',\n",
            " 'RWDD2A',\n",
            " 'RWDD2B',\n",
            " 'RWDD3',\n",
            " 'RWDD4',\n",
            " 'RXFP1',\n",
            " 'RXFP2',\n",
            " 'RXFP3',\n",
            " 'RXFP4',\n",
            " 'RXRA',\n",
            " 'RXRB',\n",
            " 'RXRG',\n",
            " 'RXYLT1',\n",
            " 'RYBP',\n",
            " 'RYK',\n",
            " 'RYR1',\n",
            " 'RYR2',\n",
            " 'RYR3',\n",
            " 'S100A1',\n",
            " 'S100A10',\n",
            " 'S100A11',\n",
            " 'S100A12',\n",
            " 'S100A13',\n",
            " 'S100A14',\n",
            " 'S100A16',\n",
            " 'S100A2',\n",
            " 'S100A3',\n",
            " 'S100A4',\n",
            " 'S100A5',\n",
            " 'S100A6',\n",
            " 'S100A7',\n",
            " 'S100A7A',\n",
            " 'S100A8',\n",
            " 'S100A9',\n",
            " 'S100B',\n",
            " 'S100G',\n",
            " 'S100P',\n",
            " 'S100PBP',\n",
            " 'S100Z',\n",
            " 'S1PR1',\n",
            " 'S1PR2',\n",
            " 'S1PR3',\n",
            " 'S1PR4',\n",
            " 'S1PR5',\n",
            " 'SAA1',\n",
            " 'SAA2',\n",
            " 'SAA4',\n",
            " 'SAAL1',\n",
            " 'SAC3D1',\n",
            " 'SACM1L',\n",
            " 'SACS',\n",
            " 'SAE1',\n",
            " 'SAFB',\n",
            " 'SAFB2',\n",
            " 'SAG',\n",
            " 'SAGE1',\n",
            " 'SALL1',\n",
            " 'SALL2',\n",
            " 'SALL3',\n",
            " 'SALL4',\n",
            " 'SAMD1',\n",
            " 'SAMD10',\n",
            " 'SAMD11',\n",
            " 'SAMD12',\n",
            " 'SAMD13',\n",
            " 'SAMD14',\n",
            " 'SAMD15',\n",
            " 'SAMD3',\n",
            " 'SAMD4A',\n",
            " 'SAMD4B',\n",
            " 'SAMD5',\n",
            " 'SAMD7',\n",
            " 'SAMD8',\n",
            " 'SAMD9',\n",
            " 'SAMD9L',\n",
            " 'SAMHD1',\n",
            " 'SAMM50',\n",
            " 'SAMSN1',\n",
            " 'SANBR',\n",
            " 'SAP130',\n",
            " 'SAP18',\n",
            " 'SAP25',\n",
            " 'SAP30',\n",
            " 'SAP30BP',\n",
            " 'SAP30L',\n",
            " 'SAPCD1',\n",
            " 'SAPCD2',\n",
            " 'SAR1A',\n",
            " 'SAR1B',\n",
            " 'SARAF',\n",
            " 'SARDH',\n",
            " 'SARM1',\n",
            " 'SARNP',\n",
            " 'SARS1',\n",
            " 'SARS2',\n",
            " 'SART1',\n",
            " 'SART3',\n",
            " 'SASH1',\n",
            " 'SASH3',\n",
            " 'SASS6',\n",
            " 'SAT1',\n",
            " 'SAT2',\n",
            " 'SATB1',\n",
            " 'SATB2',\n",
            " 'SATL1',\n",
            " 'SAV1',\n",
            " 'SAXO1',\n",
            " 'SAXO2',\n",
            " 'SAYSD1',\n",
            " 'SBDS',\n",
            " 'SBF1',\n",
            " 'SBF2',\n",
            " 'SBK1',\n",
            " 'SBK2',\n",
            " 'SBNO1',\n",
            " 'SBNO2',\n",
            " 'SBSN',\n",
            " 'SBSPON',\n",
            " 'SC5D',\n",
            " 'SCAF1',\n",
            " 'SCAF11',\n",
            " 'SCAF4',\n",
            " 'SCAF8',\n",
            " 'SCAI',\n",
            " 'SCAMP1',\n",
            " 'SCAMP2',\n",
            " 'SCAMP3',\n",
            " 'SCAMP4',\n",
            " 'SCAMP5',\n",
            " 'SCAND1',\n",
            " 'SCAND3',\n",
            " 'SCAP',\n",
            " 'SCAPER',\n",
            " 'SCARA3',\n",
            " 'SCARA5',\n",
            " 'SCARB1',\n",
            " 'SCARB2',\n",
            " 'SCARF1',\n",
            " 'SCARF2',\n",
            " 'SCCPDH',\n",
            " 'SCD',\n",
            " 'SCD5',\n",
            " 'SCEL',\n",
            " 'SCFD1',\n",
            " 'SCFD2',\n",
            " 'SCG2',\n",
            " 'SCG3',\n",
            " 'SCG5',\n",
            " 'SCGB1A1',\n",
            " 'SCGB1C1',\n",
            " 'SCGB1D1',\n",
            " 'SCGB1D2',\n",
            " 'SCGB1D4',\n",
            " 'SCGB2A1',\n",
            " 'SCGB2A2',\n",
            " 'SCGB2B2',\n",
            " 'SCGB3A1',\n",
            " 'SCGB3A2',\n",
            " 'SCGN',\n",
            " 'SCHIP1',\n",
            " 'SCIMP',\n",
            " 'SCIN',\n",
            " 'SCLT1',\n",
            " 'SCLY',\n",
            " 'SCMH1',\n",
            " 'SCML1',\n",
            " 'SCML2',\n",
            " 'SCML4',\n",
            " 'SCN10A',\n",
            " 'SCN11A',\n",
            " 'SCN1A',\n",
            " 'SCN1B',\n",
            " 'SCN2A',\n",
            " 'SCN2B',\n",
            " 'SCN3A',\n",
            " 'SCN3B',\n",
            " 'SCN4A',\n",
            " 'SCN4B',\n",
            " 'SCN5A',\n",
            " 'SCN7A',\n",
            " 'SCN8A',\n",
            " 'SCN9A',\n",
            " 'SCNM1',\n",
            " 'SCNN1A',\n",
            " 'SCNN1B',\n",
            " 'SCNN1D',\n",
            " 'SCNN1G',\n",
            " 'SCO1',\n",
            " 'SCO2',\n",
            " 'SCOC',\n",
            " 'SCP2',\n",
            " 'SCP2D1',\n",
            " 'SCP2D1-AS1',\n",
            " 'SCPEP1',\n",
            " 'SCRG1',\n",
            " 'SCRIB',\n",
            " 'SCRN1',\n",
            " 'SCRN2',\n",
            " 'SCRN3',\n",
            " 'SCRT1',\n",
            " 'SCRT2',\n",
            " 'SCT',\n",
            " 'SCTR',\n",
            " 'SCUBE1',\n",
            " 'SCUBE2',\n",
            " 'SCUBE3',\n",
            " 'SCX',\n",
            " 'SCYL1',\n",
            " 'SCYL2',\n",
            " 'SCYL3',\n",
            " 'SDAD1',\n",
            " 'SDC1',\n",
            " 'SDC2',\n",
            " 'SDC3',\n",
            " 'SDC4',\n",
            " 'SDCBP',\n",
            " 'SDCBP2',\n",
            " 'SDCCAG8',\n",
            " 'SDE2',\n",
            " 'SDF2',\n",
            " 'SDF2L1',\n",
            " 'SDF4',\n",
            " 'SDHA',\n",
            " 'SDHAF1',\n",
            " 'SDHAF2',\n",
            " 'SDHAF3',\n",
            " 'SDHAF4',\n",
            " 'SDHB',\n",
            " 'SDHC',\n",
            " 'SDHD',\n",
            " 'SDK1',\n",
            " 'SDK2',\n",
            " 'SDR16C5',\n",
            " 'SDR39U1',\n",
            " 'SDR42E1',\n",
            " 'SDR42E2',\n",
            " 'SDR9C7',\n",
            " 'SDS',\n",
            " 'SDSL',\n",
            " 'SEC11A',\n",
            " 'SEC11C',\n",
            " 'SEC13',\n",
            " 'SEC14L1',\n",
            " 'SEC14L2',\n",
            " 'SEC14L3',\n",
            " 'SEC14L4',\n",
            " 'SEC14L5',\n",
            " 'SEC14L6',\n",
            " 'SEC16A',\n",
            " 'SEC16B',\n",
            " 'SEC22A',\n",
            " 'SEC22B',\n",
            " 'SEC22C',\n",
            " 'SEC23A',\n",
            " 'SEC23B',\n",
            " 'SEC23IP',\n",
            " 'SEC24A',\n",
            " 'SEC24B',\n",
            " 'SEC24C',\n",
            " 'SEC24D',\n",
            " 'SEC31A',\n",
            " 'SEC31B',\n",
            " 'SEC61A1',\n",
            " 'SEC61A2',\n",
            " 'SEC61B',\n",
            " 'SEC61G',\n",
            " 'SEC62',\n",
            " 'SEC63',\n",
            " 'SECISBP2',\n",
            " 'SECISBP2L',\n",
            " 'SECTM1',\n",
            " 'SEH1L',\n",
            " 'SEL1L',\n",
            " 'SEL1L2',\n",
            " 'SEL1L3',\n",
            " 'SELE',\n",
            " 'SELENBP1',\n",
            " 'SELENOF',\n",
            " 'SELENOH',\n",
            " 'SELENOI',\n",
            " 'SELENOK',\n",
            " 'SELENOM',\n",
            " 'SELENON',\n",
            " 'SELENOO',\n",
            " 'SELENOP',\n",
            " 'SELENOS',\n",
            " 'SELENOT',\n",
            " 'SELENOV',\n",
            " 'SELENOW',\n",
            " 'SELL',\n",
            " 'SELP',\n",
            " 'SELPLG',\n",
            " 'SEM1',\n",
            " 'SEMA3A',\n",
            " 'SEMA3B',\n",
            " 'SEMA3C',\n",
            " 'SEMA3D',\n",
            " 'SEMA3E',\n",
            " 'SEMA3F',\n",
            " 'SEMA3G',\n",
            " 'SEMA4A',\n",
            " 'SEMA4B',\n",
            " 'SEMA4C',\n",
            " 'SEMA4D',\n",
            " 'SEMA4F',\n",
            " 'SEMA4G',\n",
            " 'SEMA5A',\n",
            " 'SEMA5B',\n",
            " 'SEMA6A',\n",
            " 'SEMA6B',\n",
            " 'SEMA6C',\n",
            " 'SEMA6D',\n",
            " 'SEMA7A',\n",
            " 'SENP1',\n",
            " 'SENP2',\n",
            " 'SENP3',\n",
            " 'SENP5',\n",
            " 'SENP6',\n",
            " 'SENP7',\n",
            " 'SENP8',\n",
            " 'SEPHS1',\n",
            " 'SEPHS2',\n",
            " 'SEPSECS',\n",
            " 'SEPTIN1',\n",
            " 'SEPTIN10',\n",
            " 'SEPTIN11',\n",
            " 'SEPTIN12',\n",
            " 'SEPTIN14',\n",
            " 'SEPTIN2',\n",
            " 'SEPTIN3',\n",
            " 'SEPTIN4',\n",
            " 'SEPTIN5',\n",
            " 'SEPTIN6',\n",
            " 'SEPTIN7',\n",
            " 'SEPTIN8',\n",
            " 'SEPTIN9',\n",
            " 'SERAC1',\n",
            " 'SERBP1',\n",
            " 'SERF2',\n",
            " 'SERGEF',\n",
            " 'SERHL2',\n",
            " 'SERINC1',\n",
            " 'SERINC2',\n",
            " 'SERINC3',\n",
            " 'SERINC4',\n",
            " 'SERINC5',\n",
            " 'SERP1',\n",
            " 'SERP2',\n",
            " 'SERPINA1',\n",
            " 'SERPINA10',\n",
            " 'SERPINA11',\n",
            " 'SERPINA12',\n",
            " 'SERPINA3',\n",
            " 'SERPINA4',\n",
            " 'SERPINA5',\n",
            " 'SERPINA6',\n",
            " 'SERPINA7',\n",
            " 'SERPINA9',\n",
            " 'SERPINB1',\n",
            " 'SERPINB10',\n",
            " 'SERPINB11',\n",
            " 'SERPINB12',\n",
            " 'SERPINB13',\n",
            " 'SERPINB2',\n",
            " 'SERPINB3',\n",
            " 'SERPINB4',\n",
            " 'SERPINB5',\n",
            " 'SERPINB6',\n",
            " 'SERPINB7',\n",
            " 'SERPINB8',\n",
            " 'SERPINB9',\n",
            " 'SERPINC1',\n",
            " 'SERPIND1',\n",
            " 'SERPINE1',\n",
            " 'SERPINE2',\n",
            " 'SERPINE3',\n",
            " 'SERPINF1',\n",
            " 'SERPINF2',\n",
            " 'SERPING1',\n",
            " 'SERPINH1',\n",
            " 'SERPINI1',\n",
            " 'SERPINI2',\n",
            " 'SERTAD1',\n",
            " 'SERTAD2',\n",
            " 'SERTAD3',\n",
            " 'SERTAD4',\n",
            " 'SERTM1',\n",
            " 'SESN1',\n",
            " 'SESN2',\n",
            " 'SESN3',\n",
            " 'SESTD1',\n",
            " 'SET',\n",
            " 'SETBP1',\n",
            " 'SETD1A',\n",
            " 'SETD1B',\n",
            " 'SETD2',\n",
            " 'SETD3',\n",
            " 'SETD4',\n",
            " 'SETD5',\n",
            " 'SETD6',\n",
            " 'SETD7',\n",
            " 'SETD9',\n",
            " 'SETDB1',\n",
            " 'SETDB2',\n",
            " 'SETMAR',\n",
            " 'SETX',\n",
            " 'SEZ6',\n",
            " 'SEZ6L',\n",
            " 'SEZ6L2',\n",
            " 'SF1',\n",
            " 'SF3A1',\n",
            " 'SF3A2',\n",
            " 'SF3A3',\n",
            " 'SF3B1',\n",
            " 'SF3B2',\n",
            " 'SF3B3',\n",
            " 'SF3B4',\n",
            " 'SF3B5',\n",
            " 'SFI1',\n",
            " 'SFMBT1',\n",
            " 'SFMBT2',\n",
            " 'SFN',\n",
            " 'SFPQ',\n",
            " 'SFR1',\n",
            " 'SFRP1',\n",
            " 'SFRP2',\n",
            " 'SFRP4',\n",
            " 'SFRP5',\n",
            " 'SFSWAP',\n",
            " 'SFT2D1',\n",
            " 'SFT2D2',\n",
            " 'SFT2D3',\n",
            " 'SFTA2',\n",
            " 'SFTA3',\n",
            " 'SFTPA1',\n",
            " 'SFTPA2',\n",
            " 'SFTPB',\n",
            " 'SFTPC',\n",
            " 'SFTPD',\n",
            " 'SFXN1',\n",
            " 'SFXN2',\n",
            " 'SFXN3',\n",
            " 'SFXN4',\n",
            " 'SFXN5',\n",
            " 'SGCA',\n",
            " 'SGCB',\n",
            " 'SGCE',\n",
            " 'SGCG',\n",
            " 'SGCZ',\n",
            " 'SGF29',\n",
            " 'SGIP1',\n",
            " 'SGK1',\n",
            " 'SGK2',\n",
            " 'SGK3',\n",
            " 'SGMS1',\n",
            " 'SGMS2',\n",
            " 'SGO1',\n",
            " 'SGO2',\n",
            " 'SGPL1',\n",
            " 'SGPP1',\n",
            " 'SGPP2',\n",
            " 'SGSH',\n",
            " 'SGSM1',\n",
            " 'SGSM2',\n",
            " 'SGSM3',\n",
            " 'SGTA',\n",
            " 'SGTB',\n",
            " 'SH2B1',\n",
            " 'SH2B2',\n",
            " 'SH2B3',\n",
            " 'SH2D1A',\n",
            " 'SH2D1B',\n",
            " 'SH2D2A',\n",
            " 'SH2D3A',\n",
            " 'SH2D3C',\n",
            " 'SH2D4A',\n",
            " 'SH2D4B',\n",
            " 'SH2D5',\n",
            " 'SH2D6',\n",
            " 'SH2D7',\n",
            " 'SH3BGR',\n",
            " 'SH3BGRL',\n",
            " 'SH3BGRL2',\n",
            " 'SH3BGRL3',\n",
            " 'SH3BP1',\n",
            " 'SH3BP2',\n",
            " 'SH3BP4',\n",
            " 'SH3BP5',\n",
            " 'SH3BP5L',\n",
            " 'SH3D19',\n",
            " 'SH3GL1',\n",
            " 'SH3GL2',\n",
            " 'SH3GL3',\n",
            " 'SH3GLB1',\n",
            " 'SH3GLB2',\n",
            " 'SH3KBP1',\n",
            " 'SH3PXD2A',\n",
            " 'SH3PXD2B',\n",
            " 'SH3RF1',\n",
            " 'SH3RF2',\n",
            " 'SH3RF3',\n",
            " 'SH3TC1',\n",
            " 'SH3TC2',\n",
            " 'SH3YL1',\n",
            " 'SHANK1',\n",
            " 'SHANK2',\n",
            " 'SHANK3',\n",
            " 'SHARPIN',\n",
            " 'SHB',\n",
            " 'SHBG',\n",
            " 'SHC1',\n",
            " 'SHC2',\n",
            " 'SHC3',\n",
            " 'SHC4',\n",
            " 'SHCBP1',\n",
            " 'SHCBP1L',\n",
            " 'SHD',\n",
            " 'SHF',\n",
            " 'SHFL',\n",
            " 'SHH',\n",
            " 'SHISA2',\n",
            " 'SHISA3',\n",
            " 'SHISA4',\n",
            " 'SHISA5',\n",
            " 'SHISA6',\n",
            " 'SHISA7',\n",
            " 'SHISA8',\n",
            " 'SHISA9',\n",
            " 'SHISAL1',\n",
            " 'SHISAL2A',\n",
            " 'SHISAL2B',\n",
            " 'SHKBP1',\n",
            " 'SHLD1',\n",
            " 'SHLD2',\n",
            " 'SHMT1',\n",
            " 'SHMT2',\n",
            " 'SHOC1',\n",
            " 'SHOC2',\n",
            " 'SHOX',\n",
            " 'SHOX2',\n",
            " 'SHPK',\n",
            " 'SHPRH',\n",
            " 'SHQ1',\n",
            " 'SHROOM1',\n",
            " 'SHROOM2',\n",
            " 'SHROOM3',\n",
            " 'SHROOM4',\n",
            " 'SHTN1',\n",
            " 'SI',\n",
            " 'SIAE',\n",
            " 'SIAH1',\n",
            " 'SIAH2',\n",
            " 'SIAH3',\n",
            " 'SIDT1',\n",
            " 'SIDT2',\n",
            " 'SIGIRR',\n",
            " 'SIGLEC1',\n",
            " 'SIGLEC10',\n",
            " 'SIGLEC11',\n",
            " 'SIGLEC12',\n",
            " 'SIGLEC14',\n",
            " 'SIGLEC15',\n",
            " 'SIGLEC5',\n",
            " 'SIGLEC6',\n",
            " 'SIGLEC7',\n",
            " 'SIGLEC8',\n",
            " 'SIGLEC9',\n",
            " 'SIGLECL1',\n",
            " 'SIGMAR1',\n",
            " 'SIK1',\n",
            " 'SIK2',\n",
            " 'SIK3',\n",
            " 'SIKE1',\n",
            " 'SIL1',\n",
            " 'SIM1',\n",
            " 'SIM2',\n",
            " 'SIMC1',\n",
            " 'SIN3A',\n",
            " 'SIN3B',\n",
            " 'SINHCAF',\n",
            " 'SIPA1',\n",
            " 'SIPA1L1',\n",
            " 'SIPA1L2',\n",
            " 'SIPA1L3',\n",
            " 'SIRPA',\n",
            " 'SIRPB1',\n",
            " 'SIRPB2',\n",
            " 'SIRPD',\n",
            " 'SIRPG',\n",
            " 'SIRT1',\n",
            " 'SIRT2',\n",
            " 'SIRT3',\n",
            " 'SIRT4',\n",
            " 'SIRT5',\n",
            " 'SIRT6',\n",
            " 'SIRT7',\n",
            " 'SIT1',\n",
            " 'SIVA1',\n",
            " 'SIX1',\n",
            " 'SIX2',\n",
            " 'SIX3',\n",
            " 'SIX4',\n",
            " 'SIX5',\n",
            " 'SIX6',\n",
            " 'SKA1',\n",
            " 'SKA2',\n",
            " 'SKA3',\n",
            " 'SKAP1',\n",
            " 'SKAP2',\n",
            " 'SKI',\n",
            " 'SKIC2',\n",
            " 'SKIC3',\n",
            " 'SKIC8',\n",
            " 'SKIDA1',\n",
            " 'SKIL',\n",
            " 'SKOR1',\n",
            " 'SKOR2',\n",
            " 'SKP1',\n",
            " 'SKP2',\n",
            " 'SLA',\n",
            " 'SLA2',\n",
            " 'SLAIN1',\n",
            " 'SLAIN2',\n",
            " 'SLAMF1',\n",
            " 'SLAMF6',\n",
            " 'SLAMF7',\n",
            " 'SLAMF8',\n",
            " 'SLAMF9',\n",
            " 'SLBP',\n",
            " 'SLC10A1',\n",
            " 'SLC10A2',\n",
            " 'SLC10A3',\n",
            " 'SLC10A4',\n",
            " 'SLC10A5',\n",
            " 'SLC10A6',\n",
            " 'SLC10A7',\n",
            " 'SLC11A1',\n",
            " 'SLC11A2',\n",
            " 'SLC12A1',\n",
            " 'SLC12A2',\n",
            " 'SLC12A3',\n",
            " 'SLC12A4',\n",
            " 'SLC12A5',\n",
            " 'SLC12A6',\n",
            " 'SLC12A7',\n",
            " 'SLC12A8',\n",
            " 'SLC12A9',\n",
            " 'SLC13A1',\n",
            " 'SLC13A2',\n",
            " 'SLC13A3',\n",
            " 'SLC13A4',\n",
            " 'SLC13A5',\n",
            " 'SLC14A1',\n",
            " 'SLC14A2',\n",
            " 'SLC15A1',\n",
            " 'SLC15A2',\n",
            " 'SLC15A3',\n",
            " 'SLC15A4',\n",
            " 'SLC15A5',\n",
            " 'SLC16A1',\n",
            " 'SLC16A10',\n",
            " 'SLC16A11',\n",
            " 'SLC16A12',\n",
            " 'SLC16A13',\n",
            " 'SLC16A14',\n",
            " 'SLC16A2',\n",
            " 'SLC16A3',\n",
            " 'SLC16A4',\n",
            " 'SLC16A5',\n",
            " 'SLC16A6',\n",
            " 'SLC16A7',\n",
            " 'SLC16A8',\n",
            " 'SLC16A9',\n",
            " 'SLC17A1',\n",
            " 'SLC17A2',\n",
            " 'SLC17A3',\n",
            " 'SLC17A4',\n",
            " 'SLC17A5',\n",
            " 'SLC17A6',\n",
            " 'SLC17A7',\n",
            " 'SLC17A8',\n",
            " 'SLC17A9',\n",
            " 'SLC18A1',\n",
            " 'SLC18A2',\n",
            " 'SLC18A3',\n",
            " 'SLC18B1',\n",
            " 'SLC19A1',\n",
            " 'SLC19A2',\n",
            " 'SLC19A3',\n",
            " 'SLC1A1',\n",
            " 'SLC1A2',\n",
            " 'SLC1A3',\n",
            " 'SLC1A4',\n",
            " 'SLC1A5',\n",
            " 'SLC1A6',\n",
            " 'SLC1A7',\n",
            " 'SLC20A1',\n",
            " 'SLC22A1',\n",
            " 'SLC22A10',\n",
            " 'SLC22A11',\n",
            " 'SLC22A12',\n",
            " 'SLC22A13',\n",
            " 'SLC22A14',\n",
            " 'SLC22A15',\n",
            " 'SLC22A16',\n",
            " 'SLC22A17',\n",
            " 'SLC22A18',\n",
            " 'SLC22A18AS',\n",
            " 'SLC22A2',\n",
            " 'SLC22A23',\n",
            " 'SLC22A24',\n",
            " 'SLC22A25',\n",
            " 'SLC22A3',\n",
            " 'SLC22A31',\n",
            " 'SLC22A4',\n",
            " 'SLC22A5',\n",
            " 'SLC22A6',\n",
            " 'SLC22A7',\n",
            " 'SLC22A8',\n",
            " 'SLC22A9',\n",
            " 'SLC23A1',\n",
            " 'SLC23A2',\n",
            " 'SLC23A3',\n",
            " 'SLC24A1',\n",
            " 'SLC24A2',\n",
            " 'SLC24A3',\n",
            " 'SLC24A4',\n",
            " 'SLC24A5',\n",
            " 'SLC25A1',\n",
            " 'SLC25A11',\n",
            " 'SLC25A12',\n",
            " 'SLC25A13',\n",
            " 'SLC25A14',\n",
            " 'SLC25A15',\n",
            " 'SLC25A16',\n",
            " 'SLC25A17',\n",
            " 'SLC25A18',\n",
            " 'SLC25A19',\n",
            " 'SLC25A2',\n",
            " 'SLC25A20',\n",
            " 'SLC25A21',\n",
            " 'SLC25A22',\n",
            " 'SLC25A23',\n",
            " 'SLC25A24',\n",
            " 'SLC25A25',\n",
            " 'SLC25A26',\n",
            " 'SLC25A27',\n",
            " 'SLC25A28',\n",
            " 'SLC25A29',\n",
            " 'SLC25A3',\n",
            " 'SLC25A30',\n",
            " 'SLC25A31',\n",
            " 'SLC25A32',\n",
            " 'SLC25A33',\n",
            " 'SLC25A34',\n",
            " 'SLC25A35',\n",
            " 'SLC25A36',\n",
            " 'SLC25A37',\n",
            " 'SLC25A38',\n",
            " 'SLC25A39',\n",
            " 'SLC25A4',\n",
            " 'SLC25A40',\n",
            " 'SLC25A41',\n",
            " 'SLC25A42',\n",
            " 'SLC25A43',\n",
            " 'SLC25A44',\n",
            " 'SLC25A45',\n",
            " 'SLC25A46',\n",
            " 'SLC25A47',\n",
            " 'SLC25A48',\n",
            " 'SLC25A5',\n",
            " 'SLC25A51',\n",
            " 'SLC25A52',\n",
            " 'SLC25A53',\n",
            " 'SLC25A6',\n",
            " 'SLC26A1',\n",
            " 'SLC26A11',\n",
            " 'SLC26A2',\n",
            " 'SLC26A3',\n",
            " 'SLC26A4',\n",
            " 'SLC26A5',\n",
            " 'SLC26A6',\n",
            " 'SLC26A7',\n",
            " 'SLC26A8',\n",
            " 'SLC26A9',\n",
            " 'SLC27A1',\n",
            " 'SLC27A2',\n",
            " 'SLC27A3',\n",
            " 'SLC27A4',\n",
            " 'SLC27A5',\n",
            " 'SLC27A6',\n",
            " 'SLC28A1',\n",
            " 'SLC28A2',\n",
            " 'SLC28A3',\n",
            " 'SLC29A1',\n",
            " 'SLC29A2',\n",
            " 'SLC29A3',\n",
            " 'SLC29A4',\n",
            " 'SLC2A1',\n",
            " 'SLC2A10',\n",
            " 'SLC2A11',\n",
            " 'SLC2A12',\n",
            " 'SLC2A13',\n",
            " 'SLC2A14',\n",
            " 'SLC2A2',\n",
            " 'SLC2A3',\n",
            " 'SLC2A4',\n",
            " 'SLC2A4RG',\n",
            " 'SLC2A5',\n",
            " 'SLC2A6',\n",
            " 'SLC2A7',\n",
            " 'SLC2A8',\n",
            " 'SLC2A9',\n",
            " 'SLC30A1',\n",
            " 'SLC30A10',\n",
            " 'SLC30A2',\n",
            " 'SLC30A3',\n",
            " 'SLC30A4',\n",
            " 'SLC30A5',\n",
            " 'SLC30A6',\n",
            " 'SLC30A7',\n",
            " 'SLC30A8',\n",
            " 'SLC30A9',\n",
            " 'SLC31A1',\n",
            " 'SLC31A2',\n",
            " 'SLC32A1',\n",
            " 'SLC33A1',\n",
            " 'SLC34A1',\n",
            " 'SLC34A2',\n",
            " 'SLC34A3',\n",
            " 'SLC35A1',\n",
            " 'SLC35A2',\n",
            " 'SLC35A3',\n",
            " 'SLC35A4',\n",
            " 'SLC35A5',\n",
            " 'SLC35B1',\n",
            " 'SLC35B2',\n",
            " 'SLC35B3',\n",
            " 'SLC35B4',\n",
            " 'SLC35C1',\n",
            " 'SLC35C2',\n",
            " 'SLC35D1',\n",
            " 'SLC35D2',\n",
            " 'SLC35D3',\n",
            " 'SLC35E1',\n",
            " 'SLC35E2B',\n",
            " 'SLC35E3',\n",
            " 'SLC35E4',\n",
            " 'SLC35F1',\n",
            " 'SLC35F2',\n",
            " 'SLC35F3',\n",
            " 'SLC35F4',\n",
            " 'SLC35F5',\n",
            " 'SLC35G1',\n",
            " 'SLC35G2',\n",
            " 'SLC35G3',\n",
            " 'SLC35G4',\n",
            " 'SLC35G5',\n",
            " 'SLC35G6',\n",
            " 'SLC36A1',\n",
            " 'SLC36A2',\n",
            " 'SLC36A3',\n",
            " 'SLC36A4',\n",
            " 'SLC37A1',\n",
            " 'SLC37A2',\n",
            " 'SLC37A3',\n",
            " 'SLC37A4',\n",
            " 'SLC38A1',\n",
            " 'SLC38A10',\n",
            " 'SLC38A11',\n",
            " 'SLC38A2',\n",
            " 'SLC38A3',\n",
            " 'SLC38A4',\n",
            " 'SLC38A5',\n",
            " 'SLC38A6',\n",
            " 'SLC38A7',\n",
            " 'SLC38A8',\n",
            " 'SLC38A9',\n",
            " 'SLC39A1',\n",
            " 'SLC39A10',\n",
            " 'SLC39A11',\n",
            " 'SLC39A12',\n",
            " 'SLC39A13',\n",
            " 'SLC39A14',\n",
            " 'SLC39A2',\n",
            " 'SLC39A3',\n",
            " 'SLC39A4',\n",
            " 'SLC39A5',\n",
            " 'SLC39A6',\n",
            " 'SLC39A7',\n",
            " 'SLC39A8',\n",
            " 'SLC39A9',\n",
            " 'SLC3A1',\n",
            " 'SLC3A2',\n",
            " 'SLC40A1',\n",
            " 'SLC41A1',\n",
            " 'SLC41A2',\n",
            " 'SLC41A3',\n",
            " 'SLC43A1',\n",
            " 'SLC43A2',\n",
            " 'SLC43A3',\n",
            " 'SLC44A1',\n",
            " 'SLC44A2',\n",
            " 'SLC44A3',\n",
            " 'SLC44A4',\n",
            " 'SLC44A5',\n",
            " 'SLC45A1',\n",
            " 'SLC45A2',\n",
            " 'SLC45A3',\n",
            " 'SLC45A4',\n",
            " 'SLC46A1',\n",
            " 'SLC46A2',\n",
            " 'SLC46A3',\n",
            " 'SLC47A1',\n",
            " 'SLC47A2',\n",
            " 'SLC48A1',\n",
            " 'SLC49A3',\n",
            " 'SLC49A4',\n",
            " 'SLC4A1',\n",
            " 'SLC4A10',\n",
            " 'SLC4A11',\n",
            " 'SLC4A1AP',\n",
            " 'SLC4A2',\n",
            " 'SLC4A3',\n",
            " 'SLC4A4',\n",
            " 'SLC4A5',\n",
            " 'SLC4A7',\n",
            " 'SLC4A8',\n",
            " 'SLC4A9',\n",
            " 'SLC50A1',\n",
            " 'SLC51A',\n",
            " 'SLC51B',\n",
            " 'SLC52A1',\n",
            " 'SLC52A2',\n",
            " 'SLC52A3',\n",
            " 'SLC5A1',\n",
            " 'SLC5A10',\n",
            " 'SLC5A11',\n",
            " 'SLC5A12',\n",
            " 'SLC5A2',\n",
            " 'SLC5A4',\n",
            " 'SLC5A5',\n",
            " 'SLC5A6',\n",
            " 'SLC5A7',\n",
            " 'SLC5A8',\n",
            " 'SLC5A9',\n",
            " 'SLC66A1',\n",
            " 'SLC66A2',\n",
            " 'SLC66A3',\n",
            " 'SLC6A1',\n",
            " 'SLC6A11',\n",
            " 'SLC6A12',\n",
            " 'SLC6A13',\n",
            " 'SLC6A14',\n",
            " 'SLC6A15',\n",
            " 'SLC6A16',\n",
            " 'SLC6A17',\n",
            " 'SLC6A18',\n",
            " 'SLC6A19',\n",
            " 'SLC6A2',\n",
            " 'SLC6A20',\n",
            " 'SLC6A3',\n",
            " 'SLC6A4',\n",
            " 'SLC6A5',\n",
            " 'SLC6A6',\n",
            " 'SLC6A7',\n",
            " 'SLC6A8',\n",
            " 'SLC6A9',\n",
            " 'SLC7A1',\n",
            " 'SLC7A10',\n",
            " 'SLC7A11',\n",
            " 'SLC7A13',\n",
            " 'SLC7A14',\n",
            " 'SLC7A2',\n",
            " 'SLC7A3',\n",
            " 'SLC7A4',\n",
            " 'SLC7A5',\n",
            " 'SLC7A6',\n",
            " 'SLC7A6OS',\n",
            " 'SLC7A7',\n",
            " 'SLC7A8',\n",
            " 'SLC7A9',\n",
            " 'SLC8A1',\n",
            " 'SLC8A2',\n",
            " 'SLC8A3',\n",
            " 'SLC8B1',\n",
            " 'SLC9A1',\n",
            " 'SLC9A2',\n",
            " 'SLC9A3',\n",
            " 'SLC9A4',\n",
            " 'SLC9A5',\n",
            " 'SLC9A6',\n",
            " 'SLC9A7',\n",
            " 'SLC9A8',\n",
            " 'SLC9A9',\n",
            " 'SLC9B1',\n",
            " 'SLC9B2',\n",
            " 'SLC9C1',\n",
            " 'SLC9C2',\n",
            " 'SLCO1A2',\n",
            " 'SLCO1B1',\n",
            " 'SLCO1B3',\n",
            " 'SLCO1C1',\n",
            " 'SLCO2A1',\n",
            " 'SLCO2B1',\n",
            " 'SLCO3A1',\n",
            " 'SLCO4A1',\n",
            " 'SLCO4C1',\n",
            " 'SLCO5A1',\n",
            " 'SLCO6A1',\n",
            " 'SLF1',\n",
            " 'SLF2',\n",
            " 'SLFN11',\n",
            " 'SLFN12',\n",
            " 'SLFN12L',\n",
            " 'SLFN13',\n",
            " 'SLFN14',\n",
            " 'SLFN5',\n",
            " 'SLFNL1',\n",
            " 'SLIRP',\n",
            " 'SLIT1',\n",
            " 'SLIT2',\n",
            " 'SLIT3',\n",
            " 'SLITRK1',\n",
            " 'SLITRK2',\n",
            " 'SLITRK3',\n",
            " 'SLITRK4',\n",
            " 'SLITRK5',\n",
            " 'SLITRK6',\n",
            " 'SLK',\n",
            " 'SLMAP',\n",
            " 'SLN',\n",
            " 'SLPI',\n",
            " 'SLTM',\n",
            " 'SLU7',\n",
            " 'SLURP1',\n",
            " 'SLX4',\n",
            " 'SLX4IP',\n",
            " 'SLX9',\n",
            " 'SMAD1',\n",
            " 'SMAD2',\n",
            " 'SMAD3',\n",
            " 'SMAD4',\n",
            " 'SMAD5',\n",
            " 'SMAD6',\n",
            " 'SMAD7',\n",
            " 'SMAD9',\n",
            " 'SMAGP',\n",
            " 'SMAP1',\n",
            " 'SMAP2',\n",
            " 'SMARCA1',\n",
            " 'SMARCA2',\n",
            " 'SMARCA4',\n",
            " 'SMARCA5',\n",
            " 'SMARCAD1',\n",
            " 'SMARCAL1',\n",
            " 'SMARCB1',\n",
            " 'SMARCC1',\n",
            " 'SMARCC2',\n",
            " 'SMARCD1',\n",
            " 'SMARCD2',\n",
            " 'SMARCD3',\n",
            " 'SMARCE1',\n",
            " 'SMC1B',\n",
            " 'SMC2',\n",
            " 'SMC3',\n",
            " 'SMC4',\n",
            " 'SMC5',\n",
            " 'SMC6',\n",
            " 'SMCHD1',\n",
            " 'SMCO1',\n",
            " 'SMCO2',\n",
            " 'SMCO3',\n",
            " 'SMCO4',\n",
            " 'SMCP',\n",
            " 'SMCR8',\n",
            " 'SMDT1',\n",
            " 'SMG1',\n",
            " 'SMG5',\n",
            " 'SMG6',\n",
            " 'SMG7',\n",
            " 'SMG8',\n",
            " 'SMG9',\n",
            " 'SMIM1',\n",
            " 'SMIM10',\n",
            " 'SMIM11',\n",
            " 'SMIM12',\n",
            " 'SMIM13',\n",
            " 'SMIM14',\n",
            " 'SMIM15',\n",
            " 'SMIM17',\n",
            " 'SMIM18',\n",
            " 'SMIM2',\n",
            " 'SMIM20',\n",
            " 'SMIM21',\n",
            " 'SMIM22',\n",
            " 'SMIM23',\n",
            " 'SMIM24',\n",
            " 'SMIM29',\n",
            " 'SMIM3',\n",
            " 'SMIM5',\n",
            " 'SMIM6',\n",
            " 'SMIM7',\n",
            " 'SMIM8',\n",
            " 'SMIM9',\n",
            " 'SMKR1',\n",
            " 'SMLR1',\n",
            " 'SMNDC1',\n",
            " 'SMO',\n",
            " 'SMOC1',\n",
            " 'SMOC2',\n",
            " 'SMOX',\n",
            " 'SMPD1',\n",
            " 'SMPD2',\n",
            " 'SMPD3',\n",
            " 'SMPD4',\n",
            " 'SMPDL3A',\n",
            " 'SMPDL3B',\n",
            " 'SMPX',\n",
            " 'SMR3A',\n",
            " 'SMR3B',\n",
            " 'SMS',\n",
            " 'SMTN',\n",
            " 'SMTNL1',\n",
            " 'SMTNL2',\n",
            " 'SMUG1',\n",
            " 'SMURF1',\n",
            " 'SMURF2',\n",
            " 'SMYD1',\n",
            " 'SMYD2',\n",
            " 'SMYD3',\n",
            " 'SMYD4',\n",
            " 'SMYD5',\n",
            " 'SNAI1',\n",
            " 'SNAI2',\n",
            " 'SNAI3',\n",
            " 'SNAP23',\n",
            " 'SNAP25',\n",
            " 'SNAP29',\n",
            " 'SNAP47',\n",
            " 'SNAP91',\n",
            " 'SNAPC1',\n",
            " 'SNAPC2',\n",
            " 'SNAPC3',\n",
            " 'SNAPC4',\n",
            " 'SNAPC5',\n",
            " 'SNAPIN',\n",
            " 'SNCA',\n",
            " 'SNCAIP',\n",
            " 'SNCB',\n",
            " 'SNCG',\n",
            " 'SND1',\n",
            " 'SNED1',\n",
            " 'SNF8',\n",
            " 'SNHG28',\n",
            " 'SNHG32',\n",
            " 'SNIP1',\n",
            " 'SNN',\n",
            " 'SNORC',\n",
            " 'SNPH',\n",
            " 'SNRK',\n",
            " 'SNRNP200',\n",
            " 'SNRNP25',\n",
            " 'SNRNP27',\n",
            " 'SNRNP35',\n",
            " 'SNRNP40',\n",
            " 'SNRNP48',\n",
            " 'SNRNP70',\n",
            " 'SNRPA',\n",
            " 'SNRPA1',\n",
            " 'SNRPB',\n",
            " 'SNRPB2',\n",
            " 'SNRPC',\n",
            " 'SNRPD1',\n",
            " 'SNRPD2',\n",
            " 'SNRPD3',\n",
            " 'SNRPE',\n",
            " 'SNRPF',\n",
            " 'SNRPG',\n",
            " 'SNTA1',\n",
            " 'SNTB1',\n",
            " 'SNTB2',\n",
            " 'SNTG1',\n",
            " 'SNTG2',\n",
            " 'SNTN',\n",
            " 'SNU13',\n",
            " 'SNUPN',\n",
            " 'SNURF',\n",
            " 'SNW1',\n",
            " 'SNX1',\n",
            " 'SNX10',\n",
            " 'SNX11',\n",
            " 'SNX12',\n",
            " 'SNX13',\n",
            " 'SNX14',\n",
            " 'SNX15',\n",
            " 'SNX16',\n",
            " 'SNX17',\n",
            " 'SNX18',\n",
            " 'SNX19',\n",
            " 'SNX2',\n",
            " 'SNX20',\n",
            " 'SNX21',\n",
            " 'SNX22',\n",
            " 'SNX24',\n",
            " 'SNX25',\n",
            " 'SNX27',\n",
            " 'SNX29',\n",
            " 'SNX3',\n",
            " 'SNX30',\n",
            " 'SNX31',\n",
            " 'SNX32',\n",
            " 'SNX33',\n",
            " 'SNX4',\n",
            " 'SNX5',\n",
            " 'SNX6',\n",
            " 'SNX7',\n",
            " 'SNX8',\n",
            " 'SNX9',\n",
            " 'SOAT1',\n",
            " 'SOAT2',\n",
            " 'SOBP',\n",
            " 'SOCS1',\n",
            " 'SOCS2',\n",
            " 'SOCS3',\n",
            " 'SOCS4',\n",
            " 'SOCS5',\n",
            " 'SOCS6',\n",
            " 'SOCS7',\n",
            " 'SOD1',\n",
            " 'SOD2',\n",
            " 'SOD3',\n",
            " 'SOGA1',\n",
            " 'SOGA3',\n",
            " 'SOHLH1',\n",
            " 'SOHLH2',\n",
            " 'SON',\n",
            " 'SORBS1',\n",
            " 'SORBS2',\n",
            " 'SORBS3',\n",
            " 'SORCS1',\n",
            " 'SORCS2',\n",
            " 'SORCS3',\n",
            " 'SORD',\n",
            " 'SORL1',\n",
            " 'SORT1',\n",
            " 'SOS1',\n",
            " 'SOS2',\n",
            " 'SOST',\n",
            " 'SOSTDC1',\n",
            " 'SOWAHA',\n",
            " 'SOWAHB',\n",
            " 'SOWAHC',\n",
            " 'SOWAHD',\n",
            " 'SOX1',\n",
            " 'SOX10',\n",
            " 'SOX11',\n",
            " 'SOX12',\n",
            " 'SOX13',\n",
            " 'SOX14',\n",
            " 'SOX15',\n",
            " 'SOX17',\n",
            " 'SOX18',\n",
            " 'SOX2',\n",
            " 'SOX21',\n",
            " 'SOX3',\n",
            " 'SOX30',\n",
            " 'SOX4',\n",
            " 'SOX5',\n",
            " 'SOX6',\n",
            " 'SOX8',\n",
            " 'SOX9',\n",
            " 'SP1',\n",
            " 'SP100',\n",
            " 'SP110',\n",
            " 'SP140',\n",
            " 'SP140L',\n",
            " 'SP2',\n",
            " 'SP3',\n",
            " 'SP4',\n",
            " 'SP5',\n",
            " 'SP6',\n",
            " 'SP7',\n",
            " 'SP8',\n",
            " 'SP9',\n",
            " 'SPA17',\n",
            " 'SPACA1',\n",
            " 'SPACA3',\n",
            " 'SPACA4',\n",
            " 'SPACA5',\n",
            " 'SPACA5B',\n",
            " 'SPACA7',\n",
            " 'SPACA9',\n",
            " 'SPACDR',\n",
            " 'SPAG1',\n",
            " 'SPAG16',\n",
            " 'SPAG17',\n",
            " 'SPAG4',\n",
            " 'SPAG5',\n",
            " 'SPAG6',\n",
            " 'SPAG7',\n",
            " 'SPAG8',\n",
            " 'SPAG9',\n",
            " 'SPAM1',\n",
            " 'SPANXC',\n",
            " 'SPANXN1',\n",
            " 'SPANXN2',\n",
            " 'SPANXN3',\n",
            " 'SPANXN4',\n",
            " 'SPANXN5',\n",
            " 'SPARC',\n",
            " 'SPARCL1',\n",
            " 'SPART',\n",
            " 'SPAST',\n",
            " 'SPATA12',\n",
            " 'SPATA13',\n",
            " 'SPATA16',\n",
            " 'SPATA17',\n",
            " 'SPATA18',\n",
            " 'SPATA19',\n",
            " 'SPATA2',\n",
            " 'SPATA20',\n",
            " 'SPATA21',\n",
            " 'SPATA22',\n",
            " 'SPATA24',\n",
            " 'SPATA25',\n",
            " 'SPATA2L',\n",
            " 'SPATA3',\n",
            " 'SPATA31D1',\n",
            " 'SPATA31E1',\n",
            " 'SPATA32',\n",
            " 'SPATA33',\n",
            " 'SPATA4',\n",
            " 'SPATA45',\n",
            " 'SPATA46',\n",
            " 'SPATA6',\n",
            " 'SPATA6L',\n",
            " 'SPATA7',\n",
            " 'SPATA8',\n",
            " 'SPATA9',\n",
            " 'SPATC1',\n",
            " 'SPATC1L',\n",
            " 'SPATS1',\n",
            " 'SPATS2',\n",
            " 'SPATS2L',\n",
            " 'SPC24',\n",
            " 'SPC25',\n",
            " 'SPCS1',\n",
            " 'SPCS2',\n",
            " 'SPCS3',\n",
            " 'SPDEF',\n",
            " 'SPDL1',\n",
            " 'SPDYA',\n",
            " 'SPDYC',\n",
            " 'SPDYE2',\n",
            " 'SPDYE2B',\n",
            " 'SPDYE4',\n",
            " 'SPECC1',\n",
            " 'SPECC1L',\n",
            " 'SPEF1',\n",
            " 'SPEF2',\n",
            " 'SPEG',\n",
            " 'SPEM1',\n",
            " 'SPEM2',\n",
            " 'SPEN',\n",
            " 'SPG11',\n",
            " 'SPG21',\n",
            " 'SPG7',\n",
            " 'SPHK1',\n",
            " 'SPHK2',\n",
            " 'SPHKAP',\n",
            " 'SPI1',\n",
            " 'SPIB',\n",
            " 'SPIC',\n",
            " 'SPICE1',\n",
            " 'SPIDR',\n",
            " 'SPIN1',\n",
            " 'SPIN2A',\n",
            " 'SPIN2B',\n",
            " 'SPIN3',\n",
            " 'SPIN4',\n",
            " 'SPINDOC',\n",
            " 'SPINK1',\n",
            " 'SPINK13',\n",
            " 'SPINK14',\n",
            " 'SPINK2',\n",
            " 'SPINK4',\n",
            " 'SPINK5',\n",
            " 'SPINK6',\n",
            " 'SPINK7',\n",
            " 'SPINK8',\n",
            " 'SPINK9',\n",
            " 'SPINT1',\n",
            " 'SPINT2',\n",
            " 'SPINT3',\n",
            " 'SPINT4',\n",
            " 'SPIRE1',\n",
            " 'SPIRE2',\n",
            " 'SPN',\n",
            " 'SPNS1',\n",
            " 'SPNS2',\n",
            " 'SPNS3',\n",
            " 'SPO11',\n",
            " 'SPOCD1',\n",
            " 'SPOCK1',\n",
            " 'SPOCK2',\n",
            " 'SPOCK3',\n",
            " 'SPON1',\n",
            " 'SPON2',\n",
            " 'SPOP',\n",
            " 'SPOPL',\n",
            " 'SPOUT1',\n",
            " 'SPP1',\n",
            " 'SPP2',\n",
            " 'SPPL2A',\n",
            " 'SPPL2B',\n",
            " 'SPPL2C',\n",
            " 'SPPL3',\n",
            " 'SPR',\n",
            " 'SPRED1',\n",
            " 'SPRED2',\n",
            " 'SPRED3',\n",
            " 'SPRING1',\n",
            " 'SPRN',\n",
            " 'SPRR1A',\n",
            " 'SPRR1B',\n",
            " 'SPRR2A',\n",
            " 'SPRR2B',\n",
            " 'SPRR2D',\n",
            " 'SPRR2E',\n",
            " 'SPRR2F',\n",
            " 'SPRR2G',\n",
            " 'SPRR3',\n",
            " 'SPRR4',\n",
            " 'SPRTN',\n",
            " 'SPRY1',\n",
            " 'SPRY2',\n",
            " 'SPRY3',\n",
            " 'SPRY4',\n",
            " 'SPRYD3',\n",
            " 'SPRYD4',\n",
            " 'SPRYD7',\n",
            " 'SPSB1',\n",
            " 'SPSB2',\n",
            " 'SPSB3',\n",
            " 'SPSB4',\n",
            " 'SPTA1',\n",
            " 'SPTAN1',\n",
            " 'SPTB',\n",
            " 'SPTBN1',\n",
            " 'SPTBN2',\n",
            " 'SPTBN4',\n",
            " 'SPTBN5',\n",
            " 'SPTLC1',\n",
            " 'SPTLC2',\n",
            " 'SPTLC3',\n",
            " 'SPTSSA',\n",
            " 'SPTSSB',\n",
            " 'SPTY2D1',\n",
            " 'SPX',\n",
            " 'SPZ1',\n",
            " 'SQLE',\n",
            " 'SQOR',\n",
            " 'SQSTM1',\n",
            " 'SRA1',\n",
            " 'SRARP',\n",
            " 'SRBD1',\n",
            " 'SRC',\n",
            " 'SRCAP',\n",
            " 'SRD5A1',\n",
            " 'SRD5A2',\n",
            " 'SRD5A3',\n",
            " 'SREBF1',\n",
            " 'SREBF2',\n",
            " 'SREK1',\n",
            " 'SREK1IP1',\n",
            " 'SRF',\n",
            " 'SRFBP1',\n",
            " 'SRGAP1',\n",
            " 'SRGAP2',\n",
            " 'SRGAP3',\n",
            " 'SRGN',\n",
            " 'SRI',\n",
            " 'SRL',\n",
            " 'SRM',\n",
            " 'SRMS',\n",
            " 'SRP14',\n",
            " 'SRP19',\n",
            " 'SRP54',\n",
            " 'SRP68',\n",
            " 'SRP72',\n",
            " 'SRP9',\n",
            " 'SRPK1',\n",
            " 'SRPK2',\n",
            " 'SRPK3',\n",
            " 'SRPRA',\n",
            " 'SRPRB',\n",
            " 'SRPX',\n",
            " 'SRPX2',\n",
            " 'SRR',\n",
            " 'SRRD',\n",
            " 'SRRM1',\n",
            " 'SRRM2',\n",
            " 'SRRM3',\n",
            " 'SRRM4',\n",
            " 'SRRM5',\n",
            " 'SRRT',\n",
            " 'SRSF1',\n",
            " 'SRSF10',\n",
            " 'SRSF11',\n",
            " 'SRSF12',\n",
            " 'SRSF2',\n",
            " 'SRSF3',\n",
            " 'SRSF4',\n",
            " 'SRSF5',\n",
            " 'SRSF6',\n",
            " 'SRSF7',\n",
            " 'SRSF8',\n",
            " 'SRSF9',\n",
            " 'SRXN1',\n",
            " 'SRY',\n",
            " 'SS18',\n",
            " 'SS18L1',\n",
            " 'SS18L2',\n",
            " 'SSB',\n",
            " 'SSBP1',\n",
            " 'SSBP2',\n",
            " 'SSBP3',\n",
            " 'SSBP4',\n",
            " 'SSC4D',\n",
            " 'SSC5D',\n",
            " 'SSH1',\n",
            " 'SSH2',\n",
            " 'SSH3',\n",
            " 'SSMEM1',\n",
            " 'SSNA1',\n",
            " 'SSPN',\n",
            " 'SSR1',\n",
            " 'SSR2',\n",
            " 'SSR3',\n",
            " 'SSR4',\n",
            " 'SSRP1',\n",
            " 'SST',\n",
            " 'SSTR1',\n",
            " 'SSTR2',\n",
            " 'SSTR3',\n",
            " 'SSTR4',\n",
            " 'SSTR5',\n",
            " 'SSU72',\n",
            " 'SSUH2',\n",
            " 'SSX1',\n",
            " 'SSX2IP',\n",
            " 'SSX3',\n",
            " 'SSX4',\n",
            " 'SSX4B',\n",
            " 'SSX7',\n",
            " 'ST13',\n",
            " 'ST14',\n",
            " 'ST18',\n",
            " 'ST20',\n",
            " 'ST3GAL1',\n",
            " 'ST3GAL2',\n",
            " 'ST3GAL3',\n",
            " 'ST3GAL4',\n",
            " 'ST3GAL5',\n",
            " 'ST3GAL6',\n",
            " 'ST6GAL1',\n",
            " 'ST6GAL2',\n",
            " 'ST6GALNAC1',\n",
            " 'ST6GALNAC2',\n",
            " 'ST6GALNAC3',\n",
            " 'ST6GALNAC4',\n",
            " 'ST6GALNAC5',\n",
            " 'ST6GALNAC6',\n",
            " 'ST7',\n",
            " 'ST7L',\n",
            " 'ST8SIA1',\n",
            " 'ST8SIA2',\n",
            " 'ST8SIA3',\n",
            " 'ST8SIA4',\n",
            " 'ST8SIA5',\n",
            " 'ST8SIA6',\n",
            " 'STAB1',\n",
            " 'STAB2',\n",
            " 'STAC',\n",
            " 'STAC2',\n",
            " 'STAC3',\n",
            " 'STAG1',\n",
            " 'STAG2',\n",
            " 'STAG3',\n",
            " 'STAM',\n",
            " 'STAM2',\n",
            " 'STAMBP',\n",
            " 'STAMBPL1',\n",
            " 'STAP1',\n",
            " 'STAP2',\n",
            " 'STAR',\n",
            " 'STARD10',\n",
            " 'STARD13',\n",
            " 'STARD3',\n",
            " 'STARD3NL',\n",
            " 'STARD4',\n",
            " 'STARD5',\n",
            " 'STARD6',\n",
            " 'STARD7',\n",
            " 'STARD8',\n",
            " 'STARD9',\n",
            " 'STAT1',\n",
            " 'STAT2',\n",
            " 'STAT3',\n",
            " 'STAT4',\n",
            " 'STAT5A',\n",
            " 'STAT5B',\n",
            " 'STAT6',\n",
            " 'STATH',\n",
            " 'STAU1',\n",
            " 'STAU2',\n",
            " 'STBD1',\n",
            " 'STC1',\n",
            " 'STC2',\n",
            " 'STEAP1',\n",
            " 'STEAP1B',\n",
            " 'STEAP2',\n",
            " 'STEAP3',\n",
            " 'STEAP4',\n",
            " 'STEEP1',\n",
            " 'STH',\n",
            " 'STIL',\n",
            " 'STIM1',\n",
            " 'STIM2',\n",
            " 'STIMATE',\n",
            " 'STING1',\n",
            " 'STIP1',\n",
            " 'STK10',\n",
            " 'STK11',\n",
            " 'STK11IP',\n",
            " 'STK16',\n",
            " 'STK17A',\n",
            " 'STK17B',\n",
            " 'STK19',\n",
            " 'STK24',\n",
            " 'STK25',\n",
            " 'STK26',\n",
            " 'STK3',\n",
            " 'STK31',\n",
            " 'STK32A',\n",
            " 'STK32B',\n",
            " 'STK32C',\n",
            " 'STK33',\n",
            " 'STK35',\n",
            " 'STK36',\n",
            " 'STK38',\n",
            " 'STK38L',\n",
            " 'STK39',\n",
            " 'STK4',\n",
            " 'STK40',\n",
            " 'STKLD1',\n",
            " 'STMN1',\n",
            " 'STMN2',\n",
            " 'STMN3',\n",
            " 'STMN4',\n",
            " 'STMND1',\n",
            " 'STMP1',\n",
            " 'STN1',\n",
            " 'STOM',\n",
            " 'STOML1',\n",
            " 'STOML2',\n",
            " 'STOML3',\n",
            " 'STON1',\n",
            " 'STON2',\n",
            " 'STOX1',\n",
            " 'STOX2',\n",
            " 'STPG1',\n",
            " 'STPG2',\n",
            " 'STPG3',\n",
            " 'STPG4',\n",
            " 'STRA6',\n",
            " 'STRA8',\n",
            " 'STRADA',\n",
            " 'STRADB',\n",
            " 'STRAP',\n",
            " 'STRBP',\n",
            " 'STRC',\n",
            " 'STRIP1',\n",
            " 'STRIP2',\n",
            " 'STRN',\n",
            " 'STRN3',\n",
            " 'STRN4',\n",
            " 'STS',\n",
            " 'STT3A',\n",
            " 'STT3B',\n",
            " 'STUB1',\n",
            " 'STUM',\n",
            " 'STX10',\n",
            " 'STX11',\n",
            " 'STX12',\n",
            " 'STX16',\n",
            " 'STX17',\n",
            " 'STX18',\n",
            " 'STX19',\n",
            " 'STX1A',\n",
            " 'STX1B',\n",
            " 'STX2',\n",
            " 'STX3',\n",
            " 'STX4',\n",
            " 'STX5',\n",
            " 'STX6',\n",
            " 'STX7',\n",
            " 'STX8',\n",
            " 'STXBP1',\n",
            " 'STXBP2',\n",
            " 'STXBP3',\n",
            " 'STXBP4',\n",
            " 'STXBP5',\n",
            " 'STXBP5L',\n",
            " 'STXBP6',\n",
            " 'STYK1',\n",
            " 'STYX',\n",
            " 'STYXL1',\n",
            " 'STYXL2',\n",
            " 'SUB1',\n",
            " 'SUCLA2',\n",
            " 'SUCLG1',\n",
            " 'SUCLG2',\n",
            " 'SUCNR1',\n",
            " 'SUCO',\n",
            " 'SUDS3',\n",
            " 'SUFU',\n",
            " 'SUGCT',\n",
            " 'SUGP1',\n",
            " 'SUGP2',\n",
            " 'SUGT1',\n",
            " 'SULF1',\n",
            " 'SULF2',\n",
            " 'SULT1A1',\n",
            " 'SULT1A2',\n",
            " 'SULT1A3',\n",
            " 'SULT1A4',\n",
            " 'SULT1B1',\n",
            " 'SULT1C2',\n",
            " 'SULT1C3',\n",
            " 'SULT1C4',\n",
            " 'SULT1E1',\n",
            " 'SULT2A1',\n",
            " 'SULT2B1',\n",
            " 'SULT4A1',\n",
            " 'SULT6B1',\n",
            " 'SUMF1',\n",
            " 'SUMF2',\n",
            " 'SUMO1',\n",
            " 'SUMO2',\n",
            " 'SUMO3',\n",
            " 'SUMO4',\n",
            " 'SUN1',\n",
            " 'SUN2',\n",
            " 'SUN3',\n",
            " 'SUN5',\n",
            " 'SUOX',\n",
            " 'SUPT16H',\n",
            " 'SUPT20H',\n",
            " 'SUPT3H',\n",
            " 'SUPT4H1',\n",
            " 'SUPT5H',\n",
            " 'SUPT6H',\n",
            " 'SUPT7L',\n",
            " 'SUPV3L1',\n",
            " 'SURF1',\n",
            " 'SURF2',\n",
            " 'SURF4',\n",
            " 'SURF6',\n",
            " 'SUSD1',\n",
            " 'SUSD2',\n",
            " 'SUSD3',\n",
            " 'SUSD4',\n",
            " 'SUSD5',\n",
            " 'SUSD6',\n",
            " 'SUV39H1',\n",
            " 'SUV39H2',\n",
            " 'SUZ12',\n",
            " 'SV2A',\n",
            " 'SV2B',\n",
            " 'SV2C',\n",
            " 'SVBP',\n",
            " 'SVEP1',\n",
            " 'SVIL',\n",
            " 'SVIP',\n",
            " 'SVOP',\n",
            " 'SVOPL',\n",
            " 'SWAP70',\n",
            " 'SWI5',\n",
            " 'SWSAP1',\n",
            " 'SWT1',\n",
            " 'SYAP1',\n",
            " 'SYBU',\n",
            " 'SYCE1',\n",
            " 'SYCE1L',\n",
            " 'SYCE2',\n",
            " 'SYCE3',\n",
            " 'SYCN',\n",
            " 'SYCP1',\n",
            " 'SYCP2',\n",
            " 'SYCP2L',\n",
            " 'SYCP3',\n",
            " 'SYDE1',\n",
            " 'SYDE2',\n",
            " 'SYF2',\n",
            " 'SYK',\n",
            " 'SYMPK',\n",
            " 'SYN1',\n",
            " 'SYN2',\n",
            " 'SYN3',\n",
            " 'SYNC',\n",
            " 'SYNCRIP',\n",
            " 'SYNDIG1',\n",
            " 'SYNDIG1L',\n",
            " 'SYNE1',\n",
            " 'SYNE2',\n",
            " 'SYNE3',\n",
            " 'SYNE4',\n",
            " 'SYNGAP1',\n",
            " 'SYNGR1',\n",
            " 'SYNGR2',\n",
            " 'SYNGR3',\n",
            " 'SYNGR4',\n",
            " 'SYNJ1',\n",
            " 'SYNJ2',\n",
            " 'SYNJ2BP',\n",
            " 'SYNM',\n",
            " 'SYNPO',\n",
            " 'SYNPO2',\n",
            " 'SYNPO2L',\n",
            " 'SYNPR',\n",
            " 'SYNRG',\n",
            " 'SYP',\n",
            " 'SYPL1',\n",
            " 'SYPL2',\n",
            " 'SYS1',\n",
            " 'SYT1',\n",
            " 'SYT10',\n",
            " 'SYT11',\n",
            " 'SYT12',\n",
            " 'SYT13',\n",
            " 'SYT14',\n",
            " 'SYT15',\n",
            " 'SYT16',\n",
            " 'SYT17',\n",
            " 'SYT2',\n",
            " 'SYT3',\n",
            " 'SYT4',\n",
            " 'SYT5',\n",
            " 'SYT6',\n",
            " 'SYT7',\n",
            " 'SYT8',\n",
            " 'SYT9',\n",
            " 'SYTL1',\n",
            " 'SYTL2',\n",
            " 'SYTL3',\n",
            " 'SYTL4',\n",
            " 'SYTL5',\n",
            " 'SYVN1',\n",
            " 'SZRD1',\n",
            " 'SZT2',\n",
            " 'TAAR1',\n",
            " 'TAAR2',\n",
            " 'TAAR5',\n",
            " 'TAAR6',\n",
            " 'TAAR8',\n",
            " 'TAAR9',\n",
            " 'TAB1',\n",
            " 'TAB2',\n",
            " 'TAB3',\n",
            " 'TAC1',\n",
            " 'TAC3',\n",
            " 'TAC4',\n",
            " 'TACC1',\n",
            " 'TACC2',\n",
            " 'TACC3',\n",
            " 'TACO1',\n",
            " 'TACR1',\n",
            " 'TACR2',\n",
            " 'TACR3',\n",
            " 'TACSTD2',\n",
            " 'TADA1',\n",
            " 'TADA2A',\n",
            " 'TADA2B',\n",
            " 'TADA3',\n",
            " 'TAF1',\n",
            " 'TAF10',\n",
            " 'TAF11',\n",
            " 'TAF12',\n",
            " 'TAF13',\n",
            " 'TAF15',\n",
            " 'TAF1A',\n",
            " 'TAF1B',\n",
            " 'TAF1C',\n",
            " 'TAF1D',\n",
            " 'TAF1L',\n",
            " 'TAF2',\n",
            " 'TAF3',\n",
            " 'TAF4',\n",
            " 'TAF4B',\n",
            " 'TAF5',\n",
            " 'TAF5L',\n",
            " 'TAF6',\n",
            " 'TAF6L',\n",
            " 'TAF7',\n",
            " 'TAF7L',\n",
            " 'TAF8',\n",
            " 'TAF9',\n",
            " 'TAF9B',\n",
            " 'TAFA1',\n",
            " 'TAFA2',\n",
            " 'TAFA3',\n",
            " 'TAFA4',\n",
            " 'TAFA5',\n",
            " 'TAFAZZIN',\n",
            " 'TAGAP',\n",
            " 'TAGLN',\n",
            " 'TAGLN3',\n",
            " 'TAL1',\n",
            " 'TAL2',\n",
            " 'TALDO1',\n",
            " 'TAMALIN',\n",
            " 'TAMM41',\n",
            " 'TANC1',\n",
            " 'TANC2',\n",
            " 'TANGO2',\n",
            " 'TANGO6',\n",
            " 'TANK',\n",
            " 'TAOK1',\n",
            " 'TAOK2',\n",
            " 'TAOK3',\n",
            " 'TAP1',\n",
            " 'TAP2',\n",
            " 'TAPBP',\n",
            " 'TAPBPL',\n",
            " 'TAPT1',\n",
            " 'TARBP1',\n",
            " 'TARBP2',\n",
            " 'TARDBP',\n",
            " 'TARM1',\n",
            " 'TARP',\n",
            " 'TARS1',\n",
            " 'TARS2',\n",
            " 'TARS3',\n",
            " 'TAS1R1',\n",
            " 'TAS1R2',\n",
            " 'TAS1R3',\n",
            " 'TAS2R1',\n",
            " 'TAS2R10',\n",
            " 'TAS2R13',\n",
            " 'TAS2R14',\n",
            " 'TAS2R16',\n",
            " 'TAS2R19',\n",
            " 'TAS2R20',\n",
            " 'TAS2R3',\n",
            " 'TAS2R30',\n",
            " 'TAS2R31',\n",
            " 'TAS2R38',\n",
            " 'TAS2R39',\n",
            " 'TAS2R4',\n",
            " 'TAS2R40',\n",
            " 'TAS2R41',\n",
            " 'TAS2R42',\n",
            " 'TAS2R43',\n",
            " 'TAS2R46',\n",
            " 'TAS2R5',\n",
            " 'TAS2R50',\n",
            " 'TAS2R60',\n",
            " 'TAS2R7',\n",
            " 'TAS2R8',\n",
            " 'TAS2R9',\n",
            " 'TASL',\n",
            " 'TASOR',\n",
            " 'TASOR2',\n",
            " 'TASP1',\n",
            " 'TAT',\n",
            " 'TATDN1',\n",
            " 'TATDN2',\n",
            " 'TATDN3',\n",
            " 'TAX1BP1',\n",
            " 'TAX1BP3',\n",
            " 'TBATA',\n",
            " 'TBC1D1',\n",
            " 'TBC1D10A',\n",
            " 'TBC1D10B',\n",
            " 'TBC1D10C',\n",
            " 'TBC1D12',\n",
            " 'TBC1D13',\n",
            " 'TBC1D14',\n",
            " 'TBC1D15',\n",
            " 'TBC1D16',\n",
            " 'TBC1D17',\n",
            " 'TBC1D19',\n",
            " 'TBC1D2',\n",
            " 'TBC1D20',\n",
            " 'TBC1D21',\n",
            " 'TBC1D22A',\n",
            " 'TBC1D22B',\n",
            " 'TBC1D23',\n",
            " 'TBC1D24',\n",
            " 'TBC1D25',\n",
            " 'TBC1D28',\n",
            " 'TBC1D2B',\n",
            " 'TBC1D3',\n",
            " 'TBC1D30',\n",
            " 'TBC1D31',\n",
            " 'TBC1D32',\n",
            " 'TBC1D4',\n",
            " 'TBC1D5',\n",
            " 'TBC1D7',\n",
            " 'TBC1D8',\n",
            " 'TBC1D8B',\n",
            " 'TBC1D9',\n",
            " 'TBC1D9B',\n",
            " 'TBCA',\n",
            " 'TBCB',\n",
            " 'TBCC',\n",
            " 'TBCCD1',\n",
            " 'TBCD',\n",
            " 'TBCE',\n",
            " 'TBCEL',\n",
            " 'TBCK',\n",
            " 'TBK1',\n",
            " 'TBKBP1',\n",
            " 'TBL1X',\n",
            " 'TBL1XR1',\n",
            " 'TBL1Y',\n",
            " 'TBL2',\n",
            " 'TBL3',\n",
            " 'TBP',\n",
            " 'TBPL1',\n",
            " 'TBPL2',\n",
            " 'TBR1',\n",
            " 'TBRG1',\n",
            " 'TBRG4',\n",
            " 'TBX1',\n",
            " 'TBX10',\n",
            " 'TBX15',\n",
            " 'TBX18',\n",
            " 'TBX19',\n",
            " 'TBX2',\n",
            " 'TBX20',\n",
            " 'TBX21',\n",
            " 'TBX22',\n",
            " 'TBX3',\n",
            " 'TBX4',\n",
            " 'TBX5',\n",
            " 'TBX6',\n",
            " 'TBXA2R',\n",
            " 'TBXAS1',\n",
            " 'TBXT',\n",
            " 'TC2N',\n",
            " 'TCAF1',\n",
            " 'TCAF2',\n",
            " 'TCAIM',\n",
            " 'TCAP',\n",
            " 'TCEA1',\n",
            " 'TCEA2',\n",
            " 'TCEA3',\n",
            " 'TCEAL1',\n",
            " 'TCEAL2',\n",
            " 'TCEAL3',\n",
            " 'TCEAL4',\n",
            " 'TCEAL5',\n",
            " 'TCEAL6',\n",
            " 'TCEAL7',\n",
            " 'TCEAL8',\n",
            " 'TCEAL9',\n",
            " 'TCEANC',\n",
            " 'TCEANC2',\n",
            " 'TCERG1',\n",
            " 'TCERG1L',\n",
            " 'TCF12',\n",
            " 'TCF15',\n",
            " 'TCF19',\n",
            " 'TCF20',\n",
            " 'TCF21',\n",
            " 'TCF23',\n",
            " 'TCF24',\n",
            " 'TCF25',\n",
            " 'TCF3',\n",
            " 'TCF4',\n",
            " 'TCF7',\n",
            " 'TCF7L1',\n",
            " 'TCF7L2',\n",
            " 'TCFL5',\n",
            " 'TCHH',\n",
            " 'TCHHL1',\n",
            " 'TCHP',\n",
            " 'TCIM',\n",
            " 'TCIRG1',\n",
            " 'TCL1A',\n",
            " 'TCL1B',\n",
            " 'TCN1',\n",
            " 'TCN2',\n",
            " 'TCOF1',\n",
            " 'TCP1',\n",
            " 'TCP10L',\n",
            " 'TCP10L3',\n",
            " 'TCP11',\n",
            " 'TCP11L1',\n",
            " 'TCP11L2',\n",
            " 'TCTA',\n",
            " 'TCTE1',\n",
            " 'TCTN1',\n",
            " 'TCTN2',\n",
            " 'TCTN3',\n",
            " 'TDG',\n",
            " 'TDO2',\n",
            " 'TDP1',\n",
            " 'TDP2',\n",
            " 'TDRD1',\n",
            " 'TDRD12',\n",
            " 'TDRD15',\n",
            " 'TDRD3',\n",
            " 'TDRD5',\n",
            " 'TDRD6',\n",
            " 'TDRD7',\n",
            " 'TDRD9',\n",
            " 'TDRKH',\n",
            " 'TDRP',\n",
            " 'TEAD1',\n",
            " 'TEAD2',\n",
            " 'TEAD3',\n",
            " 'TEAD4',\n",
            " 'TEC',\n",
            " 'TECPR1',\n",
            " 'TECPR2',\n",
            " 'TECR',\n",
            " 'TECRL',\n",
            " 'TECTA',\n",
            " 'TECTB',\n",
            " 'TEDC1',\n",
            " 'TEDC2',\n",
            " 'TEDDM1',\n",
            " 'TEF',\n",
            " 'TEFM',\n",
            " 'TEK',\n",
            " 'TEKT1',\n",
            " 'TEKT2',\n",
            " 'TEKT3',\n",
            " 'TEKT4',\n",
            " 'TEKT5',\n",
            " 'TEKTIP1',\n",
            " 'TELO2',\n",
            " 'TENM1',\n",
            " 'TENM2',\n",
            " 'TENM3',\n",
            " 'TENM4',\n",
            " 'TENT2',\n",
            " 'TENT4A',\n",
            " 'TENT4B',\n",
            " 'TENT5A',\n",
            " 'TENT5B',\n",
            " 'TENT5C',\n",
            " 'TENT5D',\n",
            " 'TEP1',\n",
            " 'TEPSIN',\n",
            " 'TERB1',\n",
            " 'TERB2',\n",
            " 'TERF1',\n",
            " 'TERF2',\n",
            " 'TERF2IP',\n",
            " 'TERT',\n",
            " 'TES',\n",
            " 'TESC',\n",
            " 'TESK1',\n",
            " 'TESK2',\n",
            " 'TESMIN',\n",
            " 'TESPA1',\n",
            " 'TET1',\n",
            " 'TET2',\n",
            " 'TET3',\n",
            " 'TEX10',\n",
            " 'TEX101',\n",
            " 'TEX11',\n",
            " 'TEX12',\n",
            " 'TEX13A',\n",
            " 'TEX13B',\n",
            " 'TEX14',\n",
            " 'TEX15',\n",
            " 'TEX19',\n",
            " 'TEX2',\n",
            " 'TEX22',\n",
            " 'TEX26',\n",
            " 'TEX261',\n",
            " 'TEX264',\n",
            " 'TEX28',\n",
            " 'TEX29',\n",
            " 'TEX30',\n",
            " 'TEX35',\n",
            " 'TEX36',\n",
            " 'TEX38',\n",
            " 'TEX44',\n",
            " 'TEX46',\n",
            " 'TEX47',\n",
            " 'TEX55',\n",
            " 'TEX9',\n",
            " 'TF',\n",
            " 'TFAM',\n",
            " 'TFAP2A',\n",
            " 'TFAP2B',\n",
            " 'TFAP2C',\n",
            " 'TFAP2D',\n",
            " 'TFAP2E',\n",
            " 'TFAP4',\n",
            " 'TFB1M',\n",
            " 'TFB2M',\n",
            " 'TFCP2',\n",
            " 'TFCP2L1',\n",
            " 'TFDP1',\n",
            " 'TFDP2',\n",
            " 'TFDP3',\n",
            " 'TFE3',\n",
            " 'TFEB',\n",
            " 'TFEC',\n",
            " 'TFF1',\n",
            " 'TFF2',\n",
            " 'TFF3',\n",
            " 'TFG',\n",
            " 'TFIP11',\n",
            " 'TFPI',\n",
            " 'TFPI2',\n",
            " 'TFPT',\n",
            " 'TFR2',\n",
            " 'TFRC',\n",
            " 'TG',\n",
            " 'TGDS',\n",
            " 'TGFA',\n",
            " 'TGFB1',\n",
            " 'TGFB1I1',\n",
            " 'TGFB2',\n",
            " 'TGFB3',\n",
            " 'TGFBI',\n",
            " 'TGFBR1',\n",
            " 'TGFBR2',\n",
            " 'TGFBR3',\n",
            " 'TGFBR3L',\n",
            " 'TGFBRAP1',\n",
            " 'TGIF1',\n",
            " 'TGIF2',\n",
            " 'TGIF2LX',\n",
            " 'TGIF2LY',\n",
            " 'TGM1',\n",
            " 'TGM2',\n",
            " 'TGM3',\n",
            " 'TGM4',\n",
            " 'TGM5',\n",
            " 'TGM6',\n",
            " 'TGM7',\n",
            " 'TGOLN2',\n",
            " 'TGS1',\n",
            " 'TH',\n",
            " 'THADA',\n",
            " 'THAP1',\n",
            " 'THAP10',\n",
            " 'THAP11',\n",
            " 'THAP12',\n",
            " 'THAP2',\n",
            " 'THAP3',\n",
            " 'THAP4',\n",
            " 'THAP5',\n",
            " 'THAP6',\n",
            " 'THAP7',\n",
            " 'THAP8',\n",
            " 'THAP9',\n",
            " 'THBD',\n",
            " 'THBS1',\n",
            " 'THBS2',\n",
            " 'THBS3',\n",
            " 'THBS4',\n",
            " 'THEM4',\n",
            " 'THEM5',\n",
            " 'THEM6',\n",
            " 'THEMIS',\n",
            " 'THEMIS2',\n",
            " 'THG1L',\n",
            " 'THNSL1',\n",
            " 'THNSL2',\n",
            " 'THOC1',\n",
            " 'THOC2',\n",
            " 'THOC3',\n",
            " 'THOC5',\n",
            " 'THOC6',\n",
            " 'THOC7',\n",
            " 'THOP1',\n",
            " 'THPO',\n",
            " 'THRA',\n",
            " 'THRAP3',\n",
            " 'THRB',\n",
            " 'THRSP',\n",
            " 'THSD1',\n",
            " 'THSD4',\n",
            " 'THSD7A',\n",
            " 'THSD7B',\n",
            " 'THTPA',\n",
            " 'THUMPD1',\n",
            " 'THUMPD2',\n",
            " 'THUMPD3',\n",
            " 'THY1',\n",
            " 'THYN1',\n",
            " 'TIA1',\n",
            " 'TIAL1',\n",
            " 'TIAM1',\n",
            " 'TIAM2',\n",
            " 'TICAM1',\n",
            " 'TICAM2',\n",
            " 'TICRR',\n",
            " 'TIE1',\n",
            " 'TIFA',\n",
            " 'TIFAB',\n",
            " 'TIGAR',\n",
            " 'TIGD1',\n",
            " 'TIGD2',\n",
            " 'TIGD3',\n",
            " 'TIGD4',\n",
            " 'TIGD5',\n",
            " 'TIGD6',\n",
            " 'TIGD7',\n",
            " 'TIGIT',\n",
            " 'TIMD4',\n",
            " 'TIMELESS',\n",
            " 'TIMM10',\n",
            " 'TIMM10B',\n",
            " 'TIMM13',\n",
            " 'TIMM17A',\n",
            " 'TIMM17B',\n",
            " 'TIMM21',\n",
            " 'TIMM22',\n",
            " 'TIMM29',\n",
            " 'TIMM44',\n",
            " 'TIMM50',\n",
            " 'TIMM8A',\n",
            " 'TIMM8B',\n",
            " 'TIMM9',\n",
            " 'TIMMDC1',\n",
            " 'TIMP1',\n",
            " 'TIMP2',\n",
            " 'TIMP3',\n",
            " 'TIMP4',\n",
            " 'TINAG',\n",
            " 'TINAGL1',\n",
            " 'TINF2',\n",
            " 'TIPARP',\n",
            " 'TIPIN',\n",
            " 'TIPRL',\n",
            " 'TIRAP',\n",
            " 'TJAP1',\n",
            " 'TJP1',\n",
            " 'TJP2',\n",
            " 'TJP3',\n",
            " 'TK1',\n",
            " 'TK2',\n",
            " 'TKFC',\n",
            " 'TKT',\n",
            " 'TKTL1',\n",
            " 'TKTL2',\n",
            " 'TLCD1',\n",
            " 'TLCD2',\n",
            " 'TLCD3A',\n",
            " 'TLCD3B',\n",
            " 'TLCD4',\n",
            " 'TLCD5',\n",
            " 'TLDC2',\n",
            " 'TLE1',\n",
            " 'TLE2',\n",
            " 'TLE3',\n",
            " 'TLE4',\n",
            " 'TLE5',\n",
            " 'TLE6',\n",
            " 'TLK1',\n",
            " 'TLK2',\n",
            " 'TLL1',\n",
            " 'TLL2',\n",
            " 'TLN1',\n",
            " 'TLN2',\n",
            " 'TLNRD1',\n",
            " 'TLR1',\n",
            " 'TLR10',\n",
            " 'TLR2',\n",
            " 'TLR3',\n",
            " 'TLR4',\n",
            " 'TLR5',\n",
            " 'TLR6',\n",
            " 'TLR7',\n",
            " 'TLR8',\n",
            " 'TLR9',\n",
            " 'TLX2',\n",
            " 'TLX3',\n",
            " 'TM2D1',\n",
            " 'TM2D2',\n",
            " 'TM2D3',\n",
            " 'TM4SF1',\n",
            " 'TM4SF18',\n",
            " 'TM4SF19',\n",
            " 'TM4SF20',\n",
            " 'TM4SF4',\n",
            " 'TM4SF5',\n",
            " 'TM6SF1',\n",
            " 'TM7SF2',\n",
            " 'TM7SF3',\n",
            " 'TM9SF1',\n",
            " 'TM9SF2',\n",
            " 'TM9SF3',\n",
            " 'TM9SF4',\n",
            " 'TMA16',\n",
            " 'TMA7',\n",
            " 'TMBIM1',\n",
            " 'TMBIM4',\n",
            " 'TMBIM6',\n",
            " 'TMC1',\n",
            " 'TMC2',\n",
            " 'TMC4',\n",
            " 'TMC5',\n",
            " 'TMC6',\n",
            " 'TMC7',\n",
            " 'TMC8',\n",
            " 'TMCC1',\n",
            " 'TMCC2',\n",
            " 'TMCC3',\n",
            " 'TMCO1',\n",
            " 'TMCO2',\n",
            " 'TMCO3',\n",
            " 'TMCO4',\n",
            " 'TMCO5A',\n",
            " 'TMCO6',\n",
            " 'TMED1',\n",
            " 'TMED10',\n",
            " 'TMED2',\n",
            " 'TMED3',\n",
            " 'TMED4',\n",
            " 'TMED5',\n",
            " 'TMED6',\n",
            " 'TMED7',\n",
            " 'TMED8',\n",
            " 'TMED9',\n",
            " 'TMEFF1',\n",
            " 'TMEFF2',\n",
            " 'TMEM100',\n",
            " 'TMEM101',\n",
            " 'TMEM102',\n",
            " 'TMEM104',\n",
            " 'TMEM105',\n",
            " 'TMEM106A',\n",
            " 'TMEM106B',\n",
            " 'TMEM106C',\n",
            " 'TMEM107',\n",
            " 'TMEM108',\n",
            " 'TMEM109',\n",
            " 'TMEM11',\n",
            " 'TMEM114',\n",
            " 'TMEM115',\n",
            " 'TMEM116',\n",
            " 'TMEM117',\n",
            " 'TMEM119',\n",
            " 'TMEM120A',\n",
            " 'TMEM120B',\n",
            " 'TMEM121',\n",
            " 'TMEM121B',\n",
            " 'TMEM123',\n",
            " 'TMEM125',\n",
            " 'TMEM126A',\n",
            " 'TMEM126B',\n",
            " 'TMEM127',\n",
            " 'TMEM128',\n",
            " 'TMEM129',\n",
            " 'TMEM130',\n",
            " 'TMEM131',\n",
            " 'TMEM131L',\n",
            " 'TMEM132A',\n",
            " 'TMEM132B',\n",
            " 'TMEM132C',\n",
            " 'TMEM132D',\n",
            " 'TMEM132E',\n",
            " 'TMEM132E-DT',\n",
            " 'TMEM134',\n",
            " 'TMEM135',\n",
            " 'TMEM138',\n",
            " 'TMEM139',\n",
            " 'TMEM140',\n",
            " 'TMEM141',\n",
            " 'TMEM143',\n",
            " 'TMEM144',\n",
            " 'TMEM145',\n",
            " 'TMEM147',\n",
            " 'TMEM14A',\n",
            " 'TMEM14B',\n",
            " 'TMEM14C',\n",
            " 'TMEM150A',\n",
            " 'TMEM150B',\n",
            " 'TMEM150C',\n",
            " 'TMEM151A',\n",
            " 'TMEM151B',\n",
            " 'TMEM154',\n",
            " 'TMEM156',\n",
            " 'TMEM158',\n",
            " 'TMEM160',\n",
            " 'TMEM161A',\n",
            " 'TMEM161B',\n",
            " 'TMEM163',\n",
            " 'TMEM164',\n",
            " 'TMEM165',\n",
            " 'TMEM167A',\n",
            " 'TMEM167B',\n",
            " 'TMEM168',\n",
            " 'TMEM169',\n",
            " 'TMEM17',\n",
            " 'TMEM170A',\n",
            " 'TMEM170B',\n",
            " 'TMEM171',\n",
            " 'TMEM174',\n",
            " 'TMEM175',\n",
            " 'TMEM176A',\n",
            " 'TMEM176B',\n",
            " 'TMEM177',\n",
            " 'TMEM178A',\n",
            " 'TMEM178B',\n",
            " 'TMEM179',\n",
            " 'TMEM179B',\n",
            " 'TMEM18',\n",
            " 'TMEM181',\n",
            " 'TMEM182',\n",
            " 'TMEM183A',\n",
            " 'TMEM184A',\n",
            " 'TMEM184B',\n",
            " 'TMEM184C',\n",
            " 'TMEM185A',\n",
            " 'TMEM185B',\n",
            " 'TMEM186',\n",
            " 'TMEM187',\n",
            " 'TMEM19',\n",
            " 'TMEM190',\n",
            " 'TMEM191B',\n",
            " 'TMEM191C',\n",
            " 'TMEM192',\n",
            " 'TMEM196',\n",
            " 'TMEM198',\n",
            " 'TMEM199',\n",
            " 'TMEM200A',\n",
            " 'TMEM200B',\n",
            " 'TMEM200C',\n",
            " 'TMEM201',\n",
            " 'TMEM202',\n",
            " 'TMEM203',\n",
            " 'TMEM204',\n",
            " 'TMEM205',\n",
            " 'TMEM207',\n",
            " 'TMEM208',\n",
            " 'TMEM209',\n",
            " 'TMEM210',\n",
            " 'TMEM212',\n",
            " 'TMEM213',\n",
            " 'TMEM214',\n",
            " 'TMEM215',\n",
            " 'TMEM216',\n",
            " 'TMEM217',\n",
            " 'TMEM218',\n",
            " 'TMEM219',\n",
            " 'TMEM220',\n",
            " 'TMEM221',\n",
            " 'TMEM222',\n",
            " 'TMEM223',\n",
            " 'TMEM225',\n",
            " 'TMEM229A',\n",
            " 'TMEM229B',\n",
            " 'TMEM230',\n",
            " 'TMEM231',\n",
            " 'TMEM232',\n",
            " 'TMEM233',\n",
            " 'TMEM234',\n",
            " 'TMEM235',\n",
            " 'TMEM236',\n",
            " 'TMEM237',\n",
            " 'TMEM238',\n",
            " 'TMEM239',\n",
            " 'TMEM240',\n",
            " 'TMEM241',\n",
            " 'TMEM242',\n",
            " 'TMEM243',\n",
            " 'TMEM244',\n",
            " 'TMEM245',\n",
            " 'TMEM247',\n",
            " 'TMEM248',\n",
            " 'TMEM249',\n",
            " 'TMEM25',\n",
            " 'TMEM250',\n",
            " 'TMEM252',\n",
            " 'TMEM253',\n",
            " 'TMEM254',\n",
            " 'TMEM255A',\n",
            " 'TMEM255B',\n",
            " 'TMEM256',\n",
            " 'TMEM258',\n",
            " 'TMEM259',\n",
            " 'TMEM26',\n",
            " 'TMEM260',\n",
            " 'TMEM263',\n",
            " 'TMEM266',\n",
            " 'TMEM267',\n",
            " 'TMEM268',\n",
            " 'TMEM270',\n",
            " 'TMEM273',\n",
            " 'TMEM30A',\n",
            " 'TMEM30B',\n",
            " 'TMEM31',\n",
            " 'TMEM33',\n",
            " 'TMEM35A',\n",
            " 'TMEM35B',\n",
            " 'TMEM37',\n",
            " 'TMEM38A',\n",
            " 'TMEM38B',\n",
            " 'TMEM39A',\n",
            " 'TMEM39B',\n",
            " 'TMEM40',\n",
            " 'TMEM41A',\n",
            " 'TMEM41B',\n",
            " 'TMEM42',\n",
            " 'TMEM43',\n",
            " 'TMEM44',\n",
            " 'TMEM45A',\n",
            " 'TMEM45B',\n",
            " 'TMEM47',\n",
            " 'TMEM50A',\n",
            " 'TMEM50B',\n",
            " 'TMEM51',\n",
            " 'TMEM51-AS2',\n",
            " 'TMEM52',\n",
            " 'TMEM52B',\n",
            " 'TMEM53',\n",
            " 'TMEM54',\n",
            " 'TMEM59',\n",
            " 'TMEM59L',\n",
            " 'TMEM60',\n",
            " 'TMEM61',\n",
            " 'TMEM62',\n",
            " 'TMEM63A',\n",
            " 'TMEM63B',\n",
            " 'TMEM63C',\n",
            " 'TMEM64',\n",
            " 'TMEM65',\n",
            " 'TMEM67',\n",
            " 'TMEM68',\n",
            " 'TMEM69',\n",
            " 'TMEM70',\n",
            " 'TMEM71',\n",
            " 'TMEM72',\n",
            " 'TMEM74',\n",
            " 'TMEM74B',\n",
            " 'TMEM79',\n",
            " 'TMEM80',\n",
            " 'TMEM81',\n",
            " 'TMEM82',\n",
            " 'TMEM86A',\n",
            " 'TMEM86B',\n",
            " 'TMEM87A',\n",
            " 'TMEM87B',\n",
            " 'TMEM88',\n",
            " 'TMEM88B',\n",
            " 'TMEM89',\n",
            " 'TMEM8B',\n",
            " 'TMEM9',\n",
            " 'TMEM91',\n",
            " 'TMEM92',\n",
            " 'TMEM94',\n",
            " 'TMEM95',\n",
            " 'TMEM97',\n",
            " 'TMEM98',\n",
            " 'TMEM9B',\n",
            " 'TMF1',\n",
            " 'TMIE',\n",
            " 'TMIGD1',\n",
            " 'TMIGD2',\n",
            " 'TMLHE',\n",
            " 'TMOD1',\n",
            " 'TMOD2',\n",
            " 'TMOD3',\n",
            " 'TMOD4',\n",
            " 'TMPO',\n",
            " 'TMPPE',\n",
            " 'TMPRSS11A',\n",
            " 'TMPRSS11B',\n",
            " 'TMPRSS11D',\n",
            " 'TMPRSS11E',\n",
            " 'TMPRSS11F',\n",
            " 'TMPRSS12',\n",
            " 'TMPRSS13',\n",
            " 'TMPRSS15',\n",
            " 'TMPRSS2',\n",
            " 'TMPRSS3',\n",
            " 'TMPRSS4',\n",
            " 'TMPRSS5',\n",
            " 'TMPRSS6',\n",
            " 'TMPRSS7',\n",
            " 'TMPRSS9',\n",
            " 'TMSB10',\n",
            " 'TMSB15A',\n",
            " 'TMSB15B',\n",
            " 'TMSB4X',\n",
            " 'TMSB4Y',\n",
            " 'TMTC1',\n",
            " 'TMTC2',\n",
            " 'TMTC3',\n",
            " 'TMTC4',\n",
            " 'TMUB1',\n",
            " 'TMUB2',\n",
            " 'TMX1',\n",
            " 'TMX2',\n",
            " 'TMX3',\n",
            " 'TMX4',\n",
            " 'TNC',\n",
            " 'TNF',\n",
            " 'TNFAIP1',\n",
            " 'TNFAIP2',\n",
            " 'TNFAIP3',\n",
            " 'TNFAIP6',\n",
            " 'TNFAIP8',\n",
            " 'TNFAIP8L1',\n",
            " 'TNFAIP8L2',\n",
            " 'TNFAIP8L3',\n",
            " 'TNFRSF10A',\n",
            " 'TNFRSF10B',\n",
            " 'TNFRSF10C',\n",
            " 'TNFRSF10D',\n",
            " 'TNFRSF11A',\n",
            " 'TNFRSF11B',\n",
            " 'TNFRSF12A',\n",
            " 'TNFRSF13B',\n",
            " 'TNFRSF13C',\n",
            " 'TNFRSF14',\n",
            " 'TNFRSF17',\n",
            " 'TNFRSF18',\n",
            " 'TNFRSF19',\n",
            " 'TNFRSF1A',\n",
            " 'TNFRSF1B',\n",
            " 'TNFRSF21',\n",
            " 'TNFRSF25',\n",
            " 'TNFRSF4',\n",
            " 'TNFRSF6B',\n",
            " 'TNFRSF8',\n",
            " 'TNFRSF9',\n",
            " 'TNFSF10',\n",
            " 'TNFSF11',\n",
            " 'TNFSF12',\n",
            " 'TNFSF13',\n",
            " 'TNFSF13B',\n",
            " 'TNFSF14',\n",
            " 'TNFSF15',\n",
            " 'TNFSF18',\n",
            " 'TNFSF4',\n",
            " 'TNFSF8',\n",
            " 'TNFSF9',\n",
            " 'TNIK',\n",
            " 'TNIP1',\n",
            " 'TNIP2',\n",
            " 'TNIP3',\n",
            " 'TNK1',\n",
            " 'TNK2',\n",
            " 'TNKS',\n",
            " 'TNKS1BP1',\n",
            " 'TNKS2',\n",
            " 'TNMD',\n",
            " 'TNN',\n",
            " 'TNNC1',\n",
            " 'TNNC2',\n",
            " 'TNNI1',\n",
            " 'TNNI2',\n",
            " 'TNNI3',\n",
            " 'TNNI3K',\n",
            " 'TNNT1',\n",
            " 'TNNT2',\n",
            " 'TNNT3',\n",
            " 'TNP1',\n",
            " 'TNP2',\n",
            " 'TNPO1',\n",
            " 'TNPO2',\n",
            " 'TNPO3',\n",
            " 'TNR',\n",
            " 'TNRC18',\n",
            " 'TNRC6A',\n",
            " 'TNRC6B',\n",
            " 'TNRC6C',\n",
            " 'TNS1',\n",
            " 'TNS2',\n",
            " 'TNS3',\n",
            " 'TNS4',\n",
            " 'TNXB',\n",
            " 'TOB1',\n",
            " 'TOB2',\n",
            " 'TOE1',\n",
            " 'TOGARAM1',\n",
            " 'TOGARAM2',\n",
            " 'TOLLIP',\n",
            " 'TOM1',\n",
            " 'TOM1L1',\n",
            " 'TOM1L2',\n",
            " 'TOMM20',\n",
            " 'TOMM20L',\n",
            " 'TOMM22',\n",
            " 'TOMM34',\n",
            " 'TOMM40',\n",
            " 'TOMM40L',\n",
            " 'TOMM5',\n",
            " 'TOMM6',\n",
            " 'TOMM7',\n",
            " 'TOMM70',\n",
            " 'TONSL',\n",
            " 'TOP1',\n",
            " 'TOP1MT',\n",
            " 'TOP2A',\n",
            " 'TOP2B',\n",
            " 'TOP3A',\n",
            " 'TOP3B',\n",
            " 'TOPAZ1',\n",
            " 'TOPBP1',\n",
            " 'TOPORS',\n",
            " 'TOR1A',\n",
            " 'TOR1AIP1',\n",
            " 'TOR1AIP2',\n",
            " 'TOR1B',\n",
            " 'TOR2A',\n",
            " 'TOR3A',\n",
            " 'TOR4A',\n",
            " 'TOX',\n",
            " 'TOX2',\n",
            " 'TOX3',\n",
            " 'TOX4',\n",
            " 'TP53',\n",
            " 'TP53AIP1',\n",
            " 'TP53BP1',\n",
            " 'TP53BP2',\n",
            " 'TP53I11',\n",
            " 'TP53I13',\n",
            " 'TP53I3',\n",
            " 'TP53INP1',\n",
            " 'TP53INP2',\n",
            " 'TP53RK',\n",
            " 'TP53TG3C',\n",
            " 'TP53TG5',\n",
            " 'TP63',\n",
            " 'TP73',\n",
            " 'TPBG',\n",
            " 'TPBGL',\n",
            " 'TPCN1',\n",
            " 'TPCN2',\n",
            " 'TPD52',\n",
            " 'TPD52L1',\n",
            " 'TPD52L2',\n",
            " 'TPD52L3',\n",
            " 'TPGS1',\n",
            " 'TPGS2',\n",
            " 'TPH1',\n",
            " 'TPH2',\n",
            " 'TPI1',\n",
            " 'TPK1',\n",
            " 'TPM1',\n",
            " 'TPM2',\n",
            " 'TPM3',\n",
            " 'TPM4',\n",
            " 'TPMT',\n",
            " 'TPO',\n",
            " 'TPP1',\n",
            " 'TPP2',\n",
            " 'TPPP',\n",
            " 'TPPP2',\n",
            " 'TPPP3',\n",
            " 'TPR',\n",
            " 'TPRA1',\n",
            " 'TPRG1',\n",
            " 'TPRG1L',\n",
            " 'TPRKB',\n",
            " 'TPRN',\n",
            " 'TPRX1',\n",
            " 'TPSAB1',\n",
            " 'TPSB2',\n",
            " 'TPSD1',\n",
            " 'TPSG1',\n",
            " 'TPST1',\n",
            " 'TPST2',\n",
            " 'TPT1',\n",
            " 'TPTE',\n",
            " 'TPTE2',\n",
            " 'TPX2',\n",
            " 'TRA2A',\n",
            " 'TRA2B',\n",
            " 'TRABD',\n",
            " 'TRABD2A',\n",
            " 'TRABD2B',\n",
            " 'TRADD',\n",
            " 'TRAF1',\n",
            " 'TRAF2',\n",
            " 'TRAF3',\n",
            " 'TRAF3IP1',\n",
            " 'TRAF3IP2',\n",
            " 'TRAF3IP3',\n",
            " 'TRAF4',\n",
            " 'TRAF5',\n",
            " 'TRAF6',\n",
            " 'TRAF7',\n",
            " 'TRAFD1',\n",
            " 'TRAIP',\n",
            " 'TRAK1',\n",
            " 'TRAK2',\n",
            " 'TRAM1',\n",
            " 'TRAM1L1',\n",
            " 'TRAM2',\n",
            " 'TRANK1',\n",
            " 'TRAP1',\n",
            " 'TRAPPC1',\n",
            " 'TRAPPC10',\n",
            " 'TRAPPC11',\n",
            " 'TRAPPC12',\n",
            " 'TRAPPC13',\n",
            " 'TRAPPC14',\n",
            " 'TRAPPC2',\n",
            " 'TRAPPC2L',\n",
            " 'TRAPPC3',\n",
            " 'TRAPPC3L',\n",
            " 'TRAPPC4',\n",
            " 'TRAPPC5',\n",
            " 'TRAPPC6A',\n",
            " 'TRAPPC6B',\n",
            " 'TRAPPC8',\n",
            " 'TRAPPC9',\n",
            " 'TRARG1',\n",
            " 'TRAT1',\n",
            " 'TRDMT1',\n",
            " 'TRDN',\n",
            " 'TREH',\n",
            " 'TREM1',\n",
            " 'TREM2',\n",
            " 'TREML1',\n",
            " 'TREML2',\n",
            " 'TREML4',\n",
            " 'TRERF1',\n",
            " 'TREX1',\n",
            " 'TRH',\n",
            " 'TRHDE',\n",
            " 'TRHR',\n",
            " 'TRIAP1',\n",
            " 'TRIB1',\n",
            " 'TRIB2',\n",
            " 'TRIB3',\n",
            " 'TRIL',\n",
            " 'TRIM10',\n",
            " 'TRIM11',\n",
            " 'TRIM13',\n",
            " 'TRIM14',\n",
            " 'TRIM15',\n",
            " 'TRIM16',\n",
            " 'TRIM16L',\n",
            " 'TRIM17',\n",
            " 'TRIM2',\n",
            " 'TRIM21',\n",
            " 'TRIM22',\n",
            " 'TRIM23',\n",
            " 'TRIM24',\n",
            " 'TRIM25',\n",
            " 'TRIM26',\n",
            " 'TRIM27',\n",
            " 'TRIM28',\n",
            " 'TRIM29',\n",
            " 'TRIM3',\n",
            " 'TRIM31',\n",
            " 'TRIM32',\n",
            " 'TRIM33',\n",
            " 'TRIM34',\n",
            " 'TRIM35',\n",
            " 'TRIM36',\n",
            " 'TRIM37',\n",
            " 'TRIM38',\n",
            " 'TRIM39',\n",
            " 'TRIM4',\n",
            " 'TRIM40',\n",
            " 'TRIM41',\n",
            " 'TRIM42',\n",
            " 'TRIM43',\n",
            " 'TRIM44',\n",
            " 'TRIM45',\n",
            " 'TRIM46',\n",
            " 'TRIM47',\n",
            " 'TRIM49',\n",
            " 'TRIM49C',\n",
            " 'TRIM49D1',\n",
            " 'TRIM5',\n",
            " 'TRIM50',\n",
            " 'TRIM51',\n",
            " 'TRIM52',\n",
            " 'TRIM54',\n",
            " 'TRIM55',\n",
            " 'TRIM56',\n",
            " 'TRIM58',\n",
            " 'TRIM59',\n",
            " 'TRIM6',\n",
            " 'TRIM60',\n",
            " 'TRIM61',\n",
            " 'TRIM62',\n",
            " 'TRIM63',\n",
            " 'TRIM65',\n",
            " 'TRIM66',\n",
            " 'TRIM67',\n",
            " 'TRIM68',\n",
            " 'TRIM69',\n",
            " 'TRIM7',\n",
            " 'TRIM71',\n",
            " 'TRIM72',\n",
            " 'TRIM73',\n",
            " 'TRIM74',\n",
            " 'TRIM77',\n",
            " 'TRIM8',\n",
            " 'TRIM9',\n",
            " 'TRIML1',\n",
            " 'TRIML2',\n",
            " 'TRIO',\n",
            " 'TRIOBP',\n",
            " 'TRIP10',\n",
            " 'TRIP11',\n",
            " 'TRIP12',\n",
            " 'TRIP13',\n",
            " 'TRIP4',\n",
            " 'TRIP6',\n",
            " 'TRIQK',\n",
            " 'TRIR',\n",
            " 'TRIT1',\n",
            " 'TRMO',\n",
            " 'TRMT1',\n",
            " 'TRMT10A',\n",
            " 'TRMT10B',\n",
            " 'TRMT10C',\n",
            " 'TRMT11',\n",
            " 'TRMT112',\n",
            " 'TRMT12',\n",
            " 'TRMT13',\n",
            " 'TRMT1L',\n",
            " 'TRMT2A',\n",
            " 'TRMT2B',\n",
            " 'TRMT44',\n",
            " 'TRMT5',\n",
            " 'TRMT6',\n",
            " 'TRMT61A',\n",
            " 'TRMT61B',\n",
            " 'TRMT9B',\n",
            " 'TRMU',\n",
            " 'TRNAU1AP',\n",
            " 'TRNP1',\n",
            " 'TRNT1',\n",
            " 'TRO',\n",
            " 'TROAP',\n",
            " 'TRPA1',\n",
            " 'TRPC1',\n",
            " 'TRPC3',\n",
            " 'TRPC4',\n",
            " 'TRPC4AP',\n",
            " 'TRPC5',\n",
            " 'TRPC5OS',\n",
            " 'TRPC6',\n",
            " 'TRPC7',\n",
            " 'TRPM1',\n",
            " 'TRPM2',\n",
            " 'TRPM3',\n",
            " 'TRPM4',\n",
            " 'TRPM5',\n",
            " 'TRPM6',\n",
            " 'TRPM7',\n",
            " 'TRPM8',\n",
            " 'TRPS1',\n",
            " 'TRPT1',\n",
            " 'TRPV1',\n",
            " 'TRPV2',\n",
            " 'TRPV3',\n",
            " 'TRPV4',\n",
            " 'TRPV5',\n",
            " 'TRPV6',\n",
            " 'TRRAP',\n",
            " 'TRUB1',\n",
            " 'TRUB2',\n",
            " 'TSACC',\n",
            " 'TSBP1',\n",
            " 'TSC1',\n",
            " 'TSC2',\n",
            " 'TSC22D1',\n",
            " 'TSC22D2',\n",
            " 'TSC22D3',\n",
            " 'TSC22D4',\n",
            " 'TSEN15',\n",
            " 'TSEN2',\n",
            " 'TSEN34',\n",
            " 'TSEN54',\n",
            " 'TSFM',\n",
            " 'TSG101',\n",
            " 'TSGA10',\n",
            " 'TSGA10IP',\n",
            " 'TSGA13',\n",
            " 'TSHB',\n",
            " 'TSHR',\n",
            " 'TSHZ1',\n",
            " 'TSHZ2',\n",
            " 'TSHZ3',\n",
            " 'TSKS',\n",
            " 'TSKU',\n",
            " 'TSLP',\n",
            " 'TSN',\n",
            " 'TSNARE1',\n",
            " 'TSNAX',\n",
            " 'TSNAXIP1',\n",
            " 'TSPAN1',\n",
            " 'TSPAN10',\n",
            " 'TSPAN11',\n",
            " 'TSPAN12',\n",
            " 'TSPAN13',\n",
            " 'TSPAN14',\n",
            " 'TSPAN15',\n",
            " 'TSPAN16',\n",
            " 'TSPAN17',\n",
            " 'TSPAN18',\n",
            " 'TSPAN19',\n",
            " 'TSPAN2',\n",
            " 'TSPAN3',\n",
            " 'TSPAN31',\n",
            " 'TSPAN32',\n",
            " 'TSPAN33',\n",
            " 'TSPAN4',\n",
            " 'TSPAN5',\n",
            " 'TSPAN6',\n",
            " 'TSPAN7',\n",
            " 'TSPAN8',\n",
            " 'TSPAN9',\n",
            " 'TSPEAR',\n",
            " 'TSPO',\n",
            " 'TSPO2',\n",
            " 'TSPOAP1',\n",
            " 'TSPY1',\n",
            " 'TSPYL1',\n",
            " 'TSPYL2',\n",
            " 'TSPYL4',\n",
            " 'TSPYL5',\n",
            " 'TSPYL6',\n",
            " 'TSR1',\n",
            " 'TSR2',\n",
            " 'TSR3',\n",
            " 'TSSC4',\n",
            " 'TSSK1B',\n",
            " 'TSSK2',\n",
            " 'TSSK3',\n",
            " 'TSSK4',\n",
            " 'TSSK6',\n",
            " 'TST',\n",
            " 'TSTD1',\n",
            " 'TSTD2',\n",
            " 'TSTD3',\n",
            " 'TTBK1',\n",
            " 'TTBK2',\n",
            " 'TTC1',\n",
            " 'TTC12',\n",
            " 'TTC13',\n",
            " 'TTC14',\n",
            " 'TTC16',\n",
            " 'TTC17',\n",
            " 'TTC19',\n",
            " 'TTC21A',\n",
            " 'TTC21B',\n",
            " 'TTC22',\n",
            " 'TTC23',\n",
            " 'TTC23L',\n",
            " 'TTC24',\n",
            " 'TTC27',\n",
            " 'TTC28',\n",
            " 'TTC29',\n",
            " 'TTC3',\n",
            " 'TTC31',\n",
            " 'TTC32',\n",
            " 'TTC33',\n",
            " 'TTC34',\n",
            " 'TTC36',\n",
            " 'TTC38',\n",
            " 'TTC39A',\n",
            " 'TTC39B',\n",
            " 'TTC39C',\n",
            " 'TTC4',\n",
            " 'TTC5',\n",
            " 'TTC6',\n",
            " 'TTC7B',\n",
            " 'TTC8',\n",
            " 'TTC9',\n",
            " 'TTC9B',\n",
            " 'TTC9C',\n",
            " 'TTF1',\n",
            " 'TTF2',\n",
            " 'TTI1',\n",
            " 'TTI2',\n",
            " 'TTK',\n",
            " 'TTL',\n",
            " 'TTLL1',\n",
            " 'TTLL10',\n",
            " 'TTLL11',\n",
            " 'TTLL12',\n",
            " 'TTLL13',\n",
            " 'TTLL2',\n",
            " 'TTLL3',\n",
            " 'TTLL4',\n",
            " 'TTLL5',\n",
            " 'TTLL6',\n",
            " 'TTLL7',\n",
            " 'TTLL8',\n",
            " 'TTLL9',\n",
            " 'TTN',\n",
            " 'TTPA',\n",
            " 'TTPAL',\n",
            " 'TTR',\n",
            " 'TTYH1',\n",
            " 'TTYH2',\n",
            " 'TTYH3',\n",
            " 'TUB',\n",
            " 'TUBA1A',\n",
            " 'TUBA1B',\n",
            " 'TUBA1C',\n",
            " 'TUBA3C',\n",
            " 'TUBA3D',\n",
            " 'TUBA3E',\n",
            " 'TUBA4A',\n",
            " 'TUBA8',\n",
            " 'TUBAL3',\n",
            " 'TUBB',\n",
            " 'TUBB1',\n",
            " 'TUBB2A',\n",
            " 'TUBB2B',\n",
            " 'TUBB3',\n",
            " 'TUBB4A',\n",
            " 'TUBB4B',\n",
            " 'TUBB6',\n",
            " 'TUBB8',\n",
            " 'TUBD1',\n",
            " 'TUBE1',\n",
            " 'TUBG1',\n",
            " 'TUBG2',\n",
            " 'TUBGCP2',\n",
            " 'TUBGCP3',\n",
            " 'TUBGCP4',\n",
            " 'TUBGCP5',\n",
            " 'TUBGCP6',\n",
            " 'TUFM',\n",
            " 'TUFT1',\n",
            " 'TULP1',\n",
            " 'TULP2',\n",
            " 'TULP3',\n",
            " 'TULP4',\n",
            " 'TUSC1',\n",
            " 'TUSC2',\n",
            " 'TUSC3',\n",
            " 'TUT1',\n",
            " 'TUT4',\n",
            " 'TUT7',\n",
            " 'TVP23A',\n",
            " 'TVP23B',\n",
            " 'TVP23C',\n",
            " 'TWF1',\n",
            " 'TWF2',\n",
            " 'TWIST1',\n",
            " 'TWIST2',\n",
            " 'TWNK',\n",
            " 'TWSG1',\n",
            " 'TXK',\n",
            " 'TXLNA',\n",
            " 'TXLNB',\n",
            " 'TXLNG',\n",
            " 'TXN',\n",
            " 'TXN2',\n",
            " 'TXNDC11',\n",
            " 'TXNDC12',\n",
            " 'TXNDC15',\n",
            " 'TXNDC16',\n",
            " 'TXNDC17',\n",
            " 'TXNDC2',\n",
            " 'TXNDC5',\n",
            " 'TXNDC8',\n",
            " 'TXNDC9',\n",
            " 'TXNIP',\n",
            " 'TXNL1',\n",
            " 'TXNL4A',\n",
            " 'TXNL4B',\n",
            " 'TXNRD1',\n",
            " 'TXNRD2',\n",
            " 'TXNRD3',\n",
            " 'TYK2',\n",
            " 'TYMP',\n",
            " 'TYR',\n",
            " 'TYRO3',\n",
            " 'TYROBP',\n",
            " 'TYRP1',\n",
            " 'TYSND1',\n",
            " 'TYW1',\n",
            " 'TYW1B',\n",
            " 'TYW3',\n",
            " 'TYW5',\n",
            " 'U2AF1',\n",
            " 'U2AF1L4',\n",
            " 'U2AF2',\n",
            " 'U2SURP',\n",
            " 'UACA',\n",
            " 'UAP1',\n",
            " 'UAP1L1',\n",
            " 'UBA1',\n",
            " 'UBA2',\n",
            " 'UBA3',\n",
            " 'UBA5',\n",
            " 'UBA52',\n",
            " 'UBA6',\n",
            " 'UBA7',\n",
            " 'UBAC1',\n",
            " 'UBAC2',\n",
            " 'UBALD1',\n",
            " 'UBALD2',\n",
            " 'UBAP1',\n",
            " 'UBAP1L',\n",
            " 'UBAP2',\n",
            " 'UBAP2L',\n",
            " 'UBASH3A',\n",
            " 'UBASH3B',\n",
            " 'UBB',\n",
            " 'UBC',\n",
            " 'UBD',\n",
            " 'UBE2A',\n",
            " 'UBE2B',\n",
            " 'UBE2C',\n",
            " 'UBE2D1',\n",
            " 'UBE2D2',\n",
            " 'UBE2D3',\n",
            " 'UBE2D4',\n",
            " 'UBE2E1',\n",
            " 'UBE2E2',\n",
            " 'UBE2E3',\n",
            " 'UBE2F',\n",
            " 'UBE2G1',\n",
            " 'UBE2G2',\n",
            " 'UBE2H',\n",
            " 'UBE2I',\n",
            " 'UBE2J1',\n",
            " 'UBE2J2',\n",
            " 'UBE2K',\n",
            " 'UBE2L3',\n",
            " 'UBE2L6',\n",
            " 'UBE2M',\n",
            " 'UBE2N',\n",
            " 'UBE2NL',\n",
            " 'UBE2O',\n",
            " 'UBE2Q1',\n",
            " 'UBE2Q2',\n",
            " 'UBE2QL1',\n",
            " 'UBE2R2',\n",
            " 'UBE2S',\n",
            " 'UBE2T',\n",
            " 'UBE2U',\n",
            " 'UBE2V1',\n",
            " 'UBE2V2',\n",
            " 'UBE2W',\n",
            " 'UBE2Z',\n",
            " 'UBE3A',\n",
            " 'UBE3B',\n",
            " 'UBE3C',\n",
            " 'UBE3D',\n",
            " 'UBE4A',\n",
            " 'UBE4B',\n",
            " 'UBFD1',\n",
            " 'UBIAD1',\n",
            " 'UBL3',\n",
            " 'UBL4A',\n",
            " 'UBL4B',\n",
            " 'UBL5',\n",
            " 'UBL7',\n",
            " 'UBLCP1',\n",
            " 'UBN1',\n",
            " 'UBN2',\n",
            " 'UBP1',\n",
            " 'UBQLN1',\n",
            " 'UBQLN2',\n",
            " 'UBQLN3',\n",
            " 'UBQLN4',\n",
            " 'UBQLNL',\n",
            " 'UBR1',\n",
            " 'UBR2',\n",
            " 'UBR3',\n",
            " 'UBR4',\n",
            " 'UBR5',\n",
            " 'UBR7',\n",
            " 'UBTD1',\n",
            " 'UBTD2',\n",
            " 'UBTF',\n",
            " 'UBXN1',\n",
            " 'UBXN10',\n",
            " 'UBXN11',\n",
            " 'UBXN2A',\n",
            " 'UBXN2B',\n",
            " 'UBXN4',\n",
            " 'UBXN6',\n",
            " 'UBXN7',\n",
            " 'UBXN8',\n",
            " 'UCHL1',\n",
            " 'UCHL3',\n",
            " 'UCHL5',\n",
            " 'UCK1',\n",
            " 'UCK2',\n",
            " 'UCKL1',\n",
            " 'UCMA',\n",
            " 'UCN',\n",
            " 'UCN2',\n",
            " 'UCN3',\n",
            " 'UCP1',\n",
            " 'UCP2',\n",
            " 'UCP3',\n",
            " 'UEVLD',\n",
            " 'UFC1',\n",
            " 'UFD1',\n",
            " 'UFL1',\n",
            " 'UFM1',\n",
            " 'UFSP1',\n",
            " 'UFSP2',\n",
            " 'UGCG',\n",
            " 'UGDH',\n",
            " 'UGGT1',\n",
            " 'UGGT2',\n",
            " 'UGP2',\n",
            " 'UGT1A1',\n",
            " 'UGT1A10',\n",
            " 'UGT1A3',\n",
            " 'UGT1A4',\n",
            " 'UGT1A5',\n",
            " 'UGT1A6',\n",
            " 'UGT1A7',\n",
            " 'UGT1A8',\n",
            " 'UGT2A3',\n",
            " 'UGT2B15',\n",
            " 'UGT2B17',\n",
            " 'UGT2B4',\n",
            " 'UGT2B7',\n",
            " 'UGT3A1',\n",
            " 'UGT3A2',\n",
            " 'UGT8',\n",
            " 'UHMK1',\n",
            " 'UHRF1',\n",
            " 'UHRF2',\n",
            " 'UIMC1',\n",
            " 'ULBP1',\n",
            " 'ULBP2',\n",
            " 'ULBP3',\n",
            " 'ULK1',\n",
            " 'ULK2',\n",
            " 'ULK3',\n",
            " 'ULK4',\n",
            " 'UMOD',\n",
            " 'UMODL1',\n",
            " 'UMPS',\n",
            " 'UNC119',\n",
            " 'UNC119B',\n",
            " 'UNC13A',\n",
            " 'UNC13B',\n",
            " 'UNC13C',\n",
            " 'UNC13D',\n",
            " 'UNC45A',\n",
            " 'UNC45B',\n",
            " 'UNC50',\n",
            " 'UNC5A',\n",
            " 'UNC5B',\n",
            " 'UNC5C',\n",
            " 'UNC5CL',\n",
            " 'UNC5D',\n",
            " 'UNC79',\n",
            " 'UNC80',\n",
            " 'UNC93A',\n",
            " 'UNC93B1',\n",
            " 'UNCX',\n",
            " 'UNG',\n",
            " 'UNK',\n",
            " 'UNKL',\n",
            " 'UPB1',\n",
            " 'UPF1',\n",
            " 'UPF2',\n",
            " 'UPF3A',\n",
            " 'UPF3B',\n",
            " 'UPK1A',\n",
            " 'UPK1B',\n",
            " 'UPK2',\n",
            " 'UPK3A',\n",
            " 'UPK3B',\n",
            " 'UPK3BL1',\n",
            " 'UPP1',\n",
            " 'UPP2',\n",
            " 'UPRT',\n",
            " 'UQCC1',\n",
            " 'UQCC2',\n",
            " 'UQCC3',\n",
            " 'UQCC4',\n",
            " 'UQCC5',\n",
            " 'UQCC6',\n",
            " 'UQCR10',\n",
            " 'UQCR11',\n",
            " 'UQCRB',\n",
            " 'UQCRC1',\n",
            " 'UQCRC2',\n",
            " 'UQCRFS1',\n",
            " 'UQCRH',\n",
            " 'UQCRHL',\n",
            " 'UQCRQ',\n",
            " 'URAD',\n",
            " 'URB1',\n",
            " 'URB2',\n",
            " 'URGCP',\n",
            " 'URI1',\n",
            " 'URM1',\n",
            " 'UROC1',\n",
            " 'UROD',\n",
            " 'UROS',\n",
            " 'USB1',\n",
            " 'USE1',\n",
            " 'USF1',\n",
            " 'USF2',\n",
            " 'USF3',\n",
            " 'USH1C',\n",
            " 'USH1G',\n",
            " 'USH2A',\n",
            " 'USHBP1',\n",
            " 'USO1',\n",
            " 'USP1',\n",
            " 'USP10',\n",
            " 'USP11',\n",
            " 'USP12',\n",
            " 'USP13',\n",
            " 'USP14',\n",
            " 'USP15',\n",
            " 'USP16',\n",
            " 'USP18',\n",
            " 'USP19',\n",
            " 'USP2',\n",
            " 'USP20',\n",
            " 'USP21',\n",
            " 'USP22',\n",
            " 'USP24',\n",
            " 'USP25',\n",
            " 'USP26',\n",
            " 'USP27X',\n",
            " 'USP28',\n",
            " 'USP29',\n",
            " 'USP3',\n",
            " 'USP30',\n",
            " 'USP31',\n",
            " 'USP32',\n",
            " 'USP33',\n",
            " 'USP34',\n",
            " 'USP35',\n",
            " 'USP36',\n",
            " 'USP37',\n",
            " 'USP38',\n",
            " 'USP39',\n",
            " 'USP4',\n",
            " 'USP40',\n",
            " 'USP42',\n",
            " 'USP43',\n",
            " 'USP44',\n",
            " 'USP45',\n",
            " 'USP46',\n",
            " 'USP47',\n",
            " 'USP48',\n",
            " 'USP49',\n",
            " 'USP5',\n",
            " 'USP50',\n",
            " 'USP51',\n",
            " 'USP53',\n",
            " 'USP54',\n",
            " 'USP6',\n",
            " 'USP6NL',\n",
            " 'USP7',\n",
            " 'USP8',\n",
            " 'USP9X',\n",
            " 'USP9Y',\n",
            " 'USPL1',\n",
            " 'UST',\n",
            " 'UTF1',\n",
            " 'UTP11',\n",
            " 'UTP14A',\n",
            " 'UTP15',\n",
            " 'UTP18',\n",
            " 'UTP20',\n",
            " 'UTP23',\n",
            " 'UTP25',\n",
            " 'UTP3',\n",
            " 'UTP4',\n",
            " 'UTP6',\n",
            " 'UTRN',\n",
            " 'UTS2',\n",
            " 'UTS2B',\n",
            " 'UTS2R',\n",
            " 'UTY',\n",
            " 'UVRAG',\n",
            " 'UVSSA',\n",
            " 'UXS1',\n",
            " 'UXT',\n",
            " 'VAC14',\n",
            " 'VAMP1',\n",
            " 'VAMP2',\n",
            " 'VAMP3',\n",
            " 'VAMP4',\n",
            " 'VAMP5',\n",
            " 'VAMP7',\n",
            " 'VAMP8',\n",
            " 'VANGL1',\n",
            " 'VANGL2',\n",
            " 'VAPA',\n",
            " 'VAPB',\n",
            " 'VARS1',\n",
            " 'VARS2',\n",
            " 'VASH1',\n",
            " 'VASH2',\n",
            " 'VASN',\n",
            " 'VASP',\n",
            " 'VAT1',\n",
            " 'VAT1L',\n",
            " 'VAV1',\n",
            " 'VAV2',\n",
            " 'VAV3',\n",
            " 'VAX1',\n",
            " 'VAX2',\n",
            " 'VBP1',\n",
            " 'VCAM1',\n",
            " 'VCAN',\n",
            " 'VCL',\n",
            " 'VCP',\n",
            " 'VCPIP1',\n",
            " 'VCPKMT',\n",
            " 'VCX',\n",
            " 'VDAC1',\n",
            " 'VDAC2',\n",
            " 'VDAC3',\n",
            " 'VDR',\n",
            " 'VEGFA',\n",
            " 'VEGFB',\n",
            " 'VEGFC',\n",
            " 'VEGFD',\n",
            " 'VENTX',\n",
            " 'VEPH1',\n",
            " 'VEZF1',\n",
            " 'VEZT',\n",
            " 'VGF',\n",
            " 'VGLL1',\n",
            " 'VGLL2',\n",
            " 'VGLL3',\n",
            " 'VGLL4',\n",
            " 'VHL',\n",
            " 'VHLL',\n",
            " 'VIL1',\n",
            " 'VILL',\n",
            " 'VIM',\n",
            " 'VIP',\n",
            " 'VIPAS39',\n",
            " 'VIPR1',\n",
            " 'VIPR2',\n",
            " 'VIRMA',\n",
            " 'VIT',\n",
            " 'VKORC1',\n",
            " 'VKORC1L1',\n",
            " 'VLDLR',\n",
            " 'VMA21',\n",
            " 'VMAC',\n",
            " 'VMO1',\n",
            " 'VMP1',\n",
            " 'VN1R1',\n",
            " 'VN1R2',\n",
            " 'VN1R4',\n",
            " 'VNN1',\n",
            " 'VNN2',\n",
            " 'VOPP1',\n",
            " 'VPREB1',\n",
            " 'VPREB3',\n",
            " 'VPS11',\n",
            " 'VPS13A',\n",
            " 'VPS13B',\n",
            " 'VPS13C',\n",
            " 'VPS13D',\n",
            " 'VPS16',\n",
            " 'VPS18',\n",
            " 'VPS25',\n",
            " 'VPS26A',\n",
            " 'VPS26B',\n",
            " 'VPS26C',\n",
            " 'VPS28',\n",
            " 'VPS33A',\n",
            " 'VPS33B',\n",
            " 'VPS35',\n",
            " 'VPS35L',\n",
            " 'VPS36',\n",
            " 'VPS37A',\n",
            " 'VPS37B',\n",
            " 'VPS37C',\n",
            " 'VPS37D',\n",
            " 'VPS39',\n",
            " 'VPS41',\n",
            " 'VPS45',\n",
            " 'VPS4A',\n",
            " 'VPS4B',\n",
            " 'VPS50',\n",
            " 'VPS51',\n",
            " 'VPS52',\n",
            " 'VPS53',\n",
            " 'VPS54',\n",
            " 'VPS72',\n",
            " 'VPS8',\n",
            " 'VPS9D1',\n",
            " 'VRK1',\n",
            " 'VRK2',\n",
            " 'VRK3',\n",
            " 'VRTN',\n",
            " 'VSIG1',\n",
            " 'VSIG10',\n",
            " 'VSIG10L',\n",
            " 'VSIG2',\n",
            " 'VSIG4',\n",
            " 'VSIG8',\n",
            " 'VSIR',\n",
            " 'VSNL1',\n",
            " 'VSTM1',\n",
            " 'VSTM2A',\n",
            " 'VSTM2B',\n",
            " 'VSTM2L',\n",
            " 'VSTM4',\n",
            " 'VSTM5',\n",
            " 'VSX1',\n",
            " 'VSX2',\n",
            " 'VTA1',\n",
            " 'VTCN1',\n",
            " 'VTI1A',\n",
            " 'VTI1B',\n",
            " 'VWA1',\n",
            " 'VWA2',\n",
            " 'VWA3A',\n",
            " 'VWA3B',\n",
            " 'VWA5A',\n",
            " 'VWA5B1',\n",
            " 'VWA5B2',\n",
            " 'VWA7',\n",
            " 'VWA8',\n",
            " 'VWC2',\n",
            " 'VWC2L',\n",
            " 'VWCE',\n",
            " 'VWDE',\n",
            " 'VWF',\n",
            " 'VXN',\n",
            " 'WAC',\n",
            " 'WAPL',\n",
            " 'WARS1',\n",
            " 'WARS2',\n",
            " 'WAS',\n",
            " 'WASF1',\n",
            " 'WASF2',\n",
            " 'WASF3',\n",
            " 'WASHC2A',\n",
            " 'WASHC2C',\n",
            " 'WASHC3',\n",
            " 'WASHC4',\n",
            " 'WASHC5',\n",
            " 'WASL',\n",
            " 'WBP1',\n",
            " 'WBP11',\n",
            " 'WBP1L',\n",
            " 'WBP2',\n",
            " 'WBP2NL',\n",
            " 'WBP4',\n",
            " 'WDCP',\n",
            " 'WDFY1',\n",
            " 'WDFY2',\n",
            " 'WDFY3',\n",
            " 'WDFY4',\n",
            " 'WDHD1',\n",
            " 'WDPCP',\n",
            " 'WDR1',\n",
            " 'WDR11',\n",
            " 'WDR12',\n",
            " 'WDR13',\n",
            " 'WDR17',\n",
            " 'WDR18',\n",
            " 'WDR19',\n",
            " 'WDR20',\n",
            " 'WDR24',\n",
            " 'WDR25',\n",
            " 'WDR26',\n",
            " 'WDR27',\n",
            " 'WDR3',\n",
            " 'WDR31',\n",
            " 'WDR33',\n",
            " 'WDR35',\n",
            " 'WDR36',\n",
            " 'WDR37',\n",
            " 'WDR38',\n",
            " 'WDR4',\n",
            " 'WDR41',\n",
            " 'WDR43',\n",
            " 'WDR44',\n",
            " 'WDR45B',\n",
            " 'WDR46',\n",
            " 'WDR47',\n",
            " 'WDR48',\n",
            " 'WDR49',\n",
            " 'WDR5',\n",
            " 'WDR53',\n",
            " 'WDR54',\n",
            " 'WDR55',\n",
            " 'WDR59',\n",
            " 'WDR5B',\n",
            " 'WDR6',\n",
            " 'WDR62',\n",
            " 'WDR64',\n",
            " 'WDR7',\n",
            " 'WDR70',\n",
            " 'WDR72',\n",
            " 'WDR73',\n",
            " 'WDR74',\n",
            " 'WDR75',\n",
            " 'WDR76',\n",
            " 'WDR77',\n",
            " 'WDR81',\n",
            " 'WDR82',\n",
            " 'WDR83',\n",
            " 'WDR83OS',\n",
            " 'WDR86',\n",
            " 'WDR87',\n",
            " 'WDR88',\n",
            " 'WDR89',\n",
            " 'WDR90',\n",
            " 'WDR91',\n",
            " 'WDR93',\n",
            " 'WDSUB1',\n",
            " 'WDTC1',\n",
            " 'WEE2',\n",
            " 'WFDC1',\n",
            " 'WFDC10A',\n",
            " 'WFDC10B',\n",
            " 'WFDC11',\n",
            " 'WFDC12',\n",
            " 'WFDC13',\n",
            " 'WFDC2',\n",
            " 'WFDC3',\n",
            " 'WFDC5',\n",
            " 'WFDC6',\n",
            " 'WFDC8',\n",
            " 'WFDC9',\n",
            " 'WFIKKN1',\n",
            " 'WFIKKN2',\n",
            " 'WFS1',\n",
            " 'WHAMM',\n",
            " 'WHRN',\n",
            " 'WIF1',\n",
            " 'WIPF1',\n",
            " 'WIPF2',\n",
            " 'WIPF3',\n",
            " 'WIPI1',\n",
            " 'WIPI2',\n",
            " 'WIZ',\n",
            " 'WLS',\n",
            " 'WNK1',\n",
            " 'WNK2',\n",
            " 'WNK3',\n",
            " 'WNK4',\n",
            " 'WNT1',\n",
            " 'WNT10A',\n",
            " 'WNT10B',\n",
            " 'WNT11',\n",
            " 'WNT16',\n",
            " 'WNT2',\n",
            " 'WNT2B',\n",
            " 'WNT3',\n",
            " 'WNT3A',\n",
            " 'WNT4',\n",
            " 'WNT5A',\n",
            " 'WNT5B',\n",
            " 'WNT6',\n",
            " 'WNT7A',\n",
            " 'WNT7B',\n",
            " 'WNT8A',\n",
            " 'WNT8B',\n",
            " 'WNT9A',\n",
            " 'WNT9B',\n",
            " 'WRAP53',\n",
            " 'WRAP73',\n",
            " 'WRN',\n",
            " 'WRNIP1',\n",
            " 'WSB1',\n",
            " 'WSB2',\n",
            " 'WSCD1',\n",
            " 'WSCD2',\n",
            " 'WT1',\n",
            " 'WTAP',\n",
            " 'WTIP',\n",
            " 'WWC1',\n",
            " 'WWC2',\n",
            " 'WWC3',\n",
            " 'WWOX',\n",
            " 'WWP1',\n",
            " 'WWP2',\n",
            " 'WWTR1',\n",
            " 'XAB2',\n",
            " 'XAF1',\n",
            " 'XAGE1B',\n",
            " 'XAGE2',\n",
            " 'XAGE3',\n",
            " 'XAGE5',\n",
            " 'XBP1',\n",
            " 'XCL1',\n",
            " 'XCL2',\n",
            " 'XCR1',\n",
            " 'XDH',\n",
            " 'XG',\n",
            " 'XIAP',\n",
            " 'XIRP1',\n",
            " 'XIRP2',\n",
            " 'XK',\n",
            " 'XKR3',\n",
            " 'XKR4',\n",
            " 'XKR5',\n",
            " 'XKR6',\n",
            " 'XKR7',\n",
            " 'XKR8',\n",
            " 'XKR9',\n",
            " 'XKRX',\n",
            " 'XPA',\n",
            " 'XPC',\n",
            " 'XPNPEP1',\n",
            " 'XPNPEP2',\n",
            " 'XPNPEP3',\n",
            " 'XPO1',\n",
            " 'XPO4',\n",
            " 'XPO5',\n",
            " 'XPO6',\n",
            " 'XPO7',\n",
            " 'XPOT',\n",
            " 'XPR1',\n",
            " 'XRCC1',\n",
            " 'XRCC2',\n",
            " 'XRCC3',\n",
            " 'XRCC4',\n",
            " 'XRCC5',\n",
            " 'XRCC6',\n",
            " 'XRN1',\n",
            " 'XRN2',\n",
            " 'XRRA1',\n",
            " 'XXYLT1',\n",
            " 'XYLB',\n",
            " 'XYLT1',\n",
            " 'XYLT2',\n",
            " 'YAE1',\n",
            " 'YAF2',\n",
            " 'YAP1',\n",
            " 'YARS1',\n",
            " 'YARS2',\n",
            " 'YBEY',\n",
            " 'YBX1',\n",
            " 'YBX2',\n",
            " 'YBX3',\n",
            " 'YDJC',\n",
            " 'YEATS2',\n",
            " 'YEATS4',\n",
            " 'YES1',\n",
            " 'YIF1A',\n",
            " 'YIF1B',\n",
            " 'YIPF1',\n",
            " 'YIPF2',\n",
            " 'YIPF3',\n",
            " 'YIPF4',\n",
            " 'YIPF5',\n",
            " 'YIPF6',\n",
            " 'YIPF7',\n",
            " 'YJEFN3',\n",
            " 'YJU2',\n",
            " 'YJU2B',\n",
            " 'YKT6',\n",
            " 'YLPM1',\n",
            " 'YME1L1',\n",
            " 'YOD1',\n",
            " 'YPEL1',\n",
            " 'YPEL2',\n",
            " 'YPEL3',\n",
            " 'YPEL4',\n",
            " 'YPEL5',\n",
            " 'YRDC',\n",
            " 'YTHDC1',\n",
            " 'YTHDC2',\n",
            " 'YTHDF1',\n",
            " 'YTHDF2',\n",
            " 'YTHDF3',\n",
            " 'YWHAB',\n",
            " 'YWHAE',\n",
            " 'YWHAG',\n",
            " 'YWHAH',\n",
            " 'YWHAH-AS1',\n",
            " 'YWHAQ',\n",
            " 'YWHAZ',\n",
            " 'YY1',\n",
            " 'YY1AP1',\n",
            " 'YY2',\n",
            " 'ZACN',\n",
            " 'ZAN',\n",
            " 'ZAP70',\n",
            " 'ZAR1',\n",
            " 'ZAR1L',\n",
            " 'ZBBX',\n",
            " 'ZBED2',\n",
            " 'ZBED3',\n",
            " 'ZBED4',\n",
            " 'ZBED5',\n",
            " 'ZBED6',\n",
            " 'ZBP1',\n",
            " 'ZBTB1',\n",
            " 'ZBTB10',\n",
            " 'ZBTB11',\n",
            " 'ZBTB12',\n",
            " 'ZBTB14',\n",
            " 'ZBTB16',\n",
            " 'ZBTB17',\n",
            " 'ZBTB18',\n",
            " 'ZBTB2',\n",
            " 'ZBTB20',\n",
            " 'ZBTB21',\n",
            " 'ZBTB22',\n",
            " 'ZBTB24',\n",
            " 'ZBTB25',\n",
            " 'ZBTB26',\n",
            " 'ZBTB3',\n",
            " 'ZBTB32',\n",
            " 'ZBTB33',\n",
            " 'ZBTB34',\n",
            " 'ZBTB37',\n",
            " 'ZBTB38',\n",
            " 'ZBTB39',\n",
            " 'ZBTB4',\n",
            " 'ZBTB40',\n",
            " 'ZBTB41',\n",
            " 'ZBTB42',\n",
            " 'ZBTB43',\n",
            " 'ZBTB44',\n",
            " 'ZBTB45',\n",
            " 'ZBTB46',\n",
            " 'ZBTB47',\n",
            " 'ZBTB48',\n",
            " 'ZBTB49',\n",
            " 'ZBTB5',\n",
            " 'ZBTB6',\n",
            " 'ZBTB7A',\n",
            " 'ZBTB7B',\n",
            " 'ZBTB7C',\n",
            " 'ZBTB8A',\n",
            " 'ZBTB8B',\n",
            " 'ZBTB8OS',\n",
            " 'ZBTB9',\n",
            " 'ZC2HC1A',\n",
            " 'ZC2HC1B',\n",
            " 'ZC2HC1C',\n",
            " 'ZC3H12A',\n",
            " 'ZC3H12B',\n",
            " 'ZC3H12C',\n",
            " 'ZC3H12D',\n",
            " 'ZC3H13',\n",
            " 'ZC3H14',\n",
            " 'ZC3H15',\n",
            " 'ZC3H18',\n",
            " 'ZC3H3',\n",
            " 'ZC3H4',\n",
            " 'ZC3H6',\n",
            " 'ZC3H7A',\n",
            " 'ZC3H7B',\n",
            " 'ZC3H8',\n",
            " 'ZC3HAV1',\n",
            " 'ZC3HAV1L',\n",
            " 'ZC3HC1',\n",
            " 'ZC4H2',\n",
            " 'ZCCHC10',\n",
            " 'ZCCHC12',\n",
            " 'ZCCHC13',\n",
            " 'ZCCHC14',\n",
            " 'ZCCHC17',\n",
            " 'ZCCHC18',\n",
            " 'ZCCHC2',\n",
            " 'ZCCHC24',\n",
            " 'ZCCHC3',\n",
            " 'ZCCHC4',\n",
            " 'ZCCHC7',\n",
            " 'ZCCHC8',\n",
            " 'ZCCHC9',\n",
            " 'ZCRB1',\n",
            " 'ZCWPW1',\n",
            " 'ZCWPW2',\n",
            " 'ZDBF2',\n",
            " 'ZDHHC1',\n",
            " 'ZDHHC11',\n",
            " 'ZDHHC11B',\n",
            " 'ZDHHC12',\n",
            " 'ZDHHC13',\n",
            " 'ZDHHC14',\n",
            " 'ZDHHC15',\n",
            " 'ZDHHC16',\n",
            " 'ZDHHC17',\n",
            " 'ZDHHC18',\n",
            " 'ZDHHC19',\n",
            " 'ZDHHC2',\n",
            " 'ZDHHC20',\n",
            " 'ZDHHC21',\n",
            " 'ZDHHC22',\n",
            " 'ZDHHC23',\n",
            " 'ZDHHC24',\n",
            " 'ZDHHC3',\n",
            " 'ZDHHC4',\n",
            " 'ZDHHC5',\n",
            " 'ZDHHC6',\n",
            " 'ZDHHC7',\n",
            " 'ZDHHC8',\n",
            " 'ZDHHC9',\n",
            " 'ZEB1',\n",
            " 'ZEB2',\n",
            " 'ZER1',\n",
            " 'ZFAND1',\n",
            " 'ZFAND2A',\n",
            " 'ZFAND2B',\n",
            " 'ZFAND3',\n",
            " 'ZFAND4',\n",
            " 'ZFAND5',\n",
            " 'ZFAND6',\n",
            " 'ZFAT',\n",
            " 'ZFC3H1',\n",
            " 'ZFHX2',\n",
            " 'ZFHX3',\n",
            " 'ZFHX4',\n",
            " 'ZFP1',\n",
            " 'ZFP14',\n",
            " 'ZFP2',\n",
            " 'ZFP28',\n",
            " 'ZFP3',\n",
            " 'ZFP30',\n",
            " 'ZFP36',\n",
            " 'ZFP36L1',\n",
            " 'ZFP36L2',\n",
            " 'ZFP37',\n",
            " 'ZFP41',\n",
            " 'ZFP42',\n",
            " 'ZFP57',\n",
            " 'ZFP62',\n",
            " 'ZFP64',\n",
            " 'ZFP69',\n",
            " 'ZFP69B',\n",
            " 'ZFP82',\n",
            " 'ZFP90',\n",
            " 'ZFP91',\n",
            " 'ZFP92',\n",
            " 'ZFPL1',\n",
            " 'ZFPM1',\n",
            " 'ZFPM2',\n",
            " 'ZFR',\n",
            " 'ZFR2',\n",
            " 'ZFTA',\n",
            " 'ZFTRAF1',\n",
            " 'ZFX',\n",
            " 'ZFY',\n",
            " 'ZFYVE1',\n",
            " 'ZFYVE16',\n",
            " 'ZFYVE19',\n",
            " 'ZFYVE21',\n",
            " 'ZFYVE26',\n",
            " 'ZFYVE27',\n",
            " 'ZFYVE28',\n",
            " 'ZFYVE9',\n",
            " 'ZG16',\n",
            " 'ZG16B',\n",
            " 'ZGLP1',\n",
            " 'ZGPAT',\n",
            " 'ZGRF1',\n",
            " 'ZHX1',\n",
            " 'ZHX2',\n",
            " 'ZHX3',\n",
            " 'ZIC1',\n",
            " 'ZIC2',\n",
            " 'ZIC3',\n",
            " 'ZIC4',\n",
            " 'ZIC5',\n",
            " 'ZIK1',\n",
            " 'ZIM2',\n",
            " 'ZIM3',\n",
            " 'ZKSCAN1',\n",
            " 'ZKSCAN2',\n",
            " 'ZKSCAN3',\n",
            " 'ZKSCAN4',\n",
            " 'ZKSCAN5',\n",
            " 'ZKSCAN7',\n",
            " 'ZKSCAN8',\n",
            " 'ZMAT1',\n",
            " 'ZMAT2',\n",
            " 'ZMAT3',\n",
            " 'ZMAT4',\n",
            " 'ZMAT5',\n",
            " 'ZMIZ1',\n",
            " 'ZMIZ2',\n",
            " 'ZMPSTE24',\n",
            " 'ZMYM1',\n",
            " 'ZMYM2',\n",
            " 'ZMYM3',\n",
            " 'ZMYM4',\n",
            " 'ZMYM5',\n",
            " 'ZMYM6',\n",
            " 'ZMYND10',\n",
            " 'ZMYND11',\n",
            " 'ZMYND12',\n",
            " 'ZMYND19',\n",
            " 'ZMYND8',\n",
            " 'ZNF10',\n",
            " 'ZNF100',\n",
            " 'ZNF101',\n",
            " 'ZNF106',\n",
            " 'ZNF107',\n",
            " 'ZNF112',\n",
            " 'ZNF114',\n",
            " 'ZNF117',\n",
            " 'ZNF12',\n",
            " 'ZNF121',\n",
            " 'ZNF124',\n",
            " 'ZNF131',\n",
            " 'ZNF132',\n",
            " 'ZNF133',\n",
            " 'ZNF134',\n",
            " 'ZNF135',\n",
            " 'ZNF136',\n",
            " 'ZNF138',\n",
            " 'ZNF14',\n",
            " 'ZNF140',\n",
            " 'ZNF141',\n",
            " 'ZNF142',\n",
            " 'ZNF143',\n",
            " 'ZNF146',\n",
            " 'ZNF148',\n",
            " 'ZNF154',\n",
            " 'ZNF155',\n",
            " 'ZNF157',\n",
            " 'ZNF16',\n",
            " 'ZNF160',\n",
            " 'ZNF165',\n",
            " 'ZNF169',\n",
            " 'ZNF17',\n",
            " 'ZNF174',\n",
            " 'ZNF175',\n",
            " 'ZNF177',\n",
            " 'ZNF18',\n",
            " 'ZNF180',\n",
            " 'ZNF181',\n",
            " 'ZNF182',\n",
            " 'ZNF184',\n",
            " 'ZNF185',\n",
            " 'ZNF189',\n",
            " 'ZNF19',\n",
            " 'ZNF195',\n",
            " 'ZNF197',\n",
            " 'ZNF2',\n",
            " 'ZNF20',\n",
            " 'ZNF200',\n",
            " 'ZNF202',\n",
            " 'ZNF205',\n",
            " 'ZNF207',\n",
            " 'ZNF208',\n",
            " 'ZNF211',\n",
            " 'ZNF212',\n",
            " 'ZNF213',\n",
            " 'ZNF214',\n",
            " 'ZNF215',\n",
            " 'ZNF217',\n",
            " 'ZNF219',\n",
            " 'ZNF22',\n",
            " 'ZNF22-AS1',\n",
            " 'ZNF221',\n",
            " 'ZNF222',\n",
            " 'ZNF223',\n",
            " 'ZNF224',\n",
            " 'ZNF225',\n",
            " 'ZNF227',\n",
            " 'ZNF229',\n",
            " 'ZNF23',\n",
            " 'ZNF230',\n",
            " 'ZNF232',\n",
            " 'ZNF233',\n",
            " 'ZNF235',\n",
            " 'ZNF236',\n",
            " 'ZNF239',\n",
            " 'ZNF24',\n",
            " 'ZNF248',\n",
            " 'ZNF25',\n",
            " 'ZNF250',\n",
            " 'ZNF251',\n",
            " 'ZNF253',\n",
            " 'ZNF254',\n",
            " 'ZNF256',\n",
            " 'ZNF257',\n",
            " 'ZNF26',\n",
            " 'ZNF260',\n",
            " 'ZNF263',\n",
            " 'ZNF264',\n",
            " 'ZNF266',\n",
            " 'ZNF267',\n",
            " 'ZNF268',\n",
            " 'ZNF273',\n",
            " 'ZNF274',\n",
            " 'ZNF275',\n",
            " 'ZNF276',\n",
            " 'ZNF277',\n",
            " 'ZNF28',\n",
            " 'ZNF280A',\n",
            " 'ZNF280B',\n",
            " 'ZNF280C',\n",
            " 'ZNF280D',\n",
            " 'ZNF281',\n",
            " 'ZNF282',\n",
            " 'ZNF283',\n",
            " 'ZNF284',\n",
            " 'ZNF285',\n",
            " 'ZNF286A',\n",
            " 'ZNF286B',\n",
            " 'ZNF287',\n",
            " 'ZNF292',\n",
            " 'ZNF296',\n",
            " 'ZNF3',\n",
            " 'ZNF30',\n",
            " 'ZNF300',\n",
            " 'ZNF302',\n",
            " 'ZNF304',\n",
            " 'ZNF311',\n",
            " 'ZNF317',\n",
            " 'ZNF318',\n",
            " 'ZNF319',\n",
            " 'ZNF32',\n",
            " 'ZNF320',\n",
            " 'ZNF322',\n",
            " 'ZNF324',\n",
            " 'ZNF324B',\n",
            " 'ZNF326',\n",
            " 'ZNF329',\n",
            " 'ZNF330',\n",
            " 'ZNF331',\n",
            " 'ZNF333',\n",
            " 'ZNF334',\n",
            " 'ZNF335',\n",
            " 'ZNF337',\n",
            " 'ZNF33A',\n",
            " 'ZNF33B',\n",
            " 'ZNF34',\n",
            " 'ZNF341',\n",
            " 'ZNF343',\n",
            " 'ZNF345',\n",
            " 'ZNF346',\n",
            " 'ZNF347',\n",
            " 'ZNF35',\n",
            " 'ZNF350',\n",
            " 'ZNF354A',\n",
            " 'ZNF354B',\n",
            " 'ZNF354C',\n",
            " 'ZNF358',\n",
            " 'ZNF362',\n",
            " 'ZNF365',\n",
            " 'ZNF366',\n",
            " 'ZNF367',\n",
            " 'ZNF37A',\n",
            " 'ZNF382',\n",
            " 'ZNF383',\n",
            " 'ZNF384',\n",
            " 'ZNF385A',\n",
            " 'ZNF385B',\n",
            " 'ZNF385C',\n",
            " 'ZNF385D',\n",
            " 'ZNF391',\n",
            " 'ZNF394',\n",
            " 'ZNF395',\n",
            " 'ZNF396',\n",
            " 'ZNF397',\n",
            " 'ZNF398',\n",
            " 'ZNF404',\n",
            " 'ZNF407',\n",
            " 'ZNF408',\n",
            " 'ZNF41',\n",
            " 'ZNF410',\n",
            " 'ZNF414',\n",
            " 'ZNF415',\n",
            " 'ZNF416',\n",
            " 'ZNF417',\n",
            " 'ZNF418',\n",
            " 'ZNF419',\n",
            " 'ZNF420',\n",
            " 'ZNF423',\n",
            " 'ZNF425',\n",
            " 'ZNF426',\n",
            " 'ZNF428',\n",
            " 'ZNF429',\n",
            " 'ZNF43',\n",
            " 'ZNF430',\n",
            " 'ZNF431',\n",
            " 'ZNF432',\n",
            " 'ZNF433',\n",
            " 'ZNF436',\n",
            " 'ZNF438',\n",
            " 'ZNF439',\n",
            " 'ZNF44',\n",
            " 'ZNF440',\n",
            " 'ZNF441',\n",
            " 'ZNF442',\n",
            " 'ZNF443',\n",
            " 'ZNF444',\n",
            " 'ZNF445',\n",
            " 'ZNF446',\n",
            " 'ZNF449',\n",
            " 'ZNF45',\n",
            " 'ZNF451',\n",
            " 'ZNF454',\n",
            " 'ZNF460',\n",
            " 'ZNF461',\n",
            " 'ZNF462',\n",
            " 'ZNF467',\n",
            " 'ZNF468',\n",
            " 'ZNF469',\n",
            " 'ZNF470',\n",
            " 'ZNF471',\n",
            " 'ZNF473',\n",
            " 'ZNF474',\n",
            " 'ZNF479',\n",
            " 'ZNF48',\n",
            " 'ZNF480',\n",
            " 'ZNF483',\n",
            " 'ZNF484',\n",
            " 'ZNF485',\n",
            " 'ZNF486',\n",
            " 'ZNF488',\n",
            " 'ZNF490',\n",
            " 'ZNF491',\n",
            " 'ZNF492',\n",
            " 'ZNF493',\n",
            " 'ZNF496',\n",
            " 'ZNF497',\n",
            " 'ZNF500',\n",
            " 'ZNF501',\n",
            " 'ZNF502',\n",
            " 'ZNF503',\n",
            " 'ZNF506',\n",
            " 'ZNF507',\n",
            " 'ZNF510',\n",
            " 'ZNF511',\n",
            " 'ZNF512',\n",
            " 'ZNF512B',\n",
            " 'ZNF513',\n",
            " 'ZNF514',\n",
            " 'ZNF516',\n",
            " 'ZNF517',\n",
            " 'ZNF518A',\n",
            " 'ZNF518B',\n",
            " 'ZNF519',\n",
            " 'ZNF521',\n",
            " 'ZNF524',\n",
            " 'ZNF526',\n",
            " 'ZNF527',\n",
            " 'ZNF528',\n",
            " 'ZNF529',\n",
            " 'ZNF530',\n",
            " 'ZNF532',\n",
            " 'ZNF534',\n",
            " 'ZNF536',\n",
            " 'ZNF540',\n",
            " 'ZNF541',\n",
            " 'ZNF543',\n",
            " 'ZNF544',\n",
            " 'ZNF546',\n",
            " 'ZNF547',\n",
            " 'ZNF548',\n",
            " 'ZNF549',\n",
            " 'ZNF550',\n",
            " 'ZNF551',\n",
            " 'ZNF552',\n",
            " 'ZNF554',\n",
            " 'ZNF555',\n",
            " 'ZNF556',\n",
            " 'ZNF557',\n",
            " 'ZNF558',\n",
            " 'ZNF559',\n",
            " 'ZNF560',\n",
            " 'ZNF561',\n",
            " 'ZNF561-AS1',\n",
            " 'ZNF562',\n",
            " 'ZNF563',\n",
            " 'ZNF565',\n",
            " 'ZNF566',\n",
            " 'ZNF567',\n",
            " 'ZNF568',\n",
            " 'ZNF569',\n",
            " 'ZNF57',\n",
            " 'ZNF570',\n",
            " 'ZNF571',\n",
            " 'ZNF572',\n",
            " 'ZNF573',\n",
            " 'ZNF574',\n",
            " 'ZNF575',\n",
            " 'ZNF577',\n",
            " 'ZNF578',\n",
            " 'ZNF579',\n",
            " 'ZNF580',\n",
            " 'ZNF581',\n",
            " 'ZNF582',\n",
            " 'ZNF583',\n",
            " 'ZNF584',\n",
            " 'ZNF585A',\n",
            " 'ZNF585B',\n",
            " 'ZNF586',\n",
            " 'ZNF587',\n",
            " 'ZNF587B',\n",
            " 'ZNF589',\n",
            " 'ZNF592',\n",
            " 'ZNF593',\n",
            " 'ZNF594',\n",
            " 'ZNF595',\n",
            " 'ZNF596',\n",
            " 'ZNF597',\n",
            " 'ZNF598',\n",
            " 'ZNF599',\n",
            " 'ZNF600',\n",
            " 'ZNF605',\n",
            " 'ZNF606',\n",
            " 'ZNF607',\n",
            " 'ZNF608',\n",
            " 'ZNF609',\n",
            " 'ZNF610',\n",
            " 'ZNF611',\n",
            " 'ZNF613',\n",
            " 'ZNF614',\n",
            " 'ZNF615',\n",
            " 'ZNF616',\n",
            " 'ZNF618',\n",
            " 'ZNF619',\n",
            " 'ZNF620',\n",
            " 'ZNF621',\n",
            " 'ZNF622',\n",
            " 'ZNF623',\n",
            " 'ZNF624',\n",
            " 'ZNF626',\n",
            " 'ZNF627',\n",
            " 'ZNF628',\n",
            " 'ZNF629',\n",
            " 'ZNF630',\n",
            " 'ZNF638',\n",
            " 'ZNF639',\n",
            " 'ZNF641',\n",
            " 'ZNF644',\n",
            " 'ZNF646',\n",
            " 'ZNF648',\n",
            " 'ZNF649',\n",
            " 'ZNF652',\n",
            " 'ZNF653',\n",
            " 'ZNF654',\n",
            " 'ZNF655',\n",
            " 'ZNF658',\n",
            " 'ZNF660',\n",
            " 'ZNF662',\n",
            " 'ZNF664',\n",
            " 'ZNF665',\n",
            " 'ZNF667',\n",
            " 'ZNF668',\n",
            " 'ZNF669',\n",
            " 'ZNF670',\n",
            " 'ZNF671',\n",
            " 'ZNF672',\n",
            " 'ZNF674',\n",
            " 'ZNF675',\n",
            " 'ZNF676',\n",
            " 'ZNF677',\n",
            " 'ZNF678',\n",
            " 'ZNF679',\n",
            " 'ZNF680',\n",
            " 'ZNF681',\n",
            " 'ZNF682',\n",
            " 'ZNF683',\n",
            " 'ZNF684',\n",
            " 'ZNF687',\n",
            " 'ZNF688',\n",
            " 'ZNF689',\n",
            " 'ZNF69',\n",
            " 'ZNF691',\n",
            " 'ZNF692',\n",
            " 'ZNF695',\n",
            " 'ZNF696',\n",
            " 'ZNF697',\n",
            " 'ZNF699',\n",
            " 'ZNF7',\n",
            " 'ZNF70',\n",
            " 'ZNF700',\n",
            " 'ZNF701',\n",
            " 'ZNF703',\n",
            " 'ZNF704',\n",
            " 'ZNF705A',\n",
            " 'ZNF705B',\n",
            " 'ZNF705D',\n",
            " 'ZNF705G',\n",
            " 'ZNF706',\n",
            " 'ZNF707',\n",
            " 'ZNF708',\n",
            " 'ZNF71',\n",
            " 'ZNF710',\n",
            " 'ZNF711',\n",
            " 'ZNF713',\n",
            " 'ZNF714',\n",
            " 'ZNF716',\n",
            " 'ZNF717',\n",
            " 'ZNF718',\n",
            " 'ZNF721',\n",
            " 'ZNF726',\n",
            " 'ZNF727',\n",
            " 'ZNF728',\n",
            " 'ZNF729',\n",
            " 'ZNF730',\n",
            " 'ZNF732',\n",
            " 'ZNF735',\n",
            " 'ZNF736',\n",
            " 'ZNF737',\n",
            " 'ZNF74',\n",
            " 'ZNF740',\n",
            " 'ZNF746',\n",
            " 'ZNF747',\n",
            " 'ZNF749',\n",
            " 'ZNF750',\n",
            " 'ZNF75A',\n",
            " 'ZNF75D',\n",
            " 'ZNF76',\n",
            " 'ZNF761',\n",
            " 'ZNF763',\n",
            " 'ZNF764',\n",
            " 'ZNF765',\n",
            " 'ZNF766',\n",
            " 'ZNF768',\n",
            " 'ZNF77',\n",
            " 'ZNF770',\n",
            " 'ZNF771',\n",
            " 'ZNF772',\n",
            " 'ZNF773',\n",
            " 'ZNF774',\n",
            " 'ZNF775',\n",
            " 'ZNF776',\n",
            " 'ZNF777',\n",
            " 'ZNF778',\n",
            " 'ZNF780A',\n",
            " 'ZNF780B',\n",
            " 'ZNF781',\n",
            " 'ZNF782',\n",
            " 'ZNF783',\n",
            " 'ZNF784',\n",
            " 'ZNF785',\n",
            " 'ZNF786',\n",
            " 'ZNF787',\n",
            " 'ZNF789',\n",
            " 'ZNF79',\n",
            " 'ZNF790',\n",
            " 'ZNF791',\n",
            " 'ZNF792',\n",
            " 'ZNF793',\n",
            " 'ZNF799',\n",
            " 'ZNF8',\n",
            " 'ZNF80',\n",
            " 'ZNF800',\n",
            " 'ZNF804A',\n",
            " 'ZNF804B',\n",
            " 'ZNF805',\n",
            " 'ZNF808',\n",
            " 'ZNF81',\n",
            " 'ZNF813',\n",
            " 'ZNF814',\n",
            " 'ZNF816',\n",
            " 'ZNF821',\n",
            " 'ZNF823',\n",
            " 'ZNF827',\n",
            " 'ZNF829',\n",
            " 'ZNF83',\n",
            " 'ZNF830',\n",
            " 'ZNF831',\n",
            " 'ZNF835',\n",
            " 'ZNF836',\n",
            " 'ZNF837',\n",
            " 'ZNF839',\n",
            " 'ZNF84',\n",
            " 'ZNF841',\n",
            " 'ZNF843',\n",
            " 'ZNF844',\n",
            " 'ZNF845',\n",
            " 'ZNF846',\n",
            " 'ZNF85',\n",
            " 'ZNF850',\n",
            " 'ZNF852',\n",
            " 'ZNF853',\n",
            " 'ZNF860',\n",
            " 'ZNF862',\n",
            " 'ZNF865',\n",
            " 'ZNF875',\n",
            " 'ZNF878',\n",
            " 'ZNF879',\n",
            " 'ZNF880',\n",
            " 'ZNF883',\n",
            " 'ZNF888',\n",
            " 'ZNF891',\n",
            " 'ZNF90',\n",
            " 'ZNF91',\n",
            " 'ZNF92',\n",
            " 'ZNF93',\n",
            " 'ZNF98',\n",
            " 'ZNF99',\n",
            " 'ZNFX1',\n",
            " 'ZNG1A',\n",
            " 'ZNHIT1',\n",
            " 'ZNHIT2',\n",
            " 'ZNHIT3',\n",
            " 'ZNHIT6',\n",
            " 'ZNRD2',\n",
            " 'ZNRF1',\n",
            " 'ZNRF2',\n",
            " 'ZNRF3',\n",
            " 'ZNRF4',\n",
            " 'ZP1',\n",
            " 'ZP2',\n",
            " 'ZP3',\n",
            " 'ZP4',\n",
            " 'ZPBP',\n",
            " 'ZPBP2',\n",
            " 'ZPLD1',\n",
            " 'ZPR1',\n",
            " 'ZRANB1',\n",
            " 'ZRANB2',\n",
            " 'ZRANB3',\n",
            " 'ZRSR2',\n",
            " 'ZSCAN1',\n",
            " 'ZSCAN10',\n",
            " 'ZSCAN12',\n",
            " 'ZSCAN16',\n",
            " 'ZSCAN18',\n",
            " 'ZSCAN2',\n",
            " 'ZSCAN20',\n",
            " 'ZSCAN21',\n",
            " 'ZSCAN22',\n",
            " 'ZSCAN23',\n",
            " 'ZSCAN25',\n",
            " 'ZSCAN26',\n",
            " 'ZSCAN29',\n",
            " 'ZSCAN30',\n",
            " 'ZSCAN31',\n",
            " 'ZSCAN32',\n",
            " 'ZSCAN5A',\n",
            " 'ZSCAN5B',\n",
            " 'ZSCAN9',\n",
            " 'ZSWIM1',\n",
            " 'ZSWIM2',\n",
            " 'ZSWIM3',\n",
            " 'ZSWIM4',\n",
            " 'ZSWIM5',\n",
            " 'ZSWIM6',\n",
            " 'ZSWIM7',\n",
            " 'ZSWIM8',\n",
            " 'ZSWIM9',\n",
            " 'ZUP1',\n",
            " 'ZW10',\n",
            " 'ZWILCH',\n",
            " 'ZWINT',\n",
            " 'ZXDA',\n",
            " 'ZXDB',\n",
            " 'ZXDC',\n",
            " 'ZYG11A',\n",
            " 'ZYG11B',\n",
            " 'ZYX',\n",
            " 'ZZEF1',\n",
            " 'ZZZ3',\n",
            " 'control_nontargeting'}\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.replace_entries(\n",
        "    slot=\"obs\",\n",
        "    column=\"perturbed_target_symbol\",\n",
        "    map_dict={\n",
        "        \"Non-Targeting\": \"control_nontargeting\"\n",
        "    }\n",
        ")\n",
        "\n",
        "cur_data.show_unique(slot = 'obs', column = 'perturbed_target_symbol')"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "f1a48435",
      "metadata": {
        "id": "f1a48435"
      },
      "source": [
        "### Standardise perturbation targets"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "id": "438c975e",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "438c975e",
        "outputId": "b9cfe614-eb64-4a30-cb50-5169cfb33cef"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/content/PerturbationCatalogue/data_exploration/curation_tools/curation_tools.py:1271: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  map_df[\"synonyms\"] = map_df[\"synonyms\"].str.split(\"|\")\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Mapped potential synonyms in perturbed_target_symbol of the provided dataframe to gene symbols\n",
            "Converted 17823/17833 gene symbols/ENSG IDs to standardized gene symbols/ENSG IDs\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.standardize_genes(\n",
        "    slot='obs',\n",
        "    input_column='perturbed_target_symbol',\n",
        "    input_column_type='gene_symbol',\n",
        "    multiple_entries=False\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "962e76b5",
      "metadata": {
        "id": "962e76b5"
      },
      "source": [
        "### Add `perturbed_target_number` column"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "id": "a8faf759",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "a8faf759",
        "outputId": "aecfb16b-557e-42b6-cd17-63d605f6dec3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Counted entries in column perturbed_target_symbol of adata.obs and stored in perturbed_target_number\n"
          ]
        }
      ],
      "source": [
        "cur_data.count_entries(\n",
        "    slot='obs',\n",
        "    input_column='perturbed_target_symbol',\n",
        "    count_column_name='perturbed_target_number',\n",
        "    sep='|'\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "97923bec",
      "metadata": {},
      "source": [
        "### Encode chromosomes as integers"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "18623d90",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Chromosome encoding applied to perturbed_target_chromosome in adata.obs and stored as 'perturbed_target_chromosome_encoding'.\n"
          ]
        }
      ],
      "source": [
        "cur_data.chromosome_encoding()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "65d4df59",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Observation data:\n",
            "DataFrame shape: (62623, 2)\n",
            "--------------------------------------------------\n",
            "               perturbation_name  perturbed_target_chromosome_encoding\n",
            "index                                                                 \n",
            "AAACATACAAGATG    63(mod)_pBA580                                     0\n",
            "AAACATACACCTAG       OST4_pDS353                                     2\n",
            "AAACATACTTCCCG    SEC61A1_pDS031                                     3\n",
            "AAACATTGAAACAG     EIF2B4_pDS491                                     2\n",
            "AAACATTGCAGCTA       SRPR_pDS482                                    11\n",
            "...                          ...                                   ...\n",
            "TTTGCATGCTTTAC      STT3A_pDS011                                    11\n",
            "TTTGCATGGAGGAC   ARHGAP22_pDS458                                    10\n",
            "TTTGCATGTAGAGA    63(mod)_pBA580                                     0\n",
            "TTTGCATGTCAAGC     KCTD16_pDS096                                     5\n",
            "TTTGCATGTGGAGG     SAMM50_pDS156                                    22\n",
            "\n",
            "[62623 rows x 2 columns]\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.show_obs(['perturbation_name', 'perturbed_target_chromosome_encoding'])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "bf0b6a5e",
      "metadata": {},
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"cur_data\",\n  \"rows\": 5,\n  \"fields\": [\n    {\n      \"column\": \"index\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"AAACCAAAGACCCAAC-HCT116_Batch1\",\n          \"AAACCAAAGGCTTAAT-HCT116_Batch1\",\n          \"AAACCAAAGAGGTACG-HCT116_Batch1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"total_counts_mt\",\n      \"properties\": {\n        \"dtype\": \"float32\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          1562.0,\n          955.0,\n          1042.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"total_counts\",\n      \"properties\": {\n        \"dtype\": \"float32\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          47916.0,\n          38366.0,\n          28435.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"num_features\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 2,\n        \"max\": 2,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pct_counts_mt\",\n      \"properties\": {\n        \"dtype\": \"float32\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          3.259871482849121\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pass_guide_filter\",\n      \"properties\": {\n        \"dtype\": \"boolean\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          true\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"sample\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"HCT116_Batch1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbation_name\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"SIGLEC5_P1P2-1|SIGLEC5_P1P2-2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"n_genes_by_counts\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1227,\n        \"min\": 4883,\n        \"max\": 8130,\n        \"num_unique_values\": 5,\n        \"samples\": [\n          8130\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_ensg\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"ENSG00000105501\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_symbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"SIGLEC5\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_biotype\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"lncRNA\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_coord\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"chr19:51630101-51645545;-1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"perturbed_target_number\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 1,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-72224600-900a-420a-bfa2-aa2958f22fc4\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>total_counts_mt</th>\n",
              "      <th>total_counts</th>\n",
              "      <th>num_features</th>\n",
              "      <th>pct_counts_mt</th>\n",
              "      <th>pass_guide_filter</th>\n",
              "      <th>sample</th>\n",
              "      <th>perturbation_name</th>\n",
              "      <th>n_genes_by_counts</th>\n",
              "      <th>perturbed_target_ensg</th>\n",
              "      <th>perturbed_target_symbol</th>\n",
              "      <th>perturbed_target_biotype</th>\n",
              "      <th>perturbed_target_coord</th>\n",
              "      <th>perturbed_target_number</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACATGTT-HCT116_Batch1</th>\n",
              "      <td>1179.0</td>\n",
              "      <td>19136.0</td>\n",
              "      <td>2</td>\n",
              "      <td>6.161162</td>\n",
              "      <td>True</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>ST14_P1P2-1|ST14_P1P2-2</td>\n",
              "      <td>4883</td>\n",
              "      <td>ENSG00000149418</td>\n",
              "      <td>ST14</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:130159782-130210362;1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGACCCAAC-HCT116_Batch1</th>\n",
              "      <td>1562.0</td>\n",
              "      <td>47916.0</td>\n",
              "      <td>2</td>\n",
              "      <td>3.259871</td>\n",
              "      <td>True</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>SIGLEC5_P1P2-1|SIGLEC5_P1P2-2</td>\n",
              "      <td>8130</td>\n",
              "      <td>ENSG00000105501</td>\n",
              "      <td>SIGLEC5</td>\n",
              "      <td>lncRNA</td>\n",
              "      <td>chr19:51630101-51645545;-1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGGTACG-HCT116_Batch1</th>\n",
              "      <td>1042.0</td>\n",
              "      <td>28435.0</td>\n",
              "      <td>2</td>\n",
              "      <td>3.664498</td>\n",
              "      <td>True</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>VSNL1_P1P2-1|VSNL1_P1P2-2</td>\n",
              "      <td>6531</td>\n",
              "      <td>ENSG00000163032</td>\n",
              "      <td>VSNL1</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr2:17539126-17657018;1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGCGATTAT-HCT116_Batch1</th>\n",
              "      <td>1087.0</td>\n",
              "      <td>26080.0</td>\n",
              "      <td>2</td>\n",
              "      <td>4.167945</td>\n",
              "      <td>True</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>KCNK7_P1P2-1|KCNK7_P1P2-2</td>\n",
              "      <td>5931</td>\n",
              "      <td>ENSG00000173338</td>\n",
              "      <td>KCNK7</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:65592836-65595996;-1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGGCTTAAT-HCT116_Batch1</th>\n",
              "      <td>955.0</td>\n",
              "      <td>38366.0</td>\n",
              "      <td>2</td>\n",
              "      <td>2.489183</td>\n",
              "      <td>True</td>\n",
              "      <td>HCT116_Batch1</td>\n",
              "      <td>APOA4_P1P2-1|APOA4_P1P2-2</td>\n",
              "      <td>7157</td>\n",
              "      <td>ENSG00000110244</td>\n",
              "      <td>APOA4</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:116820700-116823304;-1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-72224600-900a-420a-bfa2-aa2958f22fc4')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-72224600-900a-420a-bfa2-aa2958f22fc4 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-72224600-900a-420a-bfa2-aa2958f22fc4');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-b25b9951-2ae3-4c50-a8be-96ab831f4cb8\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-b25b9951-2ae3-4c50-a8be-96ab831f4cb8')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-b25b9951-2ae3-4c50-a8be-96ab831f4cb8 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "                                total_counts_mt  total_counts  num_features  \\\n",
              "index                                                                         \n",
              "AAACCAAAGACATGTT-HCT116_Batch1           1179.0       19136.0             2   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1           1562.0       47916.0             2   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1           1042.0       28435.0             2   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1           1087.0       26080.0             2   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1            955.0       38366.0             2   \n",
              "\n",
              "                                pct_counts_mt  pass_guide_filter  \\\n",
              "index                                                              \n",
              "AAACCAAAGACATGTT-HCT116_Batch1       6.161162               True   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1       3.259871               True   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1       3.664498               True   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1       4.167945               True   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1       2.489183               True   \n",
              "\n",
              "                                       sample              perturbation_name  \\\n",
              "index                                                                          \n",
              "AAACCAAAGACATGTT-HCT116_Batch1  HCT116_Batch1        ST14_P1P2-1|ST14_P1P2-2   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1  HCT116_Batch1  SIGLEC5_P1P2-1|SIGLEC5_P1P2-2   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1  HCT116_Batch1      VSNL1_P1P2-1|VSNL1_P1P2-2   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1  HCT116_Batch1      KCNK7_P1P2-1|KCNK7_P1P2-2   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1  HCT116_Batch1      APOA4_P1P2-1|APOA4_P1P2-2   \n",
              "\n",
              "                                n_genes_by_counts perturbed_target_ensg  \\\n",
              "index                                                                     \n",
              "AAACCAAAGACATGTT-HCT116_Batch1               4883       ENSG00000149418   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1               8130       ENSG00000105501   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1               6531       ENSG00000163032   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1               5931       ENSG00000173338   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1               7157       ENSG00000110244   \n",
              "\n",
              "                               perturbed_target_symbol  \\\n",
              "index                                                    \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                    ST14   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                 SIGLEC5   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                   VSNL1   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                   KCNK7   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                   APOA4   \n",
              "\n",
              "                               perturbed_target_biotype  \\\n",
              "index                                                     \n",
              "AAACCAAAGACATGTT-HCT116_Batch1           protein_coding   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                   lncRNA   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1           protein_coding   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1           protein_coding   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1           protein_coding   \n",
              "\n",
              "                                      perturbed_target_coord  \\\n",
              "index                                                          \n",
              "AAACCAAAGACATGTT-HCT116_Batch1   chr11:130159782-130210362;1   \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1    chr19:51630101-51645545;-1   \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1      chr2:17539126-17657018;1   \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1    chr11:65592836-65595996;-1   \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1  chr11:116820700-116823304;-1   \n",
              "\n",
              "                                perturbed_target_number  \n",
              "index                                                    \n",
              "AAACCAAAGACATGTT-HCT116_Batch1                        1  \n",
              "AAACCAAAGACCCAAC-HCT116_Batch1                        1  \n",
              "AAACCAAAGAGGTACG-HCT116_Batch1                        1  \n",
              "AAACCAAAGCGATTAT-HCT116_Batch1                        1  \n",
              "AAACCAAAGGCTTAAT-HCT116_Batch1                        1  "
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "cur_data.adata.obs.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "1dc8df38",
      "metadata": {
        "id": "1dc8df38"
      },
      "source": [
        "### Add guide RNA information"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "d0d8699f",
      "metadata": {
        "id": "d0d8699f"
      },
      "source": [
        "The original pre-print does not contain information about guide RNAs in the `obs` slot.\n",
        "\n",
        "However, the pre-print references the source of the library - [Maximizing CRISPRi efficacy and accessibility with dual-sgRNA libraries and optimal effectors. Replogle et al 2022](https://doi.org/10.7554/eLife.81856).\n",
        "\n",
        "Supplementary table s4 contains the information about guide RNAs used in the experiment:\n",
        "\n",
        "Supplementary table S4: https://elifesciences.org/download/aHR0cHM6Ly9jZG4uZWxpZmVzY2llbmNlcy5vcmcvYXJ0aWNsZXMvODE4NTYvZWxpZmUtODE4NTYtc3VwcDQtdjIueGxzeA--/elife-81856-supp4-v2.xlsx?_hash=AmEXp%2BOl1z1Y4JCBAaVhST5ImQmKAGdfEjGhuJbSJ9Q%3D"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "id": "41bf47fe",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 455
        },
        "id": "41bf47fe",
        "outputId": "d396f4ff-8be7-4145-d901-7d2cee1268de"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "obj_df"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-83c3b093-ccad-4b41-a8eb-e51323646149\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>guide_sequence</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAAGTAGC-HEK293T_Batch1</th>\n",
              "      <td>GGAGAGCGTCAAATGAAGAG|GGAGGCTGAAGGCTGCGTCC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGTACGG-HEK293T_Batch1</th>\n",
              "      <td>GGAGGCTATCTGCTAGACTT|GTCCACATGCGGATAGACAC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGTGGGC-HEK293T_Batch1</th>\n",
              "      <td>GGGCCGAACCGCGCCAGAGC|GCGACCGAGAGGCACTGTGT</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGTGTAA-HEK293T_Batch1</th>\n",
              "      <td>GTGCCGAAGAGGCCGGGACC|GACGACGGTTCCCGAGACAG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGATACGGG-HEK293T_Batch1</th>\n",
              "      <td>GATGTTCTTGAATCATCTAT|GGGTTCCTGGAACACTTCAG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTCCGTGGGATGG-HEK293T_Batch223</th>\n",
              "      <td>GGGGGAATCTGCAAGACCAT|GAGAAGGAGCGAGGCTAAAG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTCTTCAACAGCC-HEK293T_Batch223</th>\n",
              "      <td>GGAGCACAGGCGCTTCACCG|GGGCAGTCTCGGCAGAGCAC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTCTTCAGTACTG-HEK293T_Batch223</th>\n",
              "      <td>GAAGTTAGGTGTCCAGAGCA|GAGGTGTCCAGAGCACGGTA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTCTTCCGCCGTT-HEK293T_Batch223</th>\n",
              "      <td>GCTCCCGCCTCCTGAGGCAG|GTCCACTGCAATGCTAACAT</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGCAGGCTTGCG-HEK293T_Batch223</th>\n",
              "      <td>GGGCCTGCGACGCAGAGAAG|GCTGCGACGCAGAGAAGAGG</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>4534299 rows × 1 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-83c3b093-ccad-4b41-a8eb-e51323646149')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-83c3b093-ccad-4b41-a8eb-e51323646149 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-83c3b093-ccad-4b41-a8eb-e51323646149');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-15531f1d-3b7b-434d-ae0f-51e3b78d5921\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-15531f1d-3b7b-434d-ae0f-51e3b78d5921')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-15531f1d-3b7b-434d-ae0f-51e3b78d5921 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "  <div id=\"id_a2828f41-0656-4dda-92af-e6c9ac667233\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('obj_df')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_a2828f41-0656-4dda-92af-e6c9ac667233 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('obj_df');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "                                                              guide_sequence\n",
              "index                                                                       \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1    GGAGAGCGTCAAATGAAGAG|GGAGGCTGAAGGCTGCGTCC\n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1    GGAGGCTATCTGCTAGACTT|GTCCACATGCGGATAGACAC\n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1    GGGCCGAACCGCGCCAGAGC|GCGACCGAGAGGCACTGTGT\n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1    GTGCCGAAGAGGCCGGGACC|GACGACGGTTCCCGAGACAG\n",
              "AAACCAAAGATACGGG-HEK293T_Batch1    GATGTTCTTGAATCATCTAT|GGGTTCCTGGAACACTTCAG\n",
              "...                                                                      ...\n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223  GGGGGAATCTGCAAGACCAT|GAGAAGGAGCGAGGCTAAAG\n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223  GGAGCACAGGCGCTTCACCG|GGGCAGTCTCGGCAGAGCAC\n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223  GAAGTTAGGTGTCCAGAGCA|GAGGTGTCCAGAGCACGGTA\n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223  GCTCCCGCCTCCTGAGGCAG|GTCCACTGCAATGCTAACAT\n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223  GGGCCTGCGACGCAGAGAAG|GCTGCGACGCAGAGAAGAGG\n",
              "\n",
              "[4534299 rows x 1 columns]"
            ]
          },
          "execution_count": 21,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "# get synonyms from the gene ontology\n",
        "syn_df = cur_data.gene_ont[[\"synonyms\", \"symbol\"]].drop_duplicates()\n",
        "syn_df[\"symbol\"] = syn_df[\"symbol\"].fillna(\"\")\n",
        "syn_df[\"synonyms\"] = syn_df[\"synonyms\"].fillna(\"\")\n",
        "syn_df[\"all_genes\"] = (\n",
        "    (syn_df[\"symbol\"] + \"|\" + syn_df[\"synonyms\"]).str.lstrip(\"|\").str.rstrip(\"|\")\n",
        ")\n",
        "syn_df[\"all_genes_match\"] = syn_df[\"all_genes\"].str.split(\"|\")\n",
        "syn_df = syn_df.explode(\"all_genes_match\")\n",
        "syn_df[\"all_genes_match\"] = syn_df[\"all_genes_match\"].str.upper()\n",
        "\n",
        "# read data from supplementary files\n",
        "guide_ids_df = pd.read_csv(\n",
        "    \"Perturbseq/supplementary/orion_2025_weissman_dual_crispri_sgrnas.csv\"\n",
        ")\n",
        "guide_ids_df[\"gene\"] = guide_ids_df[\"gene\"].str.upper()\n",
        "\n",
        "guide_ids_df = guide_ids_df.merge(\n",
        "    syn_df, left_on=\"gene\", right_on=\"all_genes_match\", how=\"left\"\n",
        ")\n",
        "\n",
        "guide_ids_df[\"all_genes\"] = guide_ids_df[\"all_genes\"].str.split(\"|\")\n",
        "guide_ids_df = guide_ids_df.explode(\"all_genes\").drop_duplicates()\n",
        "\n",
        "# create a perturbation name column\n",
        "guide_ids_df[\"perturbation_name\"] = (\n",
        "    guide_ids_df[\"all_genes\"]\n",
        "    + \"_\"\n",
        "    + guide_ids_df[\"transcript\"]\n",
        "    + \"|\"\n",
        "    + guide_ids_df[\"all_genes\"]\n",
        "    + \"_\"\n",
        "    + guide_ids_df[\"transcript\"]\n",
        ")\n",
        "\n",
        "guide_ids_df[\"perturbation_name\"] = guide_ids_df[\"perturbation_name\"].str.upper()\n",
        "\n",
        "# replace commas with underscores in the perturbation name\n",
        "guide_ids_df[\"perturbation_name\"] = guide_ids_df[\"perturbation_name\"].str.replace(\n",
        "    \",\", \"_\"\n",
        ")\n",
        "\n",
        "# create a dual guide RNA sequence column\n",
        "guide_ids_df[\"dual_guide_sequence\"] = (\n",
        "    guide_ids_df[\"protospacer_A\"] + \"|\" + guide_ids_df[\"protospacer_B\"]\n",
        ").str.upper()\n",
        "\n",
        "guide_ids_df.loc[\n",
        "    guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"perturbation_name\"\n",
        "] = (\n",
        "    guide_ids_df.loc[\n",
        "        guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"gene\"\n",
        "    ]\n",
        "    + \"_\"\n",
        "    + guide_ids_df.loc[\n",
        "        guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"transcript\"\n",
        "    ]\n",
        "    + \"|\"\n",
        "    + guide_ids_df.loc[\n",
        "        guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"gene\"\n",
        "    ]\n",
        "    + \"_\"\n",
        "    + guide_ids_df.loc[\n",
        "        guide_ids_df[\"all_genes\"].isna() & guide_ids_df[\"gene\"].notna(), \"transcript\"\n",
        "    ]\n",
        ")\n",
        "\n",
        "# replace NEGATIVE_CONTROL with actual non targeting guide names\n",
        "guide_ids_df.loc[guide_ids_df[\"gene\"] == \"NEGATIVE_CONTROL\", \"perturbation_name\"] = (\n",
        "    guide_ids_df.loc[guide_ids_df[\"gene\"] == \"NEGATIVE_CONTROL\", \"sgID_AB\"].str.upper()\n",
        ")\n",
        "\n",
        "# create dictionaries for guide sequences mapping\n",
        "guide_ids_dict = (\n",
        "    guide_ids_df[[\"perturbation_name\", \"dual_guide_sequence\"]]\n",
        "    .dropna()\n",
        "    .drop_duplicates()\n",
        "    .set_index(\"perturbation_name\")[\"dual_guide_sequence\"]\n",
        "    .to_dict()\n",
        ")\n",
        "\n",
        "# add guide sequences to the obs dataframe\n",
        "obj_df = cur_data.adata.obs.copy()\n",
        "\n",
        "# remove the numeric suffices from the perturbation names\n",
        "obj_df[\"perturbation_name_matching\"] = obj_df[\"perturbation_name\"].str.replace(\n",
        "    r\"-\\d(?=\\||$)\", \"\", regex=True\n",
        ")\n",
        "obj_df[\"perturbation_name_matching\"] = obj_df[\"perturbation_name_matching\"].str.upper()\n",
        "\n",
        "# map the guide sequences to the perturbation names\n",
        "obj_df[\"guide_sequence\"] = obj_df[\"perturbation_name_matching\"].map(guide_ids_dict)\n",
        "\n",
        "obj_df = obj_df[[\"guide_sequence\"]]\n",
        "\n",
        "obj_df"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "id": "51ed3fec",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 683
        },
        "id": "51ed3fec",
        "outputId": "535d67b5-4449-43b6-b0ad-f8b77b6168be"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-02cb92a4-390c-4461-8e30-f6c0c43b1092\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>n_genes_by_counts</th>\n",
              "      <th>total_counts</th>\n",
              "      <th>pass_guide_filter</th>\n",
              "      <th>perturbation_name</th>\n",
              "      <th>num_features</th>\n",
              "      <th>sample</th>\n",
              "      <th>pct_counts_mt</th>\n",
              "      <th>total_counts_mt</th>\n",
              "      <th>perturbed_target_ensg</th>\n",
              "      <th>perturbed_target_symbol</th>\n",
              "      <th>perturbed_target_biotype</th>\n",
              "      <th>perturbed_target_coord</th>\n",
              "      <th>perturbed_target_number</th>\n",
              "      <th>guide_sequence</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAAGTAGC-HEK293T_Batch1</th>\n",
              "      <td>6856</td>\n",
              "      <td>24243.0</td>\n",
              "      <td>True</td>\n",
              "      <td>FAM180A_P1P2-1|FAM180A_P1P2-2</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch1</td>\n",
              "      <td>2.668812</td>\n",
              "      <td>647.0</td>\n",
              "      <td>ENSG00000189320</td>\n",
              "      <td>FAM180A</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr7:135728348-135748813;-1</td>\n",
              "      <td>1</td>\n",
              "      <td>GGAGAGCGTCAAATGAAGAG|GGAGGCTGAAGGCTGCGTCC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGTACGG-HEK293T_Batch1</th>\n",
              "      <td>6636</td>\n",
              "      <td>20847.0</td>\n",
              "      <td>True</td>\n",
              "      <td>non-targeting_03016|non-targeting_03214</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch1</td>\n",
              "      <td>4.250012</td>\n",
              "      <td>886.0</td>\n",
              "      <td>control_nontargeting</td>\n",
              "      <td>control_nontargeting</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1</td>\n",
              "      <td>GGAGGCTATCTGCTAGACTT|GTCCACATGCGGATAGACAC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGTGGGC-HEK293T_Batch1</th>\n",
              "      <td>5353</td>\n",
              "      <td>14035.0</td>\n",
              "      <td>True</td>\n",
              "      <td>PCDHGC4_P1P2-1|PCDHGC4_P1P2-2</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch1</td>\n",
              "      <td>4.346277</td>\n",
              "      <td>610.0</td>\n",
              "      <td>ENSG00000242419</td>\n",
              "      <td>PCDHGC4</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr5:141484997-141512975;1</td>\n",
              "      <td>1</td>\n",
              "      <td>GGGCCGAACCGCGCCAGAGC|GCGACCGAGAGGCACTGTGT</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGAGTGTAA-HEK293T_Batch1</th>\n",
              "      <td>6358</td>\n",
              "      <td>20939.0</td>\n",
              "      <td>True</td>\n",
              "      <td>USP30_P1P2-1|USP30_P1P2-2</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch1</td>\n",
              "      <td>2.812933</td>\n",
              "      <td>589.0</td>\n",
              "      <td>ENSG00000135093</td>\n",
              "      <td>USP30</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr12:109023089-109088023;1</td>\n",
              "      <td>1</td>\n",
              "      <td>GTGCCGAAGAGGCCGGGACC|GACGACGGTTCCCGAGACAG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACCAAAGATACGGG-HEK293T_Batch1</th>\n",
              "      <td>5527</td>\n",
              "      <td>14083.0</td>\n",
              "      <td>True</td>\n",
              "      <td>KLRD1_P1P2-1|KLRD1_P1P2-2</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch1</td>\n",
              "      <td>3.379961</td>\n",
              "      <td>476.0</td>\n",
              "      <td>ENSG00000134539</td>\n",
              "      <td>KLRD1</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr12:10226058-10329608;1</td>\n",
              "      <td>1</td>\n",
              "      <td>GATGTTCTTGAATCATCTAT|GGGTTCCTGGAACACTTCAG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTCCGTGGGATGG-HEK293T_Batch223</th>\n",
              "      <td>8112</td>\n",
              "      <td>29296.0</td>\n",
              "      <td>True</td>\n",
              "      <td>non-targeting_00183|non-targeting_01885</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch223</td>\n",
              "      <td>1.781813</td>\n",
              "      <td>522.0</td>\n",
              "      <td>control_nontargeting</td>\n",
              "      <td>control_nontargeting</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1</td>\n",
              "      <td>GGGGGAATCTGCAAGACCAT|GAGAAGGAGCGAGGCTAAAG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTCTTCAACAGCC-HEK293T_Batch223</th>\n",
              "      <td>10568</td>\n",
              "      <td>88551.0</td>\n",
              "      <td>True</td>\n",
              "      <td>AKAP13_P1P2-1|AKAP13_P1P2-2</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch223</td>\n",
              "      <td>3.427403</td>\n",
              "      <td>3035.0</td>\n",
              "      <td>ENSG00000170776</td>\n",
              "      <td>AKAP13</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr15:85380571-85749358;1</td>\n",
              "      <td>1</td>\n",
              "      <td>GGAGCACAGGCGCTTCACCG|GGGCAGTCTCGGCAGAGCAC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTCTTCAGTACTG-HEK293T_Batch223</th>\n",
              "      <td>10735</td>\n",
              "      <td>81286.0</td>\n",
              "      <td>True</td>\n",
              "      <td>TTF2_P1P2-1|TTF2_P1P2-2</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch223</td>\n",
              "      <td>2.224245</td>\n",
              "      <td>1808.0</td>\n",
              "      <td>ENSG00000116830</td>\n",
              "      <td>TTF2</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr1:117060326-117107453;1</td>\n",
              "      <td>1</td>\n",
              "      <td>GAAGTTAGGTGTCCAGAGCA|GAGGTGTCCAGAGCACGGTA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTCTTCCGCCGTT-HEK293T_Batch223</th>\n",
              "      <td>9248</td>\n",
              "      <td>40733.0</td>\n",
              "      <td>True</td>\n",
              "      <td>MMP7_P1P2-1|MMP7_P1P2-2</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch223</td>\n",
              "      <td>4.338006</td>\n",
              "      <td>1767.0</td>\n",
              "      <td>ENSG00000137673</td>\n",
              "      <td>MMP7</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr11:102520508-102530750;-1</td>\n",
              "      <td>1</td>\n",
              "      <td>GCTCCCGCCTCCTGAGGCAG|GTCCACTGCAATGCTAACAT</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>GTTGTGCAGGCTTGCG-HEK293T_Batch223</th>\n",
              "      <td>8682</td>\n",
              "      <td>36045.0</td>\n",
              "      <td>True</td>\n",
              "      <td>AAK1_P1P2-1|AAK1_P1P2-2</td>\n",
              "      <td>2</td>\n",
              "      <td>HEK293T_Batch223</td>\n",
              "      <td>2.621723</td>\n",
              "      <td>945.0</td>\n",
              "      <td>ENSG00000115977</td>\n",
              "      <td>AAK1</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>chr2:69457997-69674349;-1</td>\n",
              "      <td>1</td>\n",
              "      <td>GGGCCTGCGACGCAGAGAAG|GCTGCGACGCAGAGAAGAGG</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>4534299 rows × 14 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-02cb92a4-390c-4461-8e30-f6c0c43b1092')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-02cb92a4-390c-4461-8e30-f6c0c43b1092 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-02cb92a4-390c-4461-8e30-f6c0c43b1092');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-9b582926-f5da-4e8a-8241-b59d9d070e4a\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-9b582926-f5da-4e8a-8241-b59d9d070e4a')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-9b582926-f5da-4e8a-8241-b59d9d070e4a button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "                                   n_genes_by_counts  total_counts  \\\n",
              "index                                                                \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1                 6856       24243.0   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1                 6636       20847.0   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1                 5353       14035.0   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1                 6358       20939.0   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1                 5527       14083.0   \n",
              "...                                              ...           ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223               8112       29296.0   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223              10568       88551.0   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223              10735       81286.0   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223               9248       40733.0   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223               8682       36045.0   \n",
              "\n",
              "                                   pass_guide_filter  \\\n",
              "index                                                  \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1                 True   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1                 True   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1                 True   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1                 True   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1                 True   \n",
              "...                                              ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223               True   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223               True   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223               True   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223               True   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223               True   \n",
              "\n",
              "                                                         perturbation_name  \\\n",
              "index                                                                        \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1              FAM180A_P1P2-1|FAM180A_P1P2-2   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1    non-targeting_03016|non-targeting_03214   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1              PCDHGC4_P1P2-1|PCDHGC4_P1P2-2   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1                  USP30_P1P2-1|USP30_P1P2-2   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1                  KLRD1_P1P2-1|KLRD1_P1P2-2   \n",
              "...                                                                    ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223  non-targeting_00183|non-targeting_01885   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223              AKAP13_P1P2-1|AKAP13_P1P2-2   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223                  TTF2_P1P2-1|TTF2_P1P2-2   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223                  MMP7_P1P2-1|MMP7_P1P2-2   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223                  AAK1_P1P2-1|AAK1_P1P2-2   \n",
              "\n",
              "                                   num_features            sample  \\\n",
              "index                                                               \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1               2    HEK293T_Batch1   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1               2    HEK293T_Batch1   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1               2    HEK293T_Batch1   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1               2    HEK293T_Batch1   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1               2    HEK293T_Batch1   \n",
              "...                                         ...               ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223             2  HEK293T_Batch223   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223             2  HEK293T_Batch223   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223             2  HEK293T_Batch223   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223             2  HEK293T_Batch223   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223             2  HEK293T_Batch223   \n",
              "\n",
              "                                   pct_counts_mt  total_counts_mt  \\\n",
              "index                                                               \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1         2.668812            647.0   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1         4.250012            886.0   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1         4.346277            610.0   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1         2.812933            589.0   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1         3.379961            476.0   \n",
              "...                                          ...              ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223       1.781813            522.0   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223       3.427403           3035.0   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223       2.224245           1808.0   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223       4.338006           1767.0   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223       2.621723            945.0   \n",
              "\n",
              "                                  perturbed_target_ensg  \\\n",
              "index                                                     \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1         ENSG00000189320   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1    control_nontargeting   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1         ENSG00000242419   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1         ENSG00000135093   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1         ENSG00000134539   \n",
              "...                                                 ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223  control_nontargeting   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223       ENSG00000170776   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223       ENSG00000116830   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223       ENSG00000137673   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223       ENSG00000115977   \n",
              "\n",
              "                                  perturbed_target_symbol  \\\n",
              "index                                                       \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1                   FAM180A   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1      control_nontargeting   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1                   PCDHGC4   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1                     USP30   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1                     KLRD1   \n",
              "...                                                   ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223    control_nontargeting   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223                  AKAP13   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223                    TTF2   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223                    MMP7   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223                    AAK1   \n",
              "\n",
              "                                  perturbed_target_biotype  \\\n",
              "index                                                        \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1             protein_coding   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1                        NaN   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1             protein_coding   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1             protein_coding   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1             protein_coding   \n",
              "...                                                    ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223                      NaN   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223           protein_coding   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223           protein_coding   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223           protein_coding   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223           protein_coding   \n",
              "\n",
              "                                         perturbed_target_coord  \\\n",
              "index                                                             \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1     chr7:135728348-135748813;-1   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1                             NaN   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1      chr5:141484997-141512975;1   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1     chr12:109023089-109088023;1   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1       chr12:10226058-10329608;1   \n",
              "...                                                         ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223                           NaN   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223     chr15:85380571-85749358;1   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223    chr1:117060326-117107453;1   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223  chr11:102520508-102530750;-1   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223     chr2:69457997-69674349;-1   \n",
              "\n",
              "                                   perturbed_target_number  \\\n",
              "index                                                        \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1                          1   \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1                          1   \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1                          1   \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1                          1   \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1                          1   \n",
              "...                                                    ...   \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223                        1   \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223                        1   \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223                        1   \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223                        1   \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223                        1   \n",
              "\n",
              "                                                              guide_sequence  \n",
              "index                                                                         \n",
              "AAACCAAAGAAGTAGC-HEK293T_Batch1    GGAGAGCGTCAAATGAAGAG|GGAGGCTGAAGGCTGCGTCC  \n",
              "AAACCAAAGAGTACGG-HEK293T_Batch1    GGAGGCTATCTGCTAGACTT|GTCCACATGCGGATAGACAC  \n",
              "AAACCAAAGAGTGGGC-HEK293T_Batch1    GGGCCGAACCGCGCCAGAGC|GCGACCGAGAGGCACTGTGT  \n",
              "AAACCAAAGAGTGTAA-HEK293T_Batch1    GTGCCGAAGAGGCCGGGACC|GACGACGGTTCCCGAGACAG  \n",
              "AAACCAAAGATACGGG-HEK293T_Batch1    GATGTTCTTGAATCATCTAT|GGGTTCCTGGAACACTTCAG  \n",
              "...                                                                      ...  \n",
              "GTTGTCCGTGGGATGG-HEK293T_Batch223  GGGGGAATCTGCAAGACCAT|GAGAAGGAGCGAGGCTAAAG  \n",
              "GTTGTCTTCAACAGCC-HEK293T_Batch223  GGAGCACAGGCGCTTCACCG|GGGCAGTCTCGGCAGAGCAC  \n",
              "GTTGTCTTCAGTACTG-HEK293T_Batch223  GAAGTTAGGTGTCCAGAGCA|GAGGTGTCCAGAGCACGGTA  \n",
              "GTTGTCTTCCGCCGTT-HEK293T_Batch223  GCTCCCGCCTCCTGAGGCAG|GTCCACTGCAATGCTAACAT  \n",
              "GTTGTGCAGGCTTGCG-HEK293T_Batch223  GGGCCTGCGACGCAGAGAAG|GCTGCGACGCAGAGAAGAGG  \n",
              "\n",
              "[4534299 rows x 14 columns]"
            ]
          },
          "execution_count": 22,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "# merge guide sequences with main\n",
        "cur_data.adata.obs = cur_data.adata.obs.merge(\n",
        "    obj_df,\n",
        "    left_index=True,\n",
        "    right_index=True,\n",
        "    how=\"left\",\n",
        ")\n",
        "\n",
        "cur_data.adata.obs"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "20cd149e",
      "metadata": {},
      "source": [
        "### Add metadata"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "dd3c6272",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Column dataset_id added to adata.obs\n",
            "Column sample_id added to adata.obs\n",
            "Column treatment_label added to adata.obs\n",
            "Column treatment_id added to adata.obs\n",
            "Column perturbation_type_label added to adata.obs\n",
            "Column perturbation_type_id added to adata.obs\n",
            "Column model_system_label added to adata.obs\n",
            "Column model_system_id added to adata.obs\n",
            "Column tissue added to adata.obs\n",
            "Column timepoint added to adata.obs\n",
            "Column species added to adata.obs\n",
            "Column sex_label added to adata.obs\n",
            "Column sex_id added to adata.obs\n",
            "Column developmental_stage_label added to adata.obs\n",
            "Column developmental_stage_id added to adata.obs\n",
            "Column study_title added to adata.obs\n",
            "Column study_uri added to adata.obs\n",
            "Column study_year added to adata.obs\n",
            "Column first_author added to adata.obs\n",
            "Column last_author added to adata.obs\n",
            "Column experiment_title added to adata.obs\n",
            "Column experiment_summary added to adata.obs\n",
            "Column number_of_perturbed_targets added to adata.obs\n",
            "Column number_of_perturbed_samples added to adata.obs\n",
            "Column library_generation_type_id added to adata.obs\n",
            "Column library_generation_type_label added to adata.obs\n",
            "Column library_generation_method_id added to adata.obs\n",
            "Column library_generation_method_label added to adata.obs\n",
            "Column enzyme_delivery_method_id added to adata.obs\n",
            "Column enzyme_delivery_method_label added to adata.obs\n",
            "Column library_delivery_method_id added to adata.obs\n",
            "Column library_delivery_method_label added to adata.obs\n",
            "Column enzyme_integration_state_id added to adata.obs\n",
            "Column enzyme_integration_state_label added to adata.obs\n",
            "Column library_integration_state_id added to adata.obs\n",
            "Column library_integration_state_label added to adata.obs\n",
            "Column enzyme_expression_control_id added to adata.obs\n",
            "Column enzyme_expression_control_label added to adata.obs\n",
            "Column library_expression_control_id added to adata.obs\n",
            "Column library_expression_control_label added to adata.obs\n",
            "Column library_name added to adata.obs\n",
            "Column library_uri added to adata.obs\n",
            "Column library_format_id added to adata.obs\n",
            "Column library_format_label added to adata.obs\n",
            "Column library_scope_id added to adata.obs\n",
            "Column library_scope_label added to adata.obs\n",
            "Column library_perturbation_type_id added to adata.obs\n",
            "Column library_perturbation_type_label added to adata.obs\n",
            "Column library_manufacturer added to adata.obs\n",
            "Column library_lentiviral_generation added to adata.obs\n",
            "Column library_grnas_per_target added to adata.obs\n",
            "Column library_total_grnas added to adata.obs\n",
            "Column library_total_variants added to adata.obs\n",
            "Column readout_dimensionality_id added to adata.obs\n",
            "Column readout_dimensionality_label added to adata.obs\n",
            "Column readout_type_id added to adata.obs\n",
            "Column readout_type_label added to adata.obs\n",
            "Column readout_technology_id added to adata.obs\n",
            "Column readout_technology_label added to adata.obs\n",
            "Column method_name_id added to adata.obs\n",
            "Column method_name_label added to adata.obs\n",
            "Column method_uri added to adata.obs\n",
            "Column sequencing_library_kit_id added to adata.obs\n",
            "Column sequencing_library_kit_label added to adata.obs\n",
            "Column sequencing_platform_id added to adata.obs\n",
            "Column sequencing_platform_label added to adata.obs\n",
            "Column sequencing_strategy_id added to adata.obs\n",
            "Column sequencing_strategy_label added to adata.obs\n",
            "Column software_counts_id added to adata.obs\n",
            "Column software_counts_label added to adata.obs\n",
            "Column software_analysis_id added to adata.obs\n",
            "Column software_analysis_label added to adata.obs\n",
            "Column reference_genome_id added to adata.obs\n",
            "Column reference_genome_label added to adata.obs\n",
            "Column associated_datasets added to adata.obs\n"
          ]
        }
      ],
      "source": [
        "cur_data.create_columns(\n",
        "    slot=\"obs\",\n",
        "    col_dict={\n",
        "        \"dataset_id\": cur_data.dataset_id,\n",
        "        \"sample_id\": range(1, cur_data.adata.obs.shape[0] + 1),\n",
        "        # treatment\n",
        "        \"treatment_label\": None, \n",
        "        \"treatment_id\": None,\n",
        "        # perturbation type\n",
        "        \"perturbation_type_label\": \"CRISPRi\", \n",
        "        \"perturbation_type_id\": None,\n",
        "        # model system\n",
        "        \"model_system_label\": \"cell line\", \n",
        "        \"model_system_id\": None,\n",
        "        \n",
        "        'tissue': 'kidney',\n",
        "        'cell_type': 'kidney epithelial cell',\n",
        "        'cell_line': 'HEK 293T cell',\n",
        "        \"timepoint\": \"P0DT0H0M0S\",\n",
        "        'disease': 'healthy',\n",
        "        \n",
        "        \"species\": \"Homo sapiens\",\n",
        "        \"sex_label\": \"female\", \n",
        "        \"sex_id\": None,\n",
        "        \"developmental_stage_label\": \"embryo\", \n",
        "        \"developmental_stage_id\": None,\n",
        "\n",
        "        \"study_title\": \"X-Atlas/Orion: Genome-wide Perturb-seq Datasets via a Scalable Fix-Cryopreserve Platform for Training Dose-Dependent Biological Foundation Models\",\n",
        "        \"study_uri\": \"https://doi.org/10.1101/2025.06.11.659105\",\n",
        "        \"study_year\": 2025,\n",
        "        \"first_author\": \"Ann Huang\",\n",
        "        \"last_author\": \"Ci Chu\",\n",
        "        \n",
        "        \"experiment_title\": \"Over 4.5 million HEK293T cells were transfected with dual gRNAs targeting the same gene, and whole genome CRISPRi Perturb-seq was performed.\",\n",
        "        \"experiment_summary\": \"The researchers developed the Fix-Cryopreserve-ScRNAseq (FiCS) Perturb-seq platform to enable scalable and reproducible genome-wide perturbation screening in human cell lines. Using this platform, they generated a genome-wide dual-guide CRISPRi Perturb-seq dataset in HEK293T cells. The dataset includes over 4.5 million deeply sequenced cells, with each cell yielding over 16,000 unique molecular identifiers (UMIs).\",\n",
        "        \"number_of_perturbed_targets\": len(set(cur_data.adata.obs['perturbed_target_coord'])),\n",
        "        \"number_of_perturbed_samples\": cur_data.adata.obs.shape[0],\n",
        "        \n",
        "        \"library_generation_type_id\": \"EFO:0022868\",\n",
        "        \"library_generation_type_label\": \"endogenous\",\n",
        "\n",
        "        \"library_generation_method_id\": \"EFO:0022895\",\n",
        "        \"library_generation_method_label\": \"dCas9-KRAB\",\n",
        "\n",
        "        \"enzyme_delivery_method_id\": None,\n",
        "        \"enzyme_delivery_method_label\": \"nanoparticle-mediated transfection\",\n",
        "\n",
        "        \"library_delivery_method_id\": None,\n",
        "        \"library_delivery_method_label\": \"lentiviral transduction\",\n",
        "\n",
        "        \"enzyme_integration_state_id\": None,\n",
        "        \"enzyme_integration_state_label\": \"random locus integration\",\n",
        "\n",
        "        \"library_integration_state_id\": None,\n",
        "        \"library_integration_state_label\": \"random locus integration\",\n",
        "\n",
        "        \"enzyme_expression_control_id\": None,\n",
        "        \"enzyme_expression_control_label\": \"constitutive expression\",\n",
        "\n",
        "        \"library_expression_control_id\": None,\n",
        "        \"library_expression_control_label\": \"constitutive expression\",\n",
        "\n",
        "        \"library_name\": \"custom\",\n",
        "        \"library_uri\": None,\n",
        "\n",
        "        \"library_format_id\": None,\n",
        "        \"library_format_label\": \"pooled\",\n",
        "\n",
        "        \"library_scope_id\": None,\n",
        "        \"library_scope_label\": \"genome-wide\",\n",
        "\n",
        "        \"library_perturbation_type_id\": None,\n",
        "        \"library_perturbation_type_label\": \"inhibition\",\n",
        "        \n",
        "        \"library_manufacturer\": \"Weissman\",\n",
        "        \"library_lentiviral_generation\": \"3\",\n",
        "        \"library_grnas_per_target\": \"2\",\n",
        "        \"library_total_grnas\": \"20845\",\n",
        "        \"library_total_variants\": None,\n",
        "        \n",
        "        \"readout_dimensionality_id\": None,\n",
        "        \"readout_dimensionality_label\": \"high-dimensional assay\",\n",
        "\n",
        "        \"readout_type_id\": None,\n",
        "        \"readout_type_label\": \"transcriptomic\",\n",
        "\n",
        "        \"readout_technology_id\": None,\n",
        "        \"readout_technology_label\": \"single-cell rna-seq\",\n",
        "\n",
        "        \"method_name_id\": None,\n",
        "        \"method_name_label\": \"Perturb-seq\",\n",
        "\n",
        "        \"method_uri\": None,\n",
        "\n",
        "        \"sequencing_library_kit_id\": None,\n",
        "        \"sequencing_library_kit_label\": \"10x Genomics Chromium GEM-X Single Cell 5-prime kit v3\",\n",
        "\n",
        "        \"sequencing_platform_id\": None,\n",
        "        \"sequencing_platform_label\": \"Illumina NovaSeq X Plus\",\n",
        "\n",
        "        \"sequencing_strategy_id\": None,\n",
        "        \"sequencing_strategy_label\": \"barcode sequencing\",\n",
        "\n",
        "        \"software_counts_id\": None,\n",
        "        \"software_counts_label\": \"CellRanger\",\n",
        "\n",
        "        \"software_analysis_id\": None,\n",
        "        \"software_analysis_label\": \"custom\",\n",
        "\n",
        "        \"reference_genome_id\": None,\n",
        "        \"reference_genome_label\": \"GRCh38\",\n",
        "\n",
        "        \"associated_datasets\": json.dumps([\n",
        "            {\n",
        "                \"dataset_accession\": \"HEK293T_filtered_dual_guide_cells\",\n",
        "                \"dataset_uri\": \"https://plus.figshare.com/ndownloader/files/55074802\",\n",
        "                \"dataset_description\": \"Filtered raw counts\",\n",
        "                \"dataset_file_name\": \"HEK293T_filtered_dual_guide_cells.h5ad\",\n",
        "            }\n",
        "        ])\n",
        "    }\n",
        ")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "3f1421b9",
      "metadata": {},
      "outputs": [
        {
          "data": {
            "application/vnd.microsoft.datawrangler.viewer.v0+json": {
              "columns": [
                {
                  "name": "index",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "tissue_type",
                  "rawType": "category",
                  "type": "unknown"
                },
                {
                  "name": "celltype",
                  "rawType": "category",
                  "type": "unknown"
                },
                {
                  "name": "organism",
                  "rawType": "category",
                  "type": "unknown"
                },
                {
                  "name": "cancer",
                  "rawType": "bool",
                  "type": "boolean"
                },
                {
                  "name": "disease",
                  "rawType": "category",
                  "type": "unknown"
                },
                {
                  "name": "percent_ribo",
                  "rawType": "float32",
                  "type": "float"
                },
                {
                  "name": "read count",
                  "rawType": "float64",
                  "type": "float"
                },
                {
                  "name": "cell_line",
                  "rawType": "category",
                  "type": "unknown"
                },
                {
                  "name": "UMI count",
                  "rawType": "float64",
                  "type": "float"
                },
                {
                  "name": "perturbation_type",
                  "rawType": "category",
                  "type": "unknown"
                },
                {
                  "name": "guide_sequence",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "ncounts",
                  "rawType": "float32",
                  "type": "float"
                },
                {
                  "name": "nperts",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "ngenes",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "vector_id",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "perturbation_name",
                  "rawType": "category",
                  "type": "unknown"
                },
                {
                  "name": "percent_mito",
                  "rawType": "float32",
                  "type": "float"
                },
                {
                  "name": "perturbed_target_ensg",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "perturbed_target_symbol",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "perturbed_target_biotype",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "perturbed_target_coord",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "perturbed_target_chromosome",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "perturbed_target_number",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "perturbed_target_chromosome_encoding",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "dataset_id",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "sample_id",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "treatment_label",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "treatment_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "perturbation_type_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "perturbation_type_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "model_system_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "model_system_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "tissue",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "timepoint",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "species",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "sex_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "sex_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "developmental_stage_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "developmental_stage_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "study_title",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "study_uri",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "study_year",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "first_author",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "last_author",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "experiment_title",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "experiment_summary",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "number_of_perturbed_targets",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "number_of_perturbed_samples",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "library_generation_type_id",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_generation_type_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_generation_method_id",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_generation_method_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "enzyme_delivery_method_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "enzyme_delivery_method_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_delivery_method_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "library_delivery_method_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "enzyme_integration_state_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "enzyme_integration_state_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_integration_state_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "library_integration_state_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "enzyme_expression_control_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "enzyme_expression_control_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_expression_control_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "library_expression_control_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_name",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_uri",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "library_format_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "library_format_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_scope_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "library_scope_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_perturbation_type_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "library_perturbation_type_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_manufacturer",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_lentiviral_generation",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_grnas_per_target",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_total_grnas",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "library_total_variants",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "readout_dimensionality_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "readout_dimensionality_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "readout_type_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "readout_type_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "readout_technology_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "readout_technology_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "method_name_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "method_name_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "method_uri",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "sequencing_library_kit_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "sequencing_library_kit_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "sequencing_platform_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "sequencing_platform_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "sequencing_strategy_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "sequencing_strategy_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "software_counts_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "software_counts_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "software_analysis_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "software_analysis_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "reference_genome_id",
                  "rawType": "object",
                  "type": "unknown"
                },
                {
                  "name": "reference_genome_label",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "associated_datasets",
                  "rawType": "object",
                  "type": "string"
                }
              ],
              "ref": "9d94f30e-e68f-4611-839c-7cbf4f8f9abf",
              "rows": [
                [
                  "AAACATACAAGATG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "21.306112",
                  "282.0",
                  "K562",
                  "8.0",
                  "CRISPR",
                  "nan",
                  "8866.0",
                  "2",
                  "2914",
                  "pBA580",
                  "63(mod)_pBA580",
                  "4.917663",
                  null,
                  null,
                  null,
                  null,
                  null,
                  "1",
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "1",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACATACACCTAG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "19.4922",
                  "331.0",
                  "K562",
                  "7.0",
                  "CRISPR",
                  "GGCTTGTTCGCTGGTGGCGT",
                  "13785.0",
                  "2",
                  "3818",
                  "pDS353",
                  "OST4_pDS353",
                  "4.4686255",
                  "ENSG00000228474",
                  "OST4",
                  "protein_coding",
                  "chr2:27070472-27071654;-1",
                  "2",
                  "1",
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "2",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACATACTTCCCG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "23.199894",
                  "285.0",
                  "K562",
                  "10.0",
                  "CRISPR",
                  "GCTGTGCAGTGGAACGCGCT",
                  "7569.0",
                  "2",
                  "2616",
                  "pDS031",
                  "SEC61A1_pDS031",
                  "5.0601134",
                  "ENSG00000058262",
                  "SEC61A1",
                  "protein_coding",
                  "chr3:128051641-128071705;1",
                  "3",
                  "1",
                  "3",
                  "adamson_2016_upr_perturb_seq",
                  "3",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACATTGAAACAG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "28.733555",
                  "1036.0",
                  "K562",
                  "30.0",
                  "CRISPR",
                  "GCTGAGGGCGATGGCTGCTG",
                  "13834.0",
                  "2",
                  "3488",
                  "pDS491",
                  "EIF2B4_pDS491",
                  "5.0527687",
                  "ENSG00000115211",
                  "EIF2B4",
                  "protein_coding",
                  "chr2:27364352-27370338;-1",
                  "2",
                  "1",
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "4",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACATTGCAGCTA",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "26.729864",
                  "863.0",
                  "K562",
                  "25.0",
                  "CRISPR",
                  "GGCGAACGCGGCCTGAATTCC",
                  "15507.0",
                  "2",
                  "3620",
                  "pDS482",
                  "SRPR_pDS482",
                  "4.5140905",
                  "ENSG00000182934",
                  "SRPRA",
                  "protein_coding",
                  "chr11:126262938-126269144;-1",
                  "11",
                  "1",
                  "11",
                  "adamson_2016_upr_perturb_seq",
                  "5",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACATTGCATTGG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "19.329765",
                  "509.0",
                  "K562",
                  "20.0",
                  "CRISPR",
                  "nan",
                  "15696.0",
                  "2",
                  "3999",
                  "pBA580",
                  "63(mod)_pBA580",
                  "3.6314986",
                  null,
                  null,
                  null,
                  null,
                  null,
                  "1",
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "6",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACCGTGCTGTAG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "22.992075",
                  "274.0",
                  "K562",
                  "11.0",
                  "CRISPR",
                  "GGGCCGCGTGAGGGGACACCA",
                  "9338.0",
                  "2",
                  "2968",
                  "pDS002",
                  "IER3IP1_pDS002",
                  "3.116299",
                  "ENSG00000134049",
                  "IER3IP1",
                  "protein_coding",
                  "chr18:47152834-47176364;-1",
                  "18",
                  "1",
                  "18",
                  "adamson_2016_upr_perturb_seq",
                  "7",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACCGTGGCGTTA",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "25.323206",
                  "775.0",
                  "K562",
                  "24.0",
                  "CRISPR",
                  "GCTGAGGGCGATGGCTGCTG",
                  "13691.0",
                  "2",
                  "3531",
                  "pDS491",
                  "EIF2B4_pDS491",
                  "3.966109",
                  "ENSG00000115211",
                  "EIF2B4",
                  "protein_coding",
                  "chr2:27364352-27370338;-1",
                  "2",
                  "1",
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "8",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACCGTGGGACTT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "25.483871",
                  "954.0",
                  "K562",
                  "31.0",
                  "CRISPR",
                  "GCTGAGGGCGATGGCTGCTG",
                  "10540.0",
                  "2",
                  "3091",
                  "pDS491",
                  "EIF2B4_pDS491",
                  "4.193548",
                  "ENSG00000115211",
                  "EIF2B4",
                  "protein_coding",
                  "chr2:27364352-27370338;-1",
                  "2",
                  "1",
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "9",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACCGTGTCGATG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "21.545685",
                  "1153.0",
                  "K562",
                  "31.0",
                  "CRISPR",
                  "GGGTGCAGGGGACCGCGTCA",
                  "12991.0",
                  "2",
                  "3408",
                  "pDS186",
                  "YIPF5_pDS186",
                  "4.541606",
                  "ENSG00000145817",
                  "YIPF5",
                  "protein_coding",
                  "chr5:144158162-144170714;-1",
                  "5",
                  "1",
                  "5",
                  "adamson_2016_upr_perturb_seq",
                  "10",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACCGTGTTCCCG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "20.108805",
                  "762.0",
                  "K562",
                  "15.0",
                  "CRISPR",
                  "GCTCAAGTGGACAGCCGGGA",
                  "4963.0",
                  "2",
                  "2061",
                  "pDS466",
                  "HARS_pDS466",
                  "3.908926",
                  "ENSG00000170445",
                  "HARS1",
                  "protein_coding",
                  "chr5:140673035-140691537;-1",
                  "5",
                  "1",
                  "5",
                  "adamson_2016_upr_perturb_seq",
                  "11",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACGCACCCAAGT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "25.604298",
                  "378.0",
                  "K562",
                  "11.0",
                  "CRISPR",
                  "GAGATGTCCGGGTAGCGCCA",
                  "8936.0",
                  "2",
                  "2825",
                  "pDS433",
                  "SLMO2_pDS433",
                  "3.782453",
                  "ENSG00000101166",
                  "PRELID3B",
                  "protein_coding",
                  "chr20:59033145-59042809;-1",
                  "20",
                  "1",
                  "20",
                  "adamson_2016_upr_perturb_seq",
                  "12",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACGCACGACAGG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "25.780176",
                  "1064.0",
                  "K562",
                  "29.0",
                  "CRISPR",
                  "GACCTTGCGTGCAGTTATGT",
                  "9453.0",
                  "2",
                  "2975",
                  "pDS499",
                  "DAD1_pDS499",
                  "4.527663",
                  "ENSG00000129562",
                  "DAD1",
                  "protein_coding",
                  "chr14:22564907-22589224;-1",
                  "14",
                  "1",
                  "14",
                  "adamson_2016_upr_perturb_seq",
                  "13",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACGCACTGACTG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "21.82937",
                  "610.0",
                  "K562",
                  "17.0",
                  "CRISPR",
                  "GAAGTAGGCGCTGGCAACGC",
                  "17875.0",
                  "2",
                  "4159",
                  "pDS284",
                  "TIMM23_pDS284",
                  "3.5076923",
                  "ENSG00000265354",
                  "TIMM23",
                  "protein_coding",
                  "chr10:45972489-46003742;1",
                  "10",
                  "1",
                  "10",
                  "adamson_2016_upr_perturb_seq",
                  "14",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACGCTGAGGTCT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "23.440454",
                  "894.0",
                  "K562",
                  "95.0",
                  "CRISPR",
                  "GTGAGACCCCAGGGTCGGGA",
                  "9522.0",
                  "2",
                  "2798",
                  "pDS495",
                  "DARS_pDS495",
                  "1.8273472",
                  "ENSG00000115866",
                  "DARS1",
                  "protein_coding",
                  "chr2:135905881-135986100;-1",
                  "2",
                  "1",
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "15",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACGCTGGTGCAT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "25.984953",
                  "208.0",
                  "K562",
                  "4.0",
                  "CRISPR",
                  "GGGGCCCCATCGGCTTCCGT",
                  "14620.0",
                  "2",
                  "3696",
                  "pDS110",
                  "IER3IP1_pDS110",
                  "5.478796",
                  "ENSG00000134049",
                  "IER3IP1",
                  "protein_coding",
                  "chr18:47152834-47176364;-1",
                  "18",
                  "1",
                  "18",
                  "adamson_2016_upr_perturb_seq",
                  "16",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACGGCTAATGCC",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "21.365742",
                  "579.0",
                  "K562",
                  "14.0",
                  "CRISPR",
                  "GGGGCCAGCAGCCGTCTACC",
                  "14893.0",
                  "2",
                  "3947",
                  "pDS434",
                  "AMIGO3_pDS434",
                  "2.8201168",
                  "ENSG00000176020",
                  "AMIGO3",
                  "protein_coding",
                  "chr3:49716829-49719684;-1",
                  "3",
                  "1",
                  "3",
                  "adamson_2016_upr_perturb_seq",
                  "17",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACGGCTCAATCG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "27.837465",
                  "1461.0",
                  "K562",
                  "134.0",
                  "CRISPR",
                  "GCGCACAGACCCGGCGAGGA",
                  "14520.0",
                  "2",
                  "3638",
                  "pDS052",
                  "ASCC3_pDS052",
                  "4.366391",
                  "ENSG00000112249",
                  "ASCC3",
                  "protein_coding",
                  "chr6:100508194-100881372;-1",
                  "6",
                  "1",
                  "6",
                  "adamson_2016_upr_perturb_seq",
                  "18",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACGGCTGCAAGG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "26.322256",
                  "1277.0",
                  "K562",
                  "35.0",
                  "CRISPR",
                  "GTGAGACCCCAGGGTCGGGA",
                  "16279.0",
                  "2",
                  "3859",
                  "pDS495",
                  "DARS_pDS495",
                  "4.066589",
                  "ENSG00000115866",
                  "DARS1",
                  "protein_coding",
                  "chr2:135905881-135986100;-1",
                  "2",
                  "1",
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "19",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACGGCTGGATCT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "22.365671",
                  "171.0",
                  "K562",
                  "5.0",
                  "CRISPR",
                  "GGGGCCCGCGTGGATCTCTA",
                  "8133.0",
                  "2",
                  "2816",
                  "pDS393",
                  "IDH3A_pDS393",
                  "6.29534",
                  "ENSG00000166411",
                  "IDH3A",
                  "protein_coding",
                  "chr15:78131498-78171945;1",
                  "15",
                  "1",
                  "15",
                  "adamson_2016_upr_perturb_seq",
                  "20",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACTTGAAACGGG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "31.567749",
                  "1801.0",
                  "K562",
                  "100.0",
                  "CRISPR",
                  "GGGCGCCTGTGCTTGAGGTT",
                  "17860.0",
                  "2",
                  "3900",
                  "pDS026",
                  "DNAJC19_pDS026",
                  "5.330347",
                  "ENSG00000205981",
                  "DNAJC19",
                  "protein_coding",
                  "chr3:180983697-180989774;-1",
                  "3",
                  "1",
                  "3",
                  "adamson_2016_upr_perturb_seq",
                  "21",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACTTGAACGACT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "20.65235",
                  "116.0",
                  "K562",
                  "2.0",
                  "CRISPR",
                  "GCGGTAAGCAAACACTTACA",
                  "6193.0",
                  "2",
                  "2350",
                  "pDS040",
                  "UFM1_pDS040",
                  "8.832553",
                  "ENSG00000120686",
                  "UFM1",
                  "protein_coding",
                  "chr13:38349849-38363619;1",
                  "13",
                  "1",
                  "13",
                  "adamson_2016_upr_perturb_seq",
                  "22",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACTTGAACTGTG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "24.033325",
                  "571.0",
                  "K562",
                  "77.0",
                  "CRISPR",
                  "GCTGAGGGCGATGGCTGCTG",
                  "11043.0",
                  "2",
                  "3303",
                  "pDS491",
                  "EIF2B4_pDS491",
                  "3.585982",
                  "ENSG00000115211",
                  "EIF2B4",
                  "protein_coding",
                  "chr2:27364352-27370338;-1",
                  "2",
                  "1",
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "23",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACTTGAGATAAG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "25.717203",
                  "580.0",
                  "K562",
                  "26.0",
                  "CRISPR",
                  "GTGAGACCCCAGGGTCGGGA",
                  "9795.0",
                  "2",
                  "3143",
                  "pDS495",
                  "DARS_pDS495",
                  "3.5426238",
                  "ENSG00000115866",
                  "DARS1",
                  "protein_coding",
                  "chr2:135905881-135986100;-1",
                  "2",
                  "1",
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "24",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAACTTGAGCTAAC",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "24.738384",
                  "408.0",
                  "K562",
                  "12.0",
                  "CRISPR",
                  "GGCGCGTGCGCCCTCTTACT",
                  "16723.0",
                  "2",
                  "4078",
                  "pDS090",
                  "IARS2_pDS090",
                  "3.1393888",
                  "ENSG00000067704",
                  "IARS2",
                  "protein_coding",
                  "chr1:220094132-220148041;1",
                  "1",
                  "1",
                  "1",
                  "adamson_2016_upr_perturb_seq",
                  "25",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGACGAACTCAG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "28.320711",
                  "218.0",
                  "K562",
                  "8.0",
                  "CRISPR",
                  "nan",
                  "8319.0",
                  "2",
                  "2713",
                  "pBA580",
                  "63(mod)_pBA580",
                  "4.6520014",
                  null,
                  null,
                  null,
                  null,
                  null,
                  "1",
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "26",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGACGAATGGTC",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "30.856718",
                  "2682.0",
                  "K562",
                  "76.0",
                  "CRISPR",
                  "GCGCGCTCAGTGAGAGGAAA",
                  "29543.0",
                  "2",
                  "5030",
                  "pDS510",
                  "QARS_pDS510",
                  "5.4056797",
                  "ENSG00000136628",
                  "EPRS1",
                  "protein_coding",
                  "chr1:219968600-220046530;-1",
                  "1",
                  "1",
                  "1",
                  "adamson_2016_upr_perturb_seq",
                  "27",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGACGAGAGCAG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "27.600248",
                  "526.0",
                  "K562",
                  "19.0",
                  "CRISPR",
                  "GGCTTGTTCGCTGGTGGCGT",
                  "11297.0",
                  "2",
                  "3192",
                  "pDS353",
                  "OST4_pDS353",
                  "5.6829247",
                  "ENSG00000228474",
                  "OST4",
                  "protein_coding",
                  "chr2:27070472-27071654;-1",
                  "2",
                  "1",
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "28",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGAGACGATACC",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "26.82654",
                  "1262.0",
                  "K562",
                  "35.0",
                  "CRISPR",
                  "GCGGCGGACCCGGCCCTAGG",
                  "16096.0",
                  "2",
                  "3872",
                  "pDS479",
                  "SOCS1_pDS479",
                  "4.6595426",
                  "ENSG00000185338",
                  "SOCS1",
                  "protein_coding",
                  "chr16:11254417-11256204;-1",
                  "16",
                  "1",
                  "16",
                  "adamson_2016_upr_perturb_seq",
                  "29",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGAGACTCGCCT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "18.958202",
                  "914.0",
                  "K562",
                  "20.0",
                  "CRISPR",
                  "nan",
                  "14379.0",
                  "2",
                  "3678",
                  "pBA580",
                  "63(mod)_pBA580",
                  "6.711176",
                  null,
                  null,
                  null,
                  null,
                  null,
                  "1",
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "30",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGAGACTGCTTT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "23.0552",
                  "813.0",
                  "K562",
                  "37.0",
                  "CRISPR",
                  "nan",
                  "26558.0",
                  "2",
                  "4922",
                  "pBA580",
                  "63(mod)_pBA580",
                  "4.721741",
                  null,
                  null,
                  null,
                  null,
                  null,
                  "1",
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "31",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGATCTTCTTAC",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "21.62519",
                  "1011.0",
                  "K562",
                  "33.0",
                  "CRISPR",
                  "GTGACTCGCAGTAGACGCGG",
                  "19690.0",
                  "2",
                  "4176",
                  "pDS410",
                  "UFL1_pDS410",
                  "5.241239",
                  "ENSG00000014123",
                  "UFL1",
                  "protein_coding",
                  "chr6:96521595-96555276;1",
                  "6",
                  "1",
                  "6",
                  "adamson_2016_upr_perturb_seq",
                  "32",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGCAGACTTGGA",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "26.90094",
                  "1342.0",
                  "K562",
                  "41.0",
                  "CRISPR",
                  "GCGGTCCACAAAGGCTCAGA",
                  "23107.0",
                  "2",
                  "4548",
                  "pDS041",
                  "DDRGK1_pDS041",
                  "3.353962",
                  "ENSG00000198171",
                  "DDRGK1",
                  "protein_coding",
                  "chr20:3190350-3204685;-1",
                  "20",
                  "1",
                  "20",
                  "adamson_2016_upr_perturb_seq",
                  "33",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGCAGAGAAGGC",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "26.075504",
                  "235.0",
                  "K562",
                  "16.0",
                  "CRISPR",
                  "GCGCGCTCAGTGAGAGGAAA",
                  "4556.0",
                  "2",
                  "1787",
                  "pDS510",
                  "QARS_pDS510",
                  "5.1580334",
                  "ENSG00000136628",
                  "EPRS1",
                  "protein_coding",
                  "chr1:219968600-220046530;-1",
                  "1",
                  "1",
                  "1",
                  "adamson_2016_upr_perturb_seq",
                  "34",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGCCTGAAACGA",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "25.613945",
                  "1254.0",
                  "K562",
                  "43.0",
                  "CRISPR",
                  "nan",
                  "7859.0",
                  "2",
                  "2553",
                  "pBA580",
                  "63(mod)_pBA580",
                  "0.5471434",
                  null,
                  null,
                  null,
                  null,
                  null,
                  "1",
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "35",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGCCTGAGTTCG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "23.38847",
                  "216.0",
                  "K562",
                  "6.0",
                  "CRISPR",
                  "nan",
                  "8222.0",
                  "2",
                  "2733",
                  "pBA582",
                  "Gal4-4(mod)_pBA582",
                  "7.3461447",
                  null,
                  null,
                  null,
                  null,
                  null,
                  "1",
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "36",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGCCTGATTTCC",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "26.864754",
                  "1289.0",
                  "K562",
                  "36.0",
                  "CRISPR",
                  "GGGAGCCCCGCGGATCGTTT",
                  "14640.0",
                  "2",
                  "3421",
                  "pDS011",
                  "STT3A_pDS011",
                  "7.3497267",
                  "ENSG00000134910",
                  "STT3A",
                  "protein_coding",
                  "chr11:125591712-125625215;1",
                  "11",
                  "1",
                  "11",
                  "adamson_2016_upr_perturb_seq",
                  "37",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGCCTGCGAGAG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "26.49352",
                  "1353.0",
                  "K562",
                  "35.0",
                  "CRISPR",
                  "GCTGCCTCGCAGAGCTGCCC",
                  "18597.0",
                  "2",
                  "3918",
                  "pDS156",
                  "SAMM50_pDS156",
                  "3.3338711",
                  "ENSG00000100347",
                  "SAMM50",
                  "protein_coding",
                  "chr22:43955442-44010531;1",
                  "22",
                  "1",
                  "22",
                  "adamson_2016_upr_perturb_seq",
                  "38",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGCCTGGTTAGC",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "20.743885",
                  "1008.0",
                  "K562",
                  "27.0",
                  "CRISPR",
                  "GGCGGACGGATAGAAAGTTA",
                  "12018.0",
                  "2",
                  "3424",
                  "pDS162",
                  "SEC61B_pDS162",
                  "7.6302214",
                  "ENSG00000106803",
                  "SEC61B",
                  "protein_coding",
                  "chr9:99222064-99230615;1",
                  "9",
                  "1",
                  "9",
                  "adamson_2016_upr_perturb_seq",
                  "39",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGCCTGTCATTC",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "27.8403",
                  "168.0",
                  "K562",
                  "7.0",
                  "CRISPR",
                  "GGGCCGCGTGAGGGGACACCA",
                  "3732.0",
                  "2",
                  "1593",
                  "pDS002",
                  "IER3IP1_pDS002",
                  "4.608789",
                  "ENSG00000134049",
                  "IER3IP1",
                  "protein_coding",
                  "chr18:47152834-47176364;-1",
                  "18",
                  "1",
                  "18",
                  "adamson_2016_upr_perturb_seq",
                  "40",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGGCCTGTCACA",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "31.233671",
                  "936.0",
                  "K562",
                  "19.0",
                  "CRISPR",
                  "GGCGCGTGCGCCCTCTTACT",
                  "6890.0",
                  "2",
                  "2340",
                  "pDS090",
                  "IARS2_pDS090",
                  "1.9303339",
                  "ENSG00000067704",
                  "IARS2",
                  "protein_coding",
                  "chr1:220094132-220148041;1",
                  "1",
                  "1",
                  "1",
                  "adamson_2016_upr_perturb_seq",
                  "41",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGGCCTTCTGGA",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "20.25641",
                  "287.0",
                  "K562",
                  "9.0",
                  "CRISPR",
                  "nan",
                  "6630.0",
                  "2",
                  "2533",
                  "pBA580",
                  "63(mod)_pBA580",
                  "3.2579186",
                  null,
                  null,
                  null,
                  null,
                  null,
                  "1",
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "42",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGTTTGGTAGGG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "22.923056",
                  "604.0",
                  "K562",
                  "14.0",
                  "CRISPR",
                  "GGTCCGTCCGGAGCCAGGAG",
                  "14348.0",
                  "2",
                  "3810",
                  "pDS458",
                  "ARHGAP22_pDS458",
                  "3.9238918",
                  "ENSG00000128805",
                  "ARHGAP22",
                  "protein_coding",
                  "chr10:48446036-48656265;-1",
                  "10",
                  "1",
                  "10",
                  "adamson_2016_upr_perturb_seq",
                  "43",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGTTTGTCCCGT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "18.672956",
                  "64.0",
                  "K562",
                  "1.0",
                  "CRISPR",
                  "GCGCGCTCAGTGAGAGGAAA",
                  "6571.0",
                  "2",
                  "2615",
                  "pDS510",
                  "QARS_pDS510",
                  "5.432963",
                  "ENSG00000136628",
                  "EPRS1",
                  "protein_coding",
                  "chr1:219968600-220046530;-1",
                  "1",
                  "1",
                  "1",
                  "adamson_2016_upr_perturb_seq",
                  "44",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAAGTTTGTCGATG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "23.315693",
                  "1262.0",
                  "K562",
                  "35.0",
                  "CRISPR",
                  "GTCAGACAAGTCGGAGGGGA",
                  "23808.0",
                  "2",
                  "4776",
                  "pDS401",
                  "SPCS2_pDS401",
                  "4.5278897",
                  "ENSG00000118363",
                  "SPCS2",
                  "protein_coding",
                  "chr11:74949261-74979033;1",
                  "11",
                  "1",
                  "11",
                  "adamson_2016_upr_perturb_seq",
                  "45",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAATACTGCACACA",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "24.970274",
                  "248.0",
                  "K562",
                  "9.0",
                  "CRISPR",
                  "nan",
                  "3364.0",
                  "2",
                  "1467",
                  "pBA580",
                  "63(mod)_pBA580",
                  "4.3697977",
                  null,
                  null,
                  null,
                  null,
                  null,
                  "1",
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "46",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAATACTGCATACG",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "23.119303",
                  "210.0",
                  "K562",
                  "8.0",
                  "CRISPR",
                  "GCGAGCAAGGAGCCCCCGGC",
                  "14702.0",
                  "2",
                  "3658",
                  "pDS124",
                  "MRGBP_pDS124",
                  "6.7677865",
                  "ENSG00000101189",
                  "MRGBP",
                  "protein_coding",
                  "chr20:62796473-62801729;1",
                  "20",
                  "1",
                  "20",
                  "adamson_2016_upr_perturb_seq",
                  "47",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAATACTGGGACGA",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "25.275394",
                  "167.0",
                  "K562",
                  "9.0",
                  "CRISPR",
                  "GGGTGCAGGGGACCGCGTCA",
                  "16431.0",
                  "2",
                  "3927",
                  "pDS186",
                  "YIPF5_pDS186",
                  "3.761183",
                  "ENSG00000145817",
                  "YIPF5",
                  "protein_coding",
                  "chr5:144158162-144170714;-1",
                  "5",
                  "1",
                  "5",
                  "adamson_2016_upr_perturb_seq",
                  "48",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAATACTGGTCGAT",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "20.475403",
                  "114.0",
                  "K562",
                  "4.0",
                  "CRISPR",
                  "GTACGGAGCCCCCGAAAGAC",
                  "6647.0",
                  "2",
                  "2460",
                  "pDS033",
                  "SEC61B_pDS033",
                  "5.370844",
                  "ENSG00000106803",
                  "SEC61B",
                  "protein_coding",
                  "chr9:99222064-99230615;1",
                  "9",
                  "1",
                  "9",
                  "adamson_2016_upr_perturb_seq",
                  "49",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "AAATACTGTGTCGA",
                  "cell_line",
                  "lymphoblasts",
                  "human",
                  "True",
                  "chronic myelogenous leukemia",
                  "17.930334",
                  "5412.0",
                  "K562",
                  "175.0",
                  "CRISPR",
                  "GGTGGTGGCTGAGTCCGTGG",
                  "16737.0",
                  "2",
                  "4059",
                  "pDS373",
                  "SEL1L_pDS373",
                  "6.5005674",
                  "ENSG00000071537",
                  "SEL1L",
                  "protein_coding",
                  "chr14:81471547-81533853;-1",
                  "14",
                  "1",
                  "14",
                  "adamson_2016_upr_perturb_seq",
                  "50",
                  null,
                  null,
                  "CRISPRi",
                  null,
                  "cell line",
                  null,
                  "blood",
                  "P0DT0H0M0S",
                  "Homo sapiens",
                  "female",
                  null,
                  "adult",
                  null,
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ]
              ],
              "shape": {
                "columns": 99,
                "rows": 62623
              }
            },
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>tissue_type</th>\n",
              "      <th>celltype</th>\n",
              "      <th>organism</th>\n",
              "      <th>cancer</th>\n",
              "      <th>disease</th>\n",
              "      <th>percent_ribo</th>\n",
              "      <th>read count</th>\n",
              "      <th>cell_line</th>\n",
              "      <th>UMI count</th>\n",
              "      <th>perturbation_type</th>\n",
              "      <th>...</th>\n",
              "      <th>sequencing_platform_label</th>\n",
              "      <th>sequencing_strategy_id</th>\n",
              "      <th>sequencing_strategy_label</th>\n",
              "      <th>software_counts_id</th>\n",
              "      <th>software_counts_label</th>\n",
              "      <th>software_analysis_id</th>\n",
              "      <th>software_analysis_label</th>\n",
              "      <th>reference_genome_id</th>\n",
              "      <th>reference_genome_label</th>\n",
              "      <th>associated_datasets</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>AAACATACAAGATG</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>21.306112</td>\n",
              "      <td>282.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>8.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACATACACCTAG</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>19.492201</td>\n",
              "      <td>331.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>7.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACATACTTCCCG</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>23.199894</td>\n",
              "      <td>285.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>10.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACATTGAAACAG</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>28.733555</td>\n",
              "      <td>1036.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>30.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AAACATTGCAGCTA</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>26.729864</td>\n",
              "      <td>863.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>25.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>TTTGCATGCTTTAC</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>22.679703</td>\n",
              "      <td>476.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>17.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>TTTGCATGGAGGAC</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>26.983313</td>\n",
              "      <td>539.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>19.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>TTTGCATGTAGAGA</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>26.207104</td>\n",
              "      <td>647.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>35.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>TTTGCATGTCAAGC</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>20.755890</td>\n",
              "      <td>98.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>4.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>TTTGCATGTGGAGG</th>\n",
              "      <td>cell_line</td>\n",
              "      <td>lymphoblasts</td>\n",
              "      <td>human</td>\n",
              "      <td>True</td>\n",
              "      <td>chronic myelogenous leukemia</td>\n",
              "      <td>20.097370</td>\n",
              "      <td>427.0</td>\n",
              "      <td>K562</td>\n",
              "      <td>14.0</td>\n",
              "      <td>CRISPR</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>None</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>None</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>None</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>None</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>62623 rows × 99 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "               tissue_type      celltype organism  cancer  \\\n",
              "index                                                       \n",
              "AAACATACAAGATG   cell_line  lymphoblasts    human    True   \n",
              "AAACATACACCTAG   cell_line  lymphoblasts    human    True   \n",
              "AAACATACTTCCCG   cell_line  lymphoblasts    human    True   \n",
              "AAACATTGAAACAG   cell_line  lymphoblasts    human    True   \n",
              "AAACATTGCAGCTA   cell_line  lymphoblasts    human    True   \n",
              "...                    ...           ...      ...     ...   \n",
              "TTTGCATGCTTTAC   cell_line  lymphoblasts    human    True   \n",
              "TTTGCATGGAGGAC   cell_line  lymphoblasts    human    True   \n",
              "TTTGCATGTAGAGA   cell_line  lymphoblasts    human    True   \n",
              "TTTGCATGTCAAGC   cell_line  lymphoblasts    human    True   \n",
              "TTTGCATGTGGAGG   cell_line  lymphoblasts    human    True   \n",
              "\n",
              "                                     disease  percent_ribo  read count  \\\n",
              "index                                                                    \n",
              "AAACATACAAGATG  chronic myelogenous leukemia     21.306112       282.0   \n",
              "AAACATACACCTAG  chronic myelogenous leukemia     19.492201       331.0   \n",
              "AAACATACTTCCCG  chronic myelogenous leukemia     23.199894       285.0   \n",
              "AAACATTGAAACAG  chronic myelogenous leukemia     28.733555      1036.0   \n",
              "AAACATTGCAGCTA  chronic myelogenous leukemia     26.729864       863.0   \n",
              "...                                      ...           ...         ...   \n",
              "TTTGCATGCTTTAC  chronic myelogenous leukemia     22.679703       476.0   \n",
              "TTTGCATGGAGGAC  chronic myelogenous leukemia     26.983313       539.0   \n",
              "TTTGCATGTAGAGA  chronic myelogenous leukemia     26.207104       647.0   \n",
              "TTTGCATGTCAAGC  chronic myelogenous leukemia     20.755890        98.0   \n",
              "TTTGCATGTGGAGG  chronic myelogenous leukemia     20.097370       427.0   \n",
              "\n",
              "               cell_line  UMI count perturbation_type  ...  \\\n",
              "index                                                  ...   \n",
              "AAACATACAAGATG      K562        8.0            CRISPR  ...   \n",
              "AAACATACACCTAG      K562        7.0            CRISPR  ...   \n",
              "AAACATACTTCCCG      K562       10.0            CRISPR  ...   \n",
              "AAACATTGAAACAG      K562       30.0            CRISPR  ...   \n",
              "AAACATTGCAGCTA      K562       25.0            CRISPR  ...   \n",
              "...                  ...        ...               ...  ...   \n",
              "TTTGCATGCTTTAC      K562       17.0            CRISPR  ...   \n",
              "TTTGCATGGAGGAC      K562       19.0            CRISPR  ...   \n",
              "TTTGCATGTAGAGA      K562       35.0            CRISPR  ...   \n",
              "TTTGCATGTCAAGC      K562        4.0            CRISPR  ...   \n",
              "TTTGCATGTGGAGG      K562       14.0            CRISPR  ...   \n",
              "\n",
              "               sequencing_platform_label  sequencing_strategy_id  \\\n",
              "index                                                              \n",
              "AAACATACAAGATG       Illumina HiSeq 4000                    None   \n",
              "AAACATACACCTAG       Illumina HiSeq 4000                    None   \n",
              "AAACATACTTCCCG       Illumina HiSeq 4000                    None   \n",
              "AAACATTGAAACAG       Illumina HiSeq 4000                    None   \n",
              "AAACATTGCAGCTA       Illumina HiSeq 4000                    None   \n",
              "...                                  ...                     ...   \n",
              "TTTGCATGCTTTAC       Illumina HiSeq 4000                    None   \n",
              "TTTGCATGGAGGAC       Illumina HiSeq 4000                    None   \n",
              "TTTGCATGTAGAGA       Illumina HiSeq 4000                    None   \n",
              "TTTGCATGTCAAGC       Illumina HiSeq 4000                    None   \n",
              "TTTGCATGTGGAGG       Illumina HiSeq 4000                    None   \n",
              "\n",
              "                sequencing_strategy_label  software_counts_id  \\\n",
              "index                                                           \n",
              "AAACATACAAGATG         barcode sequencing                None   \n",
              "AAACATACACCTAG         barcode sequencing                None   \n",
              "AAACATACTTCCCG         barcode sequencing                None   \n",
              "AAACATTGAAACAG         barcode sequencing                None   \n",
              "AAACATTGCAGCTA         barcode sequencing                None   \n",
              "...                                   ...                 ...   \n",
              "TTTGCATGCTTTAC         barcode sequencing                None   \n",
              "TTTGCATGGAGGAC         barcode sequencing                None   \n",
              "TTTGCATGTAGAGA         barcode sequencing                None   \n",
              "TTTGCATGTCAAGC         barcode sequencing                None   \n",
              "TTTGCATGTGGAGG         barcode sequencing                None   \n",
              "\n",
              "               software_counts_label software_analysis_id  \\\n",
              "index                                                       \n",
              "AAACATACAAGATG            CellRanger                 None   \n",
              "AAACATACACCTAG            CellRanger                 None   \n",
              "AAACATACTTCCCG            CellRanger                 None   \n",
              "AAACATTGAAACAG            CellRanger                 None   \n",
              "AAACATTGCAGCTA            CellRanger                 None   \n",
              "...                              ...                  ...   \n",
              "TTTGCATGCTTTAC            CellRanger                 None   \n",
              "TTTGCATGGAGGAC            CellRanger                 None   \n",
              "TTTGCATGTAGAGA            CellRanger                 None   \n",
              "TTTGCATGTCAAGC            CellRanger                 None   \n",
              "TTTGCATGTGGAGG            CellRanger                 None   \n",
              "\n",
              "                software_analysis_label reference_genome_id  \\\n",
              "index                                                         \n",
              "AAACATACAAGATG                   MAGeCK                None   \n",
              "AAACATACACCTAG                   MAGeCK                None   \n",
              "AAACATACTTCCCG                   MAGeCK                None   \n",
              "AAACATTGAAACAG                   MAGeCK                None   \n",
              "AAACATTGCAGCTA                   MAGeCK                None   \n",
              "...                                 ...                 ...   \n",
              "TTTGCATGCTTTAC                   MAGeCK                None   \n",
              "TTTGCATGGAGGAC                   MAGeCK                None   \n",
              "TTTGCATGTAGAGA                   MAGeCK                None   \n",
              "TTTGCATGTCAAGC                   MAGeCK                None   \n",
              "TTTGCATGTGGAGG                   MAGeCK                None   \n",
              "\n",
              "               reference_genome_label  \\\n",
              "index                                   \n",
              "AAACATACAAGATG                 GRCh37   \n",
              "AAACATACACCTAG                 GRCh37   \n",
              "AAACATACTTCCCG                 GRCh37   \n",
              "AAACATTGAAACAG                 GRCh37   \n",
              "AAACATTGCAGCTA                 GRCh37   \n",
              "...                               ...   \n",
              "TTTGCATGCTTTAC                 GRCh37   \n",
              "TTTGCATGGAGGAC                 GRCh37   \n",
              "TTTGCATGTAGAGA                 GRCh37   \n",
              "TTTGCATGTCAAGC                 GRCh37   \n",
              "TTTGCATGTGGAGG                 GRCh37   \n",
              "\n",
              "                                              associated_datasets  \n",
              "index                                                              \n",
              "AAACATACAAGATG  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "AAACATACACCTAG  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "AAACATACTTCCCG  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "AAACATTGAAACAG  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "AAACATTGCAGCTA  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "...                                                           ...  \n",
              "TTTGCATGCTTTAC  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "TTTGCATGGAGGAC  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "TTTGCATGTAGAGA  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "TTTGCATGTCAAGC  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "TTTGCATGTGGAGG  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "\n",
              "[62623 rows x 99 columns]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "cur_data.adata.obs"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "81be5554",
      "metadata": {},
      "source": [
        "### Curate tissue information"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "2f3ab03a",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Mapped 1 tissue ontology terms from `tissue` column to ontology terms\n",
            "DataFrame shape: (1, 4)\n",
            "--------------------------------------------------\n",
            "  input_column input_column_lower name_lower     ontology_id\n",
            "0        blood              blood      blood  UBERON:0000178\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "\n",
        "cur_data.standardize_ontology(\n",
        "    input_column='tissue',\n",
        "    column_type='term_name',\n",
        "    ontology_type='tissue',\n",
        "    overwrite=True\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "2e6fa9a9",
      "metadata": {},
      "source": [
        "### Curate cell type information"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "31ef57cb",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Mapped 1 cell_type ontology terms from `celltype` column to ontology terms\n",
            "DataFrame shape: (1, 4)\n",
            "--------------------------------------------------\n",
            "   input_column input_column_lower    name_lower ontology_id\n",
            "0  lymphoblasts       lymphoblasts  lymphoblasts  CL:0017005\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.standardize_ontology(\n",
        "    input_column='cell_type',\n",
        "    column_type='term_name',\n",
        "    ontology_type='cell_type'\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "7dbfad9d",
      "metadata": {},
      "source": [
        "### Curate cell line information"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "aabc31b0",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Mapped 1 cell_line ontology terms from `cell_line` column to ontology terms\n",
            "DataFrame shape: (1, 4)\n",
            "--------------------------------------------------\n",
            "  input_column input_column_lower name_lower  ontology_id\n",
            "0         K562               k562       k562  CLO:0007050\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.standardize_ontology(\n",
        "    input_column='cell_line',\n",
        "    column_type='term_name',\n",
        "    ontology_type='cell_line'\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "691ae35a",
      "metadata": {},
      "source": [
        "### Curate disease information"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "a050160c",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Mapped 1 disease ontology terms from `disease` column to ontology terms\n",
            "DataFrame shape: (1, 4)\n",
            "--------------------------------------------------\n",
            "                   input_column            input_column_lower  \\\n",
            "0  chronic myelogenous leukemia  chronic myelogenous leukemia   \n",
            "\n",
            "                     name_lower    ontology_id  \n",
            "0  chronic myelogenous leukemia  MONDO:0011996  \n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.standardize_ontology(\n",
        "    input_column='disease',\n",
        "    column_type='term_name',\n",
        "    ontology_type='disease'\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "4c1fd9b7",
      "metadata": {},
      "source": [
        "### Match schema column order"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "7025f6a5",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Matched columns of adata.obs to the obs_schema.\n"
          ]
        }
      ],
      "source": [
        "cur_data.match_schema_columns(slot='obs')"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "53eb5170",
      "metadata": {},
      "source": [
        "### Validate obs metadata"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "b288f724",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "adata.obs is valid according to the obs_schema.\n",
            "Validated data:\n"
          ]
        },
        {
          "data": {
            "application/vnd.microsoft.datawrangler.viewer.v0+json": {
              "columns": [
                {
                  "name": "index",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "dataset_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "sample_id",
                  "rawType": "Int64",
                  "type": "integer"
                },
                {
                  "name": "perturbation_name",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "perturbed_target_coord",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "perturbed_target_chromosome",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "perturbed_target_chromosome_encoding",
                  "rawType": "Int64",
                  "type": "integer"
                },
                {
                  "name": "perturbed_target_number",
                  "rawType": "Int64",
                  "type": "integer"
                },
                {
                  "name": "perturbed_target_ensg",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "perturbed_target_symbol",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "perturbed_target_biotype",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "guide_sequence",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "perturbation_type_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "perturbation_type_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "timepoint",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "treatment_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "treatment_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "model_system_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "model_system_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "species",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "tissue_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "tissue_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "cell_type_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "cell_type_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "cell_line_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "cell_line_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "sex_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "sex_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "developmental_stage_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "developmental_stage_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "disease_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "disease_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "study_title",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "study_uri",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "study_year",
                  "rawType": "Int64",
                  "type": "integer"
                },
                {
                  "name": "first_author",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "last_author",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "experiment_title",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "experiment_summary",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "number_of_perturbed_targets",
                  "rawType": "Int64",
                  "type": "integer"
                },
                {
                  "name": "number_of_perturbed_samples",
                  "rawType": "Int64",
                  "type": "integer"
                },
                {
                  "name": "library_generation_type_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_generation_type_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_generation_method_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_generation_method_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "enzyme_delivery_method_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "enzyme_delivery_method_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_delivery_method_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_delivery_method_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "enzyme_integration_state_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "enzyme_integration_state_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_integration_state_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_integration_state_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "enzyme_expression_control_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "enzyme_expression_control_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_expression_control_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_expression_control_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_name",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_uri",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_format_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_format_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_scope_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_scope_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_perturbation_type_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_perturbation_type_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_manufacturer",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_lentiviral_generation",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_grnas_per_target",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "library_total_grnas",
                  "rawType": "Int64",
                  "type": "integer"
                },
                {
                  "name": "library_total_variants",
                  "rawType": "Int64",
                  "type": "integer"
                },
                {
                  "name": "readout_dimensionality_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "readout_dimensionality_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "readout_type_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "readout_type_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "readout_technology_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "readout_technology_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "method_name_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "method_name_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "method_uri",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "sequencing_library_kit_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "sequencing_library_kit_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "sequencing_platform_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "sequencing_platform_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "sequencing_strategy_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "sequencing_strategy_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "software_counts_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "software_counts_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "software_analysis_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "software_analysis_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "reference_genome_id",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "reference_genome_label",
                  "rawType": "string",
                  "type": "string"
                },
                {
                  "name": "associated_datasets",
                  "rawType": "object",
                  "type": "string"
                }
              ],
              "ref": "5ca6e2ed-617b-414d-9157-5262e554271f",
              "rows": [
                [
                  "0",
                  "adamson_2016_upr_perturb_seq",
                  "1",
                  "63(mod)_pBA580",
                  null,
                  null,
                  "0",
                  "1",
                  null,
                  null,
                  null,
                  "nan",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "1",
                  "adamson_2016_upr_perturb_seq",
                  "2",
                  "OST4_pDS353",
                  "chr2:27070472-27071654;-1",
                  "2",
                  "2",
                  "1",
                  "ENSG00000228474",
                  "OST4",
                  "protein_coding",
                  "GGCTTGTTCGCTGGTGGCGT",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "2",
                  "adamson_2016_upr_perturb_seq",
                  "3",
                  "SEC61A1_pDS031",
                  "chr3:128051641-128071705;1",
                  "3",
                  "3",
                  "1",
                  "ENSG00000058262",
                  "SEC61A1",
                  "protein_coding",
                  "GCTGTGCAGTGGAACGCGCT",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "3",
                  "adamson_2016_upr_perturb_seq",
                  "4",
                  "EIF2B4_pDS491",
                  "chr2:27364352-27370338;-1",
                  "2",
                  "2",
                  "1",
                  "ENSG00000115211",
                  "EIF2B4",
                  "protein_coding",
                  "GCTGAGGGCGATGGCTGCTG",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "4",
                  "adamson_2016_upr_perturb_seq",
                  "5",
                  "SRPR_pDS482",
                  "chr11:126262938-126269144;-1",
                  "11",
                  "11",
                  "1",
                  "ENSG00000182934",
                  "SRPRA",
                  "protein_coding",
                  "GGCGAACGCGGCCTGAATTCC",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "5",
                  "adamson_2016_upr_perturb_seq",
                  "6",
                  "63(mod)_pBA580",
                  null,
                  null,
                  "0",
                  "1",
                  null,
                  null,
                  null,
                  "nan",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "6",
                  "adamson_2016_upr_perturb_seq",
                  "7",
                  "IER3IP1_pDS002",
                  "chr18:47152834-47176364;-1",
                  "18",
                  "18",
                  "1",
                  "ENSG00000134049",
                  "IER3IP1",
                  "protein_coding",
                  "GGGCCGCGTGAGGGGACACCA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "7",
                  "adamson_2016_upr_perturb_seq",
                  "8",
                  "EIF2B4_pDS491",
                  "chr2:27364352-27370338;-1",
                  "2",
                  "2",
                  "1",
                  "ENSG00000115211",
                  "EIF2B4",
                  "protein_coding",
                  "GCTGAGGGCGATGGCTGCTG",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "8",
                  "adamson_2016_upr_perturb_seq",
                  "9",
                  "EIF2B4_pDS491",
                  "chr2:27364352-27370338;-1",
                  "2",
                  "2",
                  "1",
                  "ENSG00000115211",
                  "EIF2B4",
                  "protein_coding",
                  "GCTGAGGGCGATGGCTGCTG",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "9",
                  "adamson_2016_upr_perturb_seq",
                  "10",
                  "YIPF5_pDS186",
                  "chr5:144158162-144170714;-1",
                  "5",
                  "5",
                  "1",
                  "ENSG00000145817",
                  "YIPF5",
                  "protein_coding",
                  "GGGTGCAGGGGACCGCGTCA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "10",
                  "adamson_2016_upr_perturb_seq",
                  "11",
                  "HARS_pDS466",
                  "chr5:140673035-140691537;-1",
                  "5",
                  "5",
                  "1",
                  "ENSG00000170445",
                  "HARS1",
                  "protein_coding",
                  "GCTCAAGTGGACAGCCGGGA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "11",
                  "adamson_2016_upr_perturb_seq",
                  "12",
                  "SLMO2_pDS433",
                  "chr20:59033145-59042809;-1",
                  "20",
                  "20",
                  "1",
                  "ENSG00000101166",
                  "PRELID3B",
                  "protein_coding",
                  "GAGATGTCCGGGTAGCGCCA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "12",
                  "adamson_2016_upr_perturb_seq",
                  "13",
                  "DAD1_pDS499",
                  "chr14:22564907-22589224;-1",
                  "14",
                  "14",
                  "1",
                  "ENSG00000129562",
                  "DAD1",
                  "protein_coding",
                  "GACCTTGCGTGCAGTTATGT",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "13",
                  "adamson_2016_upr_perturb_seq",
                  "14",
                  "TIMM23_pDS284",
                  "chr10:45972489-46003742;1",
                  "10",
                  "10",
                  "1",
                  "ENSG00000265354",
                  "TIMM23",
                  "protein_coding",
                  "GAAGTAGGCGCTGGCAACGC",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "14",
                  "adamson_2016_upr_perturb_seq",
                  "15",
                  "DARS_pDS495",
                  "chr2:135905881-135986100;-1",
                  "2",
                  "2",
                  "1",
                  "ENSG00000115866",
                  "DARS1",
                  "protein_coding",
                  "GTGAGACCCCAGGGTCGGGA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "15",
                  "adamson_2016_upr_perturb_seq",
                  "16",
                  "IER3IP1_pDS110",
                  "chr18:47152834-47176364;-1",
                  "18",
                  "18",
                  "1",
                  "ENSG00000134049",
                  "IER3IP1",
                  "protein_coding",
                  "GGGGCCCCATCGGCTTCCGT",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "16",
                  "adamson_2016_upr_perturb_seq",
                  "17",
                  "AMIGO3_pDS434",
                  "chr3:49716829-49719684;-1",
                  "3",
                  "3",
                  "1",
                  "ENSG00000176020",
                  "AMIGO3",
                  "protein_coding",
                  "GGGGCCAGCAGCCGTCTACC",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "17",
                  "adamson_2016_upr_perturb_seq",
                  "18",
                  "ASCC3_pDS052",
                  "chr6:100508194-100881372;-1",
                  "6",
                  "6",
                  "1",
                  "ENSG00000112249",
                  "ASCC3",
                  "protein_coding",
                  "GCGCACAGACCCGGCGAGGA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "18",
                  "adamson_2016_upr_perturb_seq",
                  "19",
                  "DARS_pDS495",
                  "chr2:135905881-135986100;-1",
                  "2",
                  "2",
                  "1",
                  "ENSG00000115866",
                  "DARS1",
                  "protein_coding",
                  "GTGAGACCCCAGGGTCGGGA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "19",
                  "adamson_2016_upr_perturb_seq",
                  "20",
                  "IDH3A_pDS393",
                  "chr15:78131498-78171945;1",
                  "15",
                  "15",
                  "1",
                  "ENSG00000166411",
                  "IDH3A",
                  "protein_coding",
                  "GGGGCCCGCGTGGATCTCTA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "20",
                  "adamson_2016_upr_perturb_seq",
                  "21",
                  "DNAJC19_pDS026",
                  "chr3:180983697-180989774;-1",
                  "3",
                  "3",
                  "1",
                  "ENSG00000205981",
                  "DNAJC19",
                  "protein_coding",
                  "GGGCGCCTGTGCTTGAGGTT",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "21",
                  "adamson_2016_upr_perturb_seq",
                  "22",
                  "UFM1_pDS040",
                  "chr13:38349849-38363619;1",
                  "13",
                  "13",
                  "1",
                  "ENSG00000120686",
                  "UFM1",
                  "protein_coding",
                  "GCGGTAAGCAAACACTTACA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "22",
                  "adamson_2016_upr_perturb_seq",
                  "23",
                  "EIF2B4_pDS491",
                  "chr2:27364352-27370338;-1",
                  "2",
                  "2",
                  "1",
                  "ENSG00000115211",
                  "EIF2B4",
                  "protein_coding",
                  "GCTGAGGGCGATGGCTGCTG",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "23",
                  "adamson_2016_upr_perturb_seq",
                  "24",
                  "DARS_pDS495",
                  "chr2:135905881-135986100;-1",
                  "2",
                  "2",
                  "1",
                  "ENSG00000115866",
                  "DARS1",
                  "protein_coding",
                  "GTGAGACCCCAGGGTCGGGA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "24",
                  "adamson_2016_upr_perturb_seq",
                  "25",
                  "IARS2_pDS090",
                  "chr1:220094132-220148041;1",
                  "1",
                  "1",
                  "1",
                  "ENSG00000067704",
                  "IARS2",
                  "protein_coding",
                  "GGCGCGTGCGCCCTCTTACT",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "25",
                  "adamson_2016_upr_perturb_seq",
                  "26",
                  "63(mod)_pBA580",
                  null,
                  null,
                  "0",
                  "1",
                  null,
                  null,
                  null,
                  "nan",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "26",
                  "adamson_2016_upr_perturb_seq",
                  "27",
                  "QARS_pDS510",
                  "chr1:219968600-220046530;-1",
                  "1",
                  "1",
                  "1",
                  "ENSG00000136628",
                  "EPRS1",
                  "protein_coding",
                  "GCGCGCTCAGTGAGAGGAAA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "27",
                  "adamson_2016_upr_perturb_seq",
                  "28",
                  "OST4_pDS353",
                  "chr2:27070472-27071654;-1",
                  "2",
                  "2",
                  "1",
                  "ENSG00000228474",
                  "OST4",
                  "protein_coding",
                  "GGCTTGTTCGCTGGTGGCGT",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "28",
                  "adamson_2016_upr_perturb_seq",
                  "29",
                  "SOCS1_pDS479",
                  "chr16:11254417-11256204;-1",
                  "16",
                  "16",
                  "1",
                  "ENSG00000185338",
                  "SOCS1",
                  "protein_coding",
                  "GCGGCGGACCCGGCCCTAGG",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "29",
                  "adamson_2016_upr_perturb_seq",
                  "30",
                  "63(mod)_pBA580",
                  null,
                  null,
                  "0",
                  "1",
                  null,
                  null,
                  null,
                  "nan",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "30",
                  "adamson_2016_upr_perturb_seq",
                  "31",
                  "63(mod)_pBA580",
                  null,
                  null,
                  "0",
                  "1",
                  null,
                  null,
                  null,
                  "nan",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "31",
                  "adamson_2016_upr_perturb_seq",
                  "32",
                  "UFL1_pDS410",
                  "chr6:96521595-96555276;1",
                  "6",
                  "6",
                  "1",
                  "ENSG00000014123",
                  "UFL1",
                  "protein_coding",
                  "GTGACTCGCAGTAGACGCGG",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "32",
                  "adamson_2016_upr_perturb_seq",
                  "33",
                  "DDRGK1_pDS041",
                  "chr20:3190350-3204685;-1",
                  "20",
                  "20",
                  "1",
                  "ENSG00000198171",
                  "DDRGK1",
                  "protein_coding",
                  "GCGGTCCACAAAGGCTCAGA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "33",
                  "adamson_2016_upr_perturb_seq",
                  "34",
                  "QARS_pDS510",
                  "chr1:219968600-220046530;-1",
                  "1",
                  "1",
                  "1",
                  "ENSG00000136628",
                  "EPRS1",
                  "protein_coding",
                  "GCGCGCTCAGTGAGAGGAAA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "34",
                  "adamson_2016_upr_perturb_seq",
                  "35",
                  "63(mod)_pBA580",
                  null,
                  null,
                  "0",
                  "1",
                  null,
                  null,
                  null,
                  "nan",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "35",
                  "adamson_2016_upr_perturb_seq",
                  "36",
                  "Gal4-4(mod)_pBA582",
                  null,
                  null,
                  "0",
                  "1",
                  null,
                  null,
                  null,
                  "nan",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "36",
                  "adamson_2016_upr_perturb_seq",
                  "37",
                  "STT3A_pDS011",
                  "chr11:125591712-125625215;1",
                  "11",
                  "11",
                  "1",
                  "ENSG00000134910",
                  "STT3A",
                  "protein_coding",
                  "GGGAGCCCCGCGGATCGTTT",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "37",
                  "adamson_2016_upr_perturb_seq",
                  "38",
                  "SAMM50_pDS156",
                  "chr22:43955442-44010531;1",
                  "22",
                  "22",
                  "1",
                  "ENSG00000100347",
                  "SAMM50",
                  "protein_coding",
                  "GCTGCCTCGCAGAGCTGCCC",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "38",
                  "adamson_2016_upr_perturb_seq",
                  "39",
                  "SEC61B_pDS162",
                  "chr9:99222064-99230615;1",
                  "9",
                  "9",
                  "1",
                  "ENSG00000106803",
                  "SEC61B",
                  "protein_coding",
                  "GGCGGACGGATAGAAAGTTA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "39",
                  "adamson_2016_upr_perturb_seq",
                  "40",
                  "IER3IP1_pDS002",
                  "chr18:47152834-47176364;-1",
                  "18",
                  "18",
                  "1",
                  "ENSG00000134049",
                  "IER3IP1",
                  "protein_coding",
                  "GGGCCGCGTGAGGGGACACCA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "40",
                  "adamson_2016_upr_perturb_seq",
                  "41",
                  "IARS2_pDS090",
                  "chr1:220094132-220148041;1",
                  "1",
                  "1",
                  "1",
                  "ENSG00000067704",
                  "IARS2",
                  "protein_coding",
                  "GGCGCGTGCGCCCTCTTACT",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "41",
                  "adamson_2016_upr_perturb_seq",
                  "42",
                  "63(mod)_pBA580",
                  null,
                  null,
                  "0",
                  "1",
                  null,
                  null,
                  null,
                  "nan",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "42",
                  "adamson_2016_upr_perturb_seq",
                  "43",
                  "ARHGAP22_pDS458",
                  "chr10:48446036-48656265;-1",
                  "10",
                  "10",
                  "1",
                  "ENSG00000128805",
                  "ARHGAP22",
                  "protein_coding",
                  "GGTCCGTCCGGAGCCAGGAG",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "43",
                  "adamson_2016_upr_perturb_seq",
                  "44",
                  "QARS_pDS510",
                  "chr1:219968600-220046530;-1",
                  "1",
                  "1",
                  "1",
                  "ENSG00000136628",
                  "EPRS1",
                  "protein_coding",
                  "GCGCGCTCAGTGAGAGGAAA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "44",
                  "adamson_2016_upr_perturb_seq",
                  "45",
                  "SPCS2_pDS401",
                  "chr11:74949261-74979033;1",
                  "11",
                  "11",
                  "1",
                  "ENSG00000118363",
                  "SPCS2",
                  "protein_coding",
                  "GTCAGACAAGTCGGAGGGGA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "45",
                  "adamson_2016_upr_perturb_seq",
                  "46",
                  "63(mod)_pBA580",
                  null,
                  null,
                  "0",
                  "1",
                  null,
                  null,
                  null,
                  "nan",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "46",
                  "adamson_2016_upr_perturb_seq",
                  "47",
                  "MRGBP_pDS124",
                  "chr20:62796473-62801729;1",
                  "20",
                  "20",
                  "1",
                  "ENSG00000101189",
                  "MRGBP",
                  "protein_coding",
                  "GCGAGCAAGGAGCCCCCGGC",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "47",
                  "adamson_2016_upr_perturb_seq",
                  "48",
                  "YIPF5_pDS186",
                  "chr5:144158162-144170714;-1",
                  "5",
                  "5",
                  "1",
                  "ENSG00000145817",
                  "YIPF5",
                  "protein_coding",
                  "GGGTGCAGGGGACCGCGTCA",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "48",
                  "adamson_2016_upr_perturb_seq",
                  "49",
                  "SEC61B_pDS033",
                  "chr9:99222064-99230615;1",
                  "9",
                  "9",
                  "1",
                  "ENSG00000106803",
                  "SEC61B",
                  "protein_coding",
                  "GTACGGAGCCCCCGAAAGAC",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ],
                [
                  "49",
                  "adamson_2016_upr_perturb_seq",
                  "50",
                  "SEL1L_pDS373",
                  "chr14:81471547-81533853;-1",
                  "14",
                  "14",
                  "1",
                  "ENSG00000071537",
                  "SEL1L",
                  "protein_coding",
                  "GGTGGTGGCTGAGTCCGTGG",
                  "CRISPRi",
                  null,
                  "P0DT0H0M0S",
                  null,
                  null,
                  "cell line",
                  null,
                  "Homo sapiens",
                  "blood",
                  "UBERON:0000178",
                  "lymphoblast",
                  "CL:0017005",
                  "K 562 cell",
                  "CLO:0007050",
                  "female",
                  null,
                  "adult",
                  null,
                  "chronic myelogenous leukemia, BCR-ABL1 positive",
                  "MONDO:0011996",
                  "A Multiplexed Single-Cell CRISPR Screening Platform Enables Systematic Dissection of the Unfolded Protein Response",
                  "https://doi.org/10.1016/j.cell.2016.11.048",
                  "2016",
                  "Britt Adamson",
                  "Jonathan Weissman",
                  "63000 chronic myeloid leukemia (K562) cells transfected with a UPR sensor gene-targeting gRNAs.",
                  "Perturb-seq was applied to a small CRISPRi library of 91 sgRNAs targeting UPR genes in K562 cells.",
                  "91",
                  "62623",
                  "EFO:0022868",
                  "endogenous",
                  "EFO:0022895",
                  "dCas9-KRAB",
                  null,
                  "retroviral transduction",
                  null,
                  "lentiviral transduction",
                  null,
                  "random locus integration",
                  null,
                  "random locus integration",
                  null,
                  "constitutive expression",
                  null,
                  "constitutive expression",
                  "custom",
                  null,
                  null,
                  "pooled",
                  null,
                  "focused",
                  null,
                  "inhibition",
                  "Weissman",
                  "3",
                  "1",
                  "91",
                  null,
                  null,
                  "high-dimensional assay",
                  null,
                  "transcriptomic",
                  null,
                  "single-cell rna-seq",
                  null,
                  "Perturb-seq",
                  null,
                  null,
                  "10x Genomics Single Cell 3-prime",
                  null,
                  "Illumina HiSeq 4000",
                  null,
                  "barcode sequencing",
                  null,
                  "CellRanger",
                  null,
                  "MAGeCK",
                  null,
                  "GRCh37",
                  "[{\"dataset_accession\": \"GSM2406677\", \"dataset_uri\": \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2406681\", \"dataset_description\": \"Barcode, cell identities, raw gene expression matrix\", \"dataset_file_name\": \"GSM2406677_10X010\"}, {\"dataset_accession\": \"GSM2406677_10X005\", \"dataset_uri\": \"https://zenodo.org/records/7041849/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad\", \"dataset_description\": \"Processed .h5ad file\", \"dataset_file_name\": \"AdamsonWeissman2016_GSM2406681_10X010.h5ad\"}]"
                ]
              ],
              "shape": {
                "columns": 91,
                "rows": 62623
              }
            },
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>sample_id</th>\n",
              "      <th>perturbation_name</th>\n",
              "      <th>perturbed_target_coord</th>\n",
              "      <th>perturbed_target_chromosome</th>\n",
              "      <th>perturbed_target_chromosome_encoding</th>\n",
              "      <th>perturbed_target_number</th>\n",
              "      <th>perturbed_target_ensg</th>\n",
              "      <th>perturbed_target_symbol</th>\n",
              "      <th>perturbed_target_biotype</th>\n",
              "      <th>...</th>\n",
              "      <th>sequencing_platform_label</th>\n",
              "      <th>sequencing_strategy_id</th>\n",
              "      <th>sequencing_strategy_label</th>\n",
              "      <th>software_counts_id</th>\n",
              "      <th>software_counts_label</th>\n",
              "      <th>software_analysis_id</th>\n",
              "      <th>software_analysis_label</th>\n",
              "      <th>reference_genome_id</th>\n",
              "      <th>reference_genome_label</th>\n",
              "      <th>associated_datasets</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>1</td>\n",
              "      <td>63(mod)_pBA580</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>2</td>\n",
              "      <td>OST4_pDS353</td>\n",
              "      <td>chr2:27070472-27071654;-1</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "      <td>ENSG00000228474</td>\n",
              "      <td>OST4</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>3</td>\n",
              "      <td>SEC61A1_pDS031</td>\n",
              "      <td>chr3:128051641-128071705;1</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>1</td>\n",
              "      <td>ENSG00000058262</td>\n",
              "      <td>SEC61A1</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>4</td>\n",
              "      <td>EIF2B4_pDS491</td>\n",
              "      <td>chr2:27364352-27370338;-1</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "      <td>ENSG00000115211</td>\n",
              "      <td>EIF2B4</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>5</td>\n",
              "      <td>SRPR_pDS482</td>\n",
              "      <td>chr11:126262938-126269144;-1</td>\n",
              "      <td>11</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>ENSG00000182934</td>\n",
              "      <td>SRPRA</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>62618</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>62619</td>\n",
              "      <td>STT3A_pDS011</td>\n",
              "      <td>chr11:125591712-125625215;1</td>\n",
              "      <td>11</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>ENSG00000134910</td>\n",
              "      <td>STT3A</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>62619</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>62620</td>\n",
              "      <td>ARHGAP22_pDS458</td>\n",
              "      <td>chr10:48446036-48656265;-1</td>\n",
              "      <td>10</td>\n",
              "      <td>10</td>\n",
              "      <td>1</td>\n",
              "      <td>ENSG00000128805</td>\n",
              "      <td>ARHGAP22</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>62620</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>62621</td>\n",
              "      <td>63(mod)_pBA580</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>62621</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>62622</td>\n",
              "      <td>KCTD16_pDS096</td>\n",
              "      <td>chr5:144170873-144485686;1</td>\n",
              "      <td>5</td>\n",
              "      <td>5</td>\n",
              "      <td>1</td>\n",
              "      <td>ENSG00000183775</td>\n",
              "      <td>KCTD16</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>62622</th>\n",
              "      <td>adamson_2016_upr_perturb_seq</td>\n",
              "      <td>62623</td>\n",
              "      <td>SAMM50_pDS156</td>\n",
              "      <td>chr22:43955442-44010531;1</td>\n",
              "      <td>22</td>\n",
              "      <td>22</td>\n",
              "      <td>1</td>\n",
              "      <td>ENSG00000100347</td>\n",
              "      <td>SAMM50</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>...</td>\n",
              "      <td>Illumina HiSeq 4000</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>barcode sequencing</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>CellRanger</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>MAGeCK</td>\n",
              "      <td>&lt;NA&gt;</td>\n",
              "      <td>GRCh37</td>\n",
              "      <td>[{\"dataset_accession\": \"GSM2406677\", \"dataset_...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>62623 rows × 91 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "                         dataset_id  sample_id perturbation_name  \\\n",
              "0      adamson_2016_upr_perturb_seq          1    63(mod)_pBA580   \n",
              "1      adamson_2016_upr_perturb_seq          2       OST4_pDS353   \n",
              "2      adamson_2016_upr_perturb_seq          3    SEC61A1_pDS031   \n",
              "3      adamson_2016_upr_perturb_seq          4     EIF2B4_pDS491   \n",
              "4      adamson_2016_upr_perturb_seq          5       SRPR_pDS482   \n",
              "...                             ...        ...               ...   \n",
              "62618  adamson_2016_upr_perturb_seq      62619      STT3A_pDS011   \n",
              "62619  adamson_2016_upr_perturb_seq      62620   ARHGAP22_pDS458   \n",
              "62620  adamson_2016_upr_perturb_seq      62621    63(mod)_pBA580   \n",
              "62621  adamson_2016_upr_perturb_seq      62622     KCTD16_pDS096   \n",
              "62622  adamson_2016_upr_perturb_seq      62623     SAMM50_pDS156   \n",
              "\n",
              "             perturbed_target_coord perturbed_target_chromosome  \\\n",
              "0                              <NA>                        <NA>   \n",
              "1         chr2:27070472-27071654;-1                           2   \n",
              "2        chr3:128051641-128071705;1                           3   \n",
              "3         chr2:27364352-27370338;-1                           2   \n",
              "4      chr11:126262938-126269144;-1                          11   \n",
              "...                             ...                         ...   \n",
              "62618   chr11:125591712-125625215;1                          11   \n",
              "62619    chr10:48446036-48656265;-1                          10   \n",
              "62620                          <NA>                        <NA>   \n",
              "62621    chr5:144170873-144485686;1                           5   \n",
              "62622     chr22:43955442-44010531;1                          22   \n",
              "\n",
              "       perturbed_target_chromosome_encoding  perturbed_target_number  \\\n",
              "0                                         0                        1   \n",
              "1                                         2                        1   \n",
              "2                                         3                        1   \n",
              "3                                         2                        1   \n",
              "4                                        11                        1   \n",
              "...                                     ...                      ...   \n",
              "62618                                    11                        1   \n",
              "62619                                    10                        1   \n",
              "62620                                     0                        1   \n",
              "62621                                     5                        1   \n",
              "62622                                    22                        1   \n",
              "\n",
              "      perturbed_target_ensg perturbed_target_symbol perturbed_target_biotype  \\\n",
              "0                      <NA>                    <NA>                     <NA>   \n",
              "1           ENSG00000228474                    OST4           protein_coding   \n",
              "2           ENSG00000058262                 SEC61A1           protein_coding   \n",
              "3           ENSG00000115211                  EIF2B4           protein_coding   \n",
              "4           ENSG00000182934                   SRPRA           protein_coding   \n",
              "...                     ...                     ...                      ...   \n",
              "62618       ENSG00000134910                   STT3A           protein_coding   \n",
              "62619       ENSG00000128805                ARHGAP22           protein_coding   \n",
              "62620                  <NA>                    <NA>                     <NA>   \n",
              "62621       ENSG00000183775                  KCTD16           protein_coding   \n",
              "62622       ENSG00000100347                  SAMM50           protein_coding   \n",
              "\n",
              "       ... sequencing_platform_label sequencing_strategy_id  \\\n",
              "0      ...       Illumina HiSeq 4000                   <NA>   \n",
              "1      ...       Illumina HiSeq 4000                   <NA>   \n",
              "2      ...       Illumina HiSeq 4000                   <NA>   \n",
              "3      ...       Illumina HiSeq 4000                   <NA>   \n",
              "4      ...       Illumina HiSeq 4000                   <NA>   \n",
              "...    ...                       ...                    ...   \n",
              "62618  ...       Illumina HiSeq 4000                   <NA>   \n",
              "62619  ...       Illumina HiSeq 4000                   <NA>   \n",
              "62620  ...       Illumina HiSeq 4000                   <NA>   \n",
              "62621  ...       Illumina HiSeq 4000                   <NA>   \n",
              "62622  ...       Illumina HiSeq 4000                   <NA>   \n",
              "\n",
              "      sequencing_strategy_label software_counts_id software_counts_label  \\\n",
              "0            barcode sequencing               <NA>            CellRanger   \n",
              "1            barcode sequencing               <NA>            CellRanger   \n",
              "2            barcode sequencing               <NA>            CellRanger   \n",
              "3            barcode sequencing               <NA>            CellRanger   \n",
              "4            barcode sequencing               <NA>            CellRanger   \n",
              "...                         ...                ...                   ...   \n",
              "62618        barcode sequencing               <NA>            CellRanger   \n",
              "62619        barcode sequencing               <NA>            CellRanger   \n",
              "62620        barcode sequencing               <NA>            CellRanger   \n",
              "62621        barcode sequencing               <NA>            CellRanger   \n",
              "62622        barcode sequencing               <NA>            CellRanger   \n",
              "\n",
              "      software_analysis_id software_analysis_label reference_genome_id  \\\n",
              "0                     <NA>                  MAGeCK                <NA>   \n",
              "1                     <NA>                  MAGeCK                <NA>   \n",
              "2                     <NA>                  MAGeCK                <NA>   \n",
              "3                     <NA>                  MAGeCK                <NA>   \n",
              "4                     <NA>                  MAGeCK                <NA>   \n",
              "...                    ...                     ...                 ...   \n",
              "62618                 <NA>                  MAGeCK                <NA>   \n",
              "62619                 <NA>                  MAGeCK                <NA>   \n",
              "62620                 <NA>                  MAGeCK                <NA>   \n",
              "62621                 <NA>                  MAGeCK                <NA>   \n",
              "62622                 <NA>                  MAGeCK                <NA>   \n",
              "\n",
              "      reference_genome_label  \\\n",
              "0                     GRCh37   \n",
              "1                     GRCh37   \n",
              "2                     GRCh37   \n",
              "3                     GRCh37   \n",
              "4                     GRCh37   \n",
              "...                      ...   \n",
              "62618                 GRCh37   \n",
              "62619                 GRCh37   \n",
              "62620                 GRCh37   \n",
              "62621                 GRCh37   \n",
              "62622                 GRCh37   \n",
              "\n",
              "                                     associated_datasets  \n",
              "0      [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "1      [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "2      [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "3      [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "4      [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "...                                                  ...  \n",
              "62618  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "62619  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "62620  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "62621  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "62622  [{\"dataset_accession\": \"GSM2406677\", \"dataset_...  \n",
              "\n",
              "[62623 rows x 91 columns]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "cur_data.validate_data(slot='obs')"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "723924a2",
      "metadata": {
        "id": "723924a2"
      },
      "source": [
        "# VAR slot curation"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "bff48ef8",
      "metadata": {
        "id": "bff48ef8"
      },
      "source": [
        "### Standardise genes"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 36,
      "id": "3a410193",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3a410193",
        "outputId": "16ad8630-8096-4804-9e46-da19ece15294"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Variable data:\n",
            "DataFrame shape: (38606, 1)\n",
            "--------------------------------------------------\n",
            "                     gene_symbol\n",
            "DDX11L2                  DDX11L2\n",
            "MIR1302-2HG          MIR1302-2HG\n",
            "FAM138A                  FAM138A\n",
            "ENSG00000290826  ENSG00000290826\n",
            "OR4F5                      OR4F5\n",
            "...                          ...\n",
            "ENSG00000277836  ENSG00000277836\n",
            "ENSG00000278633  ENSG00000278633\n",
            "ENSG00000276017  ENSG00000276017\n",
            "ENSG00000278817  ENSG00000278817\n",
            "ENSG00000277196  ENSG00000277196\n",
            "\n",
            "[38606 rows x 1 columns]\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.show_var()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 37,
      "id": "87280a5e",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "87280a5e",
        "outputId": "4cbc2324-7fa2-4e58-b6c1-abe875d418d0"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Column genes added to adata.var\n"
          ]
        }
      ],
      "source": [
        "cur_data.create_columns(slot='var', col_dict={\n",
        "    'genes':cur_data.adata.var.index\n",
        "})"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 38,
      "id": "43821329",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "43821329",
        "outputId": "f1e02634-a77b-4247-effa-aacc9c2873cc"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/content/PerturbationCatalogue/data_exploration/curation_tools/curation_tools.py:1271: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  map_df[\"synonyms\"] = map_df[\"synonyms\"].str.split(\"|\")\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Mapped potential synonyms in genes of the provided dataframe to gene symbols\n",
            "Converted 36993/38087 gene symbols/ENSG IDs to standardized gene symbols/ENSG IDs\n",
            "--------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "cur_data.standardize_genes(\n",
        "    slot=\"var\", input_column=\"genes\", input_column_type=\"gene_symbol\"\n",
        ")"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "38bca8db",
      "metadata": {
        "id": "38bca8db"
      },
      "source": [
        "### Validate var metadata"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 39,
      "id": "3fdf4305",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 490
        },
        "id": "3fdf4305",
        "outputId": "226ac300-9300-46bb-f00a-60d8e3154986"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "adata.var is valid according to the var_schema.\n",
            "Validated data:\n"
          ]
        },
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"cur_data\",\n  \"rows\": 38606,\n  \"fields\": [\n    {\n      \"column\": \"index\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 38606,\n        \"samples\": [\n          \"GARIN5A\",\n          \"ENSG00000286348\",\n          \"ENSG00000260986\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ensembl_gene_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 36833,\n        \"samples\": [\n          \"ENSG00000010165\",\n          \"ENSG00000117533\",\n          \"ENSG00000187243\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"gene_symbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 25027,\n        \"samples\": [\n          \"MYO1D-DT\",\n          \"UCP2\",\n          \"ZC3H7A\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-c60ae961-640c-47d8-9428-40e99be44050\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>ensembl_gene_id</th>\n",
              "      <th>gene_symbol</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>index</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>DDX11L2</th>\n",
              "      <td>ENSG00000236397</td>\n",
              "      <td>DDX11L2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MIR1302-2HG</th>\n",
              "      <td>ENSG00000243485</td>\n",
              "      <td>MIR1302-2HG</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>FAM138A</th>\n",
              "      <td>ENSG00000237613</td>\n",
              "      <td>FAM138A</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000290826</th>\n",
              "      <td>ENSG00000290826</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>OR4F5</th>\n",
              "      <td>ENSG00000186092</td>\n",
              "      <td>OR4F5</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000277836</th>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000278633</th>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000276017</th>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000278817</th>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000277196</th>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>38606 rows × 2 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-c60ae961-640c-47d8-9428-40e99be44050')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-c60ae961-640c-47d8-9428-40e99be44050 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-c60ae961-640c-47d8-9428-40e99be44050');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-d974831d-6cbd-40aa-95f4-c1babd28df92\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-d974831d-6cbd-40aa-95f4-c1babd28df92')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-d974831d-6cbd-40aa-95f4-c1babd28df92 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "                 ensembl_gene_id  gene_symbol\n",
              "index                                        \n",
              "DDX11L2          ENSG00000236397      DDX11L2\n",
              "MIR1302-2HG      ENSG00000243485  MIR1302-2HG\n",
              "FAM138A          ENSG00000237613      FAM138A\n",
              "ENSG00000290826  ENSG00000290826         None\n",
              "OR4F5            ENSG00000186092        OR4F5\n",
              "...                          ...          ...\n",
              "ENSG00000277836             None         None\n",
              "ENSG00000278633             None         None\n",
              "ENSG00000276017             None         None\n",
              "ENSG00000278817             None         None\n",
              "ENSG00000277196             None         None\n",
              "\n",
              "[38606 rows x 2 columns]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "cur_data.validate_data(slot='var')"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "12cda67c",
      "metadata": {},
      "source": [
        "# Save the dataset"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "33f9ec8a",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Curated data saved to ../curated/h5ad/adamson_2016_upr_perturb_seq_curated.h5ad\n"
          ]
        }
      ],
      "source": [
        "cur_data.save_curated_data_h5ad()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "952ed212",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Starting the conversion of adata to a long format DataFrame...\n",
            "Starting the conversion to long format...\n",
            "Processing 32738 genes in 164 chunks of size 200...\n",
            "Created ParquetWriter and wrote chunk 1/164\n",
            "Appended chunk 2/164 to parquet file\n",
            "Appended chunk 3/164 to parquet file\n",
            "Appended chunk 4/164 to parquet file\n",
            "Appended chunk 5/164 to parquet file\n",
            "Appended chunk 6/164 to parquet file\n",
            "Appended chunk 7/164 to parquet file\n",
            "Appended chunk 8/164 to parquet file\n",
            "Appended chunk 9/164 to parquet file\n",
            "Appended chunk 10/164 to parquet file\n",
            "Appended chunk 11/164 to parquet file\n",
            "Appended chunk 12/164 to parquet file\n",
            "Appended chunk 13/164 to parquet file\n",
            "Appended chunk 14/164 to parquet file\n",
            "Appended chunk 15/164 to parquet file\n",
            "Appended chunk 16/164 to parquet file\n",
            "Appended chunk 17/164 to parquet file\n",
            "Appended chunk 18/164 to parquet file\n",
            "Appended chunk 19/164 to parquet file\n",
            "Appended chunk 20/164 to parquet file\n",
            "Appended chunk 21/164 to parquet file\n",
            "Appended chunk 22/164 to parquet file\n",
            "Appended chunk 23/164 to parquet file\n",
            "Appended chunk 24/164 to parquet file\n",
            "Appended chunk 25/164 to parquet file\n",
            "Appended chunk 26/164 to parquet file\n",
            "Appended chunk 27/164 to parquet file\n",
            "Appended chunk 28/164 to parquet file\n",
            "Appended chunk 29/164 to parquet file\n",
            "Appended chunk 30/164 to parquet file\n",
            "Appended chunk 31/164 to parquet file\n",
            "Appended chunk 32/164 to parquet file\n",
            "Appended chunk 33/164 to parquet file\n",
            "Appended chunk 34/164 to parquet file\n",
            "Appended chunk 35/164 to parquet file\n",
            "Appended chunk 36/164 to parquet file\n",
            "Appended chunk 37/164 to parquet file\n",
            "Appended chunk 38/164 to parquet file\n",
            "Appended chunk 39/164 to parquet file\n",
            "Appended chunk 40/164 to parquet file\n",
            "Appended chunk 41/164 to parquet file\n",
            "Appended chunk 42/164 to parquet file\n",
            "Appended chunk 43/164 to parquet file\n",
            "Appended chunk 44/164 to parquet file\n",
            "Appended chunk 45/164 to parquet file\n",
            "Appended chunk 46/164 to parquet file\n",
            "Appended chunk 47/164 to parquet file\n",
            "Appended chunk 48/164 to parquet file\n",
            "Appended chunk 49/164 to parquet file\n",
            "Appended chunk 50/164 to parquet file\n",
            "Appended chunk 51/164 to parquet file\n",
            "Appended chunk 52/164 to parquet file\n",
            "Appended chunk 53/164 to parquet file\n",
            "Appended chunk 54/164 to parquet file\n",
            "Appended chunk 55/164 to parquet file\n",
            "Appended chunk 56/164 to parquet file\n",
            "Appended chunk 57/164 to parquet file\n",
            "Appended chunk 58/164 to parquet file\n",
            "Appended chunk 59/164 to parquet file\n",
            "Appended chunk 60/164 to parquet file\n",
            "Appended chunk 61/164 to parquet file\n",
            "Appended chunk 62/164 to parquet file\n",
            "Appended chunk 63/164 to parquet file\n",
            "Appended chunk 64/164 to parquet file\n",
            "Appended chunk 65/164 to parquet file\n",
            "Appended chunk 66/164 to parquet file\n",
            "Appended chunk 67/164 to parquet file\n",
            "Appended chunk 68/164 to parquet file\n",
            "Appended chunk 69/164 to parquet file\n",
            "Appended chunk 70/164 to parquet file\n",
            "Appended chunk 71/164 to parquet file\n",
            "Appended chunk 72/164 to parquet file\n",
            "Appended chunk 73/164 to parquet file\n",
            "Appended chunk 74/164 to parquet file\n",
            "Appended chunk 75/164 to parquet file\n",
            "Appended chunk 76/164 to parquet file\n",
            "Appended chunk 77/164 to parquet file\n",
            "Appended chunk 78/164 to parquet file\n",
            "Appended chunk 79/164 to parquet file\n",
            "Appended chunk 80/164 to parquet file\n",
            "Appended chunk 81/164 to parquet file\n",
            "Appended chunk 82/164 to parquet file\n",
            "Appended chunk 83/164 to parquet file\n",
            "Appended chunk 84/164 to parquet file\n",
            "Appended chunk 85/164 to parquet file\n",
            "Appended chunk 86/164 to parquet file\n",
            "Appended chunk 87/164 to parquet file\n",
            "Appended chunk 88/164 to parquet file\n",
            "Appended chunk 89/164 to parquet file\n",
            "Appended chunk 90/164 to parquet file\n",
            "Appended chunk 91/164 to parquet file\n",
            "Appended chunk 92/164 to parquet file\n",
            "Appended chunk 93/164 to parquet file\n",
            "Appended chunk 94/164 to parquet file\n",
            "Appended chunk 95/164 to parquet file\n",
            "Appended chunk 96/164 to parquet file\n",
            "Appended chunk 97/164 to parquet file\n",
            "Appended chunk 98/164 to parquet file\n",
            "Appended chunk 99/164 to parquet file\n",
            "Appended chunk 100/164 to parquet file\n",
            "Appended chunk 101/164 to parquet file\n",
            "Appended chunk 102/164 to parquet file\n",
            "Appended chunk 103/164 to parquet file\n",
            "Appended chunk 104/164 to parquet file\n",
            "Appended chunk 105/164 to parquet file\n",
            "Appended chunk 106/164 to parquet file\n",
            "Appended chunk 107/164 to parquet file\n",
            "Appended chunk 108/164 to parquet file\n",
            "Appended chunk 109/164 to parquet file\n",
            "Appended chunk 110/164 to parquet file\n",
            "Appended chunk 111/164 to parquet file\n",
            "Appended chunk 112/164 to parquet file\n",
            "Appended chunk 113/164 to parquet file\n",
            "Appended chunk 114/164 to parquet file\n",
            "Appended chunk 115/164 to parquet file\n",
            "Appended chunk 116/164 to parquet file\n",
            "Appended chunk 117/164 to parquet file\n",
            "Appended chunk 118/164 to parquet file\n",
            "Appended chunk 119/164 to parquet file\n",
            "Appended chunk 120/164 to parquet file\n",
            "Appended chunk 121/164 to parquet file\n",
            "Appended chunk 122/164 to parquet file\n",
            "Appended chunk 123/164 to parquet file\n",
            "Appended chunk 124/164 to parquet file\n",
            "Appended chunk 125/164 to parquet file\n",
            "Appended chunk 126/164 to parquet file\n",
            "Appended chunk 127/164 to parquet file\n",
            "Appended chunk 128/164 to parquet file\n",
            "Appended chunk 129/164 to parquet file\n",
            "Appended chunk 130/164 to parquet file\n",
            "Appended chunk 131/164 to parquet file\n",
            "Appended chunk 132/164 to parquet file\n",
            "Appended chunk 133/164 to parquet file\n",
            "Appended chunk 134/164 to parquet file\n",
            "Appended chunk 135/164 to parquet file\n",
            "Appended chunk 136/164 to parquet file\n",
            "Appended chunk 137/164 to parquet file\n",
            "Appended chunk 138/164 to parquet file\n",
            "Appended chunk 139/164 to parquet file\n",
            "Appended chunk 140/164 to parquet file\n",
            "Appended chunk 141/164 to parquet file\n",
            "Appended chunk 142/164 to parquet file\n",
            "Appended chunk 143/164 to parquet file\n",
            "Appended chunk 144/164 to parquet file\n",
            "Appended chunk 145/164 to parquet file\n",
            "Appended chunk 146/164 to parquet file\n",
            "Appended chunk 147/164 to parquet file\n",
            "Appended chunk 148/164 to parquet file\n",
            "Appended chunk 149/164 to parquet file\n",
            "Appended chunk 150/164 to parquet file\n",
            "Appended chunk 151/164 to parquet file\n",
            "Appended chunk 152/164 to parquet file\n",
            "Appended chunk 153/164 to parquet file\n",
            "Appended chunk 154/164 to parquet file\n",
            "Appended chunk 155/164 to parquet file\n",
            "Appended chunk 156/164 to parquet file\n",
            "Appended chunk 157/164 to parquet file\n",
            "Appended chunk 158/164 to parquet file\n",
            "Appended chunk 159/164 to parquet file\n",
            "Appended chunk 160/164 to parquet file\n",
            "Appended chunk 161/164 to parquet file\n",
            "Appended chunk 162/164 to parquet file\n",
            "Appended chunk 163/164 to parquet file\n",
            "Appended chunk 164/164 to parquet file\n",
            "All chunks written and ParquetWriter closed.\n",
            "Metadata written to ../curated/parquet/adamson_2016_upr_perturb_seq_curated_long_metadata.parquet\n"
          ]
        }
      ],
      "source": [
        "cur_data.save_curated_data_parquet(split_metadata=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "0071652b",
      "metadata": {},
      "outputs": [],
      "source": [
        "cur_data.save_curated_data_parquet(split_metadata=False)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "1297c44f",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;33mWARNING:\u001b[0m Parallel composite upload was turned ON to get the best performance on\n",
            "uploading large objects. If you would like to opt-out and instead\n",
            "perform a normal upload, run:\n",
            "`gcloud config set storage/parallel_composite_upload_enabled False`\n",
            "If you would like to disable this warning, run:\n",
            "`gcloud config set storage/parallel_composite_upload_enabled True`\n",
            "Note that with parallel composite uploads, your object might be\n",
            "uploaded as a composite object\n",
            "(https://cloud.google.com/storage/docs/composite-objects), which means\n",
            "that any user who downloads your object will need to use crc32c\n",
            "checksums to verify data integrity. gcloud storage is capable of\n",
            "computing crc32c checksums, but this might pose a problem for other\n",
            "clients.\n",
            "\n",
            "Copying file://Perturbseq/curated/h5ad/orion_2025_hct116_curated.h5ad to gs://perturbation-catalogue-lake/perturbseq/curated/orion_2025_hct116_curated.h5ad\n",
            "\n",
            "Average throughput: 420.9MiB/s\n"
          ]
        }
      ],
      "source": [
        "!gcloud storage cp Perturbseq/curated/h5ad/orion_2025_hek293t_curated.h5ad gs://perturbation-catalogue-lake/perturbseq/curated/"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "name": "orion_2025_hek293.ipynb",
      "provenance": []
    },
    "kernelspec": {
      "display_name": ".venv",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.12.8"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}
