{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8f448a7f9c193983",
   "metadata": {},
   "source": [
    "# Colab setup\n",
    "\n",
    "Run the code below to set up Google Cloud SDK, clone the Perturbation Catalogue repository and install required packages."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8f1cfd5bd5d506c",
   "metadata": {},
   "outputs": [],
   "source": [
    "!curl -LsSf https://astral.sh/uv/install.sh | sh\n",
    "!gcloud config set compute/zone \"europe-west2-a\"\n",
    "!gcloud init --project \"prj-ext-dev-pertcat-437314\" --account \"zakirov@ebi.ac.uk\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "761fed235ae8d7ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "!git clone https://github.com/EMBL-EBI-ABC/PerturbationCatalogue.git\n",
    "%cd PerturbationCatalogue/data_exploration/\n",
    "!git pull\n",
    "!git switch aleks/curate_replogle_2022\n",
    "!pwd\n",
    "!uv pip install -r requirements.txt\n",
    "!mkdir -p Perturbseq/non_curated/h5ad/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b6526661",
   "metadata": {},
   "source": [
    "# Import"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0dc0287a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:06.024186Z",
     "start_time": "2025-11-14T15:40:06.016686Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import json\n",
    "\n",
    "from curation_tools.curation_tools import (\n",
    "    CuratedDataset, \n",
    "    ObsSchema, \n",
    "    VarSchema, \n",
    "    Experiment,\n",
    "    download_file,\n",
    "    upload_parquet_to_bq\n",
    ")\n",
    "\n",
    "import logging\n",
    "logging.basicConfig(\n",
    "    level=logging.DEBUG,\n",
    "    format=\"%(asctime)s %(levelname)s %(name)s: %(message)s\",\n",
    "    handlers=[\n",
    "        logging.FileHandler(\"curation.log\"),\n",
    "        logging.StreamHandler(),  # keep console output too\n",
    "    ],\n",
    "    force=True,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46c0f3f5",
   "metadata": {},
   "source": [
    "# Download data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6509db94",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:06.041922Z",
     "start_time": "2025-11-14T15:40:06.039404Z"
    }
   },
   "outputs": [],
   "source": [
    "noncurated_path = \"../non_curated/h5ad/replogle_2022_k562_essential_normalized.h5ad\"\n",
    "download_file(\n",
    "    url=\"https://plus.figshare.com/ndownloader/files/35773075\",\n",
    "    dest_path=noncurated_path\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1454289",
   "metadata": {},
   "source": [
    "# Initialise the dataset object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73531093",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:30.392294Z",
     "start_time": "2025-11-14T15:40:06.055763Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data = CuratedDataset(\n",
    "    obs_schema=ObsSchema,\n",
    "    var_schema=VarSchema,\n",
    "    exp_metadata_schema=Experiment,\n",
    "    noncurated_path=noncurated_path\n",
    ")\n",
    "\n",
    "cur_data.load_data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d7a5020c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:30.789007Z",
     "start_time": "2025-11-14T15:40:30.505235Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.adata.obs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14662b96",
   "metadata": {},
   "source": [
    "# OBS slot curation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4f073d6c",
   "metadata": {},
   "source": [
    "### Show unique perturbations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e829b86a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:31.696590Z",
     "start_time": "2025-11-14T15:40:31.620150Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.show_unique(slot = 'obs', column = 'sgID_AB')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "089d6117",
   "metadata": {},
   "source": [
    "### Rename `sgID_AB` to `perturbation_name`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12cb9a54",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:31.972493Z",
     "start_time": "2025-11-14T15:40:31.949742Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.rename_columns(slot = 'obs', name_dict = {'sgID_AB': 'perturbation_name'})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6bbcd0c9",
   "metadata": {},
   "source": [
    "### Add guide RNA information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "962b389f",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:36.726292Z",
     "start_time": "2025-11-14T15:40:32.004634Z"
    }
   },
   "outputs": [],
   "source": [
    "# download the guide RNA spreadsheet\n",
    "download_file(\n",
    "    url=\"https://ars.els-cdn.com/content/image/1-s2.0-S0092867422005979-mmc1.xlsx\",\n",
    "    dest_path=\"../supplementary/replogle_2022_guide_info.xlsx\"\n",
    ")\n",
    "\n",
    "# read in the guide RNA spreadsheet\n",
    "# guides for the K562 essential day 6 library are in \"TabB_K562_day6_library\"\n",
    "guide_info_df = pd.read_excel(\"../supplementary/replogle_2022_guide_info.xlsx\", sheet_name=\"TabB_K562_day6_library\")\n",
    "\n",
    "# create perturbation_name column in guide_info_df\n",
    "guide_info_df['perturbation_name'] = guide_info_df['sgID_A'] + '|' + guide_info_df['sgID_B']\n",
    "# replace commas with hyphens in perturbation_name\n",
    "guide_info_df['perturbation_name'] = guide_info_df['perturbation_name'].str.replace(',', '-')\n",
    "# check that all perturbation names in cur_data are in guide_info_df\n",
    "print(f\"All perturbation names in cur_data are in guide_info_df: {cur_data.adata.obs['perturbation_name'].isin(guide_info_df['perturbation_name']).all()}\")\n",
    "# create guide_sequence column in guide_info_df\n",
    "guide_info_df['guide_sequence'] = guide_info_df['targeting sequence A'] + '|' + guide_info_df['targeting sequence B']\n",
    "# subset for necessary columns\n",
    "guide_info_df = guide_info_df[['perturbation_name', 'guide_sequence']]\n",
    "# merge cur_data.adata.obs with guide_info_df on perturbation_name\n",
    "cur_data.adata.obs = cur_data.adata.obs.merge(guide_info_df, on='perturbation_name', how='left')\n",
    "# check that there are no missing guide sequences\n",
    "print(f\"Number of missing guide sequences: {cur_data.adata.obs['guide_sequence'].isna().sum()}\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5ded033",
   "metadata": {},
   "source": [
    "### Standardise perturbation targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0eb611d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:45.728050Z",
     "start_time": "2025-11-14T15:40:36.740672Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.standardize_genes(\n",
    "    slot='obs',\n",
    "    input_column='gene',\n",
    "    input_column_type='gene_symbol',\n",
    "    multiple_entries=False\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "022b9348",
   "metadata": {},
   "source": [
    "### Add `perturbed_target_number` column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e909e4b8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:45.849963Z",
     "start_time": "2025-11-14T15:40:45.734760Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.count_entries(\n",
    "    slot='obs',\n",
    "    input_column='perturbed_target_symbol',\n",
    "    count_column_name='perturbed_target_number',\n",
    "    sep='|'\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2fd30f95",
   "metadata": {},
   "source": [
    "### Encode chromosomes as integers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e126cdd6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:45.908929Z",
     "start_time": "2025-11-14T15:40:45.854976Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.chromosome_encoding()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09a729b8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:45.918433Z",
     "start_time": "2025-11-14T15:40:45.913558Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.show_obs(['perturbation_name', 'perturbed_target_chromosome_encoding'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "75ab6b482b9a616e",
   "metadata": {},
   "source": [
    "### Add metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1221c10f938bd198",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:46.120975Z",
     "start_time": "2025-11-14T15:40:45.932784Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.create_columns(\n",
    "    overwrite=True,\n",
    "    slot=\"obs\",\n",
    "    col_dict={\n",
    "        \"dataset_id\": cur_data.dataset_id,\n",
    "        \"sample_id\": range(1, cur_data.adata.obs.shape[0] + 1),\n",
    "        # perturbation type\n",
    "        \"perturbation_type_label\": \"CRISPRi\",\n",
    "        \"perturbation_type_id\": None,\n",
    "        \"data_modality\": \"CRISPR screen\",\n",
    "        \"significant\": None,\n",
    "        \"significance_criteria\": None,\n",
    "        \"score_interpretation\": None,\n",
    "\n",
    "        # treatment\n",
    "        \"treatment_label\": None,\n",
    "        \"treatment_id\": None,\n",
    "        # model system\n",
    "        \"model_system_label\": \"cell_line\",\n",
    "        \"model_system_id\": None,\n",
    "        \"tissue\": \"blood\",\n",
    "        \"cell_line_label\": \"K 562 cell\",\n",
    "        \"cell_type_label\": \"lymphoblast\",\n",
    "        \"disease_label\": \"chronic myelogenous leukemia, BCR-ABL1 positive\",\n",
    "\n",
    "        \"timepoint\": \"P6DT0H0M0S\",\n",
    "        \"species\": \"Homo sapiens\",\n",
    "        \"sex_label\": \"female\",\n",
    "        \"sex_id\": None,\n",
    "        \"developmental_stage_label\": \"adult\",\n",
    "        \"developmental_stage_id\": None,\n",
    "\n",
    "        \"study_title\": \"Mapping information-rich genotype-phenotype landscapes with genome-scale Perturb-seq\",\n",
    "        \"study_uri\": \"https://doi.org/10.1016/j.cell.2022.05.013\",\n",
    "        \"study_year\": 2022,\n",
    "        \"first_author\": \"Joseph M Replogle\",\n",
    "        \"last_author\": \"Jonathan S Weissman\",\n",
    "\n",
    "        \"experiment_title\": \"K562 day 6 essential-scale Perturb-seq experiment\",\n",
    "        \"experiment_summary\": \"\"\"\n",
    "            K562 chronic myeloid leukemia cells were transduced with a sgRNA library targeting a set of 2,057 common essential genes and sampled at day 6 after lentiviral transduction.\n",
    "            Multiplexed CRISPRi library containing two distinct guides targeting the same gene were used. \n",
    "            \"\"\",\n",
    "\n",
    "        \"number_of_perturbed_targets\": len(set(cur_data.adata.obs['perturbed_target_coord'])),\n",
    "        \"number_of_perturbed_samples\": cur_data.adata.obs.shape[0],\n",
    "\n",
    "        \"library_generation_type_id\": \"EFO:0022868\",\n",
    "        \"library_generation_type_label\": \"endogenous\",\n",
    "\n",
    "        \"library_generation_method_id\": \"EFO:0022895\",\n",
    "        \"library_generation_method_label\": \"dCas9-KRAB\",\n",
    "\n",
    "        \"enzyme_delivery_method_id\": None,\n",
    "        \"enzyme_delivery_method_label\": \"lentivirus transduction\",\n",
    "\n",
    "        \"library_delivery_method_id\": None,\n",
    "        \"library_delivery_method_label\": \"lentivirus transduction\",\n",
    "\n",
    "        \"enzyme_integration_state_id\": None,\n",
    "        \"enzyme_integration_state_label\": \"random locus integration\",\n",
    "\n",
    "        \"library_integration_state_id\": None,\n",
    "        \"library_integration_state_label\": \"random locus integration\",\n",
    "\n",
    "        \"enzyme_expression_control_id\": None,\n",
    "        \"enzyme_expression_control_label\": \"constitutive transgene expression\",\n",
    "\n",
    "        \"library_expression_control_id\": None,\n",
    "        \"library_expression_control_label\": \"constitutive transgene expression\",\n",
    "\n",
    "        \"library_name\": \"custom\",\n",
    "        \"library_uri\": None,\n",
    "\n",
    "        \"library_format_id\": None,\n",
    "        \"library_format_label\": \"pooled\",\n",
    "\n",
    "        \"library_scope_id\": None,\n",
    "        \"library_scope_label\": \"focused\",\n",
    "\n",
    "        \"library_perturbation_type_id\": None,\n",
    "        \"library_perturbation_type_label\": \"inhibition\",\n",
    "\n",
    "        \"library_manufacturer\": \"Weissman Lab\",\n",
    "        \"library_lentiviral_generation\": \"3\",\n",
    "        \"library_grnas_per_target\": \"2\",\n",
    "        \"library_total_grnas\": str(cur_data.adata.obs['guide_sequence'].str.split('|').explode().nunique()),\n",
    "        \"library_total_variants\": None,\n",
    "\n",
    "        \"readout_dimensionality_id\": None,\n",
    "        \"readout_dimensionality_label\": \"high-dimensional assay\",\n",
    "\n",
    "        \"readout_type_id\": None,\n",
    "        \"readout_type_label\": \"transcriptomic\",\n",
    "\n",
    "        \"readout_technology_id\": None,\n",
    "        \"readout_technology_label\": \"single-cell rna-seq\",\n",
    "\n",
    "        \"method_name_id\": None,\n",
    "        \"method_name_label\": \"Perturb-seq\",\n",
    "\n",
    "        \"method_uri\": None,\n",
    "\n",
    "        \"sequencing_library_kit_id\": None,\n",
    "        \"sequencing_library_kit_label\": \"10x Genomics Single Cell 3-prime v3\",\n",
    "\n",
    "        \"sequencing_platform_id\": None,\n",
    "        \"sequencing_platform_label\": \"Illumina NovaSeq 6000\",\n",
    "\n",
    "        \"sequencing_strategy_id\": None,\n",
    "        \"sequencing_strategy_label\": \"barcode sequencing\",\n",
    "\n",
    "        \"software_counts_id\": None,\n",
    "        \"software_counts_label\": \"CellRanger\",\n",
    "\n",
    "        \"software_analysis_id\": None,\n",
    "        \"software_analysis_label\": \"custom\",\n",
    "\n",
    "        \"reference_genome_id\": None,\n",
    "        \"reference_genome_label\": \"GRCh38\",\n",
    "\n",
    "        \"associated_datasets\": json.dumps([\n",
    "            {\n",
    "                \"dataset_accession\": \"K562_essential_raw_bulk_01\",\n",
    "                \"dataset_uri\": \"https://plus.figshare.com/ndownloader/files/35773070\",\n",
    "                \"dataset_description\": \"Raw, pseudo-bulk expression data for genes expressed at >0.01 UMI per cell\",\n",
    "                \"dataset_file_name\": \"K562_essential_raw_bulk_01.h5ad\",\n",
    "            },\n",
    "            {\n",
    "                \"dataset_accession\": \"K562_essential_normalized_bulk_01\",\n",
    "                \"dataset_uri\": \"https://plus.figshare.com/ndownloader/files/35780870\",\n",
    "                \"dataset_description\": \"Gemgroup Z-normalized pseudo-bulk expression data for genes expressed at >0.01 UMI per cell\",\n",
    "                \"dataset_file_name\": \"K562_essential_normalized_bulk_01.h5ad\",\n",
    "            },\n",
    "            {\n",
    "                \"dataset_accession\": \"K562_essential_raw_singlecell_01\",\n",
    "                \"dataset_uri\": \"https://plus.figshare.com/ndownloader/files/35773219\",\n",
    "                \"dataset_description\": \"Raw, single-cell expression data for genes expressed at >0.01 UMI per cell\",\n",
    "                \"dataset_file_name\": \"K562_essential_raw_singlecell_01.h5ad\",\n",
    "            },\n",
    "            {\n",
    "                \"dataset_accession\": \"K562_essential_normalized_singlecell_01\",\n",
    "                \"dataset_uri\": \"https://plus.figshare.com/ndownloader/files/35773075\",\n",
    "                \"dataset_description\": \"Gemgroup Z-normalized single-cell expression data for genes expressed at >0.01 UMI per cell\",\n",
    "                \"dataset_file_name\": \"K562_essential_normalized_singlecell_01.h5ad\",\n",
    "            }\n",
    "        ])\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ca7c4f39ca1b224",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:46.239799Z",
     "start_time": "2025-11-14T15:40:46.139694Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.adata.obs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e51497c7385df7d",
   "metadata": {},
   "source": [
    "### Curate tissue information\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c65d77d94fe8a6f",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:47.069592Z",
     "start_time": "2025-11-14T15:40:46.250349Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.standardize_ontology(\n",
    "    input_column='tissue',\n",
    "    column_type='term_name',\n",
    "    ontology_type='tissue',\n",
    "    overwrite=True\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d241da9",
   "metadata": {},
   "source": [
    "### Curate cell type information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92fd4ede",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:47.932346Z",
     "start_time": "2025-11-14T15:40:47.076948Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.standardize_ontology(\n",
    "    input_column='cell_type_label',\n",
    "    column_type='term_name',\n",
    "    ontology_type='cell_type',\n",
    "    overwrite=True\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "924e430e",
   "metadata": {},
   "source": [
    "### Curate cell line information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcaaca46",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:48.982649Z",
     "start_time": "2025-11-14T15:40:47.938617Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.standardize_ontology(\n",
    "    input_column='cell_line_label',\n",
    "    column_type='term_name',\n",
    "    ontology_type='cell_line',\n",
    "    overwrite=True\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5b500dc",
   "metadata": {},
   "source": [
    "### Curate disease information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a0ffcbc",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:50.135203Z",
     "start_time": "2025-11-14T15:40:48.987271Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.standardize_ontology(\n",
    "    input_column='disease_label',\n",
    "    column_type='term_name',\n",
    "    ontology_type='disease',\n",
    "    overwrite=True\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1df7177a",
   "metadata": {},
   "source": [
    "### Match schema column order"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af9eec5c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:50.441479Z",
     "start_time": "2025-11-14T15:40:50.280861Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.match_schema_columns(slot='obs')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a938e3fb",
   "metadata": {},
   "source": [
    "### Validate obs metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51b1468f",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:54.228254Z",
     "start_time": "2025-11-14T15:40:50.446888Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.validate_data(slot='obs', verbose=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80d8f2663a8a04f5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:54.240188Z",
     "start_time": "2025-11-14T15:40:54.233897Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.adata.obs[['library_total_grnas', 'library_total_variants']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8e545cf",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:54.286837Z",
     "start_time": "2025-11-14T15:40:54.279912Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.show_obs(['perturbation_name', 'perturbed_target_symbol', 'perturbed_target_ensg', 'perturbed_target_coord'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e577bf41",
   "metadata": {},
   "source": [
    "# VAR slot curation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae293c41",
   "metadata": {},
   "source": [
    "### Standardise genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24cf30ca5a36ff47",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:54.411707Z",
     "start_time": "2025-11-14T15:40:54.306909Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.gene_ont[cur_data.gene_ont['synonym'].str.contains('AC00455', na=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17c80a62",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:54.420758Z",
     "start_time": "2025-11-14T15:40:54.416696Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.show_var()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61c37ffb",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:54.448095Z",
     "start_time": "2025-11-14T15:40:54.445249Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.create_columns(\n",
    "    slot = 'var', \n",
    "    col_dict={'gene_ensembl_id': cur_data.adata.var.index},\n",
    "    overwrite=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bef90079",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:55.959813Z",
     "start_time": "2025-11-14T15:40:54.476071Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.standardize_genes(\n",
    "    slot=\"var\",\n",
    "    input_column=\"gene_ensembl_id\",\n",
    "    input_column_type=\"ensembl_gene_id\",\n",
    "    remove_version=False,\n",
    "    multiple_entries=False\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3aff3cb0",
   "metadata": {},
   "source": [
    "### Validate var metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd11729e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:40:55.989057Z",
     "start_time": "2025-11-14T15:40:55.966056Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.validate_data(slot='var')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f84f1542",
   "metadata": {},
   "source": [
    "# Save the dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "efb684a5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T14:42:59.338004Z",
     "start_time": "2025-11-14T14:41:47.437763Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.save_curated_data_h5ad()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09c59580",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:42:08.662334Z",
     "start_time": "2025-11-14T15:42:04.434202Z"
    }
   },
   "outputs": [],
   "source": [
    "cur_data.save_curated_data_parquet(split_metadata=True, save_metadata_only=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23e9425074cb7f3c",
   "metadata": {},
   "source": [
    "# Upload to BigQuery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c0bf7b524309c8e3",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-14T15:42:46.535769Z",
     "start_time": "2025-11-14T15:42:17.994834Z"
    }
   },
   "outputs": [],
   "source": [
    "upload_parquet_to_bq(\n",
    "    parquet_path='../curated/parquet/replogle_2022_k562_essential_normalized_curated_metadata.parquet',\n",
    "    bq_dataset_id='prj-ext-dev-pertcat-437314.perturb_seq',\n",
    "    bq_table_name='metadata',\n",
    "    key_columns=['dataset_id', 'sample_id'],\n",
    "    verbose=True\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6957880576a549ac",
   "metadata": {},
   "source": [
    "# Upload to GC Storage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98caae210881787c",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gcloud storage cp ../curated/h5ad/replogle_2022_k562_essential_normalized_curated.h5ad gs://perturbation-catalogue-lake/perturbseq/curated/"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "PerturbationCatalogue",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
